---
title: "Beta diversity Bow River microbiome"
author: "Emilie Diesbourg"
date: "07/08/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Introduction {.tabset}

This data comes from a study done in 2022 across 5 sites on the Bow River, Calgary AB and 3 microcosm streams from Pine Creek wastewater treatment plant with varying municipal wastewater effluent exposures (mesocosm streams: 0, 5, or 15% effluent flow). The objective of this study is to determine whether there are changes to the microbiome of freshwater macroinvertebrates and riparian Spiders with increased exposure to wastewater as well as any differences across taxa or whether the microbiome is retained across metamorphosis. 

## Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
#Change margin size to be smaller so graphs fit in the plot panel
par(mar = c(2, 2, 2, 2)) # Set the margin on all sides to 2
```

## Loading Packages
```{r Load packages}
library(rlang)
library(glue)
library(ggh4x)
library(phyloseq) #phyloseq will be the main package used for structuring microbiome data and diversity comparisons
#if (!require("BiocManager", quietly = TRUE))
#install.packages("BiocManager")
#install.packages("viridis")
library(viridis)

#BiocManager::install("apeglm")
library(apeglm)
library(car)
library(ggplot2) #For creating graphs
#install.packages("ggtext")
library(ggtext)
library(dplyr) #Helps with data wrangling 
library(cluster) #Used for cluster analyses
library(grid)
library(ape)
library(pals)
library(FSA)
library(multcomp)
library(MASS)
library(glmmTMB)
library(gridExtra)
library(vegan)
library(tidyverse) #data wrangling
library(microbiome) #For microbiome analyses
#if (!require("BiocManager", quietly = TRUE)) #Installing microbiome package
#install.packages("BiocManager")
#BiocManager::install("microbiome")
library(PERFect) #Permutation filtration for microbiome data. Used when comparing beta diversities across sites/locations. 
#if(!requireNamespace("BiocManager", quietly = TRUE))
#install.packages("BiocManager")
#BiocManager::install("PERFect")
library(knitr) #For R Markdown knitting
library(kableExtra)
library(speedyseq)
#if (!requireNamespace("remotes", quietly = TRUE))
#install.packages("remotes")
#remotes::install_github("mikemc/speedyseq")
library(colorspace)
library(RColorBrewer)
library(picante)
library(ggpubr)
library(data.table)
library(AICcmodavg)
#devtools::install_github("microsud/microbiomeutilities")
library(microbiomeutilities)
library(imsig)
library(phangorn) #For phylogenetic tree
library(metacoder) 
library(tibble)
#if (!require("BiocManager", quietly = TRUE))
#install.packages("BiocManager")

#BiocManager::install("DESeq2")
library(DESeq2)
library(emmeans)
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis", force=TRUE)
library(pairwiseAdonis)
#library(plyr) apparently if you load this after dpylr it messes up the summarize function?

#if( !require(NbClust) ){install.packages("NbClust"); library(NbClust)}
#if( !require(cobiclust) ){install.packages("cobiclust"); library(cobiclust)}
library(NbClust)
library(cobiclust)
#if (!require("BiocManager", quietly = TRUE))
#install.packages("BiocManager")

#BiocManager::install("mia")
library(mia)
library(factoextra)
#install.packages(
#"microViz",
#repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos")))
library(microViz)
library(dendextend) #Formatting dendrograms
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
#if (!require("BiocManager", quietly = TRUE))
#install.packages("BiocManager")

#BiocManager::install("decontam")
library(decontam)
#if (!requireNamespace("BiocManager", quietly = TRUE))
#install.packages("BiocManager")
#BiocManager::install("dada2", version = "3.14")
library(dada2)
library(forcats)
#install.packages(c("dbplyr", "dplyr", "dtplyr", "haven", "httr", "jsonlite", "readr", 
#"readxl", "rvest", "tibble", "tidyr", "xml2"))
library(ggpubr)
#install.packages("remotes")
#remotes::install_github("kstagaman/phyloseqCompanion")
library(phyloseqCompanion)
#if (!requireNamespace("BiocManager", quietly = TRUE))
#install.packages("BiocManager")
#BiocManager::install("ANCOMBC")
library(ANCOMBC)
#devtools::install_github("Russel88/DAtest")
library(DAtest)
#BiocManager::install(c("DESeq2","limma","edgeR","metagenomeSeq","baySeq","ALDEx2","impute","ANCOMBC"))

#install.packages(c("samr","pscl","statmod","mvabund"))
```

# Phyloseq object

For a phyloseq object you need 3 pieces of data: 
  
  1. The ASV table with the genetic barcode identifier (the ASV) of each identified bacterial taxa within each insect sample ran (ex. KK100)

2. The taxa table with the genetic barcode of each unique ASV and the respective bacterial classification "Kingdom", "Phylum", "Class", "Order", "Family", "Genus".

3. A metadata file with the sample names (ex. KK100) and any other variables that were measured. Ex. the family of insect or fish organism collected, the site the organism was collected at, the water temperature, the distance to waste water treatment plant, the weight and length of organism, etc.

The phyloseq object will merge all this data together in one object that can be called in by various functions in the phyloseq and microbiome packages. 

```{r Load data and phyloseq object}
#Importing ASV table
ASVseq<-read.csv("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\ASV_data_2022_ED_fulldataset_samplesordered.csv")

#The row names have to be the ASV identifier (genetic barcode) in order for it to be read. The following lines of code are to re-organize the row names
n1 <- ASVseq$X #Currently the ASVs are being called "X" in the first column. We want them to be unnamed in the row names column
ASVseq <- ASVseq[,-1] #This removes the first column of the ASV dataframe
rownames(ASVseq) <- n1 #This moves the ASVs to the rownames column of the dataframe

ASVtable <- as.matrix(ASVseq) #must be a matrix so we convert to a matrix
#View(ASVtable) #The ASVs are now the row names

#Importing taxa table
taxonomy <-read.csv ("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\Taxonomy_ED_2022_fulldataset.csv")

#Using the same steps as above with the ASV table to make the genetic barcode the rowname
n2 <- taxonomy$X
taxonomy <- taxonomy[,-1]
rownames(taxonomy) <- n2

taxa_table<-as.matrix(taxonomy) #Turn into a matrix
#View(taxa_table) #ASVs are now the row names

#Importing metadata file
metadata<-read.csv("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\metadata_ED_withblanks_noKK2078.csv")

metadata$Site<-factor(metadata$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Site_code<-factor(metadata$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Stage<-factor(metadata$Stage, levels=c("Larvae", "Adults", "Spiders"))

#Need to make sample names as row names so we follow the same steps as above. 
n3 <- metadata$Study_ID

# move names to row headings
metadata <- metadata[,-1]
rownames(metadata) <- n3

#View(metadata) #sample names are now row names

#Combining phyloseq object
ASVtable = otu_table(ASVtable, taxa_are_rows = TRUE) #Taxa are rows means that the ASVs of each identified microbe are the row names which we made sure was true
taxa_table = tax_table(taxa_table)

#Make sure the ASV table and taxa table are both matricies before trying to turn into phyloseq object
ps <- phyloseq(ASVtable, sample_data(metadata), taxa_table)
```

# Filtering phyloseq 

```{r}
#Need to remove PMF adult caddis because they were contaminated with contaminated laboratory reagents.
samples_to_remove<-c("POLFLT_1_Ad_Caddis_MB_1", "POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8")

ps_pruned<-subset_samples(ps, !(Sample_ID %in% samples_to_remove))
ps_pruned<-prune_taxa(taxa_sums(ps_pruned) > 0, ps_pruned)
ps_pruned #31,185 taxa, 372 samples

#Removing low confidence data
psexp<-subset_samples(ps_pruned, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,218 taxa and 349 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs

ps2<-phyloseq_filter_prevalence(ps1, prev.trh = 0.01, abund.trh = 6, threshold_condition = "AND")
ps2 #4563 taxa, 349 samples. 
any(taxa_sums(ps2) <6) #FALSE. Means it did a good job of getting rid of them. 

sample_ps <- subset_samples(ps2, sample_sums(ps2) > 2000) 
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2384 reads.
sample_ps #4563 taxa, 335 samples. 
```

# Beta Diversity

## Non-rarefied Analysis {.tabset}
```{r}
#### Bow River samples ####

#Normalize the filtered data by relative abundance 
ps_Bow<-subset_samples(sample_ps, Stream_Type=="Bow River")
Bow_rel_abun  = transform_sample_counts(ps_Bow, function(x) x/sum(x)) 
#transform to relative abundance. Seems to be the best method of normalizing.

Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray") #Bray Curtis distance matrix
ord_Bow = ordinate(Bow_rel_abun, method="PCoA", distance=Bow_bray) #Ordinate using PCoA

#Scree plot of PCoA
scree <- plot_scree(ord_Bow)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
              axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1)

metadata_Bow <- data.frame(sample_data(Bow_rel_abun))

#All together
plot_ordination(Bow_rel_abun, ord_Bow, color="Site", shape="Order") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [11.4%]")+
  ylab("PC2 [8.6%]")

#Faceted by stage (larvae, adults, spiders)
plot_ordination(Bow_rel_abun, ord_Bow, color="Site", shape="Order") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  facet_wrap(~Stage)+
  xlab("PC1 [11.4%]")+
  ylab("PC2 [8.6%]")
#Points are very clustered together and the axes don't explain aton of variation so I will try ordinating with an NMDS as well. 

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/PCoA by site and invertebrate family faceted by life stage Bow River.png", height = 7, width=8)



# NMDS 
set.seed(1)
ord_NMDS_Bow = ordinate(Bow_rel_abun, method="NMDS", distance=Bow_bray)
ord_NMDS_Bow #Stress=0.203
plot_ordination(Bow_rel_abun, ord_NMDS_Bow, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=14), 
        legend.text = element_text(size=14), 
        axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18, color="black"),
        legend.position = c(0.9, 0.3))
#stat_ellipse(type="norm", alpha=0.4, aes(group=Site, color=Site))

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites Bow only relative abundance normalized.png", height=8, width=10)


#Faceted by invertebrate taxa and life Stage
p<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Site") + 
  theme_classic() +
  geom_point(size=2)+
  facet_wrap(vars(Stage, Family), scales="free", nrow=1)+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=20), 
        legend.position = "bottom", 
        axis.text.x=element_text(size=16, color="black", angle=90, hjust=0.5, vjust=0.5),
        axis.text.y=element_text(size=16, color="black"),
        axis.title = element_text(size=18),
        strip.text.x = element_text(size=20))+
  facet_nested_wrap(~Stage + Family, scale="free_x", nrow=1)+
  stat_ellipse()

newtab = data.table(p$data)
newtab$Family <- ordered(newtab$Family,
                         levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Chironomidae ", "Araneidae", "Tetragnathidae"))
p$data <- newtab
print(p)

#ggsave(filename="Microbiome_ED/Final Analysis/Beta Diversity Graphs/All samples combined/NMDS across Bow River invertebrates and sites.png", height=10, width=20)

#Faceted by site
p1<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Site, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=20), 
        legend.position = "bottom", 
        axis.text=element_text(size=18, color="black"),
        axis.title = element_text(size=18, color="black"),
        strip.text.x = element_text(size=20))+
  stat_ellipse(linetype=1, type="norm")+
  guides(color=guide_legend(nrow=3, byrow=TRUE, title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))


newtab = data.table(p1$data)
newtab$Family <- ordered(newtab$Family,
                         levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Chironomidae", "Araneidae", "Tetragnathidae"))
p1$data <- newtab
print(p1)


#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites faceted by invertebrate taxa rarefied Bow only.png", height=10, width=16)


#Faceted by life stage (Larvae, Adults, Spiders)
plot_ordination(Bow_rel_abun, ord_NMDS_Bow, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Stage, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=14), 
        legend.text = element_text(size=12), 
        legend.position = "right")+
  stat_ellipse()

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites faceted by life stage rarefied Bow only.png", height=8, width=16)


#PERMANOVA by Site and invertebrate taxa

test.adonis.Bow <- adonis(Bow_bray ~ Site*Family, data = metadata_Bow)
test.adonis.Bow <- as.data.frame(test.adonis.Bow$aov.tab)
test.adonis.Bow #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two.

#Pairwise comparisons
pairwise.adonis2(Bow_bray ~ Site, data = metadata_Bow) #Gives the same result as above. 
#All sites except COCH-SUN are sig diff when invertebrate taxa are combined

pairwise.adonis2(Bow_bray ~ Family, data = metadata_Bow)
#All invertebrate taxa are sig diff when sites are combined


#Testing variance (beta dispersion) differences between sites, and sample families:
bd_all_site_Bow<-betadisper(Bow_bray, metadata_Bow$Site)
anova(bd_all_site_Bow) #No significant beta dispersion (p=0.313)
permutest(bd_all_site_Bow) #Gives the same result but based on a permutation test. 
#PERMANOVA results could be more driven by the spread in the variance of samples rather than the means if it was significant. (p=0.295)

bd_all_fam_Bow<-betadisper(Bow_bray, metadata_Bow$Family) 
anova(bd_all_fam_Bow) #significantly different p<0.001
permutest(bd_all_fam_Bow) #sig. different (same result)
#Significant PERMANOVA could just be because of the variance within groups rather than between groups (groups = invertebrate taxa type)



## Water quality variables ##


#Graves Bridge does not have water quality info so we have to get rid of this site so that the NAs don't mess up the rest of the analysis.
ps_Bow_no_grvbr<-subset_samples(sample_ps, Stream_Type=="Bow River" & Site!="Graves Bridge")

#Transform to relative abundances to standardize
Bow_rel_abun_no_grvbr  = transform_sample_counts(ps_Bow_no_grvbr, function(x) x/sum(x)) 

env_Bow<-data.frame(sample_data(Bow_rel_abun_no_grvbr))
env_Bow_2<-env_Bow[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Total_Phosphorus", "Total_Dissolved_Solids", "Total_Suspended_Solids", "Total_Organic_Carbon", "pH", "Nitrate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc)


correlation_matrix_env<-as.data.frame(cor(env_Bow_2), method="spearman") #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

#transform to relative abundance. Seems to be the best method of normalizing.

Bow_bray_no_grvbr = phyloseq::distance(Bow_rel_abun_no_grvbr, method="bray") #Bray Curtis distance matrix

#Convert to data frame so it can be used in vegan
Bow_bray_df_no_grvbr<-as.matrix(Bow_bray_no_grvbr)
Bow_bray_df_no_grvbr<-as.data.frame(Bow_bray_df_no_grvbr)

#Ordinate with the bray curtis distance matrix
set.seed(1)
ord_Bow_mds_no_grvbr<-metaMDS(Bow_bray_df_no_grvbr)
sppscores(ord_Bow_mds_no_grvbr) <- env_Bow_2
ord_Bow_mds_no_grvbr #2 dimensions, 0.208 same as when it's done in phyloseq



fit_all <- envfit(ord_Bow_mds_no_grvbr ~ Water_Temp + Dissolved_oxygen + Conductivity + Total_Phosphorus + Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH + Nitrate, data = env_Bow_2, perm = 999)
fit_all

#Remove highly correlated and non-significant variables

bow1<-envfit(ord_Bow_mds_no_grvbr ~ Dissolved_oxygen + Conductivity + Total_Phosphorus + Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH + Nitrate, data = env_Bow_2, perm = 999)
bow1

bow2<-envfit(ord_Bow_mds_no_grvbr ~ Conductivity + Total_Phosphorus + Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH + Nitrate, data = env_Bow_2, perm = 999)
bow2

bow3<-envfit(ord_Bow_mds_no_grvbr ~ Total_Phosphorus + Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH + Nitrate, data = env_Bow_2, perm = 999)
bow3

bow4<-envfit(ord_Bow_mds_no_grvbr ~ Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH + Nitrate, data = env_Bow_2, perm = 999)
bow4

bow5<-envfit(ord_Bow_mds_no_grvbr ~ Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon + pH, data = env_Bow_2, perm = 999)
bow5

bow6<-envfit(ord_Bow_mds_no_grvbr ~ Total_Dissolved_Solids + Total_Suspended_Solids + Total_Organic_Carbon, data = env_Bow_2, perm = 999)
bow6

bow7<-envfit(ord_Bow_mds_no_grvbr ~ Total_Dissolved_Solids + Total_Organic_Carbon, data = env_Bow_2, perm = 999)
bow7

#Total_Dissolved_Solids and Total_Organic_Carbon are significant but have low effect size (p=0.002, r^2=0.0551 and p=0.002, r^2=0.0607 respectively)

#Adjust p-value for multiple factors
bow7.adj <- bow7
pvals.adj <- p.adjust (bow7$vectors$pvals, method = 'bonferroni')
bow7.adj$vectors$pvals <- pvals.adj
bow7.adj #p=0.004 for dissolved solids, p=0.004 for total organic carbon. Still have the same effect sizes though. 

plot1 <- ordiplot(ord_Bow_mds_no_grvbr, choices=c(1,2))

sites.long1 <- sites.long(plot1, env.data=env_Bow_2)
head(sites.long1)

scores(bow7.adj, "vectors")
plot(ord_Bow_mds_no_grvbr)
text(ord_Bow_mds_no_grvbr, display = "sites", cex=0.7, col="blue")
plot(bow7.adj)
plot(bow7.adj, p.max = 0.05, col = "red")


#Total dissolved solids and total organic carbon are significant but don't explain much variance in the data. 

#Ordinating with phyloseq object and MicroViz package.
#Need a phyloseq object without Graves Bridge samples since there are no water quality data for that site

nogrvbr_ps<-subset_samples(sample_ps, Site!="Graves Bridge")
nogrvbr_ps<-prune_taxa(taxa_sums(nogrvbr_ps)>0, nogrvbr_ps)
nogrvbr_ps

tax_fix_interactive(nogrvbr_ps)

nogrvbr_ps %>%
  tax_fix(
    min_length = 4,
    unknowns = c(""),
    sep = " ", anon_unique = TRUE,
    suffix_rank = "classified"
  )


nogrvbr_ps %>%
  ps_mutate(
    Total_Dissolved_Solids = as.numeric(Total_Dissolved_Solids == "Total Dissolved Solids (mg/L)"),
    Total_Organic_Carbon = as.numeric(Total_Organic_Carbon == "Total Organic Carbon (mg/L)")) %>%
  tax_fix()%>%
  tax_transform("compositional", rank = "Genus") %>%
  ord_calc(
    constraints = c("Total Dissolved Solids (mg/L)", "Total Organic Carbon (mg/L)"),
    # method = "RDA", # Note: you can specify RDA explicitly, and it is good practice to do so, but microViz can guess automatically that you want an RDA here (helpful if you don't remember the name?)
    scale_cc = FALSE # doesn't make a difference
  ) %>%
  ord_plot(
    colour = "Family", size = 2, alpha = 0.5, shape = "Site",
    plot_taxa = 1:8
  )

#### Trying RDA with water quality variables ####

#RDA

#Bow River
Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")

no_grvbr_ps<-subset_samples(Bow_ps, Site=="Graves Bridge")

rds_Bow <- ordinate(Bow_ps, "RDA")


temp_rda_Bow <- plot_ordination(no_grvbr_ps, rda_Bow, type="split", color="Family", shape="Site") 
temp_rda

pH_rda <- plot_ordination(Bow_ps, rds_Bow, type="split", color="pH", shape="Site")
pH_rda

K_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, color="Condition.Factor", shape="Site", points=3)
K_rda

K_rda_split <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="Condition.Factor", shape="Location", label="Site")
K_rda_split

length_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="Length..cm.", shape="Location", label="Site")
length_rda

weight_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="Weight..g.", shape="Location", label="Site")
weight_rda

DO_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="DO", shape="Location", label="Site")
DO_rda

con_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="Conductivity", shape="Location", label="Site")
con_rda

tds_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="TDS", shape="Location", label="Site")
tds_rda

NH4_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="NH4", shape="Location", label="Site")
NH4_rda

NH3_rda <- plot_ordination(dat_wq_lessKK, dat_wq_rda, type="split", color="NH3", shape="Location", label="Site")
NH3_rda


bow_ps<- subset_taxa(Bow_ps, Phylum=="Proteobacteria")

Bow_rel_abun <- transform_sample_counts(Bow_ps, function(x) x/sum(x))
Bow_rel_abun_df <- psmelt(Bow_rel_abun)

cor(Bow_rel_abun_df$Abundance, Bow_rel_abun_df(c(20,23)))
cor(Bow_rel_abun_df(c(3,20,23)))


#RDA by condition factor


relabunbray.ord <- ordinate(dat_wq_lessKK, method = "RDA")

relabunbray.plot.site <- plot_ordination(dat_wq_lessKK, relabunbray.ord,
                                         color = "Condition.Factor",
                                         shape = "Site",
                                         axes = c(1,2))

site_label_order = c("1_REF1", "2_REF2", "3_REF3", "4_DSW1", "5_DSW2", "6_DSW3", "7_DSK1", "8_DSK2", "9_DSK3", "10_DSK4")

site_label_name = c("REF 1", "REF 2", "REF 3", "DSW 1", "DSW 2", "DSW 3", "DSK 1", "DSK 2", "DSK 3", "DSK 4")

relabunbray.plot.site$data$Site <- as.character(relabunbray.plot.site$data$Site)

relabunbray.plot.site$data$Site <- factor(relabunbray.plot.site$data$Site, levels=site_label_order, labels=site_label_name)

#colour and point size by condition factor; shapes by site
relabunbray.plot.site + scale_color_gradient(low="blue", high="red") + 
  geom_point(aes(size=Condition.Factor)) +
  scale_shape_manual(values=c(15:25)) +
  theme(axis.text=element_text(size=18), axis.title=element_text(size=18)) +
  theme(legend.text=element_text(size=14, color="black")) +
  theme(legend.title = element_text(size=15, color="black"))

#point size by condition factor; colour by site

relabunbray.plot.site <- plot_ordination(dat_wq_lessKK, relabunbray.ord,
                                         color = "Site",
                                         axes = c(1,2))

site_label_order = c("1_REF1", "2_REF2", "3_REF3", "4_DSW1", "5_DSW2", "6_DSW3", "7_DSK1", "8_DSK2", "9_DSK3", "10_DSK4")

site_label_name = c("REF 1", "REF 2", "REF 3", "DSW 1", "DSW 2", "DSW 3", "DSK 1", "DSK 2", "DSK 3", "DSK 4")

relabunbray.plot.site$data$Site <- as.character(relabunbray.plot.site$data$Site)

relabunbray.plot.site$data$Site <- factor(relabunbray.plot.site$data$Site, levels=site_label_order, labels=site_label_name)


relabunbray.plot.site + 
  geom_point(aes(size=Condition.Factor)) +
  scale_shape_manual(values=c(15:25)) +
  theme(axis.text=element_text(size=18), axis.title=element_text(size=18)) +
  theme(legend.text=element_text(size=14, color="black")) +
  theme(legend.title = element_text(size=15, color="black"))


#old things not used

##
relabunbray.plot <- plot_ordination(dat_wq_lessKK, relabunbray.ord,
                                    color = "Length..cm.",
                                    shape = "Location",
                                    axes = c(1,2))

relabunbray.plot+scale_color_gradient(low="blue", high="red")

##
relabunbray.plot <- plot_ordination(dat_wq_lessKK, relabunbray.ord,
                                    color = "Weight..g.",
                                    shape = "Location",
                                    axes = c(1,2))

relabunbray.plot+scale_color_gradient(low="blue", high="red")


#Bray Curtis - coloured by Abundance
#Can't do because it won't accept the df with abundance, and without the df there is no abundance?


#Bray Curtis - by proteobacteria
dat_proteo <- subset_taxa(dat_wq_lessKK, Phylum=="Proteobacteria")
rel_abun_proteo <- speedyseq::tax_glom(dat_proteo, taxrank = "Family")


relabunbray.ord <- ordinate(rel_abun_proteo, method = "PCoA", distance = "bray")

relabunbray.plot <- plot_ordination(rel_abun_proteo, relabunbray.ord,
                                    color = "Location",
                                    axes = c(4,5))

relabunbray.plot

##PERMANOVA PROTEOBACTERIA
dat_proteo <- subset_taxa(dat_wq_lessKK, Phylum=="Proteobacteria")
metadata <- as(sample_data(dat_proteo), "data.frame")
permanova<-adonis(distance(dat_proteo, method="bray", set.seed=1414) ~ Location, data = metadata)
permanova


#Constrained Analysis on PCoA


#dat_wq_lessKK_no10 <- subset_samples(dat_wq_lessKK, Site != "10_DSK4")  #gut content only and no site 10 (because no wq data for site 10)


cap_K <- ordinate(dat_wq_lessKK, method="CAP", distance="bray", formula=~Condition.Factor )

#Just CAP without formula

cap_K <- ordinate(dat_wq_lessKK, method="CAP", distance="bray")

tankbr_rda <- plot_ordination(
  physeq = dat_wq_lessKK, 
  ordination = cap_K,
  #RDA1 and RDA2?
  axes = c(1,2)) +
  xlim(-1.2, 2) +
  #Plot shape and colour according to distance bin and location
  geom_point(aes(colour = Location, shape = Site), size = 6, stroke = 1) +
  #Change legend titles for shape and colour boxes
  labs(colour = "Location", shape = "Site") +
  #Change order of distance bin appearance; made it appear sequentially. values= To change shapes
  scale_shape_manual(values=c(1:10)) +
  #Change colours
  scale_color_manual(breaks=c("Upstream", "Downstream Waterloo", "Downstream Kitchener"), values = c("#4682B4", "#FFA500", "red")) 

tankbr_rda


#Add the environmental variables/loadings as arrows
arrowmat_rar <- vegan::scores(ord_Bow_mds_no_grvbr, display = "bp") #scores, display biplot

#Add labels, make a data.frame
arrowdf_rar <- data.frame(labels = rownames(arrowmat_rar), arrowmat_rar) #add arrow labels

# Define the arrow aesthetic mapping
arrow_map_rar <- aes(xend = CAP1*1, yend = MDS1*1, #Extend arrow by magnitude
                     x = 0, y = 0, shape = NULL, color = NULL, label = labels)

label_map_rar <- aes(x = 1.2 * CAP1, y = 1.1 * MDS1, #label positioning
                     shape = NULL, color = NULL, label = labels)

arrowhead_rar = arrow(length = unit(0.02, "npc")) #define arrowhead

#Create dbRDA plot
tankbr_rda <- plot_ordination(
  physeq = dat_wq_lessKK, 
  ordination = cap_K,
  #RDA1 and RDA2?
  axes = c(1,2)) +
  xlim(-1.2, 2) +
  #Plot shape and colour according to distance bin and location
  geom_point(aes(colour = Location, shape = Site), size = 6, stroke = 1) +
  #Change legend titles for shape and colour boxes
  labs(colour = "Location", shape = "Site") +
  #Change order of distance bin appearance; made it appear sequentially. values= To change shapes
  scale_shape_manual(values=c(1:10)) +
  #Change colours
  scale_color_manual(breaks=c("Upstream", "Downstream_Waterloo", "Downstream_Kitchener"), values = c("#4682B4", "#FFA500", "red")) +
  #map arrows
  geom_segment(mapping = arrow_map_rar, #call on mapping
               size = 0.70, #arrow width
               data = arrowdf_rar, #arrow eigens presumably
               color = "grey20", 
               arrow = arrowhead_rar) + 
  geom_text(mapping = label_map_rar, #call on mapping
            size = 5.5,  #font size
            fontface=2, #bold arrow labels
            data = arrowdf_rar, #environmental variable names
            show.legend = FALSE) +
  theme(panel.border = element_blank(),
        panel.background = element_rect(fill = "white", #Default panel is grey, so make it white
                                        colour = "black", #Adds black border around plot area
                                        size = 0.5, linetype = "solid"), #Line customizations
        panel.grid.major = element_line(size = 0.5, linetype = 'solid', #Plot major grid lines
                                        colour = "lightgrey"),
        legend.text = element_text(size = 14), legend.title = element_text(size = 16),
        axis.title = element_text(size = 14, face = 2), axis.text = element_text(size = 14))

tankbr_rda

#### ACWA streams ####

ACWA_ps<-subset_samples(sample_ps, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun  = transform_sample_counts(ACWA_ps, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix

ord_ACWA = ordinate(ACWA_rel_abun, method="PCoA", distance="ACWA_bray")

#Scree plot of PCoA
scree <- plot_scree(ord_ACWA)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
              axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1 and 2)

metadata_ACWA <- data.frame(sample_data(ACWA_rel_abun))

#All together
plot_ordination(ACWA_rel_abun, ord_ACWA, color="Family", shape="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [32.8%]")+
  ylab("PC2 [17.9%]")

#Faceted by stage (larvae, adults, spiders)
plot_ordination(ACWA_rel_abun, ord_ACWA, color="Site", shape="Order") + #Plots the ordination
  theme_bw() +
  rremove("grid")+
  facet_nested_wrap(~Stage+factor(Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera")), scale="free_x", nrow=1)+
  xlab("PC1 [32.8%]")+
  ylab("PC2 [17.9%]")+
  stat_ellipse()

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/PCoA by site and invertebrate family faceted by life stage ACWA Streams.png", height = 7, width=8)



# NMDS 
set.seed(1)
ord_NMDS_ACWA = ordinate(ACWA_rel_abun, method="NMDS", distance=ACWA_bray)
ord_NMDS_ACWA #Stress=0.099


plot_ordination(ACWA_rel_abun, ord_NMDS_ACWA, color = "Site", shape="Family") + 
  theme_bw() +
  rremove("grid")+
  geom_point(size=3)+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=12), 
        legend.position = "right")+
  labs(color="Site")+
  stat_ellipse()



plot_ordination(ACWA_rel_abun, ord_NMDS_ACWA, color = "Site") + 
  theme_bw() +
  rremove("grid")+
  geom_point(size=3)+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=12), 
        legend.position = "right")+
  labs(color="Site")+
  stat_ellipse()+
  facet_nested_wrap(~Stage+factor(Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera")), scales="free_x", nrow=1)

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites ACWA only.png", height=6, width=10)


#PERMANOVA by Site and invertebrate taxa
test.adonis.ACWA <- adonis(ACWA_bray ~ Site*Family, data = metadata_ACWA)
test.adonis.ACWA <- as.data.frame(test.adonis.ACWA$aov.tab)
test.adonis.ACWA #p<0.001. Significant dissimilarity between sites and families but no significant interaction between the two (p=0.085).

#Pairwise comparisons
pairwise.adonis2(ACWA_bray ~ Family, data = metadata_ACWA) 
#All invertebrate types are different from one another when all sites are combined

pairwise.adonis2(ACWA_bray ~ Site, data = metadata_ACWA)
#PCR1 - PCR3 are not significantly different from each other but they are both different from BRR2 (control)

#Testing variance (beta dispersion) differences between sites, sampling months, and sample families:
bd_all_site_ACWA<-betadisper(ACWA_bray, metadata_ACWA$Site)
anova(bd_all_site_ACWA) #p<0.001
permutest(bd_all_site_ACWA) #P<0.001 significant beta disperison.

bd_all_fam_ACWA<-betadisper(ACWA_bray, metadata_ACWA$Family) 
anova(bd_all_fam_ACWA) #p<0.001
permutest(bd_all_fam_ACWA) #P<0.001 sig. different (same result)


## Water quality variables ##

ACWA_ps<-subset_samples(sample_ps, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun  = transform_sample_counts(ACWA_ps, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
#Convert to dataframe
ACWA_bray_df<-as.matrix(ACWA_bray)
ACWA_bray_df<-as.data.frame(ACWA_bray_df)

ord_ACWA_mds = metaMDS(ACWA_bray_df) #Ordinate using NDMS

#Extract environmental variables to correlate with
env_ACWA<-data.frame(sample_data(ACWA_rel_abun))
env_ACWA_2<-env_ACWA[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Ammonia", "pH", "Dissolved_Nitrogen", "Nitrate", "Nitrite", "Dissolved_Organic_Carbon", "Orthophosphate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc)

correlation_matrix_env_ACWA<-as.data.frame(cor(env_ACWA_2)) #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 


fit_all_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Dissolved_Organic_Carbon + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_all_ACWA

fit_1_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_1_ACWA

fit_2_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_2_ACWA

fit_3_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_3_ACWA

fit_4_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_4_ACWA

fit_5_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_5_ACWA

fit_6_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite, data = env_ACWA_2, perm = 999)
fit_6_ACWA

fit_7_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Nitrite, data = env_ACWA_2, perm = 999)
fit_7_ACWA


fit_8_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity, data = env_ACWA_2, perm = 999)
fit_8_ACWA

fit_9_ACWA <- envfit(ord_ACWA_mds ~ Conductivity, data = env_ACWA_2, perm = 999)
fit_9_ACWA

#Conductivity is a significant predictor of microbiome community structure (p=0.006, r^2=0.1932)

fitACWA.adj <- fit_9_ACWA
pvals.adj.ACWA <- p.adjust (fit_9_ACWA$vectors$pvals, method = 'bonferroni')
fitACWA.adj$vectors$pvals <- pvals.adj.ACWA
fitACWA.adj #same p-value because there is only the one predictor variable in the model

scores(fitACWA.adj, "vectors")
plot(ord_ACWA_mds) #Why are species scores not available?
plot(fitACWA.adj)
plot(fitACWA.adj, p.max = 0.05, col = "red")

```

### Sites analyzed individually 
```{r}
#Cochrane
coch_rel_abun<-subset_samples(Bow_rel_abun, Site=="Cochrane")

dist_coch<-phyloseq::distance(coch_rel_abun, method="bray")

coch_metadata <- data.frame(sample_data(coch_rel_abun))
coch_adonis <- adonis(dist_coch ~ Family, data = coch_metadata)
coch_adonis_df <- as.data.frame(coch_adonis$aov.tab)
coch_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_coch ~ Family, data = coch_metadata)

#Beta dispersion
coch_bd<-betadisper(dist_coch, coch_metadata$Family)
anova(coch_bd) #p=0.0047. 

#Sunalta 
sun_rel_abun<-subset_samples(Bow_rel_abun, Site=="Sunalta")

dist_sun<-phyloseq::distance(sun_rel_abun, method="bray")

sun_metadata <- data.frame(sample_data(sun_rel_abun))
sun_adonis <- adonis(dist_sun ~ Family, data = sun_metadata)
sun_adonis_df <- as.data.frame(sun_adonis$aov.tab)
sun_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_sun ~ Family, data = sun_metadata)

#Beta dispersion
sun_bd<-betadisper(dist_sun, sun_metadata$Family)
anova(sun_bd) #p=0.00498. 

#Cushing Bridge
cush_rel_abun<-subset_samples(Bow_rel_abun, Site=="Cushing Bridge")

dist_cush<-phyloseq::distance(cush_rel_abun, method="bray")

cush_metadata <- data.frame(sample_data(cush_rel_abun))
cush_adonis <- adonis(dist_cush ~ Family, data = cush_metadata)
cush_adonis_df <- as.data.frame(cush_adonis$aov.tab)
cush_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_cush ~ Family, data = cush_metadata)

#Beta dispersion
cush_bd<-betadisper(dist_cush, cush_metadata$Family)
anova(cush_bd) #p=0.000469. 


#Graves Bridge

grvbr_rel_abun<-subset_samples(Bow_rel_abun, Site=="Graves Bridge")

dist_grvbr<-phyloseq::distance(grvbr_rel_abun, method="bray")

grvbr_metadata <- data.frame(sample_data(grvbr_rel_abun))
grvbr_adonis <- adonis(dist_grvbr ~ Family, data = grvbr_metadata)
grvbr_adonis_df <- as.data.frame(grvbr_adonis$aov.tab)
grvbr_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_grvbr ~ Family, data = grvbr_metadata)

#Beta dispersion
grvbr_bd<-betadisper(dist_grvbr, grvbr_metadata$Family)
anova(grvbr_bd) #p=0.01124. 


#Policeman Flats

pmf_rel_abun<-subset_samples(Bow_rel_abun, Site=="Policeman Flats")

dist_pmf<-phyloseq::distance(pmf_rel_abun, method="bray")

pmf_metadata <- data.frame(sample_data(pmf_rel_abun))
pmf_adonis <- adonis(dist_pmf ~ Family, data = pmf_metadata)
pmf_adonis_df <- as.data.frame(pmf_adonis$aov.tab)
pmf_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_pmf ~ Family, data = pmf_metadata)

#Beta dispersion
pmf_bd<-betadisper(dist_pmf, pmf_metadata$Family)
anova(pmf_bd) #p=0.00102. 



#PCR1
PCR1_rel_abun<-subset_samples(ACWA_rel_abun, Site=="PCR1")

dist_PCR1<-phyloseq::distance(PCR1_rel_abun, method="bray")

PCR1_metadata <- data.frame(sample_data(PCR1_rel_abun))
PCR1_adonis <- adonis(dist_PCR1 ~ Family, data = PCR1_metadata)
PCR1_adonis_df <- as.data.frame(PCR1_adonis$aov.tab)
PCR1_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_PCR1 ~ Family, data = PCR1_metadata)

#Beta dispersion
PCR1_bd<-betadisper(dist_PCR1, PCR1_metadata$Family)
anova(PCR1_bd) #p=0.307. 


#PCR3
PCR3_rel_abun<-subset_samples(ACWA_rel_abun, Site=="PCR3")

dist_PCR3<-phyloseq::distance(PCR3_rel_abun, method="bray")

PCR3_metadata <- data.frame(sample_data(PCR3_rel_abun))
PCR3_adonis <- adonis(dist_PCR3 ~ Family, data = PCR3_metadata)
PCR3_adonis_df <- as.data.frame(PCR3_adonis$aov.tab)
PCR3_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_PCR3 ~ Family, data = PCR3_metadata)

#Beta dispersion
PCR3_bd<-betadisper(dist_PCR3, PCR3_metadata$Family)
anova(PCR3_bd) #p<0.001. 

```


## Rarefied analysis {.tabset}
```{r}

#Rarefy to even sampling depth
set.seed(1)
ps_rare<-rarefy_even_depth(sample_ps, rngseed = F)

#Normalize the filtered data by relative abundance 
ps_Bow_rare<-subset_samples(ps_rare, Stream_Type=="Bow River")
Bow_rel_abun_rare  = transform_sample_counts(ps_Bow_rare, function(x) x/sum(x)) 
#transform to relative abundance. Seems to be the best method of normalizing.

Bow_bray_rare = phyloseq::distance(Bow_rel_abun_rare, method="bray") #Bray Curtis distance matrix
ord_Bow_rare = ordinate(Bow_rel_abun_rare, method="PCoA", distance=Bow_bray_rare) #Ordinate using PCoA

metadata_Bow_rare <- data.frame(sample_data(Bow_rel_abun_rare))


plot_ordination(Bow_rel_abun_rare, ord_Bow_rare, color="Site", shape="Order") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [11.2%]")+
  ylab("PC2 [8.5%]")

#Faceted by stage (larvae, adults, spiders)
plot_ordination(Bow_rel_abun_rare, ord_Bow_rare, color="Site", shape="Order") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  facet_wrap(~Stage)+
  xlab("PC1 [11.2%]")+
  ylab("PC2 [8.5%]")
#Points are very clustered together and the axes don't explain aton of variation so I will try ordinating with an NMDS as well. 


# NMDS 
set.seed(1)
ord_NMDS_Bow_rare = ordinate(Bow_rel_abun_rare, method="NMDS", distance=Bow_bray_rare)
ord_NMDS_Bow_rare #Stress=0.119

plot_ordination(Bow_rel_abun_rare, ord_NMDS_Bow_rare, shape="Site", color = "Family", label="Sample_ID") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=12), 
        legend.position = c(0.9, 0.3))
#stat_ellipse(geom = "polygon", type="norm", alpha=0.4, aes(fill=Family))

#Need to remove outlier

ps_Bow_rare_outlier_removed<-subset_samples(ps_Bow_rare, Sample_ID!="CUSHBR_1_Ad_Caddis_MB_8" & Sample_ID!="CUSHBR_1_Ad_Caddis_MB_2")
Bow_rel_abun_rare_outlier_removed  = transform_sample_counts(ps_Bow_rare_outlier_removed, function(x) x/sum(x)) 
  
Bow_bray_rare_outlier_removed = phyloseq::distance(Bow_rel_abun_rare_outlier_removed, method="bray") #Bray Curtis distance matrix

set.seed(1)
ord_NMDS_Bow_rare_outlier_removed <- ordinate(Bow_rel_abun_rare_outlier_removed, method="NMDS", distance=Bow_bray_rare_outlier_removed) 
ord_NMDS_Bow_rare_outlier_removed #stress = 0.179

plot_ordination(Bow_rel_abun_rare_outlier_removed, ord_NMDS_Bow_rare_outlier_removed, shape="Site", color = "Family", label="Sample_ID") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=12), 
        legend.position = "right")

metadata_Bow_rare_outlier_removed <- data.frame(sample_data(Bow_rel_abun_rare_outlier_removed))
  
  

#Faceted by invertebrate taxa and life Stage
plot_ordination(Bow_rel_abun_rare_outlier_removed, ord_NMDS_Bow_rare_outlier_removed, color = "Site") + 
  theme_classic() +
  geom_point(size=2)+
  facet_wrap(vars(Stage, Family), scales="free", nrow=1)+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=10), 
        legend.position = "bottom", 
        axis.text=element_text(size=8))+
  facet_nested_wrap(~Stage + Family, scale="free_x", nrow=1)+
  stat_ellipse()

#Faceted by site
plot_ordination(Bow_rel_abun_rare_outlier_removed, ord_NMDS_Bow_rare_outlier_removed, color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Site, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=10), 
        legend.position = "bottom", 
        axis.text=element_text(size=8))+
  stat_ellipse(linetype=1, type="norm")


#Faceted by life stage (Larvae, Adults, Spiders)
plot_ordination(ps_rare, ord_NMDS_Bow_rare_outlier_removed, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Stage, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=14), 
        legend.text = element_text(size=12), 
        legend.position = "right")


#PERMANOVA by Site and invertebrate taxa
test.adonis.Bow.rare <- adonis(Bow_bray_rare_outlier_removed ~ Site*Family, data = metadata_Bow_rare_outlier_removed)
test.adonis.Bow.rare <- as.data.frame(test.adonis.Bow.rare$aov.tab)
test.adonis.Bow.rare #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two.

#Pairwise comparisons
pairwise.adonis2(Bow_bray_rare_outlier_removed ~ Site, data = metadata_Bow_rare_outlier_removed) #Gives the same result as the non-rarefied analysis (no difference between Cochrane and Sunalta)

pairwise.adonis2(Bow_bray_rare_outlier_removed ~ Family, data = metadata_Bow_rare_outlier_removed)
#All invertebrate taxa are sig diff when sites are combined. Same as non-rarefied analysis


#Testing variance (beta dispersion) differences between sites, and sample families:
bd_all_site_Bow_rare<-betadisper(Bow_bray_rare_outlier_removed, metadata_Bow_rare_outlier_removed$Site)
anova(bd_all_site_Bow_rare) #No significant beta dispersion (p=0.4301) Same as non rarefied.


bd_all_fam_Bow_rare<-betadisper(Bow_bray_rare_outlier_removed, metadata_Bow_rare_outlier_removed$Family) 
anova(bd_all_fam_Bow_rare) #significantly different p<0.001. Same as non rarefied.



###Using water quality variables to plot correlations ###

## Bow River
#Rarefying the data
ps_rare <- phyloseq::rarefy_even_depth(sample_ps, rngseed = 123, replace = FALSE)
ps_rare_Bow<-subset_samples(ps_rare, Stream_Type=="Bow River")


env_Bow_rare<-data.frame(sample_data(ps_rare_Bow))
env_Bow_rare<-select(data=env_Bow_rare, Total_Dissolved_Solids, Total_Organic_Carbon)
cor(env_Bow_rare) #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

en = envfit(ord_NMDS_Bow, env_Bow, permutations = 999, na.rm = TRUE)
en

#Total dissolved solids and total organic carbon are significant. Get rid of variables that were correlated (pH, Water_Temp, Ammonia, Total_Suspended_Solids, Total_Phosphorus) and run it again. 



### ACWA streams only ###


ps_rare_ACWA<-subset_samples(ps_rare, Stream_Type=="ACWA Streams")

#PCoA with Bray Curtis distance matrix
ACWA_rare_rel_abun  = transform_sample_counts(ps_rare_ACWA, function(x) x/sum(x)) #log transforming the data 

dist_all_ACWA_rare = phyloseq::distance(ACWA_rare_rel_abun, method="bray") #Bray Curtis distance matrix
ordination_all_ACWA_rare = ordinate(ACWA_rare_rel_abun, method="PCoA", distance=dist_all_ACWA_rare) #Ordinate using PCoA

ps_rare_metadata_ACWA <- data.frame(sample_data(ACWA_rare_rel_abun))

#All together
plot_ordination(ACWA_rare_rel_abun, ordination_all_ACWA_rare, shape="Site", color="Family") +
  theme_classic() +
  theme(strip.background = element_blank())+
xlab("PC1 [31.7%]")+
  ylab("PC2 [17.2%]")


# NMDS 
set.seed(1)
ord_NMDS_ACWA_rare = ordinate(ACWA_rare_rel_abun, method="NMDS", distance=dist_all_ACWA_rare)
ord_NMDS_ACWA_rare #Stress=0.1, higher than non rarefied

plot_ordination(ACWA_rare_rel_abun, ord_NMDS_ACWA_rare, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=12), 
        legend.position = c(0.9, 0.8))+
  labs(color="Invertebrate Type")


#Faceted by invertebrate taxa and life Stage

plot_ordination(ACWA_rare_rel_abun, ord_NMDS_ACWA_rare, color = "Site") + 
  theme_classic() +
  geom_point(size=2)+
  facet_wrap(vars(Stage, Family), scales="free", nrow=1)+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=12), 
        legend.text = element_text(size=10), 
        legend.position = "bottom", 
        axis.text=element_text(size=8))+
  facet_nested_wrap(~Stage + Family, nrow=1)+
  stat_ellipse()


#Faceted by life stage (Larvae, Adults, Spiders)
plot_ordination(ACWA_rare_rel_abun, ord_NMDS_ACWA_rare, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Stage, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=14), 
        legend.text = element_text(size=12), 
        legend.position = "right")+
  stat_ellipse()

#PERMANOVA by Site and invertebrate taxa
test.adonis.ACWA.rare <- adonis(dist_all_ACWA_rare ~ Site*Family, data = ps_rare_metadata_ACWA)
test.adonis.ACWA.rare <- as.data.frame(test.adonis.ACWA.rare$aov.tab)
test.adonis.ACWA.rare #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two p=0.086.

#Pairwise comparisons
pairwise.adonis2(dist_all_ACWA_rare ~ Site, data = ps_rare_metadata_ACWA)
#PCR1 and PCR3 are not different. Same as non rarefied analysis

pairwise.adonis2(dist_all_ACWA_rare ~ Family, data = ps_rare_metadata_ACWA)
#All invertebrate taxa are sig diff when sites are combined. Same as non rarefied analysis


#Testing variance (beta dispersion) 
bd_all_site_ACWA_rare<-betadisper(dist_all_ACWA_rare, ps_rare_metadata_ACWA$Site)
anova(bd_all_site_ACWA_rare) #Significant beta dispersion (p<0.001). Opposite of non rarefied analysis


bd_all_fam_ACWA_rare<-betadisper(dist_all_ACWA_rare, ps_rare_metadata_ACWA$Family) 
anova(bd_all_fam_ACWA_rare) #significantly different p<0.001. Same as non rarefied. 

```

### Sites analyzed individually
```{r}
#Cochrane
coch_rel_abun_rare<-subset_samples(Bow_rel_abun_rare_outlier_removed, Site=="Cochrane")

dist_coch_rare<-phyloseq::distance(coch_rel_abun_rare, method="bray")

coch_metadata_rare <- data.frame(sample_data(coch_rel_abun_rare))
coch_adonis_rare <- adonis(dist_coch_rare ~ Family, data = coch_metadata_rare)
coch_adonis_df_rare <- as.data.frame(coch_adonis_rare$aov.tab)
coch_adonis_df_rare #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_coch_rare ~ Family, data = coch_metadata_rare)

#Only difference from non-rarefied analysis is that Tetragnathids are not significantly different from Diptera adults (p=0.056) very close though.

#Beta dispersion
coch_bd_rare<-betadisper(dist_coch_rare, coch_metadata_rare$Family)
anova(coch_bd_rare) #p=0.0062. Same as non rarefied

#Sunalta 
sun_rel_abun<-subset_samples(Bow_rel_abun, Site=="Sunalta")

dist_sun<-phyloseq::distance(sun_rel_abun, method="bray")

sun_metadata <- data.frame(sample_data(sun_rel_abun))
sun_adonis <- adonis(dist_sun ~ Family, data = sun_metadata)
sun_adonis_df <- as.data.frame(sun_adonis$aov.tab)
sun_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_sun ~ Family, data = sun_metadata)

#Beta dispersion
sun_bd<-betadisper(dist_sun, sun_metadata$Family)
anova(sun_bd) #p=0.00498. 

#Cushing Bridge
cush_rel_abun<-subset_samples(Bow_rel_abun, Site=="Cushing Bridge")

dist_cush<-phyloseq::distance(cush_rel_abun, method="bray")

cush_metadata <- data.frame(sample_data(cush_rel_abun))
cush_adonis <- adonis(dist_cush ~ Family, data = cush_metadata)
cush_adonis_df <- as.data.frame(cush_adonis$aov.tab)
cush_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_cush ~ Family, data = cush_metadata)

#Beta dispersion
cush_bd<-betadisper(dist_cush, cush_metadata$Family)
anova(cush_bd) #p=0.000469. 


#Graves Bridge

grvbr_rel_abun<-subset_samples(Bow_rel_abun, Site=="Graves Bridge")

dist_grvbr<-phyloseq::distance(grvbr_rel_abun, method="bray")

grvbr_metadata <- data.frame(sample_data(grvbr_rel_abun))
grvbr_adonis <- adonis(dist_grvbr ~ Family, data = grvbr_metadata)
grvbr_adonis_df <- as.data.frame(grvbr_adonis$aov.tab)
grvbr_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_grvbr ~ Family, data = grvbr_metadata)

#Beta dispersion
grvbr_bd<-betadisper(dist_grvbr, grvbr_metadata$Family)
anova(grvbr_bd) #p=0.01124. 


#Policeman Flats

pmf_rel_abun<-subset_samples(Bow_rel_abun, Site=="Policeman Flats")

dist_pmf<-phyloseq::distance(pmf_rel_abun, method="bray")

pmf_metadata <- data.frame(sample_data(pmf_rel_abun))
pmf_adonis <- adonis(dist_pmf ~ Family, data = pmf_metadata)
pmf_adonis_df <- as.data.frame(pmf_adonis$aov.tab)
pmf_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_pmf ~ Family, data = pmf_metadata)

#Beta dispersion
pmf_bd<-betadisper(dist_pmf, pmf_metadata$Family)
anova(pmf_bd) #p=0.00102. 



#PCR1
PCR1_rel_abun<-subset_samples(ACWA_rel_abun, Site=="PCR1")

dist_PCR1<-phyloseq::distance(PCR1_rel_abun, method="bray")

PCR1_metadata <- data.frame(sample_data(PCR1_rel_abun))
PCR1_adonis <- adonis(dist_PCR1 ~ Family, data = PCR1_metadata)
PCR1_adonis_df <- as.data.frame(PCR1_adonis$aov.tab)
PCR1_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_PCR1 ~ Family, data = PCR1_metadata)

#Beta dispersion
PCR1_bd<-betadisper(dist_PCR1, PCR1_metadata$Family)
anova(PCR1_bd) #p=0.307. 


#PCR3
PCR3_rel_abun<-subset_samples(ACWA_rel_abun, Site=="PCR3")

dist_PCR3<-phyloseq::distance(PCR3_rel_abun, method="bray")

PCR3_metadata <- data.frame(sample_data(PCR3_rel_abun))
PCR3_adonis <- adonis(dist_PCR3 ~ Family, data = PCR3_metadata)
PCR3_adonis_df <- as.data.frame(PCR3_adonis$aov.tab)
PCR3_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(dist_PCR3 ~ Family, data = PCR3_metadata)

#Beta dispersion
PCR3_bd<-betadisper(dist_PCR3, PCR3_metadata$Family)
anova(PCR3_bd) #p<0.001. 

```
