---
output: html_document
editor_options: 
  chunk_output_type: console
---
title: "Bow River Data Organization and Exploration"
author: "Emilie Diesbourg"
date: "26/05/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---
# Introduction {.tabset}

This data comes from a study done in 2022 across 5 sites on the Bow River, Calgary AB and 3 microcosm streams from Pine Creek wastewater treatment plant with varying municipal wastewater effluent exposures (mesocosm streams: 0, 5, or 15% effluent flow). The objective of this study is to determine whether there are changes to the microbiome of freshwater macroinvertebrates and riparian Spiders with increased exposure to wastewater as well as any differences across taxa or whether the microbiome is retained across metamorphosis. 

## Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
#Change margin size to be smaller so graphs fit in the plot panel
par(mar = c(2, 2, 2, 2)) # Set the margin on all sides to 2
```

## Loading Packages
```{r Load packages}
library(rlang)
library(glue)
library(ggh4x)
library(phyloseq) #phyloseq will be the main package used for structuring microbiome data and diversity comparisons
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#install.packages("viridis")
library(viridis)

#BiocManager::install("apeglm")
library(apeglm)
library(geosphere)
library(car)
library(ggplot2) #For creating graphs
#install.packages("ggtext")
library(ggtext)
library(dplyr) #Helps with data wrangling 
library(cluster) #Used for cluster analyses
library(grid)
library(ape)
library(pals)
library(FSA)
library(multcomp)
library(MASS)
library(glmmTMB)
library(gridExtra)
library(vegan)
library(tidyverse) #data wrangling
library(microbiome) #For microbiome analyses
#if (!require("BiocManager", quietly = TRUE)) #Installing microbiome package
  #install.packages("BiocManager")
#BiocManager::install("microbiome")
library(PERFect) #Permutation filtration for microbiome data. Used when comparing beta diversities across sites/locations. 
#if(!requireNamespace("BiocManager", quietly = TRUE))
  #install.packages("BiocManager")
#BiocManager::install("PERFect")
library(knitr) #For R Markdown knitting
library(kableExtra)
library(speedyseq)
#if (!requireNamespace("remotes", quietly = TRUE))
  #install.packages("remotes")
#remotes::install_github("mikemc/speedyseq")
library(colorspace)
library(RColorBrewer)
library(picante)
library(ggpubr)
library(data.table)
library(AICcmodavg)
#devtools::install_github("microsud/microbiomeutilities")
library(microbiomeutilities)
library(imsig)
library(phangorn) #For phylogenetic tree
library(metacoder) 
library(tibble)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("DESeq2")
library(DESeq2)
library(emmeans)
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis", force=TRUE)
library(pairwiseAdonis)
#library(plyr) apparently if you load this after dpylr it messes up the summarize function?

#if( !require(NbClust) ){install.packages("NbClust"); library(NbClust)}
#if( !require(cobiclust) ){install.packages("cobiclust"); library(cobiclust)}
library(NbClust)
library(cobiclust)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("mia")
library(mia)
library(factoextra)
#install.packages(
  #"microViz",
  #repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos")))
library(microViz)
library(dendextend) #Formatting dendrograms
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("decontam")
library(decontam)
#if (!requireNamespace("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("dada2", version = "3.14")
library(dada2)
library(forcats)
#install.packages(c("dbplyr", "dplyr", "dtplyr", "haven", "httr", "jsonlite", "readr", 
#"readxl", "rvest", "tibble", "tidyr", "xml2"))
library(ggpubr)
#install.packages("remotes")
#remotes::install_github("kstagaman/phyloseqCompanion")
library(phyloseqCompanion)
#if (!requireNamespace("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("ANCOMBC")
library(ANCOMBC)
#devtools::install_github("Russel88/DAtest")
library(DAtest)
#BiocManager::install(c("DESeq2","limma","edgeR","metagenomeSeq","baySeq","ALDEx2","impute","ANCOMBC"))

#install.packages(c("samr","pscl","statmod","mvabund"))
```

# Phyloseq object

For a phyloseq object you need 3 pieces of data: 

1. The ASV table with the genetic barcode identifier (the ASV) of each identified bacterial taxa within each insect sample ran (ex. KK100)

2. The taxa table with the genetic barcode of each unique ASV and the respective bacterial classification "Kingdom", "Phylum", "Class", "Order", "Family", "Genus".

3. A metadata file with the sample names (ex. KK100) and any other variables that were measured. Ex. the family of insect or fish organism collected, the site the organism was collected at, the water temperature, the distance to waste water treatment plant, the weight and length of organism, etc.

The phyloseq object will merge all this data together in one object that can be called in by various functions in the phyloseq and microbiome packages. 

```{r Load data and phyloseq object}
#Importing ASV table
ASVseq<-read.csv("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\ASV_data_2022_ED_fulldataset_samplesordered.csv")

#The row names have to be the ASV identifier (genetic barcode) in order for it to be read. The following lines of code are to re-organize the row names
n1 <- ASVseq$X #Currently the ASVs are being called "X" in the first column. We want them to be unnamed in the row names column
ASVseq <- ASVseq[,-1] #This removes the first column of the ASV dataframe
rownames(ASVseq) <- n1 #This moves the ASVs to the rownames column of the dataframe

ASVtable <- as.matrix(ASVseq) #must be a matrix so we convert to a matrix
#View(ASVtable) #The ASVs are now the row names

#Importing taxa table
taxonomy <-read.csv ("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\Taxonomy_ED_2022_fulldataset.csv")

#Using the same steps as above with the ASV table to make the genetic barcode the rowname
n2 <- taxonomy$X
taxonomy <- taxonomy[,-1]
rownames(taxonomy) <- n2

taxa_table<-as.matrix(taxonomy) #Turn into a matrix
#View(taxa_table) #ASVs are now the row names

#Importing metadata file
metadata<-read.csv("C:\\Users\\Emilie\\Documents\\McMaster University 2022-2024\\Bow River Thesis\\BowRiverRProject\\Microbiome_ED\\Raw data\\metadata_ED_withblanks_noKK2078.csv")

metadata$Site<-factor(metadata$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Site_code<-factor(metadata$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Stage<-factor(metadata$Stage, levels=c("Larvae", "Adults", "Spiders"))

#Need to make sample names as row names so we follow the same steps as above. 
n3 <- metadata$Study_ID

# move names to row headings
metadata <- metadata[,-1]
rownames(metadata) <- n3

#View(metadata) #sample names are now row names

#Combining phyloseq object
ASVtable = otu_table(ASVtable, taxa_are_rows = TRUE) #Taxa are rows means that the ASVs of each identified microbe are the row names which we made sure was true
taxa_table = tax_table(taxa_table)

#Make sure the ASV table and taxa table are both matricies before trying to turn into phyloseq object
ps <- phyloseq(ASVtable, sample_data(metadata), taxa_table)
```

# Microbiome filtering steps
```{r}
#Need to remove PMF adult caddis because they were contaminated with contaminated laboratory reagents.
samples_to_remove<-c("POLFLT_1_Ad_Caddis_MB_1", "POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8")

ps_pruned<-subset_samples(ps, !(Sample_ID %in% samples_to_remove))
ps_pruned<-prune_taxa(taxa_sums(ps_pruned) > 0, ps_pruned)
ps_pruned #31,185 taxa, 372 samples

#Removing low confidence data
psexp<-subset_samples(ps_pruned, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,218 taxa and 349 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs

ps2<-phyloseq_filter_prevalence(ps1, prev.trh = 0.01, abund.trh = 6, threshold_condition = "AND")
ps2 #4563 taxa, 349 samples. 
any(taxa_sums(ps2) <6) #FALSE. Means it did a good job of getting rid of them. 

sample_ps <- subset_samples(ps2, sample_sums(ps2) > 2000) 
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2384 reads.
sample_ps #4563 taxa, 335 samples.
```

# Data diagnostics

In this section I filter out different taxonomic organizations of bacterial taxa to see the abundance at different levels of organization. Ex. how many phyla per site, what are the top phyla, families, genera and their percentage of the total. 

## Blank Samples
```{r}
blank_ps<-subset_samples(ps, Sample_Type=="Blank")
blank_ps<-prune_taxa(taxa_sums(blank_ps) > 0, blank_ps)
nsamples(blank_ps) #23
ntaxa(blank_ps) #1809

sum(sample_sums(blank_ps)) #503,506 total reads
mean(sample_sums(blank_ps)) #21,891 average reads/sample
sort(sample_sums(blank_ps)) # reads/sample ranges from 16 - 63,500 counts

#The extraction reagent blanks from the first 4 plates are relatively low (16 - 629 counts)
#The extraction reagent blanks from the last plate is very high (44,645 and 57,629 counts)
#Buffer tube blanks were run on the last (contaminated) plate and have 56,500 and 63,500 counts
#Field blanks range from 7,949 - 49,817 counts.

#Relative abundance of top 5 most abundant classification of bacteria. 
#Percent of each phyla to the total bacteria
sort((table(tax_table(blank_ps)[,2]))/ntaxa(blank_ps)*100)
#Firmicutes (42.3 %), Proteobacteria (19.6 %), Bacteroidota (13.4 %), Actinobacteriota (11.0 %), Cyanobacteria (5.25 %)

#Percent of each class to the total bacteria
sort((table(tax_table(blank_ps)[,3]))/ntaxa(blank_ps)*100)
#Clostridia (35.2 %), Bacteroidia (13.3 %), Alphaproteobacteria (10.1)

#Percent of each order to the total bacteria
sort((table(tax_table(blank_ps)[,4]))/ntaxa(blank_ps)*100)
#Lachnospirales (15.0 %), Oscillospirales (14.2 %), Burkholderiales (6.5 %)

#Percent of each family to the total bacteria
sort((table(tax_table(blank_ps)[,5]))/ntaxa(blank_ps)*100)
#Lachnospiraceae (14.9 %), Ruminococcaceae (6.9 %), Oscillospiraceae (5.5 %)

#Percent of each genus to the total bacteria
sort((table(tax_table(blank_ps)[,6]))/ntaxa(blank_ps)*100)
#Faecalibacterium (1.99 %), [Ruminococcus] torques group (1.82 %), Blautia (1.60 %), Bacteroides (1.55 %), Alistipes (1.55 %)

#Percent of each species to the total bacteria
sort((table(tax_table(blank_ps)[,7]))/ntaxa(blank_ps)*100)
#prausnitzii (1.0 %), bacterium (0.44%), hadrus (0.39%)

## Get total numbers of ASVs at each level
get_taxa_unique(blank_ps, "Phylum") #26 phyla total
get_taxa_unique(blank_ps, "Class") #57 Classes total
get_taxa_unique(blank_ps, "Order") #136 orders total
get_taxa_unique(blank_ps, "Family") #199 families total
get_taxa_unique(blank_ps, "Genus") #400 genera total

# Get total number of ASVs within each blank type

## Water July
water_july<-subset_samples(blank_ps, Blank_collection_method=="Water July")
water_july<-prune_taxa(taxa_sums(water_july)>0, water_july)
water_july #153 taxa, 3 samples

get_taxa_unique(water_july, "Phylum") #11 phyla total
get_taxa_unique(water_july, "Class") #21 Classes total
get_taxa_unique(water_july, "Order") #54 orders total
get_taxa_unique(water_july, "Family") #63 families total
get_taxa_unique(water_july, "Genus") #76 genera total

## Water September

water_sept<-subset_samples(blank_ps, Blank_collection_method=="Water September")
water_sept<-prune_taxa(taxa_sums(water_sept)>0, water_sept)
water_sept #286 taxa, 3 samples

get_taxa_unique(water_sept, "Phylum") #19 phyla total
get_taxa_unique(water_sept, "Class") #32 Classes total
get_taxa_unique(water_sept, "Order") #73 orders total
get_taxa_unique(water_sept, "Family") #95 families total
get_taxa_unique(water_sept, "Genus") #126 genera total

## Bedsheet July

bedsheet<-subset_samples(blank_ps, Blank_collection_method=="Sheet")
bedsheet<-prune_taxa(taxa_sums(bedsheet)>0, bedsheet)
bedsheet #749 taxa, 4 samples

get_taxa_unique(bedsheet, "Phylum") #17 phyla total
get_taxa_unique(bedsheet, "Class") #37 Classes total
get_taxa_unique(bedsheet, "Order") #92 orders total
get_taxa_unique(bedsheet, "Family") #133 families total
get_taxa_unique(bedsheet, "Genus") #249 genera total

## Floating emergence traps September

mesh_trap<-subset_samples(blank_ps, Blank_collection_method=="Mesh")
mesh_trap<-prune_taxa(taxa_sums(mesh_trap)>0, mesh_trap)
mesh_trap #242 taxa, 3 samples

get_taxa_unique(mesh_trap, "Phylum") #18 phyla total
get_taxa_unique(mesh_trap, "Class") #34 Classes total
get_taxa_unique(mesh_trap, "Order") #63 orders total
get_taxa_unique(mesh_trap, "Family") #86 families total
get_taxa_unique(mesh_trap, "Genus") #101 genera total

## DNA extraction reagents (plate 4)

DNAext4<-subset_samples(blank_ps, Blank_collection_method=="Extraction_neg_4")
DNAext4<-prune_taxa(taxa_sums(DNAext4)>0, DNAext4)
DNAext4 #382 taxa, 2 samples

get_taxa_unique(DNAext4, "Phylum") #9 phyla total
get_taxa_unique(DNAext4, "Class") #13 Classes total
get_taxa_unique(DNAext4, "Order") #31 orders total
get_taxa_unique(DNAext4, "Family") #45 families total
get_taxa_unique(DNAext4, "Genus") #92 genera total

## DNA extraction reagents (plates 1-3)

DNAext<-subset_samples(blank_ps, Blank_collection_method=="Extraction_neg")
DNAext<-prune_taxa(taxa_sums(DNAext)>0, DNAext)
DNAext #23 taxa, 6 samples

get_taxa_unique(DNAext, "Phylum") #3 phyla total
get_taxa_unique(DNAext, "Class") #5 Classes total
get_taxa_unique(DNAext, "Order") #10 orders total
get_taxa_unique(DNAext, "Family") #10 families total
get_taxa_unique(DNAext, "Genus") #15 genera total

## Lysis Buffer tubes

buffer<-subset_samples(blank_ps, Blank_collection_method=="Buffer")
buffer<-prune_taxa(taxa_sums(buffer)>0, buffer)
buffer #447 taxa, 2 samples

get_taxa_unique(buffer, "Phylum") #11 phyla total
get_taxa_unique(buffer, "Class") #17 Classes total
get_taxa_unique(buffer, "Order") #48 orders total
get_taxa_unique(buffer, "Family") #64 families total
get_taxa_unique(buffer, "Genus") #114 genera total



## Plotting composition of blanks
#Phylum 
blank_phy <- tax_glom(blank_ps, taxrank="Phylum")
blank_rel_phy <- microbiome::transform(blank_phy, "compositional")

blank_rel_phy_df <- psmelt(blank_rel_phy)

blank_phy_agg <- aggregate(Abundance ~ Phylum + Blank_Type, data = blank_rel_phy_df, FUN = mean) 

blank_phy_plot <- ggplot(blank_phy_agg, aes(x=Blank_Type, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  scale_fill_manual(name="Bacterial Class", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "#43464B", "lightcoral", "palevioletred2", brewer.pal(8, "Set3")))+
  rremove("grid")
blank_phy_plot

#Family
blank_fam <- tax_glom(blank_ps, taxrank="Family")
blank_rel_fam <- microbiome::transform(blank_fam, "compositional")
blank_rel_fam_prune = prune_taxa(taxa_sums(blank_rel_fam) > 0.05, blank_rel_fam)

blank_rel_fam_df <- psmelt(blank_rel_fam_prune)

blank_fam_agg <- aggregate(Abundance ~ Family + Blank_Type, data = blank_rel_fam_df, FUN = mean) 

blank_fam_plot <- ggplot(blank_fam_agg, aes(x=Blank_Type, y=Abundance, fill=Family)) + geom_bar(stat="identity")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
blank_fam_plot

#Genus
blank_genus <- tax_glom(blank_ps, taxrank="Genus")
blank_rel_genus <- microbiome::transform(blank_genus, "compositional")
blank_rel_genus_prune = prune_taxa(taxa_sums(blank_rel_genus) > 0.1, blank_rel_genus)

blank_rel_genus_df <- psmelt(blank_rel_genus_prune)

blank_genus_agg <- aggregate(Abundance ~ Genus + Blank_Type, data = blank_rel_genus_df, FUN = mean) 

blank_genus_plot <- ggplot(blank_genus_agg, aes(x=Blank_Type, y=Abundance, fill=Genus)) + geom_bar(stat="identity")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
blank_genus_plot
```

## Experimental samples {.tabset}

### Sample sizes 
```{r}
Bow<-subset_samples(sample_ps, Stream_Type=="Bow River")

#Larvae
larvae_cochrane<-subset_samples(Bow, Site=="Cochrane" & Stage=="Larvae")
nsamples(larvae_cochrane) #28

larvae_sunalta<-subset_samples(Bow, Site=="Sunalta" & Stage=="Larvae")
nsamples(larvae_sunalta) #15

larvae_cushbr<-subset_samples(Bow, Site=="Cushing Bridge" & Stage=="Larvae")
nsamples(larvae_cushbr) #24

larvae_grvbr<-subset_samples(Bow, Site=="Graves Bridge" & Stage=="Larvae")
nsamples(larvae_grvbr) #16

larvae_pmf<-subset_samples(Bow, Site=="Policeman Flats" & Stage=="Larvae")
nsamples(larvae_pmf) #28

#Adults
ad_cochrane<-subset_samples(Bow, Site=="Cochrane" & Stage=="Adults")
nsamples(ad_cochrane) #21

ad_sunalta<-subset_samples(Bow, Site=="Sunalta" & Stage=="Adults")
nsamples(ad_sunalta) #16

ad_cushbr<-subset_samples(Bow, Site=="Cushing Bridge" & Stage=="Adults")
nsamples(ad_cushbr) #24

ad_grvbr<-subset_samples(Bow, Site=="Graves Bridge" & Stage=="Adults")
nsamples(ad_grvbr) #15

ad_pmf<-subset_samples(Bow, Site=="Policeman Flats" & Stage=="Adults")
nsamples(ad_pmf) #16

#Spiders
spi_cochrane<-subset_samples(Bow, Site=="Cochrane" & Stage=="Spiders")
nsamples(spi_cochrane) #16

spi_sunalta<-subset_samples(Bow, Site=="Sunalta" & Stage=="Spiders")
nsamples(spi_sunalta) #16

spi_cushbr<-subset_samples(Bow, Site=="Cushing Bridge" & Stage=="Spiders")
nsamples(spi_cushbr) #15

spi_grvbr<-subset_samples(Bow, Site=="Graves Bridge" & Stage=="Spiders")
nsamples(spi_grvbr) #16

spi_pmf<-subset_samples(Bow, Site=="Policeman Flats" & Stage=="Spiders")
nsamples(spi_pmf) #14

```

### Relative abundance at different taxonomic levels (all samples combined)
```{r}

sort((table(tax_table(sample_ps)[,2]))/sum(ntaxa(sample_ps))*100)
#Top 5 abundant phyla are Proteobacteria (37.4 %), Bacteroidota (31.1 %), Firmicutes (10.4 %), Cyanobacteria (6.03 %), Actinobacteria (5.41 %)

sort((table(tax_table(sample_ps)[,5])/sum(ntaxa(sample_ps))*100))
#Top 5 families are Comamonadaceae (7.89 %), Saprospiraceae (6.57 %), Flavobacteriaceae (4.76 %), Rhodobacteraceae (4.23 %), and Sphingomonadaceae (3.51 %)

sort((table(tax_table(sample_ps)[,6])/ntaxa(sample_ps)*100))
#top 5 genera: Flavobacterium (4.51 %), Rhodoferax (2.19 %), Tyzzerella (1.47 %), Pseudorhodobacter (1.34), Limnohabitans (1.21 %)

#Sort by proteobacteria
proteo <- subset_taxa(sample_ps, Phylum=="Proteobacteria")
ntaxa(proteo) #1684
sort(table(tax_table(proteo)[,3])) #class. Only 2 classes. Alphaproteobacteria (48.7%) and Gammaproteobacteria (51.1%)
sort(table(tax_table(proteo)[,5])) #family Top 3: Comamonadaceae (21.2 %), Rhodobacteraceae (11.5 %), Sphingomonadaceae (9.5 %) 
sort(table(tax_table(proteo)[,6])) #genera Top 3: Rhodoferax (5.9 %), Pseudorhodobacter (3.6 %), Limnohabitans (3.3 %)

#Sort by Bacteroidota
bac <- subset_taxa(sample_ps, Phylum=="Bacteroidota")
ntaxa(bac) #1404
sort(table(tax_table(bac)[,3])) #class. Top class: Bacteroidia (97.5 %)
sort(table(tax_table(bac)[,5])) #family Top 3: Saprospiraceae (21.4 %), Flavobacteriaceae (15.5 %), Chitinophagaceae (10.3 %)  
sort(table(tax_table(bac)[,6])) #genera Top 3: Flavobacterium (14.7 %), Hymenobacter (3.0 %), Ferruginibacter (2.8 %)

#Sort by Firmicutes
firm <- subset_taxa(sample_ps, Phylum=="Firmicutes")
ntaxa(firm) #393
sort(table(tax_table(firm)[,3])) #class. Top 2: Clostridia (64.4 %), Bacilli (32.8 %) 
sort(table(tax_table(firm)[,5])) #family Top 2: Lachnospiraceae (25.2 %), Ruminococcaceae (15. 5%) 
sort(table(tax_table(firm)[,6])) #genera Top 3: Tyzzerella (17.0 %), Candidatus Soleaferrea (4.8 %), Staphylococcus (4.3 %)


##Get total numbers of ASVs at each level
get_taxa_unique(sample_ps, "Phylum") #32 phyla total
get_taxa_unique(sample_ps, "Class") #70 Classes total
get_taxa_unique(sample_ps, "Order") #170 orders total
get_taxa_unique(sample_ps, "Family") #241 families total
get_taxa_unique(sample_ps, "Genus") #467 genera total
```

### Number of ASVs per invertebrate taxa

```{r}
#Bow River sample breakdown
#Hydro larvae
hydro_lar_Bow<-subset_samples(sample_ps, Stage=="Larvae" & Family=="Hydropsychidae" & Stream_Type=="Bow River")
nsamples(hydro_lar_Bow) #36

hep_lar_Bow<-subset_samples(sample_ps, Stage=="Larvae" & Family=="Heptageniidae" & Stream_Type=="Bow River")
nsamples(hep_lar_Bow) #40

perl_Bow<-subset_samples(sample_ps, Stage=="Larvae" & Family=="Perlidae" & Stream_Type=="Bow River")
nsamples(perl_Bow) #21

chiro_lar_Bow<-subset_samples(sample_ps, Stage=="Larvae" & Family=="Chironomidae" & Stream_Type=="Bow River")
nsamples(chiro_lar_Bow) #14

araneid_Bow<-subset_samples(sample_ps, Family=="Araneidae")
nsamples(araneid_Bow) #37

tetra_Bow<-subset_samples(sample_ps, Family=="Tetragnathidae")
nsamples(tetra_Bow) #40

ad_caddis_Bow<-subset_samples(sample_ps, Stage=="Adults" & Order=="Trichoptera" & Stream_Type=="Bow River")
nsamples(ad_caddis_Bow) #31

mayfly_ad_Bow<-subset_samples(sample_ps, Stage=="Adults" & Order=="Ephemeroptera" & Stream_Type=="Bow River")
nsamples(mayfly_ad_Bow) #40

chiro_ad_Bow<-subset_samples(sample_ps, Stage=="Adults" & Order=="Diptera" & Stream_Type=="Bow River")
nsamples(chiro_ad_Bow) #21



#Hydropsychidae
hydro<-subset_samples(sample_ps, Family=="Hydropsychidae")
hydro<-prune_taxa(taxa_sums(hydro) > 0, hydro)
ntaxa(hydro) #2552

#Heptageniidae
hep<-subset_samples(sample_ps, Family=="Heptageniidae")
hep<-prune_taxa(taxa_sums(hep) > 0, hep)
ntaxa(hep) #2643

#Baetidae
baetid<-subset_samples(sample_ps, Family=="Baetidae")
baetid<-prune_taxa(taxa_sums(baetid) > 0, baetid)
ntaxa(baetid) #1450

#Perlidae
perlid<-subset_samples(sample_ps, Family=="Perlidae")
perlid<-prune_taxa(taxa_sums(perlid) > 0, perlid)
ntaxa(perlid) #1713

#Chironomidae
chiro<-subset_samples(sample_ps, Family=="Chironomidae")
chiro<-prune_taxa(taxa_sums(chiro) > 0, chiro)
ntaxa(chiro) #1137

#Araneidae
aran<-subset_samples(sample_ps, Family=="Araneidae")
aran<-prune_taxa(taxa_sums(aran) > 0, aran)
ntaxa(aran) #687

#Tetragnathidae
tetra<-subset_samples(sample_ps, Family=="Tetragnathidae")
tetra<-prune_taxa(taxa_sums(tetra) > 0, tetra)
ntaxa(tetra) #571

#Ephemeroptera
ephem<-subset_samples(sample_ps, Order=="Ephemeroptera")
ephem<-prune_taxa(taxa_sums(ephem) > 0, ephem)
ntaxa(ephem) #3501

#Trichoptera
tricop<-subset_samples(sample_ps, Order=="Trichoptera")
tricop<-prune_taxa(taxa_sums(tricop) > 0, tricop)
ntaxa(tricop) #3407

#Plecoptera
plec<-subset_samples(sample_ps, Order=="Plecoptera")
plec<-prune_taxa(taxa_sums(plec) > 0, plec)
ntaxa(plec) #1713

#Araneae
Spiders<-subset_samples(sample_ps, Order=="Araneae")
Spiders<-prune_taxa(taxa_sums(Spiders) > 0, Spiders)
ntaxa(Spiders) #816
nsamples(Spiders) #77
```

### Total ASVs per site and invertebrate taxa 

```{r}
#Total samples
nsamples(sample_ps) #335
ntaxa(sample_ps) #4,563

#Cochrane, n = 65
coch<-subset_samples(sample_ps, Site=="Cochrane")
coch<-prune_taxa(taxa_sums(coch) > 0, coch)
nsamples(coch) #65
ntaxa(coch) #2467

get_taxa_unique(coch, "Phylum") #25 phyla total
get_taxa_unique(coch, "Class") #54 Classes total
get_taxa_unique(coch, "Order") #128 orders total
get_taxa_unique(coch, "Family") #186 families total
get_taxa_unique(coch, "Genus") #363 genera total

sort((table(tax_table(coch)[,2]))/ntaxa(coch)*100)

#Trichoptera at Cochrane
tricoch<-subset_samples(coch, Order=="Trichoptera")
tricoch<-prune_taxa(taxa_sums(tricoch)>0, tricoch)
ntaxa(tricoch) #1249
nsamples(tricoch) #16

nsamples(subset_samples(ps, Family=="Chironomidae"))
nsamples(subset_samples(sample_ps, Family=="Chironomidae"))



get_taxa_unique(tricoch, "Phylum") #19 phyla total
get_taxa_unique(tricoch, "Class") #38 Classes total
get_taxa_unique(tricoch, "Order") #90 orders total
get_taxa_unique(tricoch, "Family") #130 families total
get_taxa_unique(tricoch, "Genus") #214 genera total

#ephemeroptera at Cochrane
ephcoch<-subset_samples(coch, Order=="Ephemeroptera")
ephcoch<-prune_taxa(taxa_sums(ephcoch)>0, ephcoch)
ntaxa(ephcoch) #1277
nsamples(ephcoch) #16

get_taxa_unique(ephcoch, "Phylum") #19 phyla total
get_taxa_unique(ephcoch, "Class") #41 Classes total
get_taxa_unique(ephcoch, "Order") #91 orders total
get_taxa_unique(ephcoch, "Family") #150 families total
get_taxa_unique(ephcoch, "Genus") #244 genera total

#Diptera at Cochrane
dipcoch<-subset_samples(coch, Order=="Diptera")
dipcoch<-prune_taxa(taxa_sums(dipcoch)>0, dipcoch)
ntaxa(dipcoch) #495
nsamples(dipcoch) #9

get_taxa_unique(dipcoch, "Phylum") #10 phyla total
get_taxa_unique(dipcoch, "Class") #26 Classes total
get_taxa_unique(dipcoch, "Order") #78 orders total
get_taxa_unique(dipcoch, "Family") #108 families total
get_taxa_unique(dipcoch, "Genus") #175 genera total

#Plecoptera at Cochrane
pleccoch<-subset_samples(coch, Order=="Plecoptera")
pleccoch<-prune_taxa(taxa_sums(pleccoch)>0, pleccoch)
ntaxa(pleccoch) #570
nsamples(pleccoch) #8

get_taxa_unique(pleccoch, "Phylum") #12 phyla total
get_taxa_unique(pleccoch, "Class") #21 Classes total
get_taxa_unique(pleccoch, "Order") #50 orders total
get_taxa_unique(pleccoch, "Family") #80 families total
get_taxa_unique(pleccoch, "Genus") #108 genera total

#Spiders at Cochrane
arancoch<-subset_samples(coch, Order=="Araneae")
arancoch<-prune_taxa(taxa_sums(arancoch)>0, arancoch)
ntaxa(arancoch) #340
nsamples(arancoch) #16

get_taxa_unique(arancoch, "Phylum") #9 phyla total
get_taxa_unique(arancoch, "Class") #14 Classes total
get_taxa_unique(arancoch, "Order") #49 orders total
get_taxa_unique(arancoch, "Family") #77 families total
get_taxa_unique(arancoch, "Genus") #136 genera total

#Sunalta, n = 47
sun<-subset_samples(sample_ps, Site=="Sunalta")
sun<-prune_taxa(taxa_sums(sun) > 0, sun)
ntaxa(sun) #2173
nsamples(sun) #47

get_taxa_unique(sun, "Phylum") #25 phyla total
get_taxa_unique(sun, "Class") #55 Classes total
get_taxa_unique(sun, "Order") #121 orders total
get_taxa_unique(sun, "Family") #179 families total
get_taxa_unique(sun, "Genus") #332 genera total

sort((table(tax_table(sun)[,2]))/ntaxa(sun)*100)

#Trichoptera at sunalta
trisun<-subset_samples(sun, Order=="Trichoptera")
trisun<-prune_taxa(taxa_sums(trisun) > 0, trisun)
ntaxa(trisun) #996
nsamples(trisun) #15

get_taxa_unique(trisun, "Phylum") #20 phyla total
get_taxa_unique(trisun, "Class") #33 Classes total
get_taxa_unique(trisun, "Order") #76 orders total
get_taxa_unique(trisun, "Family") #122 families total
get_taxa_unique(trisun, "Genus") #199 genera total

#Ephem in sunalta
ephsun<-subset_samples(sun, Order=="Ephemeroptera")
ephsun<-prune_taxa(taxa_sums(ephsun) > 0, ephsun)
ntaxa(ephsun) #1480
nsamples(ephsun) #16

get_taxa_unique(ephsun, "Phylum") #19 phyla total
get_taxa_unique(ephsun, "Class") #44 Classes total
get_taxa_unique(ephsun, "Order") #100 orders total
get_taxa_unique(ephsun, "Family") #149 families total
get_taxa_unique(ephsun, "Genus") #260 genera total

#Araneae in sunalta
aransun<-subset_samples(sun, Order=="Araneae")
aransun<-prune_taxa(taxa_sums(aransun) > 0, aransun)
ntaxa(aransun) #362
nsamples(aransun) #16

get_taxa_unique(aransun, "Phylum") #8 phyla total
get_taxa_unique(aransun, "Class") #15 Classes total
get_taxa_unique(aransun, "Order") #50 orders total
get_taxa_unique(aransun, "Family") #74 families total
get_taxa_unique(aransun, "Genus") #133 genera total

#Cushing Bridge, n = 63
cushbr<-subset_samples(sample_ps, Site=="Cushing Bridge")
cushbr<-prune_taxa(taxa_sums(cushbr) > 0, cushbr)
ntaxa(cushbr) #2761
nsamples(cushbr) #63

get_taxa_unique(cushbr, "Phylum") #29 phyla total
get_taxa_unique(cushbr, "Class") #58 Classes total
get_taxa_unique(cushbr, "Order") #140 orders total
get_taxa_unique(cushbr, "Family") #203 families total
get_taxa_unique(cushbr, "Genus") #400 genera total

sort((table(tax_table(cushbr)[,2]))/ntaxa(cushbr)*100)

#Trichoptera in cushing bridge
tricush<-subset_samples(cushbr, Order=="Trichoptera")
tricush<-prune_taxa(taxa_sums(tricush) > 0, tricush)
ntaxa(tricush) #1099
nsamples(tricush) #13

get_taxa_unique(tricush, "Phylum") #20 phyla total
get_taxa_unique(tricush, "Class") #34 Classes total
get_taxa_unique(tricush, "Order") #80 orders total
get_taxa_unique(tricush, "Family") #132 families total
get_taxa_unique(tricush, "Genus") #218 genera total

#Ephemeroptera in cushing bridge
ephcush<-subset_samples(cushbr, Order=="Ephemeroptera")
ephcush<-prune_taxa(taxa_sums(ephcush) > 0, ephcush)
ntaxa(ephcush) #1276
nsamples(ephcush) #16

get_taxa_unique(ephcush, "Phylum") #20 phyla total
get_taxa_unique(ephcush, "Class") #42 Classes total
get_taxa_unique(ephcush, "Order") #102 orders total
get_taxa_unique(ephcush, "Family") #148 families total
get_taxa_unique(ephcush, "Genus") #256 genera total

#Diptera in cushbr
dipcush<-subset_samples(cushbr, Order=="Diptera")
dipcush<-prune_taxa(taxa_sums(dipcush) > 0, dipcush)
ntaxa(dipcush) #687
nsamples(dipcush) #14

get_taxa_unique(dipcush, "Phylum") #16 phyla total
get_taxa_unique(dipcush, "Class") #32 Classes total
get_taxa_unique(dipcush, "Order") #81 orders total
get_taxa_unique(dipcush, "Family") #129 families total
get_taxa_unique(dipcush, "Genus") #183 genera total

#plecoptera in cush br
pleccush<-subset_samples(cushbr, Order=="Plecoptera")
pleccush<-prune_taxa(taxa_sums(pleccush) > 0, pleccush)
ntaxa(pleccush) #1190
nsamples(pleccush) #5

get_taxa_unique(pleccush, "Phylum") #20 phyla total
get_taxa_unique(pleccush, "Class") #39 Classes total
get_taxa_unique(pleccush, "Order") #90 orders total
get_taxa_unique(pleccush, "Family") #125 families total
get_taxa_unique(pleccush, "Genus") #202 genera total

#araneae in cush br
arancush<-subset_samples(cushbr, Order=="Araneae")
arancush<-prune_taxa(taxa_sums(arancush) > 0, arancush)
ntaxa(arancush) #468
nsamples(arancush) #15

get_taxa_unique(arancush, "Phylum") #12 phyla total
get_taxa_unique(arancush, "Class") #23 Classes total
get_taxa_unique(arancush, "Order") #68 orders total
get_taxa_unique(arancush, "Family") #96 families total
get_taxa_unique(arancush, "Genus") #172 genera total

#Graves Bridge, n = 47
grvbr<-subset_samples(sample_ps, Site=="Graves Bridge")
grvbr<-prune_taxa(taxa_sums(grvbr) > 0, grvbr)
ntaxa(grvbr) #2662
nsamples(grvbr) #47

get_taxa_unique(grvbr, "Phylum") #25 phyla total
get_taxa_unique(grvbr, "Class") #54 Classes total
get_taxa_unique(grvbr, "Order") #136 orders total
get_taxa_unique(grvbr, "Family") #199 families total
get_taxa_unique(grvbr, "Genus") #379 genera total

sort((table(tax_table(grvbr)[,2]))/ntaxa(grvbr)*100)

#Trichoptera at graves bridge
trigrvbr<-subset_samples(grvbr, Order=="Trichoptera")
trigrvbr<-prune_taxa(taxa_sums(trigrvbr) > 0, trigrvbr)
ntaxa(trigrvbr) #1594
nsamples(trigrvbr) #15

get_taxa_unique(trigrvbr, "Phylum") #22 phyla total
get_taxa_unique(trigrvbr, "Class") #42 Classes total
get_taxa_unique(trigrvbr, "Order") #101 orders total
get_taxa_unique(trigrvbr, "Family") #147 families total
get_taxa_unique(trigrvbr, "Genus") #245 genera total

ephgrvbr<-subset_samples(grvbr, Order=="Ephemeroptera")
ephgrvbr<-prune_taxa(taxa_sums(ephgrvbr) > 0, ephgrvbr)
ntaxa(ephgrvbr) #1472
nsamples(ephgrvbr) #16

get_taxa_unique(ephgrvbr, "Phylum") #20 phyla total
get_taxa_unique(ephgrvbr, "Class") #44 Classes total
get_taxa_unique(ephgrvbr, "Order") #98 orders total
get_taxa_unique(ephgrvbr, "Family") #153 families total
get_taxa_unique(ephgrvbr, "Genus") #269 genera total

arangrvbr<-subset_samples(grvbr, Order=="Araneae")
arangrvbr<-prune_taxa(taxa_sums(arangrvbr) > 0, arangrvbr)
ntaxa(arangrvbr) #375
nsamples(arangrvbr) #16

get_taxa_unique(arangrvbr, "Phylum") #9 phyla total
get_taxa_unique(arangrvbr, "Class") #17 Classes total
get_taxa_unique(arangrvbr, "Order") #53 orders total
get_taxa_unique(arangrvbr, "Family") #84 families total
get_taxa_unique(arangrvbr, "Genus") #148 genera total


#Policeman Flats, n = 58
pmf<-subset_samples(sample_ps, Site=="Policeman Flats")
pmf<-prune_taxa(taxa_sums(pmf) > 0, pmf)
ntaxa(pmf) #3168
nsamples(pmf) #58

get_taxa_unique(pmf, "Phylum") #27 phyla total
get_taxa_unique(pmf, "Class") #61 Classes total
get_taxa_unique(pmf, "Order") #153 orders total
get_taxa_unique(pmf, "Family") #218 families total
get_taxa_unique(pmf, "Genus") #417 genera total

sort((table(tax_table(pmf)[,2]))/ntaxa(pmf)*100)

tripmf<-subset_samples(pmf, Order=="Trichoptera")
tripmf<-prune_taxa(taxa_sums(tripmf) > 0, tripmf)
ntaxa(tripmf) #1195
nsamples(tripmf) #8

get_taxa_unique(tripmf, "Phylum") #21 phyla total
get_taxa_unique(tripmf, "Class") #41 Classes total
get_taxa_unique(tripmf, "Order") #85 orders total
get_taxa_unique(tripmf, "Family") #107 families total
get_taxa_unique(tripmf, "Genus") #139 genera total

ephpmf<-subset_samples(pmf, Order=="Ephemeroptera")
ephpmf<-prune_taxa(taxa_sums(ephpmf) > 0, ephpmf)
ntaxa(ephpmf) #1812
nsamples(ephpmf) #16

get_taxa_unique(ephpmf, "Phylum") #22 phyla total
get_taxa_unique(ephpmf, "Class") #49 Classes total
get_taxa_unique(ephpmf, "Order") #113 orders total
get_taxa_unique(ephpmf, "Family") #173 families total
get_taxa_unique(ephpmf, "Genus") #296 genera total

dippmf<-subset_samples(pmf, Order=="Diptera")
dippmf<-prune_taxa(taxa_sums(dippmf) > 0, dippmf)
ntaxa(dippmf) #865
nsamples(dippmf) #12

get_taxa_unique(dippmf, "Phylum") #16 phyla total
get_taxa_unique(dippmf, "Class") #35 Classes total
get_taxa_unique(dippmf, "Order") #87 orders total
get_taxa_unique(dippmf, "Family") #132 families total
get_taxa_unique(dippmf, "Genus") #208 genera total

plecpmf<-subset_samples(pmf, Order=="Plecoptera")
plecpmf<-prune_taxa(taxa_sums(plecpmf) > 0, plecpmf)
ntaxa(plecpmf) #753
nsamples(plecpmf) #8

get_taxa_unique(plecpmf, "Phylum") #17 phyla total
get_taxa_unique(plecpmf, "Class") #30 Classes total
get_taxa_unique(plecpmf, "Order") #71 orders total
get_taxa_unique(plecpmf, "Family") #96 families total
get_taxa_unique(plecpmf, "Genus") #133 genera total

aranpmf<-subset_samples(pmf, Order=="Araneae")
aranpmf<-prune_taxa(taxa_sums(aranpmf) > 0, aranpmf)
ntaxa(aranpmf) #522
nsamples(aranpmf) #14

get_taxa_unique(aranpmf, "Phylum") #11 phyla total
get_taxa_unique(aranpmf, "Class") #19 Classes total
get_taxa_unique(aranpmf, "Order") #67 orders total
get_taxa_unique(aranpmf, "Family") #103 families total
get_taxa_unique(aranpmf, "Genus") #187 genera total

#BRR2 (ACWA control), n = 8
BRR2<-subset_samples(sample_ps, Site=="BRR2")
BRR2<-prune_taxa(taxa_sums(BRR2) > 0, BRR2)
ntaxa(BRR2) #1324
nsamples(BRR2) #8

get_taxa_unique(BRR2, "Phylum") #22 phyla total
get_taxa_unique(BRR2, "Class") #48 Classes total
get_taxa_unique(BRR2, "Order") #113 orders total
get_taxa_unique(BRR2, "Family") #132 families total
get_taxa_unique(BRR2, "Genus") #172 genera total

sort((table(tax_table(BRR2)[,2]))/ntaxa(BRR2)*100)

#PCR1 (ACWA 5% effluent stream), n = 20
PCR1<-subset_samples(sample_ps, Site=="PCR1")
PCR1<-prune_taxa(taxa_sums(PCR1) > 0, PCR1)
ntaxa(PCR1) #2179
nsamples(PCR1) #24

get_taxa_unique(PCR1, "Phylum") #26 phyla total
get_taxa_unique(PCR1, "Class") #59 Classes total
get_taxa_unique(PCR1, "Order") #139 orders total
get_taxa_unique(PCR1, "Family") #178 families total
get_taxa_unique(PCR1, "Genus") #284 genera total

sort((table(tax_table(PCR1)[,2]))/ntaxa(PCR1)*100)

triPCR1<-subset_samples(PCR1, Order=="Trichoptera")
triPCR1<-prune_taxa(taxa_sums(triPCR1) > 0, triPCR1)
ntaxa(triPCR1) #1584
nsamples(triPCR1) #8

get_taxa_unique(triPCR1, "Phylum") #25 phyla total
get_taxa_unique(triPCR1, "Class") #54 Classes total
get_taxa_unique(triPCR1, "Order") #122 orders total
get_taxa_unique(triPCR1, "Family") #143 families total
get_taxa_unique(triPCR1, "Genus") #205 genera total

ephPCR1<-subset_samples(PCR1, Order=="Ephemeroptera")
ephPCR1<-prune_taxa(taxa_sums(ephPCR1) > 0, ephPCR1)
ntaxa(ephPCR1) #1271
nsamples(ephPCR1) #16

get_taxa_unique(ephPCR1, "Phylum") #21 phyla total
get_taxa_unique(ephPCR1, "Class") #48 Classes total
get_taxa_unique(ephPCR1, "Order") #111 orders total
get_taxa_unique(ephPCR1, "Family") #153 families total
get_taxa_unique(ephPCR1, "Genus") #237 genera total


#PCR3 (ACWA 15% effluent stream), n = 15
PCR3<-subset_samples(sample_ps, Site=="PCR3")
PCR3<-prune_taxa(taxa_sums(PCR3) > 0, PCR3)
ntaxa(PCR3) #2036
nsamples(PCR3) #23

get_taxa_unique(PCR3, "Phylum") #27 phyla total
get_taxa_unique(PCR3, "Class") #55 Classes total
get_taxa_unique(PCR3, "Order") #130 orders total
get_taxa_unique(PCR3, "Family") #178 families total
get_taxa_unique(PCR3, "Genus") #295 genera total

sort((table(tax_table(PCR3)[,2]))/ntaxa(PCR3)*100)

triPCR3<-subset_samples(PCR3, Order=="Trichoptera")
triPCR3<-prune_taxa(taxa_sums(triPCR3) > 0, triPCR3)
ntaxa(triPCR3) #1348
nsamples(triPCR3) #8

get_taxa_unique(triPCR3, "Phylum") #24 phyla total
get_taxa_unique(triPCR3, "Class") #48 Classes total
get_taxa_unique(triPCR3, "Order") #110 orders total
get_taxa_unique(triPCR3, "Family") #134 families total
get_taxa_unique(triPCR3, "Genus") #182 genera total

ephPCR3<-subset_samples(PCR3, Order=="Ephemeroptera")
ephPCR3<-prune_taxa(taxa_sums(ephPCR3) > 0, ephPCR3)
ntaxa(ephPCR3) #1140
nsamples(ephPCR3) #15

get_taxa_unique(ephPCR3, "Phylum") #24 phyla total
get_taxa_unique(ephPCR3, "Class") #45 Classes total
get_taxa_unique(ephPCR3, "Order") #104 orders total
get_taxa_unique(ephPCR3, "Family") #144 families total
get_taxa_unique(ephPCR3, "Genus") #244 genera total

#Larvae
larvae<-subset_samples(sample_ps, Stage=="Larvae")
larvae<-prune_taxa(taxa_sums(larvae) > 0, larvae)
nsamples(larvae) #150

#Adults
Adults<-subset_samples(sample_ps, Stage=="Adults")
Adults<-prune_taxa(taxa_sums(Adults) > 0, Adults)
nsamples(Adults) #108

#77 Spiders samples
```

#### Relative abundance across sites 
```{r}
#Cochrane
#Larvae
cochlar<-subset_samples(coch, Stage=="Larvae")
cochlar<-prune_taxa(taxa_sums(cochlar) > 0, cochlar)
sort((table(tax_table(cochlar)[,2]))/ntaxa(cochlar)*100)
#Adults
cochad<-subset_samples(coch, Stage=="Adults")
cochad<-prune_taxa(taxa_sums(cochad) > 0, cochad)
sort((table(tax_table(cochad)[,2]))/ntaxa(cochad)*100)
#Spiders
cochSpiders<-subset_samples(coch, Stage=="Spiderss")
cochSpiders<-prune_taxa(taxa_sums(cochSpiders) > 0, cochSpiders)
sort((table(tax_table(cochSpiders)[,2]))/ntaxa(cochSpiders)*100)
#Hydropsychidae larvae
coch_hydro_lar<-subset_samples(sample_ps, Site=="Cochrane" & Stage=="Larvae" & Family=="Hydropsychidae")
coch_hydro_lar<-prune_taxa(taxa_sums(coch_hydro_lar)> 0, coch_hydro_lar)
sort((table(tax_table(coch_hydro_lar)[,2]))/ntaxa(coch_hydro_lar)*100)

#Heptageniidae larvae
coch_hep_lar<-subset_samples(sample_ps, Site=="Cochrane" & Stage=="Larvae" & Family=="Heptageniidae")
coch_hep_lar<-prune_taxa(taxa_sums(coch_hep_lar)> 0, coch_hep_lar)
sort((table(tax_table(coch_hep_lar)[,2]))/ntaxa(coch_hep_lar)*100)

#Perlidae larvae
coch_perl_lar<-subset_samples(sample_ps, Site=="Cochrane" & Stage=="Larvae" & Family=="Perlidae")
coch_perl_lar<-prune_taxa(taxa_sums(coch_perl_lar)> 0, coch_perl_lar)
sort((table(tax_table(coch_perl_lar)[,2]))/sum(ntaxa(coch_perl_lar))*100)



hydro_lar_bow_rel_abun <- tax_glom(hydro_lar_Bow, "Phylum")
hydro_lar_bow_rel_abun_top <- merge_samples(hydro_lar_bow_rel_abun, top=5)
ps4 <- merge_samples(ps3.top, "Sex")


hydro_lar_Bow_df<-psmelt(hydro_lar_Bow)
hydro_lar_Bow_df$Phylum <- as.character(hydro_lar_Bow_df$Phylum)
hydro_lar_Bow_df <- hydro_lar_Bow_df %>%
group_by(Phylum, Site, sample_Family) %>%
mutate(median=median(Abundance))

#to get the same rows together
hydro_lar_Bow_df_sum <- hydro_lar_Bow_df %>%
group_by(Site, Phylum) %>%
summarise(Abundance=mean(Abundance))

mean(hydro_lar_Bow_df_sum$Site)




summarize_taxa(ps4)
#Heptageniidae larvae
sort((table(tax_table(hep_lar_Bow)[,2]))/ntaxa(hep_lar_Bow)*100)
#Chironomid larvae
sort((table(tax_table(chiro_lar_Bow)[,2]))/ntaxa(chiro_lar_Bow)*100)

#Sunalta
#Larvae
sunlar<-subset_samples(sun, Stage=="Larvae")
sunlar<-prune_taxa(taxa_sums(sunlar) > 0, sunlar)
sort((table(tax_table(sunlar)[,2]))/ntaxa(sunlar)*100)
#Adultss
sunad<-subset_samples(sun, Stage=="Adultss")
sunad<-prune_taxa(taxa_sums(sunad) > 0, sunad)
sort((table(tax_table(sunad)[,2]))/ntaxa(sunad)*100)
#Spiders
sunSpiders<-subset_samples(sun, Stage=="Spiders")
sunSpiders<-prune_taxa(taxa_sums(sunSpiders) > 0, sunSpiders)
sort((table(tax_table(sunSpiders)[,2]))/ntaxa(sunSpiders)*100)

#Cushing Bridge
#Larvae
cushbrlar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Cushing Bridge")
cushbrlar<-prune_taxa(taxa_sums(cushbrlar) > 0, cushbrlar)
sort((table(tax_table(cushbrlar)[,2]))/ntaxa(cushbrlar)*100)
#Adults
cushbrad<-subset_samples(cushbr, Stage=="Adults")
cushbrad<-prune_taxa(taxa_sums(cushbrad) > 0, cushbrad)
sort((table(tax_table(cushbrad)[,2]))/ntaxa(cushbrad)*100)
#Spiders
cushbrSpiders<-subset_samples(cushbr, Stage=="Spiders")
cushbrSpiders<-prune_taxa(taxa_sums(cushbrSpiders) > 0, cushbrSpiders)
sort((table(tax_table(cushbrSpiders)[,2]))/ntaxa(cushbrSpiders)*100)

#Graves Bridge
#Larvae
grvbrlar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Graves Bridge")
grvbrlar<-prune_taxa(taxa_sums(grvbrlar) > 0, grvbrlar)
sort((table(tax_table(grvbrlar)[,2]))/ntaxa(grvbrlar)*100)
#Adults
grvbrad<-subset_samples(grvbr, Stage=="Adults")
grvbrad<-prune_taxa(taxa_sums(grvbrad) > 0, grvbrad)
sort((table(tax_table(grvbrad)[,2]))/ntaxa(grvbrad)*100)
#Spiders
grvbrSpiders<-subset_samples(grvbr, Stage=="Spiders")
grvbrSpiders<-prune_taxa(taxa_sums(grvbrSpiders) > 0, grvbrSpiders)
sort((table(tax_table(grvbrSpiders)[,2]))/ntaxa(grvbrSpiders)*100)

#Policeman Flats
#Larvae
pmflar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Policeman Flats")
pmflar<-prune_taxa(taxa_sums(pmflar) > 0, pmflar)
sort((table(tax_table(pmflar)[,2]))/ntaxa(pmflar)*100)
#Adults
pmfad<-subset_samples(pmf, Stage=="Adults")
pmfad<-prune_taxa(taxa_sums(pmfad) > 0, pmfad)
sort((table(tax_table(pmfad)[,2]))/ntaxa(pmfad)*100)
#Spiders
pmfSpiders<-subset_samples(pmf, Stage=="Spiders")
pmfSpiders<-prune_taxa(taxa_sums(pmfSpiders) > 0, pmfSpiders)
sort((table(tax_table(pmfSpiders)[,2]))/ntaxa(pmfSpiders)*100)

#BRR2
#Larvae
sort((table(tax_table(BRR2)[,2]))/ntaxa(BRR2)*100)

#PCR1
#Larvae
PCR1lar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="PCR1")
PCR1lar<-prune_taxa(taxa_sums(PCR1lar) > 0, PCR1lar)
sort((table(tax_table(PCR1lar)[,2]))/ntaxa(PCR1lar)*100)
#Adults
PCR1ad<-subset_samples(PCR1, Stage=="Adults")
PCR1ad<-prune_taxa(taxa_sums(PCR1ad) > 0, PCR1ad)
sort((table(tax_table(PCR1ad)[,2]))/ntaxa(PCR1ad)*100)

#PCR3
#Larvae
PCR3lar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="PCR3")
PCR3lar<-prune_taxa(taxa_sums(PCR3lar) > 0, PCR3lar)
sort((table(tax_table(PCR3lar)[,2]))/ntaxa(PCR3lar)*100)
```

