---
title: "Bow River Microbiome Analysis 2022"
author: "Emilie Diesbourg"
date: "26/05/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---
# Introduction {.tabset}

This data comes from a study done in 2022 across 5 sites on the Bow River, Calgary AB and 3 experimental streams from Advancing Canadian Water Assets (ACWA) at the Pine Creek waste water treatment plant with varying municipal waste water effluent exposures (stream additions: 0, 5, or 15% effluent). The objective of this study was to determine whether there are changes to the microbiome of freshwater macroinvertebrates and riparian spiders with exposure to waste water effluents as well as if there were any differences in the microbial assemblage across taxa and across metamorphosis/life stage. 

## Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
#Change margin size to be smaller so graphs fit in the plot panel
par(mar = c(2, 2, 2, 2)) # Set the margin on all sides to 2
```

## Loading Packages
```{r Load packages}
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("phyloseq")
library(phyloseq) #phyloseq will be the main package used for structuring microbiome data and diversity comparisons
#BiocManager::install("apeglm")
library(apeglm)
#BiocManager::install("microbiome")
library(microbiome) #For microbiome analyses
library(rlang)
library(glue)
library(ggh4x)
library(viridis)
library(geosphere)
library(car)
library(ggplot2) #For creating graphs
library(ggtext)
library(plyr) #if you load this after dpylr it messes up the dplyr functions
library(dplyr) #Helps with data wrangling 
library(cluster) #Used for cluster analyses
library(grid)
library(ape)
library(pals)
library(FSA)
library(multcomp)
library(MASS)
library(glmmTMB)
library(gridExtra)
library(vegan)
library(tidyverse) #data wrangling
#BiocManager::install("PERFect")
library(PERFect) #Permutation filtration for microbiome data. Used when comparing beta diversities across sites/locations. 
library(knitr) #For R Markdown knitting
library(kableExtra)
#remotes::install_github("mikemc/speedyseq")
library(speedyseq)
library(colorspace)
library(RColorBrewer)
library(picante)
library(ggpubr)
library(data.table)
library(AICcmodavg)
#devtools::install_github("microsud/microbiomeutilities")
library(microbiomeutilities)
library(imsig)
library(phangorn) #For phylogenetic tree
library(metacoder) 
library(tibble)
#BiocManager::install("DESeq2")
library(DESeq2)
library(emmeans)
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)
#BiocManager::install("mia")
library(mia)
library(factoextra)
#install.packages("microViz", repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos")))
library(microViz)
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
#BiocManager::install("decontam")
library(decontam)
#BiocManager::install("dada2", version = "3.16") 3.16 with R v 4.2.3, 3.17 with R v 4.3.2
library(dada2)
library(forcats)
library(ggpubr)
#remotes::install_github("kstagaman/phyloseqCompanion")
library(phyloseqCompanion)
#BiocManager::install("ANCOMBC")
library(ANCOMBC)
#devtools::install_github("Russel88/DAtest")
library(DAtest)
#remotes::install_github("Russel88/MicEco")
library(MicEco)
library(readxl)
```

# Phyloseq object

For a phyloseq object you need 3 pieces of data: 

1. The ASV table with the genetic barcode identifier (the ASV) of each identified bacterial taxa within each insect sample ran (ex. KK100)

2. The taxa table with the genetic barcode of each unique ASV and the respective bacterial classification "Kingdom", "Phylum", "Class", "Order", "Family", "Genus".

3. A metadata file with the sample names (ex. KK100) and any other variables that were measured. Ex. the family of insect or fish organism collected, the site the organism was collected at, the water temperature, the distance to waste water treatment plant, the weight and length of organism, etc.

The phyloseq object will merge all this data together in one object that can be called in by various functions in the phyloseq and microbiome packages. 

```{r Load data and phyloseq object}
#Importing ASV table
ASVseq<-read.csv("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\ASV_data_2022_ED_fulldataset_samplesordered.csv")

#The row names have to be the ASV identifier (genetic barcode) in order for it to be read. The following lines of code are to re-organize the row names
n1 <- ASVseq$X #Currently the ASVs are being called "X" in the first column. We want them to be unnamed in the row names column
ASVseq <- ASVseq[,-1] #This removes the first column of the ASV dataframe
rownames(ASVseq) <- n1 #This moves the ASVs to the rownames column of the dataframe

ASVtable <- as.matrix(ASVseq) #must be a matrix so we convert to a matrix
#View(ASVtable) #The ASVs are now the row names

#Importing taxa table
taxonomy <-read.csv ("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\Taxonomy_ED_2022_fulldataset.csv")

#Using the same steps as above with the ASV table to make the genetic barcode the rowname
n2 <- taxonomy$X
taxonomy <- taxonomy[,-1]
rownames(taxonomy) <- n2

taxa_table<-as.matrix(taxonomy) #Turn into a matrix
#View(taxa_table) #ASVs are now the row names

#Importing metadata file
metadata<-read.csv("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\metadata_noKK2078_samples_ordered.csv")

metadata$Site<-factor(metadata$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Site_code<-factor(metadata$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Stage<-factor(metadata$Stage, levels=c("Larvae", "Adults", "Spiders"))

#Need to make sample names as row names so we follow the same steps as above. 
n3 <- metadata$Study_ID

# move names to row headings
metadata <- metadata[,-1]
rownames(metadata) <- n3

#View(metadata) #sample names are now row names

#Combining phyloseq object
ASVtable = otu_table(ASVtable, taxa_are_rows = TRUE) #Taxa are rows means that the ASVs of each identified microbe are the row names which we made sure was true
taxa_table = tax_table(taxa_table)

#Make sure the ASV table and taxa table are both matricies before trying to turn into phyloseq object
ps <- phyloseq(ASVtable, sample_data(metadata), taxa_table)
```


# Filtering/pre-processing steps {.tabset}

Experimental samples:

Removing low confidence data: I removed all ASVs identified as Eukaryota and phyla that were identified as <NA>. This removed 39 and 403 taxa respectively. 

Removing low abundance taxa (including singletons and doubletons): I removed taxa whose sums of reads were less than or equal to 6 across all samples.  

Removing low prevalence taxa: I removed taxa that were only found in less than 4 samples. 

Removing low read samples: I removed all samples whose total reads were less than 2000. 

Rarefying: Results from statistical analysis (alpha and beta diversity) will be compared before and after rarefying to the minimum sampling depth. The minimum even sampling depth is 2384 reads (the lowest read count after filtering). If there is no difference in results, other methods of normalization will be used (i.e., transform to relative abundances or log transformation) besides rarefaction since it removes a large portion of the data. 

In total, ~85 % of the taxa (ASVs) were filtered out but only 0.32 % of the entire dataset (reads) were filtered out from the original decontaminated dataset. This may have consequences for diversity measures but ensures sequencing errors and potential contaminants are removed. 

### Decontamination Steps using 'Decontam'
```{r}
## Subsetting the blanks from each invert collection method and decontaminating with a threshold of 0.1 and merging them back together ##

ps #31,556 taxa, 380 samples

## Adult mayfly from ACWA ##

ad_mesh_bl<-subset_samples(ps, Blank_collection_method=="Mesh") #Subsets all mayflies and blanks that were collected from the floating emergence traps
ad_mesh_bl<-prune_taxa(taxa_sums(ad_mesh_bl) > 0, ad_mesh_bl)
ad_mesh_bl #837 taxa, 19 samples

#Decontam steps for removing floating emergence trap contaminants from ACWA adult mayflies
ad_mesh_bl_df <- as.data.frame(sample_data(ad_mesh_bl))

sample_data(ad_mesh_bl)$is.neg <- sample_data(ad_mesh_bl)$Sample_Type == "Blank" #Specifies that the negative control are the blank samples
contamdf.prev.ad.mesh <- isContaminant(ad_mesh_bl, method="prevalence", neg="is.neg", threshold=0.1)
#Runs the decontamination with a threshold of 0.1. 
hist(contamdf.prev.ad.mesh$p) 
table(contamdf.prev.ad.mesh$contaminant)
#Threshold of 0.1: FALSE = 822, TRUE=15. Identifies 15 potential contaminants.

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.ad.mesh <- prune_taxa(!contamdf.prev.ad.mesh$contaminant, ad_mesh_bl) #Keeps the 822 taxa that were not ID as contaminants
ps.noncontam.ad.mesh 


## Larvae from ACWA ##

lar_ACWA_bl<-subset_samples(ps, Blank_collection_method=="Water September")
lar_ACWA_bl<-prune_taxa(taxa_sums(lar_ACWA_bl) > 0, lar_ACWA_bl)
lar_ACWA_bl #6556 taxa, 42 samples

#Decontam steps for removing floating emergence trap contaminants from ACWA adult mayflies
lar_ACWA_bl_df <- as.data.frame(sample_data(lar_ACWA_bl))

sample_data(lar_ACWA_bl)$is.neg <- sample_data(lar_ACWA_bl)$Sample_Type == "Blank" 
contamdf.prev.lar.ACWA <- isContaminant(lar_ACWA_bl, method="prevalence", neg="is.neg", threshold=0.1)
 
table(contamdf.prev.lar.ACWA$contaminant)
#Threshold of 0.1: FALSE = 6517, TRUE=39

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.lar.ACWA <- prune_taxa(!contamdf.prev.lar.ACWA$contaminant, lar_ACWA_bl)
ps.noncontam.lar.ACWA #6517 taxa and 42 samples


## Adults from Bow River ##

ad_Bow_bl<-subset_samples(ps, Blank_collection_method=="Sheet")
ad_Bow_bl<-prune_taxa(taxa_sums(ad_Bow_bl) > 0, ad_Bow_bl)
ad_Bow_bl #9877 taxa, 107 samples

#Decontam steps for removing floating emergence trap contaminants from ACWA adult mayflies
ad_Bow_bl_df <- as.data.frame(sample_data(ad_Bow_bl))

sample_data(ad_Bow_bl)$is.neg <- sample_data(ad_Bow_bl)$Sample_Type == "Blank" 
contamdf.prev.ad.Bow <- isContaminant(ad_Bow_bl, method="prevalence", neg="is.neg", threshold=0.1)
 
table(contamdf.prev.ad.Bow$contaminant)
#Threshold of 0.1: FALSE = 9552, TRUE=325

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.ad.Bow <- prune_taxa(!contamdf.prev.ad.Bow$contaminant, ad_Bow_bl)
ps.noncontam.ad.Bow #9552 taxa and 107 samples

#Adult caddis at Policeman Flats need to be decontaminated with extraction reagents too since they were on the last extraction plate that was contaminated from the lab
contam.caddis<-subset_samples(ps.noncontam.ad.Bow, Sample_ID %in% c("POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8"))

#Separate extraction reagents so I can combine the POLFLT caddis dataframe together
ext.reag<-subset_samples(ps, Blank_collection_method=="Extraction_neg_4")
#Merge them together
polflt.caddis<-merge_phyloseq(contam.caddis, ext.reag)

sample_data(polflt.caddis)$is.neg <- sample_data(polflt.caddis)$Sample_Type == "Blank"
contamdf.prev.caddis <- isContaminant(polflt.caddis, method="prevalence", neg="is.neg", threshold=0.1)

table(contamdf.prev.caddis$contaminant)
#Threshold of 0.1: FALSE = 31,523, TRUE=33

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.caddis <- prune_taxa(!contamdf.prev.caddis$contaminant, polflt.caddis)
ps.noncontam.caddis #31,523 taxa and 9 samples

ps.noncontam.caddis<-subset_samples(ps.noncontam.caddis, Sample_Type!="Blank")
nsamples(ps.noncontam.caddis) #7

ps.noncontam.ad.Bow <- subset_samples(ps.noncontam.ad.Bow, Sample_ID != "POLFLT_1_Ad_Caddis_MB_2" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_3" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_4" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_5" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_6" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_7" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_8")


## Larvae Bow River ##

lar_Bow_bl<-subset_samples(ps, Blank_collection_method=="Water July")
lar_Bow_bl<-prune_taxa(taxa_sums(lar_Bow_bl) > 0, lar_Bow_bl)
lar_Bow_bl #16688 taxa, 122 samples

#Decontam steps for removing floating emergence trap contaminants from ACWA adult mayflies
lar_Bow_bl_df <- as.data.frame(sample_data(lar_Bow_bl))

sample_data(lar_Bow_bl)$is.neg <- sample_data(lar_Bow_bl)$Sample_Type == "Blank" 
contamdf.prev.lar.Bow <- isContaminant(lar_Bow_bl, method="prevalence", neg="is.neg", threshold=0.1)
 
table(contamdf.prev.lar.Bow$contaminant)
#Threshold of 0.1: FALSE = 16629, TRUE=59

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.lar.Bow <- prune_taxa(!contamdf.prev.lar.Bow$contaminant, lar_Bow_bl)
ps.noncontam.lar.Bow 


## All data merged back together ##

#Rest of data that wasn't decontaminated
rest_of_data<-subset_samples(ps, Blank_collection_method == "No_blank")
rest_of_data<-prune_taxa(taxa_sums(rest_of_data) > 0, rest_of_data)
rest_of_data #4026 taxa, 80 samples (spiders)
#Merge back together
final_merge<-merge_phyloseq(ps.noncontam.ad.mesh, ps.noncontam.ad.Bow, ps.noncontam.lar.ACWA, ps.noncontam.lar.Bow, rest_of_data)

decontam_sample_ps<-subset_samples(final_merge, Sample_Type == "Sample") 

decontam_sample_ps_0 <- subset_taxa(decontam_sample_ps, Kingdom != "Eukaryota")

decontam_sample_ps_1<-subset_taxa(decontam_sample_ps_0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) 

decontam_sample_ps_2<-ps_prune(decontam_sample_ps_1, min.samples = 4, min.reads = 6)
others = c("Others")
allTaxa_decontam = taxa_names(decontam_sample_ps_2)
allTaxa_decontam <- allTaxa_decontam[!(allTaxa_decontam %in% others)] #removes others from taxa names
decontam_sample_ps_2 = prune_taxa(allTaxa_decontam, decontam_sample_ps_2)
decontam_sample_ps_2 #4520, 350 samples

decontam_ps <- subset_samples(decontam_sample_ps_2, sample_sums(decontam_sample_ps_2) > 2000) 
decontam_ps #4520 taxa, 332 samples. 
#Removed 44 potential contaminants from original dataset.
```

### Removing low confidence taxa
```{r}
#Need to remove PMF adult caddis because they were contaminated with contaminated laboratory reagents.
samples_to_remove<-c("POLFLT_1_Ad_Caddis_MB_1", "POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8")

ps_pruned<-subset_samples(ps, !(Sample_ID %in% samples_to_remove))
ps_pruned<-prune_taxa(taxa_sums(ps_pruned) > 0, ps_pruned)
ps_pruned #31,185 taxa, 372 samples


#Subsetting experimental samples only from the decontaminanted phyloseq object
psexp<-subset_samples(ps_pruned, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,218 taxa and 349 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
table(tax_table(ps0)[,"Phylum"],exclude=NULL)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs
ps1 #29,776 taxa, 349 samples
sort(sample_sums(ps1)) #Reads/sample range from 59 - 222,956
```

### Finding a good prevalence threshold
```{r}

##Prevalence threshold pruning ##
#Define prevalence of each taxa (in how many samples did each taxa appear at least once)
#2%
prev0 = apply(X = otu_table(ps1),
                MARGIN = ifelse(taxa_are_rows(ps1), yes = 1, no = 2),
                FUN = function(x){sum(x > 0)})
prevdf = data.frame(Prevalence = prev0,
                      TotalAbundance = taxa_sums(ps1),
                      tax_table(ps1))
keepPhyla = table(prevdf$Phylum)[(table(prevdf$Phylum) > 2)] #Keeping phyla with a prevalence greater than 2
prevdf1 = subset(prevdf, Phylum %in% names(keepPhyla))
# Define prevalence threshold as 2% of total samples
prevalenceThreshold = 0.02 * nsamples(ps1)
prevalenceThreshold #We are keeping phyla that occur at least 6.78 samples (2% of samples) with a prevalence of 2 or greater 

psprune = prune_taxa((prev0 > prevalenceThreshold), ps1)
psprune #2823 taxa. Cut out a very large portion of the data. Probably too high of a threshold.  

reads_pruned <- taxa_sums(psprune)
print(sum(reads_pruned)) #11,202,901 total reads left after pruning
reads_per_OTU <- taxa_sums(psexp) 
(1-(sum(reads_pruned))/(sum(reads_per_OTU)))*100 #14.6% of the total read data was removed by pruning. A decently large portion of the data was cut out from additional pruning

#Graphs prevalence of each bacterial Phylum with dotted line indicating the prevalence threshold so you get an idea of how much data would be lost.
ggplot(prevdf1, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.7) +
  scale_y_log10() + scale_x_log10() +
  xlab("Total Abundance") +
  facet_wrap(~Phylum)
#Prevalence threshold looks too high, cutting out a lot of the variation in microbial communities. I will try a lower threshold value. 


#Lowering the prevalence threshold to 1%

# Define prevalence of each taxa
# (in how many samples did each taxa appear at least once)
prev1 = apply(X = otu_table(ps1),
                MARGIN = ifelse(taxa_are_rows(ps1), yes = 1, no = 2),
                FUN = function(x){sum(x > 0)})
prevdf2 = data.frame(Prevalence = prev1,
                      TotalAbundance = taxa_sums(ps1),
                      tax_table(ps1))
keepPhyla1 = table(prevdf2$Phylum)[(table(prevdf2$Phylum) > 2)]
prevdf3 = subset(prevdf2, Phylum %in% names(keepPhyla))
# Define prevalence threshold as 1% of total samples
prevalenceThreshold1 = 0.01 * nsamples(ps1)
prevalenceThreshold1 #3.49. Present in at least 3.49 samples (round up to 4)

psprune1 = prune_taxa((prev1 > prevalenceThreshold1), ps1)
psprune1 #4572 taxa. Still a very large percentage of taxa removed but we need to look at the amount of total reads removed to understand the implications to the data.

reads_pruned1 <- taxa_sums(psprune1)
print(sum(reads_pruned1)) #11,981,219 total reads left after pruning
reads_per_OTU <- taxa_sums(psexp) 
(1-(sum(reads_pruned1))/(sum(reads_per_OTU)))*100 #8.68% of the total data was removed. Still a decnetly large percentage 

#Plot prevalence of each Phylum
ggplot(prevdf2, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold1, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.7) +
  scale_y_log10() + scale_x_log10() +
  xlab("Total Abundance") +
  facet_wrap(~Phylum)
#This threshold of filtering looks a lot better and I can be more confidence that we are not removing true taxa from samples. Looks like the ones that are removed are just artifacts of sampling or sequencing errors. 

#I will use a prevalence threshold of 1 % meaning that the bacteria has to be present in at least 4 samples 
```

### Filtering by abundance and prevalence threshold
```{r}
prevdf=apply(X=otu_table(ps1),MARGIN=ifelse(taxa_are_rows(ps1),yes=1,no=2),FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf= data.frame(Prevalence= prevdf,TotalAbundance=taxa_sums(ps1),tax_table(ps1))

plyr::ddply(prevdf,"Phylum",function(df1){cbind(mean(df1$Prevalence),sum(df1$TotalAbundance))}) 
#Shows mean and total prevalence of taxa within each phylum. The ones that are only prevalent once or twice in each phylum might be worth removing. This means they only showed up in one or two samples across all samples but might have a high abundance (total number of reads).  

#There are a couple of phyla with only a prevalence sum of one or two so these might be worth removing. 

ps1.dt.taxa = data.table(tax_table(ps1), OTUabundance = taxa_sums(ps1), OTU = taxa_names(ps1)) 
ps1.cumsum = ps1.dt.taxa[, .N, by = OTUabundance]
setkey(ps1.cumsum, OTUabundance)
ps1.cumsum[, CumSum := cumsum(N)]
# Define the plot
ps1.cumsum.plot = ggplot(ps1.cumsum, aes(OTUabundance, CumSum)) + 
  geom_point() +
  xlab("Filtering Threshold, Minimum Total Counts") +
  ylab("OTUs Filtered") +
  ggtitle("OTUs that would be filtered vs. the minimum count threshold") + theme_bw()

print(ps1.cumsum.plot) #Difficult to see the smaller values so we need to zoom in

ps1.cumsum.plot.zoom <- ps1.cumsum.plot + xlim(0, 50) + theme_bw() 

print(ps1.cumsum.plot.zoom) 
#Could maybe remove taxa with an abundance of less than 5-6 across all samples however if I go higher than that it filters out a ton of taxa (about 1/3 of all of them)

#I will filter out taxa that are only prevalent in 1% of all samples (has to be in at least 4/357 samples) and at least 6 reads across all samples
ps2<-ps_prune(ps1, min.samples = 4, min.reads = 6) #Creates an other category that needs to be removed
others = c("Others")
allTaxa = taxa_names(ps2)
allTaxa <- allTaxa[!(allTaxa %in% others)] #removes others from taxa names
ps2 = prune_taxa(allTaxa, ps2)
ps2 #4563, 349 samples

any(taxa_sums(ps2) <6) #FALSE. Means it did a good job of getting rid of them. 
sort(sample_sums(ps2))
#Should remove samples with less than 2000 reads. Will remove 14 samples.
```

### Filtering by a minimum total sample count (at least 2000 reads/sample)
```{r}
#Creating a rarefaction curve to see the spread in the sequencing depths
tab <- otu_table(ps2)
class(tab) <- "matrix"
tab <- t(tab) # transpose observations to rows
rare <- rarecurve(tab, step=5000, lwd=2, ylab="OTU",  label=F) 
#The rarefaction curves are very spread out. This means that there are different sequencing depths and standardizing the data might be needed prior to analysis. 

sort(sample_sums(ps2)) #minimum is 24, maximum is 222,630
mean(sample_sums(ps2)) #34330

#Final dataset
sample_ps <- subset_samples(ps2, sample_sums(ps2) > 2000) 
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2384 reads. 
sample_ps #4563 taxa, 335 samples. 

#Create excel file with samples that were subset out
pruned_df<-psmelt(subset_samples(ps2, sample_sums(ps2) < 2000))

#write.csv(pruned_df, file="microbiome_analysis/Samples pruned from 2000 reads per sample.csv")
```

### Amount of data that was filtered out from the original phyloseq object
```{r}
ntaxa(psexp)-ntaxa(sample_ps) #filtered 25,655 total taxa (ASVs)
(1-ntaxa(sample_ps)/ntaxa(psexp))*100 #84.9% of the taxa were removed from the original decontaminated dataset. 

reads_per_OTU <- taxa_sums(psexp) #Original ps with no filtering done.
print(sum(reads_per_OTU)) #Total number of reads is 13,120,090.

reads_per_OTU_filtered <- taxa_sums(sample_ps) #Original ps with no filtering done.
print(sum(reads_per_OTU_filtered)) #Total number of reads is now 11,966,724.

(1-11966724/13120090)*100 #8.79%. Filtered out a decently small percentage of the total data (reads). 


#Creating a distribution of reads per sample versus the sample counts. 
ps1_df= data.table(as(sample_data(sample_ps), "data.frame"), Reads_per_sample = sample_sums(sample_ps), keep.rownames = TRUE)
ps1_df_plot = ggplot(ps1_df, aes(Reads_per_sample)) + geom_histogram() + ggtitle("Distribution of reads per sample") + ylab("Sample counts") 

print(ps1_df_plot) #Looks right skewed. A couple of very large reads per sample. Probably the Perlidae individuals because they are so big compared to some of the other samples like chironomids. Most are around 20,000 - 40,000 reads per sample though.

#Now we need to observe the counts of abundance of unique sequences across samples
ps1.dt.taxa = data.table(tax_table(sample_ps), OTUabundance = taxa_sums(sample_ps), OTU = taxa_names(sample_ps)) 
#makes a data table of the sum of abundance of each taxa vs the total number of taxa (ASVs)
ps1.dt.tax.plot<- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram() + ggtitle("Histogram of OTU (unique sequence) counts") + theme_bw() #Plots the number of times an ASV is repeated from the reads (abundance) on the x axis and the number of taxa on the y axis
print(ps1.dt.tax.plot)
#You can"t see much on the plots panel in R. Save it as an image to see more. 

#write.table(ps1.dt.taxa, "Filtering phyloseq/ps1_dt_tax_plot.txt", sep = "\t") #makes a table that can open in excel

#ggsave("Filtering phyloseq/Histogram of OTU counts.pdf", height = 6, width = 7) #pdf of the plot

#This plot shows that there is a very small proportion of ASVs with a really high abundance and most have relatively low abundance. Highly right skewed. We want to see how many have very low abundance because this may indicate that further processing/filtering should be done. 

#Zoom in on the lower abundance counts to get a better idea of its distribution
plot.zoom1 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 100, by =10)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom1)
#ggsave("Filtering phyloseq/Histogram of OTU counts 1 to 100.pdf", height = 6, width = 7)
#A very high proportion of ASVs have an abundance of less than 10. This may indicate that they should be filtered out. 

plot.zoom2 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 10, by =2)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom2)
#ggsave("Filtering phyloseq/Histogram of OTU counts 1 to 10.pdf", height = 6, width = 7)
```

### Simplified filtering steps (experimental and blanks) for reloading data quickly
```{r}
#Need to remove PMF adult caddis because they were contaminated with contaminated laboratory reagents.
samples_to_remove<-c("POLFLT_1_Ad_Caddis_MB_1", "POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8")

ps_pruned<-subset_samples(ps, !(Sample_ID %in% samples_to_remove))
ps_pruned<-prune_taxa(taxa_sums(ps_pruned) > 0, ps_pruned)
ps_pruned #31,185 taxa, 372 samples

#Removing low confidence data
psexp<-subset_samples(ps_pruned, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,218 taxa and 349 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs

ps2<-ps_prune(ps1, min.samples = 4, min.reads = 6) # Remove ASVs with reads less than 6 across all samples or less than 4 within a sample. ps_prune from MicEco package. It groups all taxa that don't fit into this category into an "others" category. Need to remove that. 
others = c("Others")
allTaxa = taxa_names(ps2)
allTaxa <- allTaxa[!(allTaxa %in% others)] #removes others from taxa names
ps3 = prune_taxa(allTaxa, ps2)
# new phyloseq object with just the taxa you kept.
ps3 #4563, 349 samples

any(taxa_sums(ps3) <6) #FALSE. Means it did a good job of getting rid of them. 

sample_ps <- subset_samples(ps3, sample_sums(ps3) > 2000) 
sample_ps<-prune_taxa(taxa_sums(sample_ps) > 0, sample_ps)
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2384 reads.
sample_ps #4563 taxa, 335 samples.
```

### Re-frame filtered data to work as a data frame
```{r}
#Metadata (from filtered phyloseq) back into a useable data frame

sample_data_ps<-as.data.frame(sample_data(sample_ps))
row_names<-row.names(sample_data_ps) 
sample_data_ps_1<-cbind(row_names, sample_data_ps)
sample_data_ps_1<-sample_data_ps_1 %>% dplyr::rename(Study_ID = row_names)
sample_data_ps_2<-sample_data_ps_1[ , c("Study_ID",  "Site", "Family", "Stream_Type")]
sample_data_3<-sample_data_ps_2 %>% dplyr::rename(sample_Family = Family)
sample_data_4<-sample_data_3 %>% drop_na(Site, sample_Family)


#ASV table from filtered phyloseq
ASVseq_ps<- as.data.frame(otu_table(sample_ps))
Sequences<-row.names(ASVseq_ps) 
ASVseq_ps_1<-cbind(Sequences, ASVseq_ps)
 
ASVseq_ps_2<-ASVseq_ps_1 %>% pivot_longer(-Sequences, names_to="Study_ID", values_to = "count")

#Taxonomy table
taxonomy_modified<-as.data.frame(tax_table(sample_ps)) %>% rownames_to_column(var="Sequences")

nseqs_per_sample <- ASVseq_ps_2 %>%
  dplyr::group_by(Study_ID) %>%
  dplyr::summarize(N = sum(count), .groups="drop") %>% 
  dplyr::count(N) %>%
  dplyr::pull(N)

lod <- 100* 1/max(nseqs_per_sample) #Determines the limit of detection (0.000449)
```

# Data diagnostics

In this section I filter out different taxonomic organizations of bacteria to see their abundance at different levels of organization. Ex. how many phyla per site, what are the top phyla, families, genera and their percentage of the total. 

## Blank Samples
```{r}
blank_ps<-subset_samples(ps, Sample_Type=="Blank")
blank_ps<-prune_taxa(taxa_sums(blank_ps) > 0, blank_ps)
nsamples(blank_ps) #23
ntaxa(blank_ps) #1809

sum(sample_sums(blank_ps)) #503,506 total reads
mean(sample_sums(blank_ps)) #21,891 average reads/sample
sort(sample_sums(blank_ps)) #Reads/sample ranges from 16 - 63,500 counts

#The extraction reagent blanks from the first 4 plates are relatively low (16 - 629 counts)
#The extraction reagent blanks from the last plate are very high from lab contamination (44,645 and 57,629 counts)
#Buffer tube blanks were run on the last (contaminated) plate and have counts of 56,500 and 63,500 
#Field blanks range from 7,949 - 49,817 counts. Could have high counts due to dead bacteria from the tweezers or bug sheet being picked up and amplified in PCR. 

#Total abundance of each blank sample
blank_ps_glom<-tax_glom(blank_ps, taxrank="Phylum")
plot_bar(blank_ps_glom, fill="Phylum", "Blank_Type", "Abundance") + theme(legend.position = "bottom") + theme_bw() + rremove("grid")

#Relative abundance of top 5 most abundant classification of bacteria. 
#Percent of each phyla to the total bacteria
sort((table(tax_table(blank_ps)[,2]))/ntaxa(blank_ps)*100)
#Firmicutes (42.3 %), Proteobacteria (19.6 %), Bacteroidota (13.4 %), Actinobacteriota (11.0 %), Cyanobacteria (5.25 %)

#Percent of each class to the total bacteria
sort((table(tax_table(blank_ps)[,3]))/ntaxa(blank_ps)*100)
#Clostridia (35.2 %), Bacteroidia (13.3 %), Alphaproteobacteria (10.1)

#Percent of each order to the total bacteria
sort((table(tax_table(blank_ps)[,4]))/ntaxa(blank_ps)*100)
#Lachnospirales (15.0 %), Oscillospirales (14.2 %), Burkholderiales (6.5 %)

#Percent of each family to the total bacteria
sort((table(tax_table(blank_ps)[,5]))/ntaxa(blank_ps)*100)
#Lachnospiraceae (14.9 %), Ruminococcaceae (6.9 %), Oscillospiraceae (5.5 %)

#Percent of each genus to the total bacteria
sort((table(tax_table(blank_ps)[,6]))/ntaxa(blank_ps)*100)
#Faecalibacterium (1.99 %), [Ruminococcus] torques group (1.82 %), Blautia (1.60 %), Bacteroides (1.55 %), Alistipes (1.55 %)

#Percent of each species to the total bacteria
sort((table(tax_table(blank_ps)[,7]))/ntaxa(blank_ps)*100)
#prausnitzii (1.0 %), bacterium (0.44%), hadrus (0.39%)

## Get total numbers of ASVs at each level
get_taxa_unique(blank_ps, "Phylum") #26 phyla total
get_taxa_unique(blank_ps, "Class") #57 Classes total
get_taxa_unique(blank_ps, "Order") #136 orders total
get_taxa_unique(blank_ps, "Family") #199 families total
get_taxa_unique(blank_ps, "Genus") #400 genera total

# Get total number of ASVs within each blank type

## Water July
water_july<-subset_samples(blank_ps, Blank_collection_method=="Water July")
water_july<-prune_taxa(taxa_sums(water_july)>0, water_july)
water_july #153 taxa, 3 samples

get_taxa_unique(water_july, "Phylum") #11 phyla total
get_taxa_unique(water_july, "Class") #21 Classes total
get_taxa_unique(water_july, "Order") #54 orders total
get_taxa_unique(water_july, "Family") #63 families total
get_taxa_unique(water_july, "Genus") #76 genera total

## Water September

water_sept<-subset_samples(blank_ps, Blank_collection_method=="Water September")
water_sept<-prune_taxa(taxa_sums(water_sept)>0, water_sept)
water_sept #286 taxa, 3 samples

get_taxa_unique(water_sept, "Phylum") #19 phyla total
get_taxa_unique(water_sept, "Class") #32 Classes total
get_taxa_unique(water_sept, "Order") #73 orders total
get_taxa_unique(water_sept, "Family") #95 families total
get_taxa_unique(water_sept, "Genus") #126 genera total

## Bedsheet July

bedsheet<-subset_samples(blank_ps, Blank_collection_method=="Sheet")
bedsheet<-prune_taxa(taxa_sums(bedsheet)>0, bedsheet)
bedsheet #749 taxa, 4 samples

get_taxa_unique(bedsheet, "Phylum") #17 phyla total
get_taxa_unique(bedsheet, "Class") #37 Classes total
get_taxa_unique(bedsheet, "Order") #92 orders total
get_taxa_unique(bedsheet, "Family") #133 families total
get_taxa_unique(bedsheet, "Genus") #249 genera total

## Floating emergence traps September

mesh_trap<-subset_samples(blank_ps, Blank_collection_method=="Mesh")
mesh_trap<-prune_taxa(taxa_sums(mesh_trap)>0, mesh_trap)
mesh_trap #242 taxa, 3 samples

get_taxa_unique(mesh_trap, "Phylum") #18 phyla total
get_taxa_unique(mesh_trap, "Class") #34 Classes total
get_taxa_unique(mesh_trap, "Order") #63 orders total
get_taxa_unique(mesh_trap, "Family") #86 families total
get_taxa_unique(mesh_trap, "Genus") #101 genera total

## DNA extraction reagents (plate 4)

DNAext4<-subset_samples(blank_ps, Blank_collection_method=="Extraction_neg_4")
DNAext4<-prune_taxa(taxa_sums(DNAext4)>0, DNAext4)
DNAext4 #382 taxa, 2 samples

get_taxa_unique(DNAext4, "Phylum") #9 phyla total
get_taxa_unique(DNAext4, "Class") #13 Classes total
get_taxa_unique(DNAext4, "Order") #31 orders total
get_taxa_unique(DNAext4, "Family") #45 families total
get_taxa_unique(DNAext4, "Genus") #92 genera total

## DNA extraction reagents (plates 1-3)

DNAext<-subset_samples(blank_ps, Blank_collection_method=="Extraction_neg")
DNAext<-prune_taxa(taxa_sums(DNAext)>0, DNAext)
DNAext #23 taxa, 6 samples

get_taxa_unique(DNAext, "Phylum") #3 phyla total
get_taxa_unique(DNAext, "Class") #5 Classes total
get_taxa_unique(DNAext, "Order") #10 orders total
get_taxa_unique(DNAext, "Family") #10 families total
get_taxa_unique(DNAext, "Genus") #15 genera total

## Lysis Buffer tubes

buffer<-subset_samples(blank_ps, Blank_collection_method=="Buffer")
buffer<-prune_taxa(taxa_sums(buffer)>0, buffer)
buffer #447 taxa, 2 samples

get_taxa_unique(buffer, "Phylum") #11 phyla total
get_taxa_unique(buffer, "Class") #17 Classes total
get_taxa_unique(buffer, "Order") #48 orders total
get_taxa_unique(buffer, "Family") #64 families total
get_taxa_unique(buffer, "Genus") #114 genera total


## Plotting composition of blanks
#Phylum 
blank_phy <- tax_glom(blank_ps, taxrank="Phylum")
blank_rel_phy <- microbiome::transform(blank_phy, "compositional")

blank_rel_phy_df <- psmelt(blank_rel_phy)

blank_phy_agg <- aggregate(Abundance ~ Phylum + Blank_Type, data = blank_rel_phy_df, FUN = mean) 

blank_phy_plot <- ggplot(blank_phy_agg, aes(x=Blank_Type, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  scale_fill_manual(name="Bacterial Class", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "#43464B", "lightcoral", "palevioletred2", brewer.pal(10, "Set3")))+
  rremove("grid")
blank_phy_plot

#Family
blank_fam <- tax_glom(blank_ps, taxrank="Family")
blank_rel_fam <- microbiome::transform(blank_fam, "compositional")
blank_rel_fam_prune = prune_taxa(taxa_sums(blank_rel_fam) > 0.05, blank_rel_fam)

blank_rel_fam_df <- psmelt(blank_rel_fam_prune)

blank_fam_agg <- aggregate(Abundance ~ Family + Blank_Type, data = blank_rel_fam_df, FUN = mean) 

blank_fam_plot <- ggplot(blank_fam_agg, aes(x=Blank_Type, y=Abundance, fill=Family)) + geom_bar(stat="identity")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
blank_fam_plot

#Genus
blank_genus <- tax_glom(blank_ps, taxrank="Genus")
blank_rel_genus <- microbiome::transform(blank_genus, "compositional")
blank_rel_genus_prune = prune_taxa(taxa_sums(blank_rel_genus) > 0.1, blank_rel_genus)

blank_rel_genus_df <- psmelt(blank_rel_genus_prune)

blank_genus_agg <- aggregate(Abundance ~ Genus + Blank_Type, data = blank_rel_genus_df, FUN = mean) 

blank_genus_plot <- ggplot(blank_genus_agg, aes(x=Blank_Type, y=Abundance, fill=Genus)) + geom_bar(stat="identity")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
blank_genus_plot
```

## Experimental samples {.tabset}

### Relative abundance at different taxonomic levels (all samples combined)
```{r}
#### Bow River ####

Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
Bow_ps<-prune_taxa(taxa_sums(Bow_ps) > 0, Bow_ps)
Bow_ps #4191 taxa, 280 samples

#Number of larval samples
lar_Bow<-subset_samples(Bow_ps, Stage=="Larvae")
nsamples(lar_Bow) #111
#Number of adult samples
ad_Bow<-subset_samples(Bow_ps, Stage=="Adults")
nsamples(ad_Bow) #92
#Number of spider samples
spider_Bow<-subset_samples(Bow_ps, Stage=="Spiders")
nsamples(spider_Bow) #77

#Total number of reads
print(sum(sample_sums(Bow_ps))) #9,816,526  
#Average # of reads per sample
mean(sample_sums(Bow_ps)) #35,059 average reads/sample
sort(sample_sums(Bow_ps)) #reads/sample ranges from 2384 - 222,630 counts
ntaxa(Bow_ps) #4191

#All phyla combined
sort((table(tax_table(Bow_ps)[,2]))/sum(ntaxa(Bow_ps))*100)
#Top 5 abundant phyla are Proteobacteria (37.6 %), Bacteroidota (30.7 %), Firmicutes (11.2 %), Cyanobacteria (5.92 %), Actinobacteria (5.65 %)
sort((table(tax_table(Bow_ps)[,5])/sum(ntaxa(Bow_ps))*100))
#Top 5 families are Comamonadaceae (7.83 %), Saprospiraceae (6.28 %), Flavobacteriaceae (4.58 %), Rhodobacteraceae (4.29 %), and Sphingomonadaceae (3.79 %)
sort((table(tax_table(Bow_ps)[,6])/ntaxa(Bow_ps)*100))
#top 5 genera: Flavobacterium (4.34 %), Rhodoferax (2.29 %), Tyzzerella (1.50 %), Pseudorhodobacter (1.41), Limnohabitans (1.10 %)

#Sort by proteobacteria
proteo_Bow <- subset_taxa(Bow_ps, Phylum=="Proteobacteria")
ntaxa(proteo_Bow) #1578
(sort(table(tax_table(proteo_Bow)[,3])/ntaxa(proteo_Bow)*100)) #class. Only 2 classes. Alphaproteobacteria (49.7%) and Gammaproteobacteria (50.3%)
(sort(table(tax_table(proteo_Bow)[,5])/ntaxa(proteo_Bow)*100)) #family Top 3: Comamonadaceae (20.8 %), Rhodobacteraceae (11.4 %), Sphingomonadaceae (10.1 %) 
(sort(table(tax_table(proteo_Bow)[,6])/ntaxa(proteo_Bow)*100)) #genera Top 3: Rhodoferax (6.08 %), Pseudorhodobacter (3.74 %), Limnohabitans (2.92 %)

#Sort by Bacteroidota
bactero_Bow <- subset_taxa(Bow_ps, Phylum=="Bacteroidota")
ntaxa(bactero_Bow) #1288
(sort(table(tax_table(bactero_Bow)[,3])/ntaxa(bactero_Bow)*100)) #class. Top class: Bacteroidia (97.4 %)
(sort(table(tax_table(bactero_Bow)[,5])/ntaxa(bactero_Bow)*100)) #family Top 3: Saprospiraceae (20.4 %), Flavobacteriaceae (14.9 %), Chitinophagaceae (10.8 %)  
(sort(table(tax_table(bactero_Bow)[,6])/ntaxa(bactero_Bow)*100)) #genera Top 3: Flavobacterium (14.1 %), Hymenobacter (3.11 %), Ferruginibacter (2.87 %)

#Sort by Firmicutes
firm_Bow <- subset_taxa(Bow_ps, Phylum=="Firmicutes")
ntaxa(firm_Bow) #468
(sort(table(tax_table(firm_Bow)[,3])/ntaxa(firm_Bow)*100)) #class. Top 2: Clostridia (67.5 %), Bacilli (29.7 %) 
(sort(table(tax_table(firm_Bow)[,5])/ntaxa(firm_Bow)*100)) #family Top 2: Lachnospiraceae (28.6 %), Ruminococcaceae (15.4 %) 
(sort(table(tax_table(firm_Bow)[,6])/ntaxa(firm_Bow)*100)) #genera Top 3: Tyzzerella (13.5 %), Candidatus Soleaferrea (4.06 %), Staphylococcus (3.63 %)


##Get total numbers of ASVs at each level
get_taxa_unique(Bow_ps, "Phylum") #32 phyla total
get_taxa_unique(Bow_ps, "Class") #68 Classes total
get_taxa_unique(Bow_ps, "Order") #166 orders total
get_taxa_unique(Bow_ps, "Family") #237 families total
get_taxa_unique(Bow_ps, "Genus") #480 genera total


#### ACWA ####

ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")
ACWA_ps<-prune_taxa(taxa_sums(ACWA_ps) > 0, ACWA_ps)
ACWA_ps #2604 taxa, 55 samples

lar_ACWA<-subset_samples(ACWA_ps, Stage=="Larvae")
nsamples(lar_ACWA) #39
ad_ACWA<-subset_samples(ACWA_ps, Stage=="Adults")
nsamples(ad_ACWA) #16

#Total number of reads
print(sum(sample_sums(ACWA_ps))) #2,150,198
#Average # of reads per sample
mean(sample_sums(ACWA_ps)) #39,094 average reads/sample
sort(sample_sums(ACWA_ps)) #reads/sample ranges from 8978 - 88,543 counts
ntaxa(ACWA_ps) #2604

#All phyla combined
sort((table(tax_table(ACWA_ps)[,2]))/sum(ntaxa(ACWA_ps))*100)
#Top 5 abundant phyla are Proteobacteria (35.8 %), Bacteroidota (31.6 %), Firmicutes (10.2 %), Cyanobacteria (5.68 %), Actinobacteria (3.76 %)
sort((table(tax_table(ACWA_ps)[,5])/sum(ntaxa(ACWA_ps))*100))
#Top 5 families are Comamonadaceae (7.52 %), Saprospiraceae (6.99 %), Rhodobacteraceae (5.07 %), Flavobacteriaceae (3.99 %), and Spirosomaceae (3.30 %)
sort((table(tax_table(ACWA_ps)[,6])/ntaxa(ACWA_ps)*100))
#top 5 genera: Flavobacterium (3.88 %), Tyzzerella (2.23 %), Pseudorhodobacter (1.65), Rhodoferax (1.50), Mucinivorans (1.19 %)

#Sort by proteobacteria
proteo_ACWA <- subset_taxa(ACWA_ps, Phylum=="Proteobacteria")
ntaxa(proteo_ACWA) #933
(sort(table(tax_table(proteo_ACWA)[,3])/ntaxa(proteo_ACWA)*100)) #class. Only 2 classes. Alphaproteobacteria (49.2%) and Gammaproteobacteria (50.6%)
(sort(table(tax_table(proteo_ACWA)[,5])/ntaxa(proteo_ACWA)*100)) #family Top 3: Comamonadaceae (21.0 %), Rhodobacteraceae (14.1 %), Sphingomonadaceae (8.57 %) 
(sort(table(tax_table(proteo_ACWA)[,6])/ntaxa(proteo_ACWA)*100)) #genera Top 3: Pseudorhodobacter (4.61), Rhodoferax (4.18 %), Limnohabitans (3.32 %)

#Sort by Bacteroidota
bactero_ACWA <- subset_taxa(ACWA_ps, Phylum=="Bacteroidota")
ntaxa(bactero_ACWA) #824
(sort(table(tax_table(bactero_ACWA)[,3])/ntaxa(bactero_ACWA)*100)) #class. Top class: Bacteroidia (97.1 %)
(sort(table(tax_table(bactero_ACWA)[,5])/ntaxa(bactero_ACWA)*100)) #family Top 3: Saprospiraceae (22.1 %), Flavobacteriaceae (12.6 %), Spirosomaceae (10.4 %)  
(sort(table(tax_table(bactero_ACWA)[,6])/ntaxa(bactero_ACWA)*100)) #genera Top 3: Flavobacterium (12.3 %), Mucinivorans (3.76 %), Emticicia (2.67 %)

#Sort by Firmicutes
firm_ACWA <- subset_taxa(ACWA_ps, Phylum=="Firmicutes")
ntaxa(firm_ACWA) #265
(sort(table(tax_table(firm_ACWA)[,3])/ntaxa(firm_ACWA)*100)) #class. Top 2: Clostridia (76.2 %), Bacilli (20.8 %) 
(sort(table(tax_table(firm_ACWA)[,5])/ntaxa(firm_ACWA)*100)) #family Top 2: Lachnospiraceae (29.8 %), Ruminococcaceae (20.0 %) 
(sort(table(tax_table(firm_ACWA)[,6])/ntaxa(firm_ACWA)*100)) #genera Top 3: Tyzzerella (21.9 %), Candidatus Soleaferrea (5.66 %), Christensenellaceae R-7 group (3.40 %)


##Get total numbers of ASVs at each level
get_taxa_unique(ACWA_ps, "Phylum") #27 phyla total
get_taxa_unique(ACWA_ps, "Class") #60 Classes total
get_taxa_unique(ACWA_ps, "Order") #144 orders total
get_taxa_unique(ACWA_ps, "Family") #196 families total
get_taxa_unique(ACWA_ps, "Genus") #338 genera total
```

### Total ASVs per site and invertebrate taxa 

```{r}
#Total samples
nsamples(sample_ps) #335
ntaxa(sample_ps) #4,563

#### Bow River ####
Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")

#Cochrane, n = 65
coch<-subset_samples(Bow_ps, Site=="Cochrane")
coch<-prune_taxa(taxa_sums(coch) > 0, coch)
nsamples(coch) #65
ntaxa(coch) #2467

get_taxa_unique(coch, "Phylum") #25 phyla total
get_taxa_unique(coch, "Class") #53 Classes total
get_taxa_unique(coch, "Order") #127 orders total
get_taxa_unique(coch, "Family") #185 families total
get_taxa_unique(coch, "Genus") #362 genera total

sort((table(tax_table(coch)[,2]))/ntaxa(coch)*100)
#Proteobacteria (38.8%), Bacteroidota (30.8%), Firmicutes (13.5%).

#Hydropsychidae
hydro_coch<-subset_samples(coch, Family=="Hydropsychidae")
hydro_coch<-prune_taxa(taxa_sums(hydro_coch)>0, hydro_coch)
ntaxa(hydro_coch) #708
nsamples(hydro_coch) #8

get_taxa_unique(hydro_coch, "Phylum") #16 phyla total
get_taxa_unique(hydro_coch, "Class") #29 Classes total
get_taxa_unique(hydro_coch, "Order") #59 orders total
get_taxa_unique(hydro_coch, "Family") #80 families total
get_taxa_unique(hydro_coch, "Genus") #99 genera total

#Heptageniidae
hep_coch<-subset_samples(coch, Family=="Heptageniidae")
hep_coch<-prune_taxa(taxa_sums(hep_coch)>0, hep_coch)
ntaxa(hep_coch) #1108
nsamples(hep_coch) #8

get_taxa_unique(hep_coch, "Phylum") #19 phyla total
get_taxa_unique(hep_coch, "Class") #40 Classes total
get_taxa_unique(hep_coch, "Order") #85 orders total
get_taxa_unique(hep_coch, "Family") #125 families total
get_taxa_unique(hep_coch, "Genus") #198 genera total

#Chironomidae
chiro_coch<-subset_samples(coch, Family=="Chironomidae")
chiro_coch<-prune_taxa(taxa_sums(chiro_coch)>0, chiro_coch)
ntaxa(chiro_coch) #323
nsamples(chiro_coch) #4

get_taxa_unique(chiro_coch, "Phylum") #9 phyla total
get_taxa_unique(chiro_coch, "Class") #24 Classes total
get_taxa_unique(chiro_coch, "Order") #66 orders total
get_taxa_unique(chiro_coch, "Family") #88 families total
get_taxa_unique(chiro_coch, "Genus") #114 genera total

#Perlidae
perl_coch<-subset_samples(coch, Family=="Perlidae")
perl_coch<-prune_taxa(taxa_sums(perl_coch)>0, perl_coch)
ntaxa(perl_coch) #570
nsamples(perl_coch) #8

get_taxa_unique(perl_coch, "Phylum") #12 phyla total
get_taxa_unique(perl_coch, "Class") #20 Classes total
get_taxa_unique(perl_coch, "Order") #49 orders total
get_taxa_unique(perl_coch, "Family") #79 families total
get_taxa_unique(perl_coch, "Genus") #107 genera total

#Trichoptera 
trichop_coch<-subset_samples(coch, Family=="Trichoptera")
trichop_coch<-prune_taxa(taxa_sums(trichop_coch)>0, trichop_coch)
ntaxa(trichop_coch) #645
nsamples(trichop_coch) #8

get_taxa_unique(trichop_coch, "Phylum") #13 phyla total
get_taxa_unique(trichop_coch, "Class") #23 Classes total
get_taxa_unique(trichop_coch, "Order") #60 orders total
get_taxa_unique(trichop_coch, "Family") #93 families total
get_taxa_unique(trichop_coch, "Genus") #168 genera total

#Ephemeroptera 
ephem_coch<-subset_samples(coch, Family=="Ephemeroptera")
ephem_coch<-prune_taxa(taxa_sums(ephem_coch)>0, ephem_coch)
ntaxa(ephem_coch) #230
nsamples(ephem_coch) #8

get_taxa_unique(ephem_coch, "Phylum") #10 phyla total
get_taxa_unique(ephem_coch, "Class") #15 Classes total
get_taxa_unique(ephem_coch, "Order") #49 orders total
get_taxa_unique(ephem_coch, "Family") #80 families total
get_taxa_unique(ephem_coch, "Genus") #107 genera total

#Diptera 
dip_coch<-subset_samples(coch, Family=="Diptera")
dip_coch<-prune_taxa(taxa_sums(dip_coch)>0, dip_coch)
ntaxa(dip_coch) #192
nsamples(dip_coch) #5

get_taxa_unique(dip_coch, "Phylum") #6 phyla total
get_taxa_unique(dip_coch, "Class") #10 Classes total
get_taxa_unique(dip_coch, "Order") #38 orders total
get_taxa_unique(dip_coch, "Family") #54 families total
get_taxa_unique(dip_coch, "Genus") #88 genera total

#Araneidae
aran_coch<-subset_samples(coch, Family=="Araneidae")
aran_coch<-prune_taxa(taxa_sums(aran_coch)>0, aran_coch)
ntaxa(aran_coch) #266
nsamples(aran_coch) #8

get_taxa_unique(aran_coch, "Phylum") #8 phyla total
get_taxa_unique(aran_coch, "Class") #12 Classes total
get_taxa_unique(aran_coch, "Order") #41 orders total
get_taxa_unique(aran_coch, "Family") #66 families total
get_taxa_unique(aran_coch, "Genus") #109 genera total

#Tetragnathidae 
tetra_coch<-subset_samples(coch, Family=="Tetragnathidae")
tetra_coch<-prune_taxa(taxa_sums(tetra_coch)>0, tetra_coch)
ntaxa(tetra_coch) #174
nsamples(tetra_coch) #8

get_taxa_unique(tetra_coch, "Phylum") #7 phyla total
get_taxa_unique(tetra_coch, "Class") #11 Classes total
get_taxa_unique(tetra_coch, "Order") #37 orders total
get_taxa_unique(tetra_coch, "Family") #61 families total
get_taxa_unique(tetra_coch, "Genus") #92 genera total

#Sunalta, n = 47
sun<-subset_samples(sample_ps, Site=="Sunalta")
sun<-prune_taxa(taxa_sums(sun) > 0, sun)
ntaxa(sun) #2173
nsamples(sun) #47

get_taxa_unique(sun, "Phylum") #25 phyla total
get_taxa_unique(sun, "Class") #54 Classes total
get_taxa_unique(sun, "Order") #120 orders total
get_taxa_unique(sun, "Family") #178 families total
get_taxa_unique(sun, "Genus") #331 genera total

sort((table(tax_table(sun)[,2]))/ntaxa(sun)*100)
#Proteobacteria (38.1%), Bacteroidota (30.1%), Firmicutes (13.5%)

#Hydropsychidae 
hydro_sun<-subset_samples(sun, Family=="Hydropsychidae")
hydro_sun<-prune_taxa(taxa_sums(hydro_sun) > 0, hydro_sun)
ntaxa(hydro_sun) #603
nsamples(hydro_sun) #7

get_taxa_unique(hydro_sun, "Phylum") #17 phyla total
get_taxa_unique(hydro_sun, "Class") #26 Classes total
get_taxa_unique(hydro_sun, "Order") #48 orders total
get_taxa_unique(hydro_sun, "Family") #61 families total
get_taxa_unique(hydro_sun, "Genus") #79 genera total

#Heptageniidae 
hep_sun<-subset_samples(sun, Family=="Heptageniidae")
hep_sun<-prune_taxa(taxa_sums(hep_sun) > 0, hep_sun)
ntaxa(hep_sun) #1245
nsamples(hep_sun) #8

get_taxa_unique(hep_sun, "Phylum") #19 phyla total
get_taxa_unique(hep_sun, "Class") #42 Classes total
get_taxa_unique(hep_sun, "Order") #94 orders total
get_taxa_unique(hep_sun, "Family") #128 families total
get_taxa_unique(hep_sun, "Genus") #207 genera total

#Trichoptera 
trichop_sun<-subset_samples(sun, Family=="Trichoptera")
trichop_sun<-prune_taxa(taxa_sums(trichop_sun) > 0, trichop_sun)
ntaxa(trichop_sun) #427
nsamples(trichop_sun) #8

get_taxa_unique(trichop_sun, "Phylum") #11 phyla total
get_taxa_unique(trichop_sun, "Class") #18 Classes total
get_taxa_unique(trichop_sun, "Order") #53 orders total
get_taxa_unique(trichop_sun, "Family") #95 families total
get_taxa_unique(trichop_sun, "Genus") #152 genera total

#Ephemeroptera 
ephem_sun<-subset_samples(sun, Family=="Ephemeroptera")
ephem_sun<-prune_taxa(taxa_sums(ephem_sun) > 0, ephem_sun)
ntaxa(ephem_sun) #339
nsamples(ephem_sun) #8

get_taxa_unique(ephem_sun, "Phylum") #8 phyla total
get_taxa_unique(ephem_sun, "Class") #15 Classes total
get_taxa_unique(ephem_sun, "Order") #43 orders total
get_taxa_unique(ephem_sun, "Family") #87 families total
get_taxa_unique(ephem_sun, "Genus") #126 genera total

#Araneidae
aran_sun<-subset_samples(sun, Family=="Araneidae")
aran_sun<-prune_taxa(taxa_sums(aran_sun) > 0, aran_sun)
ntaxa(aran_sun) #341
nsamples(aran_sun) #8

get_taxa_unique(aran_sun, "Phylum") #8 phyla total
get_taxa_unique(aran_sun, "Class") #15 Classes total
get_taxa_unique(aran_sun, "Order") #49 orders total
get_taxa_unique(aran_sun, "Family") #72 families total
get_taxa_unique(aran_sun, "Genus") #131 genera total

#Tetragnathidae
tetra_sun<-subset_samples(sun, Family=="Tetragnathidae")
tetra_sun<-prune_taxa(taxa_sums(tetra_sun) > 0, tetra_sun)
ntaxa(tetra_sun) #57
nsamples(tetra_sun) #8

get_taxa_unique(tetra_sun, "Phylum") #5 phyla total
get_taxa_unique(tetra_sun, "Class") #7 Classes total
get_taxa_unique(tetra_sun, "Order") #18 orders total
get_taxa_unique(tetra_sun, "Family") #23 families total
get_taxa_unique(tetra_sun, "Genus") #25 genera total

#Cushing Bridge, n = 63
cushbr<-subset_samples(sample_ps, Site=="Cushing Bridge")
cushbr<-prune_taxa(taxa_sums(cushbr) > 0, cushbr)
ntaxa(cushbr) #2761
nsamples(cushbr) #63

get_taxa_unique(cushbr, "Phylum") #29 phyla total
get_taxa_unique(cushbr, "Class") #57 Classes total
get_taxa_unique(cushbr, "Order") #139 orders total
get_taxa_unique(cushbr, "Family") #202 families total
get_taxa_unique(cushbr, "Genus") #399 genera total

sort((table(tax_table(cushbr)[,2]))/ntaxa(cushbr)*100)
#Proteobacteria (40.5%), Bacteroidota (28.6%), Firmicutes (11.5%)

#Hydropsychidae
hydro_cush<-subset_samples(cushbr, Family=="Hydropsychidae")
hydro_cush<-prune_taxa(taxa_sums(hydro_cush) > 0, hydro_cush)
ntaxa(hydro_cush) #592
nsamples(hydro_cush) #5

get_taxa_unique(hydro_cush, "Phylum") #16 phyla total
get_taxa_unique(hydro_cush, "Class") #25 Classes total
get_taxa_unique(hydro_cush, "Order") #53 orders total
get_taxa_unique(hydro_cush, "Family") #69 families total
get_taxa_unique(hydro_cush, "Genus") #93 genera total

#Heptageniidae
hep_cush<-subset_samples(cushbr, Family=="Heptageniidae")
hep_cush<-prune_taxa(taxa_sums(hep_cush) > 0, hep_cush)
ntaxa(hep_cush) #1167
nsamples(hep_cush) #8

get_taxa_unique(hep_cush, "Phylum") #20 phyla total
get_taxa_unique(hep_cush, "Class") #41 Classes total
get_taxa_unique(hep_cush, "Order") #97 orders total
get_taxa_unique(hep_cush, "Family") #130 families total
get_taxa_unique(hep_cush, "Genus") #217 genera total

#Chironomidae
chiro_cush<-subset_samples(cushbr, Family=="Chironomidae")
chiro_cush<-prune_taxa(taxa_sums(chiro_cush) > 0, chiro_cush)
ntaxa(chiro_cush) #511
nsamples(chiro_cush) #6

get_taxa_unique(chiro_cush, "Phylum") #15 phyla total
get_taxa_unique(chiro_cush, "Class") #30 Classes total
get_taxa_unique(chiro_cush, "Order") #71 orders total
get_taxa_unique(chiro_cush, "Family") #101 families total
get_taxa_unique(chiro_cush, "Genus") #131 genera total

#Perlidae
perl_cush<-subset_samples(cushbr, Family=="Perlidae")
perl_cush<-prune_taxa(taxa_sums(perl_cush) > 0, perl_cush)
ntaxa(perl_cush) #1190
nsamples(perl_cush) #5

get_taxa_unique(perl_cush, "Phylum") #20 phyla total
get_taxa_unique(perl_cush, "Class") #38 Classes total
get_taxa_unique(perl_cush, "Order") #89 orders total
get_taxa_unique(perl_cush, "Family") #124 families total
get_taxa_unique(perl_cush, "Genus") #201 genera total

#Trichoptera
trichop_cush<-subset_samples(cushbr, Family=="Trichoptera")
trichop_cush<-prune_taxa(taxa_sums(trichop_cush) > 0, trichop_cush)
ntaxa(trichop_cush) #611
nsamples(trichop_cush) #8

get_taxa_unique(trichop_cush, "Phylum") #14 phyla total
get_taxa_unique(trichop_cush, "Class") #21 Classes total
get_taxa_unique(trichop_cush, "Order") #55 orders total
get_taxa_unique(trichop_cush, "Family") #105 families total
get_taxa_unique(trichop_cush, "Genus") #175 genera total

#Ephemeroptera 
ephem_cush<-subset_samples(cushbr, Family=="Ephemeroptera")
ephem_cush<-prune_taxa(taxa_sums(ephem_cush) > 0, ephem_cush)
ntaxa(ephem_cush) #151
nsamples(ephem_cush) #8

get_taxa_unique(ephem_cush, "Phylum") #5 phyla total
get_taxa_unique(ephem_cush, "Class") #10 Classes total
get_taxa_unique(ephem_cush, "Order") #37 orders total
get_taxa_unique(ephem_cush, "Family") #56 families total
get_taxa_unique(ephem_cush, "Genus") #76 genera total

#Diptera 
dip_cush<-subset_samples(cushbr, Family=="Diptera")
dip_cush<-prune_taxa(taxa_sums(dip_cush) > 0, dip_cush)
ntaxa(dip_cush) #206
nsamples(dip_cush) #8

get_taxa_unique(dip_cush, "Phylum") #8 phyla total
get_taxa_unique(dip_cush, "Class") #11 Classes total
get_taxa_unique(dip_cush, "Order") #36 orders total
get_taxa_unique(dip_cush, "Family") #67 families total
get_taxa_unique(dip_cush, "Genus") #86 genera total

#Araneidae
aran_cush<-subset_samples(cushbr, Family=="Araneidae")
aran_cush<-prune_taxa(taxa_sums(aran_cush) > 0, aran_cush)
ntaxa(aran_cush) #301
nsamples(aran_cush) #7

get_taxa_unique(aran_cush, "Phylum") #8 phyla total
get_taxa_unique(aran_cush, "Class") #14 Classes total
get_taxa_unique(aran_cush, "Order") #52 orders total
get_taxa_unique(aran_cush, "Family") #69 families total
get_taxa_unique(aran_cush, "Genus") #120 genera total

#Tetragnathidae
tetra_cush<-subset_samples(cushbr, Family=="Tetragnathidae")
tetra_cush<-prune_taxa(taxa_sums(tetra_cush) > 0, tetra_cush)
ntaxa(tetra_cush) #317
nsamples(tetra_cush) #8

get_taxa_unique(tetra_cush, "Phylum") #12 phyla total
get_taxa_unique(tetra_cush, "Class") #21 Classes total
get_taxa_unique(tetra_cush, "Order") #58 orders total
get_taxa_unique(tetra_cush, "Family") #83 families total
get_taxa_unique(tetra_cush, "Genus") #138 genera total

#Graves Bridge, n = 47
grvbr<-subset_samples(sample_ps, Site=="Graves Bridge")
grvbr<-prune_taxa(taxa_sums(grvbr) > 0, grvbr)
ntaxa(grvbr) #2662
nsamples(grvbr) #47

get_taxa_unique(grvbr, "Phylum") #25 phyla total
get_taxa_unique(grvbr, "Class") #53 Classes total
get_taxa_unique(grvbr, "Order") #135 orders total
get_taxa_unique(grvbr, "Family") #198 families total
get_taxa_unique(grvbr, "Genus") #378 genera total

sort((table(tax_table(grvbr)[,2]))/ntaxa(grvbr)*100)
#Proteobacteria (36.8%), Bacteroidota (32.1%), Firmicutes (13.1%)

#Hydropsychidae
hydro_grvbr<-subset_samples(grvbr, Family=="Hydropsychidae")
hydro_grvbr<-prune_taxa(taxa_sums(hydro_grvbr) > 0, hydro_grvbr)
ntaxa(hydro_grvbr) #1142
nsamples(hydro_grvbr) #8

get_taxa_unique(hydro_grvbr, "Phylum") #21 phyla total
get_taxa_unique(hydro_grvbr, "Class") #37 Classes total
get_taxa_unique(hydro_grvbr, "Order") #76 orders total
get_taxa_unique(hydro_grvbr, "Family") #98 families total
get_taxa_unique(hydro_grvbr, "Genus") #137 genera total


#Heptageniidae
hep_grvbr<-subset_samples(grvbr, Family=="Heptageniidae")
hep_grvbr<-prune_taxa(taxa_sums(hep_grvbr) > 0, hep_grvbr)
ntaxa(hep_grvbr) #1315
nsamples(hep_grvbr) #8

get_taxa_unique(hep_grvbr, "Phylum") #19 phyla total
get_taxa_unique(hep_grvbr, "Class") #42 Classes total
get_taxa_unique(hep_grvbr, "Order") #88 orders total
get_taxa_unique(hep_grvbr, "Family") #132 families total
get_taxa_unique(hydro_grvbr, "Genus") #137 genera total

#Trichoptera at graves bridge
trichop_grvbr<-subset_samples(grvbr, Family=="Trichoptera")
trichop_grvbr<-prune_taxa(taxa_sums(trichop_grvbr) > 0, trichop_grvbr)
ntaxa(trichop_grvbr) #509
nsamples(trichop_grvbr) #7

get_taxa_unique(trichop_grvbr, "Phylum") #13 phyla total
get_taxa_unique(trichop_grvbr, "Class") #19 Classes total
get_taxa_unique(trichop_grvbr, "Order") #61 orders total
get_taxa_unique(trichop_grvbr, "Family") #104 families total
get_taxa_unique(trichop_grvbr, "Genus") #160 genera total

#Ephemeroptera
ephem_grvbr<-subset_samples(grvbr, Family=="Ephemeroptera")
ephem_grvbr<-prune_taxa(taxa_sums(ephem_grvbr) > 0, ephem_grvbr)
ntaxa(ephem_grvbr) #193
nsamples(ephem_grvbr) #8

get_taxa_unique(ephem_grvbr, "Phylum") #10 phyla total
get_taxa_unique(ephem_grvbr, "Class") #12 Classes total
get_taxa_unique(ephem_grvbr, "Order") #41 orders total
get_taxa_unique(ephem_grvbr, "Family") #61 families total
get_taxa_unique(ephem_grvbr, "Genus") #88 genera total

#Araneidae
aran_grvbr<-subset_samples(grvbr, Family=="Araneidae")
aran_grvbr<-prune_taxa(taxa_sums(aran_grvbr) > 0, aran_grvbr)
ntaxa(aran_grvbr) #330
nsamples(aran_grvbr) #8

get_taxa_unique(aran_grvbr, "Phylum") #7 phyla total
get_taxa_unique(aran_grvbr, "Class") #14 Classes total
get_taxa_unique(aran_grvbr, "Order") #48 orders total
get_taxa_unique(aran_grvbr, "Family") #76 families total
get_taxa_unique(aran_grvbr, "Genus") #133 genera total

#Tetragnathidae
tetra_grvbr<-subset_samples(grvbr, Family=="Tetragnathidae")
tetra_grvbr<-prune_taxa(taxa_sums(tetra_grvbr) > 0, tetra_grvbr)
ntaxa(tetra_grvbr) #122
nsamples(tetra_grvbr) #8

get_taxa_unique(tetra_grvbr, "Phylum") #8 phyla total
get_taxa_unique(tetra_grvbr, "Class") #12 Classes total
get_taxa_unique(tetra_grvbr, "Order") #34 orders total
get_taxa_unique(tetra_grvbr, "Family") #49 families total
get_taxa_unique(tetra_grvbr, "Genus") #71 genera total


#Policeman Flats, n = 60
pmf<-subset_samples(sample_ps, Site=="Policeman Flats")
pmf<-prune_taxa(taxa_sums(pmf) > 0, pmf)
ntaxa(pmf) #3168
nsamples(pmf) #58

get_taxa_unique(pmf, "Phylum") #27 phyla total
get_taxa_unique(pmf, "Class") #60 Classes total
get_taxa_unique(pmf, "Order") #152 orders total
get_taxa_unique(pmf, "Family") #217 families total
get_taxa_unique(pmf, "Genus") #416 genera total

sort((table(tax_table(pmf)[,2]))/ntaxa(pmf)*100)
#Proteobacteria (37.4%), Bacteroidota (31.2%), Firmicutes (12.2%)

#Hydropsychidae
hydro_pmf<-subset_samples(pmf, Family=="Hydropsychidae")
hydro_pmf<-prune_taxa(taxa_sums(hydro_pmf) > 0, hydro_pmf)
ntaxa(hydro_pmf) #1195
nsamples(hydro_pmf) #8

get_taxa_unique(hydro_pmf, "Phylum") #21 phyla total
get_taxa_unique(hydro_pmf, "Class") #40 Classes total
get_taxa_unique(hydro_pmf, "Order") #84 orders total
get_taxa_unique(hydro_pmf, "Family") #106 families total
get_taxa_unique(hydro_pmf, "Genus") #138 genera total

#Heptageniidae
hep_pmf<-subset_samples(pmf, Family=="Heptageniidae")
hep_pmf<-prune_taxa(taxa_sums(hep_pmf) > 0, hep_pmf)
ntaxa(hep_pmf) #1559
nsamples(hep_pmf) #8

get_taxa_unique(hep_pmf, "Phylum") #20 phyla total
get_taxa_unique(hep_pmf, "Class") #47 Classes total
get_taxa_unique(hep_pmf, "Order") #103 orders total
get_taxa_unique(hep_pmf, "Family") #148 families total
get_taxa_unique(hep_pmf, "Genus") #233 genera total

#Chironomidae
chiro_pmf<-subset_samples(pmf, Family=="Chironomidae")
chiro_pmf<-prune_taxa(taxa_sums(chiro_pmf) > 0, chiro_pmf)
ntaxa(chiro_pmf) #684
nsamples(chiro_pmf) #4

get_taxa_unique(chiro_pmf, "Phylum") #15 phyla total
get_taxa_unique(chiro_pmf, "Class") #33 Classes total
get_taxa_unique(chiro_pmf, "Order") #78 orders total
get_taxa_unique(chiro_pmf, "Family") #108 families total
get_taxa_unique(chiro_pmf, "Genus") #151 genera total


#Perlidae
perl_pmf<-subset_samples(pmf, Family=="Perlidae")
perl_pmf<-prune_taxa(taxa_sums(perl_pmf) > 0, perl_pmf)
ntaxa(perl_pmf) #753
nsamples(perl_pmf) #8

get_taxa_unique(perl_pmf, "Phylum") #17 phyla total
get_taxa_unique(perl_pmf, "Class") #29 Classes total
get_taxa_unique(perl_pmf, "Order") #70 orders total
get_taxa_unique(perl_pmf, "Family") #95 families total
get_taxa_unique(perl_pmf, "Genus") #132 genera total

#Ephemeroptera
ephem_pmf<-subset_samples(pmf, Family=="Ephemeroptera")
ephem_pmf<-prune_taxa(taxa_sums(ephem_pmf) > 0, ephem_pmf)
ntaxa(ephem_pmf) #468
nsamples(ephem_pmf) #8

get_taxa_unique(ephem_pmf, "Phylum") #15 phyla total
get_taxa_unique(ephem_pmf, "Class") #23 Classes total
get_taxa_unique(ephem_pmf, "Order") #56 orders total
get_taxa_unique(ephem_pmf, "Family") #101 families total
get_taxa_unique(ephem_pmf, "Genus") #161 genera total

#Diptera
dippmf<-subset_samples(pmf, Family=="Diptera")
dippmf<-prune_taxa(taxa_sums(dippmf) > 0, dippmf)
ntaxa(dippmf) #224
nsamples(dippmf) #8

get_taxa_unique(dippmf, "Phylum") #7 phyla total
get_taxa_unique(dippmf, "Class") #10 Classes total
get_taxa_unique(dippmf, "Order") #35 orders total
get_taxa_unique(dippmf, "Family") #65 families total
get_taxa_unique(dippmf, "Genus") #98 genera total

#Araneidae
aranpmf<-subset_samples(pmf, Family=="Araneidae")
aranpmf<-prune_taxa(taxa_sums(aranpmf) > 0, aranpmf)
ntaxa(aranpmf) #387
nsamples(aranpmf) #6

get_taxa_unique(aranpmf, "Phylum") #9 phyla total
get_taxa_unique(aranpmf, "Class") #17 Classes total
get_taxa_unique(aranpmf, "Order") #59 orders total
get_taxa_unique(aranpmf, "Family") #87 families total
get_taxa_unique(aranpmf, "Genus") #149 genera total

#Tetragnathidae
tetrapmf<-subset_samples(pmf, Family=="Tetragnathidae")
tetrapmf<-prune_taxa(taxa_sums(tetrapmf) > 0, tetrapmf)
ntaxa(tetrapmf) #327
nsamples(tetrapmf) #8

get_taxa_unique(tetrapmf, "Phylum") #9 phyla total
get_taxa_unique(tetrapmf, "Class") #16 Classes total
get_taxa_unique(tetrapmf, "Order") #57 orders total
get_taxa_unique(tetrapmf, "Family") #83 families total
get_taxa_unique(tetrapmf, "Genus") #143 genera total

#### ACWA ####

#REF (ACWA control), n = 8
BRR2<-subset_samples(sample_ps, Site=="BRR2")
BRR2<-prune_taxa(taxa_sums(BRR2) > 0, BRR2)
ntaxa(BRR2) #1324
nsamples(BRR2) #8

get_taxa_unique(BRR2, "Phylum") #22 phyla total
get_taxa_unique(BRR2, "Class") #47 Classes total
get_taxa_unique(BRR2, "Order") #112 orders total
get_taxa_unique(BRR2, "Family") #131 families total
get_taxa_unique(BRR2, "Genus") #171 genera total

sort((table(tax_table(BRR2)[,2]))/ntaxa(BRR2)*100)
#Bacteroidota (38.9%), Proteobacteria (31.0%), Firmicutes (11.3%)

#EFF5 (ACWA 5% effluent stream), n = 24
PCR1<-subset_samples(sample_ps, Site=="PCR1")
PCR1<-prune_taxa(taxa_sums(PCR1) > 0, PCR1)
ntaxa(PCR1) #2179
nsamples(PCR1) #24

get_taxa_unique(PCR1, "Phylum") #26 phyla total
get_taxa_unique(PCR1, "Class") #58 Classes total
get_taxa_unique(PCR1, "Order") #138 orders total
get_taxa_unique(PCR1, "Family") #177 families total
get_taxa_unique(PCR1, "Genus") #283 genera total

sort((table(tax_table(PCR1)[,2]))/ntaxa(PCR1)*100)
#Proteobacteria (35.6%), Bacteroidota (32.3%), Firmicutes (8.81%)

#Hydropsychidae
hydro_EFF5<-subset_samples(PCR1, Family=="Hydropsychidae")
hydro_EFF5<-prune_taxa(taxa_sums(hydro_EFF5) > 0, hydro_EFF5)
ntaxa(hydro_EFF5) #1584
nsamples(hydro_EFF5) #8

get_taxa_unique(hydro_EFF5, "Phylum") #25 phyla total
get_taxa_unique(hydro_EFF5, "Class") #53 Classes total
get_taxa_unique(hydro_EFF5, "Order") #121 orders total
get_taxa_unique(hydro_EFF5, "Family") #142 families total
get_taxa_unique(hydro_EFF5, "Genus") #204 genera total

#Baetidae
baetid_EFF5<-subset_samples(PCR1, Family=="Baetidae")
baetid_EFF5<-prune_taxa(taxa_sums(baetid_EFF5) > 0, baetid_EFF5)
ntaxa(baetid_EFF5) #1210
nsamples(baetid_EFF5) #8

get_taxa_unique(baetid_EFF5, "Phylum") #21 phyla total
get_taxa_unique(baetid_EFF5, "Class") #47 Classes total
get_taxa_unique(baetid_EFF5, "Order") #108 orders total
get_taxa_unique(baetid_EFF5, "Family") #144 families total
get_taxa_unique(baetid_EFF5, "Genus") #216 genera total

#Ephemeroptera
ephem_EFF5<-subset_samples(PCR1, Family=="Ephemeroptera")
ephem_EFF5<-prune_taxa(taxa_sums(ephem_EFF5) > 0, ephem_EFF5)
ntaxa(ephem_EFF5) #180
nsamples(ephem_EFF5) #8

get_taxa_unique(ephem_EFF5, "Phylum") #11 phyla total
get_taxa_unique(ephem_EFF5, "Class") #20 Classes total
get_taxa_unique(ephem_EFF5, "Order") #43 orders total
get_taxa_unique(ephem_EFF5, "Family") #59 families total
get_taxa_unique(ephem_EFF5, "Genus") #74 genera total


#PCR3 (ACWA 15% effluent stream), n = 15
PCR3<-subset_samples(sample_ps, Site=="PCR3")
PCR3<-prune_taxa(taxa_sums(PCR3) > 0, PCR3)
ntaxa(PCR3) #2036
nsamples(PCR3) #23

get_taxa_unique(PCR3, "Phylum") #27 phyla total
get_taxa_unique(PCR3, "Class") #54 Classes total
get_taxa_unique(PCR3, "Order") #129 orders total
get_taxa_unique(PCR3, "Family") #177 families total
get_taxa_unique(PCR3, "Genus") #294 genera total

sort((table(tax_table(PCR3)[,2]))/ntaxa(PCR3)*100)
#Proteobacteria (37.0%), Bacteroidota (31.3%), Firmicutes (9.77%)

#Hydropsychidae
hydro_EFF15<-subset_samples(PCR3, Family=="Hydropsychidae")
hydro_EFF15<-prune_taxa(taxa_sums(hydro_EFF15) > 0, hydro_EFF15)
ntaxa(hydro_EFF15) #1348
nsamples(hydro_EFF15) #8

get_taxa_unique(hydro_EFF15, "Phylum") #24 phyla total
get_taxa_unique(hydro_EFF15, "Class") #47 Classes total
get_taxa_unique(hydro_EFF15, "Order") #109 orders total
get_taxa_unique(hydro_EFF15, "Family") #133 families total
get_taxa_unique(hydro_EFF15, "Genus") #181 genera total

#Baetidae
baetid_EFF15<-subset_samples(PCR3, Family=="Baetidae")
baetid_EFF15<-prune_taxa(taxa_sums(baetid_EFF15) > 0, baetid_EFF15)
ntaxa(baetid_EFF15) #1006
nsamples(baetid_EFF15) #8

get_taxa_unique(baetid_EFF15, "Phylum") #23 phyla total
get_taxa_unique(baetid_EFF15, "Class") #43 Classes total
get_taxa_unique(baetid_EFF15, "Order") #98 orders total
get_taxa_unique(baetid_EFF15, "Family") #126 families total
get_taxa_unique(baetid_EFF15, "Genus") #185 genera total

#Ephemeroptera
ephem_EFF15<-subset_samples(PCR3, Family=="Ephemeroptera")
ephem_EFF15<-prune_taxa(taxa_sums(ephem_EFF15) > 0, ephem_EFF15)
ntaxa(ephem_EFF15) #211
nsamples(ephem_EFF15) #8

get_taxa_unique(ephem_EFF15, "Phylum") #13 phyla total
get_taxa_unique(ephem_EFF15, "Class") #19 Classes total
get_taxa_unique(ephem_EFF15, "Order") #52 orders total
get_taxa_unique(ephem_EFF15, "Family") #78 families total
get_taxa_unique(ephem_EFF15, "Genus") #115 genera total
```

# Relative Abundance 

Mean relative abundance is beneficial for displaying the composition of taxonomic levels at different sites/across taxa in an easy to understand way (unlike absolute abundance) but has some drawbacks including:

-Difficulty in comparing the absolute abundance across studies because most literature doesn't usually put the sample size of bacteria or samples. The percent abundance isn't very useful to compare if you don't know the total sample size. 
-Difficulty in interpreting the percent abundance across groups because of a lack of a baseline. 
-No error bars, can't get an idea of the variability in the data
-Hard to decipher between small abundance groups because of the colour limitation and sometimes uneven bars

The following graphs display mean relative abundance post filtering of very low counts (all pre-processing steps are done, but rarefying was not done). 

## Blank samples 
```{r}
#Phylum level

bl_rel_abun_phylum <- tax_glom(blank_ps, taxrank="Phylum")
bl_comp_rel_abun_phylum <- microbiome::transform(bl_rel_abun_phylum, "compositional")
bl_rel_abun_all_prune_phylum = prune_taxa(taxa_sums(bl_comp_rel_abun_phylum) > 0.02, bl_comp_rel_abun_phylum) 

bl_rel_abun_prune_sample_phylum <- psmelt(bl_rel_abun_all_prune_phylum)

bl.sample.df.agg.phylum <- aggregate(Abundance ~ Phylum + Blank_Type, data = bl_rel_abun_prune_sample_phylum, FUN = mean) #aggregate to Phylum level, take the mean

bl.sample.df.agg.phylum$Blank_Type<-factor(bl.sample.df.agg.phylum$Blank_Type, levels=c("Bedsheet", "Floating emergence trap", "Tweezer water rinse", "Buffer tube", "DNA extraction reagent", "Nested PCR negative"))

bl_rel_abun_plot_phylum <- ggplot(bl.sample.df.agg.phylum, aes(x=Blank_Type, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >2% ASVs") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  rremove("grid")
bl_rel_abun_plot_phylum



bl_glom_df = data.frame(Phylum = tax_table(bl_comp_rel_abun_phylum)[,"Phylum"], Abundance = rowSums(otu_table(bl_comp_rel_abun_phylum)), row.names = NULL)
bow_glom_df = bow_glom_df[order(-bow_glom_df$Mean),]

ggplot(bl.sample.df.agg.phylum, aes(y=Phylum, x=Abundance, color=Blank_Type))+
  stat_summary(fun.data=median_hilow, geom = "pointrange", fun.args=list(conf.int=0.95), position = position_dodge(width=0.8))+
  xlab("Relative Abundance (%)")+
  scale_x_log10()



#Family Level

bl_rel_abun_fam <- tax_glom(blank_ps, taxrank="Family")
bl_comp_rel_abun_fam <- microbiome::transform(bl_rel_abun_fam, "compositional")
bl_rel_abun_all_prune_fam = prune_taxa(taxa_sums(bl_comp_rel_abun_fam) > 0.05, bl_comp_rel_abun_fam) 

bl_rel_abun_prune_sample_fam <- psmelt(bl_rel_abun_all_prune_fam)

bl.sample.df.agg.fam <- aggregate(Abundance ~ Family + Blank_Type, data = bl_rel_abun_prune_sample_fam, FUN = mean) #aggregate to Phylum level, take the mean

bl.sample.df.agg.fam$Blank_Type<-factor(bl.sample.df.agg.fam$Blank_Type, levels=c("Bedsheet ", "Floating emergence trap", "Tweezer water rinse", "Buffer tube", "DNA extraction reagent"))

top_fam_bl <- bl.sample.df.agg.fam %>%
    group_by(Blank_Type, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_bl
top5fambl <- top_fam_bl$Family[1:10]
top5fambl <- bl.sample.df.agg.fam %>%
    mutate(Family = fct_other(Family, top5fambl))


top5famplotbl<-ggplot(top5fambl, aes(x=Blank_Type, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(5, "Paired"), "#43464B"))+
  scale_y_continuous(label = scales::percent)

top5famplotbl



#Genus

bl_rel_abun_genus <- tax_glom(blank_ps, taxrank="Genus")
bl_comp_rel_abun_genus <- microbiome::transform(bl_rel_abun_genus, "compositional")

bl_rel_abun_genus_df <- psmelt(bl_comp_rel_abun_genus)

bl.sample.df.agg.genus <- aggregate(Abundance ~ Genus + Blank_Type, data = bl_rel_abun_genus_df, FUN = mean) 

bl.sample.df.agg.genus$Blank_Type<-factor(bl.sample.df.agg.genus$Blank_Type, levels=c("Bedsheet ", "Floating emergence trap", "Tweezer water rinse", "Buffer tube", "DNA extraction reagent"))


top_genus_bl <- bl.sample.df.agg.genus %>%
    group_by(Blank_Type, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_genus_bl
top5genusbl <- top_genus_bl$Genus[1:21]
top5genusbl <- bl.sample.df.agg.genus %>%
    mutate(Genus = fct_other(Genus, top5genusbl))


topgenusplotbl<-ggplot(top5genusbl, aes(x=Blank_Type, y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.7, "cm"),
        legend.text=element_text(size=14, face="italic"), legend.title=element_text(size=14)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(11, "Paired"), "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Blank Type")+
  rremove("grid")

topgenusplotbl

#ggsave(filename = "Microbiome_ED/Final analysis/Relative abundance/Blank samples/Relative abundance of bacterial genera in blank samples.png", height=10, width=12)
```

## Experimental {.tabset}

### Calculating mean relative abundance and SD of top 5 phyla 
```{r}
#### Sites ####
# Cochrane #

cochrane<-subset_samples(sample_ps, Site=="Cochrane") 
cochrane<-prune_taxa(taxa_sums(cochrane) > 0, cochrane) 

coch_glom = tax_glom(cochrane, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

coch_phylum <- psmelt(coch_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

# Sunalta #

sunalta<-subset_samples(sample_ps, Site=="Sunalta") 
sunalta<-prune_taxa(taxa_sums(sunalta) > 0, sunalta) 

sun_glom = tax_glom(sunalta, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

sun_phylum <- psmelt(sun_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

# Cushing Bridge #

cush<-subset_samples(sample_ps, Site=="Cushing Bridge") 
cush<-prune_taxa(taxa_sums(cush) > 0, cush) 

cush_glom = tax_glom(cush, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

cush_phylum <- psmelt(cush_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

#Graves Bridge

grvbr<-subset_samples(sample_ps, Site=="Graves Bridge") 
grvbr<-prune_taxa(taxa_sums(grvbr) > 0, grvbr) 

grvbr_glom = tax_glom(grvbr, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

grvbr_phylum <- psmelt(grvbr_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

# Policeman Flats #

pmf<-subset_samples(sample_ps, Site=="Policeman Flats") 
pmf<-prune_taxa(taxa_sums(pmf) > 0, pmf) 

pmf_glom = tax_glom(pmf, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

pmf_phylum <- psmelt(pmf_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

# REF ACWA Stream #

REF<-subset_samples(sample_ps, Site=="BRR2") 
REF<-prune_taxa(taxa_sums(REF) > 0, REF) 

REF_glom = tax_glom(REF, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

REF_phylum <- psmelt(REF_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 
#(Bacteroidota, Proteobacteria, Firmicutes)

# EFF5 ACWA Stream #

EFF5<-subset_samples(sample_ps, Site=="PCR1") 
EFF5<-prune_taxa(taxa_sums(EFF5) > 0, EFF5) 

EFF5_glom = tax_glom(EFF5, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

EFF5_phylum <- psmelt(EFF5_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup 

# EFF15 ACWA Stream #

EFF15<-subset_samples(sample_ps, Site=="PCR3") 
EFF15<-prune_taxa(taxa_sums(EFF15) > 0, EFF15) 

EFF15_glom = tax_glom(EFF15, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)})

EFF15_phylum <- psmelt(EFF15_glom) %>%
  as_tibble %>%
    group_by(sample_Family, Phylum) %>%
  dplyr::select(c("Abundance", "sample_Family", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>%
    ungroup

#### Hydropsychidae Bow River ####

#How to find the top three phyla across all sites
hydro_Bow<-subset_samples(sample_ps, Family=="Hydropsychidae" & Stream_Type=="Bow River") #Subset hydropsychidae

hydro_Bow_glom = tax_glom(hydro_Bow, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) #Agglomerate to phylum level

hydro_Bow_phylum <- psmelt(hydro_Bow_glom) %>% #Transform to dataframe and summarize to get the mean and standard deviation. 
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup  #Top 3 most abundant phyla across all sites (Bacteroidota, Proteobacteria, Firmicutes)

# Cochrane #
hydro_coch_phy<-subset(coch_phylum, sample_Family=="Hydropsychidae")
hydro_coch_phy<-subset(hydro_coch_phy, Mean > 0)
hydro_coch_phy<-hydro_coch_phy[order(hydro_coch_phy$Mean, decreasing = TRUE),]  

top3_phy_coch_hydro <- hydro_coch_phy$Phylum[1:3]
  
coch_phy_sum_hydro<- hydro_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_coch_hydro))
  
top_means_phy_hydro_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_coch<-merge(top_means_phy_hydro_coch, top_SD_phy_hydro_coch, by=c("Phylum", "sample_Family"))

# Sunalta #
hydro_sun_phy<-subset(sun_phylum, sample_Family=="Hydropsychidae")
hydro_sun_phy<-subset(hydro_sun_phy, Mean > 0)
hydro_sun_phy<-hydro_sun_phy[order(hydro_sun_phy$Mean, decreasing = TRUE),]  

top3_phy_sun_hydro <- hydro_sun_phy$Phylum[1:3]
  
sun_phy_sum_hydro<- hydro_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_sun_hydro))
  
top_means_phy_hydro_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_sun<-aggregate(SD ~ Phylum + sample_Family, data = sun_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_sun<-merge(top_means_phy_hydro_sun, top_SD_phy_hydro_sun, by=c("Phylum", "sample_Family"))


# Cushing Bridge #
hydro_cush_phy<-subset(cush_phylum, sample_Family=="Hydropsychidae")
hydro_cush_phy<-subset(hydro_cush_phy, Mean > 0)
hydro_cush_phy<-hydro_cush_phy[order(hydro_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_hydro <- hydro_cush_phy$Phylum[1:3]
  
cush_phy_sum_hydro<- hydro_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_hydro))
  
top_means_phy_hydro_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_cush<-merge(top_means_phy_hydro_cush, top_SD_phy_hydro_cush, by=c("Phylum", "sample_Family"))


# Graves Bridge #
hydro_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Hydropsychidae")
hydro_grvbr_phy<-subset(hydro_grvbr_phy, Mean > 0)
hydro_grvbr_phy<-hydro_grvbr_phy[order(hydro_grvbr_phy$Mean, decreasing = TRUE),]  

top3_phy_grvbr_hydro <- hydro_grvbr_phy$Phylum[1:3]
  
grvbr_phy_sum_hydro<- hydro_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_grvbr_hydro))
  
top_means_phy_hydro_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_grvbr<-aggregate(SD ~ Phylum + sample_Family, data = grvbr_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_grvbr<-merge(top_means_phy_hydro_grvbr, top_SD_phy_hydro_grvbr, by=c("Phylum", "sample_Family"))


# Policeman Flats #
hydro_pmf_phy<-subset(pmf_phylum, sample_Family=="Hydropsychidae")
hydro_pmf_phy<-subset(hydro_pmf_phy, Mean > 0)
hydro_pmf_phy<-hydro_pmf_phy[order(hydro_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_hydro <- hydro_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_hydro<- hydro_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_hydro))
  
top_means_phy_hydro_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_pmf<-merge(top_means_phy_hydro_pmf, top_SD_phy_hydro_pmf, by=c("Phylum", "sample_Family"))


#### Hydropsychidae ACWA ####

#How to find the top three phyla across all sites
hydro_ACWA<-subset_samples(sample_ps, Family=="Hydropsychidae" & Stream_Type!="Bow River") #Subset hydropsychidae

hydro_ACWA_glom = tax_glom(hydro_ACWA, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) #Agglomerate to phylum level

hydro_ACWA_phylum <- psmelt(hydro_ACWA_glom) %>% #Transform to dataframe and summarize to get the mean and standard deviation. 
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup  #Top 3 most abundant phyla across all sites (Bacteroidota, Proteobacteria, Firmicutes)

# REF #
hydro_REF_phy<-subset(REF_phylum, sample_Family=="Hydropsychidae")
hydro_REF_phy<-subset(hydro_REF_phy, Mean > 0)
hydro_REF_phy<-hydro_REF_phy[order(hydro_REF_phy$Mean, decreasing = TRUE),]  

top3_phy_REF_hydro <- hydro_REF_phy$Phylum[1:3]
  
REF_phy_sum_hydro<- hydro_REF_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_REF_hydro))
  
top_means_phy_hydro_REF<-aggregate(Mean ~ Phylum + sample_Family, data = REF_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_REF<-aggregate(SD ~ Phylum + sample_Family, data = REF_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_REF<-merge(top_means_phy_hydro_REF, top_SD_phy_hydro_REF, by=c("Phylum", "sample_Family"))

# EFF5 #

hydro_EFF5_phy<-subset(EFF5_phylum, sample_Family=="Hydropsychidae")
hydro_EFF5_phy<-subset(hydro_EFF5_phy, Mean > 0)
hydro_EFF5_phy<-hydro_EFF5_phy[order(hydro_EFF5_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF5_hydro <- hydro_EFF5_phy$Phylum[1:3]
  
EFF5_phy_sum_hydro<- hydro_EFF5_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF5_hydro))
  
top_means_phy_hydro_EFF5<-aggregate(Mean ~ Phylum + sample_Family, data = EFF5_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_EFF5<-aggregate(SD ~ Phylum + sample_Family, data = EFF5_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_EFF5<-merge(top_means_phy_hydro_EFF5, top_SD_phy_hydro_EFF5, by=c("Phylum", "sample_Family"))

# EFF15 #

hydro_EFF15_phy<-subset(EFF15_phylum, sample_Family=="Hydropsychidae")
hydro_EFF15_phy<-subset(hydro_EFF15_phy, Mean > 0)
hydro_EFF15_phy<-hydro_EFF15_phy[order(hydro_EFF15_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF15_hydro <- hydro_EFF15_phy$Phylum[1:3]
  
EFF15_phy_sum_hydro<- hydro_EFF15_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF15_hydro))
  
top_means_phy_hydro_EFF15<-aggregate(Mean ~ Phylum + sample_Family, data = EFF15_phy_sum_hydro, FUN = sum)
top_SD_phy_hydro_EFF15<-aggregate(SD ~ Phylum + sample_Family, data = EFF15_phy_sum_hydro, FUN = sum) #Not sure if this is statistically correct to sum all the standard deviations together... 

top_phy_hydro_EFF15<-merge(top_means_phy_hydro_EFF15, top_SD_phy_hydro_EFF15, by=c("Phylum", "sample_Family"))


#### Heptageniidae ####

hep<-subset_samples(sample_ps, Family=="Heptageniidae")

hep_glom = tax_glom(hep, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

hep_phylum <- psmelt(hep_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Cyanobacteria, Firmicutes)

# Cochrane #
hep_coch_phy<-subset(coch_phylum, sample_Family=="Heptageniidae")
hep_coch_phy<-subset(hep_coch_phy, Mean > 0)
hep_coch_phy<-hep_coch_phy[order(hep_coch_phy$Mean, decreasing = TRUE),]  

top3_phy_coch_hep <- hep_coch_phy$Phylum[1:3]
  
coch_phy_sum_hep<- hep_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_coch_hep))
  
top_means_phy_hep_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_hep, FUN = sum)
top_SD_phy_hep_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_hep, FUN = sum)

top_phy_hep_coch<-merge(top_means_phy_hep_coch, top_SD_phy_hep_coch, by=c("Phylum", "sample_Family"))

# Sunalta #
hep_sun_phy<-subset(sun_phylum, sample_Family=="Heptageniidae")
hep_sun_phy<-subset(hep_sun_phy, Mean > 0)
hep_sun_phy<-hep_sun_phy[order(hep_sun_phy$Mean, decreasing = TRUE),]  

top3_phy_sun_hep <- hep_sun_phy$Phylum[1:3]
  
sun_phy_sum_hep<- hep_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_sun_hep))
  
top_means_phy_hep_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_hep, FUN = sum)
top_SD_phy_hep_sun<-aggregate(SD ~ Phylum + sample_Family, data = sun_phy_sum_hep, FUN = sum)

top_phy_hep_sun<-merge(top_means_phy_hep_sun, top_SD_phy_hep_sun, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
hep_cush_phy<-subset(cush_phylum, sample_Family=="Heptageniidae")
hep_cush_phy<-subset(hep_cush_phy, Mean > 0)
hep_cush_phy<-hep_cush_phy[order(hep_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_hep <- hep_cush_phy$Phylum[1:3]
  
cush_phy_sum_hep<- hep_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_hep))
  
top_means_phy_hep_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_hep, FUN = sum)
top_SD_phy_hep_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_hep, FUN = sum)

top_phy_hep_cush<-merge(top_means_phy_hep_cush, top_SD_phy_hep_cush, by=c("Phylum", "sample_Family"))

# Graves Bridge #
hep_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Heptageniidae")
hep_grvbr_phy<-subset(hep_grvbr_phy, Mean > 0)
hep_grvbr_phy<-hep_grvbr_phy[order(hep_grvbr_phy$Mean, decreasing = TRUE),]  

top3_phy_grvbr_hep <- hep_grvbr_phy$Phylum[1:3]
  
grvbr_phy_sum_hep<- hep_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_grvbr_hep))
  
top_means_phy_hep_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_hep, FUN = sum)
top_SD_phy_hep_grvbr<-aggregate(SD ~ Phylum + sample_Family, data = grvbr_phy_sum_hep, FUN = sum)

top_phy_hep_grvbr<-merge(top_means_phy_hep_grvbr, top_SD_phy_hep_grvbr, by=c("Phylum", "sample_Family"))

# Policeman Flats #
hep_pmf_phy<-subset(pmf_phylum, sample_Family=="Heptageniidae")
hep_pmf_phy<-subset(hep_pmf_phy, Mean > 0)
hep_pmf_phy<-hep_pmf_phy[order(hep_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_hep <- hep_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_hep<- hep_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_hep))
  
top_means_phy_hep_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_hep, FUN = sum)
top_SD_phy_hep_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_hep, FUN = sum)

top_phy_hep_pmf<-merge(top_means_phy_hep_pmf, top_SD_phy_hep_pmf, by=c("Phylum", "sample_Family"))


#### Chironomidae ####

chiro<-subset_samples(sample_ps, Family=="Chironomidae")

chiro_glom = tax_glom(chiro, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

chiro_phylum <- psmelt(chiro_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Cyanobacteria, Bacteroidota)

# Cochrane #
chiro_coch_phy<-subset(coch_phylum, sample_Family=="Chironomidae")
chiro_coch_phy<-subset(chiro_coch_phy, Mean > 0)
chiro_coch_phy<-chiro_coch_phy[order(chiro_coch_phy$Mean, decreasing = TRUE),]  

coch_phy_sum_chiro<- chiro_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_chiro_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_chiro, FUN = sum)
top_SD_phy_chiro_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_chiro, FUN = sum)

top_phy_chiro_coch<-merge(top_means_phy_chiro_coch, top_SD_phy_chiro_coch, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
chiro_cush_phy<-subset(cush_phylum, sample_Family=="Chironomidae")
chiro_cush_phy<-subset(chiro_cush_phy, Mean > 0)
chiro_cush_phy<-chiro_cush_phy[order(chiro_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_chiro <- chiro_cush_phy$Phylum[1:3]
  
cush_phy_sum_chiro<- chiro_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_chiro))
  
top_means_phy_chiro_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_chiro, FUN = sum)
top_SD_phy_chiro_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_chiro, FUN = sum)

top_phy_chiro_cush<-merge(top_means_phy_chiro_cush, top_SD_phy_chiro_cush, by=c("Phylum", "sample_Family"))

# Policeman Flats #
chiro_pmf_phy<-subset(pmf_phylum, sample_Family=="Chironomidae")
chiro_pmf_phy<-subset(chiro_pmf_phy, Mean > 0)
chiro_pmf_phy<-chiro_pmf_phy[order(chiro_pmf_phy$Mean, decreasing = TRUE),]  

pmf_phy_sum_chiro<- chiro_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_chiro_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_chiro, FUN = sum)
top_SD_phy_chiro_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_chiro, FUN = sum)

top_phy_chiro_pmf<-merge(top_means_phy_chiro_pmf, top_SD_phy_chiro_pmf, by=c("Phylum", "sample_Family"))


#### Perlidae ####
perl<-subset_samples(sample_ps, Family=="Perlidae")

perl_glom = tax_glom(perl, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

perl_phylum <- psmelt(perl_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Bacteroidota, Cyanobacteria)

# Cochrane #
perl_coch_phy<-subset(coch_phylum, sample_Family=="Perlidae")
perl_coch_phy<-subset(perl_coch_phy, Mean > 0)
perl_coch_phy<-perl_coch_phy[order(perl_coch_phy$Mean, decreasing = TRUE),]  

top3_phy_coch_perl <- perl_coch_phy$Phylum[1:3]
  
coch_phy_sum_perl<- perl_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_coch_perl))
  
top_means_phy_perl_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_perl, FUN = sum)
top_SD_phy_perl_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_perl, FUN = sum)

top_phy_perl_coch<-merge(top_means_phy_perl_coch, top_SD_phy_perl_coch, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
perl_cush_phy<-subset(cush_phylum, sample_Family=="Perlidae")
perl_cush_phy<-subset(perl_cush_phy, Mean > 0)
perl_cush_phy<-perl_cush_phy[order(perl_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_perl <- perl_cush_phy$Phylum[1:3]
  
cush_phy_sum_perl<- perl_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_perl))
  
top_means_phy_perl_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_perl, FUN = sum)
top_SD_phy_perl_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_perl, FUN = sum)

top_phy_perl_cush<-merge(top_means_phy_perl_cush, top_SD_phy_perl_cush, by=c("Phylum", "sample_Family"))

# Policeman Flats #
perl_pmf_phy<-subset(pmf_phylum, sample_Family=="Perlidae")
perl_pmf_phy<-subset(perl_pmf_phy, Mean > 0)
perl_pmf_phy<-perl_pmf_phy[order(perl_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_perl <- perl_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_perl<- perl_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_perl))
  
top_means_phy_perl_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_perl, FUN = sum)
top_SD_phy_perl_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_perl, FUN = sum)

top_phy_perl_pmf<-merge(top_means_phy_perl_pmf, top_SD_phy_perl_pmf, by=c("Phylum", "sample_Family"))


#### Trichoptera ####
trichop<-subset_samples(sample_ps, Family=="Trichoptera")

trichop_glom = tax_glom(trichop, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

trichop_phylum <- psmelt(trichop_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Cyanobacteria, Bacteroidota)

# Cochrane #
trichop_coch_phy<-subset(coch_phylum, sample_Family=="Trichoptera")
trichop_coch_phy<-subset(trichop_coch_phy, Mean > 0)
trichop_coch_phy<-trichop_coch_phy[order(trichop_coch_phy$Mean, decreasing = TRUE),]  

top3_phy_coch_trichop <- trichop_coch_phy$Phylum[1:3]
  
coch_phy_sum_trichop<- trichop_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_coch_trichop))
  
top_means_phy_trichop_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_trichop, FUN = sum)
top_SD_phy_trichop_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_trichop, FUN = sum)

top_phy_trichop_coch<-merge(top_means_phy_trichop_coch, top_SD_phy_trichop_coch, by=c("Phylum", "sample_Family"))

# Sunalta #

trichop_sun_phy<-subset(sun_phylum, sample_Family=="Trichoptera")
trichop_sun_phy<-subset(trichop_sun_phy, Mean > 0)
trichop_sun_phy<-trichop_sun_phy[order(trichop_sun_phy$Mean, decreasing = TRUE),]  
  
sun_phy_sum_trichop<- trichop_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Proteobacteria", "Cyanobacteria", "Bacteroidota")))
  
top_means_phy_trichop_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_trichop, FUN = sum)
top_SD_phy_trichop_sun<-aggregate(SD ~ Phylum + sample_Family, data = sun_phy_sum_trichop, FUN = sum)

top_phy_trichop_sun<-merge(top_means_phy_trichop_sun, top_SD_phy_trichop_sun, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
trichop_cush_phy<-subset(cush_phylum, sample_Family=="Trichoptera")
trichop_cush_phy<-subset(trichop_cush_phy, Mean > 0)
trichop_cush_phy<-trichop_cush_phy[order(trichop_cush_phy$Mean, decreasing = TRUE),]  

cush_phy_sum_trichop<- trichop_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Proteobacteria", "Cyanobacteria", "Bacteroidota")))
  
top_means_phy_trichop_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_trichop, FUN = sum)
top_SD_phy_trichop_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_trichop, FUN = sum)

top_phy_trichop_cush<-merge(top_means_phy_trichop_cush, top_SD_phy_trichop_cush, by=c("Phylum", "sample_Family"))

# Graves Bridge #
trichop_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Trichoptera")
trichop_grvbr_phy<-subset(trichop_grvbr_phy, Mean > 0)
trichop_grvbr_phy<-trichop_grvbr_phy[order(trichop_grvbr_phy$Mean, decreasing = TRUE),]  

top3_phy_grvbr_trichop <- trichop_grvbr_phy$Phylum[1:3]
  
grvbr_phy_sum_trichop<- trichop_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_grvbr_trichop))
  
top_means_phy_trichop_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_trichop, FUN = sum)
top_SD_phy_trichop_grvbr<-aggregate(SD ~ Phylum + sample_Family, data = grvbr_phy_sum_trichop, FUN = sum)

top_phy_trichop_grvbr<-merge(top_means_phy_trichop_grvbr, top_SD_phy_trichop_grvbr, by=c("Phylum", "sample_Family"))

#### Ephemeroptera Bow River ####

ephem_Bow<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type=="Bow River")

ephem_Bow_glom = tax_glom(ephem_Bow, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

ephem_Bow_phylum <- psmelt(ephem_Bow_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Cyanobacteria, Firmicutes)

# Cochrane #
ephem_coch_phy<-subset(coch_phylum, sample_Family=="Ephemeroptera")
ephem_coch_phy<-subset(ephem_coch_phy, Mean > 0)
ephem_coch_phy<-ephem_coch_phy[order(ephem_coch_phy$Mean, decreasing = TRUE),]  

coch_phy_sum_ephem<- ephem_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Proteobacteria", "Cyanobacteria", "Firmicutes")))
  
top_means_phy_ephem_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_ephem, FUN = sum)

top_phy_ephem_coch<-merge(top_means_phy_ephem_coch, top_SD_phy_ephem_coch, by=c("Phylum", "sample_Family"))

# Sunalta #
ephem_sun_phy<-subset(sun_phylum, sample_Family=="Ephemeroptera")
ephem_sun_phy<-subset(ephem_sun_phy, Mean > 0)
ephem_sun_phy<-ephem_sun_phy[order(ephem_sun_phy$Mean, decreasing = TRUE),]  

top3_phy_sun_ephem <- ephem_sun_phy$Phylum[1:3]
  
sun_phy_sum_ephem<- ephem_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_sun_ephem))
  
top_means_phy_ephem_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_sun<-aggregate(SD ~ Phylum + sample_Family, data =sun_phy_sum_ephem, FUN = sum)

top_phy_ephem_sun<-merge(top_means_phy_ephem_sun, top_SD_phy_ephem_sun, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
ephem_cush_phy<-subset(cush_phylum, sample_Family=="Ephemeroptera")
ephem_cush_phy<-subset(ephem_cush_phy, Mean > 0)
ephem_cush_phy<-ephem_cush_phy[order(ephem_cush_phy$Mean, decreasing = TRUE),]  

cush_phy_sum_ephem<- ephem_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Proteobacteria", "Cyanobacteria", "Firmicutes")))
  
top_means_phy_ephem_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_cush<-aggregate(SD ~ Phylum + sample_Family, data =cush_phy_sum_ephem, FUN = sum)

top_phy_ephem_cush<-merge(top_means_phy_ephem_cush, top_SD_phy_ephem_cush, by=c("Phylum", "sample_Family"))

# Graves Bridge #
ephem_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Ephemeroptera")
ephem_grvbr_phy<-subset(ephem_grvbr_phy, Mean > 0)
ephem_grvbr_phy<-ephem_grvbr_phy[order(ephem_grvbr_phy$Mean, decreasing = TRUE),]  

top3_phy_grvbr_ephem <- ephem_grvbr_phy$Phylum[1:3]
  
grvbr_phy_sum_ephem<- ephem_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_grvbr_ephem))
  
top_means_phy_ephem_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_grvbr<-aggregate(SD ~ Phylum + sample_Family, data =grvbr_phy_sum_ephem, FUN = sum)

top_phy_ephem_grvbr<-merge(top_means_phy_ephem_grvbr, top_SD_phy_ephem_grvbr, by=c("Phylum", "sample_Family"))

# Policeman Flats #
ephem_pmf_phy<-subset(pmf_phylum, sample_Family=="Ephemeroptera")
ephem_pmf_phy<-subset(ephem_pmf_phy, Mean > 0)
ephem_pmf_phy<-ephem_pmf_phy[order(ephem_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_ephem <- ephem_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_ephem<- ephem_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_ephem))
  
top_means_phy_ephem_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_pmf<-aggregate(SD ~ Phylum + sample_Family, data =pmf_phy_sum_ephem, FUN = sum)

top_phy_ephem_pmf<-merge(top_means_phy_ephem_pmf, top_SD_phy_ephem_pmf, by=c("Phylum", "sample_Family"))


#### Ephemeroptera ACWA ####
ephem_ACWA<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type!="Bow River")

ephem_ACWA_glom = tax_glom(ephem_ACWA, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

ephem_ACWA_phylum <- psmelt(ephem_ACWA_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites ()

# EFF5 #
ephem_EFF5_phy<-subset(EFF5_phylum, sample_Family=="Ephemeroptera")
ephem_EFF5_phy<-subset(ephem_EFF5_phy, Mean > 0)
ephem_EFF5_phy<-ephem_EFF5_phy[order(ephem_EFF5_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF5_ephem <- ephem_EFF5_phy$Phylum[1:3]
  
EFF5_phy_sum_ephem<- ephem_EFF5_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF5_ephem))
  
top_means_phy_ephem_EFF5<-aggregate(Mean ~ Phylum + sample_Family, data = EFF5_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_EFF5<-aggregate(SD ~ Phylum + sample_Family, data = EFF5_phy_sum_ephem, FUN = sum)

top_phy_ephem_EFF5<-merge(top_means_phy_ephem_EFF5, top_SD_phy_ephem_EFF5, by=c("Phylum", "sample_Family"))

# EFF15 #
ephem_EFF15_phy<-subset(EFF15_phylum, sample_Family=="Ephemeroptera")
ephem_EFF15_phy<-subset(ephem_EFF15_phy, Mean > 0)
ephem_EFF15_phy<-ephem_EFF15_phy[order(ephem_EFF15_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF15_ephem <- ephem_EFF15_phy$Phylum[1:3]
  
EFF15_phy_sum_ephem<- ephem_EFF15_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF15_ephem))
  
top_means_phy_ephem_EFF15<-aggregate(Mean ~ Phylum + sample_Family, data = EFF15_phy_sum_ephem, FUN = sum)
top_SD_phy_ephem_EFF15<-aggregate(SD ~ Phylum + sample_Family, data = EFF15_phy_sum_ephem, FUN = sum)

top_phy_ephem_EFF15<-merge(top_means_phy_ephem_EFF15, top_SD_phy_ephem_EFF15, by=c("Phylum", "sample_Family"))


#### Baetidae ####

baetid_ACWA<-subset_samples(sample_ps, Family=="Baetidae" & Stream_Type!="Bow River")

baetid_ACWA_glom = tax_glom(baetid_ACWA, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

baetid_ACWA_phylum <- psmelt(baetid_ACWA_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites ()

# EFF5 #
baetid_EFF5_phy<-subset(EFF5_phylum, sample_Family=="Baetidae")
baetid_EFF5_phy<-subset(baetid_EFF5_phy, Mean > 0)
baetid_EFF5_phy<-baetid_EFF5_phy[order(baetid_EFF5_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF5_baetid <- baetid_EFF5_phy$Phylum[1:3]
  
EFF5_phy_sum_baetid<- baetid_EFF5_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF5_baetid))
  
top_means_phy_baetid_EFF5<-aggregate(Mean ~ Phylum + sample_Family, data = EFF5_phy_sum_baetid, FUN = sum)
top_SD_phy_baetid_EFF5<-aggregate(SD ~ Phylum + sample_Family, data = EFF5_phy_sum_baetid, FUN = sum)

top_phy_baetid_EFF5<-merge(top_means_phy_baetid_EFF5, top_SD_phy_baetid_EFF5, by=c("Phylum", "sample_Family"))

# EFF15 #
baetid_EFF15_phy<-subset(EFF15_phylum, sample_Family=="Baetidae")
baetid_EFF15_phy<-subset(baetid_EFF15_phy, Mean > 0)
baetid_EFF15_phy<-baetid_EFF15_phy[order(baetid_EFF15_phy$Mean, decreasing = TRUE),]  

top3_phy_EFF15_baetid <- baetid_EFF15_phy$Phylum[1:3]
  
EFF15_phy_sum_baetid<- baetid_EFF15_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_EFF15_baetid))
  
top_means_phy_baetid_EFF15<-aggregate(Mean ~ Phylum + sample_Family, data = EFF15_phy_sum_baetid, FUN = sum)
top_SD_phy_baetid_EFF15<-aggregate(SD ~ Phylum + sample_Family, data = EFF15_phy_sum_baetid, FUN = sum)

top_phy_baetid_EFF15<-merge(top_means_phy_baetid_EFF15, top_SD_phy_baetid_EFF15, by=c("Phylum", "sample_Family"))


#### Diptera ####
dip<-subset_samples(sample_ps, Family=="Diptera")

dip_glom = tax_glom(dip, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

dip_phylum <- psmelt(dip_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Firmicutes, Cyanobacteria)

# Cochrane #
dip_coch_phy<-subset(coch_phylum, sample_Family=="Diptera")
dip_coch_phy<-subset(dip_coch_phy, Mean > 0)
dip_coch_phy<-dip_coch_phy[order(dip_coch_phy$Mean, decreasing = TRUE),]  

top3_phy_coch_dip <- dip_coch_phy$Phylum[1:3]
  
coch_phy_sum_dip<- dip_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_coch_dip))
  
top_means_phy_dip_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_dip, FUN = sum)
top_SD_phy_dip_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_dip, FUN = sum)

top_phy_dip_coch<-merge(top_means_phy_dip_coch, top_SD_phy_dip_coch, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
dip_cush_phy<-subset(cush_phylum, sample_Family=="Diptera")
dip_cush_phy<-subset(dip_cush_phy, Mean > 0)
dip_cush_phy<-dip_cush_phy[order(dip_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_dip <- dip_cush_phy$Phylum[1:3]
  
cush_phy_sum_dip<- dip_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_dip))
  
top_means_phy_dip_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_dip, FUN = sum)
top_SD_phy_dip_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_dip, FUN = sum)

top_phy_dip_cush<-merge(top_means_phy_dip_cush, top_SD_phy_dip_cush, by=c("Phylum", "sample_Family"))

# Policeman Flats #
dip_pmf_phy<-subset(pmf_phylum, sample_Family=="Diptera")
dip_pmf_phy<-subset(dip_pmf_phy, Mean > 0)
dip_pmf_phy<-dip_pmf_phy[order(dip_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_dip <- dip_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_dip<- dip_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_dip))
  
top_means_phy_dip_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_dip, FUN = sum)
top_SD_phy_dip_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_dip, FUN = sum)

top_phy_dip_pmf<-merge(top_means_phy_dip_pmf, top_SD_phy_dip_pmf, by=c("Phylum", "sample_Family"))


#### Araneidae ####
aran<-subset_samples(sample_ps, Family=="Araneidae")

aran_glom = tax_glom(aran, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

aran_phylum <- psmelt(aran_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Cyanobacteria, Bacteroidota, Proteobacteria)

# Cochrane #
aran_coch_phy<-subset(coch_phylum, sample_Family=="Araneidae")
aran_coch_phy<-subset(aran_coch_phy, Mean > 0)
aran_coch_phy<-aran_coch_phy[order(aran_coch_phy$Mean, decreasing = TRUE),]  

coch_phy_sum_aran<- aran_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_aran_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_aran, FUN = sum)
top_SD_phy_aran_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_aran, FUN = sum)

top_phy_aran_coch<-merge(top_means_phy_aran_coch, top_SD_phy_aran_coch, by=c("Phylum", "sample_Family"))

# Sunalta #
aran_sun_phy<-subset(sun_phylum, sample_Family=="Araneidae")
aran_sun_phy<-subset(aran_sun_phy, Mean > 0)
aran_sun_phy<-aran_sun_phy[order(aran_sun_phy$Mean, decreasing = TRUE),]  

top3_phy_sun_aran <- aran_sun_phy$Phylum[1:3]
  
sun_phy_sum_aran<- aran_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_sun_aran))
  
top_means_phy_aran_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_aran, FUN = sum)
top_SD_phy_aran_sun<-aggregate(SD ~ Phylum + sample_Family, data = sun_phy_sum_aran, FUN = sum)

top_phy_aran_sun<-merge(top_means_phy_aran_sun, top_SD_phy_aran_sun, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
aran_cush_phy<-subset(cush_phylum, sample_Family=="Araneidae")
aran_cush_phy<-subset(aran_cush_phy, Mean > 0)
aran_cush_phy<-aran_cush_phy[order(aran_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_aran <- aran_cush_phy$Phylum[1:3]
  
cush_phy_sum_aran<- aran_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_aran))
  
top_means_phy_aran_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_aran, FUN = sum)
top_SD_phy_aran_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_aran, FUN = sum)

top_phy_aran_cush<-merge(top_means_phy_aran_cush, top_SD_phy_aran_cush, by=c("Phylum", "sample_Family"))

# Graves Bridge #
aran_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Araneidae")
aran_grvbr_phy<-subset(aran_grvbr_phy, Mean > 0)
aran_grvbr_phy<-aran_grvbr_phy[order(aran_grvbr_phy$Mean, decreasing = TRUE),]  

top3_phy_grvbr_aran <- aran_grvbr_phy$Phylum[1:3]
  
grvbr_phy_sum_aran<- aran_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_grvbr_aran))
  
top_means_phy_aran_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_aran, FUN = sum)
top_SD_phy_aran_grvbr<-aggregate(SD ~ Phylum + sample_Family, data = grvbr_phy_sum_aran, FUN = sum)

top_phy_aran_grvbr<-merge(top_means_phy_aran_grvbr, top_SD_phy_aran_grvbr, by=c("Phylum", "sample_Family"))

# Policeman Flats #
aran_pmf_phy<-subset(pmf_phylum, sample_Family=="Araneidae")
aran_pmf_phy<-subset(aran_pmf_phy, Mean > 0)
aran_pmf_phy<-aran_pmf_phy[order(aran_pmf_phy$Mean, decreasing = TRUE),]  

pmf_phy_sum_aran<- aran_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_aran_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_aran, FUN = sum)
top_SD_phy_aran_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_aran, FUN = sum)

top_phy_aran_pmf<-merge(top_means_phy_aran_pmf, top_SD_phy_aran_pmf, by=c("Phylum", "sample_Family"))


#### Tetragnathidae ####
tetra<-subset_samples(sample_ps, Family=="Tetragnathidae")

tetra_glom = tax_glom(tetra, "Phylum") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 

tetra_phylum <- psmelt(tetra_glom) %>%  
  as_tibble %>%
    group_by(Phylum) %>%
  dplyr::select(c("Abundance", "Phylum")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance)) %>% ungroup #Top 3 most abundant phyla across all sites (Proteobacteria, Bacteroidota, Cyanobacteria)

# Cochrane #
tetra_coch_phy<-subset(coch_phylum, sample_Family=="Tetragnathidae")
tetra_coch_phy<-subset(tetra_coch_phy, Mean > 0)
tetra_coch_phy<-tetra_coch_phy[order(tetra_coch_phy$Mean, decreasing = TRUE),]  

coch_phy_sum_tetra<- tetra_coch_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))

top_means_phy_tetra_coch<-aggregate(Mean ~ Phylum + sample_Family, data = coch_phy_sum_tetra, FUN = sum)
top_SD_phy_tetra_coch<-aggregate(SD ~ Phylum + sample_Family, data = coch_phy_sum_tetra, FUN = sum)

top_phy_tetra_coch<-merge(top_means_phy_tetra_coch, top_SD_phy_tetra_coch, by=c("Phylum", "sample_Family"))

# Sunalta #
tetra_sun_phy<-subset(sun_phylum, sample_Family=="Tetragnathidae")
tetra_sun_phy<-subset(tetra_sun_phy, Mean > 0)
tetra_sun_phy<-tetra_sun_phy[order(tetra_sun_phy$Mean, decreasing = TRUE),]  

sun_phy_sum_tetra<- tetra_sun_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_tetra_sun<-aggregate(Mean ~ Phylum + sample_Family, data = sun_phy_sum_tetra, FUN = sum)
top_SD_phy_tetra_sun<-aggregate(SD ~ Phylum + sample_Family, data = sun_phy_sum_tetra, FUN = sum)

top_phy_tetra_sun<-merge(top_means_phy_tetra_sun, top_SD_phy_tetra_sun, by=c("Phylum", "sample_Family"))

# Cushing Bridge #
tetra_cush_phy<-subset(cush_phylum, sample_Family=="Tetragnathidae")
tetra_cush_phy<-subset(tetra_cush_phy, Mean > 0)
tetra_cush_phy<-tetra_cush_phy[order(tetra_cush_phy$Mean, decreasing = TRUE),]  

top3_phy_cush_tetra <- tetra_cush_phy$Phylum[1:3]
  
cush_phy_sum_tetra<- tetra_cush_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_cush_tetra))
  
top_means_phy_tetra_cush<-aggregate(Mean ~ Phylum + sample_Family, data = cush_phy_sum_tetra, FUN = sum)
top_SD_phy_tetra_cush<-aggregate(SD ~ Phylum + sample_Family, data = cush_phy_sum_tetra, FUN = sum)

top_phy_tetra_cush<-merge(top_means_phy_tetra_cush, top_SD_phy_tetra_cush, by=c("Phylum", "sample_Family"))

# Graves Bridge #
tetra_grvbr_phy<-subset(grvbr_phylum, sample_Family=="Tetragnathidae")
tetra_grvbr_phy<-subset(tetra_grvbr_phy, Mean > 0)
tetra_grvbr_phy<-tetra_grvbr_phy[order(tetra_grvbr_phy$Mean, decreasing = TRUE),]  

grvbr_phy_sum_tetra<- tetra_grvbr_phy %>%
  mutate(Phylum = fct_other(Phylum, keep=c("Cyanobacteria", "Bacteroidota", "Proteobacteria")))
  
top_means_phy_tetra_grvbr<-aggregate(Mean ~ Phylum + sample_Family, data = grvbr_phy_sum_tetra, FUN = sum)
top_SD_phy_tetra_grvbr<-aggregate(SD ~ Phylum + sample_Family, data = grvbr_phy_sum_tetra, FUN = sum)

top_phy_tetra_grvbr<-merge(top_means_phy_tetra_grvbr, top_SD_phy_tetra_grvbr, by=c("Phylum", "sample_Family"))

# Policeman Flats #
tetra_pmf_phy<-subset(pmf_phylum, sample_Family=="Tetragnathidae")
tetra_pmf_phy<-subset(tetra_pmf_phy, Mean > 0)
tetra_pmf_phy<-tetra_pmf_phy[order(tetra_pmf_phy$Mean, decreasing = TRUE),]  

top3_phy_pmf_tetra <- tetra_pmf_phy$Phylum[1:3]
  
pmf_phy_sum_tetra<- tetra_pmf_phy %>%
  mutate(Phylum = fct_other(Phylum, top3_phy_pmf_tetra))
  
top_means_phy_tetra_pmf<-aggregate(Mean ~ Phylum + sample_Family, data = pmf_phy_sum_tetra, FUN = sum)
top_SD_phy_tetra_pmf<-aggregate(SD ~ Phylum + sample_Family, data = pmf_phy_sum_tetra, FUN = sum)

top_phy_tetra_pmf<-merge(top_means_phy_tetra_pmf, top_SD_phy_tetra_pmf, by=c("Phylum", "sample_Family"))
```

### Testing significant differences across sites and invert taxa
```{r}
#### Bow River ####

### Phylum level ###

metadata_Bow<-subset(sample_data_4, Stream_Type=="Bow River")

otu_rel_abund_Bow <- dplyr::inner_join(ASVseq_ps_2, metadata_Bow, by="Study_ID") %>%
  dplyr::inner_join(., taxonomy_modified, by="Sequences") %>% #joins taxonomy, abundances, and metadata
  dplyr::group_by(Study_ID) %>%
  dplyr::mutate(rel_abund = count / sum(count)) %>% #transforms to relative abundance
  dplyr::ungroup() %>%
  dplyr::select(-count) %>%
  pivot_longer(
    c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Sequences"),
    names_to="level",
    values_to="taxon") 

#Phylum
taxon_rel_abund_Phylum_Bow <- otu_rel_abund_Bow %>%
  dplyr::filter(level=="Phylum") %>%
  dplyr::group_by(Site, sample_Family, Study_ID, taxon) %>%
  dplyr::summarize(rel_abund = 100*sum(rel_abund), .groups="drop") %>%
  drop_na(Site, sample_Family, taxon)

taxon_pool_Phylum_Bow <- taxon_rel_abund_Phylum_Bow %>%
  dplyr::group_by(Site, sample_Family, taxon) %>%
  dplyr::summarize(median=median(rel_abund), .groups="drop") %>%
  dplyr::group_by(taxon) %>%
  dplyr::summarize(pool = max(median) < 25,
            median = median(median),
            .groups="drop")

#Graphing pointrange plot with median and 50% quartiles
Phylum_Bow_graphing_df<- dplyr::inner_join(taxon_rel_abund_Phylum_Bow, taxon_pool_Phylum_Bow, by="taxon") %>%
 dplyr::mutate(taxon = if_else(pool, "Other", taxon)) %>%
  dplyr::group_by(Study_ID, Site, sample_Family, taxon) %>%
  dplyr::summarize(rel_abund = sum(rel_abund),
            median = min(median),
            .groups="drop") %>%
  dplyr::mutate(taxon = factor(taxon),
         taxon = fct_reorder(taxon, median, .desc=T)) %>%
  dplyr::mutate(rel_abund = if_else(rel_abund == 0,
                             2/3 * lod,
                             rel_abund))
Phylum_Bow_graphing_df

## Hydropsychidae ##
hydro_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Hydropsychidae")

hydro_phylum_Bow$taxon<-factor(hydro_phylum_Bow$taxon, levels=c("Other", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Bacteroidota"))

hydro_phylum_Bow$Site<-factor(hydro_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

hydro_rel_abun_plot_Bow <- ggplot(hydro_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

hydro_rel_abun_plot_Bow

#Statistical tests

hist(hydro_phylum_Bow$rel_abund)
shapiro.test(hydro_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

kruskal.test(data=hydro_phylum_Bow, rel_abund ~ Site) #p = 0.9779
dunnTest(data=hydro_phylum_Bow, rel_abund ~ Site) #no sig. diff

#Proteobacteria
proteo_hydro_Bow<-subset(hydro_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_hydro_Bow, rel_abund ~ Site) #p = 0.778 no diff.

#Bacteroidota
bacter_hydro_Bow<-subset(hydro_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_hydro_Bow, rel_abund ~ Site) #p = 0.410 no diff.

#Firmicutes
firm_hydro_Bow<-subset(hydro_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_hydro_Bow, rel_abund ~ Site) #p = 0.0267
dunnTest(data=firm_hydro_Bow, rel_abund ~ Site) #COCH - PMF p=0.0188, z = 3.11
#Policeman Flats has significantly less Firmicutes than Cochrane

#Cyanobacteria
cyano_hydro_Bow<-subset(hydro_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_hydro_Bow, rel_abund ~ Site) #p = 0.000141
dunnTest(data=cyano_hydro_Bow, rel_abund ~ Site) 
#Cochrane sig. lower than all sites except Sunalta
#Sunalta sig. lower than Graves Bridge


## Heptageniidae ##

hep_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Heptageniidae")

hep_phylum_Bow$taxon<-factor(hep_phylum_Bow$taxon, levels=c("Other", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

hep_phylum_Bow$Site<-factor(hep_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

hep_rel_abun_plot_Bow <- ggplot(hep_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

hep_rel_abun_plot_Bow

#Statistical tests

shapiro.test(hep_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_hep_Bow<-subset(hep_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_hep_Bow, rel_abund ~ Site) #p = 0.013 
dunnTest(data=proteo_hep_Bow, rel_abund ~ Site) 
#CUSHBR - PMF p=0.022
#CUSHBR - GRVBR p=0.0489

#Bacteroidota
bacter_hep_Bow<-subset(hep_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_hep_Bow, rel_abund ~ Site) #p = 0.00687 possible diff
dunnTest(data=bacter_hep_Bow, rel_abund ~ Site) 
#CUSHBR - PMF p = 0.00327

#Firmicutes
firm_hep_Bow<-subset(hep_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_hep_Bow, rel_abund ~ Site) #p = 0.014
dunnTest(data=firm_hep_Bow, rel_abund ~ Site) 
#COCH - CUSHBR p = 0.0295
#COCH - GRVBR p = 0.0350
#Cochrane has higher Firmicutes

#Cyanobacteria
cyano_hep_Bow<-subset(hep_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_hep_Bow, rel_abund ~ Site) #p < 0.001, chi-squared: 24.1 
dunnTest(data=cyano_hep_Bow, rel_abund ~ Site) 
#COCH - CUSHBR p < 0.001
#SUN - CUSHBR p = 0.0015

#GRVBR is very close to being sig. diff from COCH


## Chironomidae ##
chiro_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Chironomidae")

chiro_phylum_Bow$taxon<-factor(chiro_phylum_Bow$taxon, levels=c("Other", "Firmicutes", "Bacteroidota", "Proteobacteria", "Cyanobacteria"))

chiro_phylum_Bow$Site<-factor(chiro_phylum_Bow$Site, levels=c("Cochrane", "Cushing Bridge", "Policeman Flats"))

chiro_rel_abun_plot_Bow <- ggplot(chiro_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Cushing Bridge", "Policeman Flats"), values=c("#66CCFF", "#CC9966", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

chiro_rel_abun_plot_Bow

#Statistical tests

shapiro.test(chiro_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_chiro_Bow<-subset(chiro_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_chiro_Bow, rel_abund ~ Site) #p = 0.074 possible difference
dunnTest(data=proteo_chiro_Bow, rel_abund ~ Site) #no diff. 

#Bacteroidota
bacter_chiro_Bow<-subset(chiro_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_chiro_Bow, rel_abund ~ Site) #p = 0.062
dunnTest(data=bacter_chiro_Bow, rel_abund ~ Site) #no diff. 

#Firmicutes
firm_chiro_Bow<-subset(chiro_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_chiro_Bow, rel_abund ~ Site) #p = 0.054
dunnTest(data=firm_chiro_Bow, rel_abund ~ Site) 
#Very close to sig. between CUSHBR - PMF but no diff. 

#Cyanobacteria
cyano_chiro_Bow<-subset(chiro_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_chiro_Bow, rel_abund ~ Site) #p = 0.034 possible diff. 
dunnTest(data=cyano_chiro_Bow, rel_abund ~ Site) 
#No diff. but close btw CUSHBR - PMF and COCH - CUSHBR


## Perlidae ##

perl_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Perlidae")

perl_phylum_Bow$taxon<-factor(perl_phylum_Bow$taxon, levels=c("Other", "Firmicutes", "Cyanobacteria", "Bacteroidota", "Proteobacteria"))

perl_phylum_Bow$Site<-factor(perl_phylum_Bow$Site, levels=c("Cochrane", "Cushing Bridge", "Policeman Flats"))

perl_rel_abun_plot_Bow <- ggplot(perl_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Cushing Bridge", "Policeman Flats"), values=c("#66CCFF", "#CC9966", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

perl_rel_abun_plot_Bow

#Statistical tests

shapiro.test(perl_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_perl_Bow<-subset(perl_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_perl_Bow, rel_abund ~ Site) #p = 0.0429 possible difference
dunnTest(data=proteo_perl_Bow, rel_abund ~ Site) #COCH - CUSHBR p = 0.049 

#Bacteroidota
bacter_perl_Bow<-subset(perl_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_perl_Bow, rel_abund ~ Site) #p = 0.328 no diff 

#Firmicutes
firm_perl_Bow<-subset(perl_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_perl_Bow, rel_abund ~ Site) #p = 0.11

#Cyanobacteria
cyano_perl_Bow<-subset(perl_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_perl_Bow, rel_abund ~ Site) #p = 0.00186 
dunnTest(data=cyano_perl_Bow, rel_abund ~ Site) 
#COCH - CUSHBR
#CUSHBR - PMF


## Trichoptera ##

trichop_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Trichoptera")

trichop_phylum_Bow$taxon<-factor(trichop_phylum_Bow$taxon, levels=c("Other", "Cyanobacteria", "Firmicutes", "Bacteroidota", "Proteobacteria"))

trichop_phylum_Bow$Site<-factor(trichop_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge"))

trichop_rel_abun_plot_Bow <- ggplot(trichop_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

trichop_rel_abun_plot_Bow

#Statistical tests

shapiro.test(trichop_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_trichop_Bow<-subset(trichop_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_trichop_Bow, rel_abund ~ Site) #p = 0.98 

#Bacteroidota
bacter_trichop_Bow<-subset(trichop_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_trichop_Bow, rel_abund ~ Site) #p = 0.638 

#Firmicutes
firm_trichop_Bow<-subset(trichop_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_trichop_Bow, rel_abund ~ Site) #p = 0.605

#Cyanobacteria
cyano_trichop_Bow<-subset(trichop_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_trichop_Bow, rel_abund ~ Site) #p = 0.249
#No diff. in Trichoptera


## Ephemeroptera ##

ephem_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Ephemeroptera")

ephem_phylum_Bow$taxon<-factor(ephem_phylum_Bow$taxon, levels=c("Other", "Bacteroidota", "Firmicutes", "Cyanobacteria", "Proteobacteria"))

ephem_phylum_Bow$Site<-factor(ephem_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

ephem_rel_abun_plot_Bow <- ggplot(ephem_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

ephem_rel_abun_plot_Bow

#Statistical tests

shapiro.test(ephem_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_ephem_Bow<-subset(ephem_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_ephem_Bow, rel_abund ~ Site) #p = 0.010 
dunnTest(data=proteo_ephem_Bow, rel_abund ~ Site) 
#CUSHBR - GRVBR p = 0.0042

#Bacteroidota
bacter_ephem_Bow<-subset(ephem_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_ephem_Bow, rel_abund ~ Site) #p = 0.010
dunnTest(data=bacter_ephem_Bow, rel_abund ~ Site)
#COCH - CUSHBR p =0.019
#CUSHBR - PMF p = 0.0294
#CUSHBR - SUN p = 0.0434

#Firmicutes
firm_ephem_Bow<-subset(ephem_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_ephem_Bow, rel_abund ~ Site) #p = 0.176

#Cyanobacteria
cyano_ephem_Bow<-subset(ephem_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_ephem_Bow, rel_abund ~ Site) #p = 0.0036
dunnTest(data=cyano_ephem_Bow, rel_abund ~ Site)
#CUSHBR - GRVBR p = 0.0035
#CUSHBR - PMF p = 0.037

#Cushing Bridge had the lowest amount of Cyanobacteria in mayfly adults compared to mayfly larvae which had the most at that site. Interesting.

## Diptera ##

dip_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Diptera")

dip_phylum_Bow$taxon<-factor(dip_phylum_Bow$taxon, levels=c("Other", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

dip_phylum_Bow$Site<-factor(dip_phylum_Bow$Site, levels=c("Cochrane", "Cushing Bridge", "Policeman Flats"))

dip_rel_abun_plot_Bow <- ggplot(dip_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Cushing Bridge", "Policeman Flats"), values=c("#66CCFF", "#CC9966", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

dip_rel_abun_plot_Bow

#Statistical tests

shapiro.test(dip_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_dip_Bow<-subset(dip_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_dip_Bow, rel_abund ~ Site) #p = 0.42 

#Bacteroidota
bacter_dip_Bow<-subset(dip_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_dip_Bow, rel_abund ~ Site) #p = 0.369 

#Firmicutes
firm_dip_Bow<-subset(dip_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_dip_Bow, rel_abund ~ Site) #p = 0.366

#Cyanobacteria
cyano_dip_Bow<-subset(dip_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_dip_Bow, rel_abund ~ Site) #p = 0.843 


## Araneidae ##

aran_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Araneidae")

aran_phylum_Bow$taxon<-factor(aran_phylum_Bow$taxon, levels=c("Other", "Firmicutes", "Proteobacteria", "Cyanobacteria", "Bacteroidota"))

aran_phylum_Bow$Site<-factor(aran_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

aran_rel_abun_plot_Bow <- ggplot(aran_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

aran_rel_abun_plot_Bow

#Statistical tests

shapiro.test(aran_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_aran_Bow<-subset(aran_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_aran_Bow, rel_abund ~ Site) #p = 0.010 
dunnTest(data=proteo_aran_Bow, rel_abund ~ Site) 
#No sig. differences. CUSHBR - SUN and GRVBR - SUN very close.

#Bacteroidota
bacter_aran_Bow<-subset(aran_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_aran_Bow, rel_abund ~ Site) #p < 0.001
dunnTest(data=bacter_aran_Bow, rel_abund ~ Site)
#COCH - PMF p = 0.0033
#CUSHBR - PMF p = 0.0033
#GRVBR - PMF p = 0.00699

#Firmicutes
firm_aran_Bow<-subset(aran_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_aran_Bow, rel_abund ~ Site) #p = 0.619

#Cyanobacteria
cyano_aran_Bow<-subset(aran_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_aran_Bow, rel_abund ~ Site) #p < 0.001
dunnTest(data=cyano_aran_Bow, rel_abund ~ Site)
#COCH - PMF p = 0.0129
#COCH - SUN p < 0.001
#CUSHBR - SUN p = 0.0208


## Tetragnathidae ##

tetra_phylum_Bow<-subset(Phylum_Bow_graphing_df, sample_Family=="Tetragnathidae")

tetra_phylum_Bow$taxon<-factor(tetra_phylum_Bow$taxon, levels=c("Other", "Firmicutes", "Bacteroidota", "Cyanobacteria", "Proteobacteria"))

tetra_phylum_Bow$Site<-factor(tetra_phylum_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

tetra_rel_abun_plot_Bow <- ggplot(tetra_phylum_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  coord_trans(x="log10")+
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.001, 0.01, 0.1, 1, 10, 100),
                     labels=c(0.001, 0.01, 0.1, 1, 10, 100)) +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

tetra_rel_abun_plot_Bow

#Statistical tests

shapiro.test(tetra_phylum_Bow$rel_abund) #not normal. Right skewed. Use Kruskal Wallis

#Proteobacteria
proteo_tetra_Bow<-subset(tetra_phylum_Bow, taxon=="Proteobacteria")
kruskal.test(data=proteo_tetra_Bow, rel_abund ~ Site) #p = 0.263

#Bacteroidota
bacter_tetra_Bow<-subset(tetra_phylum_Bow, taxon=="Bacteroidota")
kruskal.test(data=bacter_tetra_Bow, rel_abund ~ Site) #p = 0.172

#Firmicutes
firm_tetra_Bow<-subset(tetra_phylum_Bow, taxon=="Firmicutes")
kruskal.test(data=firm_tetra_Bow, rel_abund ~ Site) #p = 0.139

#Cyanobacteria
cyano_tetra_Bow<-subset(tetra_phylum_Bow, taxon=="Cyanobacteria")
kruskal.test(data=cyano_tetra_Bow, rel_abund ~ Site) #p = 0.216


### Genus level ###

taxon_rel_abund_Genus_Bow <- otu_rel_abund_Bow %>%
  dplyr::filter(level=="Genus") %>%
  dplyr::group_by(Site, sample_Family, Study_ID, taxon) %>%
  dplyr::summarize(rel_abund = 100*sum(rel_abund), .groups="drop") %>%
  drop_na(Site, sample_Family, taxon)

taxon_pool_Genus_Bow <- taxon_rel_abund_Genus_Bow %>%
  dplyr::group_by(Site, sample_Family, taxon) %>%
  dplyr::summarize(median=median(rel_abund), .groups="drop") %>%
  dplyr::group_by(taxon) %>%
  dplyr::summarize(pool = max(median) < 3,
            median = median(median),
            .groups="drop")

#Graphing pointrange plot with median and 50% quartiles
Genus_Bow_graphing_df<- dplyr::inner_join(taxon_rel_abund_Genus_Bow, taxon_pool_Genus_Bow, by="taxon") %>%
 dplyr::mutate(taxon = if_else(pool, "Other", taxon)) %>%
  dplyr::group_by(Study_ID, Site, sample_Family, taxon) %>%
  dplyr::summarize(rel_abund = sum(rel_abund),
            median = min(median),
            .groups="drop") %>%
  dplyr::mutate(taxon = factor(taxon),
         taxon = fct_reorder(taxon, median, .desc=F)) %>%
  dplyr::mutate(rel_abund = if_else(rel_abund == 0,
                             2/3 * lod,
                             rel_abund))
Genus_Bow_graphing_df


## Hydropsychidae ##

hydro_genus_Bow<-subset(Genus_Bow_graphing_df, sample_Family=="Hydropsychidae")

hydro_genus_Bow$Site<-factor(hydro_genus_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

hydro_rel_abun_plot_Bow_genus <- ggplot(hydro_genus_Bow, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  scale_color_manual(name="Site", labels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"), values=c("#66CCFF", "#003399", "#CC9966", "#FF6666", "#990000")) +
  labs(y="Genus", x="Relative Abundance (%)") +
  theme_bw() +
  coord_trans(x="log10")+
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

hydro_rel_abun_plot_Bow_genus

#Statistical tests

kruskal.test(data=hydro_genus_Bow, rel_abund ~ Site) #p = 0.98
dunnTest(data=hydro_genus_Bow, rel_abund ~ Site) #no sig. diff

#### Bow River graphs ####

#Phylum level

hydro_plot<-hydro_rel_abun_plot_Bow + rremove("axis.title")
hep_plot<-hep_rel_abun_plot_Bow + rremove("axis.title")
chiro_plot<-chiro_rel_abun_plot_Bow + rremove("axis.title")
perl_plot<-perl_rel_abun_plot_Bow + rremove("axis.title")
trichop_plot<-trichop_rel_abun_plot_Bow + rremove("axis.title")
ephem_plot<-ephem_rel_abun_plot_Bow + rremove("axis.title")
dip_plot<-dip_rel_abun_plot_Bow + rremove("axis.title")
aran_plot<-aran_rel_abun_plot_Bow + rremove("axis.title")
tetra_plot<-tetra_rel_abun_plot_Bow + rremove("axis.title")

rel_abun_fig_Bow<-ggarrange(hydro_plot, hep_plot, chiro_plot, perl_plot, trichop_plot, ephem_plot, dip_plot, aran_plot, tetra_plot, legend = "top", common.legend = T, labels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathiade"), font.label=list(size=10), vjust = -0, nrow=3, ncol=3)

annotate_figure(rel_abun_fig_Bow, left = textGrob("Phylum", rot = 90, vjust = 1, gp = gpar(cex = 1.3)),
                    bottom = textGrob("Relative Abundance (%)", gp = gpar(cex = 1.3)))


#ggsave(filename="Microbiome_Analysis/Final analysis/Relative abundance/All samples/point range rel abund graph Bow River across taxa.png", height=10, width=10, bg="white")


#### ACWA Streams ####

metadata_ACWA<-subset(sample_data_4, Stream_Type!="Bow River")

otu_rel_abund_ACWA <- dplyr::inner_join(ASVseq_ps_2, metadata_ACWA, by="Study_ID") %>%
  dplyr::inner_join(., taxonomy_modified, by="Sequences") %>% #joins taxonomy, abundances, and metadata
  dplyr::group_by(Study_ID) %>%
  dplyr::mutate(rel_abund = count / sum(count)) %>% #transforms to relative abundance
  dplyr::ungroup() %>%
  dplyr::select(-count) %>%
  pivot_longer(
    c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Sequences"),
    names_to="level",
    values_to="taxon") 

#Phylum
taxon_rel_abund_Phylum_ACWA <- otu_rel_abund_ACWA %>%
  dplyr::filter(level=="Phylum") %>%
  dplyr::group_by(Site, sample_Family, Study_ID, taxon) %>%
  dplyr::summarize(rel_abund = 100*sum(rel_abund), .groups="drop") %>%
  drop_na(Site, sample_Family, taxon)

taxon_pool_Phylum_ACWA <- taxon_rel_abund_Phylum_ACWA %>%
  dplyr::group_by(Site, sample_Family, taxon) %>%
  dplyr::summarize(median=median(rel_abund), .groups="drop") %>%
  dplyr::group_by(taxon) %>%
  dplyr::summarize(pool = max(median) < 3,
            median = median(median),
            .groups="drop")

#Graphing pointrange plot with median and 50% quartiles
Phylum_ACWA_graphing_df<- dplyr::inner_join(taxon_rel_abund_Phylum_ACWA, taxon_pool_Phylum_ACWA, by="taxon") %>%
 dplyr::mutate(taxon = if_else(pool, "Other", taxon)) %>%
  dplyr::group_by(Study_ID, Site, sample_Family, taxon) %>%
  dplyr::summarize(rel_abund = sum(rel_abund),
            median = min(median),
            .groups="drop") %>%
  dplyr::mutate(taxon = factor(taxon),
         taxon = fct_reorder(taxon, median, .desc=T)) %>%
  dplyr::mutate(rel_abund = if_else(rel_abund == 0,
                             2/3 * lod,
                             rel_abund))
Phylum_ACWA_graphing_df

## Hydropsychidae ##

hydro_ACWA_phylum<-subset(Phylum_ACWA_graphing_df, sample_Family=="Hydropsychidae")

hydro_ACWA_phylum$taxon<-factor(hydro_ACWA_phylum$taxon, levels=c("Other", "Cyanobacteria", "Firmicutes", "Proteobacteria", "Bacteroidota"))

hydro_ACWA_phylum$Site<-factor(hydro_ACWA_phylum$Site, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

hydro_rel_abun_plot_ACWA <- ggplot(hydro_ACWA_phylum, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", values=c("#CC9966", "#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

hydro_rel_abun_plot_ACWA

#Statistical tests

#Proteobacteria
proteo_hydro_ACWA<-subset(hydro_ACWA_phylum, taxon=="Proteobacteria")
kruskal.test(data=proteo_hydro_ACWA, rel_abund ~ Site) #p = 0.0767 no diff.
dunnTest(data=proteo_hydro_ACWA, rel_abund ~ Site) #no diff

#Bacteroidota
bacter_hydro_ACWA<-subset(hydro_ACWA_phylum, taxon=="Bacteroidota")
kruskal.test(data=bacter_hydro_ACWA, rel_abund ~ Site) #p = 0.117 no diff.

#Firmicutes
firm_hydro_Bow<-subset(hydro_ACWA_phylum, taxon=="Firmicutes")
kruskal.test(data=firm_hydro_Bow, rel_abund ~ Site) #p = 0.0701
dunnTest(data=firm_hydro_Bow, rel_abund ~ Site) #No diff

#Cyanobacteria
cyano_hydro_Bow<-subset(hydro_ACWA_phylum, taxon=="Cyanobacteria")
kruskal.test(data=cyano_hydro_Bow, rel_abund ~ Site) #p = 0.032
dunnTest(data=cyano_hydro_Bow, rel_abund ~ Site) 
#EXP5 has higher compared to REF Z = 2.55, P = 0.0327


## Baetidae ##

baetid_ACWA_phylum<-subset(Phylum_ACWA_graphing_df, sample_Family=="Baetidae")

baetid_ACWA_phylum$taxon<-factor(baetid_ACWA_phylum$taxon, levels=c("Firmicutes", "Other","Cyanobacteria", "Bacteroidota", "Proteobacteria"))

baetid_ACWA_phylum$Site<-factor(baetid_ACWA_phylum$Site, levels=c("PCR1", "PCR3"), labels=c("EXP5", "EXP15"))

baetid_rel_abun_plot_ACWA <- ggplot(baetid_ACWA_phylum, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", values=c("#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

baetid_rel_abun_plot_ACWA

#Statistical tests

#Proteobacteria
proteo_baetid_ACWA<-subset(baetid_ACWA_phylum, taxon=="Proteobacteria")
kruskal.test(data=proteo_baetid_ACWA, rel_abund ~ Site) #p = 0.355

#Bacteroidota
bacter_baetid_ACWA<-subset(baetid_ACWA_phylum, taxon=="Bacteroidota")
kruskal.test(data=bacter_baetid_ACWA, rel_abund ~ Site) #p = 0.488 no diff.

#Firmicutes
firm_baetid_Bow<-subset(baetid_ACWA_phylum, taxon=="Firmicutes")
kruskal.test(data=firm_baetid_Bow, rel_abund ~ Site) #p = 0.563

#Cyanobacteria
cyano_baetid_Bow<-subset(baetid_ACWA_phylum, taxon=="Cyanobacteria")
kruskal.test(data=cyano_baetid_Bow, rel_abund ~ Site) #p = 0.418


## Ephemeroptera ##

ephem_ACWA_phylum<-subset(Phylum_ACWA_graphing_df, sample_Family=="Ephemeroptera")

ephem_ACWA_phylum$taxon<-factor(ephem_ACWA_phylum$taxon, levels=c("Cyanobacteria", "Other", "Firmicutes", "Proteobacteria", "Bacteroidota"))

ephem_ACWA_phylum$Site<-factor(ephem_ACWA_phylum$Site, levels=c("PCR1", "PCR3"), labels=c("EXP5", "EXP15"))

ephem_rel_abun_plot_ACWA <- ggplot(ephem_ACWA_phylum, aes(y=taxon, x=rel_abund, color=Site)) +
  stat_summary(fun.data=median_hilow, geom = "pointrange",
               fun.args=list(conf.int=0.5),
               position = position_dodge(width=1)) +
  coord_trans(x="log10") +
  scale_x_continuous(limits=c(NA, 100),
                     breaks=c(0.01, 0.1, 1, 10, 100),
                     labels=c(0.01, 0.1, 1, 10, 100)) +
  scale_color_manual(name="Site", values=c("#FF6666", "#990000")) +
  labs(y="Phylum", x="Relative Abundance (%)") +
  theme_bw() +
  rremove("grid")+
  theme(axis.text.y = element_text(color="black"), axis.text.x = element_text(color="black"))

ephem_rel_abun_plot_ACWA

#Statistical tests

#Proteobacteria
proteo_ephem_ACWA<-subset(ephem_ACWA_phylum, taxon=="Proteobacteria")
kruskal.test(data=proteo_ephem_ACWA, rel_abund ~ Site) #p = 0.115

#Bacteroidota
bacter_ephem_ACWA<-subset(ephem_ACWA_phylum, taxon=="Bacteroidota")
kruskal.test(data=bacter_ephem_ACWA, rel_abund ~ Site) #p = 0.4008

#Firmicutes
firm_ephem_Bow<-subset(ephem_ACWA_phylum, taxon=="Firmicutes")
kruskal.test(data=firm_ephem_Bow, rel_abund ~ Site) #p = 0.00865

#Cyanobacteria
cyano_ephem_Bow<-subset(ephem_ACWA_phylum, taxon=="Cyanobacteria")
kruskal.test(data=cyano_ephem_Bow, rel_abund ~ Site) #p = 0.916


#### ACWA graphs ####

#Phylum level

hydro_plot_ACWA<-hydro_rel_abun_plot_ACWA + rremove("axis.title")
baetid_plot_ACWA<-baetid_rel_abun_plot_ACWA + rremove("axis.title")
ephem_plot_ACWA<-ephem_rel_abun_plot_ACWA + rremove("axis.title")

rel_abun_fig_ACWA<-ggarrange(hydro_plot_ACWA, baetid_plot_ACWA, ephem_plot_ACWA, legend = "bottom", common.legend = T, labels = c("Hydropsychidae", "Baetidae", "Ephemeroptera"), font.label=list(size=10), vjust = 2, hjust = -1.6, nrow=1, ncol=3)

annotate_figure(rel_abun_fig_ACWA, left = textGrob("Phylum", rot = 90, vjust = 1, gp = gpar(cex = 1.3)),
                    bottom = textGrob("Relative Abundance (%)", gp = gpar(cex = 1.3)))


#ggsave(filename="Microbiome_Analysis/Final analysis/Relative abundance/All samples/point range rel abund graph ACWA across taxa.png", height=8, width=10, bg="white")


```

### Plotting relative abundance {.tabset}
#### Bow River - faceted across taxa and sites
```{r}
#### Phylum ####

Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
rel_abun_phylum_bow <- speedyseq::tax_glom(Bow_ps, taxrank = "Phylum")
comp_rel_abun_phylum_bow <- transform_sample_counts(rel_abun_phylum_bow, function(x) x/sum(x))

rel_abun_prune_sample_bow <- psmelt(comp_rel_abun_phylum_bow)

sample.df.agg_bow <- aggregate(Abundance ~ Phylum + Site_code + sample_Family + Stage, data = rel_abun_prune_sample_bow, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_bow$Site_code <- factor(sample.df.agg_bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))
sample.df.agg_bow$sample_Family<-factor(sample.df.agg_bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))


#Top 5 phyla relative abundance, all inverts included (Bow River and ACWA combined)
top_phyla_bow <- sample.df.agg_bow %>%
    group_by(Site_code, Phylum) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phyla_bow
top5_bow <- top_phyla_bow$Phylum[1:23]
df0_bow <- sample.df.agg_bow %>%
    mutate(Phylum = fct_other(Phylum, top5_bow))
df0_bow
df0_bow<-aggregate(Abundance~Phylum + Site_code + sample_Family+Stage, data=df0_bow, FUN="sum") #Aggregate again to combine all the 'Other' categories into a single bar on the graph.

df0_bow$Phylum <- factor(df0_bow$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"), labels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"))

df0_bow$sample_Family<-factor(df0_bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

top5phylumplot_bow <- ggplot(df0_bow, aes(x=factor(Site_code, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Phylum)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(1, "cm"),
        legend.text=element_text(size=14), legend.title=element_text(size=14), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "#43464B", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")+
  facet_nested_wrap(~Stage+sample_Family, scales="free_x", nrow=1)
top5phylumplot_bow

#ggsave(filename="Microbiome_ED/Final Analysis/Relative abundance/All samples/Mean relative abundance of top 5 phyla across invertebrate taxa Bow only.png", height=10, width=22)

#### Class Level ####
rel_abun_class <- speedyseq::tax_glom(sample_ps, taxrank = "Class")
rel_abun_class <- transform_sample_counts(rel_abun_class, function(x){x / sum(x)})
rel_abun_all_prune_class = prune_taxa(taxa_sums(rel_abun_class) > 0.05, rel_abun_class) 

rel_abun_prune_sample_class <- psmelt(rel_abun_all_prune_class)

sample.df.agg.class <- aggregate(Abundance ~ Class + Site_code  + sample_Family+Stage, data = rel_abun_prune_sample_class, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg.class$Site <- factor(sample.df.agg.class$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))
sample.df.agg.class$Site_code <- factor(sample.df.agg.class$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3"))


rel_abun_plot_class <- ggplot(sample.df.agg.class, aes(x=Site_code, y=Abundance, fill=Class)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_wrap(.~sample_Family, scale="free_x", nrow=2) + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_discrete(name="Bacterial Class")

rel_abun_plot_class #Too many classes

top_class <- sample.df.agg.class %>%
    group_by(Site_code, Class) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_class
top5class <- top_class$Class[1:30]
top5classdf <- sample.df.agg.class %>%
    mutate(Class = fct_other(Class, top5class))
top5classdf <- aggregate(Abundance ~ Class + Site_code  + sample_Family+Stage, data = top5classdf, FUN = sum)
top5classdf

top5classdf$Class <- factor(top5classdf$Class, levels=c("Alphaproteobacteria", "Bacteroidia", "Gammaproteobacteria", "Bacilli", "Other", "Clostridia", "Cyanobacteriia"))

top5classdf$sample_Family<-factor(top5classdf$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

top5classplot <- ggplot(top5classdf, aes(x=Site_code, y=Abundance, fill=Class)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scales="free_x", nrow=1) + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=6)) +
  theme_bw()+
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Bacterial Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "#43464B", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")
top5classplot

#ggsave(filename="Microbiome_ED/Final Analysis/Relative abundance/All samples/Mean relative abundance of top 5 classes across invertebrates Bow only.png", height=10, width=22)


#### Family Level ####

sample_all_Bow<-subset_samples(sample_ps, Stream_Type=="Bow River")

rel_abun_fam_Bow <- tax_glom(sample_all_Bow, taxrank = "Family")
comp_rel_abun_fam_Bow <- transform_sample_counts(rel_abun_fam_Bow, function(x){x / sum(x)})

rel_abun_prune_sample_fam_Bow <- psmelt(comp_rel_abun_fam_Bow)

sample.df.agg_fam_Bow <- aggregate(Abundance ~ Family + Site_code + sample_Family+Stage, data = rel_abun_prune_sample_fam_Bow, FUN = mean) #aggregate to Phylum level, take the mean

sample.df.agg_fam_Bow$Site <- factor(sample.df.agg_fam_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

sample.df.agg_fam_Bow$Site_code <- factor(sample.df.agg_fam_Bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

rel_abun_plot_fam_Bow <- ggplot(sample.df.agg_fam_Bow, aes(x=Site_code, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >10% ASVs") +
  facet_grid(~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.2, "cm"),
        legend.text=element_text(size=8)) +
  guides(fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)

#Top ~22 families, all inverts included
top_fam_Bow <- sample.df.agg_fam_Bow %>%
    group_by(Site_code, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_Bow <- top_fam_Bow$Family[1:75]
top5famdf_Bow <- sample.df.agg_fam_Bow %>%
    mutate(Family = fct_other(Family, top5fam_Bow))

top5famdf_Bow <- aggregate(Abundance ~ Family + Site_code + sample_Family+Stage, data = top5famdf_Bow, FUN = sum)

top5famdf_Bow$sample_Family<-factor(top5famdf_Bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))


top5famplotBow <- ggplot(top5famdf_Bow, aes(x=factor(Site_code, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=22))+
  scale_fill_manual(name="Family", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5famplotBow 

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/Top 22 families larvae, adults, spiders separated by invertebrate taxa type Bow only.png", height=10, width=22)

#### Genus level ####

rel_abun_genus_Bow <- tax_glom(Bow_ps, taxrank = "Genus")
comp_rel_abun_genus_Bow <- transform_sample_counts(rel_abun_genus_Bow, function(x) x/sum(x))

rel_abun_prune_sample_genus_Bow <- psmelt(comp_rel_abun_genus_Bow)

sample.df.agg_genus_Bow <- aggregate(Abundance ~ Genus + Site_code + sample_Family+Stage, data = rel_abun_prune_sample_genus_Bow, FUN = mean) #aggregate to Phylum level, take the mean

sample.df.agg_genus_Bow$Site <- factor(sample.df.agg_genus_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

sample.df.agg_genus_Bow$Site_code <- factor(sample.df.agg_genus_Bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

sample.df.agg_genus_Bow$Stage<-factor(sample.df.agg_genus_Bow$Stage, levels=c("Larvae", "Adults", "Spiders"))

rel_abun_plot_genus_Bow <- ggplot(sample.df.agg_genus_Bow, aes(x=Site_code, y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >10% ASVs") +
  facet_grid(~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.2, "cm"),
        legend.text=element_text(size=8)) +
  guides(fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)

#Top ~22 genus, all inverts included
top_genus_Bow <- sample.df.agg_genus_Bow %>%
    group_by(Site_code, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_Bow <- top_genus_Bow$Genus[1:63]
top5genusdf_Bow <- sample.df.agg_genus_Bow %>%
    mutate(Genus = fct_other(Genus, top5genus_Bow))

top5genusdf_Bow <- aggregate(Abundance ~ Genus + Site_code + sample_Family+Stage, data = top5genusdf_Bow, FUN = sum)

top5genusdf_Bow$sample_Family<-factor(top5genusdf_Bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Chironomidae ", "Araneidae", "Tetragnathidae"))


top5genusplotBow <- ggplot(top5genusdf_Bow, aes(x=factor(Site_code, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=20, face="italic"), legend.title=element_text(size=20)) +
  guides (fill=guide_legend(nrow=24)) +
  theme(axis.title.y=element_text(size=26),
        axis.text.y=element_text(size=22, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=22, color="black"),
                        axis.title=element_text(size=26))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5genusplotBow 

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/Top 22 genera separated by invertebrate taxa type Bow only, caddis PMF included.png", height=16, width=26)


#Decontaminated data:

sample_all_Bow_decontam<-subset_samples(decontam_ps, Stream_Type=="Bow River")

rel_abun_genus_Bow_decontam <- tax_glom(sample_all_Bow_decontam, taxrank = "Genus")
comp_rel_abun_genus_Bow_decontam <- microbiome::transform(rel_abun_genus_Bow_decontam, "compositional")

rel_abun_prune_sample_genus_Bow_decontam <- psmelt(comp_rel_abun_genus_Bow_decontam)

sample.df.agg_genus_Bow_decontam <- aggregate(Abundance ~ Genus + Site_code + sample_Family+Stage, data = rel_abun_prune_sample_genus_Bow_decontam, FUN = mean) #aggregate to Phylum level, take the mean

#Top ~22 genera, all inverts included
top_genus_Bow_decontam <- sample.df.agg_genus_Bow_decontam %>%
    group_by(Site_code, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_Bow_decontam<- top_genus_Bow_decontam$Genus[1:60]
top5genusdf_Bow_decontam <- sample.df.agg_genus_Bow_decontam %>%
    mutate(Genus = fct_other(Genus, top5genus_Bow_decontam))
top5genusdf_Bow_decontam<-aggregate(Abundance~ Genus + Site_code + sample_Family+Stage, data = top5genusdf_Bow_decontam, FUN = sum)

top5genusdf_Bow_decontam$Site_code <- factor(top5genusdf_Bow_decontam$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

top5genusdf_Bow_decontam$sample_Family<-factor(top5genusdf_Bow_decontam$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Chironomidae ", "Araneidae", "Tetragnathidae"))

top5genusdf_Bow_decontam$Stage<-factor(top5genusdf_Bow_decontam$Stage, levels=c("Larvae", "Adults", "Spiders"))

top5genusplot_Bow_decontam <- ggplot(top5genusdf_Bow_decontam, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5genusplot_Bow_decontam

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/Top 22 genera separated by invertebrate taxa type Bow and ACWA decontaminated caddis PMF included.png", height=10, width=24)


#Both Bow River graphs combined

top5genusplotBow_1<- top5genusplotBow + rremove("xlab") +rremove("ylab")
top5genusplotBow_decontam_1<- top5genusplot_Bow_decontam + rremove("xlab") +rremove("ylab")

Bow_arrange<- ggarrange(top5genusplotBow_1, top5genusplotBow_decontam_1, ncol=1, nrow=2, align="v")

annotate_figure(Bow_arrange, left = textGrob("Mean Relative Abundance", rot = 90, vjust = 1, gp = gpar(cex = 2.5)), bottom = textGrob("Site", gp = gpar(cex = 2.5)), bgcolor("white"))

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/Bow decontam vs non decontam caddis PMF included.png", height=12, width=20, bg="white")

```

#### ACWA - faceted across taxa and sites
```{r}
#### Phylum ####
#ACWA only
ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")
rel_abun_phylum_ACWA <- speedyseq::tax_glom(ACWA_ps, taxrank = "Phylum")
comp_rel_abun_phylum_ACWA <- transform_sample_counts(rel_abun_phylum_ACWA, function(x) x/sum(x))

rel_abun_prune_sample_ACWA <- psmelt(comp_rel_abun_phylum_ACWA)

sample.df.agg.ACWA <- aggregate(Abundance ~ Phylum + Site_code + sample_Family + Stage , data = rel_abun_prune_sample_ACWA, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg.ACWA$Site_code <- factor(sample.df.agg.ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))
sample.df.agg.ACWA$sample_Family<-factor(sample.df.agg.ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

#Top 5 phyla relative abundance, all inverts included (Bow River and ACWA combined)
top_phyla_ACWA <- sample.df.agg.ACWA %>%
    group_by(Site_code, Phylum) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phyla_ACWA
top5_ACWA <- top_phyla_ACWA$Phylum[1:13]
df0_ACWA <- sample.df.agg.ACWA %>%
    mutate(Phylum = fct_other(Phylum, top5_ACWA))
df0_ACWA<-aggregate(Abundance~Phylum+sample_Family+Site_code+Stage, data=df0_ACWA, FUN="sum")

df0_ACWA$Phylum <- factor(df0_ACWA$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Deferribacterota", "Other", "Proteobacteria"), labels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Deferribacterota", "Other", "Proteobacteria"))

top5phylumplot_ACWA <- ggplot(df0_ACWA, aes(x=Site_code, y=Abundance, fill=Phylum)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  facet_nested_wrap(~Stage+sample_Family, scales="free_x", nrow=1)+ 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(1, "cm"),
        legend.text=element_text(size=16), legend.title=element_text(size=16), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_markdown(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=22))+
  scale_fill_manual(name="Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "#43464B", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("ACWA Stream")
top5phylumplot_ACWA

#ggsave(filename="Microbiome_ED/Final Analysis/Relative abundance/All samples/Mean relative abundance of top 5 phyla across invertebrate taxa ACWA only.png", height=10, width=22)
#### Family ####

sample_all_ACWA<-subset_samples(sample_ps, Stream_Type!="Bow River")

rel_abun_fam_ACWA <- tax_glom(sample_all_ACWA, taxrank = "Family")
comp_rel_abun_fam_ACWA <- transform_sample_counts(rel_abun_fam_ACWA, function(x) x/sum(x))

rel_abun_prune_sample_fam_ACWA <- psmelt(comp_rel_abun_fam_ACWA)

sample.df.agg_fam_ACWA <- aggregate(Abundance ~ Family + Site_code + sample_Family+Stage, data = rel_abun_prune_sample_fam_ACWA, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_fam_ACWA$Site <- factor(sample.df.agg_fam_ACWA$Site, levels=c("BRR2", "PCR1", "PCR3"))
sample.df.agg_fam_ACWA$sample_Family<-factor(sample.df.agg_fam_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

rel_abun_plot_fam_ACWA <- ggplot(sample.df.agg_fam_ACWA, aes(x=Site_code, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >10% ASVs") +
  facet_grid(~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.2, "cm"),
        legend.text=element_text(size=8)) +
  guides(fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)

#Top ~22 families, all inverts included
top_fam_ACWA <- sample.df.agg_fam_ACWA %>%
    group_by(Site_code, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_ACWA
top5fam_ACWA <- top_fam_ACWA$Family[1:53]
top5famdf_ACWA <- sample.df.agg_fam_ACWA %>%
    mutate(Family = fct_other(Family, top5fam_ACWA))


top5famdf_ACWA <- aggregate(Abundance ~ Family + Site_code + sample_Family+Stage, data = top5famdf_ACWA, FUN = sum)


top5famplotACWA<- ggplot(top5famdf_ACWA, aes(x=factor(Site_code, labels = c("BRR2", "PCR1", "PCR3")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=22))+
  scale_fill_manual(name="Family", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5famplotACWA

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/Top 22 families Hydropsychidae, Baetidae ACWA.png", height=10, width=22)

#### Genus ####
ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")
rel_abun_genus_ACWA <- tax_glom(ACWA_ps, taxrank = "Genus")
comp_rel_abun_genus_ACWA <- microbiome::transform(rel_abun_genus_ACWA, "compositional")

rel_abun_prune_sample_genus_ACWA <- psmelt(comp_rel_abun_genus_ACWA)

sample.df.agg_genus_ACWA <- aggregate(Abundance ~ Genus + Site + sample_Family+Stage, data = rel_abun_prune_sample_genus_ACWA, FUN = mean) #aggregate to Phylum level, take the mean

sample.df.agg_genus_ACWA$sample_Family<-factor(sample.df.agg_genus_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

sample.df.agg_genus_ACWA$Site<-factor(sample.df.agg_genus_ACWA$Site, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

#Top ~23 genera
top_genus_ACWA <- sample.df.agg_genus_ACWA %>%
    group_by(Site, Genus) %>% #groups together the abundances by genera of bacteria
    summarize(Mean = mean(Abundance)) %>% #takes the mean abundance of each genus per sample
    arrange(-Mean) #arrange by decreasing mean abundance per genus of bacteria
top_genus_ACWA
top5genus_ACWA <- top_genus_ACWA$Genus[1:44] #Specifies the number of lines of data to take
top5genus_ACWA
top5genusdf_ACWA <- sample.df.agg_genus_ACWA %>%
    mutate(Genus = fct_other(Genus, top5genus_ACWA)) 
top5genusdf_ACWA <- aggregate(Abundance ~ Genus + Site + sample_Family+Stage, data = top5genusdf_ACWA, FUN = sum)
    

top5genusplotACWA <- ggplot(top5genusdf_ACWA, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=16, face="italic"), legend.title=element_text(size=16)) +
  guides (fill=guide_legend(nrow=24)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=20, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, vjust=1, size=20, color="black"),
        axis.title.x=element_text(size=22, vjust= -1))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("ACWA Stream")+
  rremove("grid")
top5genusplotACWA

ggsave(filename="microbiome_analysis/Final analysis/Relative abundance/All samples/Top 22 genera ACWA across inverts.png", height=10, width=22)

#ACWA decontam

ACWA_ps_decontam<-subset_samples(decontam_ps, Stream_Type!="Bow River")
rel_abun_genus_ACWA_decontam <- tax_glom(ACWA_ps_decontam, taxrank = "Genus")
comp_rel_abun_genus_ACWA_decontam <- microbiome::transform(rel_abun_genus_ACWA_decontam, "compositional")

rel_abun_prune_sample_genus_ACWA_decontam <- psmelt(comp_rel_abun_genus_ACWA_decontam)

sample.df.agg_genus_ACWA_decontam <- aggregate(Abundance ~ Genus + Site + sample_Family+Stage, data = rel_abun_prune_sample_genus_ACWA_decontam, FUN = mean) #aggregate to Phylum level, take the mean

sample.df.agg_genus_ACWA_decontam$sample_Family<-factor(sample.df.agg_genus_ACWA_decontam$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"), labels=c("Hydropsychidae", "Baetidae", "Baetidae "))

sample.df.agg_genus_ACWA_decontam$Site<-factor(sample.df.agg_genus_ACWA_decontam$Site, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))


sample.df.agg_genus_ACWA_decontam$Stage<-factor(sample.df.agg_genus_ACWA_decontam$Stage, levels=c("Larvae", "Adults"))

#Top ~23 genera
top_genus_ACWA_decontam <- sample.df.agg_genus_ACWA_decontam %>%
    group_by(Site, Genus) %>% #groups together the abundances by genera of bacteria
    summarize(Mean = mean(Abundance)) %>% #takes the mean abundance of each genus per sample
    arrange(-Mean) #arrange by decreasing mean abundance per genus of bacteria
top5genus_ACWA_decontam <- top_genus_ACWA_decontam$Genus[1:44] #Specifies the number of lines of data to take
top5genusdf_ACWA_decontam <- sample.df.agg_genus_ACWA_decontam %>%
    mutate(Genus = fct_other(Genus, top5genus_ACWA_decontam)) 
top5genusdf_ACWA_decontam <- aggregate(Abundance ~ Genus + Site + sample_Family+Stage, data = top5genusdf_ACWA_decontam, FUN = sum)
    

top5genusplotACWA_decontam <- ggplot(top5genusdf_ACWA_decontam, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=16, face="italic"), legend.title=element_text(size=16)) +
  guides (fill=guide_legend(nrow=24)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=20, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, vjust=1, size=20, color="black"),
        axis.title.x=element_text(size=22, vjust= -1))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("ACWA Stream")+
  rremove("grid")
top5genusplotACWA_decontam

#Both ACWA graphs combined

top5genusplotACWA_1<- top5genusplotACWA + rremove("xlab") +rremove("ylab")
top5genusplotACWA_decontam_1<- top5genusplotACWA_decontam + rremove("xlab") +rremove("ylab")

ACWA_arrange<- ggarrange(top5genusplotACWA_1, top5genusplotACWA_decontam_1, ncol=1, nrow=2, align="v", common.legend = TRUE, legend = "right")

annotate_figure(ACWA_arrange, left = textGrob("Mean Relative Abundance", rot = 90, vjust = 0.5, gp = gpar(cex = 2.5)), bottom = textGrob("ACWA Stream", gp = gpar(cex = 2.5), hjust=0.7), bgcolor("white"))

#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/All samples/ACWA decontam vs non decontam.png", height=10, width=18, bg="white")
```

#### Cyanobacteria
```{r}
Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")

cyano_Bow<-subset_taxa(Bow_ps, Phylum=="Cyanobacteria")
cyano_Bow<-prune_taxa(taxa_sums(cyano_Bow)>0, cyano_Bow)
cyano_Bow #248 taxa, 284 samples
cyano_ACWA<-subset_taxa(ACWA_ps, Phylum=="Cyanobacteria")
cyano_ACWA<-prune_taxa(taxa_sums(cyano_ACWA)>0, cyano_ACWA)
cyano_ACWA #148 taxa, 55 samples

## Filtering for the cyanotoxin producing genera (Gugger et al., 2023 ; Millar et al., 2020; Zanchett and Oliveira-Filho, 2013) ##

cyanotox_Bow<-subset_taxa(Bow_ps, Genus=="Dolichospermum" | Genus=="Anabaena" | Genus=="Planktothrix NIVA-CYA 15" | Genus=="Synechocytis" | Genus=="Synechococcus" | Genus=="Microcystis PCC-7914" | Genus=="Aphanizomenon MDT14a" | Genus=="Schizothrix LEGE 07164" | Genus=="Chroococus" | Genus=="Hydrocoleum" | Genus=="Aphanizomenon flos-aquae" | Genus=="Aphanizomenon" | Genus=="Fischerella" | Genus=="Planktothrix" | Genus=="Nostoc" | Genus=="Anabaenopsis" | Genus=="Nodularia" | Genus=="Cylindrospermopsis raciborskii" | Genus=="Aphanizomenon ovalisporum" | Genus=="Aphanizomenon zflos-aquae" | Genus=="Alexandrium" | Genus=="Pyrodinium" | Genus=="Gymnodinium" | Genus=="Lyngbya wollei" | Genus=="Lyngbya")
cyanotox_Bow<-prune_taxa(taxa_sums(cyanotox_Bow) > 0, cyanotox_Bow)
cyanotox_Bow #2 taxa, 284 samples
sort(table(tax_table(cyanotox_Bow)[,6])) #1 genera: Planktothrix NIVA-CYA 15

cyanotox_ACWA<-subset_taxa(ACWA_ps, Genus=="Dolichospermum" | Genus=="Anabaena" | Genus=="Planktothrix NIVA-CYA 15" | Genus=="Synechocytis" | Genus=="Synechococcus" | Genus=="Microcystis PCC-7914" | Genus=="Aphanizomenon MDT14a" | Genus=="Schizothrix LEGE 07164" | Genus=="Chroococus" | Genus=="Hydrocoleum" | Genus=="Aphanizomenon flos-aquae" | Genus=="Aphanizomenon" | Genus=="Fischerella" | Genus=="Planktothrix" | Genus=="Nostoc" | Genus=="Anabaenopsis" | Genus=="Nodularia" | Genus=="Cylindrospermopsis raciborskii" | Genus=="Aphanizomenon ovalisporum" | Genus=="Aphanizomenon zflos-aquae" | Genus=="Alexandrium" | Genus=="Pyrodinium" | Genus=="Gymnodinium" | Genus=="Lyngbya wollei" | Genus=="Lyngbya")
cyanotox_ACWA<-prune_taxa(taxa_sums(cyanotox_ACWA) > 0, cyanotox_ACWA)
cyanotox_ACWA #3 taxa, 55 samples
sort(table(tax_table(cyanotox_ACWA)[,6])) #1 genera: Planktothrix NIVA-CYA 15

#### Bow River only ####

cyano_genus_Bow <- tax_glom(cyano_Bow, taxrank = "Genus")
cyano_genus_Bow #16 genera

rel_abun_cyano_genus_Bow <- transform_sample_counts(cyano_genus_Bow, function(x){x / sum(x)})

cyano_genus_Bow_df <- psmelt(rel_abun_cyano_genus_Bow)

cyano_genus_Bow_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = cyano_genus_Bow_df, FUN = mean) 


cyano_genus_Bow_agg$Site_code <- factor(cyano_genus_Bow_agg$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

cyano_genus_Bow_agg$Stage<-factor(cyano_genus_Bow_agg$Stage, levels=c("Larvae", "Adults", "Spiders"))

cyano_genus_Bow_agg$sample_Family<-factor(cyano_genus_Bow_agg$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Chironomidae ", "Araneidae", "Tetragnathidae"))

ggplot(cyano_genus_Bow_agg, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage + sample_Family, scales="free_x", nrow=1) +  
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=24, face="italic"), legend.title=element_text(size=24)) +
  guides (fill=guide_legend(nrow=24)) +
  theme(axis.title.y=element_text(size=26),
        axis.text.y=element_text(size=22, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=22, color="black"),
                        axis.title=element_text(size=26))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")

#ggsave(filename="microbiome_analysis/Final analysis/Relative abundance/Cyanobacteria/Cyanobacteria genera across taxa and sites.png", height=12, width=26)

cyano_Bow_df = data.frame(Genus = tax_table(cyano_genus_Bow)[,"Genus"], Mean = rowMeans(otu_table(cyano_genus_Bow)), row.names = NULL)
cyano_Bow_df = cyano_Bow_df[order(-cyano_Bow_df$Mean),]

cyano_Bow_df$percentage <- (cyano_Bow_df$Mean/sum(cyano_Bow_df$Mean)*100)


#### ACWA streams only ####
cyano_genus_ACWA <- tax_glom(cyano_ACWA, taxrank = "Genus")
cyano_genus_ACWA #12 genera
rel_abun_cyano_genus_ACWA <- transform_sample_counts(cyano_genus_ACWA, function(x){x / sum(x)})

cyano_genus_ACWA_df <- psmelt(rel_abun_cyano_genus_ACWA)

cyano_genus_ACWA_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = cyano_genus_ACWA_df, FUN = mean) 


cyano_genus_ACWA_agg$Site_code <- factor(cyano_genus_ACWA_agg$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

cyano_genus_ACWA_agg$Stage<-factor(cyano_genus_ACWA_agg$Stage, levels=c("Larvae", "Adults"))

cyano_genus_ACWA_agg$sample_Family<-factor(cyano_genus_ACWA_agg$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

ggplot(cyano_genus_ACWA_agg, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1) + 
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=24, face="italic"), legend.title=element_text(size=24)) +
  guides (fill=guide_legend(nrow=24)) +
  theme(axis.title.y=element_text(size=26),
        axis.text.y=element_text(size=22, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=22, color="black"),
                        axis.title=element_text(size=26))+
  scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")

#ggsave(filename="microbiome_analysis/Final analysis/Relative abundance/Cyanobacteria/Cyanobacteria genera ACWA across taxa and sites.png", height=14, width=26)

cyano_ACWA_df = data.frame(Genus = tax_table(cyano_genus_ACWA)[,"Genus"], Mean = rowMeans(otu_table(cyano_genus_ACWA)), row.names = NULL)
cyano_ACWA_df = cyano_ACWA_df[order(-cyano_ACWA_df$Mean),]

cyano_ACWA_df$percentage <- (cyano_ACWA_df$Mean/sum(cyano_ACWA_df$Mean)*100)
```

# Rarefying the data

Rarefying is done to allow comparisons between samples with high versus low sequencing depth. It involves subsampling the samples to an even number of reads for diversity comparisons. The sequencing depth has to be large enough that most samples retain their diversity, otherwise results can become biased. Some people don't agree with rarefying because it removes a lot of the data and can bias results. I will try diversity analyses with and without rarefying to see if there is a difference to the interpretation of results. 

Papers that support rarefying/repeated rarefying: 
-Cameron et al., 2021. Scientific reports.

Papers that do not support rarefaction:
-McCurdie & Holmes, 2014. PLOS Computational Biology.
-Willis, 2019. Frontiers in Microbiology. 


``` {r Rarefying the data}
#Rarefying to the minimum sequencing depth
#Rarefies to an even sequencing depth (usually the minimum read so that no sequences are removed)
(ps_rare <- phyloseq::rarefy_even_depth(sample_ps, rngseed = 123, replace = FALSE)) 
#4420 taxa, 339 samples
#Rarefying to the minimum sample depth removed 102 ASVs (not many) because we already filtered out the rare bacteria.

head(phyloseq::sample_sums(ps_rare)) #2283 bacterial taxa subsampled per sample. Cuts a lot of sequencing depth but some people argue it is better to retain more samples rather than a larger sequencing depth. 
```

# Alpha Diversity 

Observed richness is the number of unique ASVs per sample (species richness). Count data ranging from 0 - infinity.

Shannon diversity is a measure of the richness and relative abundance (evenness) of bacteria in a sample (H=[(pi)log(pi)]). Usually ranges between 1.5 - 3.5, rarely above 4.5.

Simpson diversity is a measure of the richness and relative abundance (evenness) of bacteria in a sample (D=n i(ni1)/N(N1)). Ranges between 0-1 (1 being the most diverse, 0 being not diverse). 

## Non-rarefied alpha diversity {.tabset}

From Paul McCurdie on GitHub: You do not want to transform or filter your data before estimating richness, other than quality assurance filtering that would remove non-target sequences. Usually that sort of filtering would be done on the sequence data before it is in the form of a contingency table (table of counts). Basically, the very first table of counts that you have in your workflow is probably the one that you want to use for estimate_richness and plot_richness. There are several different alpha diversity estimators supported in the estimate_richness function, and they all incorporate library size into their estimates in different ways... Except for the "Observed" option, which is simply showing you graphically the OTUs that were observed at least once in each sample. To be clear, the other methods require that you use raw counts. Do not use rarefied counts for the Chao-I estimate, or the Shannon index, for example. The "Observed" option is not a method at all, just showing you what is in your data as you provided. If you transform your data, especially if you rarefy, and then you want to estimate richness, the "Observed" result is now the only one still available to you at all. Rarefying reduces the precision with which you would estimate the diversity in the first place, and so this generally shouldn't be done. All alpha-diversity indices/estimates are aware of differences in sample size (library size, number of reads in this case), because this has always been a problem when attempting to count things, even trees in a forest (see Sanders original paper describing rarefaction, a different technique than rarefying, if rarefying can be called a technique).

### Bow River - Shannon index faceted by taxa and sites

```{r}
Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")

adiv_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(Bow_ps, measures = "Shannon"),
  "Family" = phyloseq::sample_data(Bow_ps)$Family,
  "Stage" = phyloseq::sample_data(Bow_ps)$Stage,
  "Site" = phyloseq::sample_data(Bow_ps)$Site)
head(adiv_Bow)


#Shannon diversity across taxa
shan.plot.taxa<-ggplot(adiv_Bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Shannon, fill=Site))+
  facet_nested_wrap(~Stage+ factor(Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scales= "free_x", nrow=1)+
  geom_boxplot() +
  labs(x = "Site", y = "Shannon Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
        strip.text.x = element_text(size=20), 
        axis.text.y = element_text(size=16, color="black"), axis.title.x = element_text(size=18), axis.title.y = element_text(size=18))+
  rremove("grid")+
  scale_fill_brewer(palette="Accent")

shan.plot.taxa

#ggsave(filename="Microbiome_ED/Final analysis/Alpha Diversity Graphs/Non rarefied shannon diversity Bow River.png", height=12, width=22)

#Shannon diversity is slightly higher at PMF in most taxa. 
#Spiders have a lower Shannon diversity than insects. Larvae have the highest, adults are more variable. 


                                    ### Statistical tests ###


#Shannon diversity

shapiro.test(adiv_Bow$Shannon) #Not normal, use Kruskal-Wallis tests. 

#Kruskal-Wallis tests:

#Life stage (larvae, adults, spiders)
kruskal.test(Shannon~Stage, data=adiv_Bow) #p<0.001
dunnTest(Shannon~Stage, data=adiv_Bow) #All diff from each other

#Invertebrate Taxa
kruskal.test(Shannon~Family, data=adiv_Bow) #p<0.001
dunnTest(Shannon~Family, data=adiv_Bow)
#Chironomidae - Ephemeroptera  
#Araneidae - Heptageniidae
#Diptera - Heptageniidae
#Ephemeroptera - Heptageniidae 
#Araneidae - Hydropsychidae 
#Chironomidae - Hydropsychidae 
#Diptera - Hydropsychidae 
#Ephemeroptera - Hydropsychidae 
#Araneidae - Perlidae 
#Diptera - Perlidae 
#Ephemeroptera - Perlidae 
#Chironomidae - Tetragnathidae
#Heptageniidae - Tetragnathidae  
#Hydropsychidae - Tetragnathidae 
#Perlidae - Tetragnathidae 
#Heptageniidae - Trichoptera  
#Hydropsychidae - Trichoptera  
#Tetragnathidae - Trichoptera 


#Difference across sites: (have to separate by invert taxa)

#Larval Hydropsychidae 
adiv_hydro_Bow<-subset(adiv_Bow, Family=="Hydropsychidae")
shapiro.test(adiv_hydro_Bow$Shannon) #Normal dist. p = 0.069
leveneTest(Shannon~Site, data=adiv_hydro_Bow) #Homogeneity. p = 0.257.
#Use ANOVA
summary(hydro_aov<-aov(Shannon~Site, data=adiv_hydro_Bow)) #p<0.001
TukeyHSD(hydro_aov)
#GRVBR - COCH
#PMF - COCH
#GRVBR - SUN
#PMF - SUN
#GRVBR - CUSH
#PMF - CUSH

#Might be an effect of effluent exposure on the alpha diversity in larval Hydropsychidae because effluent impacted sites tend to have higher alpha diversity than all upstream comparisons. 

#Adult Trichoptera
div.df.Bow.Tricho<-subset(adiv_Bow, Family=="Trichoptera")
shapiro.test(div.df.Bow.Tricho$Shannon) #Not normal
leveneTest(Shannon~Site, data=div.df.Bow.Tricho) #Heterogeneity
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.Tricho) #p= 0.753
dunnTest(Shannon~Site, data=div.df.Bow.Tricho) #no significant diff. across sites.

#Larval Heptageniidae (order Ephemeroptera)
div.df.Bow.Hep<-subset(adiv_Bow, Family=="Heptageniidae")
shapiro.test(div.df.Bow.Hep$Shannon) #Normal dist. p = 0.286
leveneTest(Shannon~Site, data=div.df.Bow.Hep) #Homogeneity
#Use ANOVA
summary(hep_aov<-aov(Shannon~Site, data=div.df.Bow.Hep)) #p=0.037
TukeyHSD(hep_aov)
#COCH - PMF

#Adult Ephemeroptera
div.df.Bow.Ephem<-subset(adiv_Bow, Family=="Ephemeroptera")
shapiro.test(div.df.Bow.Ephem$Shannon) #p=0.012, not normal
leveneTest(Shannon~Site, data=div.df.Bow.Ephem) #Homogeneity p = 0.49
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.Ephem) #p=0.129
dunnTest(Shannon~Site, data=div.df.Bow.Ephem) #No sig diff across sites

#Larval Chironomidae (Order Diptera)
div.df.Bow.chiro<-subset(adiv_Bow, Family=="Chironomidae")
shapiro.test(div.df.Bow.chiro$Shannon) #0.366 Normal
leveneTest(Shannon~Site, data=div.df.Bow.chiro) #Homogeneity. 
#Use AnOVA
summary(chiro_aov<-aov(Shannon~Site, data=div.df.Bow.chiro)) #p=0.261
TukeyHSD(chiro_aov)
#No differences across sites

#Adult Chironomidae (Order Diptera)
div.df.Bow.dip<-subset(adiv_Bow, Family=="Diptera")
shapiro.test(div.df.Bow.dip$Shannon) #p=0.166 Normal
leveneTest(Shannon~Site, data=div.df.Bow.dip) #0.605. Homogeneity.
#Use ANOVA
summary(dip_aov<-aov(Shannon~Site, data=div.df.Bow.dip)) #p=0.652
TukeyHSD(dip_aov)
#No differences across sites

#Larval Perlidae (Order Plecoptera)
div.df.Bow.perl<-subset(adiv_Bow, Family=="Perlidae")
shapiro.test(div.df.Bow.perl$Shannon) #0.089 normal
leveneTest(Shannon~Site, data=div.df.Bow.perl) #0.04. Heterogeneity
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.perl) #p=0.211
dunnTest(Shannon~Site, data=div.df.Bow.perl) #No sig diff across sites

summary(perl_aov<-aov(Shannon~Site, data=div.df.Bow.perl)) #p=0.106
TukeyHSD(perl_aov) #ANOVA also indicates no differences. 

#Araneidae
div.df.Bow.aran<-subset(adiv_Bow, Family=="Araneidae")
shapiro.test(div.df.Bow.aran$Shannon) #0.407. Normal
leveneTest(Shannon~Site, data=div.df.Bow.aran) #0.75 homogeneity
#Use ANOVA
summary(aran_aov<-aov(Shannon~Site, data=div.df.Bow.aran)) #p=0.365
TukeyHSD(aran_aov)
#No differences across sites

#Tetragnathidae
div.df.Bow.tetra<-subset(adiv_Bow, Family=="Tetragnathidae")
shapiro.test(div.df.Bow.tetra$Shannon) #P<0.001 not normal
#Use KW
kruskal.test(Shannon~Site, data=div.df.Bow.tetra) #p=0.26
dunnTest(Shannon~Site, data=div.df.Bow.tetra) #No differences across sites 
```

### ACWA - Shannon index faceted by taxa and sites

```{r}
ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")

adiv_ACWA <- data.frame(
  "Shannon" = phyloseq::estimate_richness(ACWA_ps, measures = "Shannon"),
  "Family" = phyloseq::sample_data(ACWA_ps)$Family,
  "Stage" = phyloseq::sample_data(ACWA_ps)$Stage,
  "Site" = phyloseq::sample_data(ACWA_ps)$Site)
head(adiv_ACWA)

#Shannon index
shan.plot.taxa.ACWA<-ggplot(adiv_ACWA, aes(x=factor(Site, labels=c("REF", "EXP5", "EXP15")), y=Shannon, fill=Site))+
  facet_nested_wrap(~Stage+ factor(Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera")), scales= "free_x", nrow=1)+
  geom_boxplot() +
  labs(x = "Site", y = "Shannon Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
        strip.text.x = element_text(size=20), 
        axis.text.y = element_text(size=16, color="black"), axis.title.x = element_text(size=18), axis.title.y = element_text(size=18))+
  rremove("grid")+
  scale_fill_manual(values=c("#7E8071", "#EDBC81", "#A76B56"))

shan.plot.taxa.ACWA

#ggsave(filename="Microbiome_ED/Final analysis/Alpha Diversity Graphs/Non rarefied shannon diversity ACWA.png", height=12, width=22)


                                    ### Statistical tests ###

#Shannon diversity

shapiro.test(adiv_ACWA$Shannon) #Not normal

#Kruskal-Wallis tests:

#Test diff btw larvae and adults
wilcox.test(Shannon~Stage, data=adiv_ACWA) #W = 594, p < 0.001
kruskal.test(Shannon~Stage, data=adiv_ACWA) #p<0.001
#Significant difference in Shannon index btw larvae and adults 

#Diff btw families
kruskal.test(Shannon~Family, data=adiv_ACWA)#p<0.001
dunnTest(Shannon~Family, data=adiv_ACWA) #All diff from each other

#Hydropsychidae
hydro_ACWA<-subset(adiv_ACWA, Family=="Hydropsychidae")
shapiro.test(hydro_ACWA$Shannon) # normal
leveneTest(data=hydro_ACWA, Shannon~Site) #0.197. homogeneity
#Use ANOVA
summary(hydro_ACWA_aov<-aov(data=hydro_ACWA, Shannon~Site)) #0.313 no diff
#No difference between streams

#Baetidae
baetid_ACWA<-subset(adiv_ACWA, Family=="Baetidae")
shapiro.test(baetid_ACWA$Shannon) # normal
leveneTest(data=baetid_ACWA, Shannon~Site) #0.683. homogeneity
#Use t test
t.test(data=baetid_ACWA, Shannon~Site) #p=0.791
#No difference between streams

#Ephemeroptera
ephem_ACWA<-subset(adiv_ACWA, Family=="Ephemeroptera")
shapiro.test(ephem_ACWA$Shannon) #Not normal
#Use Wilcoxon test
wilcox.test(data=ephem_ACWA, Shannon~Site) # W = 19, p = 0.195
#No difference between streams
```

### Calculating median +/- IQR Shannon Index

```{r Calculating median +/- IQR Shannon diversity for each taxa at each site}

adiv_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(Bow_ps, measures = "Shannon"),
  "Observed" = phyloseq::estimate_richness(Bow_ps, measures = "Observed"),
  "Family" = phyloseq::sample_data(Bow_ps)$Family,
  "Stage" = phyloseq::sample_data(Bow_ps)$Stage,
  "Site" = phyloseq::sample_data(Bow_ps)$Site)
head(adiv_Bow)

#All Bow River larvae combined
lar_ps_Bow<-subset_samples(sample_ps, Stream_Type=="Bow River" & Stage=="Larvae")

adiv_lar_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(lar_ps_Bow, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(lar_ps_Bow)$Stream_Type,
  "Site" = phyloseq::sample_data(lar_ps_bow)$Site)

adiv_lar_Bow %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_lar_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

#All Bow River adults combined
ad_Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River" & Stage=="Adults")

adiv_ad_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(ad_Bow_ps, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(ad_Bow_ps)$Stream_Type,
  "Site" = phyloseq::sample_data(ad_Bow_ps)$Site)

adiv_ad_Bow %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_ad_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

#All Bow River spiders combined
spiders_ps<-subset_samples(sample_ps, Stage=="Spiders" & Stream_Type=="Bow River")

adiv_spiders <- data.frame(
  "Shannon" = phyloseq::estimate_richness(spiders_ps, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(spiders_ps)$Stream_Type,
  "Site" = phyloseq::sample_data(spiders_ps)$Site)

adiv_spiders %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_ad_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Larval Hydropsychidae ##

hydro_lar<-subset_samples(sample_ps, Family=="Hydropsychidae")

adiv_hydro <- data.frame(
  "Shannon" = phyloseq::estimate_richness(hydro_lar, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(hydro_lar)$Stream_Type,
  "Site" = phyloseq::sample_data(hydro_lar)$Site)
head(adiv_hydro)

adiv_hydro %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_hydro %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

#Bow River
hydro_Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River" & Family=="Hydropsychidae")

adiv_hydro_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(hydro_Bow_ps, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(hydro_Bow_ps)$Stream_Type,
  "Site" = phyloseq::sample_data(hydro_Bow_ps)$Site)
head(adiv_hydro_Bow)

adiv_hydro_Bow %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_hydro_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))


#ACWA

hydro_ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River" & Family=="Hydropsychidae")

adiv_hydro_ACWA <- data.frame(
  "Shannon" = phyloseq::estimate_richness(hydro_ACWA_ps, measures = "Shannon"),
  "Stream_Type" = phyloseq::sample_data(hydro_ACWA_ps)$Stream_Type,
  "Site" = phyloseq::sample_data(hydro_ACWA_ps)$Site)

adiv_hydro_ACWA %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_hydro_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))


## Larval Heptageniidae ##

hep_lar<-subset_samples(sample_ps, Family=="Heptageniidae")

adiv_hep <- data.frame(
  "Shannon" = phyloseq::estimate_richness(hep_lar, measures = "Shannon"),
  "Site" = phyloseq::sample_data(hep_lar)$Site)

adiv_hep %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_hep %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Larval Chironomidae ##

chiro_lar<-subset_samples(sample_ps, Family=="Chironomidae")

adiv_chiro <- data.frame(
  "Shannon" = phyloseq::estimate_richness(chiro_lar, measures = "Shannon"),
  "Site" = phyloseq::sample_data(chiro_lar)$Site)

adiv_chiro %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_chiro %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Larval Perlidae ##

perl_lar<-subset_samples(sample_ps, Family=="Perlidae")

adiv_perl <- data.frame(
  "Shannon" = phyloseq::estimate_richness(perl_lar, measures = "Shannon"),
  "Site" = phyloseq::sample_data(perl_lar)$Site)

adiv_perl %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_perl %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Larval Baetidae ##

baetid_lar<-subset_samples(sample_ps, Family=="Baetidae")

adiv_baetid <- data.frame(
  "Shannon" = phyloseq::estimate_richness(baetid_lar, measures = "Shannon"),
  "Site" = phyloseq::sample_data(baetid_lar)$Site)

adiv_baetid %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_baetid %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))


## Adult Trichoptera ##

trichop_ad<-subset_samples(sample_ps, Family=="Trichoptera")

adiv_trichop <- data.frame(
  "Shannon" = phyloseq::estimate_richness(trichop_ad, measures = "Shannon"),
  "Site" = phyloseq::sample_data(trichop_ad)$Site)

adiv_trichop %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_trichop %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Adult Ephemeroptera ##

ephem_ad_Bow<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type=="Bow River")

adiv_ephem_Bow <- data.frame(
  "Shannon" = phyloseq::estimate_richness(ephem_ad_Bow, measures = "Shannon"),
  "Site" = phyloseq::sample_data(ephem_ad_Bow)$Site)

adiv_ephem_Bow %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_ephem_Bow %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

# ACWA #

ephem_ad_ACWA<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type!="Bow River")

adiv_ephem_ACWA <- data.frame(
  "Shannon" = phyloseq::estimate_richness(ephem_ad_ACWA, measures = "Shannon"),
  "Site" = phyloseq::sample_data(ephem_ad_ACWA)$Site)

adiv_ephem_ACWA %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_ephem_ACWA %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))



## Adult Diptera ##

dip_ad<-subset_samples(sample_ps, Family=="Diptera")

adiv_dip <- data.frame(
  "Shannon" = phyloseq::estimate_richness(dip_ad, measures = "Shannon"),
  "Site" = phyloseq::sample_data(dip_ad)$Site)

adiv_dip %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_dip %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Araneidae ##

aran<-subset_samples(sample_ps, Family=="Araneidae")

adiv_aran <- data.frame(
  "Shannon" = phyloseq::estimate_richness(aran, measures = "Shannon"),
  "Site" = phyloseq::sample_data(aran)$Site)

adiv_aran %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))


adiv_aran %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

## Tetragnathidae ##

tetra<-subset_samples(sample_ps, Family=="Tetragnathidae")

adiv_tetra <- data.frame(
  "Shannon" = phyloseq::estimate_richness(tetra, measures = "Shannon"),
  "Site" = phyloseq::sample_data(tetra)$Site)

adiv_tetra %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))

adiv_tetra %>%
  group_by(Site) %>%
  dplyr::summarise(median_shannon = median(Shannon),
            IQR_Shannon= IQR(Shannon))
```

## Rarefied alpha diversity {.tabset}

### Bow River - separated by site and invertebrate taxa type 

```{r}
Bow_ps_rare<-subset_samples(ps_rare, Stream_Type=="Bow River")

adiv_Bow_rare <- data.frame(
  "Shannon" = phyloseq::estimate_richness(Bow_ps_rare, measures = "Shannon"),
  "Family" = phyloseq::sample_data(Bow_ps_rare)$Family,
  "Stage" = phyloseq::sample_data(Bow_ps_rare)$Stage,
  "Site" = phyloseq::sample_data(Bow_ps_rare)$Site)
head(adiv_Bow_rare)


#Shannon diversity across taxa
shan.plot.taxa.rare<-ggplot(adiv_Bow_rare, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Shannon, fill=Site))+
  facet_nested_wrap(~Stage+ factor(Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scales= "free_x", nrow=1)+
  geom_boxplot() +
  labs(x = "Site", y = "Shannon Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
        strip.text.x = element_text(size=20), 
        axis.text.y = element_text(size=16, color="black"), axis.title.x = element_text(size=18), axis.title.y = element_text(size=18))+
  rremove("grid")+
  scale_fill_brewer(palette="Accent")

shan.plot.taxa.rare

#ggsave(filename="Microbiome_ED/Final analysis/Alpha Diversity Graphs/Rarefied shannon diversity Bow River.png", height=12, width=22)

#Looks the same as the non-rarefied plot


                                    ### Statistical tests ###


#Shannon diversity

shapiro.test(adiv_Bow_rare$Shannon) #Not normal, use Kruskal-Wallis tests. 

#Kruskal-Wallis tests:

#Life stage (larvae, adults, spiders)
kruskal.test(Shannon~Stage, data=adiv_Bow_rare) #p<0.001
dunnTest(Shannon~Stage, data=adiv_Bow_rare) #All diff from each other

#Difference across sites: (have to separate by invert taxa)

#Larval Hydropsychidae 
adiv_hydro_Bow_rare<-subset(adiv_Bow_rare, Family=="Hydropsychidae")
shapiro.test(adiv_hydro_Bow_rare$Shannon) #Normal dist. p = 0.09
leveneTest(Shannon~Site, data=adiv_hydro_Bow_rare) #Homogeneity. p = 0.276.
#Use ANOVA
summary(hydro_aov_rare<-aov(Shannon~Site, data=adiv_hydro_Bow_rare)) #p<0.001
TukeyHSD(hydro_aov_rare)
#GRVBR - COCH
#PMF - COCH
#GRVBR - SUN
#PMF - SUN
#GRVBR - CUSH
#PMF - CUSH
#PMF - GRVBR

#One extra difference with rarefied data (between GRVBR and PMF)

#Adult Trichoptera
div.df.Bow.Tricho.rare<-subset(adiv_Bow_rare, Family=="Trichoptera")
shapiro.test(div.df.Bow.Tricho.rare$Shannon) #Not normal
leveneTest(Shannon~Site, data=div.df.Bow.Tricho.rare) #Heterogeneity
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.Tricho.rare) #p= 0.709
dunnTest(Shannon~Site, data=div.df.Bow.Tricho.rare) #no significant diff. across sites.

#Larval Heptageniidae (order Ephemeroptera)
div.df.Bow.Hep.rare<-subset(adiv_Bow_rare, Family=="Heptageniidae")
shapiro.test(div.df.Bow.Hep.rare$Shannon) #Normal dist. p = 0.2118
leveneTest(Shannon~Site, data=div.df.Bow.Hep.rare) #Homogeneity
#Use ANOVA
summary(hep_aov_rare<-aov(Shannon~Site, data=div.df.Bow.Hep.rare)) #p=0.0435
TukeyHSD(hep_aov_rare)
#COCH - PMF

#Adult Ephemeroptera
div.df.Bow.Ephem.rare<-subset(adiv_Bow_rare, Family=="Ephemeroptera")
shapiro.test(div.df.Bow.Ephem.rare$Shannon) #p=0.012, not normal
leveneTest(Shannon~Site, data=div.df.Bow.Ephem.rare) #Homogeneity p = 0.49
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.Ephem.rare) #p=0.135
dunnTest(Shannon~Site, data=div.df.Bow.Ephem.rare) #No sig diff across sites

#Larval Chironomidae (Order Diptera)
div.df.Bow.chiro.rare<-subset(adiv_Bow_rare, Family=="Chironomidae")
shapiro.test(div.df.Bow.chiro.rare$Shannon) #0.308 Normal
leveneTest(Shannon~Site, data=div.df.Bow.chiro.rare) #Homogeneity. 
#Use AnOVA
summary(chiro_aov_rare<-aov(Shannon~Site, data=div.df.Bow.chiro.rare)) #p=0.257
TukeyHSD(chiro_aov_rare)
#No differences across sites

#Adult Chironomidae (Order Diptera)
div.df.Bow.dip.rare<-subset(adiv_Bow_rare, Family=="Diptera")
shapiro.test(div.df.Bow.dip.rare$Shannon) #p=0.166 Normal
leveneTest(Shannon~Site, data=div.df.Bow.dip.rare) #0.605. Homogeneity.
#Use ANOVA
summary(dip_aov_rare<-aov(Shannon~Site, data=div.df.Bow.dip.rare)) #p=0.652
TukeyHSD(dip_aov_rare)
#No differences across sites

#Larval Perlidae (Order Plecoptera)
div.df.Bow.perl.rare<-subset(adiv_Bow_rare, Family=="Perlidae")
shapiro.test(div.df.Bow.perl.rare$Shannon) #0.089 normal
leveneTest(Shannon~Site, data=div.df.Bow.perl.rare) #0.04. Heterogeneity
#Use Kruskal-Wallis
kruskal.test(Shannon~Site, data=div.df.Bow.perl.rare) #p=0.216
dunnTest(Shannon~Site, data=div.df.Bow.perl.rare) #No sig diff across sites

summary(perl_aov_rare<-aov(Shannon~Site, data=div.df.Bow.perl.rare)) #p=0.106
TukeyHSD(perl_aov_rare) #ANOVA also indicates no differences. 

#Araneidae
div.df.Bow.aran.rare<-subset(adiv_Bow_rare, Family=="Araneidae")
shapiro.test(div.df.Bow.aran.rare$Shannon) #0.532. Normal
leveneTest(Shannon~Site, data=div.df.Bow.aran.rare) #0.75 homogeneity
#Use ANOVA
summary(aran_aov_rare<-aov(Shannon~Site, data=div.df.Bow.aran.rare)) #p=0.404
TukeyHSD(aran_aov_rare)
#No differences across sites

#Tetragnathidae
div.df.Bow.tetra.rare<-subset(adiv_Bow_rare, Family=="Tetragnathidae")
shapiro.test(div.df.Bow.tetra.rare$Shannon) #P<0.001 not normal
#Use KW
kruskal.test(Shannon~Site, data=div.df.Bow.tetra.rare) #p=0.25
dunnTest(Shannon~Site, data=div.df.Bow.tetra.rare) #No differences across sites 

#Rarefied data resulted in only one extra difference in larval hydropsychidae. Pretty much the same as non rarefied data.

```

### ACWA - faceted by stream and invertebrate type 

```{r}
ACWA_ps_rare<-subset_samples(ps_rare, Stream_Type!="Bow River")

adiv_ACWA_rare <- data.frame(
  "Shannon" = phyloseq::estimate_richness(ACWA_ps_rare, measures = "Shannon"),
  "Family" = phyloseq::sample_data(ACWA_ps_rare)$Family,
  "Stage" = phyloseq::sample_data(ACWA_ps_rare)$Stage,
  "Site" = phyloseq::sample_data(ACWA_ps_rare)$Site)

#Shannon index
shan.plot.taxa.ACWA.rare<-ggplot(adiv_ACWA_rare, aes(x=factor(Site, labels=c("REF", "EXP5", "EXP15")), y=Shannon, fill=Site))+
  facet_nested_wrap(~Stage+ factor(Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera")), scales= "free_x", nrow=1)+
  geom_boxplot() +
  labs(x = "Site", y = "Shannon Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
        strip.text.x = element_text(size=20), 
        axis.text.y = element_text(size=16, color="black"), axis.title.x = element_text(size=18), axis.title.y = element_text(size=18))+
  rremove("grid")+
  scale_fill_manual(values=c("#7E8071", "#EDBC81", "#A76B56"))

shan.plot.taxa.ACWA.rare

#ggsave(filename="Microbiome_ED/Final analysis/Alpha Diversity Graphs/Non rarefied shannon diversity ACWA.png", height=12, width=22)


                                    ### Statistical tests ###

#Shannon diversity

shapiro.test(adiv_ACWA_rare$Shannon) #Not normal

#Kruskal-Wallis tests:

#Test diff btw larvae and adults
wilcox.test(Shannon~Stage, data=adiv_ACWA_rare) #W = 0.84, p < 0.001

#Diff btw families
kruskal.test(Shannon~Family, data=adiv_ACWA_rare)#p<0.001
dunnTest(Shannon~Family, data=adiv_ACWA_rare) #All diff from each other

#Hydropsychidae
hydro_ACWA_rare<-subset(adiv_ACWA_rare, Family=="Hydropsychidae")
shapiro.test(hydro_ACWA_rare$Shannon) # normal
leveneTest(data=hydro_ACWA_rare, Shannon~Site) #0.3. homogeneity
#Use ANOVA
summary(hydro_ACWA_aov_rare<-aov(data=hydro_ACWA_rare, Shannon~Site)) #0.326 no diff
#No difference between streams

#Baetidae
baetid_ACWA_rare<-subset(adiv_ACWA_rare, Family=="Baetidae")
shapiro.test(baetid_ACWA_rare$Shannon) # normal
leveneTest(data=baetid_ACWA, Shannon~Site) #0.683. homogeneity
#Use t test
t.test(data=baetid_ACWA_rare, Shannon~Site) #p=0.809
#No difference between streams

#Ephemeroptera
ephem_ACWA_rare<-subset(adiv_ACWA_rare, Family=="Ephemeroptera")
shapiro.test(ephem_ACWA_rare$Shannon) #Not normal
#Use Wilcoxon test
wilcox.test(data=ephem_ACWA_rare, Shannon~Site) # W = 18, p = 0.195
#No difference between streams

#Same results as non rarefied data
```

# Beta Diversity

Beta diversity is a measure of dissimilarity between communities. It is useful to see how the bacterial community changes across sites/invertebrate families/exposures/etc. This analysis was split by invertebrate taxa to tease out some variability in the data. A PERMANOVA (function 'adonis') was used to calculate significant differences across sites and beta dispersion (measure of variance within groups) was also tested to determine if there was a location or dispersion effect. The results were ordinated with an NMDS using Bray-Curtis dissimilarity matrices. 

Transforming the data to proportions (relative abundance) is the best method of normalizing the data prior to beta diversity analysis according to McCurdie & Holmes, 2014; Weiss et al., 2017; McKnight et al., 2019. 


## Blanks and experimental samples 
```{r}
#Combine filtered experimental samples with blanks
blank_ps<-subset_samples(ps, Sample_Type=="Blank")

exp_bl_ps<-merge_phyloseq(blank_ps, sample_ps)
nsamples(exp_bl_ps) #362 (339 samples, 23 blanks)

#Ordinating blanks with experimental samples to see how closely related they are

#PCoA with Bray Curtis distance matrix
ps_rel_abun = transform_sample_counts(exp_bl_ps, function(x) x/sum(x))

dist_all = phyloseq::distance(ps_rel_abun, method="bray") #Bray Curtis distance matrix
ordination_all = ordinate(ps_rel_abun, method="PCoA", distance=dist_all) #Ordinate using PCoA

plot_ordination(ps_rel_abun, ordination_all, shape="Blank_collection_method", color="Sample_Type") +
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1[12.9%]")+
  ylab("PC2[10.7%]")
#Doesn't explain much of the data, need to do an NMDS
 
set.seed(1)
ord_NMDS_blanks = ordinate(ps_rel_abun, method="NMDS", distance=dist_all)
ord_NMDS_blanks #stress=0.213


blank_nmds<-plot_ordination(ps_rel_abun, ord_NMDS_blanks, color="Sample_Type", shape="Blank_Type") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=10), 
        legend.text = element_text(size=8), 
        legend.position = "bottom")+
  scale_shape_manual(name="Blank type", values=c(1:7))+
  scale_color_manual(name="Sample type", values=c("red", "blue"))

newtab$Blank_Type <- factor(newtab$Blank_Type, levels = c("Tweezer water rinse","Bedsheet","Floating emergence trap","Buffer tube","DNA extraction reagent","Nested PCR negative"))
blank_nmds$Blank_Type <- newtab
print(blank_nmds)

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/blank and experimental samples.png", height=10, width=12)

metadata_blanks <- data.frame(sample_data(ps_rel_abun))

adonis.blanks <- adonis2(dist_all ~ Sample_Type, data = metadata_blanks)
adonis.blanks #p<0.001. Significant dissimilarity between blank samples and invertebrate samples
#F(1, 360) = 6.91, P < 0.001

#Testing variance (beta dispersion) differences between samples and blanks:
bd_blanks<-betadisper(dist_all, metadata_blanks$Sample_Type)
anova(bd_blanks) #Sig. dispersion F(1,360) = 17.7, p<0.001

## Separating out respective blanks for each sample type ##

#Water July (larvae and blanks from July)
water_July<-subset_samples(ps_rel_abun, Blank_collection_method=="Water July")
dist_wat_July<-phyloseq::distance(water_July, method="bray")
metadata_wat_July<-data.frame(sample_data(water_July))
adonis.blank.lar.July<-adonis2(dist_wat_July ~ Sample_Type, data = metadata_wat_July)
adonis.blank.lar.July #p=0.001 sig. diff. 

bd_wat_July<-betadisper(dist_wat_July, metadata_wat_July$Sample_Type)
anova(bd_wat_July) #Sig. dispersion

#Water September (larvae and blanks from September)
water_Sept<-subset_samples(ps_rel_abun, Blank_collection_method=="Water September")
dist_wat_Sept<-phyloseq::distance(water_Sept, method="bray")
metadata_wat_Sept<-data.frame(sample_data(water_Sept))
adonis.blank.lar.Sept<-adonis2(dist_wat_Sept ~ Sample_Type, data = metadata_wat_Sept)
adonis.blank.lar.Sept #p=0.001 sig. diff. 

bd_wat_Sept<-betadisper(dist_wat_Sept, metadata_wat_Sept$Sample_Type)
anova(bd_wat_Sept) #Sig. dispersion

#Bedsheet 
bedsheet_ps<-subset_samples(ps_rel_abun, Blank_collection_method=="Sheet")
bedsheet_bray<-phyloseq::distance(bedsheet_ps, method="bray")
metadata_bedsheet<-data.frame(sample_data(bedsheet_ps))
adonis.blank.bedsheet<-adonis2(bedsheet_bray ~ Sample_Type, data = metadata_bedsheet)
adonis.blank.bedsheet #p=0.001 sig. diff. 

bd_bedsheet<-betadisper(bedsheet_bray, metadata_bedsheet$Sample_Type)
anova(bd_bedsheet) #Sig. dispersion

#Floating emergence traps
emerg_trap_ps<-subset_samples(ps_rel_abun, Blank_collection_method=="Mesh")
emerg_trap_bray<-phyloseq::distance(emerg_trap_ps, method="bray")
metadata_emerg_trap<-data.frame(sample_data(emerg_trap_ps))
adonis.blank.emerg_trap<-adonis2(emerg_trap_bray ~ Sample_Type, data = metadata_emerg_trap)
adonis.blank.emerg_trap #p=0.043 sig. diff. 

bd_emerg_trap<-betadisper(emerg_trap_bray, metadata_emerg_trap$Sample_Type)
anova(bd_emerg_trap) #no sig. dispersion


#Pairwise comparisons between all blank collection methods
pairwise.adonis2(dist_all~ Blank_collection_method, data = metadata_blanks) 
#Difference between all blanks and samples

#Testing variance (beta dispersion) differences between samples and blanks:
bd_blanks_type<-betadisper(dist_all, metadata_blanks$Blank_collection_method)
anova(bd_blanks_type) #Sig. dispersion

#Seems like the blank community composition differs from that of the invertebrate samples however, there is significant beta dispersion so we can't be entirely sure. We can compare the results with and without prior decontamination steps to be cautious. 
```

## Experimental samples {.tabset}

### Non-rarefied 

#### Bow River 
```{r}
#### Beta Diversity Graphs + PERMANOVA ####
#Normalize the filtered data by relative abundance 
Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
Bow_rel_abun  = transform_sample_counts(Bow_ps, function(x) x/sum(x)) 
#transform to relative abundance. Seems to be the best method of normalizing.

Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray") #Bray Curtis distance matrix
ord_Bow = ordinate(Bow_rel_abun, method="PCoA", distance=Bow_bray) #Ordinate using PCoA

#Scree plot of PCoA
scree <- plot_scree(ord_Bow)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
        axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1)

metadata_Bow <- data.frame(sample_data(Bow_rel_abun))

#All together
plot_ordination(Bow_rel_abun, ord_Bow, shape="Site", color="Family") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [12.8%]")+
  ylab("PC2 [10.9%]")

#Faceted by stage (larvae, adults, spiders)
plot_ordination(Bow_rel_abun, ord_Bow, shape="Site", color="Family") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  facet_wrap(~Stage)+
  xlab("PC1 [12.8%]")+
  ylab("PC2 [10.9%]")

#Axes don't explain a ton of variation so I will try ordinating with an NMDS as well. 

#ggsave(filename="microbiome_analysis/Final analysis/Beta Diversity Graphs/PCoA by site and invert Bow River.png", height = 7, width=8)


# NMDS 
set.seed(1)
ord_NMDS_Bow = ordinate(Bow_rel_abun, method="NMDS", distance=Bow_bray)
ord_NMDS_Bow #Stress=0.197

plotNMDS1<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.3, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=18), 
        axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18, color="black"),
        legend.position = "right")+
  guides(color=guide_legend(title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  ylim(-0.5, 0.5)+
  xlim(-0.5, 0.5)

newtab1 = data.table(plotNMDS1$data)
newtab1$Family <- ordered(newtab1$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Chironomidae", "Araneidae", "Tetragnathidae"))
plotNMDS1$data <- newtab1
print(plotNMDS1)

#ggsave(filename="microbiome_analysis/Final analysis/Beta Diversity Graphs/NMDS across sites Bow only relative abundance normalized.png", height=10, width=14)

#Ellipses by taxa
plotNMDStaxa<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Family", shape="Site") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.3, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=18), 
        axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18, color="black"),
        legend.position = "right")+
  guides(color=guide_legend(title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  stat_ellipse(type="norm", alpha=1, aes(group=Family, color=Family))+
  ylim(-0.5, 0.5)+
  xlim(-0.5, 0.5)

newtabtaxa = data.table(plotNMDStaxa$data)
newtabtaxa$Family <- ordered(newtabtaxa$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Chironomidae", "Araneidae", "Tetragnathidae"))
plotNMDStaxa$data <- newtabtaxa
print(plotNMDStaxa)

#ggsave(filename="microbiome_analysis/Final analysis/Beta Diversity Graphs/NMDS across taxa Bow.png", height=10, width=14)


#Ellipses by Site
plotNMDSsite<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Family", shape="Site") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.3, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=18), 
        axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18, color="black"),
        legend.position = "right")+
  guides(color=guide_legend(title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  stat_ellipse(type="norm", alpha=1, aes(group=Site))+
   ylim(-0.5, 0.5)+
  xlim(-0.5, 0.5)

newtabtaxa = data.table(plotNMDSsite$data)
newtabtaxa$Family <- ordered(newtabtaxa$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Chironomidae", "Araneidae", "Tetragnathidae"))
plotNMDSsite$data <- newtabtaxa
print(plotNMDSsite)

#ggsave(filename="microbiome_analysis/Final analysis/Beta Diversity Graphs/NMDS across sites Bow.png", height=10, width=14)



#Faceted by invertebrate taxa and life Stage
p<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Site") + 
  theme_classic() +
  geom_point(size=2)+
  facet_wrap(vars(Stage, Family), scales="free", nrow=1)+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=20), 
        legend.position = "bottom", 
        axis.text.x=element_text(size=16, color="black", angle=90, hjust=0.5, vjust=0.5),
        axis.text.y=element_text(size=16, color="black"),
        axis.title = element_text(size=18),
        strip.text.x = element_text(size=20))+
  facet_nested_wrap(~Stage + Family, scale="free_x", nrow=1)+
  stat_ellipse()

newtab = data.table(p$data)
newtab$Family <- ordered(newtab$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))
p$data <- newtab
print(p)

#ggsave(filename="microbiome_analysis/Final Analysis/Beta Diversity Graphs/All samples combined/NMDS across Bow River invertebrates and sites.png", height=10, width=20)

#Faceted by site
p1<-plot_ordination(Bow_rel_abun, ord_NMDS_Bow, color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Site, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=20), 
        legend.position = "bottom", 
        axis.text=element_text(size=18, color="black", angle=90, hjust=1, vjust=1),
        axis.title = element_text(size=18, color="black"),
        strip.text.x = element_text(size=20))+
  stat_ellipse(linetype=1, type="norm")+
  guides(color=guide_legend(nrow=3, byrow=TRUE, title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))
 

newtab2 = data.table(p1$data)
newtab2$Family <- ordered(newtab2$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Diptera", "Araneidae", "Tetragnathidae"))
p1$data <- newtab2
print(p1)


#ggsave(filename="microbiome_analysis/Final analysis/Beta Diversity Graphs/NMDS across sites faceted by invertebrate taxa non-rarefied Bow only.png", height=10, width=16)


#Faceted by life stage (Larvae, Adults, Spiders)
plot_ordination(Bow_rel_abun, ord_NMDS_Bow, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=2)+
  facet_grid(~Stage, scales="free")+
  theme_bw()+
  rremove("grid")+
  theme(legend.key.size = unit(0.7, 'cm'), 
        legend.title = element_text(size=14), 
        legend.text = element_text(size=12), 
        legend.position = "right")+
  stat_ellipse()

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites faceted by life stage rarefied Bow only.png", height=8, width=16)


#PERMANOVA by Site and invertebrate taxa

test.adonis.Bow <- adonis2(Bow_bray ~ Site*Family, data = metadata_Bow)
test.adonis.Bow #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two.

#Pairwise comparisons
pairwise.adonis2(Bow_bray ~ Site, data = metadata_Bow)  
#All sites except COCH-SUN are sig diff when invertebrate taxa are combined

pairwise.adonis2(Bow_bray ~ Family, data = metadata_Bow)
#All invertebrate taxa are sig diff when sites are combined

#Testing variance (beta dispersion) differences between sites, and sample families:
bd_all_site_Bow<-betadisper(Bow_bray, metadata_Bow$Site)
anova(bd_all_site_Bow) #No significant beta dispersion (p=0.193)
permutest(bd_all_site_Bow) #Gives the same result but based on a permutation test. 
#PERMANOVA results could be more driven by the spread in the variance of samples rather than the means if it was significant. (p=0.201)

bd_all_fam_Bow<-betadisper(Bow_bray, metadata_Bow$Family) 
anova(bd_all_fam_Bow) #significantly different p=0.0014
permutest(bd_all_fam_Bow) #sig. different (same result)
#Significant PERMANOVA could just be because of the variance within groups rather than between groups (groups = invertebrate taxa type)


## Hydropsychidae ##

hydro_Bow<-subset_samples(Bow_ps, Family=="Hydropsychidae")
Bow_rel_abun_hydro  = transform_sample_counts(hydro_Bow, function(x) x/sum(x)) 
Bow_bray_hydro = phyloseq::distance(Bow_rel_abun_hydro, method="bray") 

metadata_Bow_hydro <- data.frame(sample_data(Bow_rel_abun_hydro))

test.adonis.Bow.hydro <- adonis2(Bow_bray_hydro ~ Site, data = metadata_Bow_hydro)
test.adonis.Bow.hydro #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_hydro ~ Site, data = metadata_Bow_hydro)
#All sites except CUSHBR - SUN are sig diff

bd_all_fam_Bow_hydro<-betadisper(Bow_bray_hydro, metadata_Bow_hydro$Site) 
anova(bd_all_fam_Bow_hydro) #no beta dispersion p = 0.063

## Heptageniidae ##

hep_Bow<-subset_samples(ps_Bow, Family=="Heptageniidae")
Bow_rel_abun_hep  = transform_sample_counts(hep_Bow, function(x) x/sum(x)) 
Bow_bray_hep = phyloseq::distance(Bow_rel_abun_hep, method="bray") 

metadata_Bow_hep <- data.frame(sample_data(Bow_rel_abun_hep))

test.adonis.Bow.hep <- adonis2(Bow_bray_hep ~ Site, data = metadata_Bow_hep)
test.adonis.Bow.hep #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_hep ~ Site, data = metadata_Bow_hep)
#All sites are sig. diff from each other

bd_all_fam_Bow_hep<-betadisper(Bow_bray_hep, metadata_Bow_hep$Site) 
anova(bd_all_fam_Bow_hep) #sig beta dispersion p = 0.0126


## Chironomidae ##

chiro_Bow<-subset_samples(ps_Bow, Family=="Chironomidae")
Bow_rel_abun_chiro  = transform_sample_counts(chiro_Bow, function(x) x/sum(x)) 
Bow_bray_chiro = phyloseq::distance(Bow_rel_abun_chiro, method="bray") 

metadata_Bow_chiro <- data.frame(sample_data(Bow_rel_abun_chiro))

test.adonis.Bow.chiro <- adonis2(Bow_bray_chiro ~ Site, data = metadata_Bow_chiro)
test.adonis.Bow.chiro #p=0.002. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_chiro ~ Site, data = metadata_Bow_chiro)
#Cochrane - Policeman Flats are not diff from each other

bd_all_fam_Bow_chiro<-betadisper(Bow_bray_chiro, metadata_Bow_chiro$Site) 
anova(bd_all_fam_Bow_chiro) #no beta dispersion p = 0.34

## Perlidae ##

perl_Bow<-subset_samples(ps_Bow, Family=="Perlidae")
Bow_rel_abun_perl  = transform_sample_counts(perl_Bow, function(x) x/sum(x)) 
Bow_bray_perl = phyloseq::distance(Bow_rel_abun_perl, method="bray") 

metadata_Bow_perl <- data.frame(sample_data(Bow_rel_abun_perl))

test.adonis.Bow.perl <- adonis2(Bow_bray_perl ~ Site, data = metadata_Bow_perl)
test.adonis.Bow.perl #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_perl ~ Site, data = metadata_Bow_perl)
#All sites were significantly different from each other

bd_all_fam_Bow_perl<-betadisper(Bow_bray_perl, metadata_Bow_perl$Site) 
anova(bd_all_fam_Bow_perl) #sig beta dispersion p = 0.0048

## Trichoptera (adult caddis) ##

trichop_Bow<-subset_samples(ps_Bow, Family=="Trichoptera")
Bow_rel_abun_trichop  = transform_sample_counts(trichop_Bow, function(x) x/sum(x)) 
Bow_bray_trichop = phyloseq::distance(Bow_rel_abun_trichop, method="bray") 

metadata_Bow_trichop <- data.frame(sample_data(Bow_rel_abun_trichop))

test.adonis.Bow.trichop <- adonis2(Bow_bray_trichop ~ Site, data = metadata_Bow_trichop)
test.adonis.Bow.trichop #p = 0.016 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_trichop ~ Site, data = metadata_Bow_trichop)
#Only COCH - GRVBR and CUSHBR - GRVBR were different from each other (not all upstream sites were different from GRVBR)

bd_all_fam_Bow_trichop<-betadisper(Bow_bray_trichop, metadata_Bow_trichop$Site) 
anova(bd_all_fam_Bow_trichop) #beta dispersion p = 0.034

## Ephemeroptera (adult mayfly) ##

ephem_Bow<-subset_samples(ps_Bow, Family=="Ephemeroptera")
Bow_rel_abun_ephem = transform_sample_counts(ephem_Bow, function(x) x/sum(x)) 
Bow_bray_ephem = phyloseq::distance(Bow_rel_abun_ephem, method="bray") 

metadata_Bow_ephem <- data.frame(sample_data(Bow_rel_abun_ephem))

test.adonis.Bow.ephem <- adonis2(Bow_bray_ephem ~ Site, data = metadata_Bow_ephem)
test.adonis.Bow.ephem #p < 0.001 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_ephem ~ Site, data = metadata_Bow_ephem)
#Sig diff btw sites include:
#COCH - GRVBR
#COCH - CUSHBR
#COCH - PMF
#GRVBR - CUSHBR
#SUN - CUSHBR
#CUSHBR - PMF

bd_all_fam_Bow_ephem<-betadisper(Bow_bray_ephem, metadata_Bow_ephem$Site) 
anova(bd_all_fam_Bow_ephem) #no beta dispersion p = 0.064


## Diptera (adult chironomidae) ##
dip_Bow<-subset_samples(ps_Bow, Family=="Diptera")
Bow_rel_abun_dip = transform_sample_counts(dip_Bow, function(x) x/sum(x)) 
Bow_bray_dip = phyloseq::distance(Bow_rel_abun_dip, method="bray") 

metadata_Bow_dip <- data.frame(sample_data(Bow_rel_abun_dip))

test.adonis.Bow.dip <- adonis2(Bow_bray_dip ~ Site, data = metadata_Bow_dip)
test.adonis.Bow.dip #p = 0.122

#Pairwise comparisons
pairwise.adonis2(Bow_bray_dip ~ Site, data = metadata_Bow_dip)
#sig diff between COCH - CUSHBR

bd_all_fam_Bow_dip<-betadisper(Bow_bray_dip, metadata_Bow_dip$Site) 
anova(bd_all_fam_Bow_dip) #no beta dispersion p = 0.593


## Araneidae ##

aran_Bow<-subset_samples(ps_Bow, Family=="Araneidae")
Bow_rel_abun_aran = transform_sample_counts(aran_Bow, function(x) x/sum(x)) 
Bow_bray_aran = phyloseq::distance(Bow_rel_abun_aran, method="bray") 

metadata_Bow_aran <- data.frame(sample_data(Bow_rel_abun_aran))

test.adonis.Bow.aran <- adonis2(Bow_bray_aran ~ Site, data = metadata_Bow_aran)
test.adonis.Bow.aran #p < 0.001 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_aran ~ Site, data = metadata_Bow_aran)
#Sig diff btw sites include:
#COCH - SUN
#COCH - PMF
#GRVBR - SUN
#GRVBR - PMF
#CUSHBR - SUN
#CUSHBR - PMF

bd_all_fam_Bow_aran<-betadisper(Bow_bray_aran, metadata_Bow_aran$Site) 
anova(bd_all_fam_Bow_aran) #no beta dispersion p = 0.825

## Tetragnathidae ##

tetra_Bow<-subset_samples(ps_Bow, Family=="Tetragnathidae")
Bow_rel_abun_tetra = transform_sample_counts(tetra_Bow, function(x) x/sum(x)) 
Bow_bray_tetra = phyloseq::distance(Bow_rel_abun_tetra, method="bray") 

metadata_Bow_tetra <- data.frame(sample_data(Bow_rel_abun_tetra))

test.adonis.Bow.tetra <- adonis2(Bow_bray_tetra ~ Site, data = metadata_Bow_tetra)
test.adonis.Bow.tetra #p = 0.541

#Pairwise comparisons
pairwise.adonis2(Bow_bray_tetra ~ Site, data = metadata_Bow_tetra)
#No significant differences between any of the sites


bd_all_fam_Bow_tetra<-betadisper(Bow_bray_tetra, metadata_Bow_tetra$Site) 
anova(bd_all_fam_Bow_tetra) #no beta dispersion p = 0.510

## Cochrane ##

coch_Bow<-subset_samples(Bow_ps, Site=="Cochrane")
coch_rel_abun = transform_sample_counts(coch_Bow, function(x) x/sum(x)) 
coch_bray = phyloseq::distance(coch_rel_abun, method="bray") 

metadata_coch <- data.frame(sample_data(coch_rel_abun))

test.adonis.Bow.coch <- adonis2(coch_bray ~ Family, data = metadata_coch)
test.adonis.Bow.coch #p < 0.001

#Pairwise comparisons
pairwise.adonis2(coch_bray ~ Family, data = metadata_coch)
#Most inverts are diff. from each other

bd_all_fam_Bow_coch<-betadisper(coch_bray, metadata_coch$Family) 
anova(bd_all_fam_Bow_coch) #no beta dispersion p = 0.1559.

## Sunalta ##

sun_Bow<-subset_samples(Bow_ps, Site=="Sunalta")
sun_rel_abun = transform_sample_counts(sun_Bow, function(x) x/sum(x)) 
sun_bray = phyloseq::distance(sun_rel_abun, method="bray") 

metadata_sun <- data.frame(sample_data(sun_rel_abun))

test.adonis.Bow.sun <- adonis2(sun_bray ~ Family, data = metadata_sun)
test.adonis.Bow.sun #p < 0.001

#Pairwise comparisons
pairwise.adonis2(sun_bray ~ Family, data = metadata_sun)
#Most inverts are diff. from each other

bd_all_fam_Bow_sun<-betadisper(sun_bray, metadata_sun$Family) 
anova(bd_all_fam_Bow_sun) #yes beta dispersion p = 0.0057.


## Cushing Bridge ##

cush_Bow<-subset_samples(Bow_ps, Site=="Cushing Bridge")
cush_rel_abun = transform_sample_counts(cush_Bow, function(x) x/sum(x)) 
cush_bray = phyloseq::distance(cush_rel_abun, method="bray") 

metadata_cush <- data.frame(sample_data(cush_rel_abun))

test.adonis.Bow.cush <- adonis2(cush_bray ~ Family, data = metadata_cush)
test.adonis.Bow.cush #p < 0.001

#Pairwise comparisons
pairwise.adonis2(cush_bray ~ Family, data = metadata_cush)
#Most inverts are diff. from each other

bd_all_fam_Bow_cush<-betadisper(cush_bray, metadata_cush$Family) 
anova(bd_all_fam_Bow_cush) #yes beta dispersion p = 0.0068.


## Graves Bridge ##

grv_Bow<-subset_samples(Bow_ps, Site=="Graves Bridge")
grv_rel_abun = transform_sample_counts(grv_Bow, function(x) x/sum(x)) 
grv_bray = phyloseq::distance(grv_rel_abun, method="bray") 

metadata_grv <- data.frame(sample_data(grv_rel_abun))

test.adonis.Bow.grv <- adonis2(grv_bray ~ Family, data = metadata_grv)
test.adonis.Bow.grv #p < 0.001

#Pairwise comparisons
pairwise.adonis2(grv_bray ~ Family, data = metadata_grv)
#Most inverts are diff. from each other

bd_all_fam_Bow_grv<-betadisper(grv_bray, metadata_grv$Family) 
anova(bd_all_fam_Bow_grv) #no beta dispersion p = 0.335.


## Policeman Flats ##

pmf_Bow<-subset_samples(Bow_ps, Site=="Policeman Flats")
pmf_rel_abun = transform_sample_counts(pmf_Bow, function(x) x/sum(x)) 
pmf_bray = phyloseq::distance(pmf_rel_abun, method="bray") 

metadata_pmf <- data.frame(sample_data(pmf_rel_abun))

test.adonis.Bow.pmf <- adonis2(pmf_bray ~ Family, data = metadata_pmf)
test.adonis.Bow.pmf #p < 0.001

#Pairwise comparisons
pairwise.adonis2(pmf_bray ~ Family, data = metadata_pmf)
#Most inverts are diff. from each other

bd_all_fam_Bow_pmf<-betadisper(pmf_bray, metadata_pmf$Family) 
anova(bd_all_fam_Bow_pmf) #no beta dispersion p = 0.0797.


#### Water quality variables Bow ####

## EnvFit function ##

#Graves Bridge does not have water quality info so we have to get rid of this site so that the NAs don't mess up the rest of the analysis.
ps_Bow_no_grvbr<-subset_samples(sample_ps, Stream_Type=="Bow River" & Site!="Graves Bridge")

#Transform to relative abundances to standardize
Bow_rel_abun_no_grvbr  = transform_sample_counts(ps_Bow_no_grvbr, function(x) x/sum(x)) 

env_Bow_no_grvbr<-data.frame(sample_data(Bow_rel_abun_no_grvbr))
env_Bow_no_grvbr_2<-env_Bow_no_grvbr[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Total_Phosphorus", "Total_Dissolved_Solids", "Total_Suspended_Solids", "Total_Organic_Carbon", "pH", "Nitrate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc). Ammonia and nitrite weren't selected since they did not differ at all between sites.

env_Bow_no_grvbr_3<-as.data.frame(scale(env_Bow_no_grvbr_2))

#All environmental data isn't normally distributed. Use Spearman correlations. 
shapiro.test(env_Bow_no_grvbr_3$Water_Temp) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Dissolved_oxygen) #non normal
shapiro.test(env_Bow_no_grvbr_3$Conductivity) #non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Phosphorus) #non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Dissolved_Solids) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Suspended_Solids) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Organic_Carbon) #Non normal
shapiro.test(env_Bow_no_grvbr_3$pH) #non normal
shapiro.test(env_Bow_no_grvbr_3$Nitrate) #non normal


correlation_matrix_env<-as.data.frame(cor(env_Bow_no_grvbr_3), method="spearman") #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

#Transform to Bray Curtis distance
Bow_bray_no_grvbr = phyloseq::distance(Bow_rel_abun_no_grvbr, method="bray") #Bray Curtis distance matrix

#Convert to data frame so it can be used in vegan
Bow_bray_df_no_grvbr<-as.matrix(Bow_bray_no_grvbr)
Bow_bray_df_no_grvbr<-as.data.frame(Bow_bray_df_no_grvbr)

#Ordinate with the bray curtis distance matrix
set.seed(123)
ord_Bow_mds_no_grvbr<-metaMDS(Bow_bray_df_no_grvbr) 
ord_Bow_mds_no_grvbr #stress=0.198

#install.packages("ggordiplots")
library(ggordiplots)

gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_3, perm = 9999, pt.size = 2, alpha = 0.05)
#Determines that TOC, pH, Conductivity, TDS, and TSS are significant predictors of microbiome composition. But some of these variables are highly correlated.

envfit(ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_3, permutations = 9999)
#Same conclusions here as above.

#Check correlations between significant variables:
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Dissolved_Solids) #0.998 Highly correlated. One variable must be removed. 
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Suspended_Solids) #0.733 Correlated
cor(env_Bow_no_grvbr_3$Total_Dissolved_Solids, env_Bow_no_grvbr_3$Total_Suspended_Solids) #0.763 Correlated. 
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Organic_Carbon) #0.0944. Not correlated. These can both stay.
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$pH) #-0.692 #Somewhat correlated
cor(env_Bow_no_grvbr_3$pH, env_Bow_no_grvbr_3$Total_Organic_Carbon) #0.276. Not correlated
#Remove Total dissolved solids and total suspended solids

env_Bow_no_grvbr_4<-env_Bow_no_grvbr_3[ , c("Conductivity", "pH", "Total_Organic_Carbon")]

envfit_2<-envfit(ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, permutations = 9999)
envfit_2 #All significant but have low R squared.

#Plotting ordination with sites as colours
ggenvfit_Bow_Sites<-gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, perm = 9999, groups = env_Bow_no_grvbr$Site, pt.size = 2, alpha = 0.05, len=0.05, unit="cm")
ggenvfit_Bow_Sites

#ggsave(file="microbiome_analysis/Final analysis/Beta Diversity Graphs/Envfit water quality Bow River sites.png", height=8, width=12)

#Plotting ordination with invertebrate taxa as colours
ggenvfit_Bow_taxa<-gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, perm = 9999, groups = env_Bow_no_grvbr$Family, pt.size = 2, alpha = 0.2)

#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality Bow River invert taxa.png", height=8, width=12)

ggarrange(ggenvfit_Bow_Sites, ggenvfit_Bow_taxa, nrow=1, ncool=2)


## Mantel correlations ##

#Abundance distance matrix
#Bow River without GRVBR
#Subset out Graves Bridge because there was no environmental data for that site
ps_Bow_no_grvbr<-subset_samples(sample_ps, Stream_Type=="Bow River" & Site!="Graves Bridge")

#Transform to relative abundances to standardize
Bow_rel_abun_no_grvbr  = transform_sample_counts(ps_Bow_no_grvbr, function(x) x/sum(x)) 

#Convert to Bray-Curtis distance
Bow_bray_no_grvbr = phyloseq::distance(Bow_rel_abun_no_grvbr, method="bray")
#Convert to data frame so it can be used in vegan
Bow_bray_df_no_grvbr<-as.matrix(Bow_bray_no_grvbr)
Bow_bray_df_no_grvbr<-as.data.frame(Bow_bray_df_no_grvbr)


#Bow River with GRVBR included for geographic matrix
#Transform to relative abundances to standardize
Bow_rel_abun  = transform_sample_counts(Bow_ps, function(x) x/sum(x)) 

#Convert to Bray-Curtis distance
Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray")
#Convert to data frame so it can be used in vegan
Bow_bray_df<-as.matrix(Bow_bray)
Bow_bray_df<-as.data.frame(Bow_bray_df)

#Geographic variable distance matrix
geo_Bow<-sample_data(Bow_rel_abun)
geo_loc_Bow<-geo_Bow[ ,4:5]

geo_Bow_mat<-as.matrix(geo_loc_Bow)
geo_Bow_df<-as.data.frame(geo_Bow_mat)

#longitude and latitude 
geo_coord_Bow = data.frame(geo_loc_Bow$Long, geo_loc_Bow$Lat) 

d.geo.Bow = distm(geo_coord_Bow, fun = distHaversine)
dist.geo.Bow = as.dist(d.geo.Bow)

abund_geo  = mantel(Bow_bray_df, dist.geo.Bow, method = "spearman", permutations = 999, na.rm = TRUE)
abund_geo #r = -0.00207, p = 0.842


#Make environmental variable distance matrix
env<-sample_data(Bow_rel_abun_no_grvbr)
env_data<-env[,16:29]

# Environmental data all together # 

env_scale<-scale(env_data)
env_dist<-dist(env_scale, method="euclidean")

abund_env = mantel(Bow_bray_df_no_grvbr, env_dist, method = "spearman", permutations = 999, na.rm = TRUE)
abund_env #r = -0.0004722, p = 0.481. Environment not selecting for the microbiome composition

# Environmental data separated #

#Water temperature
temp <-env_data$Water_Temp
dist.temp = dist(temp, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_temp = mantel(Bow_bray_df_no_grvbr, dist.temp, method = "spearman", permutations = 999, na.rm = TRUE)
abund_temp # Mantel r=0.00564, p=0.342

#Dissolved oxygen
DO<-env_data$Dissolved_oxygen

dist.DO = dist(DO, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DO = mantel(Bow_bray_df_no_grvbr, dist.DO, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DO # Mantel r=-0.01185, p=0.718

#Conductivity
cond<-env_data$Conductivity

dist.cond = dist(cond, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_cond = mantel(Bow_bray_df_no_grvbr, dist.cond, method = "spearman", permutations = 999, na.rm = TRUE)
abund_cond # Mantel r=0.00988, p=0.213

#Total Phosphorus
TP<-env_data$Total_Phosphorus

dist.TP = dist(TP, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TP = mantel(Bow_bray_df_no_grvbr, dist.TP, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TP # Mantel r=0.00502, p=0.381

#Total Dissolved Solids
TDS<-env_data$Total_Dissolved_Solids

dist.TDS = dist(TDS, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TDS = mantel(Bow_bray_df_no_grvbr, dist.TDS, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TDS # Mantel r=0.00988, p=0.197

#Total Suspended Solids
TSS<-env_data$Total_Suspended_Solids

dist.TSS = dist(TSS, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TSS = mantel(Bow_bray_df_no_grvbr, dist.TSS, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TSS # Mantel r=0.01127, p=0.22

#Total Organic Carbon
TOC<-env_data$Total_Organic_Carbon

dist.TOC = dist(TOC, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TOC = mantel(Bow_bray_df_no_grvbr, dist.TOC, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TOC # Mantel r=0.0503, p=0.023

#pH
pH<-env_data$pH

dist.pH = dist(pH, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_pH = mantel(Bow_bray_df_no_grvbr, dist.pH, method = "spearman", permutations = 999, na.rm = TRUE)
abund_pH # Mantel r=0.0102, p=0.346

#NO3
NO3<-env_data$Nitrate

dist.NO3 = dist(NO3, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO3 = mantel(Bow_bray_df_no_grvbr, dist.NO3, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO3 # Mantel r=0.01127, p=0.221

#Total organic carbon was a significant predictor of microbiome composition in the Bow River


#Plot scatterplot of abundance Bray Curtis dissimilarity matrix and environmental variables

#Abundance variable distance matrix
Bow_rel_abun  = transform_sample_counts(Bow_ps, function(x) x/sum(x)) 
Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray")
Bow_bray_mat<-as.matrix(Bow_bray)
Bow_bray_df<-as.data.frame(Bow_bray_mat)

#Geographic variable distance matrix
geo_Bow<-sample_data(Bow_rel_abun)
geo_loc_Bow<-geo_Bow[ ,4:5]
geo_Bow_df<-as.data.frame(geo_loc_Bow)

d.geo.Bow = distm(geo_Bow_df, fun = distHaversine)


#temperature - Euclidean
temp_Bow<-sample_data[ ,16]
temp_Bow_df<-as.data.frame(temp_Bow)

dist.temp = dist(temp_Bow_df, method = "euclidean")


#Combine distances matricies into vectors in one combined matrix
aa = as.vector(Bow_bray)
tt = as.vector(dist.temp)
gg = as.vector(d.geo.Bow)

#new data frame with vectorized distance matrices
mat = data.frame(aa,tt)

#Plotting difference in temp with dissimilarity matrix
mm = ggplot(mat, aes(y = aa, x = tt)) + 
    geom_point(size = 3, alpha = 0.5) + 
    labs(x = "Difference in Temperature (C)", y = "Bray-Curtis Dissimilarity") + 
    theme( axis.text.x = element_text(face = "bold",colour = "black", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11, colour = "black"), 
        axis.title= element_text(face = "bold", size = 14, colour = "black"), 
        panel.background = element_blank(), 
        panel.border = element_rect(fill = NA, colour = "black"))
mm


```

#### ACWA
```{r}
#### Beta Diversity Graphs + PERMANOVA ####

ACWA_ps<-subset_samples(sample_ps, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun  = transform_sample_counts(ACWA_ps, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix

ord_ACWA = ordinate(ACWA_rel_abun, method="PCoA", distance="ACWA_bray")

#Scree plot of PCoA
scree <- plot_scree(ord_ACWA)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
        axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1 and 2)

metadata_ACWA <- data.frame(sample_data(ACWA_rel_abun))

#All together
plot_ordination(ACWA_rel_abun, ord_ACWA, color="Family", shape="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [32.8%]")+
  ylab("PC2 [17.9%]")

#Faceted by family and stage (larvae, adults)
plot_ordination(ACWA_rel_abun, ord_ACWA, color="Site", shape="Family") + #Plots the ordination
  theme_bw() +
  rremove("grid")+
  facet_nested_wrap(~Stage+factor(Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera")), scale="free_x", nrow=1)+
  xlab("PC1 [32.8%]")+
  ylab("PC2 [17.9%]")+
  stat_ellipse()

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/PCoA by site and invertebrate family faceted by life stage ACWA Streams.png", height = 7, width=8)


# NMDS 
set.seed(1)
ord_NMDS_ACWA = ordinate(ACWA_rel_abun, method="NMDS", distance=ACWA_bray)
ord_NMDS_ACWA #Stress=0.111


plot_ACWA<-plot_ordination(ACWA_rel_abun, ord_NMDS_ACWA, shape = "Site", color="Family") + 
  theme_bw() +
  rremove("grid")+
  geom_point(size=3)+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=16), 
        legend.text = element_text(size=16), 
        legend.position = "right")+
  labs(color="Family")+
  guides(color=guide_legend(nrow=3, byrow=TRUE, title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  ylim(-0.65, 0.45)+
  xlim(-0.85, 0.95)
  
newtab3 = data.table(plot_ACWA$data)
newtab3$Family <- ordered(newtab3$Family,
levels = c("Hydropsychidae", "Baetidae", "Ephemeroptera"), labels=c("Larval Hydropsychidae", "Larval Baetidae", "Adult Baetidae"))
plot_ACWA$data <- newtab3
print(plot_ACWA)

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites and taxa ACWA only.png", height=8, width=10)


plot_ACWA_sites<-plot_ordination(ACWA_rel_abun, ord_NMDS_ACWA, shape = "Site", color="Family") + 
  theme_bw() +
  rremove("grid")+
  geom_point(size=3)+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=16), 
        legend.text = element_text(size=16), 
        legend.position = "right")+
  labs(color="Family")+
  guides(color=guide_legend(nrow=3, byrow=TRUE, title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  stat_ellipse(type="norm", alpha=1, aes(group=Site))+
  ylim(-0.65, 0.45)+
   xlim(-0.85, 0.95)
  
newtab3 = data.table(plot_ACWA_sites$data)
newtab3$Family <- ordered(newtab3$Family,
levels = c("Hydropsychidae", "Baetidae", "Ephemeroptera"), labels=c("Larval Hydropsychidae", "Larval Baetidae", "Adult Baetidae"))
plot_ACWA_sites$data <- newtab3
print(plot_ACWA_sites)

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across sites ACWA only.png", height=8, width=10)

plot_ACWA_taxa<-plot_ordination(ACWA_rel_abun, ord_NMDS_ACWA, color="Site") +
  theme_bw() +
  rremove("grid")+
  facet_nested_wrap(~Stage+Family, scales="free_x")+
  geom_point(size=3)+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=16), 
        legend.text = element_text(size=16), 
        legend.position = "right")+
  labs(color="Family")+
  guides(color=guide_legend(nrow=3, byrow=TRUE, title="ACWA Stream")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
 stat_ellipse(type="norm", alpha=1, aes(group=Site))

plot_ACWA_taxa
  

newtab5 = data.table(plot_ACWA_taxa$data)
newtab5$Site <- ordered(newtab5$Site,
levels = c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))
newtab5$Family <- ordered(newtab5$Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))
plot_ACWA_taxa$data <- newtab5
print(plot_ACWA_taxa)
#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across taxa and sites with ellipses ACWA only.png", height=8, width=10)

#ggsave(filename="Microbiome_ED/Final analysis/Beta Diversity Graphs/All samples combined/NMDS across taxa ACWA only.png", height=8, width=10)


#PERMANOVA by Site and invertebrate taxa
test.adonis.ACWA <- adonis2(ACWA_bray ~ Site*Family, data = metadata_ACWA)
test.adonis.ACWA #p<0.001. Significant dissimilarity between sites and families but no significant interaction between the two (p=0.088).

#Pairwise comparisons
pairwise.adonis2(ACWA_bray ~ Family, data = metadata_ACWA) 
#All invertebrate types are different from one another when all sites are combined

pairwise.adonis2(ACWA_bray ~ Site, data = metadata_ACWA)
#PCR1 - PCR3 are not significantly different from each other but they are both different from BRR2 (control)

bd_sites_ACWA<-betadisper(ACWA_bray, metadata_ACWA$Site)
anova(bd_sites_ACWA) #p<0.001, F (2, 52) = 23.7

bd_fam_ACWA<-betadisper(ACWA_bray, metadata_ACWA$Family)
anova(bd_fam_ACWA) #p<0.001, F (2, 52) = 16.4


# Hydropsychidae larvae 

hydro_ACWA_ps<-subset_samples(ACWA_ps, Family=="Hydropsychidae")
hydro_ACWA_rel_abun  = transform_sample_counts(hydro_ACWA_ps, function(x) x/sum(x)) 
hydro_ACWA_bray = phyloseq::distance(hydro_ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
metadata_hydro_ACWA <- data.frame(sample_data(hydro_ACWA_rel_abun))

hydro.sites.ACWA <- adonis2(hydro_ACWA_bray ~ Site, data = metadata_hydro_ACWA)
hydro.sites.ACWA #F(2,21)=1.91, P<0.001

pairwise.adonis2(hydro_ACWA_bray ~ Site, data = metadata_hydro_ACWA) 
#All streams are significantly different from each other
bd_hydro_ACWA<-betadisper(hydro_ACWA_bray, metadata_hydro_ACWA$Site)
anova(bd_hydro_ACWA) #p=0.035, F (2, 21) = 3.94

# Baetidae larvae 

baetid_ACWA_ps<-subset_samples(ACWA_ps, Family=="Baetidae")
baetid_ACWA_rel_abun  = transform_sample_counts(baetid_ACWA_ps, function(x) x/sum(x)) 
baetid_ACWA_bray = phyloseq::distance(baetid_ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
metadata_baetid_ACWA <- data.frame(sample_data(baetid_ACWA_rel_abun))

pairwise.adonis2(baetid_ACWA_bray ~ Site, data = metadata_baetid_ACWA) 
#No differences
bd_baetid_ACWA<-betadisper(baetid_ACWA_bray, metadata_baetid_ACWA$Site)
anova(bd_baetid_ACWA) #p=0.98

#Ephemeroptera

ephem_ACWA_ps<-subset_samples(ACWA_ps, Family=="Ephemeroptera")
ephem_ACWA_rel_abun  = transform_sample_counts(ephem_ACWA_ps, function(x) x/sum(x)) 
ephem_ACWA_bray = phyloseq::distance(ephem_ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
metadata_ephem_ACWA <- data.frame(sample_data(ephem_ACWA_rel_abun))

pairwise.adonis2(ephem_ACWA_bray ~ Site, data = metadata_ephem_ACWA) 
#No differences
bd_ephem_ACWA<-betadisper(ephem_ACWA_bray, metadata_ephem_ACWA$Site)
anova(bd_ephem_ACWA) #p=0.07


#### Water quality variables ACWA ####

ACWA_ps<-subset_samples(sample_ps, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun  = transform_sample_counts(ACWA_ps, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
#Convert to dataframe
ACWA_bray_df<-as.matrix(ACWA_bray)
ACWA_bray_df<-as.data.frame(ACWA_bray_df)

set.seed(1)
ord_ACWA_mds = metaMDS(ACWA_bray_df) #Ordinate using NDMS
ord_ACWA_mds #0.111

#Extract environmental variables to correlate with
env_ACWA<-data.frame(sample_data(ACWA_rel_abun))
env_ACWA_2<-env_ACWA[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Ammonia", "pH", "Dissolved_Nitrogen", "Nitrate", "Nitrite", "Dissolved_Organic_Carbon", "Orthophosphate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc)

correlation_matrix_env_ACWA<-as.data.frame(cor(env_ACWA_2), method="spearman") #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

#Sites as colours

env_ACWA$Site<-factor(env_ACWA$Site, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

gg_envfit(ord = ord_ACWA_mds, env = env_ACWA_2, perm = 9999, pt.size = 2, alpha = 0.05, groups=env_ACWA$Site)
#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality ACWA sites.png", height=8, width=12)

#Invertebrate type as colours
gg_envfit(ord = ord_ACWA_mds, env = env_ACWA_2, perm = 9999, pt.size = 2, alpha = 0.05, groups=env_ACWA$Family)

#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality ACWA invert taxa.png", height=8, width=12)

#Fits all environmental variables together
envfit(ord_ACWA_mds, env = env_ACWA_2, permutations = 9999)

#Fitting environmental variables individually (dropping non significant terms)
fit_all_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Dissolved_Organic_Carbon + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_all_ACWA

fit_1_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_1_ACWA

fit_2_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_2_ACWA

fit_3_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_3_ACWA

fit_4_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_4_ACWA

fit_5_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_5_ACWA

fit_6_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite, data = env_ACWA_2, perm = 999)
fit_6_ACWA

fit_7_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Nitrite, data = env_ACWA_2, perm = 999)
fit_7_ACWA

fit_8_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity, data = env_ACWA_2, perm = 999)
fit_8_ACWA

fit_9_ACWA <- envfit(ord_ACWA_mds ~ Conductivity, data = env_ACWA_2, perm = 999)
fit_9_ACWA 


#Conductivity is a significant predictor of microbiome community structure (p=0.005, r^2=0.184)

#Plotting conductivity on microbiome ordination

plot(fit_9_ACWA)

NMDS_ACWA <- metaMDS(ACWA_bray_df, k = 2, trymax = 100, trace = F, autotransform = FALSE, distance="bray")
NMDS_ACWA

arrowmat <- vegan::scores(fit_9_ACWA, display = "bp")
arrowmat #NMDS1 (0.339), NDMS2(-0.262)

arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
# Define the arrow aesthetic mapping
arrow_map = aes(xend = NMDS1, yend = NMDS2, x = 0, y = 0, shape = NULL, color = NULL, 
    label = labels)
label_map = aes(x = 0.339 * NMDS1, y = -0.262 * NMDS2, shape = NULL, color = NULL, 
    label = labels)
# Make a new graphic
arrowhead = arrow(length = unit(0.05, "npc"))
p1 = NMDS_ACWA + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "gray", 
    arrow = arrowhead) + geom_text(label_map, size = 2, data = arrowdf)
p1

## Mantel correlations ##

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
#Convert to dataframe
ACWA_bray_df<-as.matrix(ACWA_bray)
ACWA_bray_df<-as.data.frame(ACWA_bray_df)

#Make environmental variable distance matrix
env_ACWA<-sample_data(ACWA_rel_abun)

#Water temperature
temp_ACWA <-env_ACWA$Water_Temp
dist.temp.ACWA = dist(temp_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_temp_ACWA = mantel(ACWA_bray_df, dist.temp.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_temp_ACWA # Mantel r=0.0282, p=0.146

#Dissolved oxygen
DO_ACWA <-env_ACWA$Dissolved_oxygen
dist.DO.ACWA = dist(DO_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DO_ACWA = mantel(ACWA_bray_df, dist.DO.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DO_ACWA # Mantel r=0.0282, p=0.139

#Conductivity
cond_ACWA <-env_ACWA$Conductivity
dist.cond.ACWA = dist(cond_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_cond_ACWA = mantel(ACWA_bray_df, dist.cond.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_cond_ACWA # Mantel r=0.0554, p=0.088

#Ammonia
NH3_ACWA <-env_ACWA$Ammonia
dist.NH3.ACWA = dist(NH3_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NH3_ACWA = mantel(ACWA_bray_df, dist.NH3.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NH3_ACWA # Mantel r=0.0282, p=0.173

#pH
pH_ACWA <-env_ACWA$pH
dist.pH.ACWA = dist(pH_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_pH_ACWA = mantel(ACWA_bray_df, dist.pH.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_pH_ACWA # Mantel r=0.00598, p=0.314


#Dissolved nitrogen
DN_ACWA <-env_ACWA$Dissolved_Nitrogen
dist.DN.ACWA = dist(DN_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DN_ACWA = mantel(ACWA_bray_df, dist.DN.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DN_ACWA # Mantel r=0.0282, p=0.152


#Nitrate
NO3_ACWA <-env_ACWA$Nitrate
dist.NO3.ACWA = dist(NO3_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO3_ACWA = mantel(ACWA_bray_df, dist.NO3.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO3_ACWA # Mantel r=0.0282, p=0.149


#Nitrite
NO2_ACWA <-env_ACWA$Nitrite
dist.NO2.ACWA = dist(NO2_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO2_ACWA = mantel(ACWA_bray_df, dist.NO2.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO2_ACWA # Mantel r=0.0282, p=0.142


#Dissolved organic carbon
DOC_ACWA <-env_ACWA$Dissolved_Organic_Carbon
dist.DOC.ACWA = dist(DOC_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DOC_ACWA = mantel(ACWA_bray_df, dist.DOC.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DOC_ACWA # Mantel r=0.0195, p=0.161


#Orthophosphate
PO4_ACWA <-env_ACWA$Orthophosphate
dist.PO4.ACWA = dist(PO4_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_PO4_ACWA = mantel(ACWA_bray_df, dist.PO4.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_PO4_ACWA # Mantel r=0.0282, p=0.147

# Plotting Spearman correlations

ACWA_rel_abun_df <- psmelt(ACWA_rel_abun)

ACWA_rel_abun_df_subset<-subset(ACWA_rel_abun_df, select = -c(Total_Phosphorus, Total_Dissolved_Solids, Total_Suspended_Solids, Total_Organic_Carbon))

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Water_Temp, method="spearman")
#-0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Dissolved_oxygen, method="spearman")
#0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Conductivity, method="spearman")
#-0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Ammonia, method="spearman")
#0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Dissolved_Organic_Carbon, method="spearman")
#-0.0107

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Orthophosphate, method="spearman")
#-0.044

library(GGally)

ACWA_cor_plot<-ggcorr(ACWA_rel_abun_df_subset, method =c("everything" ,"spearman"), label=TRUE, hjust = 0.9, size = 3, layout.exp=1)
ACWA_cor_plot
```

### Rarefied

#### Bow River 
```{r}
#Normalize the filtered data by relative abundance 
Bow_ps_rare<-subset_samples(ps_rare, Stream_Type=="Bow River")
Bow_rel_abun_rare  = transform_sample_counts(Bow_ps_rare, function(x) x/sum(x)) 
#transform to relative abundance. Seems to be the best method of normalizing.

Bow_bray_rare = phyloseq::distance(Bow_rel_abun_rare, method="bray") #Bray Curtis distance matrix
ord_Bow_rare = ordinate(Bow_rel_abun_rare, method="PCoA", distance=Bow_bray_rare) #Ordinate using PCoA

#Scree plot of PCoA
scree <- plot_scree(ord_Bow_rare)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
        axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1)

metadata_Bow_rare <- data.frame(sample_data(Bow_rel_abun_rare))

#All together
plot_ordination(Bow_rel_abun_rare, ord_Bow_rare, shape="Site", color="Family") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [11.4%]")+
  ylab("PC2 [8.6%]")

#Faceted by stage (larvae, adults, spiders)
plot_ordination(Bow_rel_abun_rare, ord_Bow_rare, shape="Site", color="Family") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  facet_wrap(~Stage)+
  xlab("PC1 [11.4%]")+
  ylab("PC2 [8.6%]")


# NMDS 
set.seed(1)
ord_NMDS_Bow_rare = ordinate(Bow_rel_abun_rare, method="NMDS", distance=Bow_bray_rare)
ord_NMDS_Bow_rare #Stress=0.177

plotNMDS1_rare<-plot_ordination(Bow_rel_abun_rare, ord_NMDS_Bow_rare, shape="Site", color = "Family") + 
  theme_classic() +
  geom_point(size=3)+
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.3, 'cm'), 
        legend.title = element_text(size=20), 
        legend.text = element_text(size=18), 
        axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18, color="black"),
        legend.position = "right")+
  guides(color=guide_legend(title="Invertebrate Taxa")) +
  scale_color_discrete(palette(c("orange","blue","red","black","magenta","peachpuff")))+
  ylim(-0.5, 0.5)+
  xlim(-0.5, 0.5)

newtab1_rare = data.table(plotNMDS1_rare$data)
newtab1_rare$Family <- ordered(newtab1_rare$Family,
levels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"), labels=c("Larval Hydropsychidae", "Larval Heptageniidae", "Larval Chironomidae", "Larval Perlidae", "Adult Trichoptera", "Adult Ephemeroptera", "Adult Chironomidae", "Araneidae", "Tetragnathidae"))
plotNMDS1_rare$data <- newtab1_rare
print(plotNMDS1_rare)

#PERMANOVA by Site and invertebrate taxa

test.adonis.Bow.rare <- adonis2(Bow_bray_rare ~ Site*Family, data = metadata_Bow_rare)
test.adonis.Bow.rare #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two.

#Pairwise comparisons
pairwise.adonis2(Bow_bray_rare ~ Site, data = metadata_Bow_rare) 
#All sites are diff from eachother, slightly different from the non-rarefied results

pairwise.adonis2(Bow_bray_rare ~ Family, data = metadata_Bow_rare)
#All invertebrate taxa except for Diptera - Ephemeroptera are sig diff when sites are combined. Slightly different from non rarefied data

#Testing variance (beta dispersion) differences between sites, and sample families:
bd_all_site_Bow_rare<-betadisper(Bow_bray_rare, metadata_Bow_rare$Site)
anova(bd_all_site_Bow_rare) #No significant beta dispersion (p=0.204)

bd_all_fam_Bow_rare<-betadisper(Bow_bray_rare, metadata_Bow_rare$Family) 
anova(bd_all_fam_Bow_rare) #significantly different p=0.0036

## Hydropsychidae ##

hydro_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Hydropsychidae")
Bow_rel_abun_hydro_rare  = transform_sample_counts(hydro_Bow_rare, function(x) x/sum(x)) 
Bow_bray_hydro_rare = phyloseq::distance(Bow_rel_abun_hydro_rare, method="bray") 

metadata_Bow_hydro_rare <- data.frame(sample_data(Bow_rel_abun_hydro_rare))

test.adonis.Bow.hydro.rare <- adonis2(Bow_bray_hydro_rare ~ Site, data = metadata_Bow_hydro_rare)
test.adonis.Bow.hydro.rare #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_hydro_rare ~ Site, data = metadata_Bow_hydro_rare)
#All sites except CUSHBR - SUN are sig diff. Same as non rarefied

bd_all_fam_Bow_hydro_rare<-betadisper(Bow_bray_hydro_rare, metadata_Bow_hydro_rare$Site) 
anova(bd_all_fam_Bow_hydro_rare) #no beta dispersion p = 0.15

## Heptageniidae ##

hep_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Heptageniidae")
Bow_rel_abun_hep_rare  = transform_sample_counts(hep_Bow_rare, function(x) x/sum(x)) 
Bow_bray_hep_rare = phyloseq::distance(Bow_rel_abun_hep_rare, method="bray") 

metadata_Bow_hep_rare <- data.frame(sample_data(Bow_rel_abun_hep_rare))

test.adonis.Bow.hep.rare <- adonis2(Bow_bray_hep_rare ~ Site, data = metadata_Bow_hep_rare)
test.adonis.Bow.hep.rare #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_hep_rare ~ Site, data = metadata_Bow_hep_rare)
#All sites are sig. diff from each other. Same as non rarefied

bd_all_fam_Bow_hep_rare<-betadisper(Bow_bray_hep_rare, metadata_Bow_hep_rare$Site) 
anova(bd_all_fam_Bow_hep_rare) #sig beta dispersion p = 0.0074


## Chironomidae ##

chiro_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Chironomidae")
Bow_rel_abun_chiro_rare  = transform_sample_counts(chiro_Bow_rare, function(x) x/sum(x)) 
Bow_bray_chiro_rare = phyloseq::distance(Bow_rel_abun_chiro_rare, method="bray") 

metadata_Bow_chiro_rare <- data.frame(sample_data(Bow_rel_abun_chiro_rare))

test.adonis.Bow.chiro.rare <- adonis2(Bow_bray_chiro_rare ~ Site, data = metadata_Bow_chiro_rare)
test.adonis.Bow.chiro.rare #p=0.003. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_chiro_rare ~ Site, data = metadata_Bow_chiro_rare)
#Cochrane - Policeman Flats are not diff from each other

bd_all_fam_Bow_chiro_rare<-betadisper(Bow_bray_chiro_rare, metadata_Bow_chiro_rare$Site) 
anova(bd_all_fam_Bow_chiro_rare) #no beta dispersion p = 0.28

## Perlidae ##

perl_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Perlidae")
Bow_rel_abun_perl_rare  = transform_sample_counts(perl_Bow_rare, function(x) x/sum(x)) 
Bow_bray_perl_rare = phyloseq::distance(Bow_rel_abun_perl_rare, method="bray") 

metadata_Bow_perl_rare <- data.frame(sample_data(Bow_rel_abun_perl_rare))

test.adonis.Bow.perl.rare <- adonis2(Bow_bray_perl_rare ~ Site, data = metadata_Bow_perl_rare)
test.adonis.Bow.perl.rare #p<0.001. 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_perl_rare ~ Site, data = metadata_Bow_perl_rare)
#All sites were significantly different from each other

bd_all_fam_Bow_perl_rare<-betadisper(Bow_bray_perl_rare, metadata_Bow_perl_rare$Site) 
anova(bd_all_fam_Bow_perl_rare) #sig beta dispersion p = 0.0050

## Trichoptera (adult caddis) ##

trichop_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Trichoptera")
Bow_rel_abun_trichop_rare  = transform_sample_counts(trichop_Bow_rare, function(x) x/sum(x)) 
Bow_bray_trichop_rare = phyloseq::distance(Bow_rel_abun_trichop_rare, method="bray") 

metadata_Bow_trichop_rare <- data.frame(sample_data(Bow_rel_abun_trichop_rare))

test.adonis.Bow.trichop.rare <- adonis2(Bow_bray_trichop_rare ~ Site, data = metadata_Bow_trichop_rare)
test.adonis.Bow.trichop.rare #p = 0.015

#Pairwise comparisons
pairwise.adonis2(Bow_bray_trichop_rare ~ Site, data = metadata_Bow_trichop_rare)
#Only COCH - GRVBR and CUSHBR - GRVBR were different from each other (not all upstream sites were different from GRVBR)

bd_all_fam_Bow_trichop_rare<-betadisper(Bow_bray_trichop_rare, metadata_Bow_trichop_rare$Site) 
anova(bd_all_fam_Bow_trichop_rare) #beta dispersion p = 0.035

## Ephemeroptera (adult mayfly) ##

ephem_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Ephemeroptera")
Bow_rel_abun_ephem_rare = transform_sample_counts(ephem_Bow_rare, function(x) x/sum(x)) 
Bow_bray_ephem_rare = phyloseq::distance(Bow_rel_abun_ephem_rare, method="bray") 

metadata_Bow_ephem_rare <- data.frame(sample_data(Bow_rel_abun_ephem_rare))

test.adonis.Bow.ephem.rare <- adonis2(Bow_bray_ephem_rare ~ Site, data = metadata_Bow_ephem_rare)
test.adonis.Bow.ephem.rare #p < 0.001 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_ephem_rare ~ Site, data = metadata_Bow_ephem_rare)
#Sig diff btw sites include:
#COCH - GRVBR
#COCH - CUSHBR
#COCH - PMF
#GRVBR - CUSHBR
#SUN - CUSHBR
#CUSHBR - PMF

bd_all_fam_Bow_ephem_rare<-betadisper(Bow_bray_ephem_rare, metadata_Bow_ephem_rare$Site) 
anova(bd_all_fam_Bow_ephem_rare) #no beta dispersion p = 0.067


## Diptera (adult chironomidae) ##
dip_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Diptera")
Bow_rel_abun_dip_rare = transform_sample_counts(dip_Bow_rare, function(x) x/sum(x)) 
Bow_bray_dip_rare = phyloseq::distance(Bow_rel_abun_dip_rare, method="bray") 

metadata_Bow_dip_rare <- data.frame(sample_data(Bow_rel_abun_dip_rare))

test.adonis.Bow.dip.rare <- adonis2(Bow_bray_dip_rare ~ Site, data = metadata_Bow_dip_rare)
test.adonis.Bow.dip.rare #p = 0.157

#Pairwise comparisons
pairwise.adonis2(Bow_bray_dip_rare ~ Site, data = metadata_Bow_dip_rare)
#no diff. Slightly different from the non rarefied dataset

bd_all_fam_Bow_dip_rare<-betadisper(Bow_bray_dip_rare, metadata_Bow_dip_rare$Site) 
anova(bd_all_fam_Bow_dip_rare) #no beta dispersion p = 0.589


## Araneidae ##

aran_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Araneidae")
Bow_rel_abun_aran_rare = transform_sample_counts(aran_Bow_rare, function(x) x/sum(x)) 
Bow_bray_aran_rare = phyloseq::distance(Bow_rel_abun_aran_rare, method="bray") 

metadata_Bow_aran_rare <- data.frame(sample_data(Bow_rel_abun_aran_rare))

test.adonis.Bow.aran.rare <- adonis2(Bow_bray_aran_rare ~ Site, data = metadata_Bow_aran_rare)
test.adonis.Bow.aran.rare #p < 0.001 

#Pairwise comparisons
pairwise.adonis2(Bow_bray_aran_rare ~ Site, data = metadata_Bow_aran_rare)
#Sig diff btw sites include:
#COCH - SUN
#COCH - PMF
#GRVBR - SUN
#GRVBR - PMF
#CUSHBR - SUN
#CUSHBR - PMF

bd_all_fam_Bow_aran_rare<-betadisper(Bow_bray_aran_rare, metadata_Bow_aran_rare$Site) 
anova(bd_all_fam_Bow_aran_rare) #no beta dispersion p = 0.85

## Tetragnathidae ##

tetra_Bow_rare<-subset_samples(Bow_ps_rare, Family=="Tetragnathidae")
Bow_rel_abun_tetra_rare = transform_sample_counts(tetra_Bow_rare, function(x) x/sum(x)) 
Bow_bray_tetra_rare = phyloseq::distance(Bow_rel_abun_tetra_rare, method="bray") 

metadata_Bow_tetra_rare <- data.frame(sample_data(Bow_rel_abun_tetra_rare))

test.adonis.Bow.tetra.rare <- adonis2(Bow_bray_tetra_rare ~ Site, data = metadata_Bow_tetra_rare)
test.adonis.Bow.tetra.rare #p = 0.55

#Pairwise comparisons
pairwise.adonis2(Bow_bray_tetra_rare ~ Site, data = metadata_Bow_tetra_rare)
#No significant differences between any of the sites


bd_all_fam_Bow_tetra_rare<-betadisper(Bow_bray_tetra_rare, metadata_Bow_tetra_rare$Site) 
anova(bd_all_fam_Bow_tetra_rare) #no beta dispersion p = 0.503

## Cochrane ##

coch_Bow_rare<-subset_samples(Bow_ps_rare, Site=="Cochrane")
coch_rel_abun_rare = transform_sample_counts(coch_Bow_rare, function(x) x/sum(x)) 
coch_bray_rare = phyloseq::distance(coch_rel_abun_rare, method="bray") 

metadata_coch_rare <- data.frame(sample_data(coch_rel_abun_rare))

test.adonis.Bow.coch.rare <- adonis2(coch_bray_rare ~ Family, data = metadata_coch_rare)
test.adonis.Bow.coch.rare #p < 0.001

#Pairwise comparisons
pairwise.adonis2(coch_bray_rare ~ Family, data = metadata_coch_rare)
#Diptera - Ephemeroptera
#Diptera - Trichoptera
#Ephemeroptera - Trichoptera
#All not sig. diff. Similar to non rarefied

bd_all_fam_Bow_coch_rare<-betadisper(coch_bray_rare, metadata_coch_rare$Family) 
anova(bd_all_fam_Bow_coch_rare) #no beta dispersion p = 0.17.

## Sunalta ##

sun_Bow_rare<-subset_samples(Bow_ps_rare, Site=="Sunalta")
sun_rel_abun_rare = transform_sample_counts(sun_Bow_rare, function(x) x/sum(x)) 
sun_bray_rare = phyloseq::distance(sun_rel_abun_rare, method="bray") 

metadata_sun_rare <- data.frame(sample_data(sun_rel_abun_rare))

test.adonis.Bow.sun.rare <- adonis2(sun_bray_rare ~ Family, data = metadata_sun_rare)
test.adonis.Bow.sun.rare #p < 0.001

#Pairwise comparisons
pairwise.adonis2(sun_bray_rare ~ Family, data = metadata_sun_rare)
#Only Ephemeroptera - Trichoptera not diff from each other

bd_all_fam_Bow_sun_rare<-betadisper(sun_bray_rare, metadata_sun_rare$Family) 
anova(bd_all_fam_Bow_sun_rare) #yes beta dispersion p = 0.0059.


## Cushing Bridge ##

cush_Bow_rare<-subset_samples(Bow_ps_rare, Site=="Cushing Bridge")
cush_rel_abun_rare = transform_sample_counts(cush_Bow_rare, function(x) x/sum(x)) 
cush_bray_rare = phyloseq::distance(cush_rel_abun_rare, method="bray") 

metadata_cush_rare <- data.frame(sample_data(cush_rel_abun_rare))

test.adonis.Bow.cush.rare <- adonis2(cush_bray_rare ~ Family, data = metadata_cush_rare)
test.adonis.Bow.cush.rare #p < 0.001

#Pairwise comparisons
pairwise.adonis2(cush_bray_rare ~ Family, data = metadata_cush_rare)
#Only Trichoptera - Diptera not diff from each other

bd_all_fam_Bow_cush_rare<-betadisper(cush_bray_rare, metadata_cush_rare$Family) 
anova(bd_all_fam_Bow_cush_rare) #yes beta dispersion p = 0.0079.


## Graves Bridge ##

grv_Bow_rare<-subset_samples(Bow_ps_rare, Site=="Graves Bridge")
grv_rel_abun_rare = transform_sample_counts(grv_Bow_rare, function(x) x/sum(x)) 
grv_bray_rare = phyloseq::distance(grv_rel_abun_rare, method="bray") 

metadata_grv_rare <- data.frame(sample_data(grv_rel_abun_rare))

test.adonis.Bow.grv.rare <- adonis2(grv_bray_rare ~ Family, data = metadata_grv_rare)
test.adonis.Bow.grv.rare #p < 0.001

#Pairwise comparisons
pairwise.adonis2(grv_bray_rare ~ Family, data = metadata_grv_rare)
#Only Ephemeroptera - Trichoptera not diff from each other

bd_all_fam_Bow_grv_rare<-betadisper(grv_bray_rare, metadata_grv_rare$Family) 
anova(bd_all_fam_Bow_grv_rare) #no beta dispersion p = 0.396.


## Policeman Flats ##

pmf_Bow_rare<-subset_samples(Bow_ps_rare, Site=="Policeman Flats")
pmf_rel_abun_rare = transform_sample_counts(pmf_Bow_rare, function(x) x/sum(x)) 
pmf_bray_rare = phyloseq::distance(pmf_rel_abun_rare, method="bray") 

metadata_pmf_rare <- data.frame(sample_data(pmf_rel_abun_rare))

test.adonis.Bow.pmf.rare <- adonis2(pmf_bray_rare ~ Family, data = metadata_pmf_rare)
test.adonis.Bow.pmf.rare #p < 0.001

#Pairwise comparisons
pairwise.adonis2(pmf_bray_rare ~ Family, data = metadata_pmf_rare)
#Araneidae - Diptera, Araneidae - Ephemeroptera, and Diptera - Ephemeroptera not diff from each other

bd_all_fam_Bow_pmf_rare<-betadisper(pmf_bray_rare, metadata_pmf_rare$Family) 
anova(bd_all_fam_Bow_pmf_rare) #no beta dispersion p = 0.163.

#A bit different from non rarefied data


#### Water quality variables Bow ####

## EnvFit function ##

#Graves Bridge does not have water quality info so we have to get rid of this site so that the NAs don't mess up the rest of the analysis.
ps_Bow_no_grvbr<-subset_samples(sample_ps, Stream_Type=="Bow River" & Site!="Graves Bridge")

#Transform to relative abundances to standardize
Bow_rel_abun_no_grvbr  = transform_sample_counts(ps_Bow_no_grvbr, function(x) x/sum(x)) 

env_Bow_no_grvbr<-data.frame(sample_data(Bow_rel_abun_no_grvbr))
env_Bow_no_grvbr_2<-env_Bow_no_grvbr[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Total_Phosphorus", "Total_Dissolved_Solids", "Total_Suspended_Solids", "Total_Organic_Carbon", "pH", "Nitrate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc). Ammonia and nitrite weren't selected since they did not differ at all between sites.

env_Bow_no_grvbr_3<-as.data.frame(scale(env_Bow_no_grvbr_2))

#All environmental data isn't normally distributed. Use Spearman correlations. 
shapiro.test(env_Bow_no_grvbr_3$Water_Temp) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Dissolved_oxygen) #non normal
shapiro.test(env_Bow_no_grvbr_3$Conductivity) #non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Phosphorus) #non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Dissolved_Solids) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Suspended_Solids) #Non normal
shapiro.test(env_Bow_no_grvbr_3$Total_Organic_Carbon) #Non normal
shapiro.test(env_Bow_no_grvbr_3$pH) #non normal
shapiro.test(env_Bow_no_grvbr_3$Nitrate) #non normal


correlation_matrix_env<-as.data.frame(cor(env_Bow_no_grvbr_3), method="spearman") #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

#Transform to Bray Curtis distance
Bow_bray_no_grvbr = phyloseq::distance(Bow_rel_abun_no_grvbr, method="bray") #Bray Curtis distance matrix

#Convert to data frame so it can be used in vegan
Bow_bray_df_no_grvbr<-as.matrix(Bow_bray_no_grvbr)
Bow_bray_df_no_grvbr<-as.data.frame(Bow_bray_df_no_grvbr)

#Ordinate with the bray curtis distance matrix
set.seed(123)
ord_Bow_mds_no_grvbr<-metaMDS(Bow_bray_df_no_grvbr) 
ord_Bow_mds_no_grvbr #stress=0.198

#install.packages("ggordiplots")
library(ggordiplots)

gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_3, perm = 9999, pt.size = 2, alpha = 0.05)
#Determines that TOC, pH, Conductivity, TDS, and TSS are significant predictors of microbiome composition. But some of these variables are highly correlated.

envfit(ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_3, permutations = 9999)
#Same conclusions here as above.

#Check correlations between significant variables:
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Dissolved_Solids) #0.998 Highly correlated. One variable must be removed. 
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Suspended_Solids) #0.733 Correlated
cor(env_Bow_no_grvbr_3$Total_Dissolved_Solids, env_Bow_no_grvbr_3$Total_Suspended_Solids) #0.763 Correlated. 
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$Total_Organic_Carbon) #0.0944. Not correlated. These can both stay.
cor(env_Bow_no_grvbr_3$Conductivity, env_Bow_no_grvbr_3$pH) #-0.692 #Somewhat correlated
cor(env_Bow_no_grvbr_3$pH, env_Bow_no_grvbr_3$Total_Organic_Carbon) #0.276. Not correlated
#Remove Total dissolved solids and total suspended solids

env_Bow_no_grvbr_4<-env_Bow_no_grvbr_3[ , c("Conductivity", "pH", "Total_Organic_Carbon")]

envfit_2<-envfit(ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, permutations = 9999)
envfit_2 #All significant but have low R squared.

#Plotting ordination with sites as colours
ggenvfit_Bow_Sites<-gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, perm = 9999, groups = env_Bow_no_grvbr$Site, pt.size = 2, alpha = 0.05, len=0.05, unit="cm")
ggenvfit_Bow_Sites

#ggsave(file="microbiome_analysis/Final analysis/Beta Diversity Graphs/Envfit water quality Bow River sites.png", height=8, width=12)

#Plotting ordination with invertebrate taxa as colours
ggenvfit_Bow_taxa<-gg_envfit(ord = ord_Bow_mds_no_grvbr, env = env_Bow_no_grvbr_4, perm = 9999, groups = env_Bow_no_grvbr$Family, pt.size = 2, alpha = 0.2)

#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality Bow River invert taxa.png", height=8, width=12)

ggarrange(ggenvfit_Bow_Sites, ggenvfit_Bow_taxa, nrow=1, ncool=2)


## Mantel correlations ##

#Abundance distance matrix
#Bow River without GRVBR
#Subset out Graves Bridge because there was no environmental data for that site
ps_Bow_no_grvbr<-subset_samples(sample_ps, Stream_Type=="Bow River" & Site!="Graves Bridge")

#Transform to relative abundances to standardize
Bow_rel_abun_no_grvbr  = transform_sample_counts(ps_Bow_no_grvbr, function(x) x/sum(x)) 

#Convert to Bray-Curtis distance
Bow_bray_no_grvbr = phyloseq::distance(Bow_rel_abun_no_grvbr, method="bray")
#Convert to data frame so it can be used in vegan
Bow_bray_df_no_grvbr<-as.matrix(Bow_bray_no_grvbr)
Bow_bray_df_no_grvbr<-as.data.frame(Bow_bray_df_no_grvbr)


#Bow River with GRVBR included for geographic matrix
#Transform to relative abundances to standardize
Bow_rel_abun  = transform_sample_counts(Bow_ps, function(x) x/sum(x)) 

#Convert to Bray-Curtis distance
Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray")
#Convert to data frame so it can be used in vegan
Bow_bray_df<-as.matrix(Bow_bray)
Bow_bray_df<-as.data.frame(Bow_bray_df)

#Geographic variable distance matrix
geo_Bow<-sample_data(Bow_rel_abun)
geo_loc_Bow<-geo_Bow[ ,4:5]

geo_Bow_mat<-as.matrix(geo_loc_Bow)
geo_Bow_df<-as.data.frame(geo_Bow_mat)

#longitude and latitude 
geo_coord_Bow = data.frame(geo_loc_Bow$Long, geo_loc_Bow$Lat) 

d.geo.Bow = distm(geo_coord_Bow, fun = distHaversine)
dist.geo.Bow = as.dist(d.geo.Bow)

abund_geo  = mantel(Bow_bray_df, dist.geo.Bow, method = "spearman", permutations = 999, na.rm = TRUE)
abund_geo #r = -0.00207, p = 0.842


#Make environmental variable distance matrix
env<-sample_data(Bow_rel_abun_no_grvbr)
env_data<-env[,16:29]

# Environmental data all together # 

env_scale<-scale(env_data)
env_dist<-dist(env_scale, method="euclidean")

abund_env = mantel(Bow_bray_df_no_grvbr, env_dist, method = "spearman", permutations = 999, na.rm = TRUE)
abund_env #r = -0.0004722, p = 0.481. Environment not selecting for the microbiome composition

# Environmental data separated #

#Water temperature
temp <-env_data$Water_Temp
dist.temp = dist(temp, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_temp = mantel(Bow_bray_df_no_grvbr, dist.temp, method = "spearman", permutations = 999, na.rm = TRUE)
abund_temp # Mantel r=0.00564, p=0.342

#Dissolved oxygen
DO<-env_data$Dissolved_oxygen

dist.DO = dist(DO, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DO = mantel(Bow_bray_df_no_grvbr, dist.DO, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DO # Mantel r=-0.01185, p=0.718

#Conductivity
cond<-env_data$Conductivity

dist.cond = dist(cond, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_cond = mantel(Bow_bray_df_no_grvbr, dist.cond, method = "spearman", permutations = 999, na.rm = TRUE)
abund_cond # Mantel r=0.00988, p=0.213

#Total Phosphorus
TP<-env_data$Total_Phosphorus

dist.TP = dist(TP, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TP = mantel(Bow_bray_df_no_grvbr, dist.TP, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TP # Mantel r=0.00502, p=0.381

#Total Dissolved Solids
TDS<-env_data$Total_Dissolved_Solids

dist.TDS = dist(TDS, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TDS = mantel(Bow_bray_df_no_grvbr, dist.TDS, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TDS # Mantel r=0.00988, p=0.197

#Total Suspended Solids
TSS<-env_data$Total_Suspended_Solids

dist.TSS = dist(TSS, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TSS = mantel(Bow_bray_df_no_grvbr, dist.TSS, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TSS # Mantel r=0.01127, p=0.22

#Total Organic Carbon
TOC<-env_data$Total_Organic_Carbon

dist.TOC = dist(TOC, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_TOC = mantel(Bow_bray_df_no_grvbr, dist.TOC, method = "spearman", permutations = 999, na.rm = TRUE)
abund_TOC # Mantel r=0.0503, p=0.023

#pH
pH<-env_data$pH

dist.pH = dist(pH, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_pH = mantel(Bow_bray_df_no_grvbr, dist.pH, method = "spearman", permutations = 999, na.rm = TRUE)
abund_pH # Mantel r=0.0102, p=0.346

#NO3
NO3<-env_data$Nitrate

dist.NO3 = dist(NO3, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO3 = mantel(Bow_bray_df_no_grvbr, dist.NO3, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO3 # Mantel r=0.01127, p=0.221

#Total organic carbon was a significant predictor of microbiome composition in the Bow River


#Plot scatterplot of abundance Bray Curtis dissimilarity matrix and environmental variables

#Abundance variable distance matrix
Bow_rel_abun  = transform_sample_counts(Bow_ps, function(x) x/sum(x)) 
Bow_bray = phyloseq::distance(Bow_rel_abun, method="bray")
Bow_bray_mat<-as.matrix(Bow_bray)
Bow_bray_df<-as.data.frame(Bow_bray_mat)

#Geographic variable distance matrix
geo_Bow<-sample_data(Bow_rel_abun)
geo_loc_Bow<-geo_Bow[ ,4:5]
geo_Bow_df<-as.data.frame(geo_loc_Bow)

d.geo.Bow = distm(geo_Bow_df, fun = distHaversine)


#temperature - Euclidean
temp_Bow<-sample_data[ ,16]
temp_Bow_df<-as.data.frame(temp_Bow)

dist.temp = dist(temp_Bow_df, method = "euclidean")


#Combine distances matricies into vectors in one combined matrix
aa = as.vector(Bow_bray)
tt = as.vector(dist.temp)
gg = as.vector(d.geo.Bow)

#new data frame with vectorized distance matrices
mat = data.frame(aa,tt)

#Plotting difference in temp with dissimilarity matrix
mm = ggplot(mat, aes(y = aa, x = tt)) + 
    geom_point(size = 3, alpha = 0.5) + 
    labs(x = "Difference in Temperature (C)", y = "Bray-Curtis Dissimilarity") + 
    theme( axis.text.x = element_text(face = "bold",colour = "black", size = 12), 
        axis.text.y = element_text(face = "bold", size = 11, colour = "black"), 
        axis.title= element_text(face = "bold", size = 14, colour = "black"), 
        panel.background = element_blank(), 
        panel.border = element_rect(fill = NA, colour = "black"))
mm


```

#### ACWA
```{r}
#### ACWA streams ####

ACWA_ps_rare<-subset_samples(ps_rare, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun_rare  = transform_sample_counts(ACWA_ps_rare, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray_rare = phyloseq::distance(ACWA_rel_abun_rare, method="bray") #Bray Curtis distance matrix

ord_ACWA_rare = ordinate(ACWA_rel_abun_rare, method="PCoA", distance="ACWA_bray_rare")

metadata_ACWA_rare <- data.frame(sample_data(ACWA_rel_abun_rare))

#All together
plot_ordination(ACWA_rel_abun_rare, ord_ACWA_rare, color="Family", shape="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [31.5%]")+
  ylab("PC2 [17.1%]")


# NMDS 
set.seed(1)
ord_NMDS_ACWA_rare = ordinate(ACWA_rel_abun_rare, method="NMDS", distance=ACWA_bray_rare)
ord_NMDS_ACWA_rare #Stress=0.11

#PERMANOVA by Site and invertebrate taxa
test.adonis.ACWA.rare <- adonis2(ACWA_bray_rare ~ Site*Family, data = metadata_ACWA_rare)
test.adonis.ACWA.rare #p<0.001. Significant dissimilarity between sites and families but no significant interaction between the two (p=0.099).

#Pairwise comparisons
pairwise.adonis2(ACWA_bray_rare ~ Family, data = metadata_ACWA_rare) 
#All invertebrate types are different from one another when all sites are combined

pairwise.adonis2(ACWA_bray_rare ~ Site, data = metadata_ACWA_rare)
#PCR1 - PCR3 are not significantly different from each other but they are both different from BRR2 (control)

bd_sites_ACWA_rare<-betadisper(ACWA_bray_rare, metadata_ACWA_rare$Site)
anova(bd_sites_ACWA_rare) #p<0.001, F (2, 52) = 17.7

bd_fam_ACWA_rare<-betadisper(ACWA_bray_rare, metadata_ACWA_rare$Family)
anova(bd_fam_ACWA_rare) #p<0.001, F (2, 52) = 17.7


# Hydropsychidae larvae 

hydro_ACWA_ps_rare<-subset_samples(ACWA_ps_rare, Family=="Hydropsychidae")
hydro_ACWA_rel_abun_rare  = transform_sample_counts(hydro_ACWA_ps_rare, function(x) x/sum(x)) 
hydro_ACWA_bray_rare = phyloseq::distance(hydro_ACWA_rel_abun_rare, method="bray") #Bray Curtis distance matrix
metadata_hydro_ACWA_rare <- data.frame(sample_data(hydro_ACWA_rel_abun_rare))

hydro.sites.ACWA.rare <- adonis2(hydro_ACWA_bray_rare ~ Site, data = metadata_hydro_ACWA_rare)
hydro.sites.ACWA.rare #F(2,21)=1.85, P<0.001

pairwise.adonis2(hydro_ACWA_bray_rare ~ Site, data = metadata_hydro_ACWA_rare) 
#All streams are significantly different from each other
bd_hydro_ACWA_rare<-betadisper(hydro_ACWA_bray_rare, metadata_hydro_ACWA_rare$Site)
anova(bd_hydro_ACWA_rare) #p=0.020

# Baetidae larvae 

baetid_ACWA_ps_rare<-subset_samples(ACWA_ps_rare, Family=="Baetidae")
baetid_ACWA_rel_abun_rare  = transform_sample_counts(baetid_ACWA_ps_rare, function(x) x/sum(x)) 
baetid_ACWA_bray_rare = phyloseq::distance(baetid_ACWA_rel_abun_rare, method="bray") #Bray Curtis distance matrix
metadata_baetid_ACWA_rare <- data.frame(sample_data(baetid_ACWA_rel_abun_rare))

pairwise.adonis2(baetid_ACWA_bray_rare ~ Site, data = metadata_baetid_ACWA_rare) 
#No differences
bd_baetid_ACWA_rare<-betadisper(baetid_ACWA_bray_rare, metadata_baetid_ACWA_rare$Site)
anova(bd_baetid_ACWA_rare) #p=0.99

#Ephemeroptera

ephem_ACWA_ps_rare<-subset_samples(ACWA_ps_rare, Family=="Ephemeroptera")
ephem_ACWA_rel_abun_rare  = transform_sample_counts(ephem_ACWA_ps_rare, function(x) x/sum(x)) 
ephem_ACWA_bray_rare = phyloseq::distance(ephem_ACWA_rel_abun_rare, method="bray") #Bray Curtis distance matrix
metadata_ephem_ACWA_rare <- data.frame(sample_data(ephem_ACWA_rel_abun_rare))

pairwise.adonis2(ephem_ACWA_bray_rare ~ Site, data = metadata_ephem_ACWA_rare) 
#No differences
bd_ephem_ACWA_rare<-betadisper(ephem_ACWA_bray_rare, metadata_ephem_ACWA_rare$Site)
anova(bd_ephem_ACWA_rare) #p=0.072

#Same as non rarefied results


#### Water quality variables ACWA ####

ACWA_ps<-subset_samples(sample_ps, Stream_Type=="ACWA Streams")

#Transform to relative abundance to standardize data
ACWA_rel_abun  = transform_sample_counts(ACWA_ps, function(x) x/sum(x)) #log transforming the data 

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
#Convert to dataframe
ACWA_bray_df<-as.matrix(ACWA_bray)
ACWA_bray_df<-as.data.frame(ACWA_bray_df)

set.seed(1)
ord_ACWA_mds = metaMDS(ACWA_bray_df) #Ordinate using NDMS
ord_ACWA_mds #0.111

#Extract environmental variables to correlate with
env_ACWA<-data.frame(sample_data(ACWA_rel_abun))
env_ACWA_2<-env_ACWA[ ,c("Water_Temp", "Dissolved_oxygen", "Conductivity", "Ammonia", "pH", "Dissolved_Nitrogen", "Nitrate", "Nitrite", "Dissolved_Organic_Carbon", "Orthophosphate")] #Selects only the numerical variables (water temp, conductivity, dissolved oxygen, etc)

correlation_matrix_env_ACWA<-as.data.frame(cor(env_ACWA_2), method="spearman") #Some variables are highly correlated. Test with all of them to see if any are significant then get rid of the correlated variable that is less significant based on p value. 

#Sites as colours

env_ACWA$Site<-factor(env_ACWA$Site, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

gg_envfit(ord = ord_ACWA_mds, env = env_ACWA_2, perm = 9999, pt.size = 2, alpha = 0.05, groups=env_ACWA$Site)
#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality ACWA sites.png", height=8, width=12)

#Invertebrate type as colours
gg_envfit(ord = ord_ACWA_mds, env = env_ACWA_2, perm = 9999, pt.size = 2, alpha = 0.05, groups=env_ACWA$Family)

#ggsave(file="Microbiome_ED/Final analysis/Beta Diversity Graphs/Envfit water quality ACWA invert taxa.png", height=8, width=12)

#Fits all environmental variables together
envfit(ord_ACWA_mds, env = env_ACWA_2, permutations = 9999)

#Fitting environmental variables individually (dropping non significant terms)
fit_all_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Dissolved_Organic_Carbon + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_all_ACWA

fit_1_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Ammonia + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_1_ACWA

fit_2_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + pH + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_2_ACWA

fit_3_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Dissolved_oxygen + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_3_ACWA

fit_4_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrate + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_4_ACWA

fit_5_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite + Orthophosphate, data = env_ACWA_2, perm = 999)
fit_5_ACWA

fit_6_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Dissolved_Nitrogen + Nitrite, data = env_ACWA_2, perm = 999)
fit_6_ACWA

fit_7_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity + Nitrite, data = env_ACWA_2, perm = 999)
fit_7_ACWA

fit_8_ACWA <- envfit(ord_ACWA_mds ~ Water_Temp + Conductivity, data = env_ACWA_2, perm = 999)
fit_8_ACWA

fit_9_ACWA <- envfit(ord_ACWA_mds ~ Conductivity, data = env_ACWA_2, perm = 999)
fit_9_ACWA 


#Conductivity is a significant predictor of microbiome community structure (p=0.005, r^2=0.184)

#Plotting conductivity on microbiome ordination

plot(fit_9_ACWA)

NMDS_ACWA <- metaMDS(ACWA_bray_df, k = 2, trymax = 100, trace = F, autotransform = FALSE, distance="bray")
NMDS_ACWA

arrowmat <- vegan::scores(fit_9_ACWA, display = "bp")
arrowmat #NMDS1 (0.339), NDMS2(-0.262)

arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
# Define the arrow aesthetic mapping
arrow_map = aes(xend = NMDS1, yend = NMDS2, x = 0, y = 0, shape = NULL, color = NULL, 
    label = labels)
label_map = aes(x = 0.339 * NMDS1, y = -0.262 * NMDS2, shape = NULL, color = NULL, 
    label = labels)
# Make a new graphic
arrowhead = arrow(length = unit(0.05, "npc"))
p1 = NMDS_ACWA + geom_segment(arrow_map, linewidth = 0.5, data = arrowdf, color = "gray", 
    arrow = arrowhead) + geom_text(label_map, size = 2, data = arrowdf)
p1

## Mantel correlations ##

#Bray Curtis distance matrix
ACWA_bray = phyloseq::distance(ACWA_rel_abun, method="bray") #Bray Curtis distance matrix
#Convert to dataframe
ACWA_bray_df<-as.matrix(ACWA_bray)
ACWA_bray_df<-as.data.frame(ACWA_bray_df)

#Make environmental variable distance matrix
env_ACWA<-sample_data(ACWA_rel_abun)

#Water temperature
temp_ACWA <-env_ACWA$Water_Temp
dist.temp.ACWA = dist(temp_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_temp_ACWA = mantel(ACWA_bray_df, dist.temp.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_temp_ACWA # Mantel r=0.0282, p=0.146

#Dissolved oxygen
DO_ACWA <-env_ACWA$Dissolved_oxygen
dist.DO.ACWA = dist(DO_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DO_ACWA = mantel(ACWA_bray_df, dist.DO.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DO_ACWA # Mantel r=0.0282, p=0.139

#Conductivity
cond_ACWA <-env_ACWA$Conductivity
dist.cond.ACWA = dist(cond_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_cond_ACWA = mantel(ACWA_bray_df, dist.cond.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_cond_ACWA # Mantel r=0.0554, p=0.088

#Ammonia
NH3_ACWA <-env_ACWA$Ammonia
dist.NH3.ACWA = dist(NH3_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NH3_ACWA = mantel(ACWA_bray_df, dist.NH3.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NH3_ACWA # Mantel r=0.0282, p=0.173

#pH
pH_ACWA <-env_ACWA$pH
dist.pH.ACWA = dist(pH_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_pH_ACWA = mantel(ACWA_bray_df, dist.pH.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_pH_ACWA # Mantel r=0.00598, p=0.314


#Dissolved nitrogen
DN_ACWA <-env_ACWA$Dissolved_Nitrogen
dist.DN.ACWA = dist(DN_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DN_ACWA = mantel(ACWA_bray_df, dist.DN.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DN_ACWA # Mantel r=0.0282, p=0.152


#Nitrate
NO3_ACWA <-env_ACWA$Nitrate
dist.NO3.ACWA = dist(NO3_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO3_ACWA = mantel(ACWA_bray_df, dist.NO3.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO3_ACWA # Mantel r=0.0282, p=0.149


#Nitrite
NO2_ACWA <-env_ACWA$Nitrite
dist.NO2.ACWA = dist(NO2_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_NO2_ACWA = mantel(ACWA_bray_df, dist.NO2.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_NO2_ACWA # Mantel r=0.0282, p=0.142


#Dissolved organic carbon
DOC_ACWA <-env_ACWA$Dissolved_Organic_Carbon
dist.DOC.ACWA = dist(DOC_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_DOC_ACWA = mantel(ACWA_bray_df, dist.DOC.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_DOC_ACWA # Mantel r=0.0195, p=0.161


#Orthophosphate
PO4_ACWA <-env_ACWA$Orthophosphate
dist.PO4.ACWA = dist(PO4_ACWA, method = "euclidean")

#Mantel test with Spearman non parametric correlations
abund_PO4_ACWA = mantel(ACWA_bray_df, dist.PO4.ACWA, method = "spearman", permutations = 999, na.rm = TRUE)
abund_PO4_ACWA # Mantel r=0.0282, p=0.147

# Plotting Spearman correlations

ACWA_rel_abun_df <- psmelt(ACWA_rel_abun)

ACWA_rel_abun_df_subset<-subset(ACWA_rel_abun_df, select = -c(Total_Phosphorus, Total_Dissolved_Solids, Total_Suspended_Solids, Total_Organic_Carbon))

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Water_Temp, method="spearman")
#-0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Dissolved_oxygen, method="spearman")
#0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Conductivity, method="spearman")
#-0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Ammonia, method="spearman")
#0.044

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Dissolved_Organic_Carbon, method="spearman")
#-0.0107

cor(ACWA_rel_abun_df$Abundance, ACWA_rel_abun_df$Orthophosphate, method="spearman")
#-0.044

library(GGally)

ACWA_cor_plot<-ggcorr(ACWA_rel_abun_df_subset, method =c("everything" ,"spearman"), label=TRUE, hjust = 0.9, size = 3, layout.exp=1)
ACWA_cor_plot
```


# Differential Abundance

Done on non-rarefied data to compare the mean absolute abundance between two ecosystems/samples/sites. The DESeq2 model internally corrects for library size, so transformed or normalized values such as counts scaled by library size should not be used as input. It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order.

Ignoring excess zeros in zero inflated data can lead to overestimation of dispersion and subsequent reduction in statistical power to detect differentially abundant features. To correct for this I used 'poscounts' which calculates a modified geometric mean by taking the nth root of the product of the non-zero counts. The prevalence threshold was increased to 3% to reduce the chance of false positives. Adjusted p-values using Benjamini-Hochberg correction were used to determine significance with an alpha of 0.001.

Most supported DA tests:
ANCOM-BC (Lin & Peddada, 2020. Biofilms and Microbiomes.) (Nearing et al., 2022. Nature Communications.) (Yang & Chen, 2022. Microbiome.)

ALDEX2 (Nearing et al., 2022. Nature Communications.) (Yang & Chen, 2022. Microbiome.)

DAtest package can test a bunch of different DA methods and see which one is the best

## Comparing un-exposed sites to exposed sites - Bow River (inverts separated)
```{r}
#### All invertebrate taxa ####
#Wastewater exposed sites include Policeman Flats and Graves Bridge whereas unexposed sites are Cochrane, Sunalta, and Cushing Bridge.

deseq_exposed <- phyloseq_to_deseq2(sample_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed$Exposure <-relevel(deseq_exposed$Exposure, "Unexposed")

deseq_exposed <- DESeq(deseq_exposed, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed <- results(deseq_exposed, cooksCutoff = FALSE)
res_deseq_exposed
alpha = 0.001
sigtab_exposed = res_deseq_exposed[which(res_deseq_exposed$padj < alpha), ]
sigtab_exposed = cbind(as(sigtab_exposed, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_exposed), ], "matrix"))

#write.csv(as.data.frame(sigtab_exposed), file="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_Wastewater_Exposed_vs_Unexposed_Sites.csv")


x_exposed = tapply(sigtab_exposed$log2FoldChange, sigtab_exposed$Phylum, function(x) max(x))
x_exposed = sort(x_exposed, TRUE)
sigtab_exposed$Phylum = factor(as.character(sigtab_exposed$Phylum), levels=names(x_exposed))
# Genus
x_exposed = tapply(sigtab_exposed$log2FoldChange, sigtab_exposed$Genus, function(x) max(x))
x_exposed = sort(x_exposed, TRUE)
sigtab_exposed$Genus = factor(as.character(sigtab_exposed$Genus), levels=sort(names(x_exposed)))

sigtab_exposed$Phylum<-factor(sigtab_exposed$Phylum, levels=c("Proteobacteria", "Bacteroidota", "Firmicutes", "Cyanobacteria", "Myxococcota", "Actinobacteriota", "Chloroflexi", "Synergistota"))

sigtab_exposed_2<-subset(sigtab_exposed, Genus=="AAP99" | Genus=="Candidatus Bacilloplasma"| Genus=="Ellin6067" | Genus=="Emticicia" | Genus=="Flavobacterium" | Genus=="Fluviicola" | Genus=="Gemmobacter" | Genus=="Haliangium" | Genus=="Hydrogenophaga" | Genus=="Methylotenera" | Genus=="Nitrosomonas" | Genus=="Nocardioides" | Genus=="Pseudorhodobacter" | Genus=="Rhizorhapis" | Genus=="Rhodobacter" | Genus=="Rhodoferax" | Genus=="Rurimicrobium" | Genus=="Serratia" | Genus=="Sphingorhabdus" | Genus=="Tyzzerella")

#All differentially abundant bacteria
ggplot(sigtab_exposed, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple", "pink", "lightcoral", "grey"))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites.png", height=15, width=22)

# Exposed sites contain many of the same effluent bacteria from the ACWA streams. Could indicate that these sites are affected by wastewater to some extent.


#Only wastewater associated bacteria subsetted

ggplot(sigtab_exposed_2, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple", "pink", "lightcoral", "grey"))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites effluent associated bacteria subset.png", height=15, width=22)

#### Spiders ####

spider_ps<-subset_samples(sample_ps, Stage=="Spiders")

deseq_exposed_spi <- phyloseq_to_deseq2(spider_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_spi$Exposure <-relevel(deseq_exposed_spi$Exposure, "Unexposed")

deseq_exposed_spi <- DESeq(deseq_exposed_spi, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_spi <- results(deseq_exposed_spi, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_spi = res_deseq_exposed_spi[which(res_deseq_exposed_spi$padj < alpha), ]
sigtab_exposed_spi = cbind(as(sigtab_exposed_spi, "data.frame"), as(tax_table(spider_ps)[rownames(sigtab_exposed_spi), ], "matrix"))

x_exposed_spi = tapply(sigtab_exposed_spi$log2FoldChange, sigtab_exposed_spi$Phylum, function(x) max(x))
x_exposed_spi = sort(x_exposed_spi, TRUE)
sigtab_exposed_spi$Phylum = factor(as.character(sigtab_exposed_spi$Phylum), levels=names(x_exposed_spi))
# Genus
x_exposed_spi = tapply(sigtab_exposed_spi$log2FoldChange, sigtab_exposed_spi$Genus, function(x) max(x))
x_exposed_spi = sort(x_exposed_spi, TRUE)
sigtab_exposed_spi$Genus = factor(as.character(sigtab_exposed_spi$Genus), levels=sort(names(x_exposed_spi)))


ggplot(sigtab_exposed_spi, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple", "pink", "lightcoral", "grey"))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites spiders only.png", height=15, width=22)


#### Aquatic adults ####

aq_ad_ps<-subset_samples(sample_ps, Stage=="Adults")

deseq_exposed_ad <- phyloseq_to_deseq2(aq_ad_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_ad$Exposure <-relevel(deseq_exposed_ad$Exposure, "Unexposed")

deseq_exposed_ad <- DESeq(deseq_exposed_ad, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_ad <- results(deseq_exposed_ad, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_ad = res_deseq_exposed_ad[which(res_deseq_exposed_ad$padj < alpha), ]
sigtab_exposed_ad = cbind(as(sigtab_exposed_ad, "data.frame"), as(tax_table(aq_ad_ps)[rownames(sigtab_exposed_ad), ], "matrix"))

x_exposed_ad = tapply(sigtab_exposed_ad$log2FoldChange, sigtab_exposed_ad$Phylum, function(x) max(x))
x_exposed_ad = sort(x_exposed_ad, TRUE)
sigtab_exposed_ad$Phylum = factor(as.character(sigtab_exposed_ad$Phylum), levels=names(x_exposed_ad))
# Genus
x_exposed_ad = tapply(sigtab_exposed_ad$log2FoldChange, sigtab_exposed_ad$Genus, function(x) max(x))
x_exposed_ad = sort(x_exposed_ad, TRUE)
sigtab_exposed_ad$Genus = factor(as.character(sigtab_exposed_ad$Genus), levels=sort(names(x_exposed_ad)))


ggplot(sigtab_exposed_ad, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple", "pink", "lightcoral", "grey"))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites aquatic adults only.png", height=15, width=22)


#### Aquatic larvae ####

aq_lar_ps<-subset_samples(sample_ps, Stage=="Larvae")

deseq_exposed_lar <- phyloseq_to_deseq2(aq_lar_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_lar$Exposure <-relevel(deseq_exposed_lar$Exposure, "Unexposed")

deseq_exposed_lar <- DESeq(deseq_exposed_lar, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_lar <- results(deseq_exposed_lar, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_lar = res_deseq_exposed_lar[which(res_deseq_exposed_lar$padj < alpha), ]
sigtab_exposed_lar = cbind(as(sigtab_exposed_lar, "data.frame"), as(tax_table(aq_lar_ps)[rownames(sigtab_exposed_lar), ], "matrix"))

x_exposed_lar = tapply(sigtab_exposed_lar$log2FoldChange, sigtab_exposed_lar$Phylum, function(x) max(x))
x_exposed_lar = sort(x_exposed_lar, TRUE)
sigtab_exposed_lar$Phylum = factor(as.character(sigtab_exposed_lar$Phylum), levels=names(x_exposed_lar))
# Genus
x_exposed_lar = tapply(sigtab_exposed_lar$log2FoldChange, sigtab_exposed_lar$Genus, function(x) max(x))
x_exposed_lar = sort(x_exposed_lar, TRUE)
sigtab_exposed_lar$Genus = factor(as.character(sigtab_exposed_lar$Genus), levels=sort(names(x_exposed_lar)))
#170 differentially abundant taxa

ggplot(sigtab_exposed_lar, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple", "pink", "lightcoral", "grey"))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites aquatic larvae only.png", height=15, width=22)


#### Hydropsychidae larvae ####

#Exposed vs non wastewater exposed sites

hydro_ps<-subset_samples(sample_ps, Family=="Hydropsychidae" & Stream_Type=="Bow River")

deseq_exposed_hydro <- phyloseq_to_deseq2(hydro_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_hydro$Exposure <-relevel(deseq_exposed_hydro$Exposure, "Unexposed")

deseq_exposed_hydro <- DESeq(deseq_exposed_hydro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_hydro <- results(deseq_exposed_hydro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_hydro = res_deseq_exposed_hydro[which(res_deseq_exposed_hydro$padj < alpha), ]
sigtab_exposed_hydro = cbind(as(sigtab_exposed_hydro, "data.frame"), as(tax_table(hydro_ps)[rownames(sigtab_exposed_hydro), ], "matrix"))

x_exposed_hydro = tapply(sigtab_exposed_hydro$log2FoldChange, sigtab_exposed_hydro$Phylum, function(x) max(x))
x_exposed_hydro = sort(x_exposed_hydro, TRUE)
sigtab_exposed_hydro$Phylum = factor(as.character(sigtab_exposed_hydro$Phylum), levels=names(x_exposed_hydro))
# Genus
x_exposed_hydro = tapply(sigtab_exposed_hydro$log2FoldChange, sigtab_exposed_hydro$Genus, function(x) max(x))
x_exposed_hydro = sort(x_exposed_hydro, TRUE)
sigtab_exposed_hydro$Genus = factor(as.character(sigtab_exposed_hydro$Genus), levels=sort(names(x_exposed_hydro)))
#110 differentially abundant taxa

diff_abund_hydro_plot<-ggplot(sigtab_exposed_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hydro_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites hydropsychidae larvae only.png", height=15, width=22)


#COCH - SUN

hydro_sun_coch<-subset_samples(hydro_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_hydro <- phyloseq_to_deseq2(hydro_sun_coch, ~ Site)

deseq_sun_coch_hydro <- DESeq(deseq_sun_coch_hydro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_sun_coch_hydro <- results(deseq_sun_coch_hydro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_hydro = res_deseq_sun_coch_hydro[which(res_deseq_sun_coch_hydro$padj < alpha), ]
sigtab_sun_coch_hydro = cbind(as(sigtab_sun_coch_hydro, "data.frame"), as(tax_table(hydro_sun_coch)[rownames(sigtab_sun_coch_hydro), ], "matrix"))

x_sun_coch_hydro = tapply(sigtab_sun_coch_hydro$log2FoldChange, sigtab_sun_coch_hydro$Phylum, function(x) max(x))
x_sun_coch_hydro = sort(x_sun_coch_hydro, TRUE)
sigtab_sun_coch_hydro$Phylum = factor(as.character(sigtab_sun_coch_hydro$Phylum), levels=names(x_sun_coch_hydro))
# Genus
x_sun_coch_hydro = tapply(sigtab_sun_coch_hydro$log2FoldChange, sigtab_sun_coch_hydro$Genus, function(x) max(x))
x_sun_coch_hydro = sort(x_sun_coch_hydro, TRUE)
sigtab_sun_coch_hydro$Genus = factor(as.character(sigtab_sun_coch_hydro$Genus), levels=sort(names(x_sun_coch_hydro)))
#6 observations

diff_abund_hydro_sun_coch_plot<-ggplot(sigtab_sun_coch_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hydro_sun_coch_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River Sunalta vs Cochrane hydropsychidae larvae only.png", height=15, width=22)


#CUSHBR - COCH

hydro_cush_coch<-subset_samples(hydro_ps, Site=="Cushing Bridge" | Site=="Cochrane")

deseq_cush_coch_hydro <- phyloseq_to_deseq2(hydro_cush_coch, ~ Site)

deseq_cush_coch_hydro <- DESeq(deseq_cush_coch_hydro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_cush_coch_hydro <- results(deseq_cush_coch_hydro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_hydro = res_deseq_cush_coch_hydro[which(res_deseq_cush_coch_hydro$padj < alpha), ]
sigtab_cush_coch_hydro = cbind(as(sigtab_cush_coch_hydro, "data.frame"), as(tax_table(hydro_cush_coch)[rownames(sigtab_cush_coch_hydro), ], "matrix"))

x_cush_coch_hydro = tapply(sigtab_cush_coch_hydro$log2FoldChange, sigtab_cush_coch_hydro$Phylum, function(x) max(x))
x_cush_coch_hydro = sort(x_cush_coch_hydro, TRUE)
sigtab_cush_coch_hydro$Phylum = factor(as.character(sigtab_cush_coch_hydro$Phylum), levels=names(x_cush_coch_hydro))
# Genus
x_cush_coch_hydro = tapply(sigtab_cush_coch_hydro$log2FoldChange, sigtab_cush_coch_hydro$Genus, function(x) max(x))
x_cush_coch_hydro = sort(x_cush_coch_hydro, TRUE)
sigtab_cush_coch_hydro$Genus = factor(as.character(sigtab_cush_coch_hydro$Genus), levels=sort(names(x_cush_coch_hydro)))
#11 observations

diff_abund_hydro_cush_coch_plot<-ggplot(sigtab_cush_coch_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hydro_cush_coch_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River Cushing Bridge vs Cochrane hydropsychidae larvae only.png", height=15, width=22)

#GRVBR - COCH

hydro_grvbr_coch<-subset_samples(hydro_ps, Site=="Graves Bridge" | Site=="Cochrane")

deseq_grvbr_coch_hydro <- phyloseq_to_deseq2(hydro_grvbr_coch, ~ Site)

deseq_grvbr_coch_hydro <- DESeq(deseq_grvbr_coch_hydro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_grvbr_coch_hydro <- results(deseq_grvbr_coch_hydro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_hydro = res_deseq_grvbr_coch_hydro[which(res_deseq_grvbr_coch_hydro$padj < alpha), ]
sigtab_grvbr_coch_hydro = cbind(as(sigtab_grvbr_coch_hydro, "data.frame"), as(tax_table(hydro_grvbr_coch)[rownames(sigtab_grvbr_coch_hydro), ], "matrix"))

x_grvbr_coch_hydro = tapply(sigtab_grvbr_coch_hydro$log2FoldChange, sigtab_grvbr_coch_hydro$Phylum, function(x) max(x))
x_grvbr_coch_hydro = sort(x_grvbr_coch_hydro, TRUE)
sigtab_grvbr_coch_hydro$Phylum = factor(as.character(sigtab_grvbr_coch_hydro$Phylum), levels=names(x_grvbr_coch_hydro))
# Genus
x_grvbr_coch_hydro = tapply(sigtab_grvbr_coch_hydro$log2FoldChange, sigtab_grvbr_coch_hydro$Genus, function(x) max(x))
x_grvbr_coch_hydro = sort(x_grvbr_coch_hydro, TRUE)
sigtab_grvbr_coch_hydro$Genus = factor(as.character(sigtab_grvbr_coch_hydro$Genus), levels=sort(names(x_grvbr_coch_hydro)))
#19 observations

diff_abund_hydro_grvbr_coch_plot<-ggplot(sigtab_grvbr_coch_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hydro_grvbr_coch_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River Graves Bridge vs Cochrane hydropsychidae larvae only.png", height=15, width=22)


#PMF - COCH

hydro_pmf_coch<-subset_samples(hydro_ps, Site=="Policeman Flats" | Site=="Cochrane")

deseq_pmf_coch_hydro <- phyloseq_to_deseq2(hydro_pmf_coch, ~ Site)

deseq_pmf_coch_hydro <- DESeq(deseq_pmf_coch_hydro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_pmf_coch_hydro <- results(deseq_pmf_coch_hydro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_hydro = res_deseq_pmf_coch_hydro[which(res_deseq_pmf_coch_hydro$padj < alpha), ]
sigtab_pmf_coch_hydro = cbind(as(sigtab_pmf_coch_hydro, "data.frame"), as(tax_table(hydro_pmf_coch)[rownames(sigtab_pmf_coch_hydro), ], "matrix"))

x_pmf_coch_hydro = tapply(sigtab_pmf_coch_hydro$log2FoldChange, sigtab_pmf_coch_hydro$Phylum, function(x) max(x))
x_pmf_coch_hydro = sort(x_pmf_coch_hydro, TRUE)
sigtab_pmf_coch_hydro$Phylum = factor(as.character(sigtab_pmf_coch_hydro$Phylum), levels=names(x_pmf_coch_hydro))
# Genus
x_pmf_coch_hydro = tapply(sigtab_pmf_coch_hydro$log2FoldChange, sigtab_pmf_coch_hydro$Genus, function(x) max(x))
x_pmf_coch_hydro = sort(x_pmf_coch_hydro, TRUE)
sigtab_pmf_coch_hydro$Genus = factor(as.character(sigtab_pmf_coch_hydro$Genus), levels=sort(names(x_pmf_coch_hydro)))
#156 observations

diff_abund_hydro_pmf_coch_plot<-ggplot(sigtab_pmf_coch_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hydro_pmf_coch_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River PMF vs Cochrane hydropsychidae larvae only.png", height=15, width=22)


#### Adult Trichoptera ####

trichop_ps<-subset_samples(sample_ps, Family=="Trichoptera")

deseq_exposed_trichop <- phyloseq_to_deseq2(trichop_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_trichop$Exposure <-relevel(deseq_exposed_trichop$Exposure, "Unexposed")

deseq_exposed_trichop <- DESeq(deseq_exposed_trichop, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_trichop <- results(deseq_exposed_trichop, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_trichop = res_deseq_exposed_trichop[which(res_deseq_exposed_trichop$padj < alpha), ]
sigtab_exposed_trichop = cbind(as(sigtab_exposed_trichop, "data.frame"), as(tax_table(trichop_ps)[rownames(sigtab_exposed_trichop), ], "matrix"))

x_exposed_trichop = tapply(sigtab_exposed_trichop$log2FoldChange, sigtab_exposed_trichop$Phylum, function(x) max(x))
x_exposed_trichop = sort(x_exposed_trichop, TRUE)
sigtab_exposed_trichop$Phylum = factor(as.character(sigtab_exposed_trichop$Phylum), levels=names(x_exposed_trichop))
# Genus
x_exposed_trichop = tapply(sigtab_exposed_trichop$log2FoldChange, sigtab_exposed_trichop$Genus, function(x) max(x))
x_exposed_trichop = sort(x_exposed_trichop, TRUE)
sigtab_exposed_trichop$Genus = factor(as.character(sigtab_exposed_trichop$Genus), levels=sort(names(x_exposed_trichop)))
#23 differentially abundant taxa

diff_abund_trichop_plot<-ggplot(sigtab_exposed_trichop, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")

diff_abund_trichop_plot

#COCH - SUN

trichop_sun_coch<-subset_samples(trichop_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_trichop <- phyloseq_to_deseq2(trichop_sun_coch, ~ Site)

deseq_sun_coch_trichop <- DESeq(deseq_sun_coch_trichop, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_sun_coch_trichop <- results(deseq_sun_coch_trichop, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_trichop = res_deseq_sun_coch_trichop[which(res_deseq_sun_coch_trichop$padj < alpha), ]
sigtab_sun_coch_trichop = cbind(as(sigtab_sun_coch_trichop, "data.frame"), as(tax_table(trichop_sun_coch)[rownames(sigtab_sun_coch_trichop), ], "matrix"))

x_sun_coch_trichop = tapply(sigtab_sun_coch_trichop$log2FoldChange, sigtab_sun_coch_trichop$Phylum, function(x) max(x))
x_sun_coch_trichop = sort(x_sun_coch_trichop, TRUE)
sigtab_sun_coch_trichop$Phylum = factor(as.character(sigtab_sun_coch_trichop$Phylum), levels=names(x_sun_coch_trichop))
# Genus
x_sun_coch_trichop = tapply(sigtab_sun_coch_trichop$log2FoldChange, sigtab_sun_coch_trichop$Genus, function(x) max(x))
x_sun_coch_trichop = sort(x_sun_coch_trichop, TRUE)
sigtab_sun_coch_trichop$Genus = factor(as.character(sigtab_sun_coch_trichop$Genus), levels=sort(names(x_sun_coch_trichop)))
#3 observations

diff_abund_trichop_sun_coch_plot<-ggplot(sigtab_sun_coch_trichop, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_trichop_sun_coch_plot


#COCH - CUSHBR

trichop_cush_coch<-subset_samples(trichop_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_trichop <- phyloseq_to_deseq2(trichop_cush_coch, ~ Site)

deseq_cush_coch_trichop <- DESeq(deseq_cush_coch_trichop, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_cush_coch_trichop <- results(deseq_cush_coch_trichop, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_trichop = res_deseq_cush_coch_trichop[which(res_deseq_cush_coch_trichop$padj < alpha), ]
sigtab_cush_coch_trichop = cbind(as(sigtab_cush_coch_trichop, "data.frame"), as(tax_table(trichop_cush_coch)[rownames(sigtab_cush_coch_trichop), ], "matrix"))

x_cush_coch_trichop = tapply(sigtab_cush_coch_trichop$log2FoldChange, sigtab_cush_coch_trichop$Phylum, function(x) max(x))
x_cush_coch_trichop = sort(x_cush_coch_trichop, TRUE)
sigtab_cush_coch_trichop$Phylum = factor(as.character(sigtab_cush_coch_trichop$Phylum), levels=names(x_cush_coch_trichop))
# Genus
x_cush_coch_trichop = tapply(sigtab_cush_coch_trichop$log2FoldChange, sigtab_cush_coch_trichop$Genus, function(x) max(x))
x_cush_coch_trichop = sort(x_cush_coch_trichop, TRUE)
sigtab_cush_coch_trichop$Genus = factor(as.character(sigtab_cush_coch_trichop$Genus), levels=sort(names(x_cush_coch_trichop)))
#3 observations

diff_abund_trichop_cush_coch_plot<-ggplot(sigtab_cush_coch_trichop, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_trichop_cush_coch_plot


#COCH - GRVBR

trichop_grvbr_coch<-subset_samples(trichop_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_trichop <- phyloseq_to_deseq2(trichop_grvbr_coch, ~ Site)

deseq_grvbr_coch_trichop <- DESeq(deseq_grvbr_coch_trichop, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_grvbr_coch_trichop <- results(deseq_grvbr_coch_trichop, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_trichop = res_deseq_grvbr_coch_trichop[which(res_deseq_grvbr_coch_trichop$padj < alpha), ]
sigtab_grvbr_coch_trichop = cbind(as(sigtab_grvbr_coch_trichop, "data.frame"), as(tax_table(trichop_grvbr_coch)[rownames(sigtab_grvbr_coch_trichop), ], "matrix"))

x_grvbr_coch_trichop = tapply(sigtab_grvbr_coch_trichop$log2FoldChange, sigtab_grvbr_coch_trichop$Phylum, function(x) max(x))
x_grvbr_coch_trichop = sort(x_grvbr_coch_trichop, TRUE)
sigtab_grvbr_coch_trichop$Phylum = factor(as.character(sigtab_grvbr_coch_trichop$Phylum), levels=names(x_grvbr_coch_trichop))
# Genus
x_grvbr_coch_trichop = tapply(sigtab_grvbr_coch_trichop$log2FoldChange, sigtab_grvbr_coch_trichop$Genus, function(x) max(x))
x_grvbr_coch_trichop = sort(x_grvbr_coch_trichop, TRUE)
sigtab_grvbr_coch_trichop$Genus = factor(as.character(sigtab_grvbr_coch_trichop$Genus), levels=sort(names(x_grvbr_coch_trichop)))
#20 observations

diff_abund_trichop_grvbr_coch_plot<-ggplot(sigtab_grvbr_coch_trichop, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_trichop_grvbr_coch_plot


#### Larval Heptageniidae ####

hep_ps<-subset_samples(sample_ps, Family=="Heptageniidae")

deseq_exposed_hep <- phyloseq_to_deseq2(hep_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_hep$Exposure <-relevel(deseq_exposed_hep$Exposure, "Unexposed")

deseq_exposed_hep <- DESeq(deseq_exposed_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_hep <- results(deseq_exposed_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_hep = res_deseq_exposed_hep[which(res_deseq_exposed_hep$padj < alpha), ]
sigtab_exposed_hep = cbind(as(sigtab_exposed_hep, "data.frame"), as(tax_table(hep_ps)[rownames(sigtab_exposed_hep), ], "matrix"))

x_exposed_hep = tapply(sigtab_exposed_hep$log2FoldChange, sigtab_exposed_hep$Phylum, function(x) max(x))
x_exposed_hep = sort(x_exposed_hep, TRUE)
sigtab_exposed_hep$Phylum = factor(as.character(sigtab_exposed_hep$Phylum), levels=names(x_exposed_hep))
# Genus
x_exposed_hep = tapply(sigtab_exposed_hep$log2FoldChange, sigtab_exposed_hep$Genus, function(x) max(x))
x_exposed_hep = sort(x_exposed_hep, TRUE)
sigtab_exposed_hep$Genus = factor(as.character(sigtab_exposed_hep$Genus), levels=sort(names(x_exposed_hep)))
#45 differentially abundant taxa

diff_abund_hep_plot<-ggplot(sigtab_exposed_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")
diff_abund_hep_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites larval heptageniidae only.png", height=15, width=22)


#COCH - SUN

hep_sun_coch<-subset_samples(hep_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_hep <- phyloseq_to_deseq2(hep_sun_coch, ~ Site)

deseq_sun_coch_hep <- DESeq(deseq_sun_coch_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_sun_coch_hep <- results(deseq_sun_coch_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_hep = res_deseq_sun_coch_hep[which(res_deseq_sun_coch_hep$padj < alpha), ]
sigtab_sun_coch_hep = cbind(as(sigtab_sun_coch_hep, "data.frame"), as(tax_table(hep_sun_coch)[rownames(sigtab_sun_coch_hep), ], "matrix"))

x_sun_coch_hep = tapply(sigtab_sun_coch_hep$log2FoldChange, sigtab_sun_coch_hep$Phylum, function(x) max(x))
x_sun_coch_hep = sort(x_sun_coch_hep, TRUE)
sigtab_sun_coch_hep$Phylum = factor(as.character(sigtab_sun_coch_hep$Phylum), levels=names(x_sun_coch_hep))
# Genus
x_sun_coch_hep = tapply(sigtab_sun_coch_hep$log2FoldChange, sigtab_sun_coch_hep$Genus, function(x) max(x))
x_sun_coch_hep = sort(x_sun_coch_hep, TRUE)
sigtab_sun_coch_hep$Genus = factor(as.character(sigtab_sun_coch_hep$Genus), levels=sort(names(x_sun_coch_hep)))
#2 observations

diff_abund_hep_sun_coch_plot<-ggplot(sigtab_sun_coch_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hep_sun_coch_plot


#COCH - CUSHBR

hep_cush_coch<-subset_samples(hep_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_hep <- phyloseq_to_deseq2(hep_cush_coch, ~ Site)

deseq_cush_coch_hep <- DESeq(deseq_cush_coch_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_hep <- results(deseq_cush_coch_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_hep = res_deseq_cush_coch_hep[which(res_deseq_cush_coch_hep$padj < alpha), ]
sigtab_cush_coch_hep = cbind(as(sigtab_cush_coch_hep, "data.frame"), as(tax_table(hep_cush_coch)[rownames(sigtab_cush_coch_hep), ], "matrix"))

x_cush_coch_hep = tapply(sigtab_cush_coch_hep$log2FoldChange, sigtab_cush_coch_hep$Phylum, function(x) max(x))
x_cush_coch_hep = sort(x_cush_coch_hep, TRUE)
sigtab_cush_coch_hep$Phylum = factor(as.character(sigtab_cush_coch_hep$Phylum), levels=names(x_cush_coch_hep))
# Genus
x_cush_coch_hep = tapply(sigtab_cush_coch_hep$log2FoldChange, sigtab_cush_coch_hep$Genus, function(x) max(x))
x_cush_coch_hep = sort(x_cush_coch_hep, TRUE)
sigtab_cush_coch_hep$Genus = factor(as.character(sigtab_cush_coch_hep$Genus), levels=sort(names(x_cush_coch_hep)))
#3 observations

diff_abund_hep_cush_coch_plot<-ggplot(sigtab_cush_coch_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hep_cush_coch_plot


#COCH - GRVBR

hep_grvbr_coch<-subset_samples(hep_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_hep <- phyloseq_to_deseq2(hep_grvbr_coch, ~ Site)

deseq_grvbr_coch_hep <- DESeq(deseq_grvbr_coch_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_grvbr_coch_hep <- results(deseq_grvbr_coch_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_hep = res_deseq_grvbr_coch_hep[which(res_deseq_grvbr_coch_hep$padj < alpha), ]
sigtab_grvbr_coch_hep = cbind(as(sigtab_grvbr_coch_hep, "data.frame"), as(tax_table(hep_grvbr_coch)[rownames(sigtab_grvbr_coch_hep), ], "matrix"))

x_grvbr_coch_hep = tapply(sigtab_grvbr_coch_hep$log2FoldChange, sigtab_grvbr_coch_hep$Phylum, function(x) max(x))
x_grvbr_coch_hep = sort(x_grvbr_coch_hep, TRUE)
sigtab_grvbr_coch_hep$Phylum = factor(as.character(sigtab_grvbr_coch_hep$Phylum), levels=names(x_grvbr_coch_hep))
# Genus
x_grvbr_coch_hep = tapply(sigtab_grvbr_coch_hep$log2FoldChange, sigtab_grvbr_coch_hep$Genus, function(x) max(x))
x_grvbr_coch_hep = sort(x_grvbr_coch_hep, TRUE)
sigtab_grvbr_coch_hep$Genus = factor(as.character(sigtab_grvbr_coch_hep$Genus), levels=sort(names(x_grvbr_coch_hep)))
#35 observations

diff_abund_hep_grvbr_coch_plot<-ggplot(sigtab_grvbr_coch_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hep_grvbr_coch_plot

#COCH - GRVBR

hep_grvbr_coch<-subset_samples(hep_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_hep <- phyloseq_to_deseq2(hep_grvbr_coch, ~ Site)

deseq_grvbr_coch_hep <- DESeq(deseq_grvbr_coch_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_grvbr_coch_hep <- results(deseq_grvbr_coch_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_hep = res_deseq_grvbr_coch_hep[which(res_deseq_grvbr_coch_hep$padj < alpha), ]
sigtab_grvbr_coch_hep = cbind(as(sigtab_grvbr_coch_hep, "data.frame"), as(tax_table(hep_grvbr_coch)[rownames(sigtab_grvbr_coch_hep), ], "matrix"))

x_grvbr_coch_hep = tapply(sigtab_grvbr_coch_hep$log2FoldChange, sigtab_grvbr_coch_hep$Phylum, function(x) max(x))
x_grvbr_coch_hep = sort(x_grvbr_coch_hep, TRUE)
sigtab_grvbr_coch_hep$Phylum = factor(as.character(sigtab_grvbr_coch_hep$Phylum), levels=names(x_grvbr_coch_hep))
# Genus
x_grvbr_coch_hep = tapply(sigtab_grvbr_coch_hep$log2FoldChange, sigtab_grvbr_coch_hep$Genus, function(x) max(x))
x_grvbr_coch_hep = sort(x_grvbr_coch_hep, TRUE)
sigtab_grvbr_coch_hep$Genus = factor(as.character(sigtab_grvbr_coch_hep$Genus), levels=sort(names(x_grvbr_coch_hep)))
#35 observations

diff_abund_hep_grvbr_coch_plot<-ggplot(sigtab_grvbr_coch_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hep_grvbr_coch_plot


#COCH - PMF

hep_pmf_coch<-subset_samples(hep_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_hep <- phyloseq_to_deseq2(hep_pmf_coch, ~ Site)

deseq_pmf_coch_hep <- DESeq(deseq_pmf_coch_hep, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_hep <- results(deseq_pmf_coch_hep, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_hep = res_deseq_pmf_coch_hep[which(res_deseq_pmf_coch_hep$padj < alpha), ]
sigtab_pmf_coch_hep = cbind(as(sigtab_pmf_coch_hep, "data.frame"), as(tax_table(hep_pmf_coch)[rownames(sigtab_pmf_coch_hep), ], "matrix"))

x_pmf_coch_hep = tapply(sigtab_pmf_coch_hep$log2FoldChange, sigtab_pmf_coch_hep$Phylum, function(x) max(x))
x_pmf_coch_hep = sort(x_pmf_coch_hep, TRUE)
sigtab_pmf_coch_hep$Phylum = factor(as.character(sigtab_pmf_coch_hep$Phylum), levels=names(x_pmf_coch_hep))
# Genus
x_pmf_coch_hep = tapply(sigtab_pmf_coch_hep$log2FoldChange, sigtab_pmf_coch_hep$Genus, function(x) max(x))
x_pmf_coch_hep = sort(x_pmf_coch_hep, TRUE)
sigtab_pmf_coch_hep$Genus = factor(as.character(sigtab_pmf_coch_hep$Genus), levels=sort(names(x_pmf_coch_hep)))
#71 observations

diff_abund_hep_pmf_coch_plot<-ggplot(sigtab_pmf_coch_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_hep_pmf_coch_plot

#### Adult Ephemeroptera ####

ephem_ps<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type=="Bow River")

deseq_exposed_ephem <- phyloseq_to_deseq2(ephem_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_ephem$Exposure <-relevel(deseq_exposed_ephem$Exposure, "Unexposed")

deseq_exposed_ephem <- DESeq(deseq_exposed_ephem, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_ephem <- results(deseq_exposed_ephem, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_ephem = res_deseq_exposed_ephem[which(res_deseq_exposed_ephem$padj < alpha), ]
sigtab_exposed_ephem = cbind(as(sigtab_exposed_ephem, "data.frame"), as(tax_table(ephem_ps)[rownames(sigtab_exposed_ephem), ], "matrix"))

x_exposed_ephem = tapply(sigtab_exposed_ephem$log2FoldChange, sigtab_exposed_ephem$Phylum, function(x) max(x))
x_exposed_ephem = sort(x_exposed_ephem, TRUE)
sigtab_exposed_ephem$Phylum = factor(as.character(sigtab_exposed_ephem$Phylum), levels=names(x_exposed_ephem))
# Genus
x_exposed_ephem = tapply(sigtab_exposed_ephem$log2FoldChange, sigtab_exposed_ephem$Genus, function(x) max(x))
x_exposed_ephem = sort(x_exposed_ephem, TRUE)
sigtab_exposed_ephem$Genus = factor(as.character(sigtab_exposed_ephem$Genus), levels=sort(names(x_exposed_ephem)))
#8 differentially abundant taxa

diff_abund_ephem_plot<-ggplot(sigtab_exposed_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")


#COCH - SUN

ephem_sun_coch<-subset_samples(ephem_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_ephem <- phyloseq_to_deseq2(ephem_sun_coch, ~ Site)

deseq_sun_coch_ephem <- DESeq(deseq_sun_coch_ephem, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_sun_coch_ephem <- results(deseq_sun_coch_ephem, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_ephem = res_deseq_sun_coch_ephem[which(res_deseq_sun_coch_ephem$padj < alpha), ]
sigtab_sun_coch_ephem = cbind(as(sigtab_sun_coch_ephem, "data.frame"), as(tax_table(ephem_sun_coch)[rownames(sigtab_sun_coch_ephem), ], "matrix"))

x_sun_coch_ephem = tapply(sigtab_sun_coch_ephem$log2FoldChange, sigtab_sun_coch_ephem$Phylum, function(x) max(x))
x_sun_coch_ephem = sort(x_sun_coch_ephem, TRUE)
sigtab_sun_coch_ephem$Phylum = factor(as.character(sigtab_sun_coch_ephem$Phylum), levels=names(x_sun_coch_ephem))
# Genus
x_sun_coch_ephem = tapply(sigtab_sun_coch_ephem$log2FoldChange, sigtab_sun_coch_ephem$Genus, function(x) max(x))
x_sun_coch_ephem = sort(x_sun_coch_ephem, TRUE)
sigtab_sun_coch_ephem$Genus = factor(as.character(sigtab_sun_coch_ephem$Genus), levels=sort(names(x_sun_coch_ephem)))
#12 observations

diff_abund_ephem_sun_coch_plot<-ggplot(sigtab_sun_coch_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_ephem_sun_coch_plot


#COCH - CUSHBR

ephem_cush_coch<-subset_samples(ephem_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_ephem <- phyloseq_to_deseq2(ephem_cush_coch, ~ Site)

deseq_cush_coch_ephem <- DESeq(deseq_cush_coch_ephem, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_ephem <- results(deseq_cush_coch_ephem, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_ephem = res_deseq_cush_coch_ephem[which(res_deseq_cush_coch_ephem$padj < alpha), ]
sigtab_cush_coch_ephem = cbind(as(sigtab_cush_coch_ephem, "data.frame"), as(tax_table(ephem_cush_coch)[rownames(sigtab_cush_coch_ephem), ], "matrix"))

x_cush_coch_ephem = tapply(sigtab_cush_coch_ephem$log2FoldChange, sigtab_cush_coch_ephem$Phylum, function(x) max(x))
x_cush_coch_ephem = sort(x_cush_coch_ephem, TRUE)
sigtab_cush_coch_ephem$Phylum = factor(as.character(sigtab_cush_coch_ephem$Phylum), levels=names(x_cush_coch_ephem))
# Genus
x_cush_coch_ephem = tapply(sigtab_cush_coch_ephem$log2FoldChange, sigtab_cush_coch_ephem$Genus, function(x) max(x))
x_cush_coch_ephem = sort(x_cush_coch_ephem, TRUE)
sigtab_cush_coch_ephem$Genus = factor(as.character(sigtab_cush_coch_ephem$Genus), levels=sort(names(x_cush_coch_ephem)))
#8 observations

diff_abund_ephem_cush_coch_plot<-ggplot(sigtab_cush_coch_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_ephem_cush_coch_plot


#COCH - GRVBR

ephem_grvbr_coch<-subset_samples(ephem_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_ephem <- phyloseq_to_deseq2(ephem_grvbr_coch, ~ Site)

deseq_grvbr_coch_ephem <- DESeq(deseq_grvbr_coch_ephem, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_grvbr_coch_ephem <- results(deseq_grvbr_coch_ephem, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_ephem = res_deseq_grvbr_coch_ephem[which(res_deseq_grvbr_coch_ephem$padj < alpha), ]
sigtab_grvbr_coch_ephem = cbind(as(sigtab_grvbr_coch_ephem, "data.frame"), as(tax_table(ephem_grvbr_coch)[rownames(sigtab_grvbr_coch_ephem), ], "matrix"))

x_grvbr_coch_ephem = tapply(sigtab_grvbr_coch_ephem$log2FoldChange, sigtab_grvbr_coch_ephem$Phylum, function(x) max(x))
x_grvbr_coch_ephem = sort(x_grvbr_coch_ephem, TRUE)
sigtab_grvbr_coch_ephem$Phylum = factor(as.character(sigtab_grvbr_coch_ephem$Phylum), levels=names(x_grvbr_coch_ephem))
# Genus
x_grvbr_coch_ephem = tapply(sigtab_grvbr_coch_ephem$log2FoldChange, sigtab_grvbr_coch_ephem$Genus, function(x) max(x))
x_grvbr_coch_ephem = sort(x_grvbr_coch_ephem, TRUE)
sigtab_grvbr_coch_ephem$Genus = factor(as.character(sigtab_grvbr_coch_ephem$Genus), levels=sort(names(x_grvbr_coch_ephem)))
#10 observations

diff_abund_ephem_grvbr_coch_plot<-ggplot(sigtab_grvbr_coch_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_ephem_grvbr_coch_plot

#COCH - PMF

ephem_pmf_coch<-subset_samples(ephem_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_ephem <- phyloseq_to_deseq2(ephem_pmf_coch, ~ Site)

deseq_pmf_coch_ephem <- DESeq(deseq_pmf_coch_ephem, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_ephem <- results(deseq_pmf_coch_ephem, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_ephem = res_deseq_pmf_coch_ephem[which(res_deseq_pmf_coch_ephem$padj < alpha), ]
sigtab_pmf_coch_ephem = cbind(as(sigtab_pmf_coch_ephem, "data.frame"), as(tax_table(ephem_pmf_coch)[rownames(sigtab_pmf_coch_ephem), ], "matrix"))

x_pmf_coch_ephem = tapply(sigtab_pmf_coch_ephem$log2FoldChange, sigtab_pmf_coch_ephem$Phylum, function(x) max(x))
x_pmf_coch_ephem = sort(x_pmf_coch_ephem, TRUE)
sigtab_pmf_coch_ephem$Phylum = factor(as.character(sigtab_pmf_coch_ephem$Phylum), levels=names(x_pmf_coch_ephem))
# Genus
x_pmf_coch_ephem = tapply(sigtab_pmf_coch_ephem$log2FoldChange, sigtab_pmf_coch_ephem$Genus, function(x) max(x))
x_pmf_coch_ephem = sort(x_pmf_coch_ephem, TRUE)
sigtab_pmf_coch_ephem$Genus = factor(as.character(sigtab_pmf_coch_ephem$Genus), levels=sort(names(x_pmf_coch_ephem)))
#1 observation

diff_abund_ephem_pmf_coch_plot<-ggplot(sigtab_pmf_coch_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_ephem_pmf_coch_plot

#### Larval Chironomidae ####

chiro_ps<-subset_samples(sample_ps, Family=="Chironomidae")

deseq_exposed_chiro <- phyloseq_to_deseq2(chiro_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_chiro$Exposure <-relevel(deseq_exposed_chiro$Exposure, "Unexposed")

deseq_exposed_chiro <- DESeq(deseq_exposed_chiro, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_chiro <- results(deseq_exposed_chiro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_chiro = res_deseq_exposed_chiro[which(res_deseq_exposed_chiro$padj < alpha), ]
sigtab_exposed_chiro = cbind(as(sigtab_exposed_chiro, "data.frame"), as(tax_table(chiro_ps)[rownames(sigtab_exposed_chiro), ], "matrix"))

x_exposed_chiro = tapply(sigtab_exposed_chiro$log2FoldChange, sigtab_exposed_chiro$Phylum, function(x) max(x))
x_exposed_chiro = sort(x_exposed_chiro, TRUE)
sigtab_exposed_chiro$Phylum = factor(as.character(sigtab_exposed_chiro$Phylum), levels=names(x_exposed_chiro))
# Genus
x_exposed_chiro = tapply(sigtab_exposed_chiro$log2FoldChange, sigtab_exposed_chiro$Genus, function(x) max(x))
x_exposed_chiro = sort(x_exposed_chiro, TRUE)
sigtab_exposed_chiro$Genus = factor(as.character(sigtab_exposed_chiro$Genus), levels=sort(names(x_exposed_chiro)))
#21 differentially abundant taxa

diff_abund_chiro_plot<-ggplot(sigtab_exposed_chiro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites chironomid larvae only.png", height=15, width=22)

#COCH - CUSHBR

chiro_cush_coch<-subset_samples(chiro_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_chiro <- phyloseq_to_deseq2(chiro_cush_coch, ~ Site)

deseq_cush_coch_chiro <- DESeq(deseq_cush_coch_chiro, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_chiro <- results(deseq_cush_coch_chiro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_chiro = res_deseq_cush_coch_chiro[which(res_deseq_cush_coch_chiro$padj < alpha), ]
sigtab_cush_coch_chiro = cbind(as(sigtab_cush_coch_chiro, "data.frame"), as(tax_table(chiro_cush_coch)[rownames(sigtab_cush_coch_chiro), ], "matrix"))

x_cush_coch_chiro = tapply(sigtab_cush_coch_chiro$log2FoldChange, sigtab_cush_coch_chiro$Phylum, function(x) max(x))
x_cush_coch_chiro = sort(x_cush_coch_chiro, TRUE)
sigtab_cush_coch_chiro$Phylum = factor(as.character(sigtab_cush_coch_chiro$Phylum), levels=names(x_cush_coch_chiro))
# Genus
x_cush_coch_chiro = tapply(sigtab_cush_coch_chiro$log2FoldChange, sigtab_cush_coch_chiro$Genus, function(x) max(x))
x_cush_coch_chiro = sort(x_cush_coch_chiro, TRUE)
sigtab_cush_coch_chiro$Genus = factor(as.character(sigtab_cush_coch_chiro$Genus), levels=sort(names(x_cush_coch_chiro)))
#20 observations

diff_abund_chiro_cush_coch_plot<-ggplot(sigtab_cush_coch_chiro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_chiro_cush_coch_plot


#COCH - PMF

chiro_pmf_coch<-subset_samples(chiro_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_chiro <- phyloseq_to_deseq2(chiro_pmf_coch, ~ Site)

deseq_pmf_coch_chiro <- DESeq(deseq_pmf_coch_chiro, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_chiro <- results(deseq_pmf_coch_chiro, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_chiro = res_deseq_pmf_coch_chiro[which(res_deseq_pmf_coch_chiro$padj < alpha), ]
sigtab_pmf_coch_chiro = cbind(as(sigtab_pmf_coch_chiro, "data.frame"), as(tax_table(chiro_pmf_coch)[rownames(sigtab_pmf_coch_chiro), ], "matrix"))

x_pmf_coch_chiro = tapply(sigtab_pmf_coch_chiro$log2FoldChange, sigtab_pmf_coch_chiro$Phylum, function(x) max(x))
x_pmf_coch_chiro = sort(x_pmf_coch_chiro, TRUE)
sigtab_pmf_coch_chiro$Phylum = factor(as.character(sigtab_pmf_coch_chiro$Phylum), levels=names(x_pmf_coch_chiro))
# Genus
x_pmf_coch_chiro = tapply(sigtab_pmf_coch_chiro$log2FoldChange, sigtab_pmf_coch_chiro$Genus, function(x) max(x))
x_pmf_coch_chiro = sort(x_pmf_coch_chiro, TRUE)
sigtab_pmf_coch_chiro$Genus = factor(as.character(sigtab_pmf_coch_chiro$Genus), levels=sort(names(x_pmf_coch_chiro)))
#31 observations

diff_abund_chiro_pmf_coch_plot<-ggplot(sigtab_pmf_coch_chiro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_chiro_pmf_coch_plot


#### Adult Diptera ####

dip_ps<-subset_samples(sample_ps, Family=="Diptera")

deseq_exposed_dip <- phyloseq_to_deseq2(dip_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_dip$Exposure <-relevel(deseq_exposed_dip$Exposure, "Unexposed")

deseq_exposed_dip <- DESeq(deseq_exposed_dip, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_dip <- results(deseq_exposed_dip, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_dip = res_deseq_exposed_dip[which(res_deseq_exposed_dip$padj < alpha), ]
sigtab_exposed_dip = cbind(as(sigtab_exposed_dip, "data.frame"), as(tax_table(dip_ps)[rownames(sigtab_exposed_dip), ], "matrix"))

x_exposed_dip = tapply(sigtab_exposed_dip$log2FoldChange, sigtab_exposed_dip$Phylum, function(x) max(x))
x_exposed_dip = sort(x_exposed_dip, TRUE)
sigtab_exposed_dip$Phylum = factor(as.character(sigtab_exposed_dip$Phylum), levels=names(x_exposed_dip))
# Genus
x_exposed_dip = tapply(sigtab_exposed_dip$log2FoldChange, sigtab_exposed_dip$Genus, function(x) max(x))
x_exposed_dip = sort(x_exposed_dip, TRUE)
sigtab_exposed_dip$Genus = factor(as.character(sigtab_exposed_dip$Genus), levels=sort(names(x_exposed_dip)))
#8 differentially abundant taxa

diff_abund_dip_plot<-ggplot(sigtab_exposed_dip, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")
diff_abund_dip_plot

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites chironomidae adults only.png", height=15, width=22)

#COCH - CUSHBR

dip_cush_coch<-subset_samples(dip_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_dip <- phyloseq_to_deseq2(dip_cush_coch, ~ Site)

deseq_cush_coch_dip <- DESeq(deseq_cush_coch_dip, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_dip <- results(deseq_cush_coch_dip, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_dip = res_deseq_cush_coch_dip[which(res_deseq_cush_coch_dip$padj < alpha), ]
sigtab_cush_coch_dip = cbind(as(sigtab_cush_coch_dip, "data.frame"), as(tax_table(dip_cush_coch)[rownames(sigtab_cush_coch_dip), ], "matrix"))

x_cush_coch_dip = tapply(sigtab_cush_coch_dip$log2FoldChange, sigtab_cush_coch_dip$Phylum, function(x) max(x))
x_cush_coch_dip = sort(x_cush_coch_dip, TRUE)
sigtab_cush_coch_dip$Phylum = factor(as.character(sigtab_cush_coch_dip$Phylum), levels=names(x_cush_coch_dip))
# Genus
x_cush_coch_dip = tapply(sigtab_cush_coch_dip$log2FoldChange, sigtab_cush_coch_dip$Genus, function(x) max(x))
x_cush_coch_dip = sort(x_cush_coch_dip, TRUE)
sigtab_cush_coch_dip$Genus = factor(as.character(sigtab_cush_coch_dip$Genus), levels=sort(names(x_cush_coch_dip)))
#71 observations

diff_abund_dip_cush_coch_plot<-ggplot(sigtab_cush_coch_dip, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_dip_cush_coch_plot


#COCH - PMF

dip_pmf_coch<-subset_samples(dip_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_dip <- phyloseq_to_deseq2(dip_pmf_coch, ~ Site)

deseq_pmf_coch_dip <- DESeq(deseq_pmf_coch_dip, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_dip <- results(deseq_pmf_coch_dip, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_dip = res_deseq_pmf_coch_dip[which(res_deseq_pmf_coch_dip$padj < alpha), ]
sigtab_pmf_coch_dip = cbind(as(sigtab_pmf_coch_dip, "data.frame"), as(tax_table(dip_pmf_coch)[rownames(sigtab_pmf_coch_dip), ], "matrix"))

x_pmf_coch_dip = tapply(sigtab_pmf_coch_dip$log2FoldChange, sigtab_pmf_coch_dip$Phylum, function(x) max(x))
x_pmf_coch_dip = sort(x_pmf_coch_dip, TRUE)
sigtab_pmf_coch_dip$Phylum = factor(as.character(sigtab_pmf_coch_dip$Phylum), levels=names(x_pmf_coch_dip))
# Genus
x_pmf_coch_dip = tapply(sigtab_pmf_coch_dip$log2FoldChange, sigtab_pmf_coch_dip$Genus, function(x) max(x))
x_pmf_coch_dip = sort(x_pmf_coch_dip, TRUE)
sigtab_pmf_coch_dip$Genus = factor(as.character(sigtab_pmf_coch_dip$Genus), levels=sort(names(x_pmf_coch_dip)))
#75 observations

diff_abund_dip_pmf_coch_plot<-ggplot(sigtab_pmf_coch_dip, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_dip_pmf_coch_plot


#### Larval Perlidae ####

perl_ps<-subset_samples(sample_ps, Family=="Perlidae")

deseq_exposed_perl <- phyloseq_to_deseq2(perl_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_perl$Exposure <-relevel(deseq_exposed_perl$Exposure, "Unexposed")

deseq_exposed_perl <- DESeq(deseq_exposed_perl, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_perl <- results(deseq_exposed_perl, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_perl = res_deseq_exposed_perl[which(res_deseq_exposed_perl$padj < alpha), ]
sigtab_exposed_perl = cbind(as(sigtab_exposed_perl, "data.frame"), as(tax_table(perl_ps)[rownames(sigtab_exposed_perl), ], "matrix"))

x_exposed_perl = tapply(sigtab_exposed_perl$log2FoldChange, sigtab_exposed_perl$Phylum, function(x) max(x))
x_exposed_perl = sort(x_exposed_perl, TRUE)
sigtab_exposed_perl$Phylum = factor(as.character(sigtab_exposed_perl$Phylum), levels=names(x_exposed_perl))
# Genus
x_exposed_perl = tapply(sigtab_exposed_perl$log2FoldChange, sigtab_exposed_perl$Genus, function(x) max(x))
x_exposed_perl = sort(x_exposed_perl, TRUE)
sigtab_exposed_perl$Genus = factor(as.character(sigtab_exposed_perl$Genus), levels=sort(names(x_exposed_perl)))
#17 differentially abundant taxa

diff_abund_perl_plot<-ggplot(sigtab_exposed_perl, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/Bow River exposed vs unexposed sites larval perlidae only.png", height=15, width=22)


#COCH - CUSHBR

perl_cush_coch<-subset_samples(perl_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_perl <- phyloseq_to_deseq2(perl_cush_coch, ~ Site)

deseq_cush_coch_perl <- DESeq(deseq_cush_coch_perl, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_perl <- results(deseq_cush_coch_perl, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_perl = res_deseq_cush_coch_perl[which(res_deseq_cush_coch_perl$padj < alpha), ]
sigtab_cush_coch_perl = cbind(as(sigtab_cush_coch_perl, "data.frame"), as(tax_table(perl_cush_coch)[rownames(sigtab_cush_coch_perl), ], "matrix"))

x_cush_coch_perl = tapply(sigtab_cush_coch_perl$log2FoldChange, sigtab_cush_coch_perl$Phylum, function(x) max(x))
x_cush_coch_perl = sort(x_cush_coch_perl, TRUE)
sigtab_cush_coch_perl$Phylum = factor(as.character(sigtab_cush_coch_perl$Phylum), levels=names(x_cush_coch_perl))
# Genus
x_cush_coch_perl = tapply(sigtab_cush_coch_perl$log2FoldChange, sigtab_cush_coch_perl$Genus, function(x) max(x))
x_cush_coch_perl = sort(x_cush_coch_perl, TRUE)
sigtab_cush_coch_perl$Genus = factor(as.character(sigtab_cush_coch_perl$Genus), levels=sort(names(x_cush_coch_perl)))
#31 observations

diff_abund_perl_cush_coch_plot<-ggplot(sigtab_cush_coch_perl, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_perl_cush_coch_plot

#COCH - PMF

perl_pmf_coch<-subset_samples(perl_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_perl <- phyloseq_to_deseq2(perl_pmf_coch, ~ Site)

deseq_pmf_coch_perl <- DESeq(deseq_pmf_coch_perl, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_perl <- results(deseq_pmf_coch_perl, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_perl = res_deseq_pmf_coch_perl[which(res_deseq_pmf_coch_perl$padj < alpha), ]
sigtab_pmf_coch_perl = cbind(as(sigtab_pmf_coch_perl, "data.frame"), as(tax_table(perl_pmf_coch)[rownames(sigtab_pmf_coch_perl), ], "matrix"))

x_pmf_coch_perl = tapply(sigtab_pmf_coch_perl$log2FoldChange, sigtab_pmf_coch_perl$Phylum, function(x) max(x))
x_pmf_coch_perl = sort(x_pmf_coch_perl, TRUE)
sigtab_pmf_coch_perl$Phylum = factor(as.character(sigtab_pmf_coch_perl$Phylum), levels=names(x_pmf_coch_perl))
# Genus
x_pmf_coch_perl = tapply(sigtab_pmf_coch_perl$log2FoldChange, sigtab_pmf_coch_perl$Genus, function(x) max(x))
x_pmf_coch_perl = sort(x_pmf_coch_perl, TRUE)
sigtab_pmf_coch_perl$Genus = factor(as.character(sigtab_pmf_coch_perl$Genus), levels=sort(names(x_pmf_coch_perl)))
#12 observations

diff_abund_perl_pmf_coch_plot<-ggplot(sigtab_pmf_coch_perl, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette = "Set3")

diff_abund_perl_pmf_coch_plot

#### Araneidae ####

aran_ps<-subset_samples(sample_ps, Family=="Araneidae")

deseq_exposed_aran <- phyloseq_to_deseq2(aran_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_aran$Exposure <-relevel(deseq_exposed_aran$Exposure, "Unexposed")

deseq_exposed_aran <- DESeq(deseq_exposed_aran, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_aran <- results(deseq_exposed_aran, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_aran = res_deseq_exposed_aran[which(res_deseq_exposed_aran$padj < alpha), ]
sigtab_exposed_aran = cbind(as(sigtab_exposed_aran, "data.frame"), as(tax_table(aran_ps)[rownames(sigtab_exposed_aran), ], "matrix"))
#no differentially abundant taxa!

#COCH - SUN

aran_sun_coch<-subset_samples(aran_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_aran <- phyloseq_to_deseq2(aran_sun_coch, ~ Site)

deseq_sun_coch_aran <- DESeq(deseq_sun_coch_aran, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_sun_coch_aran <- results(deseq_sun_coch_aran, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_aran = res_deseq_sun_coch_aran[which(res_deseq_sun_coch_aran$padj < alpha), ]
sigtab_sun_coch_aran = cbind(as(sigtab_sun_coch_aran, "data.frame"), as(tax_table(aran_sun_coch)[rownames(sigtab_sun_coch_aran), ], "matrix"))
#No diff abund taxa

#COCH - CUSHBR

aran_cush_coch<-subset_samples(aran_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_aran <- phyloseq_to_deseq2(aran_cush_coch, ~ Site)

deseq_cush_coch_aran <- DESeq(deseq_cush_coch_aran, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_aran <- results(deseq_cush_coch_aran, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_aran = res_deseq_cush_coch_aran[which(res_deseq_cush_coch_aran$padj < alpha), ]
sigtab_cush_coch_aran = cbind(as(sigtab_cush_coch_aran, "data.frame"), as(tax_table(aran_cush_coch)[rownames(sigtab_cush_coch_aran), ], "matrix"))
#1 obs

#COCH - GRVBR

aran_grvbr_coch<-subset_samples(aran_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_aran <- phyloseq_to_deseq2(aran_grvbr_coch, ~ Site)

deseq_grvbr_coch_aran <- DESeq(deseq_grvbr_coch_aran, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_grvbr_coch_aran <- results(deseq_grvbr_coch_aran, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_aran = res_deseq_grvbr_coch_aran[which(res_deseq_grvbr_coch_aran$padj < alpha), ]
sigtab_grvbr_coch_aran = cbind(as(sigtab_grvbr_coch_aran, "data.frame"), as(tax_table(aran_grvbr_coch)[rownames(sigtab_grvbr_coch_aran), ], "matrix"))
#none

#COCH - PMF

aran_pmf_coch<-subset_samples(aran_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_aran <- phyloseq_to_deseq2(aran_pmf_coch, ~ Site)

deseq_pmf_coch_aran <- DESeq(deseq_pmf_coch_aran, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_aran <- results(deseq_pmf_coch_aran, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_aran = res_deseq_pmf_coch_aran[which(res_deseq_pmf_coch_aran$padj < alpha), ]
sigtab_pmf_coch_aran = cbind(as(sigtab_pmf_coch_aran, "data.frame"), as(tax_table(aran_pmf_coch)[rownames(sigtab_pmf_coch_aran), ], "matrix"))
#6 obs



#### Tetragnathidae ####

tetra_ps<-subset_samples(sample_ps, Family=="Tetragnathidae")

deseq_exposed_tetra <- phyloseq_to_deseq2(tetra_ps, ~ Exposure)
#Re level the reference variable so that the comparison is exposed vs unexposed
deseq_exposed_tetra$Exposure <-relevel(deseq_exposed_tetra$Exposure, "Unexposed")

deseq_exposed_tetra <- DESeq(deseq_exposed_tetra, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed_tetra <- results(deseq_exposed_tetra, cooksCutoff = FALSE)
alpha = 0.001
sigtab_exposed_tetra = res_deseq_exposed_tetra[which(res_deseq_exposed_tetra$padj < alpha), ]
sigtab_exposed_tetra = cbind(as(sigtab_exposed_tetra, "data.frame"), as(tax_table(tetra_ps)[rownames(sigtab_exposed_tetra), ], "matrix"))

x_exposed_tetra = tapply(sigtab_exposed_tetra$log2FoldChange, sigtab_exposed_tetra$Phylum, function(x) max(x))
x_exposed_tetra = sort(x_exposed_tetra, TRUE)
sigtab_exposed_tetra$Phylum = factor(as.character(sigtab_exposed_tetra$Phylum), levels=names(x_exposed_tetra))
# Genus
x_exposed_tetra = tapply(sigtab_exposed_tetra$log2FoldChange, sigtab_exposed_tetra$Genus, function(x) max(x))
x_exposed_tetra = sort(x_exposed_tetra, TRUE)
sigtab_exposed_tetra$Genus = factor(as.character(sigtab_exposed_tetra$Genus), levels=sort(names(x_exposed_tetra)))

diff_abund_tetra_plot<-ggplot(sigtab_exposed_tetra, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")

diff_abund_tetra_plot


#COCH - SUN

tetra_sun_coch<-subset_samples(tetra_ps, Site=="Cochrane" | Site=="Sunalta")

deseq_sun_coch_tetra <- phyloseq_to_deseq2(tetra_sun_coch, ~ Site)

deseq_sun_coch_tetra <- DESeq(deseq_sun_coch_tetra, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_sun_coch_tetra <- results(deseq_sun_coch_tetra, cooksCutoff = FALSE)
alpha = 0.001
sigtab_sun_coch_tetra = res_deseq_sun_coch_tetra[which(res_deseq_sun_coch_tetra$padj < alpha), ]
sigtab_sun_coch_tetra = cbind(as(sigtab_sun_coch_tetra, "data.frame"), as(tax_table(tetra_sun_coch)[rownames(sigtab_sun_coch_tetra), ], "matrix"))
#1

x_sun_coch_tetra = tapply(sigtab_sun_coch_tetra$log2FoldChange, sigtab_sun_coch_tetra$Phylum, function(x) max(x))
x_sun_coch_tetra = sort(x_sun_coch_tetra, TRUE)
sigtab_sun_coch_tetra$Phylum = factor(as.character(sigtab_sun_coch_tetra$Phylum), levels=names(x_sun_coch_tetra))
# Genus
x_sun_coch_tetra = tapply(sigtab_sun_coch_tetra$log2FoldChange, sigtab_sun_coch_tetra$Genus, function(x) max(x))
x_sun_coch_tetra = sort(x_sun_coch_tetra, TRUE)
sigtab_sun_coch_tetra$Genus = factor(as.character(sigtab_sun_coch_tetra$Genus), levels=sort(names(x_sun_coch_tetra)))

ggplot(sigtab_sun_coch_tetra, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")

#COCH - CUSHBR

tetra_cush_coch<-subset_samples(tetra_ps, Site=="Cochrane" | Site=="Cushing Bridge")

deseq_cush_coch_tetra <- phyloseq_to_deseq2(tetra_cush_coch, ~ Site)

deseq_cush_coch_tetra <- DESeq(deseq_cush_coch_tetra, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_cush_coch_tetra <- results(deseq_cush_coch_tetra, cooksCutoff = FALSE)
alpha = 0.001
sigtab_cush_coch_tetra = res_deseq_cush_coch_tetra[which(res_deseq_cush_coch_tetra$padj < alpha), ]
sigtab_cush_coch_tetra = cbind(as(sigtab_cush_coch_tetra, "data.frame"), as(tax_table(tetra_cush_coch)[rownames(sigtab_cush_coch_tetra), ], "matrix"))
#none

#COCH - GRVBR

tetra_grvbr_coch<-subset_samples(tetra_ps, Site=="Cochrane" | Site=="Graves Bridge")

deseq_grvbr_coch_tetra <- phyloseq_to_deseq2(tetra_grvbr_coch, ~ Site)

deseq_grvbr_coch_tetra <- DESeq(deseq_grvbr_coch_tetra, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_grvbr_coch_tetra <- results(deseq_grvbr_coch_tetra, cooksCutoff = FALSE)
alpha = 0.001
sigtab_grvbr_coch_tetra = res_deseq_grvbr_coch_tetra[which(res_deseq_grvbr_coch_tetra$padj < alpha), ]
sigtab_grvbr_coch_tetra = cbind(as(sigtab_grvbr_coch_tetra, "data.frame"), as(tax_table(tetra_grvbr_coch)[rownames(sigtab_grvbr_coch_tetra), ], "matrix"))
#none

#COCH - PMF

tetra_pmf_coch<-subset_samples(tetra_ps, Site=="Cochrane" | Site=="Policeman Flats")

deseq_pmf_coch_tetra <- phyloseq_to_deseq2(tetra_pmf_coch, ~ Site)

deseq_pmf_coch_tetra <- DESeq(deseq_pmf_coch_tetra, test = "Wald", fitType = "parametric", sfType = "poscounts")


res_deseq_pmf_coch_tetra <- results(deseq_pmf_coch_tetra, cooksCutoff = FALSE)
alpha = 0.001
sigtab_pmf_coch_tetra = res_deseq_pmf_coch_tetra[which(res_deseq_pmf_coch_tetra$padj < alpha), ]
sigtab_pmf_coch_tetra = cbind(as(sigtab_pmf_coch_tetra, "data.frame"), as(tax_table(tetra_pmf_coch)[rownames(sigtab_pmf_coch_tetra), ], "matrix"))
#none

x_pmf_coch_tetra = tapply(sigtab_pmf_coch_tetra$log2FoldChange, sigtab_pmf_coch_tetra$Phylum, function(x) max(x))
x_pmf_coch_tetra = sort(x_pmf_coch_tetra, TRUE)
sigtab_pmf_coch_tetra$Phylum = factor(as.character(sigtab_pmf_coch_tetra$Phylum), levels=names(x_pmf_coch_tetra))
# Genus
x_pmf_coch_tetra = tapply(sigtab_pmf_coch_tetra$log2FoldChange, sigtab_pmf_coch_tetra$Genus, function(x) max(x))
x_pmf_coch_tetra = sort(x_pmf_coch_tetra, TRUE)
sigtab_pmf_coch_tetra$Genus = factor(as.character(sigtab_pmf_coch_tetra$Genus), levels=sort(names(x_pmf_coch_tetra)))

ggplot(sigtab_pmf_coch_tetra, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_brewer(palette="Set3")



#### Arranging invert plots all together ####

#Hydropsychidae

sigtab_exposed_hydro$Phylum<-factor(sigtab_exposed_hydro$Phylum, levels=c("Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Myxococcota", "Proteobacteria", "Rs-K70 termite group", "Synergistota"))

diff_abund_hydro_plot<-ggplot(sigtab_exposed_hydro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#f22020", "#991919", "#e68f66", "#f47a22", "#fcff5d", "#ffc413", "#228c68", "#235b54", "#3998f5", "#2f2aa0"))
diff_abund_hydro_plot

#Acidobacteriota, Actinobacteriota, Bacteroidota, Cyanobacteria, Firmicutes, Myxococcota, Proteobacteria, Rs-K70 termite group, Synergistota

#Heptageniidae

sigtab_exposed_hep$Phylum<-factor(sigtab_exposed_hep$Phylum, levels=c("Actinobacteriota", "Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

diff_abund_hep_plot<-ggplot(sigtab_exposed_hep, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=16, face="italic")) + theme(axis.text=element_text(size=16, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#991919", "#e68f66", "#f47a22", "#fcff5d", "#228c68"))
diff_abund_hep_plot

#Actinobacteriota, Bacteroidota, Cyanobacteria, Firmicutes, Proteobacteria

#Chironomidae

sigtab_exposed_chiro$Phylum<-factor(sigtab_exposed_chiro$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

diff_abund_chiro_plot<-ggplot(sigtab_exposed_chiro, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#e68f66", "#f47a22", "#fcff5d", "#228c68"))
diff_abund_chiro_plot

#Bacteroidota, Cyanobacteria, Firmicutes, Proteobacteria

#Perlidae

sigtab_exposed_perl$Phylum<-factor(sigtab_exposed_perl$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Proteobacteria", "Verrucomicrobiota"))

diff_abund_perl_plot<-ggplot(sigtab_exposed_perl, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#e68f66", "#f47a22", "#228c68", "#2f2aa0"))
diff_abund_perl_plot

#Bacteroidota, Cyanobacteria, Proteobacteria, Verrucomicrobiota

#Trichoptera

sigtab_exposed_trichop$Phylum<-factor(sigtab_exposed_trichop$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

diff_abund_trichop_plot<-ggplot(sigtab_exposed_trichop, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=18, face="italic")) + theme(axis.text=element_text(size=18, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#e68f66", "#f47a22", "#fcff5d", "#228c68"))
diff_abund_trichop_plot

#Bacteroidota, Cyanobacteria, Firmicutes, Proteobacteria

#Ephemeroptera

sigtab_exposed_ephem$Phylum<-factor(sigtab_exposed_ephem$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Proteobacteria"))

diff_abund_ephem_plot<-ggplot(sigtab_exposed_ephem, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#e68f66", "#f47a22", "#fcff5d", "#228c68"))
diff_abund_ephem_plot

#Bacteroidota, Cyanobacteria, Firmicutes, Proteobacteria

#Diptera

sigtab_exposed_dip$Phylum<-factor(sigtab_exposed_dip$Phylum, levels=c("Cyanobacteria", "Firmicutes", "Proteobacteria"))

diff_abund_dip_plot<-ggplot(sigtab_exposed_dip, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values=c("#f47a22", "#fcff5d", "#228c68"))
diff_abund_dip_plot

#Cyanobacteria, Firmicutes, Proteobacteria

#Tetragnathidae

diff_abund_tetra_plot<-ggplot(sigtab_exposed_tetra, aes(x=Genus, y=log2FoldChange, color=Phylum))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=24, face="italic")) + theme(axis.text=element_text(size=24, color="black"), axis.title=element_text(size=28)) +
          theme(legend.text=element_text(size=24, color="black")) + 
          theme(legend.title = element_text(size=24, color="black"), legend.position="bottom")+
  guides(nrow=8)+
  rremove("grid")+
 labs(y = expression(Log[2]~ "Ratio of Absolute Abundance"))+
  scale_color_manual(values="#228c68")
diff_abund_tetra_plot


diff_abund_hydro_plot_1<- diff_abund_hydro_plot + rremove("axis.title") + rremove("legend")
diff_abund_hydro_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_hydro_exposed.png", height=12, width=30, bg="white")
diff_abund_hep_plot_1<- diff_abund_hep_plot + rremove("axis.title")+ rremove("legend")
diff_abund_hep_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_hep_exposed.png", height=16, width=18, bg="white")
diff_abund_chiro_plot_1<- diff_abund_chiro_plot + rremove("axis.title")+ rremove("legend")
diff_abund_chiro_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_chiro_exposed.png", height=12, width=10, bg="white")
diff_abund_perl_plot_1<- diff_abund_perl_plot + rremove("axis.title")+ rremove("legend")
diff_abund_perl_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_perl_exposed.png", height=10, width=8, bg="white")
diff_abund_trichop_plot_1<- diff_abund_trichop_plot + rremove("axis.title")+ rremove("legend")
diff_abund_trichop_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_trichop_exposed.png", height=14, width=26, bg="white")
diff_abund_ephem_plot_1<- diff_abund_ephem_plot + rremove("axis.title")+ rremove("legend")
diff_abund_ephem_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_ephem_exposed.png", height=6, width=8, bg="white")
diff_abund_dip_plot_1<- diff_abund_dip_plot + rremove("axis.title")+ rremove("legend")
diff_abund_dip_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_dip_exposed.png", height=10, width=16, bg="white")
diff_abund_tetra_plot_1<- diff_abund_tetra_plot + rremove("axis.title")+ rremove("legend")
diff_abund_tetra_plot_1
#ggsave("microbiome_analysis/Final analysis/Differential Abundance Results/diff_abund_tetra_exposed.png", height=6, width=8, bg="white")

ggarrange(diff_abund_hydro_plot_1, diff_abund_hep_plot_1, diff_abund_chiro_plot_1, diff_abund_perl_plot_1, diff_abund_trichop_plot_1, diff_abund_ephem_plot_1, diff_abund_dip_plot_1, diff_abund_tetra_plot_1, nrow=3, ncol=3, common.legend = T, legend="bottom")

#ggsave(filename="microbiome_analysis/Final analysis/Differential Abundance Results/diff abund across taxa Bow River faceted.png", height=20, width=26, bg="white")

```

## Differential abundance in ACWA streams, separated by invert taxa {.tabset}

### REF vs EXP5 
```{r}
#### DESeq2 ####
diff_abun_BRR2_PCR1 = subset_samples(sample_ps, Stream_Type=="ACWA Streams")
diff_abun_BRR2_PCR1 = subset_samples(diff_abun_BRR2_PCR1, Site != "PCR3")

deseq_BRR2_PCR1 <- phyloseq_to_deseq2(diff_abun_BRR2_PCR1, ~ Site)
deseq_BRR2_PCR1 <- DESeq(deseq_BRR2_PCR1, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_BRR2_PCR1 <- results(deseq_BRR2_PCR1, cooksCutoff = FALSE)
res_deseq_BRR2_PCR1
alpha = 0.001
sigtab_BRR2_PCR1 = res_deseq_BRR2_PCR1[which(res_deseq_BRR2_PCR1$padj < alpha), ]
sigtab_BRR2_PCR1 = cbind(as(sigtab_BRR2_PCR1, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_BRR2_PCR1), ], "matrix"))

#write.csv(as.data.frame(sigtab_BRR2_PCR1), filename="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_BRR2_PCR1.csv")

#Plotting the log2fold change in bacteria: PCR1 vs BRR2

x_BRR2_PCR1 = tapply(sigtab_BRR2_PCR1$log2FoldChange, sigtab_BRR2_PCR1$Phylum, function(x) max(x))
x_BRR2_PCR1 = sort(x_BRR2_PCR1, TRUE)
sigtab_BRR2_PCR1$Phylum = factor(as.character(sigtab_BRR2_PCR1$Phylum), levels=names(x_BRR2_PCR1))
# Genus order
x_BRR2_PCR1 = tapply(sigtab_BRR2_PCR1$log2FoldChange, sigtab_BRR2_PCR1$Genus, function(x) max(x))
x_BRR2_PCR1 = sort(x_BRR2_PCR1, TRUE)
sigtab_BRR2_PCR1$Genus = factor(as.character(sigtab_BRR2_PCR1$Genus), levels=sort(names(x_BRR2_PCR1)))

sigtab_BRR2_PCR1$Phylum<-factor(sigtab_BRR2_PCR1$Phylum, levels=c("Proteobacteria", "Bacteroidota", "Firmicutes", "Cyanobacteria", "Myxococcota"))

sigtab_BRR2_PCR1$Genus<-factor(sigtab_BRR2_PCR1$Genus, labels=c("Alistipes", "Rhizobium", "Flavobacterium", "Ideonella", "Limnohabitans", "Nevskia", "Novosphingobium", "Pajaroellobacter", "Rhodoferax", "Romboutsia", "Sediminibacterium", "Vibrionimonas"))

PCR1_BRR2<-ggplot(sigtab_BRR2_PCR1, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of PCR1/BRR2")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")

PCR1_BRR2

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_PCR1_compared_to_BRR2.png", height=6.5, width=7.5)


## Hydropsychidae ##
diff_abun_hydro_REF_EXP5 = subset_samples(diff_abun_BRR2_PCR1, Family == "Hydropsychidae")

deseq_hydro_REF_EXP5 <- phyloseq_to_deseq2(diff_abun_hydro_REF_EXP5, ~ Site)
deseq_hydro_REF_EXP5 <- DESeq(deseq_hydro_REF_EXP5, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_hydro_REF_EXP5 <- results(deseq_hydro_REF_EXP5, cooksCutoff = FALSE)
res_deseq_hydro_REF_EXP5
alpha = 0.001
sigtab_hydro_REF_EXP5 = res_deseq_hydro_REF_EXP5[which(res_deseq_hydro_REF_EXP5$padj < alpha), ]
sigtab_hydro_REF_EXP5 = cbind(as(sigtab_hydro_REF_EXP5, "data.frame"), as(tax_table(diff_abun_hydro_REF_EXP5)[rownames(sigtab_hydro_REF_EXP5), ], "matrix"))


x_hydro_REF_EXP5 = tapply(sigtab_hydro_REF_EXP5$log2FoldChange, sigtab_hydro_REF_EXP5$Phylum, function(x) max(x))
x_hydro_REF_EXP5 = sort(x_hydro_REF_EXP5, TRUE)
sigtab_hydro_REF_EXP5$Phylum = factor(as.character(sigtab_hydro_REF_EXP5$Phylum), levels=names(x_hydro_REF_EXP5))
# Genus order
x_hydro_REF_EXP5 = tapply(sigtab_hydro_REF_EXP5$log2FoldChange, sigtab_hydro_REF_EXP5$Genus, function(x) max(x))
x_hydro_REF_EXP5 = sort(x_hydro_REF_EXP5, TRUE)
sigtab_hydro_REF_EXP5$Genus = factor(as.character(sigtab_hydro_REF_EXP5$Genus), levels=sort(names(x_hydro_REF_EXP5)))

ggplot(sigtab_hydro_REF_EXP5, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of PCR1/BRR2")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")




#### ALDEx2 ####

diff_abun_BRR2_PCR1 = subset_samples(sample_ps, Stream_Type=="ACWA Streams")
diff_abun_BRR2_PCR1 = subset_samples(diff_abun_BRR2_PCR1, Site != "PCR3")

tax_names<-rownames_to_column(taxonomy, var="Sequences")


aldex2_BRR2_PCR1 <- ALDEx2::aldex(data.frame(phyloseq::otu_table(diff_abun_BRR2_PCR1)), phyloseq::sample_data(diff_abun_BRR2_PCR1)$Site, test="t", effect = TRUE, denom="iqlr")

sig_aldex2_BRR2_PCR1 <- aldex2_BRR2_PCR1 %>%
  rownames_to_column(var = "Sequences") %>%
  filter(wi.eBH < 0.01) %>%
  arrange(effect, wi.eBH) %>%
  dplyr::select(Sequences, diff.btw, diff.win, effect, wi.ep, wi.eBH)
sig_aldex2_BRR2_PCR1 <- dplyr::inner_join(sig_aldex2_BRR2_PCR1, tax_names, by="Sequences")


aldex_BRR2_PCR1 = tapply(sig_aldex2_BRR2_PCR1$effect, sig_aldex2_BRR2_PCR1$Phylum, function(x) max(x))
aldex_BRR2_PCR1 = sort(aldex_BRR2_PCR1, TRUE)
sig_aldex2_BRR2_PCR1$Phylum = factor(as.character(sig_aldex2_BRR2_PCR1$Phylum), levels=names(aldex_BRR2_PCR1))
# Genus order
aldex_BRR2_PCR1 = tapply(sig_aldex2_BRR2_PCR1$effect, sig_aldex2_BRR2_PCR1$Genus, function(x) max(x))
aldex_BRR2_PCR1 = sort(aldex_BRR2_PCR1, TRUE)
sig_aldex2_BRR2_PCR1$Genus = factor(as.character(sig_aldex2_BRR2_PCR1$Genus), levels=sort(names(aldex_BRR2_PCR1)))


ggplot(sig_aldex2_BRR2_PCR1, aes(x=Genus, y=effect, color=Phylum))+
  geom_point(size=4) + 
  theme_bw()+
  theme(axis.text.x= element_text(angle = 45, hjust = 1, vjust=1, size=14, color="black"), axis.text.y = element_text(size=14, color="black"))+
  #theme(legend.text=element_text(size=6, color="black")) + 
  #theme(legend.key.size = unit(0.5, "cm"), legend.title = element_text(size=8, color="black"),  legend.position="right")+
  guides(color=guide_legend(nrow=30, byrow=TRUE, ncol=1))+
  rremove("grid")+
  labs(y = expression(Log[2]~Difference))+
  xlab("Metabolic Pathways")

#Finds different taxa that are differentially abundant

# Hydropsychidae #
diff_abun_hydro_REF_EXP5 = subset_samples(diff_abun_BRR2_PCR1, Family == "Hydropsychidae")
```

### REF vs EXP15
```{r}
diff_abun_BRR2_PCR3 = subset_samples(sample_ps, Stream_Type=="ACWA Streams")
diff_abun_BRR2_PCR3 = subset_samples(diff_abun_BRR2_PCR3, Site != "PCR1")

deseq_BRR2_PCR3 <- phyloseq_to_deseq2(diff_abun_BRR2_PCR3, ~ Site)
deseq_BRR2_PCR3 <- DESeq(deseq_BRR2_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_BRR2_PCR3 <- results(deseq_BRR2_PCR3, cooksCutoff = FALSE)
head(res_deseq_BRR2_PCR3)
alpha = 0.001
sigtab_BRR2_PCR3 = res_deseq_BRR2_PCR3[which(res_deseq_BRR2_PCR3$padj < alpha), ]
sigtab_BRR2_PCR3 = cbind(as(sigtab_BRR2_PCR3, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_BRR2_PCR3), ], "matrix"))

#write.csv(as.data.frame(sigtab_BRR2_PCR3), file="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_BRR2_PCR3.csv")

#Plotting the log2fold change in bacteria: PCR3 vs BRR2

x_BRR2_PCR3 = tapply(sigtab_BRR2_PCR3$log2FoldChange, sigtab_BRR2_PCR3$Phylum, function(x) max(x))
x_BRR2_PCR3 = sort(x_BRR2_PCR3, TRUE)
sigtab_BRR2_PCR3$Phylum = factor(as.character(sigtab_BRR2_PCR3$Phylum), levels=names(x_BRR2_PCR3))
# Genus order
x_BRR2_PCR3 = tapply(sigtab_BRR2_PCR3$log2FoldChange, sigtab_BRR2_PCR3$Genus, function(x) max(x))
x_BRR2_PCR3 = sort(x_BRR2_PCR3, TRUE)
sigtab_BRR2_PCR3$Genus = factor(as.character(sigtab_BRR2_PCR3$Genus), levels=sort(names(x_BRR2_PCR3)))

sigtab_BRR2_PCR3$Phylum<-factor(sigtab_BRR2_PCR3$Phylum, levels=c("Proteobacteria", "Bacteroidota", "Firmicutes"))

PCR3_BRR2<-ggplot(sigtab_BRR2_PCR3, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title=element_text(size=30)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position="right")+
  ggtitle("Differential Abundance of PCR3/BRR2")+
  scale_color_manual(values=c("red", "#234F1E", "blue"))+
  rremove("grid")
PCR3_BRR2

#ggsave(filename="Microbiome_ED/Non decontaminated analysis/deseq_PCR3_BRR2.png", height=6.5, width=7.5)

## Hydropsychidae ##
diff_abun_hydro_REF_EXP15 = subset_samples(diff_abun_BRR2_PCR3, Family == "Hydropsychidae")

deseq_hydro_REF_EXP15 <- phyloseq_to_deseq2(diff_abun_hydro_REF_EXP15, ~ Site)
deseq_hydro_REF_EXP15 <- DESeq(deseq_hydro_REF_EXP15, test = "Wald", fitType = "parametric", sfType = "poscounts")  

res_deseq_hydro_REF_EXP15 <- results(deseq_hydro_REF_EXP15, cooksCutoff = FALSE)
res_deseq_hydro_REF_EXP15
alpha = 0.001
sigtab_hydro_REF_EXP15 = res_deseq_hydro_REF_EXP15[which(res_deseq_hydro_REF_EXP15$padj < alpha), ]
sigtab_hydro_REF_EXP15 = cbind(as(sigtab_hydro_REF_EXP15, "data.frame"), as(tax_table(diff_abun_hydro_REF_EXP15)[rownames(sigtab_hydro_REF_EXP15), ], "matrix"))


x_hydro_REF_EXP15 = tapply(sigtab_hydro_REF_EXP15$log2FoldChange, sigtab_hydro_REF_EXP15$Phylum, function(x) max(x))
x_hydro_REF_EXP15 = sort(x_hydro_REF_EXP15, TRUE)
sigtab_hydro_REF_EXP15$Phylum = factor(as.character(sigtab_hydro_REF_EXP15$Phylum), levels=names(x_hydro_REF_EXP15))
# Genus order
x_hydro_REF_EXP15 = tapply(sigtab_hydro_REF_EXP15$log2FoldChange, sigtab_hydro_REF_EXP15$Genus, function(x) max(x))
x_hydro_REF_EXP15 = sort(x_hydro_REF_EXP15, TRUE)
sigtab_hydro_REF_EXP15$Genus = factor(as.character(sigtab_hydro_REF_EXP15$Genus), levels=sort(names(x_hydro_REF_EXP15)))

ggplot(sigtab_hydro_REF_EXP15, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of EXP15/REF")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")
#2 diff abund bacteria

```

### EXP5 vs EXP15
```{r}
diff_abun_PCR1_PCR3 = subset_samples(sample_ps, Stream_Type=="ACWA Streams")
diff_abun_PCR1_PCR3 = subset_samples(diff_abun_PCR1_PCR3, Site != "BRR2")

deseq_PCR1_PCR3 <- phyloseq_to_deseq2(diff_abun_PCR1_PCR3, ~ Site)
deseq_PCR1_PCR3 <- DESeq(deseq_PCR1_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_PCR1_PCR3 <- results(deseq_PCR1_PCR3, cooksCutoff = FALSE)
res_deseq_PCR1_PCR3 #PCR3/PCR1
alpha = 0.001
sigtab_PCR1_PCR3 = res_deseq_PCR1_PCR3[which(res_deseq_PCR1_PCR3$padj < alpha), ]
sigtab_PCR1_PCR3 = cbind(as(sigtab_PCR1_PCR3, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_PCR1_PCR3), ], "matrix"))

#write.csv(as.data.frame(sigtab_PCR1_PCR3), file="Microbiome_ED/Non decontaminated analysis/Differential Abundance Results/deseq_PCR1_PCR3.csv")

#Plotting the log2fold change in bacteria: PCR3 vs PCR1

x_PCR1_PCR3 = tapply(sigtab_PCR1_PCR3$log2FoldChange, sigtab_PCR1_PCR3$Phylum, function(x) max(x))
x_PCR1_PCR3 = sort(x_PCR1_PCR3, TRUE)
sigtab_PCR1_PCR3$Phylum = factor(as.character(sigtab_PCR1_PCR3$Phylum), levels=names(x_PCR1_PCR3))
# Family order
x_PCR1_PCR3 = tapply(sigtab_PCR1_PCR3$log2FoldChange, sigtab_PCR1_PCR3$Genus, function(x) max(x))
x_PCR1_PCR3 = sort(x_PCR1_PCR3, TRUE)
sigtab_PCR1_PCR3$Genus = factor(as.character(sigtab_PCR1_PCR3$Genus), levels=sort(names(x_PCR1_PCR3)))

sigtab_PCR1_PCR3$Phylum<-factor(sigtab_PCR1_PCR3$Phylum, levels=c("Proteobacteria", "Cyanobacteria", "Nitrospirota"))

PCR3_PCR1<-ggplot(sigtab_PCR1_PCR3, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  scale_color_manual(values=c("red", "#fdb915", "brown"))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1)) + theme(axis.text=element_text(size=26, color="black"), axis.title=element_text(size=26)) +
          theme(legend.text=element_text(size=26, color="black")) + 
          theme(legend.title = element_text(size=26, color="black"), legend.position="right")+
  ggtitle("Differential Abundance of PCR3/PCR1")+
  rremove("grid")

PCR3_PCR1

#ggsave(filename="Differential Abundance Results/deseq_PCR3_PCR1.png", height=6.5, width=7.5)

## Hydropsychidae ##
diff_abun_hydro_EXP5_EXP15 = subset_samples(diff_abun_PCR1_PCR3, Family == "Hydropsychidae")

deseq_hydro_EXP5_EXP15 <- phyloseq_to_deseq2(diff_abun_hydro_EXP5_EXP15, ~ Site)
deseq_hydro_EXP5_EXP15 <- DESeq(deseq_hydro_EXP5_EXP15, test = "Wald", fitType = "parametric", sfType = "poscounts")  

res_deseq_hydro_EXP5_EXP15 <- results(deseq_hydro_EXP5_EXP15, cooksCutoff = FALSE)
res_deseq_hydro_EXP5_EXP15
alpha = 0.001
sigtab_hydro_EXP5_EXP15 = res_deseq_hydro_EXP5_EXP15[which(res_deseq_hydro_EXP5_EXP15$padj < alpha), ]
sigtab_hydro_EXP5_EXP15 = cbind(as(sigtab_hydro_EXP5_EXP15, "data.frame"), as(tax_table(diff_abun_hydro_EXP5_EXP15)[rownames(sigtab_hydro_EXP5_EXP15), ], "matrix"))


x_hydro_EXP5_EXP15 = tapply(sigtab_hydro_EXP5_EXP15$log2FoldChange, sigtab_hydro_EXP5_EXP15$Phylum, function(x) max(x))
x_hydro_EXP5_EXP15 = sort(x_hydro_EXP5_EXP15, TRUE)
sigtab_hydro_EXP5_EXP15$Phylum = factor(as.character(sigtab_hydro_EXP5_EXP15$Phylum), levels=names(x_hydro_EXP5_EXP15))
# Genus order
x_hydro_EXP5_EXP15 = tapply(sigtab_hydro_EXP5_EXP15$log2FoldChange, sigtab_hydro_EXP5_EXP15$Genus, function(x) max(x))
x_hydro_EXP5_EXP15 = sort(x_hydro_EXP5_EXP15, TRUE)
sigtab_hydro_EXP5_EXP15$Genus = factor(as.character(sigtab_hydro_EXP5_EXP15$Genus), levels=sort(names(x_hydro_EXP5_EXP15)))

ggplot(sigtab_hydro_EXP5_EXP15, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of EXP15/REF")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")
#2 diff abund bacteria

## Baetidae ##
diff_abun_baetid_EXP5_EXP15 = subset_samples(diff_abun_PCR1_PCR3, Family == "Baetidae")

deseq_baetid_EXP5_EXP15 <- phyloseq_to_deseq2(diff_abun_baetid_EXP5_EXP15, ~ Site)
deseq_baetid_EXP5_EXP15 <- DESeq(deseq_baetid_EXP5_EXP15, test = "Wald", fitType = "parametric", sfType = "poscounts")  

res_deseq_baetid_EXP5_EXP15 <- results(deseq_baetid_EXP5_EXP15, cooksCutoff = FALSE)
res_deseq_baetid_EXP5_EXP15
alpha = 0.001
sigtab_baetid_EXP5_EXP15 = res_deseq_baetid_EXP5_EXP15[which(res_deseq_baetid_EXP5_EXP15$padj < alpha), ]
sigtab_baetid_EXP5_EXP15 = cbind(as(sigtab_baetid_EXP5_EXP15, "data.frame"), as(tax_table(diff_abun_baetid_EXP5_EXP15)[rownames(sigtab_baetid_EXP5_EXP15), ], "matrix"))


x_baetid_EXP5_EXP15 = tapply(sigtab_baetid_EXP5_EXP15$log2FoldChange, sigtab_baetid_EXP5_EXP15$Phylum, function(x) max(x))
x_baetid_EXP5_EXP15 = sort(x_baetid_EXP5_EXP15, TRUE)
sigtab_baetid_EXP5_EXP15$Phylum = factor(as.character(sigtab_baetid_EXP5_EXP15$Phylum), levels=names(x_baetid_EXP5_EXP15))
# Genus order
x_baetid_EXP5_EXP15 = tapply(sigtab_baetid_EXP5_EXP15$log2FoldChange, sigtab_baetid_EXP5_EXP15$Genus, function(x) max(x))
x_baetid_EXP5_EXP15 = sort(x_baetid_EXP5_EXP15, TRUE)
sigtab_baetid_EXP5_EXP15$Genus = factor(as.character(sigtab_baetid_EXP5_EXP15$Genus), levels=sort(names(x_baetid_EXP5_EXP15)))

ggplot(sigtab_baetid_EXP5_EXP15, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of EXP15/REF")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")
#4 diff abund bacteria

## Ephemeroptera ##
diff_abun_ephem_EXP5_EXP15 = subset_samples(diff_abun_PCR1_PCR3, Family == "Ephemeroptera")

deseq_ephem_EXP5_EXP15 <- phyloseq_to_deseq2(diff_abun_ephem_EXP5_EXP15, ~ Site)
deseq_ephem_EXP5_EXP15 <- DESeq(deseq_ephem_EXP5_EXP15, test = "Wald", fitType = "parametric", sfType = "poscounts")  

res_deseq_ephem_EXP5_EXP15 <- results(deseq_ephem_EXP5_EXP15, cooksCutoff = FALSE)
res_deseq_ephem_EXP5_EXP15
alpha = 0.001
sigtab_ephem_EXP5_EXP15 = res_deseq_ephem_EXP5_EXP15[which(res_deseq_ephem_EXP5_EXP15$padj < alpha), ]
sigtab_ephem_EXP5_EXP15 = cbind(as(sigtab_ephem_EXP5_EXP15, "data.frame"), as(tax_table(diff_abun_ephem_EXP5_EXP15)[rownames(sigtab_ephem_EXP5_EXP15), ], "matrix"))


x_ephem_EXP5_EXP15 = tapply(sigtab_ephem_EXP5_EXP15$log2FoldChange, sigtab_ephem_EXP5_EXP15$Phylum, function(x) max(x))
x_ephem_EXP5_EXP15 = sort(x_ephem_EXP5_EXP15, TRUE)
sigtab_ephem_EXP5_EXP15$Phylum = factor(as.character(sigtab_ephem_EXP5_EXP15$Phylum), levels=names(x_ephem_EXP5_EXP15))
# Genus order
x_ephem_EXP5_EXP15 = tapply(sigtab_ephem_EXP5_EXP15$log2FoldChange, sigtab_ephem_EXP5_EXP15$Genus, function(x) max(x))
x_ephem_EXP5_EXP15 = sort(x_ephem_EXP5_EXP15, TRUE)
sigtab_ephem_EXP5_EXP15$Genus = factor(as.character(sigtab_ephem_EXP5_EXP15$Genus), levels=sort(names(x_ephem_EXP5_EXP15)))

ggplot(sigtab_ephem_EXP5_EXP15, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=30, face="italic")) + theme(axis.text=element_text(size=30, color="black"), axis.title.x=element_text(size=30, vjust=10)) +
          theme(legend.text=element_text(size=30, color="black")) + 
          theme(legend.title = element_text(size=30, color="black"), legend.position = "right")+
  ggtitle("Differential Abundance of EXP15/REF")+
  scale_color_manual(values=c("red", "#234F1E", "blue", "#fdb915", "purple"))+
  rremove("grid")
#2 diff abund bacteria
```

### ANCOMBC (All ACWA streams run together) ###
```{r}
#### Hydropsychidae ####
hydro_ACWA<-subset_samples(ACWA_ps, Family=="Hydropsychidae")
nsamples(hydro_ACWA) #24

# ancombc
hydro_ancom_out <- ancombc(phyloseq = hydro_ACWA, formula = "Site",
              p_adj_method = "holm", lib_cut = 1000, 
              group = "Site", struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.01, global = TRUE)
hydro_ancom_res <- hydro_ancom_out$res

tab_lfc = hydro_ancom_res$lfc

hydro_tax<-as.data.frame(tax_table(hydro_ACWA))
hydro_tax<-rownames_to_column(hydro_tax, var="taxon")

hydro_ancom_res_combined<-dplyr::inner_join(hydro_tax, tab_lfc, by="taxon")



```

### Putting all ACWA graphs together ###
```{r}
PCR3_BRR2_2<-PCR3_BRR2 + theme(plot.title = element_blank(), axis.title.y = element_blank(), axis.title.x=element_blank())
PCR1_BRR2_2<-PCR1_BRR2 + theme(plot.title = element_blank(), axis.title.y = element_blank(), axis.title.x=element_blank())
PCR3_PCR1_2<-PCR3_PCR1 + theme(plot.title = element_blank(), axis.title.y = element_blank(), axis.title.x=element_blank())

ACWA_deseq<-ggarrange(PCR1_BRR2_2, NULL, PCR3_BRR2_2, NULL, PCR3_PCR1_2, nrow=5, ncol=1, align="v", labels=c("EXP 5 % - REF", "", "EXP 15 % - REF", "", "EXP 15 % - EXP 5 %"), font.label = list(size = 30), vjust=-0.3, hjust=0, heights=c(1, 0.05, 1, 0.05, 1))+theme(plot.margin = margin(1,1,1,1, "cm")) 

annotate_figure(ACWA_deseq, left = text_grob(bquote(""*~Log[2]~Ratio~of~Absolute~Abundance*""), size=36, rot = 90))

#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_all_ACWA.png", height=18, width=22, bg="white")

#Only treatment streams vs reference

ACWA_treat<-ggarrange(PCR1_BRR2_2, NULL, PCR3_BRR2_2, nrow=3, ncol=1, align="v", labels=c("EXP 5 % - REF", "", "EXP 15 % - REF"), font.label = list(size = 26), vjust=-0.3, hjust=0, heights=c(1, 0.05, 1, 0.05, 1))+theme(plot.margin = margin(1,1,1,1, "cm")) 

ACWA_treat

#annotate_figure(ACWA_treat, left = text_grob(bquote(""*~Log[2]~Ratio~of~Absolute~Abundance*""), size=40, rot = 90)) + theme(axis.title.y = element_text(vjust = -1))


#ggsave(filename="Microbiome_ED/Final analysis/Differential Abundance Results/deseq_treatment_vs_reference_ACWA.png", height=18, width=24, bg="white")

```

### Effluent bacteria in Bow River samples ###
```{r}

#Checking for Flavobacterium in Bow River samples
Bow_Flav<-subset_taxa(Bow, Genus=="Flavobacterium")
Bow_Flav<-prune_taxa(taxa_sums(Bow_Flav) > 0, Bow_Flav)
Bow_Flav_df<-psmelt(Bow_Flav)

Bow_Flav_agg<-aggregate(Abundance~Genus+Site, data=Bow_Flav_df, FUN=mean)

ggplot(Bow_Flav_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Sunalta but the mean total abundance is super low in Bow River sites. 

#Flavobacterium can sometimes be found in wastewater 
#Flavobacterium have a prominent role in freshwater bacterioplankton communities due to their high rates of substrate uptake and growth, growth on algal-derived substrates and high mortality rates from bacterivory.


#Limnohabitans
ACWA_Lim<-subset_taxa(ACWA, Genus=="Limnohabitans")
ACWA_Lim_df<-psmelt(ACWA_Lim)
ACWA_Lim_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Lim_df, FUN=mean)

ggplot(ACWA_Lim_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at PCR1 then PCR3. Low in BRR2. Overall mean total abundance is low for all 3 streams. 

#Checking for Limnohabitans in Bow River samples
Bow_Lim<-subset_taxa(Bow, Genus=="Limnohabitans")
Bow_Lim<-prune_taxa(taxa_sums(Bow_Lim) > 0, Bow_Lim)
Bow_Lim_df<-psmelt(Bow_Lim)

Bow_Lim_agg<-aggregate(Abundance~Genus+Site, data=Bow_Lim_df, FUN=mean)

ggplot(Bow_Lim_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge then Policeman Flats. 


#Ideonella
#Not sure of the species but probably is Ideonella dechloratans because it is perchlorate reducing and perchlorate is found in wastewater since it's commonly used to treat effluent. 
#Could also be Ideonella sakaiensis which able to break down plastic. Potentially has more plastic in the wastewater streams? 
#Or Ideonella azotifigens? It is nitrogen fixing

#PCR1 and PCR3 had a higher abundance of this bacteria compared to the control streams.
ACWA_Ideon<-subset_taxa(ACWA, Genus=="Ideonella")
table(tax_table(ACWA_Ideon)[,7])
ACWA_Ideon_df<-psmelt(ACWA_Ideon)
ACWA_Ideon_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Ideon_df, FUN=mean)
ggplot(ACWA_Ideon_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")+
  ylab("Mean Total Abundance")
  #Much higher in PCR1 and PCR3 streams

#Checking for Ideonella in Bow River samples
Bow<-subset_samples(sample_ps2, Stream_Type=="Bow River")
Bow_Ideon<-subset_taxa(Bow, Genus=="Ideonella")
Bow_Ideon<-prune_taxa(taxa_sums(Bow_Ideon) > 0, Bow_Ideon)
Bow_Ideon_df<-psmelt(Bow_Ideon)

Bow_Ideon_agg<-aggregate(Abundance~Genus+Site, data=Bow_Ideon_df, FUN=mean)

ggplot(Bow_Ideon_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Cushing Bridge had the highest mean total abundance of Ideonella. Lowest at Cochrane and Sunalta.

#Rhodoferax
#Oxidizes acetate with with the reduction of Fe(3)
ACWA_Rhodo<-subset_taxa(ACWA, Genus=="Rhodoferax")
ACWA_Rhodo_df<-psmelt(ACWA_Rhodo)
ACWA_Rhodo_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Rhodo_df, FUN=mean)
ggplot(ACWA_Rhodo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest mean total abundance at PCR3. Could potentially indicate an anoxic environment.

#Checking for Rhodoferax in Bow River samples
Bow_Rhodo<-subset_taxa(Bow, Genus=="Rhodoferax")
Bow_Rhodo<-prune_taxa(taxa_sums(Bow_Rhodo) > 0, Bow_Rhodo)
Bow_Rhodo_df<-psmelt(Bow_Rhodo)

Bow_Rhodo_agg<-aggregate(Abundance~Genus+Site, data=Bow_Rhodo_df, FUN=mean)

ggplot(Bow_Rhodo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge. Similar across all other sites. 


#Pajaroellobacter (abortibovis?) only one known species
#Is the agent for Epizootic bovine abortion (EBA), an arthropod-borne bacterial disease that causes significant economic loss for cattle producers in the western United States

ACWA_paja<-subset_taxa(ACWA, Genus=="Pajaroellobacter")
ACWA_paja_df<-psmelt(ACWA_paja)
ACWA_paja_agg<-aggregate(Abundance~Genus+Site, data=ACWA_paja_df, FUN=mean)
ggplot(ACWA_paja_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Mean total abundance was highest at PCR1 but the same between PCR3 and BRR2

#Checking for Paja in Bow River samples
Bow_paja<-subset_taxa(Bow, Genus=="Pajaroellobacter")
Bow_paja<-prune_taxa(taxa_sums(Bow_paja) > 0, Bow_paja)
Bow_paja_df<-psmelt(Bow_paja)

Bow_paja_agg<-aggregate(Abundance~Genus+Site, data=Bow_paja_df, FUN=mean)

ggplot(Bow_paja_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Mean abundance is much higher at Policeman Flats compared to other sites. Cochrane is virtually 0. 


#Novosphingobium
ACWA_novo<-subset_taxa(ACWA, Genus=="Novosphingobium")
ACWA_novo_df<-psmelt(ACWA_novo)
ACWA_novo_agg<-aggregate(Abundance~Genus+Site, data=ACWA_novo_df, FUN=mean)
ggplot(ACWA_novo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at BRR2... but the differential abundance test said the opposite.

#Checking for Novosphingobium in Bow River samples
Bow_novo<-subset_taxa(Bow, Genus=="Novosphingobium")
Bow_novo<-prune_taxa(taxa_sums(Bow_novo) > 0, Bow_novo)
Bow_novo_df<-psmelt(Bow_novo)

Bow_novo_agg<-aggregate(Abundance~Genus+Site, data=Bow_novo_df, FUN=mean)

ggplot(Bow_novo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Policeman Flats


#Alistipes
ACWA_Ali<-subset_taxa(ACWA, Genus=="Alistipes")
ACWA_Ali_df<-psmelt(ACWA_Ali)

ACWA_Ali_agg<-aggregate(Abundance ~ Genus + Site, data = ACWA_Ali_df, FUN = mean)

ggplot(ACWA_Ali_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Lowest at PCR1. Same at BRR2 and PCR3. 

#Checking for Alistipes in Bow River samples
Bow_ali<-subset_taxa(Bow, Genus=="Alistipes")
Bow_ali<-prune_taxa(taxa_sums(Bow_ali) > 0, Bow_ali)
Bow_ali_df<-psmelt(Bow_ali)

Bow_ali_agg<-aggregate(Abundance~Genus+Site, data=Bow_ali_df, FUN=mean)

ggplot(Bow_ali_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge and second highest at Policeman Flats.

#Alistipes is known to be pathogenic in colorectal cancer and is associated with mentalsigns of depression (Parker et al., 2020)

```

# Endosymbionts

```{r Relative abundance of endosymbionts to total abundance/site and taxa}
endo<-subset_taxa(sample_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Rickettsia" | Genus =="Wolbachia" | Genus=="Spiroplasma" | Genus=="Diplorickettsia")
endo<-prune_taxa(taxa_sums(endo) > 0, endo) 
endo #31 taxa

#### Larval Hydropsychidae Bow River ####

hydro_Bow<-subset_samples(sample_ps, Family=="Hydropsychidae" & Stream_Type=="Bow River")

#Cochrane 

hydro_coch<-subset_samples(hydro_Bow, Site=="Cochrane")

hydro_coch_glom = tax_glom(hydro_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hydro_coch_glom<-prune_taxa(taxa_sums(hydro_coch_glom) > 0, hydro_coch_glom)

hydro_coch_genus <- psmelt(hydro_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_hydro<- hydro_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_hydro$SD_squared ='^'(coch_endo_hydro$SD,2)
coch_endo_hydro$SD_squared_divided =(coch_endo_hydro$SD_squared/8) 
  
top_means_endo_hydro_coch<-aggregate(Mean ~ Genus, data = coch_endo_hydro, FUN = sum)
top_SD_endo_hydro_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_hydro, FUN = sum)

top_endo_hydro_coch<-merge(top_means_endo_hydro_coch, top_SD_endo_hydro_coch, by=c("Genus"))
top_endo_hydro_coch$SD_Final<-sqrt(top_endo_hydro_coch$SD_squared_divided)
View(top_endo_hydro_coch)

#Sunalta
hydro_sun<-subset_samples(hydro_Bow, Site=="Sunalta")

hydro_sun_glom = tax_glom(hydro_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hydro_sun_glom<-prune_taxa(taxa_sums(hydro_sun_glom) > 0, hydro_sun_glom)

hydro_sun_genus <- psmelt(hydro_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_hydro<- hydro_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_hydro$SD_squared ='^'(sun_endo_hydro$SD,2)
sun_endo_hydro$SD_squared_divided =(sun_endo_hydro$SD_squared/7) 
  
top_means_endo_hydro_sun<-aggregate(Mean ~ Genus, data = sun_endo_hydro, FUN = sum)
top_SD_endo_hydro_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_hydro, FUN = sum)

top_endo_hydro_sun<-merge(top_means_endo_hydro_sun, top_SD_endo_hydro_sun, by=c("Genus"))
top_endo_hydro_sun$SD_Final<-sqrt(top_endo_hydro_sun$SD_squared_divided)
View(top_endo_hydro_sun)

#Cushing Bridge
hydro_cush<-subset_samples(hydro_Bow, Site=="Cushing Bridge")

hydro_cush_glom = tax_glom(hydro_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hydro_cush_glom<-prune_taxa(taxa_sums(hydro_cush_glom) > 0, hydro_cush_glom)

hydro_cush_genus <- psmelt(hydro_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_hydro<- hydro_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_hydro$SD_squared ='^'(cush_endo_hydro$SD,2)
cush_endo_hydro$SD_squared_divided =(cush_endo_hydro$SD_squared/5) 
  
top_means_endo_hydro_cush<-aggregate(Mean ~ Genus, data = cush_endo_hydro, FUN = sum)
top_SD_endo_hydro_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_hydro, FUN = sum)

top_endo_hydro_cush<-merge(top_means_endo_hydro_cush, top_SD_endo_hydro_cush, by=c("Genus"))
top_endo_hydro_cush$SD_Final<-sqrt(top_endo_hydro_cush$SD_squared_divided)
View(top_endo_hydro_cush)

#Graves Bridge
hydro_grvbr<-subset_samples(hydro_Bow, Site=="Graves Bridge")

hydro_grvbr_glom = tax_glom(hydro_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hydro_grvbr_glom<-prune_taxa(taxa_sums(hydro_grvbr_glom) > 0, hydro_grvbr_glom)

hydro_grvbr_genus <- psmelt(hydro_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_hydro<- hydro_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_hydro$SD_squared ='^'(grvbr_endo_hydro$SD,2)
grvbr_endo_hydro$SD_squared_divided =(grvbr_endo_hydro$SD_squared/8) 
  
top_means_endo_hydro_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_hydro, FUN = sum)
top_SD_endo_hydro_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_hydro, FUN = sum)

top_endo_hydro_grvbr<-merge(top_means_endo_hydro_grvbr, top_SD_endo_hydro_grvbr, by=c("Genus"))
top_endo_hydro_grvbr$SD_Final<-sqrt(top_endo_hydro_grvbr$SD_squared_divided)
View(top_endo_hydro_grvbr)

#Policeman Flats
hydro_pmf<-subset_samples(hydro_Bow, Site=="Policeman Flats")

hydro_pmf_glom = tax_glom(hydro_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hydro_pmf_glom<-prune_taxa(taxa_sums(hydro_pmf_glom) > 0, hydro_pmf_glom)

hydro_pmf_genus <- psmelt(hydro_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_hydro<- hydro_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_hydro$SD_squared ='^'(pmf_endo_hydro$SD,2)
pmf_endo_hydro$SD_squared_divided =(pmf_endo_hydro$SD_squared/8) 
  
top_means_endo_hydro_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_hydro, FUN = sum)
top_SD_endo_hydro_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_hydro, FUN = sum)

top_endo_hydro_pmf<-merge(top_means_endo_hydro_pmf, top_SD_endo_hydro_pmf, by=c("Genus"))
top_endo_hydro_pmf$SD_Final<-sqrt(top_endo_hydro_pmf$SD_squared_divided)
View(top_endo_hydro_pmf)

## Plotting relative abundance of endosymbionts versus non endosymbionts ##

hydro_Bow_genus <- tax_glom(hydro_lar_Bow, taxrank = "Genus")
comp_hydro_Bow_genus <- transform_sample_counts(hydro_Bow_genus, function(x) x/sum(x))

hydro_Bow_genus_df <- psmelt(comp_hydro_Bow_genus)

hydro_agg <- aggregate(Abundance ~ Genus + Site_code, data = hydro_Bow_genus_df, FUN = mean) #aggregate to Phylum level, take the mean

#Sort abundance of each genus by mean at each site
sort_genus_hydro <- hydro_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_hydro

endosymbionts <- sort_genus_hydro$Genus[1901:1905]

  
endo_df_hydro <- hydro_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts))
endo_df_hydro

endo_agg_df<-aggregate(Abundance ~ Genus + Site_code, data = endo_df_hydro, FUN = sum)

endo_agg_df$Genus<-factor(endo_agg_df$Genus, levels=c("Rickettsia", "Other"), labels=c("Rickettsia", "Non-endosymbionts"))

endo_plot_hydro <- ggplot(endo_agg_df, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Genus", values=c("lightgoldenrod1", "honeydew4"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_hydro

#### Larval Heptageniidae Bow River ####
hep<-subset_samples(sample_ps, Family=="Heptageniidae")

#Cochrane 
hep_coch<-subset_samples(hep, Site=="Cochrane")

hep_coch_glom = tax_glom(hep_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hep_coch_glom<-prune_taxa(taxa_sums(hep_coch_glom) > 0, hep_coch_glom)

hep_coch_genus <- psmelt(hep_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_hep<- hep_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_hep$SD_squared ='^'(coch_endo_hep$SD,2)
coch_endo_hep$SD_squared_divided =(coch_endo_hep$SD_squared/8) 
  
top_means_endo_hep_coch<-aggregate(Mean ~ Genus, data = coch_endo_hep, FUN = sum)
top_SD_endo_hep_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_hep, FUN = sum)

top_endo_hep_coch<-merge(top_means_endo_hep_coch, top_SD_endo_hep_coch, by=c("Genus"))
top_endo_hep_coch$SD_Final<-sqrt(top_endo_hep_coch$SD_squared_divided)
View(top_endo_hep_coch)

#Sunalta
hep_sun<-subset_samples(hep, Site=="Sunalta")

hep_sun_glom = tax_glom(hep_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hep_sun_glom<-prune_taxa(taxa_sums(hep_sun_glom) > 0, hep_sun_glom)

hep_sun_genus <- psmelt(hep_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_hep<- hep_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_hep$SD_squared ='^'(sun_endo_hep$SD,2)
sun_endo_hep$SD_squared_divided =(sun_endo_hep$SD_squared/8) 
  
top_means_endo_hep_sun<-aggregate(Mean ~ Genus, data = sun_endo_hep, FUN = sum)
top_SD_endo_hep_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_hep, FUN = sum)

top_endo_hep_sun<-merge(top_means_endo_hep_sun, top_SD_endo_hep_sun, by=c("Genus"))
top_endo_hep_sun$SD_Final<-sqrt(top_endo_hep_sun$SD_squared_divided)
View(top_endo_hep_sun)

#Cushing Bridge
hep_cush<-subset_samples(hep, Site=="Cushing Bridge")

hep_cush_glom = tax_glom(hep_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hep_cush_glom<-prune_taxa(taxa_sums(hep_cush_glom) > 0, hep_cush_glom)

hep_cush_genus <- psmelt(hep_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_hep<- hep_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_hep$SD_squared ='^'(cush_endo_hep$SD,2)
cush_endo_hep$SD_squared_divided =(cush_endo_hep$SD_squared/8) 
  
top_means_endo_hep_cush<-aggregate(Mean ~ Genus, data = cush_endo_hep, FUN = sum)
top_SD_endo_hep_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_hep, FUN = sum)

top_endo_hep_cush<-merge(top_means_endo_hep_cush, top_SD_endo_hep_cush, by=c("Genus"))
top_endo_hep_cush$SD_Final<-sqrt(top_endo_hep_cush$SD_squared_divided)
View(top_endo_hep_cush)

#Graves Bridge
hep_grvbr<-subset_samples(hep, Site=="Graves Bridge")

hep_grvbr_glom = tax_glom(hep_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hep_grvbr_glom<-prune_taxa(taxa_sums(hep_grvbr_glom) > 0, hep_grvbr_glom)

hep_grvbr_genus <- psmelt(hep_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_hep<- hep_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_hep$SD_squared ='^'(grvbr_endo_hep$SD,2)
grvbr_endo_hep$SD_squared_divided =(grvbr_endo_hep$SD_squared/8) 
  
top_means_endo_hep_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_hep, FUN = sum)
top_SD_endo_hep_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_hep, FUN = sum)

top_endo_hep_grvbr<-merge(top_means_endo_hep_grvbr, top_SD_endo_hep_grvbr, by=c("Genus"))
top_endo_hep_grvbr$SD_Final<-sqrt(top_endo_hep_grvbr$SD_squared_divided)
View(top_endo_hep_grvbr)

#Policeman Flats
hep_pmf<-subset_samples(hep, Site=="Policeman Flats")

hep_pmf_glom = tax_glom(hep_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
hep_pmf_glom<-prune_taxa(taxa_sums(hep_pmf_glom) > 0, hep_pmf_glom)

hep_pmf_genus <- psmelt(hep_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_hep<- hep_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_hep$SD_squared ='^'(pmf_endo_hep$SD,2)
pmf_endo_hep$SD_squared_divided =(pmf_endo_hep$SD_squared/8) 
  
top_means_endo_hep_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_hep, FUN = sum)
top_SD_endo_hep_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_hep, FUN = sum)

top_endo_hep_pmf<-merge(top_means_endo_hep_pmf, top_SD_endo_hep_pmf, by=c("Genus"))
top_endo_hep_pmf$SD_Final<-sqrt(top_endo_hep_pmf$SD_squared_divided)
View(top_endo_hep_pmf)


## Plotting relative abundance of endosymbionts versus non endosymbionts ##

hep_Bow_genus <- tax_glom(hep_lar_Bow, taxrank = "Genus")
comp_hep_Bow_genus <- microbiome::transform(hep_Bow_genus, "compositional")

hep_Bow_genus_df <- psmelt(comp_hep_Bow_genus)

hep_agg <- aggregate(Abundance ~ Genus + Site_code, data = hep_Bow_genus_df, FUN = mean)

#Sort abundance of each genus by mean at each site
sort_genus_hep <- hep_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_hep

endosymbionts_hep <- sort_genus_hep$Genus[c(1901:1905, 2346:2350)]
endosymbionts_hep
  
endo_df_hep <- hep_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts_hep))
endo_df_hep

endo_agg_df_hep<-aggregate(Abundance ~ Genus + Site_code, data = endo_df_hep, FUN = sum)

endo_agg_df_hep$Genus<-factor(endo_agg_df_hep$Genus, levels=c("Rickettsia", "Wolbachia", "Other"), labels=c("Rickettsia", "Wolbachia", "Non-endosymbionts"))

endo_plot_hep <- ggplot(endo_agg_df_hep, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Genus", values=c("lightgoldenrod1", "lightpink2", "honeydew4"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_hep

#### Larval Chironomidae Bow River ####
chiro<-subset_samples(sample_ps, Family=="Chironomidae")

#Cochrane 
chiro_coch<-subset_samples(chiro, Site=="Cochrane")

chiro_coch_glom = tax_glom(chiro_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
chiro_coch_glom<-prune_taxa(taxa_sums(chiro_coch_glom) > 0, chiro_coch_glom)

chiro_coch_genus <- psmelt(chiro_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_chiro<- chiro_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_chiro$SD_squared ='^'(coch_endo_chiro$SD,2)
coch_endo_chiro$SD_squared_divided =(coch_endo_chiro$SD_squared/4) 
  
top_means_endo_chiro_coch<-aggregate(Mean ~ Genus, data = coch_endo_chiro, FUN = sum)
top_SD_endo_chiro_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_chiro, FUN = sum)

top_endo_chiro_coch<-merge(top_means_endo_chiro_coch, top_SD_endo_chiro_coch, by=c("Genus"))
top_endo_chiro_coch$SD_Final<-sqrt(top_endo_chiro_coch$SD_squared_divided)
View(top_endo_chiro_coch)

#Cushing Bridge
chiro_cush<-subset_samples(chiro, Site=="Cushing Bridge")

chiro_cush_glom = tax_glom(chiro_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
chiro_cush_glom<-prune_taxa(taxa_sums(chiro_cush_glom) > 0, chiro_cush_glom)

chiro_cush_genus <- psmelt(chiro_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_chiro<- chiro_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_chiro$SD_squared ='^'(cush_endo_chiro$SD,2)
cush_endo_chiro$SD_squared_divided =(cush_endo_chiro$SD_squared/6) 
  
top_means_endo_chiro_cush<-aggregate(Mean ~ Genus, data = cush_endo_chiro, FUN = sum)
top_SD_endo_chiro_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_chiro, FUN = sum)

top_endo_chiro_cush<-merge(top_means_endo_chiro_cush, top_SD_endo_chiro_cush, by=c("Genus"))
top_endo_chiro_cush$SD_Final<-sqrt(top_endo_chiro_cush$SD_squared_divided)
View(top_endo_chiro_cush)

#Policeman Flats
chiro_pmf<-subset_samples(chiro, Site=="Policeman Flats")

chiro_pmf_glom = tax_glom(chiro_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
chiro_pmf_glom<-prune_taxa(taxa_sums(chiro_pmf_glom) > 0, chiro_pmf_glom)

chiro_pmf_genus <- psmelt(chiro_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_chiro<- chiro_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_chiro$SD_squared ='^'(pmf_endo_chiro$SD,2)
pmf_endo_chiro$SD_squared_divided =(pmf_endo_chiro$SD_squared/4) 
  
top_means_endo_chiro_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_chiro, FUN = sum)
top_SD_endo_chiro_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_chiro, FUN = sum)

top_endo_chiro_pmf<-merge(top_means_endo_chiro_pmf, top_SD_endo_chiro_pmf, by=c("Genus"))
top_endo_chiro_pmf$SD_Final<-sqrt(top_endo_chiro_pmf$SD_squared_divided)
View(top_endo_chiro_pmf)

## Plotting relative abundance of endosymbionts versus non endosymbionts ##

chiro_Bow_genus <- tax_glom(chiro_lar_Bow, taxrank = "Genus")
comp_chiro_Bow_genus <- transform_sample_counts(chiro_Bow_genus, function(x) x/sum(x))

chiro_Bow_genus_df <- psmelt(comp_chiro_Bow_genus)

chiro_agg <- aggregate(Abundance ~ Genus + Site_code, data = chiro_Bow_genus_df, FUN = mean)

#Sort abundance of each genus by mean at each site
sort_genus_chiro <- chiro_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_chiro

endosymbionts_chiro <- sort_genus_chiro$Genus[1901:1905]

endo_df_chiro <- chiro_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts_chiro))
endo_df_chiro

endo_agg_df_chiro<-aggregate(Abundance ~ Genus + Site_code, data = endo_df_chiro, FUN = sum)

endo_agg_df_chiro$Genus<-factor(endo_agg_df_chiro$Genus, levels=c("Rickettsia", "Other"), labels=c("Rickettsia", "Non-endosymbionts"))

endo_plot_chiro <- ggplot(endo_agg_df_chiro, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Genus", values=c("honeydew4"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_chiro


#### Larval Perlidae Bow River ####
perl<-subset_samples(sample_ps, Family=="Perlidae")

#Cochrane 
perl_coch<-subset_samples(perl, Site=="Cochrane")

perl_coch_glom = tax_glom(perl_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
perl_coch_glom<-prune_taxa(taxa_sums(perl_coch_glom) > 0, perl_coch_glom)

perl_coch_genus <- psmelt(perl_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_perl<- perl_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_perl$SD_squared ='^'(coch_endo_perl$SD,2)
coch_endo_perl$SD_squared_divided =(coch_endo_perl$SD_squared/8) 
  
top_means_endo_perl_coch<-aggregate(Mean ~ Genus, data = coch_endo_perl, FUN = sum)
top_SD_endo_perl_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_perl, FUN = sum)

top_endo_perl_coch<-merge(top_means_endo_perl_coch, top_SD_endo_perl_coch, by=c("Genus"))
top_endo_perl_coch$SD_Final<-sqrt(top_endo_perl_coch$SD_squared_divided)
View(top_endo_perl_coch)

#Cushing Bridge
perl_cush<-subset_samples(perl, Site=="Cushing Bridge")

perl_cush_glom = tax_glom(perl_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
perl_cush_glom<-prune_taxa(taxa_sums(perl_cush_glom) > 0, perl_cush_glom)

perl_cush_genus <- psmelt(perl_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_perl<- perl_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_perl$SD_squared ='^'(cush_endo_perl$SD,2)
cush_endo_perl$SD_squared_divided =(cush_endo_perl$SD_squared/5) 
  
top_means_endo_perl_cush<-aggregate(Mean ~ Genus, data = cush_endo_perl, FUN = sum)
top_SD_endo_perl_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_perl, FUN = sum)

top_endo_perl_cush<-merge(top_means_endo_perl_cush, top_SD_endo_perl_cush, by=c("Genus"))
top_endo_perl_cush$SD_Final<-sqrt(top_endo_perl_cush$SD_squared_divided)
View(top_endo_perl_cush)

#Policeman Flats
perl_pmf<-subset_samples(perl, Site=="Policeman Flats")

perl_pmf_glom = tax_glom(perl_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
perl_pmf_glom<-prune_taxa(taxa_sums(perl_pmf_glom) > 0, perl_pmf_glom)

perl_pmf_genus <- psmelt(perl_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_perl<- perl_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_perl$SD_squared ='^'(pmf_endo_perl$SD,2)
pmf_endo_perl$SD_squared_divided =(pmf_endo_perl$SD_squared/8) 
  
top_means_endo_perl_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_perl, FUN = sum)
top_SD_endo_perl_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_perl, FUN = sum)

top_endo_perl_pmf<-merge(top_means_endo_perl_pmf, top_SD_endo_perl_pmf, by=c("Genus"))
top_endo_perl_pmf$SD_Final<-sqrt(top_endo_perl_pmf$SD_squared_divided)
View(top_endo_perl_pmf)

## Plotting relative abundance of endosymbionts versus non endosymbionts ##

perl_Bow_genus <- tax_glom(perl_lar_Bow, taxrank = "Genus")
comp_perl_Bow_genus <- transform_sample_counts(perl_Bow_genus, function(x) x/sum(x))

perl_Bow_genus_df <- psmelt(comp_perl_Bow_genus)

perl_agg <- aggregate(Abundance ~ Genus + Site_code, data = perl_Bow_genus_df, FUN = mean)

#Sort abundance of each genus by mean at each site
sort_genus_perl <- perl_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_perl

endosymbionts_perl <- sort_genus_perl$Genus[1901:1905]

endo_df_perl <- perl_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts_perl))
endo_df_perl

endo_agg_df_perl<-aggregate(Abundance ~ Genus + Site_code, data = endo_df_perl, FUN = sum)

endo_agg_df_perl$Genus<-factor(endo_agg_df_perl$Genus, levels=c("Rickettsia", "Other"), labels=c("Rickettsia", "Non-endosymbionts"))

endo_plot_perl <- ggplot(endo_agg_df_perl, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=22),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title.x=element_text(size=22))+
  scale_fill_manual(name="Genus", values=c("honeydew4"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_perl

#### Adult Trichoptera Bow River ####
trichop<-subset_samples(sample_ps, Family=="Trichoptera")

#Cochrane 
trichop_coch<-subset_samples(trichop, Site=="Cochrane")

trichop_coch_glom = tax_glom(trichop_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
trichop_coch_glom<-prune_taxa(taxa_sums(trichop_coch_glom) > 0, trichop_coch_glom)

trichop_coch_genus <- psmelt(trichop_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_trichop<- trichop_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_trichop$SD_squared ='^'(coch_endo_trichop$SD,2)
coch_endo_trichop$SD_squared_divided =(coch_endo_trichop$SD_squared/8) 
  
top_means_endo_trichop_coch<-aggregate(Mean ~ Genus, data = coch_endo_trichop, FUN = sum)
top_SD_endo_trichop_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_trichop, FUN = sum)

top_endo_trichop_coch<-merge(top_means_endo_trichop_coch, top_SD_endo_trichop_coch, by=c("Genus"))
top_endo_trichop_coch$SD_Final<-sqrt(top_endo_trichop_coch$SD_squared_divided)
View(top_endo_trichop_coch)

#Sunalta 
trichop_sun<-subset_samples(trichop, Site=="Sunalta")

trichop_sun_glom = tax_glom(trichop_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
trichop_sun_glom<-prune_taxa(taxa_sums(trichop_sun_glom) > 0, trichop_sun_glom)

trichop_sun_genus <- psmelt(trichop_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_trichop<- trichop_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_trichop$SD_squared ='^'(sun_endo_trichop$SD,2)
sun_endo_trichop$SD_squared_divided =(sun_endo_trichop$SD_squared/8) 
  
top_means_endo_trichop_sun<-aggregate(Mean ~ Genus, data = sun_endo_trichop, FUN = sum)
top_SD_endo_trichop_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_trichop, FUN = sum)

top_endo_trichop_sun<-merge(top_means_endo_trichop_sun, top_SD_endo_trichop_sun, by=c("Genus"))
top_endo_trichop_sun$SD_Final<-sqrt(top_endo_trichop_sun$SD_squared_divided)
View(top_endo_trichop_sun)

#Cushing Bridge
trichop_cush<-subset_samples(trichop, Site=="Cushing Bridge")

trichop_cush_glom = tax_glom(trichop_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
trichop_cush_glom<-prune_taxa(taxa_sums(trichop_cush_glom) > 0, trichop_cush_glom)

trichop_cush_genus <- psmelt(trichop_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_trichop<- trichop_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_trichop$SD_squared ='^'(cush_endo_trichop$SD,2)
cush_endo_trichop$SD_squared_divided =(cush_endo_trichop$SD_squared/8) 
  
top_means_endo_trichop_cush<-aggregate(Mean ~ Genus, data = cush_endo_trichop, FUN = sum)
top_SD_endo_trichop_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_trichop, FUN = sum)

top_endo_trichop_cush<-merge(top_means_endo_trichop_cush, top_SD_endo_trichop_cush, by=c("Genus"))
top_endo_trichop_cush$SD_Final<-sqrt(top_endo_trichop_cush$SD_squared_divided)
View(top_endo_trichop_cush)

#Graves Bridge
trichop_grvbr<-subset_samples(trichop, Site=="Graves Bridge")

trichop_grvbr_glom = tax_glom(trichop_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
trichop_grvbr_glom<-prune_taxa(taxa_sums(trichop_grvbr_glom) > 0, trichop_grvbr_glom)

trichop_grvbr_genus <- psmelt(trichop_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_trichop<- trichop_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_trichop$SD_squared ='^'(grvbr_endo_trichop$SD,2)
grvbr_endo_trichop$SD_squared_divided =(grvbr_endo_trichop$SD_squared/7) 
  
top_means_endo_trichop_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_trichop, FUN = sum)
top_SD_endo_trichop_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_trichop, FUN = sum)

top_endo_trichop_grvbr<-merge(top_means_endo_trichop_grvbr, top_SD_endo_trichop_grvbr, by=c("Genus"))
top_endo_trichop_grvbr$SD_Final<-sqrt(top_endo_trichop_grvbr$SD_squared_divided)
View(top_endo_trichop_grvbr)

#### Adult Ephemeroptera Bow River ####

ephem_Bow<-subset_samples(sample_ps, Family=="Ephemeroptera" & Stream_Type=="Bow River")

#Cochrane 
ephem_Bow_coch<-subset_samples(ephem_Bow, Site=="Cochrane")

ephem_Bow_coch_glom = tax_glom(ephem_Bow_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
ephem_Bow_coch_glom<-prune_taxa(taxa_sums(ephem_Bow_coch_glom) > 0, ephem_Bow_coch_glom)

ephem_Bow_coch_genus <- psmelt(ephem_Bow_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_ephem_Bow<- ephem_Bow_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_ephem_Bow$SD_squared ='^'(coch_endo_ephem_Bow$SD,2)
coch_endo_ephem_Bow$SD_squared_divided =(coch_endo_ephem_Bow$SD_squared/8) 
  
top_means_endo_ephem_Bow_coch<-aggregate(Mean ~ Genus, data = coch_endo_ephem_Bow, FUN = sum)
top_SD_endo_ephem_Bow_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_ephem_Bow, FUN = sum)

top_endo_ephem_Bow_coch<-merge(top_means_endo_ephem_Bow_coch, top_SD_endo_ephem_Bow_coch, by=c("Genus"))
top_endo_ephem_Bow_coch$SD_Final<-sqrt(top_endo_ephem_Bow_coch$SD_squared_divided)
View(top_endo_ephem_Bow_coch)

#Sunalta
ephem_Bow_sun<-subset_samples(ephem_Bow, Site=="Sunalta")

ephem_Bow_sun_glom = tax_glom(ephem_Bow_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
ephem_Bow_sun_glom<-prune_taxa(taxa_sums(ephem_Bow_sun_glom) > 0, ephem_Bow_sun_glom)

ephem_Bow_sun_genus <- psmelt(ephem_Bow_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_ephem_Bow<- ephem_Bow_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_ephem_Bow$SD_squared ='^'(sun_endo_ephem_Bow$SD,2)
sun_endo_ephem_Bow$SD_squared_divided =(sun_endo_ephem_Bow$SD_squared/8) 
  
top_means_endo_ephem_Bow_sun<-aggregate(Mean ~ Genus, data = sun_endo_ephem_Bow, FUN = sum)
top_SD_endo_ephem_Bow_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_ephem_Bow, FUN = sum)

top_endo_ephem_Bow_sun<-merge(top_means_endo_ephem_Bow_sun, top_SD_endo_ephem_Bow_sun, by=c("Genus"))
top_endo_ephem_Bow_sun$SD_Final<-sqrt(top_endo_ephem_Bow_sun$SD_squared_divided)
View(top_endo_ephem_Bow_sun)

#Cushing Bridge
ephem_Bow_cush<-subset_samples(ephem_Bow, Site=="Cushing Bridge")

ephem_Bow_cush_glom = tax_glom(ephem_Bow_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
ephem_Bow_cush_glom<-prune_taxa(taxa_sums(ephem_Bow_cush_glom) > 0, ephem_Bow_cush_glom)

ephem_Bow_cush_genus <- psmelt(ephem_Bow_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_ephem_Bow<- ephem_Bow_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_ephem_Bow$SD_squared ='^'(cush_endo_ephem_Bow$SD,2)
cush_endo_ephem_Bow$SD_squared_divided =(cush_endo_ephem_Bow$SD_squared/8) 
  
top_means_endo_ephem_Bow_cush<-aggregate(Mean ~ Genus, data = cush_endo_ephem_Bow, FUN = sum)
top_SD_endo_ephem_Bow_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_ephem_Bow, FUN = sum)

top_endo_ephem_Bow_cush<-merge(top_means_endo_ephem_Bow_cush, top_SD_endo_ephem_Bow_cush, by=c("Genus"))
top_endo_ephem_Bow_cush$SD_Final<-sqrt(top_endo_ephem_Bow_cush$SD_squared_divided)
View(top_endo_ephem_Bow_cush)

#Graves Bridge
ephem_Bow_grvbr<-subset_samples(ephem_Bow, Site=="Graves Bridge")

ephem_Bow_grvbr_glom = tax_glom(ephem_Bow_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
ephem_Bow_grvbr_glom<-prune_taxa(taxa_sums(ephem_Bow_grvbr_glom) > 0, ephem_Bow_grvbr_glom)

ephem_Bow_grvbr_genus <- psmelt(ephem_Bow_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_ephem_Bow<- ephem_Bow_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_ephem_Bow$SD_squared ='^'(grvbr_endo_ephem_Bow$SD,2)
grvbr_endo_ephem_Bow$SD_squared_divided =(grvbr_endo_ephem_Bow$SD_squared/8) 
  
top_means_endo_ephem_Bow_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_ephem_Bow, FUN = sum)
top_SD_endo_ephem_Bow_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_ephem_Bow, FUN = sum)

top_endo_ephem_Bow_grvbr<-merge(top_means_endo_ephem_Bow_grvbr, top_SD_endo_ephem_Bow_grvbr, by=c("Genus"))
top_endo_ephem_Bow_grvbr$SD_Final<-sqrt(top_endo_ephem_Bow_grvbr$SD_squared_divided)
View(top_endo_ephem_Bow_grvbr)

#Policeman Flats
ephem_Bow_pmf<-subset_samples(ephem_Bow, Site=="Policeman Flats")

ephem_Bow_pmf_glom = tax_glom(ephem_Bow_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
ephem_Bow_pmf_glom<-prune_taxa(taxa_sums(ephem_Bow_pmf_glom) > 0, ephem_Bow_pmf_glom)

ephem_Bow_pmf_genus <- psmelt(ephem_Bow_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_ephem_Bow<- ephem_Bow_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_ephem_Bow$SD_squared ='^'(pmf_endo_ephem_Bow$SD,2)
pmf_endo_ephem_Bow$SD_squared_divided =(pmf_endo_ephem_Bow$SD_squared/8) 
  
top_means_endo_ephem_Bow_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_ephem_Bow, FUN = sum)
top_SD_endo_ephem_Bow_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_ephem_Bow, FUN = sum)

top_endo_ephem_Bow_pmf<-merge(top_means_endo_ephem_Bow_pmf, top_SD_endo_ephem_Bow_pmf, by=c("Genus"))
top_endo_ephem_Bow_pmf$SD_Final<-sqrt(top_endo_ephem_Bow_pmf$SD_squared_divided)
View(top_endo_ephem_Bow_pmf)

#### Adult Diptera Bow River ####

dip_Bow<-subset_samples(sample_ps, Family=="Diptera")

#Cochrane 
dip_coch<-subset_samples(dip_Bow, Site=="Cochrane")

dip_coch_glom = tax_glom(dip_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
dip_coch_glom<-prune_taxa(taxa_sums(dip_coch_glom) > 0, dip_coch_glom)

dip_coch_genus <- psmelt(dip_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_dip<- dip_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_dip$SD_squared ='^'(coch_endo_dip$SD,2)
coch_endo_dip$SD_squared_divided =(coch_endo_dip$SD_squared/5) 
  
top_means_endo_dip_coch<-aggregate(Mean ~ Genus, data = coch_endo_dip, FUN = sum)
top_SD_endo_dip_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_dip, FUN = sum)

top_endo_dip_coch<-merge(top_means_endo_dip_coch, top_SD_endo_dip_coch, by=c("Genus"))
top_endo_dip_coch$SD_Final<-sqrt(top_endo_dip_coch$SD_squared_divided)
View(top_endo_dip_coch)

#Cushing Bridge
dip_cush<-subset_samples(dip_Bow, Site=="Cushing Bridge")

dip_cush_glom = tax_glom(dip_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
dip_cush_glom<-prune_taxa(taxa_sums(dip_cush_glom) > 0, dip_cush_glom)

dip_cush_genus <- psmelt(dip_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_dip<- dip_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_dip$SD_squared ='^'(cush_endo_dip$SD,2)
cush_endo_dip$SD_squared_divided =(cush_endo_dip$SD_squared/8) 
  
top_means_endo_dip_cush<-aggregate(Mean ~ Genus, data = cush_endo_dip, FUN = sum)
top_SD_endo_dip_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_dip, FUN = sum)

top_endo_dip_cush<-merge(top_means_endo_dip_cush, top_SD_endo_dip_cush, by=c("Genus"))
top_endo_dip_cush$SD_Final<-sqrt(top_endo_dip_cush$SD_squared_divided)
View(top_endo_dip_cush)

#Policeman Flats
dip_pmf<-subset_samples(dip_Bow, Site=="Policeman Flats")

dip_pmf_glom = tax_glom(dip_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
dip_pmf_glom<-prune_taxa(taxa_sums(dip_pmf_glom) > 0, dip_pmf_glom)

dip_pmf_genus <- psmelt(dip_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_dip<- dip_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_dip$SD_squared ='^'(pmf_endo_dip$SD,2)
pmf_endo_dip$SD_squared_divided =(pmf_endo_dip$SD_squared/8) 
  
top_means_endo_dip_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_dip, FUN = sum)
top_SD_endo_dip_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_dip, FUN = sum)

top_endo_dip_pmf<-merge(top_means_endo_dip_pmf, top_SD_endo_dip_pmf, by=c("Genus"))
top_endo_dip_pmf$SD_Final<-sqrt(top_endo_dip_pmf$SD_squared_divided)
View(top_endo_dip_pmf)


#### Araneidae ####
aran<-subset_samples(sample_ps, Family=="Araneidae")

#Cochrane 
aran_coch<-subset_samples(aran, Site=="Cochrane")

aran_coch_glom = tax_glom(aran_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
aran_coch_glom<-prune_taxa(taxa_sums(aran_coch_glom) > 0, aran_coch_glom)

aran_coch_genus <- psmelt(aran_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_aran<- aran_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_aran$SD_squared ='^'(coch_endo_aran$SD,2)
coch_endo_aran$SD_squared_divided =(coch_endo_aran$SD_squared/8) 
  
top_means_endo_aran_coch<-aggregate(Mean ~ Genus, data = coch_endo_aran, FUN = sum)
top_SD_endo_aran_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_aran, FUN = sum)

top_endo_aran_coch<-merge(top_means_endo_aran_coch, top_SD_endo_aran_coch, by=c("Genus"))
top_endo_aran_coch$SD_Final<-sqrt(top_endo_aran_coch$SD_squared_divided)
View(top_endo_aran_coch)

#Sunalta
aran_sun<-subset_samples(aran, Site=="Sunalta")

aran_sun_glom = tax_glom(aran_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
aran_sun_glom<-prune_taxa(taxa_sums(aran_sun_glom) > 0, aran_sun_glom)

aran_sun_genus <- psmelt(aran_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_aran<- aran_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_aran$SD_squared ='^'(sun_endo_aran$SD,2)
sun_endo_aran$SD_squared_divided =(sun_endo_aran$SD_squared/8) 
  
top_means_endo_aran_sun<-aggregate(Mean ~ Genus, data = sun_endo_aran, FUN = sum)
top_SD_endo_aran_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_aran, FUN = sum)

top_endo_aran_sun<-merge(top_means_endo_aran_sun, top_SD_endo_aran_sun, by=c("Genus"))
top_endo_aran_sun$SD_Final<-sqrt(top_endo_aran_sun$SD_squared_divided)
View(top_endo_aran_sun)

#Cushing Bridge
aran_cush<-subset_samples(aran, Site=="Cushing Bridge")

aran_cush_glom = tax_glom(aran_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
aran_cush_glom<-prune_taxa(taxa_sums(aran_cush_glom) > 0, aran_cush_glom)

aran_cush_genus <- psmelt(aran_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_aran<- aran_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_aran$SD_squared ='^'(cush_endo_aran$SD,2)
cush_endo_aran$SD_squared_divided =(cush_endo_aran$SD_squared/7) 
  
top_means_endo_aran_cush<-aggregate(Mean ~ Genus, data = cush_endo_aran, FUN = sum)
top_SD_endo_aran_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_aran, FUN = sum)

top_endo_aran_cush<-merge(top_means_endo_aran_cush, top_SD_endo_aran_cush, by=c("Genus"))
top_endo_aran_cush$SD_Final<-sqrt(top_endo_aran_cush$SD_squared_divided)
View(top_endo_aran_cush)

#Graves Bridge
aran_grvbr<-subset_samples(aran, Site=="Graves Bridge")

aran_grvbr_glom = tax_glom(aran_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
aran_grvbr_glom<-prune_taxa(taxa_sums(aran_grvbr_glom) > 0, aran_grvbr_glom)

aran_grvbr_genus <- psmelt(aran_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_aran<- aran_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_aran$SD_squared ='^'(grvbr_endo_aran$SD,2)
grvbr_endo_aran$SD_squared_divided =(grvbr_endo_aran$SD_squared/8) 
  
top_means_endo_aran_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_aran, FUN = sum)
top_SD_endo_aran_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_aran, FUN = sum)

top_endo_aran_grvbr<-merge(top_means_endo_aran_grvbr, top_SD_endo_aran_grvbr, by=c("Genus"))
top_endo_aran_grvbr$SD_Final<-sqrt(top_endo_aran_grvbr$SD_squared_divided)
View(top_endo_aran_grvbr)

#Policeman Flats

aran_pmf<-subset_samples(aran, Site=="Policeman Flats")

aran_pmf_glom = tax_glom(aran_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
aran_pmf_glom<-prune_taxa(taxa_sums(aran_pmf_glom) > 0, aran_pmf_glom)

aran_pmf_genus <- psmelt(aran_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_aran<- aran_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_aran$SD_squared ='^'(pmf_endo_aran$SD,2)
pmf_endo_aran$SD_squared_divided =(pmf_endo_aran$SD_squared/6) 
  
top_means_endo_aran_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_aran, FUN = sum)
top_SD_endo_aran_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_aran, FUN = sum)

top_endo_aran_pmf<-merge(top_means_endo_aran_pmf, top_SD_endo_aran_pmf, by=c("Genus"))
top_endo_aran_pmf$SD_Final<-sqrt(top_endo_aran_pmf$SD_squared_divided)
View(top_endo_aran_pmf)

#### Tetragnathidae ####

tetra<-subset_samples(sample_ps, Family=="Tetragnathidae")

#Cochrane 
tetra_coch<-subset_samples(tetra, Site=="Cochrane")

tetra_coch_glom = tax_glom(tetra_coch, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
tetra_coch_glom<-prune_taxa(taxa_sums(tetra_coch_glom) > 0, tetra_coch_glom)

tetra_coch_genus <- psmelt(tetra_coch_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

coch_endo_tetra<- tetra_coch_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

coch_endo_tetra$SD_squared ='^'(coch_endo_tetra$SD,2)
coch_endo_tetra$SD_squared_divided =(coch_endo_tetra$SD_squared/8) 
  
top_means_endo_tetra_coch<-aggregate(Mean ~ Genus, data = coch_endo_tetra, FUN = sum)
top_SD_endo_tetra_coch<-aggregate(SD_squared_divided ~ Genus, data = coch_endo_tetra, FUN = sum)

top_endo_tetra_coch<-merge(top_means_endo_tetra_coch, top_SD_endo_tetra_coch, by=c("Genus"))
top_endo_tetra_coch$SD_Final<-sqrt(top_endo_tetra_coch$SD_squared_divided)
View(top_endo_tetra_coch)

#Sunalta
tetra_sun<-subset_samples(tetra, Site=="Sunalta")

tetra_sun_glom = tax_glom(tetra_sun, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
tetra_sun_glom<-prune_taxa(taxa_sums(tetra_sun_glom) > 0, tetra_sun_glom)

tetra_sun_genus <- psmelt(tetra_sun_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

sun_endo_tetra<- tetra_sun_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

sun_endo_tetra$SD_squared ='^'(sun_endo_tetra$SD,2)
sun_endo_tetra$SD_squared_divided =(sun_endo_tetra$SD_squared/8) 
  
top_means_endo_tetra_sun<-aggregate(Mean ~ Genus, data = sun_endo_tetra, FUN = sum)
top_SD_endo_tetra_sun<-aggregate(SD_squared_divided ~ Genus, data = sun_endo_tetra, FUN = sum)

top_endo_tetra_sun<-merge(top_means_endo_tetra_sun, top_SD_endo_tetra_sun, by=c("Genus"))
top_endo_tetra_sun$SD_Final<-sqrt(top_endo_tetra_sun$SD_squared_divided)
View(top_endo_tetra_sun)

#Cushing Bridge
tetra_cush<-subset_samples(tetra, Site=="Cushing Bridge")

tetra_cush_glom = tax_glom(tetra_cush, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
tetra_cush_glom<-prune_taxa(taxa_sums(tetra_cush_glom) > 0, tetra_cush_glom)

tetra_cush_genus <- psmelt(tetra_cush_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

cush_endo_tetra<- tetra_cush_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

cush_endo_tetra$SD_squared ='^'(cush_endo_tetra$SD,2)
cush_endo_tetra$SD_squared_divided =(cush_endo_tetra$SD_squared/8) 
  
top_means_endo_tetra_cush<-aggregate(Mean ~ Genus, data = cush_endo_tetra, FUN = sum)
top_SD_endo_tetra_cush<-aggregate(SD_squared_divided ~ Genus, data = cush_endo_tetra, FUN = sum)

top_endo_tetra_cush<-merge(top_means_endo_tetra_cush, top_SD_endo_tetra_cush, by=c("Genus"))
top_endo_tetra_cush$SD_Final<-sqrt(top_endo_tetra_cush$SD_squared_divided)
View(top_endo_tetra_cush)


#Graves Bridge
tetra_grvbr<-subset_samples(tetra, Site=="Graves Bridge")

tetra_grvbr_glom = tax_glom(tetra_grvbr, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
tetra_grvbr_glom<-prune_taxa(taxa_sums(tetra_grvbr_glom) > 0, tetra_grvbr_glom)

tetra_grvbr_genus <- psmelt(tetra_grvbr_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

grvbr_endo_tetra<- tetra_grvbr_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

grvbr_endo_tetra$SD_squared ='^'(grvbr_endo_tetra$SD,2)
grvbr_endo_tetra$SD_squared_divided =(grvbr_endo_tetra$SD_squared/8) 
  
top_means_endo_tetra_grvbr<-aggregate(Mean ~ Genus, data = grvbr_endo_tetra, FUN = sum)
top_SD_endo_tetra_grvbr<-aggregate(SD_squared_divided ~ Genus, data = grvbr_endo_tetra, FUN = sum)

top_endo_tetra_grvbr<-merge(top_means_endo_tetra_grvbr, top_SD_endo_tetra_grvbr, by=c("Genus"))
top_endo_tetra_grvbr$SD_Final<-sqrt(top_endo_tetra_grvbr$SD_squared_divided)
View(top_endo_tetra_grvbr)

#Policeman Flats
tetra_pmf<-subset_samples(tetra, Site=="Policeman Flats")

tetra_pmf_glom = tax_glom(tetra_pmf, "Genus") %>% transform_sample_counts(function(x) {x * 100/sum(x)}) 
tetra_pmf_glom<-prune_taxa(taxa_sums(tetra_pmf_glom) > 0, tetra_pmf_glom)

tetra_pmf_genus <- psmelt(tetra_pmf_glom) %>%  
  as_tibble %>%
    group_by(Genus) %>%
  dplyr::select(c("Abundance", "Genus")) %>%
  summarize(Mean = mean(Abundance), SD = sd(Abundance), sample_size= n()) %>% ungroup 

pmf_endo_tetra<- tetra_pmf_genus %>%
  mutate(Genus = fct_other(Genus, keep=c("Rickettsia", "Spiroplasma", "Buchnera", "Wolbachia", "Candidatus cardinium")))

pmf_endo_tetra$SD_squared ='^'(pmf_endo_tetra$SD,2)
pmf_endo_tetra$SD_squared_divided =(pmf_endo_tetra$SD_squared/8) 
  
top_means_endo_tetra_pmf<-aggregate(Mean ~ Genus, data = pmf_endo_tetra, FUN = sum)
top_SD_endo_tetra_pmf<-aggregate(SD_squared_divided ~ Genus, data = pmf_endo_tetra, FUN = sum)

top_endo_tetra_pmf<-merge(top_means_endo_tetra_pmf, top_SD_endo_tetra_pmf, by=c("Genus"))
top_endo_tetra_pmf$SD_Final<-sqrt(top_endo_tetra_pmf$SD_squared_divided)
View(top_endo_tetra_pmf)

#### Plotting all taxa faceted ####

#Bow River 

Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")

endo<-subset_taxa(sample_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Rickettsia" | Genus =="Wolbachia" | Genus=="Spiroplasma")
endo<-prune_taxa(taxa_sums(endo) > 0, endo) 
endo #31 taxa 

endo_Bow<-subset_samples(endo, Stream_Type=="Bow River")
endo_Bow<-prune_taxa(taxa_sums(endo_Bow) > 0, endo_Bow) 
sort((table(tax_table(endo_Bow)[,6]))/sum(ntaxa(endo_Bow))*100)
#Buchnera: 3.23%, Candidatus Cardinium: 3.23%, Rickettsia: 25.8%, Wolbachia: 32.3%, Spiroplasma: 35.5%
sum(sample_sums(endo_Bow))/sum(sample_sums(Bow_ps))*100
#Endosymbionts comprise 37.1% of the total microbiome in the Bow River (across all invertebrate types)

## Plotting relative abundance of endosymbionts versus non endosymbionts ##

Bow_genus <- tax_glom(Bow_ps, taxrank = "Genus")
comp_Bow_genus <- transform_sample_counts(Bow_genus, function(x) x/sum(x))

Bow_genus_df <- psmelt(comp_Bow_genus)

Bow_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = Bow_genus_df, FUN = mean) #aggregate to Phylum level, take the mean

#Sort abundance of each genus by mean at each site
sort_genus_Bow <- Bow_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_Bow

endosymbionts_Bow <- sort_genus_Bow$Genus[c(316:320, 391:395, 1901:1905, 2076:2080, 2346:2350)]
  
endo_df_Bow <- Bow_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts_Bow))

endo_agg_df_Bow<-aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = endo_df_Bow, FUN = sum)

endo_agg_df_Bow$Genus<-factor(endo_agg_df_Bow$Genus, levels=c("Other", "Buchnera", "Candidatus Cardinium", "Rickettsia", "Spiroplasma", "Wolbachia"), labels=c( "Non-endosymbionts", "Buchnera", "Candidatus Cardinium", "Rickettsia", "Spiroplasma", "Wolbachia"))

endo_agg_df_Bow$Site_code <- factor(endo_agg_df_Bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

endo_agg_df_Bow$sample_Family<-factor(endo_agg_df_Bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

endo_plot_Bow <- ggplot(endo_agg_df_Bow, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.8, "cm"),
        legend.text=element_text(size=28, face="italic"), legend.title=element_text(size=28), legend.position = "bottom") +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=30),
        axis.text.y=element_text(size=24, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=24, color="black"),
                        axis.title.x=element_text(size=30))+
  scale_fill_manual(name="Genus", values=c("honeydew4", "tan1","seagreen1", "skyblue1", "lightgoldenrod1","lightpink1"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_Bow


#ggsave(endo_plot_Bow, filename="Microbiome_ED/Final analysis/Relative abundance/Endosymbionts/Endosymbiont composition across sites and taxa Bow only.png", height=12, width=22)


#ACWA Streams

ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")

endo<-subset_taxa(sample_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Rickettsia" | Genus =="Wolbachia" | Genus=="Spiroplasma")
endo<-prune_taxa(taxa_sums(endo) > 0, endo) 
endo #31 taxa 

endo_ACWA<-subset_samples(endo, Stream_Type!="Bow River")
endo_ACWA<-prune_taxa(taxa_sums(endo_ACWA) > 0, endo_ACWA) 
sort((table(tax_table(endo_ACWA)[,6]))/sum(ntaxa(endo_ACWA))*100)
#Rickettsia: 50, Wolbachia: 50
sum(sample_sums(endo_ACWA))/sum(sample_sums(ACWA_ps))*100
#Endosymbionts comprise 0.009% of the total microbiome within ACWA streams (across all invertebrate types)

## Plotting relative abundance of endosymbionts versus non endosymbionts ##
ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")

ACWA_genus <- tax_glom(ACWA_ps, taxrank = "Genus")
comp_ACWA_genus <- transform_sample_counts(ACWA_genus, function(x) x/sum(x))

ACWA_genus_df <- psmelt(comp_ACWA_genus)

ACWA_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = ACWA_genus_df, FUN = mean) #aggregate to Phylum level, take the mean

#Sort abundance of each genus by mean at each site
sort_genus_ACWA <- ACWA_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_ACWA

endosymbionts_ACWA <- sort_genus_ACWA$Genus[c(1141:1143, 1408:1410)]
#Rickettsia              #Wolbachia 
#BRR2: 0.0000345 %       #BRR2: 0 %
#PCR1: 0.0000680 %       #PCR1: 0.0000547 %
#PCR3: 0.000258 %        #PCR3: 0.0000229 %
  
endo_df_ACWA <- ACWA_agg %>%
    mutate(Genus = fct_other(Genus, endosymbionts_ACWA))

endo_agg_df_ACWA<-aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = endo_df_ACWA, FUN = sum)

endo_agg_df_ACWA$Genus<-factor(endo_agg_df_ACWA$Genus, levels=c("Other", "Rickettsia", "Wolbachia"), labels=c("Non-endosymbionts", "Rickettsia", "Wolbachia"))

endo_agg_df_ACWA$Site_code <- factor(endo_agg_df_ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

endo_agg_df_ACWA$sample_Family<-factor(endo_agg_df_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

endo_plot_ACWA <- ggplot(endo_agg_df_ACWA, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.8, "cm"),
        legend.text=element_text(size=28, face="italic"), legend.title=element_text(size=28), legend.position = "bottom") +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=30),
        axis.text.y=element_text(size=24, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=24, color="black"),
                        axis.title.x=element_text(size=30))+
  scale_fill_manual(name="Genus", values=c("honeydew4", "skyblue1","lightpink1"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
endo_plot_ACWA


#ggsave(endo_plot_ACWA, filename="Microbiome_ED/Final analysis/Relative abundance/Endosymbionts/Endosymbiont composition across sites and taxa ACWA only.png", height=12, width=22)

```

### Graphing Endosymbiont relative abundance across sites and taxa
```{r}

## Filtering for endosymbiont bacteria ##
tax_select(sample_ps, tax_list="Buchnera") #1 taxa.
tax_select(sample_ps, tax_list="Candidatus Cardinium") #1 taxa
tax_select(sample_ps, tax_list="Rickettsia") #58 taxa. 
tax_select(sample_ps, tax_list="Rickettsiella") #No taxa matched
tax_select(sample_ps, tax_list="Wolbachia") #10 taxa. 
tax_select(sample_ps, tax_list="Spiroplasma") #11 taxa 
tax_select(sample_ps, tax_list="Hamiltonella") #none matched
tax_select(sample_ps, tax_list="Arsenophonus") #none matched

endo<-subset_taxa(sample_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Rickettsia" | Genus =="Wolbachia" | Genus=="Spiroplasma")
endo<-prune_taxa(taxa_sums(endo) > 0, endo) 
endo #31 taxa 

#### Bow River ####

endo_Bow<-subset_taxa(Bow_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Rickettsia" | Genus =="Wolbachia" | Genus=="Spiroplasma")
endo_Bow<-prune_taxa(taxa_sums(endo_Bow) > 0, endo_Bow) 
endo_Bow #31 taxa, 280 samples

## All samples plotted separately faceted by invert taxa and site ##

endo_agglom_Bow_all <- speedyseq::tax_glom(endo_Bow, taxrank = "Genus")
endo_rel_abun_Bow_all<- transform_sample_counts(endo_agglom_Bow_all, function(x) x/sum(x))
endo_df_Bow_all <- psmelt(endo_rel_abun_Bow_all)
sample.df.agg.endo_phy_Bow_all <- aggregate(Abundance ~ Genus + Site_code + sample_Family+Sample_ID, data = endo_df_Bow_all, FUN = mean)

#Faceted by site and invert family. Hard to see because there's not much going on.
ggplot(sample.df.agg.endo_phy_Bow_all, aes(x=Sample_ID , y=Abundance, fill=Genus, label=Sample_ID)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean relative abundance of endosymbionts") +
  facet_nested_wrap(~Site_code + sample_Family, scales = "free", nrow=4, ncol=8)+
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=8)) +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=8),
        axis.text.y=element_text(size=8),
        axis.text.x=element_blank(),
        #axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Genus", values=c("tan1","seagreen1", "skyblue1", "lightgoldenrod1","lightpink1"))+
  rremove("grid")

## Faceted by average relative abundance per invert taxa and site ##

sample.df.agg.endo_phy_Bow_site <- aggregate(Abundance ~ Genus + Site_code + Stage + sample_Family, data = endo_df_Bow_all, FUN = mean)

sample.df.agg.endo_phy_Bow_site$sample_Family<-factor(sample.df.agg.endo_phy_Bow_site$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

ggplot(sample.df.agg.endo_phy_Bow_site, aes(x=Site_code , y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  facet_nested_wrap(~Stage + sample_Family, scales = "free_x", nrow=1)+
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=8)) +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_blank(),
        axis.text.y=element_text(size=8, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Genus", values=c("tan1","seagreen1", "skyblue1", "lightgoldenrod1","lightpink1"))+
  xlab("Site")



#ggsave(filename="Microbiome_ED/Final analysis/Relative abundance/Endosymbionts/Bow River and ACWA stream endosymbionts.png", height=6, width=5, bg="white")


#### ACWA ####

endo_ACWA<-subset_taxa(ACWA_ps, Genus=="Buchnera" | Genus=="Candidatus Cardinium" | Genus=="Hamiltonella" | Genus=="Rickettsia" | Genus=="Rickettsiella" | Genus =="Wolbachia" | Genus=="Spiroplasma" | Genus=="Arsenophonus"| Genus=="Diplorickettsia")

endo_ACWA<-prune_taxa(taxa_sums(endo_ACWA) > 0, endo_ACWA) 
endo_ACWA #4 taxa, 55 samples

endo_agglom_ACWA <- speedyseq::tax_glom(endo_ACWA, taxrank = "Genus")
endo_rel_abun_ACWA<- transform_sample_counts(endo_agglom_ACWA, function(x) x/sum(x))
endo_df_ACWA <- psmelt(endo_rel_abun_ACWA)
sample.df.agg.endo_phy_ACWA <- aggregate(Abundance ~ Genus + Site_code + Stage + sample_Family, data = endo_df_ACWA, FUN = mean)

sample.df.agg.endo_phy_ACWA$sample_Family<-factor(sample.df.agg.endo_phy_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

sample.df.agg.endo_phy_ACWA$Site_code<-factor(sample.df.agg.endo_phy_ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EFF5", "EFF15"))

#Endosymbiont genera across sites and taxa
ggplot(sample.df.agg.endo_phy_ACWA, aes(x=Site_code , y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  xlab("ACWA Stream")+
  theme_bw()+
  facet_nested_wrap(~Stage + sample_Family, scales="free_x")+
  #facet_wrap(vars(Stage, Site_code), scales = "free")+
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=8)) +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_blank(),
        axis.text.y=element_text(size=8, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=14))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Genus", values = c("skyblue1", "lightpink1"))+
  rremove("grid")


#ACWA streams

endo_ACWA<-subset_samples(endo, Stream_Type!="Bow River")
endo_ACWA<-prune_taxa(taxa_sums(endo_ACWA)>0, endo_ACWA)

endo_agglom_ACWA <- speedyseq::tax_glom(endo_ACWA, taxrank = "Genus")
endo_rel_abun_ACWA<- transform_sample_counts(endo_agglom_ACWA, function(x) x/sum(x))
endo_df_ACWA <- psmelt(endo_rel_abun_ACWA)
sample.df.agg.endo_phy_ACWA <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Sample_ID, data = endo_df_ACWA, FUN = mean)

sample.df.agg.endo_phy_ACWA$sample_Family<-factor(sample.df.agg.endo_phy_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

sample.df.agg.endo_phy_ACWA$Site_code<-factor(sample.df.agg.endo_phy_ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EFF5", "EFF15"))

#Endosymbiont genera
ggplot(sample.df.agg.endo_phy_ACWA, aes(x=Sample_ID , y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  facet_nested_wrap(~Site_code + sample_Family, scales = "free")+
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=8), axis.text.x = element_blank()) +
  guides(fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=8),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Genus", values = c("skyblue1", "lightpink1"))+
  rremove("grid")

```

### Statistically testing the difference in endosymbiont relative abundance
```{r}
#First transform the endosymbiont phyloseq data to a dataframe as we did with all the rel. abund. data

#Metadata

sample_data_endo<-as.data.frame(sample_data(endo))
row_names_endo<-row.names(sample_data_endo) 
sample_data_endo_1<-cbind(row_names_endo, sample_data_endo)
sample_data_endo_1<-sample_data_endo_1 %>% dplyr::rename(Study_ID = row_names_endo)
sample_data_endo_2<-sample_data_endo_1[ , c("Study_ID",  "Site", "Family", "Stream_Type")]
sample_data_endo_3<-sample_data_endo_2 %>% dplyr::rename(sample_Family = Family)
sample_data_endo_4<-sample_data_endo_3 %>% drop_na(Site, sample_Family)


#ASV table from filtered phyloseq
ASVseq_endo<- as.data.frame(otu_table(endo))
Sequences<-row.names(ASVseq_endo) 
ASVseq_endo_1<-cbind(Sequences, ASVseq_endo)
 
ASVseq_endo_2<-ASVseq_endo_1 %>% pivot_longer(-Sequences, names_to="Study_ID", values_to = "count")

#Taxonomy table
taxonomy_endo<-as.data.frame(tax_table(endo)) %>% rownames_to_column(var="Sequences")

nseqs_per_sample_endo <- ASVseq_endo_2 %>%
  dplyr::group_by(Study_ID) %>%
  dplyr::summarize(N = sum(count), .groups="drop") %>% 
  dplyr::count(N) %>%
  dplyr::pull(N)

lod_endo <- 100* 1/max(nseqs_per_sample_endo) #Determines the limit of detection (0.000449)

metadata_endo_Bow<-subset(sample_data_endo_4, Stream_Type=="Bow River")

otu_rel_abund_endo_Bow <- dplyr::inner_join(ASVseq_endo_2, metadata_endo_Bow, by="Study_ID") %>%
  dplyr::inner_join(., taxonomy_endo, by="Sequences") %>% #joins taxonomy, abundances, and metadata
  dplyr::group_by(Study_ID) %>%
  dplyr::mutate(rel_abund = count / sum(count)) %>% #transforms to relative abundance
  dplyr::ungroup() %>%
  dplyr::select(-count) %>%
  pivot_longer(
    c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "Sequences"),
    names_to="level",
    values_to="taxon") 

#Genus
taxon_rel_abund_Genus_endo_Bow <- otu_rel_abund_endo_Bow %>%
  dplyr::filter(level=="Genus") %>%
  dplyr::group_by(Site, sample_Family, Study_ID, taxon) %>%
  dplyr::summarize(rel_abund = 100*sum(rel_abund), .groups="drop") %>%
  drop_na(Site, sample_Family, taxon)

taxon_pool_Genus_endo_Bow <- taxon_rel_abund_Genus_endo_Bow %>%
  dplyr::group_by(Site, sample_Family, taxon) %>%
  dplyr::summarize(median=median(rel_abund), .groups="drop") %>%
  dplyr::group_by(taxon) %>%
  dplyr::summarize(pool = max(median) < 0,
            median = median(median),
            .groups="drop")

#Graphing pointrange plot with median and 50% quartiles
Genus_endo_Bow_graphing_df<- dplyr::inner_join(taxon_rel_abund_Genus_endo_Bow, taxon_pool_Genus_endo_Bow, by="taxon") %>%
 dplyr::mutate(taxon = if_else(pool, "Other", taxon)) %>%
  dplyr::group_by(Study_ID, Site, sample_Family, taxon) %>%
  dplyr::summarize(rel_abund = sum(rel_abund),
            median = min(median),
            .groups="drop") %>%
  dplyr::mutate(taxon = factor(taxon),
         taxon = fct_reorder(taxon, median, .desc=T)) %>%
  dplyr::mutate(rel_abund = if_else(rel_abund == 0,
                             2/3 * lod_endo,
                             rel_abund))
Genus_endo_Bow_graphing_df



```

# Wastewater associated bacteria 

```{r}
#### Filtering for wastewater effluent-associated bacteria and potential pathogens ####

#First aggregate by genus and see how many we have. 
genus_all <- tax_glom(sample_ps, taxrank = "Genus")
genus_all #493 genera

rel_abun_genus_all <- transform_sample_counts(genus_all, function(x) x/sum(x))

genus_df_all <- psmelt(rel_abun_genus_all)

agg_all <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = genus_df_all, FUN = mean) #aggregate to Phylum level, take the mean

#Sort abundance of each genus by mean at each site
sort_genus_all <- agg_all %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_all

# Researched all genera on Google Scholar to see if they were wastewater associated #
# Achromobacter - Jin et al., (2015). Int. J. Environ. Res. Public Health. Kundu et al., (2011). Bioprocess and Biosystems Engineering. 
# Acidovorax - Wang et al., (2009). Water Science and Technology. Heylen et al., (2008). Int. J. Sys. Evol. Biology. 
# Acinetobacter - Higgins et al., (2018). Water Research. Liu et al., (2017). Bioresource Technology. 
# Aeromonas - Harnisz and Korzeniewska, (2018). Sci. Total Environ. Figueira et al., (2011). Water Research. 
# Afipia - Zhang et al., (2009). FEMS Microbiology. Kim et al., (2014). J. Microbiol. Biotechnol.
# Ahniella - Tang et al., (2023). Environ. Research. 
# Alcaligenes (faecalis or aquatilis) - Joo et al., (2006). Water Research. Cao et al., (2022). Bioresource Technol. 
# Amaricoccus - Maszenan et al., (2000). J Appl. Microbiology. Maszenan et al., (1997). INT. J. SYST. BACTERIology.
#Anaerocella - Yan et al., (2021). Environ. Science and Pollution Research. Abe et al., (2012). J Gen. Appl. Microbiol.
#Anaerovorax - Zhu et al., (2018). Biochem Engineering J. Chu et al., (2021). Environ. Science and Pollut. research. 
#	Arenimonas - Liu et al., (2018). Int J Microbiol. Research. Feng et al., (2022). J Water Process eng.
#Arthrobacter - Roh et al., (2008). J Microbiol. Agunbiade et al., (2017). BMC Biotechn. 
#Bacillus - Ertugrul et al., (2007). J Hazardous Mater. Deng et al., (2003). App. Microbiol. and biotech.
#Bacteroides - Srinivasan et al., (2011). Water Research. Niestpski et al., (2020). J Hazardous Mat. 
# Barnesiella - Morotomi et al., (2008). Int J Microbol. Research.
#Blastocatella - Ouyang et al., (2017). Environ. Technol. Huang et al., (2017). Bioresource Technol.
#	Blautia - Cai & Zhang (2014). Appl. Microbiol. and biotechnol.
#Bordetella - Sanuth et al., (2013). Indian J Microbiol. Awad et al., (2022). Sci total environ.
#Bosea - Lee et al., (2017). Kor. J. Fisheries. Aquat. Sci.
#Brevundimonas - Ryu et al., (2007). Int. J. Sys. Evol. Biology. Sforza et al., (2018). Bioresource Technol. 
#Candidatus Accumulibacter - He & McMahon, 2011. Microbial biotechnology. 
#Candidatus Competibacter - Rubio-Rincon et al., (2017). Water Research.
#Candidatus Contendobacter - Zhou et al., (2023). Int. Biodeterioration and biodegredation.
#Candidatus Microthrix - Blackall et al., (1996). Int J Sys Evol Microbiology
#Candidatus Nitrotoga - Zheng et al., (2021). Environ Sci Tech
#Cetobacterium - Wang et al., 2022. Ecotox.  Environ. safety
#Chitinivorax - Begmatov et al., (2022). Scientific reports. 
#Chryseobacterium - Kundu et al., (2014). Biomed research. Kamper et al., (2003). Int. J Sys. Evol Microbiology
#Chthoniobacter - Feng et al., (2020). Bioresource technol.
#Citrobacter - Owoseni and Okoh, (2017). Environ monitroing assessment. 
#	Cloacibacterium - Allen et al., (2006). Int J Sys Evol Microbiology
#Clostridium sensu stricto 1 - Wang et al., (2003). J Biotechnol. 
#	Clostridium sensu stricto 13 - Yin et al., (2021). Int J Hydro Energy
#Comamonas - Gumaelius et al., (2001). Int J Sys Evol biology
#Corynebacterium - Liu et al., (2023). Front. Bioengineer. biotechnol. 
#Curtobacterium - Li et al., (2021). Sci total environ.
#Curvibacter - Zielinska et al., 2016. Environ. technol.
#	Cytophaga - Manz et al., 1994. Water research.
#Dechloromonas - Mcllroy et al., (2016). Environ. Microbiology
#Defluviicoccus - Maszenan et al., (2022). Water Research. Burow et al., (2007). Microbiology research.
#Delftia - Shigematsu et al., (2003). Int. J Sys. Evol. Microbiology
#Desulfatirhabdium - Balk et al., (2008). Int. J Sys. Evol. Micro
#Desulfobacca - oude Elferink et al., (1999). Int. J Sys. Evol. Micro
#Desulfobulbus - Houari et al., (2017). Int. J Sys. Evol. Micro
#Desulfoprunum - Junghare et al., (2015). Int. J Sys. Evol. Micro
#Desulfosporosinus - Houari et al., (2020). Processes
#Ellin6067 - Feng et al., (2022). J Water Process engineering
#Enterococcus - Tauer-Kapteijn et al., (2017). Delft University of Technology.
#Erysipelothrix - Ma et al., (2023). J. Environ. Manage.
#	Escherichia-Shigella - Yang et al., (2021). J Cleaner Product.
#Faecalibacterium - Wery et al., (2010). Water research.
#Ferruginibacter - Wijaya et al., (2023). Environ. Research. 
#Flavobacterium sasangense - Yoon et al., (2009).  Int. J Sys. Evol. Micro.
#Flavobacterium anseonense - Chen et al., (2019). Water.
#Fusicatenibacter - Takada et al., (2013). Int. J. Sys. Evol. Micro.
#Haliangium - Mcllroy et al., (2016). Environ. Micro.
#Haliscomenobacter - Van Veen et al., (1973). Antonie van Leeuwenhoek
#Halomonas - Li et al., (2023). Environ. Sci. Pollut.
#Hydrogenophaga - Kampfer et al., (2005). Int. J. Sys. Evol. Micro
#Hyphomicrobium - Leyton et al., (2000). Appl. Environ. Microbiology
#Ideonella - Kraigher et al., (2008). Water Research
#Inhella - Zeng et al., (2019). Environ. Engin. research
#Intestinibacter - Major et al., (2022). Waste Management. 
#Lachnoclostridium - Liang et al., (2020). Gut. 
#Lacunisphaera - Wang et al., (2020). Water Research.
#Leptothrix - Hirota et al., (2016). J Environ. Sci.
#Longivirga - Ren et al., (2024). J Environ. Sci.
#Luteimonas - Cydzik-Kwiatkowska & Zielinska, (2017). J Environ. Sci and Health
#Massilia - Zhang et al., (2016). Bioresource technol
#Methylobacillus - Bassin et al., (2017). Int biodeterioration biodegredation
#Methylocystis - Siniscalchi et al., (2014). Environ. Technol
#Methylotenera - Zhang et al., (2017). Scientific reports
#Micrococcus - Zheng et al., (2009). J Hazardous Mat.
#Monoglobus - Kim et al., (2017). Int J Sys Evol Micro
#Moraxella - Batinovic et al., (2020). Microbiology Resource. Qu et al., (2022). Bioprocess Biosystems
#Mycobacterium abscessus - To et al., (2020). J Clinical Med.
#Nitratireductor - Zhang et al., (2023). J Hazardous Mat
#Nitrosomonas - Silyn-Roberts et al., (2001). Water Research
#Nitrospira - Daims et al., (2001). Appl. Environ. Micro.
#Nocardioides - Ma et al., (2023). Molecules.
#Novosphingobium - Hashimotot et al., (2010). J Environ. Eng
#OLB12 - Hei et al., (2022). J Cleaner Prod.
#OLB8 - Aqueel et al., (2020). Front. Micro.
#Opitutus - Meng et al., (2015). Chem Eng J.
#P3OB-42 - Chen et al., (2022). Bioresource technol
#Paludibacter - Liang et al., (2013). Chem eng J
#Paraclostridium - Fang et al., (2021). Water Research
#Paracoccus - Zheng et al., (2010). Int J Sys Evol Micro
#Pelomonas - Trebuch et al., (2020). Water Research
#Phaeodactylibacter - Wang et al., (2022). Sci Total Env
#Phenylobacterium - Yang et al., (2022). Proc. Safety Environ. Protec.
#Planococcus - Chanwala et al., (2019). Environ. Technol. Innova.
#Polaromonas - Jankowski et al., (2022). Environ. Micro
#Polyangium - Liu et al., (2021). Bioresource technol
#Propionivibrio - Wu et al., (2023). Water Research.
#Pseudarcobacter - Aoki et al., (2023). Plos One.
#Pseudomonas - Yong et al., (2015). Chemosphere.
#Pseudorhodobacter - Bian et al., (2023). J Hazardous Mat
#Pseudoxanthomonas - Meng et al., (2015). Bioresource Tecnol.
#Ralstonia - Gan et al., (2011). Chemosphere
#Rhizorhapis - Carbreros et al., (2023). Sci. total envir.
#Rhodobacter - Hiraishi et al., (1995). J ferment. bioeng.
#Rhodoferax - Mcllroy et al., (2014). Environ. Micro
#Roseomonas - Xin et al., (2008). Earth Sci Front.
#Rubellimicrobium - Xing et al., (2018). Environ. Sci Technol
#Rubrivivax - Ponsano et al., (2008). Bioresource technol
#Ruminococcus - Mansfeldt et al., (2020). Sci Total Environ
#Runella - Ryu et al., (2006). J Sys Evol Micro
#Saprospira - Chen et al., (2016). Scientific reports.
#Sediminibacterium - Lee et al., (2013). Micro Technol
#Serratia - Chojinak et al., (2015). Acta Biochem.
#Sphingobium - Qin et al., (2020). Int J Sys Evol Micro
#Staphylococcus - Zielinski et al., (2020). Environ. Int.
#Stenotrophomonas - Wierzba, (2015). Polish J CHem Technol
#Sulfurospirillum - Shi et al., (2022). Environ. Poll.
#Syntrophobacter - Kong et al., (2022). Sci total environ
#Trichococcus - Wang et al., (2016). Environ. Micro
#Youngiibacter - Shi et al., (2023). Fuel
#Zoogloea - Lajoie, et al., (2000). Water environ. research

effluent_bac<-subset_taxa(sample_ps, Genus=="Achromobacter" | Genus=="Acidovorax" | Genus=="Acinetobacter"| Genus=="Aeromonas" | Genus=="Ahniella" | Genus=="Alcaligenes" | Genus=="Amaricoccus" | Genus=="Anaerocella" | Genus=="Anaerovorax" | Genus=="Arenimonas" | Genus=="Arthrobacter" | Genus=="Bacillus" | Genus == "Barnesiella" | Genus=="Blastocatella" | Genus=="Blautia" | Genus=="Bordetella" | Genus=="Bosea" | Genus=="Brevundimonas" | Genus=="Candidatus Accumulibacter" | Genus=="Candidatus Competibacter" | Genus=="Candidatus Contendobacter" | Genus=="Candidatus Microthrix" | Genus=="Candidatus Nitrotoga" | Genus=="Cetobacterium" | Genus =="Chitinivorax" | Genus=="Chryseobacterium" | Genus=="Chthoniobacter" | Genus=="Citrobacter" | Genus=="Cloacibacterium" | Genus=="Clostridium sensu stricto 1" | Genus=="Clostridium sensu stricto 13" | Genus=="Clostridium sensu stricto 5" | Genus=="Corynebacterium" | Genus=="Curtobacterium" | Genus=="Curvibacter" | Genus=="Cytophaga" | Genus=="Dechloromonas" | Genus=="Defluviicoccus" | Genus=="Delftia" | Genus=="Desulfatirhabdium" | Genus=="Desulfobacca" | Genus=="Desulfobulbus" | Genus=="Desulfoprunum" | Genus=="Desulfosporosinus" | Genus=="Ellin6067" | Genus=="Enterococcus" | Genus=="Erysipelothrix" | Genus=="Escherichia-Shigella" | Genus=="Faecalibacterium" | Genus=="Ferruginibacter" | Species=="sasangense" | Species=="anseonense" | Genus=="Fusicatenibacter" | Genus=="Haliangium" | Genus=="Haliscomenobacter" | Genus=="Halomonas" | Genus=="Hydrogenophaga" | Genus=="Hyphomicrobium" | Genus=="Ideonella" | Genus=="Inhella" | Genus=="Intestinibacter" | Genus=="Lachnoclostridium" | Genus=="Lacunisphaera" | Genus=="Leptothrix" | Genus=="Longivirga" | Genus=="Luteimonas" | Genus=="Massilia" | Genus=="Methylobacillus" | Genus=="Methylocystis" | Genus=="Methylotenera" | Genus=="Micrococcus" | Genus=="Monoglobus" | Genus=="Moraxella" | Genus=="Mycobacterium" | Genus=="Nitratireductor" | Genus=="Nitrosomonas" | Genus=="Nitrospira" | Genus=="Nocardioides" | Genus=="Novosphingobium" | Genus=="OLB12" | Genus=="OLB8" | Genus=="Opitutus" | Genus=="P3OB-42" | Genus=="Paludibacter" | Genus=="Paraclostridium" | Genus=="Paracoccus" | Genus=="Pelomonas" | Genus=="Phaeodactylibacter" | Genus=="Phenylobacterium" | Genus=="Planococcus" | Genus=="Polaromonas" | Genus=="Polyangium" | Genus=="Propionivibrio" | Genus=="Pseudarcobacter" | Genus=="Pseudomonas" | Genus=="Pseudorhodobacter" | Genus=="Pseudoxanthomonas" | Genus=="Ralstonia" | Genus=="Rhizorhapis" | Genus=="Rhodobacter" | Genus=="Rhodoferax" | Genus=="Roseomonas" | Genus=="Rubellimicrobium" | Genus=="Rubrivivax" | Genus=="Ruminococcus" | Genus=="Runella" | Genus=="Saprospira" | Genus=="Sediminibacterium" | Genus=="Serratia" | Genus=="Sphingobium" | Genus=="Staphylococcus" | Genus=="Stenotrophomonas" | Genus=="Sulfurospirillum" | Genus=="Syntrophobacter" | Genus=="Trichococcus" | Genus=="Thermomonas" | Genus=="Youngiibacter" | Genus=="Zoogloea")
effluent_bac<-prune_samples(sample_sums(effluent_bac) > 0, effluent_bac) 
effluent_bac #974 taxa 326 samples


#Cochrane
effluent_coch<-subset_samples(effluent_bac, Site=="Cochrane")
effluent_coch<-prune_taxa(taxa_sums(effluent_coch) > 0, effluent_coch) 
effluent_coch #567 taxa, 64 samples /67 = 95.5%

sort((table(tax_table(effluent_coch)[,6]))/ntaxa(effluent_coch))*100
#Rhodoferax (12.9%), Pseudorhodobacter (6.17%), Ferruginibacter (5.11%)

#Hydropsychidae larvae
nsamples(hydro_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Hydropsychidae")) #8 samples

eff_hydro_coch<-subset_samples(effluent_coch, Family=="Hydropsychidae")
eff_hydro_coch<-prune_taxa(taxa_sums(eff_hydro_coch) > 0, eff_hydro_coch) 
eff_hydro_coch #123 taxa, 8/8 samples = 100% prevalence

sample_sums(eff_hydro_coch) #Present in all 8 samples (100% prevalence)

sort((table(tax_table(eff_hydro_coch)[,6]))/ntaxa(eff_hydro_coch))*100
#Rhodoferax (21.1%), Pseudorhodobacter (13.0%), Hyphomicrobium (11.4%)

#Heptageniidae larvae
nsamples(hep_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Heptageniidae")) #8

eff_hep_coch<-subset_samples(effluent_coch, Family=="Heptageniidae")
eff_hep_coch<-prune_taxa(taxa_sums(eff_hep_coch) > 0, eff_hep_coch) 
eff_hep_coch #277 taxa, 8/8 samples = 100% prevalence

sample_sums(eff_hep_coch) #Present in all 8 samples (100% prevalence)

sort((table(tax_table(eff_hep_coch)[,6]))/ntaxa(eff_hep_coch))*100
#Rhodoferax (18.8%), Pseudorhodobacter (6.86%), Ferruginibacter (6.14%)

#Chironomidae larvae 
nsamples(chiro_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Chironomidae")) #5

eff_chiro_coch<-subset_samples(effluent_coch, Family=="Chironomidae")
eff_chiro_coch<-prune_taxa(taxa_sums(eff_chiro_coch) > 0, eff_chiro_coch) 
eff_chiro_coch #90 taxa, 5/5 samples = 100% prevalence

sample_sums(eff_chiro_coch) #Present in all 5 samples (100% prevalence)

sort((table(tax_table(eff_chiro_coch)[,6]))/ntaxa(eff_chiro_coch))*100
#Rhodoferax (11.1%), Ferruginibacter (10.0%), Roseomonas (5.56%)

#Perlidae larvae 
nsamples(hydro_perl<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Perlidae")) #8

eff_perl_coch<-subset_samples(effluent_coch, Family=="Perlidae")
eff_perl_coch<-prune_taxa(taxa_sums(eff_perl_coch) > 0, eff_perl_coch) 
eff_perl_coch #150 taxa, 8 samples

sample_sums(eff_perl_coch) #Present in all 8 samples (100% prevalence)

sort((table(tax_table(eff_perl_coch)[,6]))/ntaxa(eff_perl_coch))*100
#Rhodoferax (15.3%), Pseudorhodobacter (12.7%), Methylotenera (12.0%)

#Trichoptera
nsamples(trichop_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Trichoptera")) #8

eff_trichop_coch<-subset_samples(effluent_coch, Family=="Trichoptera")
eff_trichop_coch<-prune_taxa(taxa_sums(eff_trichop_coch) > 0, eff_trichop_coch) 
eff_trichop_coch #169 taxa, 7 samples = 87.5 %

sample_sums(eff_trichop_coch) #Present in all 7 samples 

sort((table(tax_table(eff_trichop_coch)[,6]))/ntaxa(eff_trichop_coch))*100
#Rhodoferax (13.0%), Massilia (11.8%), Brevundimonas (7.69%)

#Ephemeroptera 
nsamples(ephem_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Ephemeroptera")) #8

eff_ephem_coch<-subset_samples(effluent_coch, Family=="Ephemeroptera")
eff_ephem_coch<-prune_taxa(taxa_sums(eff_ephem_coch) > 0, eff_ephem_coch) 
eff_ephem_coch #74 taxa, 8 samples

sample_sums(eff_ephem_coch) #Present in all 8 samples (100% prevalence)

sort((table(tax_table(eff_ephem_coch)[,6]))/ntaxa(eff_ephem_coch))*100
#Staphylococcus (12.2%), Corynebacterium (8.11%), Brevundimonas (8.11%)

#Diptera
nsamples(dip_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Diptera")) #8

eff_dip_coch<-subset_samples(effluent_coch, Family=="Diptera")
eff_dip_coch<-prune_taxa(taxa_sums(eff_dip_coch) > 0, eff_dip_coch) 
eff_dip_coch #60 taxa, 6 samples

sample_sums(eff_dip_coch) #Present in all 6 samples (75% prevalence)

sort((table(tax_table(eff_dip_coch)[,6]))/ntaxa(eff_dip_coch))*100
#Blautia (13.3%), Staphylococcus (8.33%), Faecalibacterium (8.33%)

#Araneidae
nsamples(aran_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Araneidae")) #8

eff_aran_coch<-subset_samples(effluent_coch, Family=="Araneidae")
eff_aran_coch<-prune_taxa(taxa_sums(eff_aran_coch) > 0, eff_aran_coch) 
eff_aran_coch #69 taxa, 8 samples

sample_sums(eff_aran_coch) #Present in all 8 samples (100% prevalence)

sort((table(tax_table(eff_aran_coch)[,6]))/ntaxa(eff_aran_coch))*100
#Blautia (14.5%), Faecalibacterium (8.70%), Staphylococcus (7.25%)


#Tetragnathidae
nsamples(tetra_coch<-subset_samples(sample_ps, Site=="Cochrane" & Family=="Tetragnathidae")) #8

eff_tetra_coch<-subset_samples(effluent_coch, Family=="Tetragnathidae")
eff_tetra_coch<-prune_taxa(taxa_sums(eff_tetra_coch) > 0, eff_tetra_coch) 
eff_tetra_coch #53 taxa, 6 samples

sample_sums(eff_tetra_coch) #Present in 6 samples = 75% prevalence

sort((table(tax_table(eff_tetra_coch)[,6]))/ntaxa(eff_tetra_coch))*100
#Massilia (9.43%), Blautia (9.43%%), Staphylococcus (5.66%)


## Sunalta ## 

effluent_sun<-subset_samples(effluent_bac, Site=="Sunalta")
effluent_sun<-prune_taxa(taxa_sums(effluent_sun) > 0, effluent_sun) 
effluent_sun #522 taxa, 44 samples/47 = 93.6 %

sort((table(tax_table(effluent_sun)[,6]))/sum(ntaxa(effluent_sun))*100)
#Rhodoferax (12.6%), Massilia (6.32%), Pseudorhodobacter (6.13%)

#Hydropsychidae larvae

nsamples(hydro_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Hydropsychidae")) #7

eff_hydro_sun<-subset_samples(effluent_sun, Family=="Hydropsychidae")
eff_hydro_sun<-prune_taxa(taxa_sums(eff_hydro_sun) > 0, eff_hydro_sun) 
eff_hydro_sun #124 taxa, 7 samples = 100% prevalence

sample_sums(eff_hydro_sun) #Present in all 7 samples (100% prevalence)

sort((table(tax_table(eff_hydro_sun)[,6]))/ntaxa(eff_hydro_sun))*100
#Rhodoferax (21.8%), Pseudorhodobacter (14.5%), Rhizorhapis (9.68%)

#Heptageniidae larvae
nsamples(hep_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Heptageniidae")) #8


eff_hep_sun<-subset_samples(effluent_sun, Family=="Heptageniidae")
eff_hep_sun<-prune_taxa(taxa_sums(eff_hep_sun) > 0, eff_hep_sun) 
eff_hep_sun #333 taxa, 8 samples 

sort((table(tax_table(eff_hep_sun)[,6]))/ntaxa(eff_hep_sun))*100
#Rhodoferax (15.9), Pseudorhodobacter (8.41), Hyphomicrobium (6.01)

#Trichoptera
nsamples(trichop_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Trichoptera")) #8


eff_trichop_sun<-subset_samples(effluent_sun, Family=="Trichoptera")
eff_trichop_sun<-prune_taxa(taxa_sums(eff_trichop_sun) > 0, eff_trichop_coch) 
eff_trichop_sun #522 taxa, 7 samples = 87.5%


sort((table(tax_table(eff_trichop_sun)[,6]))/ntaxa(eff_trichop_sun))*100
#Rhodoferax (12.6%), Massilia (6.32), Pseudorhodobacter (6.13%)

#Ephemeroptera 
nsamples(ephem_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Ephemeroptera")) #8


eff_ephem_sun<-subset_samples(effluent_sun, Family=="Ephemeroptera")
eff_ephem_sun<-prune_taxa(taxa_sums(eff_ephem_sun) > 0, eff_ephem_sun) 
eff_ephem_sun #98 taxa, 8 samples

sort((table(tax_table(eff_ephem_sun)[,6]))/ntaxa(eff_ephem_sun))*100
#Massilia (16.3%), Staphylococcus (11.2%), Corynebacterium (7.14%)

#Araneidae
nsamples(aran_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Araneidae")) #8


eff_aran_sun<-subset_samples(effluent_sun, Family=="Araneidae")
eff_aran_sun<-prune_taxa(taxa_sums(eff_aran_sun) > 0, eff_aran_sun) 
eff_aran_sun #109 taxa, 8 samples

sort((table(tax_table(eff_aran_sun)[,6]))/ntaxa(eff_aran_sun))*100
#Massilia (24.8), Blautia (7.34), Arthtobacter (6.42)


#Tetragnathidae
nsamples(tetra_sun<-subset_samples(sample_ps, Site=="Sunalta" & Family=="Tetragnathidae")) #8


eff_tetra_sun<-subset_samples(effluent_sun, Family=="Tetragnathidae")
eff_tetra_sun<-prune_taxa(taxa_sums(eff_tetra_sun) > 0, eff_tetra_sun) 
eff_tetra_sun #14 taxa, 6 samples = 75

sort((table(tax_table(eff_tetra_sun)[,6]))/ntaxa(eff_tetra_sun))*100
#Massilia (35.7), Clostridium sensu stricto 1 (14.3), Blautia (14.3)


## Cushing Bridge ##
effluent_cush<-subset_samples(effluent_bac, Site=="Cushing Bridge")
effluent_cush<-prune_taxa(taxa_sums(effluent_cush) > 0, effluent_cush) 
effluent_cush #669 taxa, 59 samples/63 = 93.7 %

sort((table(tax_table(effluent_cush)[,6]))/sum(ntaxa(effluent_cush))*100)

#Hydropsychidae larvae
nsamples(hydro_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Hydropsychidae")) #5


eff_hydro_cush<-subset_samples(effluent_cush, Family=="Hydropsychidae")
eff_hydro_cush<-prune_taxa(taxa_sums(eff_hydro_cush) > 0, eff_hydro_cush) 
eff_hydro_cush #124 taxa, 5 samples = 100% prevalence

sort((table(tax_table(eff_hydro_cush)[,6]))/ntaxa(eff_hydro_cush))*100
#Rhodoferax (19.4), Pseudorhodobacter (17.7), Hyphomicrobium (9.68)

#Heptageniidae larvae
nsamples(hep_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Heptageniidae")) #8


eff_hep_cush<-subset_samples(effluent_cush, Family=="Heptageniidae")
eff_hep_cush<-prune_taxa(taxa_sums(eff_hep_cush) > 0, eff_hep_cush) 
eff_hep_cush #295 taxa, 8 samples 

sort((table(tax_table(eff_hep_cush)[,6]))/ntaxa(eff_hep_cush))*100
#Rhodoferax (16.9), Pseudorhodobacter (7.80), Arenimonas (5.76)

#Chironomidae larvae 
nsamples(chiro_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Chironomidae")) #6


eff_chiro_cush<-subset_samples(effluent_cush, Family=="Chironomidae")
eff_chiro_cush<-prune_taxa(taxa_sums(eff_chiro_cush) > 0, eff_chiro_cush) 
eff_chiro_cush #131 taxa, 6 samples

sort((table(tax_table(eff_chiro_cush)[,6]))/ntaxa(eff_chiro_cush))*100
#Rhodoferax (9.92), Rhodobacter (7.63), Roseomonas (5.34)

#Perlidae larvae 
nsamples(perl_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Perlidae")) #5


eff_perl_cush<-subset_samples(effluent_cush, Family=="Perlidae")
eff_perl_cush<-prune_taxa(taxa_sums(eff_perl_cush) > 0, eff_perl_cush) 
eff_perl_cush #301 taxa, 5 samples, 100%


sort((table(tax_table(eff_perl_cush)[,6]))/ntaxa(eff_perl_cush))*100
#Rhodoferax (13.3), Pseudorhodobacter (9.97), Leptothrix (5.98)

#Trichoptera
nsamples(trichop_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Trichoptera")) #8


eff_trichop_cush<-subset_samples(effluent_cush, Family=="Trichoptera")
eff_trichop_cush<-prune_taxa(taxa_sums(eff_trichop_cush) > 0, eff_trichop_cush) 
eff_trichop_cush #181 taxa, 7 samples = 87.5%


sort((table(tax_table(eff_trichop_cush)[,6]))/ntaxa(eff_trichop_cush))*100
#Pseudorhodobacter (9.94), Massilia (9.94), Rhodoferax (7.18)

#Ephemeroptera 
nsamples(ephem_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Ephemeroptera")) #8


eff_ephem_cush<-subset_samples(effluent_cush, Family=="Ephemeroptera")
eff_ephem_cush<-prune_taxa(taxa_sums(eff_ephem_cush) > 0, eff_ephem_cush) 
eff_ephem_cush #65 taxa, 5 samples = 62.5

sort((table(tax_table(eff_ephem_cush)[,6]))/ntaxa(eff_ephem_cush))*100
#Staphylococcus (20.0), Nocardioides (9.23), Arthrobacter (7.69)

#Diptera
nsamples(dip_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Diptera")) #8


eff_dip_cush<-subset_samples(effluent_cush, Family=="Diptera")
eff_dip_cush<-prune_taxa(taxa_sums(eff_dip_cush) > 0, eff_dip_cush) 
eff_dip_cush #79 taxa, 8 samples =100%

sort((table(tax_table(eff_dip_cush)[,6]))/ntaxa(eff_dip_cush))*100
#Staphylococcus (19.0), Massilia (10.1), Corynebacterium (8.86)

#Araneidae
nsamples(aran_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Araneidae")) #7

eff_aran_cush<-subset_samples(effluent_cush, Family=="Araneidae")
eff_aran_cush<-prune_taxa(taxa_sums(eff_aran_cush) > 0, eff_aran_cush) 
eff_aran_cush #84 taxa, 7 samples =100%

sort((table(tax_table(eff_aran_cush)[,6]))/ntaxa(eff_aran_cush))*100
#Massilia (25.0), Blautia (9.52), Faecalibacterium (7.14)


#Tetragnathidae
nsamples(tetra_cush<-subset_samples(sample_ps, Site=="Cushing Bridge" & Family=="Tetragnathidae")) #8


eff_tetra_cush<-subset_samples(effluent_cush, Family=="Tetragnathidae")
eff_tetra_cush<-prune_taxa(taxa_sums(eff_tetra_cush) > 0, eff_tetra_cush) 
eff_tetra_cush #89 taxa, 8 samples=100%

sort((table(tax_table(eff_tetra_cush)[,6]))/ntaxa(eff_tetra_cush))*100
#Massilia (10.1), Blautia (7.87), Rubellimicrobium (6.74). 

## Graves Bridge ##

effluent_grv<-subset_samples(effluent_bac, Site=="Graves Bridge")
effluent_grv<-prune_taxa(taxa_sums(effluent_grv) > 0, effluent_grv) 
effluent_grv #52 taxa, 40 samples/47 = 72.3 %

sort((table(tax_table(effluent_grv)[,6]))/sum(ntaxa(effluent_grv))*100)

#Hydropsychidae larvae
nsamples(hydro_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Hydropsychidae")) #8


eff_hydro_grv<-subset_samples(effluent_grv, Family=="Hydropsychidae")
eff_hydro_grv<-prune_taxa(taxa_sums(eff_hydro_grv) > 0, eff_hydro_grv) 
eff_hydro_grv #212 taxa, 8 samples =100% prev

sort((table(tax_table(eff_hydro_grv)[,6]))/ntaxa(eff_hydro_grv))*100
#Rhodoferax (17.9), Pseudorhodobacter (13.7), Hyphomicrobium (8.49)

#Heptageniidae larvae
nsamples(hep_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Heptageniidae")) #8


eff_hep_grv<-subset_samples(effluent_grv, Family=="Heptageniidae")
eff_hep_grv<-prune_taxa(taxa_sums(eff_hep_grv) > 0, eff_hep_grv) 
eff_hep_grv #342 taxa, 8 samples 

sort((table(tax_table(eff_hep_grv)[,6]))/ntaxa(eff_hep_grv))*100
#Rhodoferax (18.1), Pseudorhodobacter (8.19), Hyphomicrobium (5.85)

#Trichoptera
nsamples(trichop_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Trichoptera")) #7


eff_trichop_grv<-subset_samples(effluent_grv, Family=="Trichoptera")
eff_trichop_grv<-prune_taxa(taxa_sums(eff_trichop_grv) > 0, eff_trichop_grv) 
eff_trichop_grv #141 taxa, 7 samples


sort((table(tax_table(eff_trichop_grv)[,6]))/ntaxa(eff_trichop_grv))*100
#Brevundimonas (8.51), Massilia (7.80), Staphylococcus (5.67)

#Ephemeroptera 
nsamples(ephem_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Ephemeroptera")) #8


eff_ephem_grv<-subset_samples(effluent_grv, Family=="Ephemeroptera")
eff_ephem_grv<-prune_taxa(taxa_sums(eff_ephem_grv) > 0, eff_ephem_grv) 
eff_ephem_grv #54 taxa, 8 samples

sort((table(tax_table(eff_ephem_grv)[,6]))/ntaxa(eff_ephem_grv))*100
#Staphylococcus (18.5), Corynebacterium (7.41), Nocardioides (5.56)

#Araneidae
nsamples(aran_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Araneidae")) #8


eff_aran_grv<-subset_samples(effluent_grv, Family=="Araneidae")
eff_aran_grv<-prune_taxa(taxa_sums(eff_aran_grv) > 0, eff_aran_grv) 
eff_aran_grv #98 taxa, 8 samples=100%

sort((table(tax_table(eff_aran_grv)[,6]))/ntaxa(eff_aran_grv))*100
#Massilia (21.4), Blautia (6.12), Arthrobacter (6.12)


#Tetragnathidae
nsamples(tetra_grv<-subset_samples(sample_ps, Site=="Graves Bridge" & Family=="Tetragnathidae")) #8


eff_tetra_grv<-subset_samples(effluent_grv, Family=="Tetragnathidae")
eff_tetra_grv<-prune_taxa(taxa_sums(eff_tetra_grv) > 0, eff_tetra_grv) 
eff_tetra_grv #26 taxa, 6 samples = 62.5

sort((table(tax_table(eff_tetra_grv)[,6]))/ntaxa(eff_tetra_grv))*100
#Faecalibacterium (11.5), Ruminococcus (7.69), Corynebacterium (7.69)

## Policeman Flats ##

effluent_pmf<-subset_samples(effluent_bac, Site=="Policeman Flats")
effluent_pmf<-prune_taxa(taxa_sums(effluent_pmf) > 0, effluent_pmf) 
effluent_pmf #708 taxa, 59 samples/60 

sort((table(tax_table(effluent_pmf)[,6]))/sum(ntaxa(effluent_pmf))*100)

#Hydropsychidae larvae
nsamples(hydro_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Hydropsychidae")) #8


eff_hydro_pmf<-subset_samples(effluent_pmf, Family=="Hydropsychidae")
eff_hydro_pmf<-prune_taxa(taxa_sums(eff_hydro_pmf) > 0, eff_hydro_pmf) 
eff_hydro_pmf #197 taxa, 8 samples = 100% prevalence

sort((table(tax_table(eff_hydro_pmf)[,6]))/ntaxa(eff_hydro_pmf))*100
#Rhodoferax (14.2), Pseudorhodobacter (10.7), Rhizorhapis (5.58)

#Heptageniidae larvae
nsamples(hep_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Heptageniidae")) #8

eff_hep_pmf<-subset_samples(effluent_pmf, Family=="Heptageniidae")
eff_hep_pmf<-prune_taxa(taxa_sums(eff_hep_pmf) > 0, eff_hep_pmf) 
eff_hep_pmf #392 taxa, 8 samples 

sort((table(tax_table(eff_hep_pmf)[,6]))/ntaxa(eff_hep_pmf))*100
#Rhodoferax (13.3), Pseudorhodobacter (11.2), Hyphomicrobium (7.65)

#Chironomidae larvae 
nsamples(chiro_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Chironomidae")) #5

eff_chiro_pmf<-subset_samples(effluent_pmf, Family=="Chironomidae")
eff_chiro_pmf<-prune_taxa(taxa_sums(eff_chiro_pmf) > 0, eff_chiro_pmf) 
eff_chiro_pmf #168 taxa, 5 samples

sort((table(tax_table(eff_chiro_pmf)[,6]))/ntaxa(eff_chiro_pmf))*100
#Rhodoferax (12.5), Pseudorhodobacter (8.93), Leptothrix (5.95)

#Perlidae larvae 
nsamples(perl_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Perlidae")) #8

eff_perl_pmf<-subset_samples(effluent_pmf, Family=="Perlidae")
eff_perl_pmf<-prune_taxa(taxa_sums(eff_perl_pmf) > 0, eff_perl_pmf) 
eff_perl_pmf #192 taxa, 8 samples


sort((table(tax_table(eff_perl_pmf)[,6]))/ntaxa(eff_perl_pmf))*100
#Methylotenera (10.9), Rhodoferax (9.38), Pseudorhodobacter (9.38)

#Ephemeroptera 
nsamples(ephem_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Ephemeroptera")) #8

eff_ephem_pmf<-subset_samples(effluent_pmf, Family=="Ephemeroptera")
eff_ephem_pmf<-prune_taxa(taxa_sums(eff_ephem_pmf) > 0, eff_ephem_pmf) 
eff_ephem_pmf #119 taxa, 7 samples = 87.5

sort((table(tax_table(eff_ephem_pmf)[,6]))/ntaxa(eff_ephem_pmf))*100
#Staphylococcus (7.56), Massilia (7.56), Pseudorhodobacter (6.72)

#Diptera
nsamples(dip_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Diptera")) #8

eff_dip_pmf<-subset_samples(effluent_pmf, Family=="Diptera")
eff_dip_pmf<-prune_taxa(taxa_sums(eff_dip_pmf) > 0, eff_dip_pmf) 
eff_dip_pmf #66 taxa, 8 samples

sort((table(tax_table(eff_dip_pmf)[,6]))/ntaxa(eff_dip_pmf))*100
#Staphylococcus (15.2), Serratia (6.06), Halomonas (6.06)

#Araneidae
nsamples(aran_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Araneidae")) #7

eff_aran_pmf<-subset_samples(effluent_pmf, Family=="Araneidae")
eff_aran_pmf<-prune_taxa(taxa_sums(eff_aran_pmf) > 0, eff_aran_pmf) 
eff_aran_pmf #114 taxa, 7 samples

sort((table(tax_table(eff_aran_pmf)[,6]))/ntaxa(eff_aran_pmf))*100
#Massilia (20.2), Brevundimonas (10.5), Rubellimicrobium (5.26)


#Tetragnathidae
nsamples(tetra_pmf<-subset_samples(sample_ps, Site=="Policeman Flats" & Family=="Tetragnathidae")) #8

eff_tetra_pmf<-subset_samples(effluent_pmf, Family=="Tetragnathidae")
eff_tetra_pmf<-prune_taxa(taxa_sums(eff_tetra_pmf) > 0, eff_tetra_pmf) 
eff_tetra_pmf #82 taxa, 8 samples

sort((table(tax_table(eff_tetra_pmf)[,6]))/ntaxa(eff_tetra_pmf))*100
#Blautia (9.76), Brevundimonas (8.54), Massilia (7.32)

## REF ACWA Stream ##

effluent_BRR2<-subset_samples(effluent_bac, Site=="BRR2")
effluent_BRR2<-prune_taxa(taxa_sums(effluent_BRR2) > 0, effluent_BRR2) 
effluent_BRR2 #212 taxa, 8 samples/8=100%

sort((table(tax_table(effluent_BRR2)[,6]))/sum(ntaxa(effluent_BRR2))*100)

## EXP5 ACWA Stream ##

effluent_PCR1<-subset_samples(effluent_bac, Site=="PCR1")
effluent_PCR1<-prune_taxa(taxa_sums(effluent_PCR1) > 0, effluent_PCR1) 
effluent_PCR1 #29 taxa, 24 samples=100%

sort((table(tax_table(effluent_PCR1)[,6]))/sum(ntaxa(effluent_PCR1))*100)

## EXP15 ACWA Stream ##

effluent_PCR3<-subset_samples(effluent_bac, Site=="PCR3")
effluent_PCR3<-prune_taxa(taxa_sums(effluent_PCR3) > 0, effluent_PCR3) 
effluent_PCR3 #28 taxa, 23 samples

sort((table(tax_table(effluent_PCR3)[,6]))/sum(ntaxa(effluent_PCR3))*100)

## Bow River ##

effluent_Bow<-subset_samples(effluent_bac, Stream_Type=="Bow River")

effluent_genus_Bow<-speedyseq::tax_glom(effluent_Bow, taxrank = "Genus")
effluent_rel_abun_genus_Bow <- transform_sample_counts(effluent_genus_Bow, function(x) x/sum(x))
effluent_genus_df_Bow <- psmelt(effluent_rel_abun_genus_Bow)
sample.df.agg.effluent.genus.Bow <- aggregate(Abundance ~ Site  + Genus, data = effluent_genus_df_Bow, FUN = mean)

ggplot(sample.df.agg.effluent.genus.Bow, aes(x=Site, y=Abundance, fill=Genus)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=17)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  #scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)


sample.df.agg.effluent.genus.Bow.invert <- aggregate(Abundance ~ Site  + Genus +sample_Family+Stage, data = effluent_genus_df_Bow, FUN = mean)

sample.df.agg.effluent.genus.Bow.invert$sample_Family<-factor(sample.df.agg.effluent.genus.Bow.invert$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))



ggplot(sample.df.agg.effluent.genus.Bow.invert, aes(x=Site, y=Abundance, fill=Genus)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  facet_nested_wrap(~Stage+sample_Family, nrow=1, scales="free_x")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=50)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  #scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)

## ACWA ##

effluent_ACWA<-subset_samples(effluent_bac, Stream_Type!="Bow River")

effluent_genus_ACWA<-speedyseq::tax_glom(effluent_ACWA, taxrank = "Genus")
effluent_rel_abun_genus_ACWA <- transform_sample_counts(effluent_genus_ACWA, function(x) x/sum(x))
effluent_genus_df_ACWA <- psmelt(effluent_rel_abun_genus_ACWA)
sample.df.agg.effluent.genus.ACWA <- aggregate(Abundance ~ Site  + Genus + sample_Family +Stage, data = effluent_genus_df_ACWA, FUN = mean)

sample.df.agg.effluent.genus.ACWA$sample_Family<-factor(sample.df.agg.effluent.genus.ACWA$sample_Family, levels=c("Hydropsychidae","Baetidae", "Ephemeroptera"))

ggplot(sample.df.agg.effluent.genus.ACWA, aes(x=Site, y=Abundance, fill=Genus)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  facet_nested_wrap(~Stage+sample_Family, nrow=1, scales="free_x")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=50)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  #scale_fill_manual(name="Genus", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#43464B"))+
  scale_y_continuous(label = scales::percent)


#### Plotting eff-assoc. bac. relative to non eff-assoc. bac. across sites + taxa ####


effluents_bac<-subset_taxa(sample_ps, Genus=="Achromobacter" | Genus=="Acidovorax" | Genus=="Acinetobacter"| Genus=="Aeromonas" | Genus=="Ahniella" | Genus=="Alcaligenes" | Genus=="Amaricoccus" | Genus=="Anaerocella" | Genus=="Anaerovorax" | Genus=="Arenimonas" | Genus=="Arthrobacter" | Genus=="Bacillus" | Genus == "Barnesiella" | Genus=="Blastocatella" | Genus=="Blautia" | Genus=="Bordetella" | Genus=="Bosea" | Genus=="Brevundimonas" | Genus=="Candidatus Accumulibacter" | Genus=="Candidatus Competibacter" | Genus=="Candidatus Contendobacter" | Genus=="Candidatus Microthrix" | Genus=="Candidatus Nitrotoga" | Genus=="Cetobacterium" | Genus =="Chitinivorax" | Genus=="Chryseobacterium" | Genus=="Chthoniobacter" | Genus=="Citrobacter" | Genus=="Cloacibacterium" | Genus=="Clostridium sensu stricto 1" | Genus=="Clostridium sensu stricto 13" | Genus=="Clostridium sensu stricto 5" | Genus=="Corynebacterium" | Genus=="Curtobacterium" | Genus=="Curvibacter" | Genus=="Cytophaga" | Genus=="Dechloromonas" | Genus=="Defluviicoccus" | Genus=="Delftia" | Genus=="Desulfatirhabdium" | Genus=="Desulfobacca" | Genus=="Desulfobulbus" | Genus=="Desulfoprunum" | Genus=="Desulfosporosinus" | Genus=="Ellin6067" | Genus=="Enterococcus" | Genus=="Erysipelothrix" | Genus=="Escherichia-Shigella" | Genus=="Faecalibacterium" | Genus=="Ferruginibacter" | Genus=="Fusicatenibacter" | Genus=="Haliangium" | Genus=="Haliscomenobacter" | Genus=="Halomonas" | Genus=="Hydrogenophaga" | Genus=="Hyphomicrobium" | Genus=="Ideonella" | Genus=="Inhella" | Genus=="Intestinibacter" | Genus=="Lachnoclostridium" | Genus=="Lacunisphaera" | Genus=="Leptothrix" | Genus=="Longivirga" | Genus=="Luteimonas" | Genus=="Massilia" | Genus=="Methylobacillus" | Genus=="Methylocystis" | Genus=="Methylotenera" | Genus=="Micrococcus" | Genus=="Monoglobus" | Genus=="Moraxella" | Genus=="Mycobacterium" | Genus=="Nitratireductor" | Genus=="Nitrosomonas" | Genus=="Nitrospira" | Genus=="Nocardioides" | Genus=="Novosphingobium" | Genus=="OLB12" | Genus=="OLB8" | Genus=="Opitutus" | Genus=="P3OB-42" | Genus=="Paludibacter" | Genus=="Paraclostridium" | Genus=="Paracoccus" | Genus=="Pelomonas" | Genus=="Phaeodactylibacter" | Genus=="Phenylobacterium" | Genus=="Planococcus" | Genus=="Polaromonas" | Genus=="Polyangium" | Genus=="Propionivibrio" | Genus=="Pseudarcobacter" | Genus=="Pseudomonas" | Genus=="Pseudorhodobacter" | Genus=="Pseudoxanthomonas" | Genus=="Ralstonia" | Genus=="Rhizorhapis" | Genus=="Rhodobacter" | Genus=="Rhodoferax" | Genus=="Roseomonas" | Genus=="Rubellimicrobium" | Genus=="Rubrivivax" | Genus=="Ruminococcus" | Genus=="Runella" | Genus=="Saprospira" | Genus=="Sediminibacterium" | Genus=="Serratia" | Genus=="Sphingobium" | Genus=="Staphylococcus" | Genus=="Stenotrophomonas" | Genus=="Sulfurospirillum" | Genus=="Syntrophobacter" | Genus=="Trichococcus" | Genus=="Thermomonas" | Genus=="Youngiibacter" | Genus=="Zoogloea")


#Bow River 

Bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
Bow_ps<-prune_taxa(taxa_sums(Bow_ps) > 0, Bow_ps) 
Bow_ps #4194 taxa, 284 samples

effluent_Bow<-subset_samples(effluent_bac, Stream_Type=="Bow River")
effluent_Bow<-prune_taxa(taxa_sums(effluent_Bow) > 0, effluent_Bow) 
sort((table(tax_table(effluent_Bow)[,6]))/sum(ntaxa(effluent_Bow))*100) #117 genera
#Rhodoferax (10.4%), Pseudorhodobacter (6.39%), Hyphomicrobium (4.33%)
sum(sample_sums(effluent_Bow))/sum(sample_sums(Bow_ps))*100
#Effluent bacteria comprises 9.17% of the total microbiome in the Bow River (across all invertebrate types). Decently small amount.

## Plotting relative abundance of effluent bacteria versus non eff-assoc. bacteria ##

Bow_genus <- tax_glom(Bow_ps, taxrank = "Genus")
Bow_genus<-prune_taxa(taxa_sums(Bow_genus) > 0, Bow_genus) 
Bow_genus #481 genera, 284 samples
comp_Bow_genus <- transform_sample_counts(Bow_genus, function(x) x/sum(x))

Bow_genus_df <- psmelt(comp_Bow_genus)

Bow_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = Bow_genus_df, FUN = mean) #aggregate to Genus level, take the mean

#Sort abundance of each genus by mean at each site
sort_genus_Bow <- Bow_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_Bow

effluents_Bow <- sort_genus_Bow$Genus[c(31:35, 46:55, 81:85, 101:105, 111:115, 146:150, 156:160, 166:170, 191:195, 201:205, 241:260, 276:290, 301:305, 361:365, 396:405, 426:430, 436:440, 501:505, 516:520, 531:540, 546:550, 556:575, 601:605, 616:625, 636:640, 656:660, 666:670, 681:685, 691:715, 771:775, 796:800, 806:810, 821:825, 831:835, 856:860, 931:935, 1006:1020, 1026:1030, 1036:1040, 1066:1070, 1086:1095, 1141:1145, 1191:1195, 1226:1230, 1246:1255, 1296:1300, 1306:1310, 1326:1335, 1351:1360, 1376:1380, 1431:1450, 1461:1475, 1486:1490, 1506:1510, 1531:1535, 1561:1570, 1606:1610, 1621:1625, 1636:1640, 1656:1660, 1671:1680, 1711:1715, 1736:1740, 1756:1760, 1766:1770, 1776:1780, 1791:1795, 1826:1835, 1841:1845, 1881:1885, 1896:1900, 1911:1915, 1926:1935, 1961:1965, 1976:1985, 2006:2010, 2056:2060, 2071:2075, 2101:2105, 2121:2125, 2161:2165, 2176:2180, 2306:2310, 2321:2325)] 
  
eff_df_Bow <- Bow_agg %>%
    mutate(Genus = fct_other(Genus, effluents_Bow))

eff_agg_df_Bow<-aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = eff_df_Bow, FUN = sum)

eff_agg_df_Bow$Site_code <- factor(eff_agg_df_Bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

eff_agg_df_Bow$sample_Family<-factor(eff_agg_df_Bow$sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

pal <- gg_color_hue(117)
pal[117] <- "#43464B"

eff_plot_Bow <- ggplot(eff_agg_df_Bow, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.8, "cm"),
        legend.text=element_text(size=12, face="italic"), legend.title=element_text(size=16), legend.position = "right") +
  guides(fill=guide_legend(nrow=39, ncol=3)) +
  theme(axis.title.y=element_text(size=30),
        axis.text.y=element_text(size=24, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=24, color="black"),
                        axis.title.x=element_text(size=30))+
  scale_fill_manual(name="Genus", values=pal)+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
eff_plot_Bow

#ggsave(eff_plot_Bow, filename="microbiome_analysis/Final analysis/Relative abundance/Effluent bacteria/Effluent associated bacteria Bow river.png", height=14, width=30)

#Can't differentiate between colors but it gives an idea of the total relative abundance of effluent associated bacteria across invertebrates (are there certain ones that accumulate more than others) nad across sites (do organisms from downstream exposed sites have more of these bacteria than upstream sites?)


# ACWA Streams

ACWA_ps<-subset_samples(sample_ps, Stream_Type!="Bow River")

effluent_ACWA<-subset_samples(effluent_bac, Stream_Type!="Bow River")
effluent_ACWA<-prune_taxa(taxa_sums(effluent_ACWA) > 0, effluent_ACWA) 
sort((table(tax_table(effluent_ACWA)[,6]))/sum(ntaxa(effluent_ACWA))*100) #96 genera
#Pseudorhodobacter (8.58%), Rhodoferax (7.78%), Rhodobacter (5.59%)
sum(sample_sums(effluent_ACWA))/sum(sample_sums(ACWA_ps))*100
#Effluent bacteria comprises 18.3% of the total microbiome in the ACWA streams (across all invertebrate types). Much larger than Bow River.

## Plotting relative abundance of effluent bacteria versus non eff-assoc. bacteria ##

ACWA_genus <- tax_glom(ACWA_ps, taxrank = "Genus")
ACWA_genus<-prune_taxa(taxa_sums(ACWA_genus) > 0, ACWA_genus) 
ACWA_genus #338 genera, 55 samples

comp_ACWA_genus <- transform_sample_counts(ACWA_genus, function(x) x/sum(x))

ACWA_genus_df <- psmelt(comp_ACWA_genus)

ACWA_agg <- aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = ACWA_genus_df, FUN = mean) 

#Sort abundance of each genus by mean at each site
sort_genus_ACWA <- ACWA_agg %>%
    group_by(Genus, Site_code) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange()
sort_genus_ACWA

#effluent_bac<-subset_taxa(sample_ps, Genus=="Achromobacter" | Genus=="Acidovorax" | Genus=="Acinetobacter"| Genus=="Aeromonas" | Genus=="Ahniella" | Genus=="Alcaligenes" | Genus=="Amaricoccus" | Genus=="Anaerocella" | Genus=="Anaerovorax" | Genus=="Arenimonas" | Genus=="Arthrobacter" | Genus=="Bacillus" | Genus == "Barnesiella" | Genus=="Blastocatella" | Genus=="Blautia" | Genus=="Bordetella" | Genus=="Bosea" | Genus=="Brevundimonas" | Genus=="Candidatus Accumulibacter" | Genus=="Candidatus Competibacter" | Genus=="Candidatus Contendobacter" | Genus=="Candidatus Microthrix" | Genus=="Candidatus Nitrotoga" | Genus=="Cetobacterium" | Genus =="Chitinivorax" | Genus=="Chryseobacterium" | Genus=="Chthoniobacter" | Genus=="Citrobacter" | Genus=="Cloacibacterium" | Genus=="Clostridium sensu stricto 1" | Genus=="Clostridium sensu stricto 13" | Genus=="Clostridium sensu stricto 5" | Genus=="Corynebacterium" | Genus=="Curtobacterium" | Genus=="Curvibacter" | Genus=="Cytophaga" | Genus=="Dechloromonas" | Genus=="Defluviicoccus" | Genus=="Delftia" | Genus=="Desulfatirhabdium" | Genus=="Desulfobacca" | Genus=="Desulfobulbus" | Genus=="Desulfoprunum" | Genus=="Desulfosporosinus" | Genus=="Ellin6067" | Genus=="Enterococcus" | Genus=="Erysipelothrix" | Genus=="Escherichia-Shigella" | Genus=="Faecalibacterium" | Genus=="Ferruginibacter" | Genus=="Fusicatenibacter" | Genus=="Haliangium" | Genus=="Haliscomenobacter" | Genus=="Halomonas" | Genus=="Hydrogenophaga" | Genus=="Hyphomicrobium" | Genus=="Ideonella" | Genus=="Inhella" | Genus=="Intestinibacter" | Genus=="Lachnoclostridium" | Genus=="Lacunisphaera" | Genus=="Leptothrix" | Genus=="Longivirga" | Genus=="Luteimonas" | Genus=="Massilia" | Genus=="Methylobacillus" | Genus=="Methylocystis" | Genus=="Methylotenera" | Genus=="Micrococcus" | Genus=="Monoglobus" | Genus=="Moraxella" | Genus=="Mycobacterium" | Genus=="Nitratireductor" | Genus=="Nitrosomonas" | Genus=="Nitrospira" | Genus=="Nocardioides" | Genus=="Novosphingobium" | Genus=="OLB12" | Genus=="OLB8" | Genus=="Opitutus" | Genus=="P3OB-42" | Genus=="Paludibacter" | Genus=="Paraclostridium" | Genus=="Paracoccus" | Genus=="Pelomonas" | Genus=="Phaeodactylibacter" | Genus=="Phenylobacterium" | Genus=="Planococcus" | Genus=="Polaromonas" | Genus=="Polyangium" | Genus=="Propionivibrio" | Genus=="Pseudarcobacter" | Genus=="Pseudomonas" | Genus=="Pseudorhodobacter" | Genus=="Pseudoxanthomonas" | Genus=="Ralstonia" | Genus=="Rhizorhapis" | Genus=="Rhodobacter" | Genus=="Rhodoferax" | Genus=="Roseomonas" | Genus=="Rubellimicrobium" | Genus=="Rubrivivax" | Genus=="Ruminococcus" | Genus=="Runella" | Genus=="Saprospira" | Genus=="Sediminibacterium" | Genus=="Serratia" | Genus=="Sphingobium" | Genus=="Staphylococcus" | Genus=="Stenotrophomonas" | Genus=="Sulfurospirillum" | Genus=="Syntrophobacter" | Genus=="Trichococcus" | Genus=="Thermomonas" | Genus=="Youngiibacter" | Genus=="Zoogloea")

effluents_ACWA <- sort_genus_ACWA$Genus[c(7:9, 13:18, 31:33, 37:42, 61:63, 67:72, 82:84, 88:90, 106:108, 112:120, 124:129, 133:135, 163:165, 178:183, 190:192, 196:198, 229:231, 238:252, 256:258, 265:270, 277:279, 286:288, 292:294, 298:300, 304:318, 340:342, 352:357, 361:366, 433:441, 445:447, 451:453, 463:465, 472:477, 493:495, 514:516, 526:528, 535:540, 559:561, 565:567, 571:579, 583:588, 595:597, 607:618, 622:630, 637:639, 646:648, 655:657, 667:672, 685:687, 694:696, 718:720, 748:750, 757:759, 769:774, 778:780, 799:801, 817:822, 835:837, 865:867, 871:873, 886:888, 898:900, 925:927, 931:933)]

  
eff_df_ACWA <- ACWA_agg %>%
    mutate(Genus = fct_other(Genus, effluents_ACWA))

eff_agg_df_ACWA<-aggregate(Abundance ~ Genus + Site_code + sample_Family + Stage, data = eff_df_ACWA, FUN = sum)

eff_agg_df_ACWA$Site_code <- factor(eff_agg_df_ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

eff_agg_df_ACWA$sample_Family<-factor(eff_agg_df_ACWA$sample_Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

pal_ACWA <- gg_color_hue(98)
pal_ACWA[98] <- "#43464B"

eff_plot_ACWA <- ggplot(eff_agg_df_ACWA, aes(x=Site_code, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage+sample_Family, scale="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 22, color="black")) + 
  theme(legend.key.size = unit(0.8, "cm"),
        legend.text=element_text(size=12, face="italic"), legend.title=element_text(size=16), legend.position = "right") +
  guides(fill=guide_legend(nrow=33, ncol=3)) +
  theme(axis.title.y=element_text(size=30),
        axis.text.y=element_text(size=24, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=24, color="black"),
                        axis.title.x=element_text(size=30))+
  scale_fill_manual(name="Genus", values=pal_ACWA)+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
eff_plot_ACWA


#ggsave(eff_plot_ACWA, filename="microbiome_analysis/Final analysis/Relative abundance/Effluent bacteria/Effluent associated bacteria ACWA.png", height=14, width=30)

```

# Picrust2

## Data prep from PICRUSt2 output files

```{r}
#Need to create a fasta file which requires an ASV identifier in the first column (ex. 'ASV1', 'ASV2', etc.) and the bacterial DNA sequence in the second column.

#Also need a .biom file with the otu/asv table (ASVs are labeleld with a unique identifier that matches the fasta file)

#.biom file

taxa_names(sample_ps) <- paste0("ASV", seq(ntaxa(sample_ps)))
otu<-as(otu_table(sample_ps),"matrix")
otu_df<-as.data.frame(otu)
otu_df<- rownames_to_column(otu_df, "ASV")
otu_biom<-make_biom(data=otu_df)
write_biom(otu_biom,"Microbiome_ED/PICRUSt2 Data/otu_biom.biom")

#After filtering steps, convert the phyloseq object back into an ASV table and save as data frame. 

# Extract abundance matrix from the phyloseq object
OTU1 = as(otu_table(sample_ps), "matrix")
# Coerce to data.frame
OTUdf = as.data.frame(OTU1)
#Move row names into the first column and call it 'Sequences' so that it can be pulled into another data frame
OTUdf<- rownames_to_column(OTUdf, "Sequences")

#Take the first column of the filtered OTU table to retrieve bacterial DNA sequences for the fasta file. 
seq<-OTUdf[,1] #Makes a column of sequences

#Now we need to create a new column with an ASV identifer for those sequences that we pulled.
taxa_names(sample_ps) <- paste0("ASV", seq(ntaxa(sample_ps))) #this will change the sequences to a numbered ASV ID


# Extract abundance matrix from the phyloseq object
OTU2 = as(otu_table(sample_ps), "matrix")
# Coerce to data.frame
OTUdf2 = as.data.frame(OTU2)
#Change the rownames to a column called ASVs
OTUdf2<- rownames_to_column(OTUdf2, "ASV")

ASV<-OTUdf2[,1] #Makes a column of ASV identifiers

#Now we add the columns together to create the fasta dataframe
ASVfasta<-data.frame(c(rbind(ASV, seq))) #Combines the ASV IDs and the sequences in a dataframe
#Remove headers

#install_github("helixcn/seqRFLP")
library("seqRFLP") #Used to save files as .fasta

#Converts data frame to fasta file
ASV.fasta = dataframe2fas(ASVfasta, file="Microbiome_ED/PICRUSt2 Data/ASV_Sequences_4.fna")
asv_csv<-read.csv("C:\\Users\\Emilie\\OneDrive - McMaster University\\PICRUSt2 data\\ASV_Sequences.csv")
ASV_fasta = dataframe2fas(asv_csv, file="Microbiome_ED/PICRUSt2 Data/ASV_Sequences_2.fna")

ASV_table<-read.table("C:\\Users\\Emilie\\OneDrive - McMaster University\\PICRUSt2 data\\ASV_table.tsv")
ASV_table_df<-as.data.frame(ASV_table)

#### BIOM file/ tab separated values file (tsv) ####

#We also need a .biom file of the ASV table. The sequences have to turned into an ASV ID that matches the fasta file we just made. So I will use the same dataframe that we pulled the ASV IDs from.

#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#BiocManager::install("biomformat")

library(biomformat) #Need this package for converting to biom files
OTUbiom<-make_biom(OTUdf1) #Turns dataframe to biom format
write_biom(OTUbiom, biom_file="Microbiome_ED/PICRUSt2 Data/ASV_table.biom") #Saves biom format as file
write.table(OTUdf1, file="Microbiome_ED/PICRUSt2 Data/ASV_table.tsv", sep="\t") #Saves dataframe to text file (tab separated values)




#### Preparing taxa table, otu table, and metadata ####
# Export taxonomy table as "tax.txt"

sample_ps_1<-phyloseq_rm_na_tax(sample_ps)
tax<-as(tax_table(sample_ps),"matrix")
tax_cols <- colnames(tax)
tax<-as.data.frame(tax)
tax$taxonomy<-do.call(paste, c(tax[tax_cols], sep=";"))
for(co in tax_cols) tax[co]<-NULL

write.table(tax, "Microbiome_ED/PICRUSt2 Data/tax.txt", quote=FALSE, col.names=FALSE, sep="\t")
tax_fna<-dataframe2fas(tax, file="Microbiome_ED/PICRUSt2 Data/tax.fna")

# Export feature/OTU table

# As a biom file

library(biomformat);packageVersion("biomformat")
## [1] 1.6.0

#if taxa_are_rows=TRUE
otu<-as(otu_table(sample_ps),"matrix")
otu_biom<-make_biom(data=otu)
write_biom(otu_biom,"Microbiome_ED/PICRUSt2 Data/otu_biom.biom")

# As a text file

write.table(otu_table(sample_ps), "Microbiome_ED/PICRUSt2 Data/seqtab.tsv", sep="\t", row.names=TRUE, col.names=NA, quote=FALSE)

# Export metadata (if you have a properly formatted metadata file that you imported in your phyloseq pipeline, you can skip this step and just use that text file directly in QIIME 2)

write.table(sample_data(sample_ps),"Microbiome_ED/PICRUSt2 Data/sample-metadata.txt", sep="\t", row.names=FALSE, col.names=TRUE, quote=FALSE)

otu_matrix<-as.matrix(sample_ps@otu_table)
otu_df<-as.data.frame(otu_matrix)
print(FWD.orients)

uniquesToFasta(otu_matrix, ids=colnames(otu_matrix))

sample_ps %>%
      taxa_names(sample_ps) %>%
      Biostrings::writeXStringSet("Microbiome_ED/PICRUSt2 Data/~/asv.fna", append=FALSE,
                                  compress=FALSE, compression_level=NA, format="fasta")

```


## Load and modify data format

'description' is the description of the individual pathway and 'Pathway_type' is a higher order classification (pathway class). Multiple individual pathways can be cateogrized into a pathway class.

```{r}
#### Merge picrust data to phyloseq ####

#The MetaCyc abundance data acts as the "ASV" file
metacyc<-read.delim("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\PICRUSt2 Data\\Output files\\pathways_descrip.tsv")

#Rename column
metacyc<-rename(metacyc, "function." = "Pathways")

#Names of pathways and classification
metacyc_ont<-read.delim("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\PICRUSt2 Data\\MetaCyc_Ontology.txt")

metacyc_ont<-metacyc_ont[,-2]

#Combining metacyc abundance data with the higher order classifications
metacyc_combined<-dplyr::inner_join(metacyc, metacyc_ont, by="Pathways")

#Creating "otu" table
metacyc_only<-metacyc[,!names(metacyc) %in% c("description")] #Remove columns with descriptions so the values can be rounded for the "ASV" table

metacyc_only_column_to_row<-column_to_rownames(metacyc_only, "Pathways")

metacyc_rounded<-ceiling(metacyc_only_column_to_row) #Have to round it to nearest value for it to work
metacyc_rounded_mat<-as.matrix(metacyc_rounded) #Final table we will use for otu table

#Descriptions of the pathways acts as the taxonomy file
descriptions<-metacyc_combined[,names(metacyc_combined) %in% c("Pathways", "description", "Pathway_type")]
descriptions_1<-column_to_rownames(descriptions, "Pathways")
descriptions_mat<-as.matrix(descriptions_1)


#Metadata is treated the same
metadata_picrust<-read_delim("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\PICRUSt2 Data\\sample-metadata-2.txt", delim="\t", escape_double = FALSE, trim_ws = TRUE)

metadata_picrust<-column_to_rownames(metadata_picrust, "Study_ID")

#Putting data together as phyloseq object
metacyc_table<-otu_table(metacyc_rounded_mat, taxa_are_rows = TRUE)
descrip_table = tax_table(descriptions_mat)

picrust_ps<-phyloseq(metacyc_table, sample_data(metadata_picrust), descrip_table)
```


## Relative abundance of top MetaCyc pathway classes across sites

```{r}
#### Bow River ####

Bow_ps_picrust<-subset_samples(picrust_ps, Stream_Type=="Bow River")

rel_abun_bow <- speedyseq::tax_glom(Bow_ps_picrust, taxrank = "Pathway_type")
comp_rel_abun_bow <- transform_sample_counts(rel_abun_bow, function(x) x/sum(x))

rel_abun_bow_df <- psmelt(comp_rel_abun_bow)

sample.df.agg_bow <- aggregate(Abundance ~ Pathway_type + Site_code + Family + Stage, data = rel_abun_bow_df, FUN = mean) 


#Top pathways
top_pathways_bow <- sample.df.agg_bow %>%
    group_by(Pathway_type, Site_code) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_24_pathways_bow <- top_pathways_bow$Pathway_type[1:106]
df0_bow <- sample.df.agg_bow %>%
    mutate(Pathway_type = fct_other(Pathway_type, top_24_pathways_bow))
df0_bow

df0_bow<-aggregate(Abundance~Pathway_type + Site_code + Family + Stage, data=df0_bow, FUN="sum") #Aggregate again to combine all the 'Other' categories into a single bar on the graph.

length(unique(df0_bow$Pathway_type))

df0_bow$Site_code<-factor(df0_bow$Site_code, levels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF"))

df0_bow$Stage<-factor(df0_bow$Stage, levels=c("Larvae", "Adults", "Spiders"))

df0_bow$Family<-factor(df0_bow$Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae"))

toppathwaysplot_bow <- ggplot(df0_bow, aes(x=Site_code, y=Abundance, fill=Pathway_type)) + 
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage + Family, scales="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=14), legend.title=element_text(size=14), legend.position = "right") +
  guides (fill=guide_legend(nrow=25)) +
  theme(axis.title.y=element_text(size=24),
        axis.text.y=element_text(size=18, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=18, color="black"),
                        axis.title.x=element_text(size=24))+
  scale_fill_manual(name="MetaCyc Pathways", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#faf0e6", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")
toppathwaysplot_bow

#ggsave(filename="microbiome_analysis/PICRUSt2 Data/Figures/Top 24 MetaCyc pathway classes across sites and taxa Bow.png", height=10, width=24)


#All taxa together 

bow_all_picrust_agg <- aggregate(Abundance ~ Pathway_type, data = rel_abun_bow_df, FUN = mean) 

ggplot(bow_all_picrust_agg, aes(x=Pathway_type, y=Abundance)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.01, "cm"),
        legend.text=element_text(size=1), legend.title=element_text(size=11), legend.position = "right") +
  guides (fill=guide_legend(nrow=25)) +
  theme(axis.title.y=element_text(size=24),
        axis.text.y=element_text(size=14, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=3, color="black"),
                        axis.title.x=element_text(size=24))+
  #scale_fill_manual(name="MetaCyc Pathways", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#faf0e6", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")


df0_bow %>%
  group_by(Pathway_type) %>%
  dplyr::summarise(mean = mean(Abundance),
            sd = sd(Abundance))%>%
  arrange(desc(mean))


#### ACWA ####

ACWA_ps_picrust<-subset_samples(picrust_ps, Stream_Type!="Bow River")

rel_abun_ACWA<- speedyseq::tax_glom(ACWA_ps_picrust, taxrank = "Pathway_type")
comp_rel_abun_ACWA <- transform_sample_counts(rel_abun_ACWA, function(x) x/sum(x))

rel_abun_ACWA_df <- psmelt(comp_rel_abun_ACWA)

sample.df.agg_ACWA <- aggregate(Abundance ~ Pathway_type + Site_code + Family + Stage, data = rel_abun_ACWA_df, FUN = mean) 


#Top pathways
top_pathways_ACWA <- sample.df.agg_ACWA %>%
    group_by(Pathway_type, Site_code) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_24_pathways_ACWA <- top_pathways_ACWA$Pathway_type[1:62]
df0_ACWA <- sample.df.agg_ACWA %>%
    mutate(Pathway_type = fct_other(Pathway_type, top_24_pathways_ACWA))
df0_ACWA

df0_ACWA<-aggregate(Abundance~Pathway_type + Site_code + Family + Stage, data=df0_ACWA, FUN="sum") #Aggregate again to combine all the 'Other' categories into a single bar on the graph.

length(unique(df0_ACWA$Pathway_type))

df0_ACWA$Site_code<-factor(df0_ACWA$Site_code, levels=c("BRR2", "PCR1", "PCR3"), labels=c("REF", "EXP5", "EXP15"))

df0_ACWA$Stage<-factor(df0_ACWA$Stage, levels=c("Larvae", "Adults"))

df0_ACWA$Family<-factor(df0_ACWA$Family, levels=c("Hydropsychidae", "Baetidae", "Ephemeroptera"))

toppathwaysplot_ACWA <- ggplot(df0_ACWA, aes(x=Site_code, y=Abundance, fill=Pathway_type)) + 
  geom_bar(stat="identity", color="black") + 
  facet_nested_wrap(~Stage + Family, scales="free_x", nrow=1)+
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=14), legend.title=element_text(size=14), legend.position = "right") +
  guides (fill=guide_legend(nrow=25)) +
  theme(axis.title.y=element_text(size=24),
        axis.text.y=element_text(size=18, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=18, color="black"),
                        axis.title.x=element_text(size=24))+
  scale_fill_manual(name="MetaCyc Pathways", values=c(as.vector(stepped(20)),"#C8BC78", "#D5CE9D", "#F7F2CE", "#faf0e6", "#43464B"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")
toppathwaysplot_ACWA

#ggsave(filename="microbiome_analysis/PICRUSt2 Data/Figures/Top 24 MetaCyc pathway classes across sites and taxa ACWA.png", height=10, width=24)


#top pathways with all samples combined 

ACWA_all_picrust_agg <- aggregate(Abundance ~ Pathway_type, data = rel_abun_ACWA_df, FUN = mean) 

ggplot(ACWA_all_picrust_agg, aes(x=Pathway_type, y=Abundance)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme_bw()+
  rremove("grid")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.01, "cm"),
        legend.text=element_text(size=1), legend.title=element_text(size=11), legend.position = "right") +
  guides (fill=guide_legend(nrow=25)) +
  theme(axis.title.y=element_text(size=24),
        axis.text.y=element_text(size=14, color="black"),
        axis.text.x=element_text(angle=45, hjust=1, size=3, color="black"),
                        axis.title.x=element_text(size=24))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")


df0_ACWA %>%
  group_by(Pathway_type) %>%
  dplyr::summarise(mean = mean(Abundance),
            sd = sd(Abundance))%>%
  arrange(desc(mean))

```

## Differential Abundance of individual MetaCyc pathways across sites 

```{r}
#### ACWA comparisons ####
### REF vs EXP5 ###

## Hydropyschidae ##

hydro_BRR2_PCR1<-subset_samples(picrust_ps, Site %in% c("BRR2", "PCR1"))
hydro_BRR2_PCR1<-subset_samples(hydro_BRR2_PCR1, Family=="Hydropsychidae")
nsamples(hydro_BRR2_PCR1) #16

dds_hydro_BRR2_PCR1 <- phyloseq_to_deseq2(hydro_BRR2_PCR1, ~ Site)

deseq_hydro_BRR2_PCR1 <- DESeq(dds_hydro_BRR2_PCR1, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_hydro_BRR2_PCR1 <- results(deseq_hydro_BRR2_PCR1, cooksCutoff = FALSE)
res_deseq_hydro_BRR2_PCR1
alpha = 0.001
sigtab_hydro_BRR2_PCR1 = res_deseq_hydro_BRR2_PCR1[which(res_deseq_hydro_BRR2_PCR1$padj < alpha), ]
sigtab_hydro_BRR2_PCR1 = cbind(as(sigtab_hydro_BRR2_PCR1, "data.frame"), as(tax_table(hydro_BRR2_PCR1)[rownames(sigtab_hydro_BRR2_PCR1), ], "matrix"))

# Adding colours as classes
x_hydro_BRR2_PCR1 = tapply(sigtab_hydro_BRR2_PCR1$log2FoldChange, sigtab_hydro_BRR2_PCR1$Pathway_type, function(x) max(x))
x_hydro_BRR2_PCR1 = sort(x_hydro_BRR2_PCR1, TRUE)
sigtab_hydro_BRR2_PCR1$Pathway_type = factor(as.character(sigtab_hydro_BRR2_PCR1$Pathway_type), levels=sort(names(x_hydro_BRR2_PCR1)))
#Points as descriptions
x_hydro_BRR2_PCR1 = tapply(sigtab_hydro_BRR2_PCR1$log2FoldChange, sigtab_hydro_BRR2_PCR1$description, function(x) max(x))
x_hydro_BRR2_PCR1 = sort(x_hydro_BRR2_PCR1, TRUE)
sigtab_hydro_BRR2_PCR1$description = factor(as.character(sigtab_hydro_BRR2_PCR1$description), levels=names(x_hydro_BRR2_PCR1))


diff_abun_hydro_EXP5_REF<-ggplot(sigtab_hydro_BRR2_PCR1, aes(x=description, y=log2FoldChange, color=Pathway_type))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=12, face="italic")) + theme(axis.text=element_text(size=12, color="black"), axis.title=element_text(size=16)) +
  rremove("grid")+
  ylab("Log2 Fold Change")+
  xlab("Metabolic Pathways")+
  scale_color_brewer(name = "Pathway Class", palette = "Pastel1")+
   labs(y = expression(Log[2]~ "Ratio of Differential Expression"))
diff_abun_hydro_EXP5_REF

#ggsave(filename="microbiome_analysis/Final analysis/PICRUSt2 Data/Figures/diff abun pathways hydro PCR1-BRR2.png", height=10, width=12)


#BRR2 vs PCR3

hydro_BRR2_PCR3<-subset_samples(picrust_ps, Site %in% c("BRR2", "PCR3"))
hydro_BRR2_PCR3<-subset_samples(hydro_BRR2_PCR3, Family=="Hydropsychidae")
nsamples(hydro_BRR2_PCR3) #16

dds_hydro_BRR2_PCR3 <- phyloseq_to_deseq2(hydro_BRR2_PCR3, ~ Site)

deseq_hydro_BRR2_PCR3 <- DESeq(dds_hydro_BRR2_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_hydro_BRR2_PCR3 <- results(deseq_hydro_BRR2_PCR3, cooksCutoff = FALSE)
res_deseq_hydro_BRR2_PCR3
alpha = 0.001
sigtab_hydro_BRR2_PCR3 = res_deseq_hydro_BRR2_PCR3[which(res_deseq_hydro_BRR2_PCR3$padj < alpha), ]
sigtab_hydro_BRR2_PCR3 = cbind(as(sigtab_hydro_BRR2_PCR3, "data.frame"), as(tax_table(hydro_BRR2_PCR3)[rownames(sigtab_hydro_BRR2_PCR3), ], "matrix"))

# Adding colours as classes
x_hydro_BRR2_PCR3 = tapply(sigtab_hydro_BRR2_PCR3$log2FoldChange, sigtab_hydro_BRR2_PCR3$Pathway_type, function(x) max(x))
x_hydro_BRR2_PCR3 = sort(x_hydro_BRR2_PCR3, TRUE)
sigtab_hydro_BRR2_PCR3$Pathway_type = factor(as.character(sigtab_hydro_BRR2_PCR3$Pathway_type), levels=sort(names(x_hydro_BRR2_PCR3)))
#Points as descriptions
x_hydro_BRR2_PCR3 = tapply(sigtab_hydro_BRR2_PCR3$log2FoldChange, sigtab_hydro_BRR2_PCR3$description, function(x) max(x))
x_hydro_BRR2_PCR3 = sort(x_hydro_BRR2_PCR3, TRUE)
sigtab_hydro_BRR2_PCR3$description = factor(as.character(sigtab_hydro_BRR2_PCR3$description), levels=names(x_hydro_BRR2_PCR3))

diff_abun_hydro_EXP15_REF<-ggplot(sigtab_hydro_BRR2_PCR3, aes(x=description, y=log2FoldChange, color=Pathway_type))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=12, face="italic")) + theme(axis.text=element_text(size=12, color="black"), axis.title=element_text(size=16)) +
  rremove("grid")+
  ylab("Log2 Fold Change")+
  xlab("Metabolic Pathways")+
  scale_color_brewer(name = "Pathway Class", palette = "Pastel1")+
   labs(y = expression(Log[2]~ "Ratio of Differential Expression"))
diff_abun_hydro_EXP15_REF

#ggsave(filename="microbiome_analysis/Final analysis/PICRUSt2 Data/Figures/diff abun pathways hydro PCR3-BRR2.png", height=10, width=12)

#PCR1 vs PCR3

hydro_PCR1_PCR3<-subset_samples(picrust_ps, Site %in% c("PCR1", "PCR3"))
hydro_PCR1_PCR3<-subset_samples(hydro_PCR1_PCR3, Family=="Hydropsychidae")
nsamples(hydro_PCR1_PCR3) #16

dds_hydro_PCR1_PCR3 <- phyloseq_to_deseq2(hydro_PCR1_PCR3, ~ Site)

deseq_hydro_PCR1_PCR3 <- DESeq(dds_hydro_PCR1_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_hydro_PCR1_PCR3 <- results(deseq_hydro_PCR1_PCR3, cooksCutoff = FALSE)
res_deseq_hydro_PCR1_PCR3
alpha = 0.001
sigtab_hydro_PCR1_PCR3 = res_deseq_hydro_PCR1_PCR3[which(res_deseq_hydro_PCR1_PCR3$padj < alpha), ]
sigtab_hydro_PCR1_PCR3 = cbind(as(sigtab_hydro_PCR1_PCR3, "data.frame"), as(tax_table(hydro_PCR1_PCR3)[rownames(sigtab_hydro_PCR1_PCR3), ], "matrix"))

# Adding colours as classes
x_hydro_PCR1_PCR3 = tapply(sigtab_hydro_PCR1_PCR3$log2FoldChange, sigtab_hydro_PCR1_PCR3$Pathway_type, function(x) max(x))
x_hydro_PCR1_PCR3 = sort(x_hydro_PCR1_PCR3, TRUE)
sigtab_hydro_PCR1_PCR3$Pathway_type = factor(as.character(sigtab_hydro_PCR1_PCR3$Pathway_type), levels=sort(names(x_hydro_PCR1_PCR3)))
#Points as descriptions
x_hydro_PCR1_PCR3 = tapply(sigtab_hydro_PCR1_PCR3$log2FoldChange, sigtab_hydro_PCR1_PCR3$description, function(x) max(x))
x_hydro_PCR1_PCR3 = sort(x_hydro_PCR1_PCR3, TRUE)
sigtab_hydro_PCR1_PCR3$description = factor(as.character(sigtab_hydro_PCR1_PCR3$description), levels=names(x_hydro_PCR1_PCR3))

diff_abun_hydro_EXP15_EXP5<-ggplot(sigtab_hydro_PCR1_PCR3, aes(x=description, y=log2FoldChange, color=Pathway_type))+
  geom_point(size=6) + 
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=12, face="italic")) + theme(axis.text=element_text(size=12, color="black"), axis.title=element_text(size=16)) +
  rremove("grid")+
  ylab("Log2 Fold Change")+
  xlab("Metabolic Pathways")+
  scale_color_brewer(name = "Pathway Class", palette = "Pastel1")+
   labs(y = expression(Log[2]~ "Ratio of Differential Expression"))
diff_abun_hydro_EXP15_EXP5

#ggsave(filename="microbiome_analysis/Final analysis/PICRUSt2 Data/Figures/diff abun pathways hydro PCR3-PCR1.png", height=10, width=12)


# Combining hydro graphs #

a<-diff_abun_hydro_EXP5_REF + rremove("axis.title")
b<-diff_abun_hydro_EXP15_REF + rremove("axis.title")
c<-diff_abun_hydro_EXP15_EXP5 + rremove("axis.title")

ggarrange(a, b, c, nrow=1)

#ggsave(filename="microbiome_analysis/PICRUSt2 Data/Figures/ACWA diff abun pathways taxa facets.png", height=10, width=18)


#### Comparing between Bow River wastewater exposed sites vs non exposed sites ####

#ALDEx2

bow<-subset_samples(picrust_ps, Stream_Type=="Bow River")

aldex2_exposure <- ALDEx2::aldex(data.frame(phyloseq::otu_table(bow)), phyloseq::sample_data(bow)$Exposure, test="t", effect = TRUE, denom="iqlr")

sig_aldex2_exposure <- aldex2_exposure %>%
  rownames_to_column(var = "Pathways") %>%
  filter(wi.eBH < 0.05) %>%
  arrange(effect, wi.eBH) %>%
  dplyr::select(Pathways, diff.btw, diff.win, effect, wi.ep, wi.eBH)
sig_aldex2_exposure <- inner_join(sig_aldex2_exposure, descriptions, by="Pathways")
#No differentially abundant pathways when alpha is 0.01 but 86 when alpha is 0.05. 

sig_aldex2_exposure_1<-inner_join(sig_aldex2_exposure, metacyc_ont, by="Pathways")


aldex_bow = tapply(sig_aldex2_exposure_1$effect, sig_aldex2_exposure_1$description, function(x) max(x))
aldex_bow = sort(aldex_bow, TRUE)
sig_aldex2_exposure_1$description = factor(as.character(sig_aldex2_exposure_1$description), levels=names(aldex_bow))

aldex_bow = tapply(sig_aldex2_exposure_1$effect, sig_aldex2_exposure_1$Pathway_type, function(x) max(x))
aldex_bow = sort(aldex_bow, TRUE)
sig_aldex2_exposure_1$Pathway_type = factor(as.character(sig_aldex2_exposure_1$Pathway_type), levels=sort(names(aldex_bow)))


ggplot(sig_aldex2_exposure_1, aes(x=description, y=effect))+
  geom_point(size=4) + 
  theme_bw()+
  theme(axis.text.x= sort(element_text(angle = 45, hjust = 1, vjust=1, size=14, color="black")), axis.text.y = element_text(size=14, color="black"))+
  #theme(legend.text=element_text(size=6, color="black")) + 
  #theme(legend.key.size = unit(0.5, "cm"), legend.title = element_text(size=8, color="black"),  legend.position="right")+
  guides(color=guide_legend(nrow=30, byrow=TRUE, ncol=1))+
  rremove("grid")+
  labs(y = expression(Log[2]~Difference))+
  xlab("Metabolic Pathways")

#ggsave(filename="Microbiome_ED/PICRUSt2 Data/Figures/ALDEx2 log2 fold changes Bow River Unexposed vs exposed.png", height=14, width=24)


#DESeq2

bow<-subset_samples(picrust_ps, Stream_Type=="Bow River")

dds_bow <- phyloseq_to_deseq2(bow, ~ Exposure)
dds_bow$Exposure <-relevel(dds_bow$Exposure, "Unexposed")

deseq_bow <- DESeq(dds_bow, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_bow <- results(deseq_bow, cooksCutoff = FALSE)
res_deseq_bow
alpha = 0.001
sigtab_bow = res_deseq_bow[which(res_deseq_bow$padj < alpha), ]
sigtab_bow = cbind(as(sigtab_bow, "data.frame"), as(tax_table(bow)[rownames(sigtab_bow), ], "matrix"))

sigtab_bow_class<-rownames_to_column(sigtab_bow, "Pathways")

sigtab_bow_combined<-inner_join(sigtab_bow_class, metacyc_ont, by="Pathways")


x_bow = tapply(sigtab_bow_combined$log2FoldChange, sigtab_bow_combined$description, function(x) max(x))
x_bow = sort(x_bow, TRUE)
sigtab_bow_combined$description = factor(as.character(sigtab_bow_combined$description), levels=names(x_bow))

x_bow = tapply(sigtab_bow_combined$log2FoldChange, sigtab_bow_combined$Pathway_type, function(x) max(x))
x_bow = sort(x_bow, TRUE)
sigtab_bow_combined$Pathway_type = factor(as.character(sigtab_bow_combined$Pathway_type), levels=sort(names(x_bow)))

ggplot(sigtab_bow_combined, aes(x=description, y=log2FoldChange, color=Pathway_type))+
  geom_point(size=6) + 
  theme_bw()+
   guides(fill=guide_legend(nrow=29))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust=1, size=12, face="italic"))+
  theme(legend.text=element_text(size=10, color="black")) + 
  theme(legend.title = element_text(size=12, color="black"), legend.position="right")+
  scale_fill_discrete(name="Phylum")+
  rremove("grid")+
  labs(y = expression(Log[2]~Difference))+
  xlab("Metabolic Pathways")

#ggsave(filename="Microbiome_ED/PICRUSt2 Data/Figures/DESeq2 log2 fold changes Bow River Unexposed vs exposed.png", height=12, width=22)


```

We are getting different results when using ALDEx2 and DESeq2. It's hard to tell which one is better because DESeq2 is prone to false positives but ALDEx2 doesn't have strong statistical power (Nearing et al., 2022; Yang & Chen, 2022). Because we are using a low alpha threshold for DESeq2, I think we can be confident that these are true results and continue to use this method. 