---
title: "Bow River Microbiome Analysis 2022"
author: "Emilie Diesbourg"
date: "01/04/2023"
output: 
  html_document:
    toc: yes
    toc_depth: 2
    df_print: paged
    toc_float: yes
editor_options: 
  chunk_output_type: console
---
# Introduction {.tabset}

This data comes from a study done in 2022 across 5 sites on the Bow River, Calgary AB and 3 microcosm streams from Pine Creek wastewater treatment plant with varying municipal wastewater effluent exposures (mesocosm streams: 0, 5, or 15% effluent flow). The objective of this study is to determine whether there are changes to the microbiome of freshwater macroinvertebrates and riparian Spiders with increased exposure to wastewater. 

## Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
```

## Loading Packages
```{r Load packages}
#setwd("Microbiome_ED")
library(phyloseq) #phyloseq will be the main package used for structuring microbiome data and diversity comparisons
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")
#install.packages("viridis")
library(viridis)

#BiocManager::install("apeglm")
library(apeglm)
library(ggplot2) #For creating graphs
library(dplyr) #Helps with data wrangling 
library(cluster) #Used for cluster analyses
library(grid)
library(ape)
library(FSA)
library(multcomp)
library(MASS)
library(glmmTMB)
library(gridExtra)
library(vegan)
library(tidyverse) #data wrangling
library(microbiome) #For microbiome analyses
#if (!require("BiocManager", quietly = TRUE)) #Installing microbiome package
  #install.packages("BiocManager")
#BiocManager::install("microbiome")
library(PERFect) #Permutation filtration for microbiome data. Used when comparing beta diversities across sites/locations. 
#if(!requireNamespace("BiocManager", quietly = TRUE))
  #install.packages("BiocManager")
#BiocManager::install("PERFect")
library(knitr) #For R Markdown knitting
library(kableExtra)
library(speedyseq)
#if (!requireNamespace("remotes", quietly = TRUE))
  #install.packages("remotes")
#remotes::install_github("mikemc/speedyseq")
library(colorspace)
library(RColorBrewer)
library(picante)
library(ggpubr)
library(data.table)
#devtools::install_github("microsud/microbiomeutilities")
library(microbiomeutilities)
library(imsig)
library(phangorn) #For phylogenetic tree
library(metacoder) 
library(tibble)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("DESeq2")
library(DESeq2)
library(emmeans)
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis", force=TRUE)
library(pairwiseAdonis)
#library(plyr) apparently if you load this after dpylr it messes up the summarize function?

#if( !require(NbClust) ){install.packages("NbClust"); library(NbClust)}
#if( !require(cobiclust) ){install.packages("cobiclust"); library(cobiclust)}
library(NbClust)
library(cobiclust)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("mia")
library(mia)
library(factoextra)
#install.packages(
  #"microViz",
  #repos = c(davidbarnett = "https://david-barnett.r-universe.dev", getOption("repos")))
library(microViz)
library(dendextend) #Formatting dendrograms
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
#install.packages("castor")
library(castor)
#if (!require("BiocManager", quietly = TRUE))
    #install.packages("BiocManager")

#BiocManager::install("decontam")
library(decontam)
```

# Phyloseq object

For a phyloseq object you need 3 pieces of data: 

1. The ASV table with the genetic barcode identifier (the ASV) of each identified bacterial taxa within each insect sample ran (ex. KK100)

2. The taxa table with the genetic barcode of each unique ASV and the respective bacterial classification "Kingdom", "Phylum", "Class", "Order", "Family", "Genus".

3. A metadata file with the sample names (ex. KK100) and any other variables that were measured. Ex. the family of insect or fish organism collected, the site the organism was collected at, the water temperature, the distance to waste water treatment plant, the weight and length of organism, etc.

The phyloseq object will merge all this data together in one object that can be called in by various functions in the phyloseq and microbiome packages. 

```{r Load data and phyloseq object}
#Importing ASV table
ASVseq<-read.csv("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\ASV_data_2022_ED_fulldataset_samplesordered.csv")

#The row names have to be the ASV identifier (genetic barcode) in order for it to be read. The following lines of code are to re-organize the row names
n1 <- ASVseq$X #Currently the ASVs are being called "X" in the first column. We want them to be unnamed in the row names column
ASVseq <- ASVseq[,-1] #This removes the first column of the ASV dataframe
rownames(ASVseq) <- n1 #This moves the ASVs to the rownames column of the dataframe

ASVtable <- as.matrix(ASVseq) #must be a matrix so we convert to a matrix
#View(ASVtable) #The ASVs are now the row names

#Importing taxa table
taxonomy <-read.csv ("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\Taxonomy_ED_2022_fulldataset.csv")

#Using the same steps as above with the ASV table to make the genetic barcode the rowname
n2 <- taxonomy$X
taxonomy <- taxonomy[,-1]
rownames(taxonomy) <- n2

taxa_table<-as.matrix(taxonomy) #Turn into a matrix
#View(taxa_table) #ASVs are now the row names

#Importing metadata file
metadata<-read.csv("C:\\Users\\Emilie\\Documents\\GitHub\\Bow-River-Microbiome-2022\\microbiome_analysis\\Raw data\\metadata_ED_withblanks_noKK2078.csv")

metadata$Site<-factor(metadata$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3")) #Order sites
metadata$Stage<-factor(metadata$Stage, levels=c("Larvae", "Adults", "Spiders"))

#Need to make sample names as row names so we follow the same steps as above. 
n3 <- metadata$Study_ID

# move names to row headings
metadata <- metadata[,-1]
rownames(metadata) <- n3

#View(metadata) #sample names are now row names

#Combining phyloseq object
ASVtable = otu_table(ASVtable, taxa_are_rows = TRUE) #Taxa are rows means that the ASVs of each identified microbe are the row names which we made sure was true
taxa_table = tax_table(taxa_table)

#Make sure the ASV table and taxa table are both matricies before trying to turn into phyloseq object
ps <- phyloseq(ASVtable, sample_data(metadata), taxa_table)
```

# Filtering/pre-processing steps {.tabset}

Experimental samples:

Used package decontam to get rid of contamination from blank samples. Used a prevalence threshold of 0.1 to identify bacteria that were found in high prevalence of blank samples. 

Removing low confidence data: I removed all ASVs identified as Eukaryota and phyla that were identified as <NA>. This removed 39 and 403 taxa respectively. 

Removing low abundance taxa (including singletons and doubletons): I removed taxa whose sums of reads were less than or equal to 5 across all samples.  

Removing low prevalence taxa: I removed taxa that were only found in less than 3 samples (1 or 2). 

Removing low read samples: I removed all samples whose total reads were less than 2000. There was not much difference between a minimum sum of reads as 2000 or 5000 so I decided to use the smaller number of reads (2000) to keep in more samples.

Rarefying: Before statistical analysis (alpha and beta diversity) the reads will be rarefied to the minimum sampling depth to ensure even sampling depth. The minimum even sampling depth is 2048 reads (the lowest read count after filtering). 

In total, 85.3 % of the taxa (ASVs) were filtered out but only 9.5 % of the entire dataset (reads) were filtered out from the original decontaminated dataset. This may have consequences for diversity measures but ensures sequencing errors and contaminants are removed. 


Blank Samples:

Removing low abundance taxa (including singletons and doubletons): I removed taxa whose sums of reads were less than or equal to 10 across all samples. 

Rarefying: Before statistical analysis the reads will be rarefied to the minimum sampling depth to ensure even sampling depth. The minimum even sampling depth is 7946 reads.

In total, 20.7% of the taxa (ASVs) were filtered out but only 0.81% of the entire dataset (reads) were filtered out.

## Decontamination Steps {.tabset}
### Larvae {.tabset}
#### Larvae July
```{r}

#All larvae were collected with tweezers that were rinsed in the DNase/RNase water

#Number of ASVs in the MBDI rinse water from July
KK1952<-subset_samples(ps, Sample_ID=="CUSHBR_MBDI_RINSE_MB_1")
KK1952<-prune_taxa(taxa_sums(KK1952) > 0, KK1952)
ntaxa(KK1952) #85

KK1953<-subset_samples(ps, Sample_ID=="CUSHBR_MBDI_RINSE_MB_2")	
KK1953<-prune_taxa(taxa_sums(KK1953) > 0, KK1953)
ntaxa(KK1953) #78

KK1954<-subset_samples(ps, Sample_ID=="CUSHBR_MBDI_RINSE_MB_3")	
KK1954<-prune_taxa(taxa_sums(KK1954) > 0, KK1954)
ntaxa(KK1954) #25


lar_water_bl<-subset_samples(ps, Blank_collection_method=="Water July")
lar_water_bl_df <- as.data.frame(sample_data(lar_water_bl))
lar_water_bl_df$LibrarySize <- sample_sums(lar_water_bl)
lar_water_bl_df <- lar_water_bl_df[order(lar_water_bl_df$LibrarySize),]
lar_water_bl_df$Index <- seq(nrow(lar_water_bl_df))
ggplot(data=lar_water_bl_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()
#Blank library sizes are very spread out but most of them are lower than experimental sample libraries

sample_data(lar_water_bl)$is.neg <- sample_data(lar_water_bl)$Sample_Type == "Blank"
contamdf.prev.lar.July <- isContaminant(lar_water_bl, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.lar.July$p)
table(contamdf.prev.lar.July$contaminant)
#Threshold of 0.1: FALSE = 31,497, TRUE=59
head(which(contamdf.prev.lar.July$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.lar.July <- prune_taxa(!contamdf.prev.lar.July$contaminant, lar_water_bl)
ps.noncontam.lar.July #31,497 taxa and 122 samples

ps.noncontam.lar.July<-subset_samples(ps.noncontam.lar.July, Sample_Type!="Blank")
nsamples(ps.noncontam.lar.July) #119
```

#### Larvae September
```{r}
#Number of ASVs in the MBDI rinse water from September
KK1955<-subset_samples(ps, Sample_ID=="ACWA_5%_MBDI_RINSE_MB_1")
KK1955<-prune_taxa(taxa_sums(KK1955) > 0, KK1955)
ntaxa(KK1955) #94

KK1956<-subset_samples(ps, Sample_ID=="ACWA_5%_MBDI_RINSE_MB_2")	
KK1956<-prune_taxa(taxa_sums(KK1956) > 0, KK1956)
ntaxa(KK1956) #82

KK1957<-subset_samples(ps, Sample_ID=="ACWA_5%_MBDI_RINSE_MB_3")	
KK1957<-prune_taxa(taxa_sums(KK1957) > 0, KK1957)
ntaxa(KK1957) #150

lar_water_bl_sept<-subset_samples(ps, Blank_collection_method=="Water September")
lar_water_bl_sept_df <- as.data.frame(sample_data(lar_water_bl_sept))
lar_water_bl_sept_df$LibrarySize <- sample_sums(lar_water_bl_sept)
lar_water_bl_sept_df <- lar_water_bl_sept_df[order(lar_water_bl_sept_df$LibrarySize),]
lar_water_bl_sept_df$Index <- seq(nrow(lar_water_bl_sept_df))
ggplot(data=lar_water_bl_sept_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()

sample_data(lar_water_bl_sept)$is.neg <- sample_data(lar_water_bl_sept)$Sample_Type == "Blank"
contamdf.prev.lar.sept <- isContaminant(lar_water_bl_sept, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.lar.sept$p)
table(contamdf.prev.lar.sept$contaminant)
#Threshold of 0.1: FALSE = 31,517, TRUE=39
head(which(contamdf.prev.lar.sept$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.lar.sept <- prune_taxa(!contamdf.prev.lar.sept$contaminant, lar_water_bl_sept)
ps.noncontam.lar.sept #31,534 taxa and 27 samples

ps.noncontam.lar.sept<-subset_samples(ps.noncontam.lar.sept, Sample_Type!="Blank")
nsamples(ps.noncontam.lar.sept) #39
```

### Adults {.tabset}
#### Adults July
```{r}
#Adults collected in July came into contact with the bedsheet
#Sheet blanks from Cochrane and Policeman Flats
KK2072<-subset_samples(ps, Sample_ID=="COCH_MB_SHEET_BL_1")	
KK2072<-prune_taxa(taxa_sums(KK2072) > 0, KK2072)
ntaxa(KK2072) #233

KK2073<-subset_samples(ps, Sample_ID=="COCH_MB_SHEET_BL_2")
KK2073<-prune_taxa(taxa_sums(KK2073) > 0, KK2073)
ntaxa(KK2073) #201

KK2074<-subset_samples(ps, Sample_ID=="POLFLT_MB_SHEET_BL_1")	
KK2074<-prune_taxa(taxa_sums(KK2074) > 0, KK2074)
ntaxa(KK2074) #365

KK2075<-subset_samples(ps, Sample_ID=="POLFLT_MB_SHEET_BL_2")	
KK2075<-prune_taxa(taxa_sums(KK2075) > 0, KK2075)
ntaxa(KK2075) #314

#Subset adults collected from July with the bed sheet and UV light emergence trap
ad_sheet_bl<-subset_samples(ps, Blank_collection_method=="Sheet")
ad_sheet_bl_df <- as.data.frame(sample_data(ad_sheet_bl))
ad_sheet_bl_df$LibrarySize <- sample_sums(ad_sheet_bl)
ad_sheet_bl_df <- ad_sheet_bl_df[order(ad_sheet_bl_df$LibrarySize),]
ad_sheet_bl_df$Index <- seq(nrow(ad_sheet_bl_df))
ggplot(data=ad_sheet_bl_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()

sample_data(ad_sheet_bl)$is.neg <- sample_data(ad_sheet_bl)$Sample_Type == "Blank"
contamdf.prev.ad.sheet <- isContaminant(ad_sheet_bl, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.ad.sheet$p)
table(contamdf.prev.ad.sheet$contaminant)
#Threshold of 0.1: FALSE = 31,231, TRUE=325
head(which(contamdf.prev.ad.sheet$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.ad.sheet <- prune_taxa(!contamdf.prev.ad.sheet$contaminant, ad_sheet_bl)
ps.noncontam.ad.sheet #31,231 taxa and 107 samples

ps.noncontam.ad.sheet<-subset_samples(ps.noncontam.ad.sheet, Sample_Type!="Blank")
nsamples(ps.noncontam.ad.sheet) #103? should be 104?

#Subset POLFLT_1_Ad_Caddis_MB_2 : POLFLT_1_Ad_Caddis_MB_8 from this field decontaminated dataframe because these individuals also got contaminated from the extraction reagents during DNA extraction. No other insect samples were contaminated by this. 

contam.caddis<-subset_samples(ps.noncontam.ad.sheet, Sample_ID %in% c("POLFLT_1_Ad_Caddis_MB_2", "POLFLT_1_Ad_Caddis_MB_3", "POLFLT_1_Ad_Caddis_MB_4", "POLFLT_1_Ad_Caddis_MB_5", "POLFLT_1_Ad_Caddis_MB_6", "POLFLT_1_Ad_Caddis_MB_7", "POLFLT_1_Ad_Caddis_MB_8"))

#Separate extraction reagents so I can combine the POLFLT caddis dataframe together
ext.reag<-subset_samples(ps, Blank_collection_method=="Extraction_neg_4")
#Merge them together
polflt.caddis<-merge_phyloseq(contam.caddis, ext.reag)

caddis_bl_df <- as.data.frame(sample_data(polflt.caddis))
caddis_bl_df$LibrarySize <- sample_sums(polflt.caddis)
caddis_bl_df <- caddis_bl_df[order(caddis_bl_df$LibrarySize),]
caddis_bl_df$Index <- seq(nrow(caddis_bl_df))
ggplot(data=caddis_bl_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()

sample_data(polflt.caddis)$is.neg <- sample_data(polflt.caddis)$Sample_Type == "Blank"
contamdf.prev.caddis <- isContaminant(polflt.caddis, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.caddis$p)
table(contamdf.prev.caddis$contaminant)
#Threshold of 0.1: FALSE = 31,523, TRUE=33
head(which(contamdf.prev.caddis$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.caddis <- prune_taxa(!contamdf.prev.caddis$contaminant, polflt.caddis)
ps.noncontam.caddis #31,523 taxa and 9 samples

ps.noncontam.caddis<-subset_samples(ps.noncontam.caddis, Sample_Type!="Blank")
nsamples(ps.noncontam.caddis) #7

#Now we have to remove those caddis individuals from the other ps object (ps.noncontam.ad.sheet) so that there aren't duplicates when we merge them all back together. 

ps.noncontam.ad.sheet <- subset_samples(ps.noncontam.ad.sheet, Sample_ID != "POLFLT_1_Ad_Caddis_MB_2" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_3" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_4" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_5" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_6" & Sample_ID !="POLFLT_1_Ad_Caddis_MB_7" & Sample_ID != "POLFLT_1_Ad_Caddis_MB_8")

nsamples(ps.noncontam.ad.sheet) #96

```

#### Adults September

```{r}
#Mesh blanks
#Adults collected in September came into contact with the mesh from floating emergence traps
KK1958<-subset_samples(ps, Sample_ID=="ACWA_5%_MB_MESH_1")	
KK1958<-prune_taxa(taxa_sums(KK1958) > 0, KK1958)
ntaxa(KK1958) #72

KK1959<-subset_samples(ps, Sample_ID=="ACWA_5%_MB_MESH_2")	
KK1959<-prune_taxa(taxa_sums(KK1959) > 0, KK1959)
ntaxa(KK1959) #78

KK1960<-subset_samples(ps, Sample_ID=="ACWA_5%_MB_MESH_3")	
KK1960<-prune_taxa(taxa_sums(KK1960) > 0, KK1960)
ntaxa(KK1960) #133


ad_mesh_bl<-subset_samples(ps, Blank_collection_method=="Mesh")
ad_mesh_bl_df <- as.data.frame(sample_data(ad_mesh_bl))
ad_mesh_bl_df$LibrarySize <- sample_sums(ad_mesh_bl)
ad_mesh_bl_df <- ad_mesh_bl_df[order(ad_mesh_bl_df$LibrarySize),]
ad_mesh_bl_df$Index <- seq(nrow(ad_mesh_bl_df))
ggplot(data=ad_mesh_bl_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()

sample_data(ad_mesh_bl)$is.neg <- sample_data(ad_mesh_bl)$Sample_Type == "Blank"
contamdf.prev.ad.mesh <- isContaminant(ad_mesh_bl, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.ad.mesh$p)
table(contamdf.prev.ad.mesh$contaminant)
#Threshold of 0.1: FALSE = 31,541, TRUE=15
head(which(contamdf.prev.ad.mesh$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.ad.mesh <- prune_taxa(!contamdf.prev.ad.mesh$contaminant, ad_mesh_bl)
ps.noncontam.ad.mesh #31,541 taxa and 19 samples

ps.noncontam.ad.mesh<-subset_samples(ps.noncontam.ad.mesh, Sample_Type!="Blank")
nsamples(ps.noncontam.ad.mesh) #16
```

### Buffer tubes
```{r}
buffer_bl<-subset_samples(ps, Blank_collection_method %in% c("Buffer", "Extraction_neg"))
buffer_bl_df <- as.data.frame(sample_data(buffer_bl))
buffer_bl_df$LibrarySize <- sample_sums(buffer_bl)
buffer_bl_df <- buffer_bl_df[order(buffer_bl_df$LibrarySize),]
buffer_bl_df$Index <- seq(nrow(buffer_bl_df))
ggplot(data=buffer_bl_df, aes(x=Index, y=LibrarySize, color=Blank_collection_method)) + geom_point()

sample_data(buffer_bl)$is.neg <- sample_data(buffer_bl)$Blank_collection_method == "Extraction_neg"
contamdf.prev.buffer <- isContaminant(buffer_bl, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.buffer$p)
table(contamdf.prev.buffer$contaminant)
#Threshold of 0.1: FALSE = 31,543, TRUE=13
head(which(contamdf.prev.buffer$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.buffer <- prune_taxa(!contamdf.prev.buffer$contaminant, buffer_bl)
ps.noncontam.buffer #31,543 taxa and 4 samples

#Only want to keep the buffer blanks not the extraction negs
ps.noncontam.buffer<-subset_samples(ps.noncontam.buffer, Blank_collection_method!="Extraction_neg")
nsamples(ps.noncontam.buffer) #2
```

### Merge all the decontaminated phyloseqs back together
```{r}

spi_ps<-subset_samples(ps, Order=="Araneae")

nsamples(ps.noncontam.lar.July) #119
nsamples(ps.noncontam.lar.sept) #39
nsamples(ps.noncontam.ad.sheet) #96
nsamples(ps.noncontam.ad.mesh) #16
nsamples(ps.noncontam.caddis) #7
nsamples(ps.noncontam.buffer) #2
nsamples(spi_ps) #80

decontam_ps<-merge_phyloseq(ps.noncontam.lar.July, ps.noncontam.lar.sept, ps.noncontam.ad.sheet, ps.noncontam.ad.mesh, ps.noncontam.buffer, ps.noncontam.caddis, spi_ps)
ntaxa(decontam_ps) #31,556
nsamples(decontam_ps) #359. 357 samples and 2 buffer tubes

#Remove contaminants in all samples due to buffer tubes
sample_contam_df <- as.data.frame(sample_data(decontam_ps))
sample_contam_df$LibrarySize <- sample_sums(decontam_ps)
sample_contam_df <- sample_contam_df[order(sample_contam_df$LibrarySize),]
sample_contam_df$Index <- seq(nrow(sample_contam_df))
ggplot(data=sample_contam_df, aes(x=Index, y=LibrarySize, color=Sample_Type)) + geom_point()

sample_data(decontam_ps)$is.neg <- sample_data(decontam_ps)$Sample_Type == "Blank"
contamdf.prev.sample_contam <- isContaminant(decontam_ps, method="prevalence", neg="is.neg", threshold=0.1)
hist(contamdf.prev.sample_contam$p)
table(contamdf.prev.sample_contam$contaminant)
#Threshold of 0.1: FALSE = 31,209, TRUE=347
head(which(contamdf.prev.sample_contam$contaminant)) 

#Prune samples to get rid of contaminants and continue with filtering steps
ps.noncontam.sample.contam <- prune_taxa(!contamdf.prev.sample_contam$contaminant, decontam_ps)
ps.noncontam.sample.contam #31,209 taxa and 359 samples

#Only want to keep the buffer blanks not the extraction negs
ps.noncontam.sample.contam<-subset_samples(ps.noncontam.sample.contam, Sample_Type!="Blank")
nsamples(ps.noncontam.sample.contam) #357
```

## Experimental Samples {.tabset}
### Removing low confidence taxa
``` {r}
#Subsetting experimental samples only from the decontaminanted phyloseq object
psexp<-subset_samples(ps.noncontam.sample.contam, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,181 taxa and 357 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
table(tax_table(ps0)[,"Phylum"],exclude=NULL)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs
ps1 #29.739 taxa, 357 samples
```

### Filtering by abundance and prevalence threshold

```{r}
prevdf=apply(X=otu_table(ps1),MARGIN=ifelse(taxa_are_rows(ps1),yes=1,no=2),FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf= data.frame(Prevalence= prevdf,TotalAbundance=taxa_sums(ps1),tax_table(ps1))

plyr::ddply(prevdf,"Phylum",function(df1){cbind(mean(df1$Prevalence),sum(df1$TotalAbundance))}) 
#Shows mean and total prevalence of taxa within each phylum. The ones that are only prevalent once in each phylum might be worth removing. This means they only showed up in one sample across all samples but might have a high abundance (total number of reads).  

#There are a couple of phyla with only a prevalence sum of one or two so these might be worth removing. 

ps1.dt.taxa = data.table(tax_table(ps1), OTUabundance = taxa_sums(ps1), OTU = taxa_names(ps1)) 
ps1.cumsum = ps1.dt.taxa[, .N, by = OTUabundance]
setkey(ps1.cumsum, OTUabundance)
ps1.cumsum[, CumSum := cumsum(N)]
# Define the plot
ps1.cumsum.plot = ggplot(ps1.cumsum, aes(OTUabundance, CumSum)) + 
  geom_point() +
  xlab("Filtering Threshold, Minimum Total Counts") +
  ylab("OTUs Filtered") +
  ggtitle("OTUs that would be filtered vs. the minimum count threshold") + theme_bw()

print(ps1.cumsum.plot) #Difficult to see the smaller values so we need to zoom in

ps1.cumsum.plot.zoom <- ps1.cumsum.plot + xlim(0, 50) + theme_bw() 

print(ps1.cumsum.plot.zoom) 
#Could maybe remove taxa with an abundance of less than 5-6 across all samples however if I go higher than that it filters out a ton of taxa (about 1/3 of all of them)

#I will filter out taxa that are only prevalent in 1% of all samples (has to be in at least 4/357 samples)
ps2<-phyloseq_filter_prevalence(ps1, prev.trh = 0.01, abund.trh = 6, threshold_condition = "AND")
ps2 #4434 taxa, 357 samples. Got rid of 25,227 ASVs
any(taxa_sums(ps2) <6) #FALSE. Means it did a good job of getting rid of them. 
```

### Filtering by a minimum total sample count (at least 2000 reads/sample)

```{r}
#Creating a rarefaction curve
tab <- otu_table(ps2)
class(tab) <- "matrix" # as.matrix() will do nothing
## you get a warning here, but this is what we need to have
tab <- t(tab) # transpose observations to rows
rare <- rarecurve(tab, step=5000, lwd=2, ylab="OTU",  label=F) 
#The rarefaction curves are very spread out. This means that there are different sequencing depths and standardizing the data will be required for analysis. 

sort(sample_sums(ps2)) #minimum is 24, maximum is 222,584
mean(sample_sums(ps2)) #31,088

#Final dataset
sample_ps <- subset_samples(ps2, sample_sums(ps2) > 2000) 
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2048 reads. Got rid of 38 samples (319 samples left).
sample_ps #4434 taxa, 319 samples. 

#Excel file with samples that were pruned out
pruned_df<-psmelt(subset_samples(ps2, sample_sums(ps2) < 2000))
#write.csv(pruned_df, file="Filtering phyloseq/Samples pruned from 2000 reads per sample.csv")
```

### Amount of data that was filtered out from the original phyloseq object

```{r}
ntaxa(psexp)-ntaxa(sample_ps) #filtered 25,747 total taxa (ASVs)
(1-ntaxa(sample_ps)/ntaxa(psexp))*100 #85.3% of the taxa were removed from the original decontaminated dataset. 

reads_per_OTU <- taxa_sums(psexp) #Original ps with no filtering done.
print(sum(reads_per_OTU)) #Total number of reads is 12222269.

reads_per_OTU_filtered <- taxa_sums(sample_ps) #Original ps with no filtering done.
print(sum(reads_per_OTU_filtered)) #Total number of reads is now 11065513.

(1-11065513/12222269)*100 #9.5%. Filtered out a smaller percentage of the total data (reads). 

#From this we can conclude that a small part of the data (~10%) can have a significant effect on downstream analyses (~85% of taxa removed), if  methods that use ASVs instead of the information that is contained in the sequences is implemented, such as Unifrac and phylogenetic diversity.


#Creating a distribution of reads per sample versus the sample counts. 
ps1_df= data.table(as(sample_data(sample_ps), "data.frame"), Reads_per_sample = sample_sums(sample_ps), keep.rownames = TRUE)
ps1_df_plot = ggplot(ps1_df, aes(Reads_per_sample)) + geom_histogram() + ggtitle("Distribution of reads per sample") + ylab("Sample counts") 

print(ps1_df_plot) #Looks right skewed. A couple of very large reads per sample. Probably the perlidae individuals because they are so big compared to some of the other samples like chironomids. Most are around 20,000 - 40,000 reads per sample though.

#Now we need to observe the counts of abundance of unique sequences across samples
ps1.dt.taxa = data.table(tax_table(sample_ps), OTUabundance = taxa_sums(sample_ps), OTU = taxa_names(sample_ps)) 
#makes a data table of the sum of abundance of each taxa vs the total number of taxa (ASVs)
ps1.dt.tax.plot<- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram() + ggtitle("Histogram of OTU (unique sequence) counts") + theme_bw() #Plots the number of times an ASV is repeated from the reads (abundance) on the x axis and the number of taxa on the y axis
print(ps1.dt.tax.plot)
#You can"t see much on the plots panel in R

#write.table(ps1.dt.taxa, "Filtering phyloseq/ps1_dt_tax_plot.txt", sep = "\t") #makes a table that can open in excel

#ggsave("Filtering phyloseq/Histogram of OTU counts.pdf", height = 6, width = 7) #pdf of the plot

#This plot shows that there is a very small proportion of ASVs with a really high abundance and most have relatively low abundance. Highly right skewed. We want to see how many have very low abundance because this may indicate that further processing/filtering should be done. 

#Zoom in on the lower abundance counts to get a better idea of its distribution
plot.zoom1 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 100, by =10)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom1)
#ggsave("Filtering phyloseq/Histogram of OTU counts 1 to 100.pdf", height = 6, width = 7)
#A very high proportion of ASVs have an abundance of less than 10. This may indicate that they should be filtered out. 

plot.zoom2 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 10, by =2)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom2)
#ggsave("Filtering phyloseq/Histogram of OTU counts 1 to 10.pdf", height = 6, width = 7)
```

## Blank Samples
```{r}
#Subsetting experimental samples only from the original phyloseq object
psbl<-subset_samples(ps, Sample_Type != "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psbl<-prune_taxa(taxa_sums(psbl) > 0, psbl)
psbl #1809 taxa and 23 samples

#Removing low confidence data (where phylum could not be assigned)
table(tax_table(psbl)[,"Phylum"],exclude=NULL) #No taxa assigned as a phyla of NA
#Seems like some of these phyla with a prevalence of 1 also only showed up once in the experimental samples. These ones may need to be filtered out of the experimental samples if it originated just from the sheet/mesh/water blanks.

#Getting rid of singletons and doubletons
any(taxa_sums(psbl)>=2) #TRUE. Means there are ASVs with a sum of sequences less than 2 across all samples. 
psbl0 <- prune_taxa(taxa_sums(psbl) > 2, psbl) #Getting rid of singletons and doubletons. (Taxa that only showed up in one or two sequences across all samples)
any(taxa_sums(psbl0) <2) #FALSE. Means they were removed. 
ntaxa(psbl)- ntaxa(psbl0) #Got rid of 44 ASVs. 

#How many bacterial taxa do we have/did we remove
ntaxa(psbl) #1809 before any filtering was done.
ntaxa(psbl0) #1765 after sum of reads less than 2 were removed (44 removed).


# Compute prevalence of each feature, store as data.frame
prevdf=apply(X=otu_table(psbl0),MARGIN=ifelse(taxa_are_rows(psbl0),yes=1,no=2),FUN = function(x){sum(x > 0)})
# Add taxonomy and total read counts to this data.frame
prevdf= data.frame(Prevalence= prevdf,TotalAbundance=taxa_sums(psbl0),tax_table(psbl0))

plyr::ddply(prevdf,"Phylum",function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))}) 
#The low prevalence phyla showed up in the experimental samples as well. Could be due to sequencing errors, contamination during extraction, or real observations in the blanks. Not sure. 

plyr::ddply(prevdf, "Phylum", function(df1){
  data.frame(mean_prevalence=mean(df1$Prevalence),total_abundance=sum(df1$TotalAbundance,na.rm = T),stringsAsFactors = F)
  })

#Creating a rarefaction curve
tab <- otu_table(psbl0)
class(tab) <- "matrix" # as.matrix() will do nothing
## you get a warning here, but this is what we need to have
tab <- t(tab) # transpose observations to rows
library(vegan)
rare <- rarecurve(tab, step=5000, lwd=2, ylab="OTU",  label=F) 

sort(sample_sums(psbl0)) #minimum is 16, maximum is 63,491
mean(sample_sums(psbl0)) #mean is 21,888

#Remove samples with a sequencing depth less than 7000 (most of the laboratory blanks)
psbl1<-subset_samples(psbl0, sample_sums(psbl0) > 7000) 
sort(sample_sums(psbl1)) #lowest is 7949 reads.

#How much data did we filter out from the original phyloseq object?

ntaxa(psbl)-ntaxa(psbl1) #filtered 44 total taxa (ASVs)
(1-ntaxa(psbl1)/ntaxa(psbl))*100 #2.43% of the taxa were removed from the original dataset. 

reads_per_OTU <- taxa_sums(psbl) #Original ps with no filtering done.
print(sum(reads_per_OTU)) #Total number of reads is 503,506.

reads_per_OTU_filtered <- taxa_sums(psbl1) #Original ps with no filtering done.
print(sum(reads_per_OTU_filtered)) #Total number of reads is now 501,503.

(1-501503/503506)*100 #0.40%. Only filtered out a very small percentage of the total data (reads). 


#Creating a distribution of reads per sample versus the sample counts. 
ps1_df= data.table(as(sample_data(psbl1), "data.frame"), Reads_per_sample = sample_sums(psbl1), keep.rownames = TRUE)
ps1_df_plot = ggplot(ps1_df, aes(Reads_per_sample)) + geom_histogram() + ggtitle("Distribution of reads per sample") + ylab("Sample counts") 

print(ps1_df_plot) #All over the place. Not really any sequencing depth that is consistent across all samples

#Now we need to observe the counts of abundance of unique sequences across samples
ps1.dt.taxa = data.table(tax_table(psbl1), OTUabundance = taxa_sums(psbl1), OTU = taxa_names(psbl1)) 
#makes a data table of the sum of abundance of each taxa vs the total number of taxa (ASVs)
ps1.dt.tax.plot<- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram() + ggtitle("Histogram of OTU (unique sequence) counts") + theme_bw() #Plots the number of times an ASV is repeated from the reads (abundance) on the x axis and the number of taxa on the y axis
print(ps1.dt.tax.plot)
#You can"t see much on the plots panel in R but not a very high abundance in most samples. One sample has a really high abundance.

#This plot shows that there is a very small proportion of ASVs with a really high abundance and most have relatively low abundance. Highly right skewed. We want to see how many have very low abundance because this may indicate that further processing/filtering should be done. 

#Zoom in on the lower abundance counts to get a better idea of its distribution
plot.zoom1 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 100, by =10)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom1)

#A small proportion of ASVs have an abundance of less than 10. These may be able to be filtered out. 

plot.zoom2 <- ggplot(ps1.dt.taxa, aes(OTUabundance)) + geom_histogram(breaks=seq(0, 10, by =2)) + ggtitle("Histogram of Total Counts") + theme_bw()
print(plot.zoom2)

#Only about 200 taxa have a prevalence below 10. 

#Finding total number of reads vs # of reads with low abundance counts
reads_per_OTU <- taxa_sums(psbl1)
print(sum(reads_per_OTU)) #Total number of reads is 501,503.

#How many taxa (ASVs) contain less than 10 reads (an abundance of 10 or less)?
print(length(reads_per_OTU[reads_per_OTU <=10])) #330. 

#How many reads do these 330 ASVs contain in total?
print(sum(reads_per_OTU[reads_per_OTU <=10])) #2066

#What percentage of the total reads is that?
print((sum(reads_per_OTU[reads_per_OTU <=10])/sum(reads_per_OTU))*100) #0.41 %. A very small proportion of the entire dataset. 

#How many ASVs out of the total number of ASVs contains less than or equal to 10 reads?
ntaxa(psbl1) #1765

print((330/1765)*100) #18.7% 


#Plotting the number of ASVs that would be filtered out given a minimum read threshold (min abundance of ASVs)
ps1.dt.taxa = data.table(tax_table(psbl1), OTUabundance = taxa_sums(psbl1), OTU = taxa_names(psbl1)) 
ps1.cumsum = ps1.dt.taxa[, .N, by = OTUabundance]
setkey(ps1.cumsum, OTUabundance)
ps1.cumsum[, CumSum := cumsum(N)]
# Define the plot
ps1.cumsum.plot = ggplot(ps1.cumsum, aes(OTUabundance, CumSum)) + 
  geom_point() +
  xlab("Filtering Threshold, Minimum Total Counts") +
  ylab("OTUs Filtered") +
  ggtitle("OTUs that would be filtered vs. the minimum count threshold") + theme_bw()

print(ps1.cumsum.plot) #Difficult to see the smaller values so we need to zoom in

ps1.cumsum.plot.zoom <- ps1.cumsum.plot + xlim(0, 50) + theme_bw() 

print(ps1.cumsum.plot.zoom) 
#Could maybe remove taxa with an abundance of less than 10 across all samples

#Can also check the coefficient of variation for spurious observations. *How do you tell which ones are spurious observations?*
ps1.rel <- microbiome::transform(psbl1, "compositional") # transform to relative abundance
p <- plot_taxa_cv(ps1.rel, plot.type="scatter")
print(p) #The low abundance ASVs have a very high coefficient of variation which may affect the differential abundance testing. 

#I will filter out taxa with an abundance of 10 or less. 


blank_ps <- prune_taxa(taxa_sums(psbl1) > 10, psbl1)
blank_ps<- prune_taxa(taxa_sums(blank_ps) > 0, blank_ps)
blank_ps #1435 taxa instead of 1809. Removed 374 ASVs in total.
any(taxa_sums(blank_ps)<=10) #FALSE. Removed them adequately. 

sort(sample_sums(blank_ps)) #Lowest was 7946, highest was 63246. Seems pretty high for a blank sample but not too variable.
mean(sample_sums(blank_ps)) #mean is 29378

#How much data did we filter out from the original phyloseq object?

ntaxa(psbl)-ntaxa(blank_ps) #filtered 374 total taxa (ASVs)
(1-ntaxa(blank_ps)/ntaxa(psbl))*100 #20.7% of the taxa were removed from the original dataset. 

reads_per_OTU <- taxa_sums(psbl) #Original ps with no filtering done.
print(sum(reads_per_OTU)) #Total number of reads is 503506

reads_per_OTU_filtered <- taxa_sums(blank_ps) #Original ps with no filtering done.
print(sum(reads_per_OTU_filtered)) #Total number of reads is now 499437.

(1-499437/503506)*100 #0.81%. Only filtered out a small percentage of the total data (reads). 


```

## Finding a good prevalence threshold
```{r}

##Prevalence threshold pruning ##
# Define prevalence of each taxa (in how many samples did each taxa appear at least once)
prev0 = apply(X = otu_table(sample_ps),
                MARGIN = ifelse(taxa_are_rows(sample_ps), yes = 1, no = 2),
                FUN = function(x){sum(x > 0)})
prevdf = data.frame(Prevalence = prev0,
                      TotalAbundance = taxa_sums(sample_ps),
                      tax_table(sample_ps))
keepPhyla = table(prevdf$Phylum)[(table(prevdf$Phylum) > 2)] #Keeping phyla with a prevalence greater than 2
prevdf1 = subset(prevdf, Phylum %in% names(keepPhyla))
# Define prevalence threshold as 5% of total samples
prevalenceThreshold = 0.02 * nsamples(sample_ps)
prevalenceThreshold #We are keeping phyla that occur at least 6.88 samples (2% of samples) with a prevalence of 2 or greater 

psprune = prune_taxa((prev0 > prevalenceThreshold), sample_ps)
psprune #2849 taxa instead of 27,069. Cut out a very large portion of the data. Too high of a threshold.  

reads_pruned <- taxa_sums(psprune)
print(sum(reads_pruned)) #11,306,751 total reads left after pruning
(1-(sum(reads_pruned))/(sum(reads_per_OTU)))*100 #14.55% of the total read data was removed by pruning. A decently large portion of the data was cut out from additional pruning

ggplot(prevdf1, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.7) +
  scale_y_log10() + scale_x_log10() +
  xlab("Total Abundance") +
  facet_wrap(~Phylum)
#Prevalence threshold looks too high, cutting out a lot of the variation in microbial communities. I will try a lower threshold value. 


#Lowering the prevalence threshold to 0.2%

# Define prevalence of each taxa
# (in how many samples did each taxa appear at least once)
prev1 = apply(X = otu_table(sample_ps),
                MARGIN = ifelse(taxa_are_rows(sample_ps), yes = 1, no = 2),
                FUN = function(x){sum(x > 0)})
prevdf2 = data.frame(Prevalence = prev1,
                      TotalAbundance = taxa_sums(sample_ps),
                      tax_table(sample_ps))
keepPhyla1 = table(prevdf2$Phylum)[(table(prevdf2$Phylum) > 2)]
prevdf3 = subset(prevdf2, Phylum %in% names(keepPhyla))
# Define prevalence threshold as 5% of total samples
prevalenceThreshold1 = 0.01 * nsamples(sample_ps)
prevalenceThreshold1 #3.44. 

psprune1 = prune_taxa((prev1 > prevalenceThreshold1), sample_ps)
psprune1 #4614 taxa instead of 27,069. Got rid of 22455 ASVs. Still a very large percentage of taxa removed.

reads_pruned1 <- taxa_sums(psprune1)
print(sum(reads_pruned1)) #13 231 428 total reads left after pruning
(1-(sum(reads_pruned1))/(sum(reads_per_OTU)))*100 #0% of data was removed. Why did it remove 297 ASVs but none of the reads??

ggplot(prevdf2, aes(TotalAbundance, Prevalence, color = Phylum)) +
  geom_hline(yintercept = prevalenceThreshold1, alpha = 0.5, linetype = 2) +
  geom_point(size = 2, alpha = 0.7) +
  scale_y_log10() + scale_x_log10() +
  xlab("Total Abundance") +
  facet_wrap(~Phylum)
#This threshold of filtering looks a lot better and I can be more confidence that we are not removing true taxa from samples. Looks like the ones that are removed are just artifacts of sampling or sequencing errors. 
```

## Simplified filtering steps for reloading data quickly
```{r}

psexp<-subset_samples(ps.noncontam.sample.contam, Sample_Type == "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psexp<-prune_taxa(taxa_sums(psexp) > 0, psexp)
psexp #30,181 taxa and 357 samples

#Removing Eukaryotes
ps0 <- subset_taxa(psexp, Kingdom != "Eukaryota")
ntaxa(psexp)-ntaxa(ps0) #Got rid of 39 ASVs

#Removing low confidence data (where phylum could not be assigned)
ps1<-subset_taxa(ps0, !is.na(Phylum) & !Phylum %in% c("","uncharacterized")) #Removes the phyla characterized as NA
ntaxa(ps0) - ntaxa(ps1) #Got rid of 403 ASVs

ps2<-phyloseq_filter_prevalence(ps1, prev.trh = 0.01, abund.trh = 6, threshold_condition = "AND")
ps2 #4434 taxa, 357 samples. 
any(taxa_sums(ps2) <6) #FALSE. Means it did a good job of getting rid of them. 

sample_ps <- subset_samples(ps2, sample_sums(ps2) > 2000) 
sort(sample_sums(sample_ps)) #Minimum sampling depth is now 2048 reads. Got rid of 38 samples (319 samples left).
sample_ps #4434 taxa, 319 samples. 
psexp #Got rid of 85.3 % of ASVs

reads_per_OTU_ps <- taxa_sums(sample_ps) 
print(sum(reads_per_OTU_ps)) #11,065,513 reads. 
#Removed ~10% of the experimental samples after decontamination. 
reads_per_OTU_expps <- taxa_sums(psexp) 
print(sum(reads_per_OTU_expps)) #12,222,269
#removed 9.5% of reads


#Blank samples

psbl<-subset_samples(ps, Sample_Type != "Sample") #Removing ASVs where the abundance is 0 (after removing the blank samples)
psbl<-prune_taxa(taxa_sums(psbl) > 0, psbl)
psbl

any(taxa_sums(psbl)>=2) #TRUE. Means there are ASVs with a sum of sequences less than 2 across all samples. 
psbl0 <- prune_taxa(taxa_sums(psbl) > 2, psbl) #Getting rid of singletons and doubletons. (Taxa that only showed up in one or two sequences across all samples)
any(taxa_sums(psbl0) <2) #FALSE. Means they were removed. 
ntaxa(psbl)- ntaxa(psbl0) #Got rid of 32 ASVs. 

#How many bacterial taxa do we have/did we remove
ntaxa(psbl) #1809 before any filtering was done.
ntaxa(psbl0) #1765 after sum of reads less than 2 were removed (32 removed).

psbl1<-subset_samples(psbl0, sample_sums(psbl0) > 7000) 
sort(sample_sums(psbl1)) #lowest is 7949 reads.

blank_ps <- prune_taxa(taxa_sums(psbl1) > 10, psbl1)
blank_ps<- prune_taxa(taxa_sums(blank_ps) > 0, blank_ps)
blank_ps #1435 taxa instead of 1809. 
```

# Data diagnostics

In this section I filter out different taxonomic organizations of bacterial taxa to see the abundance at different levels of organization. Ex. how many phyla per site, what are the top phyla, families, genera and their percentage of the total. 

## Blank Samples (not decontaminated)
```{r}
blank_ps<-subset_samples(ps, Sample_Type=="Blank")
blank_ps<-prune_taxa(taxa_sums(blank_ps) > 0, blank_ps)
ntaxa(blank_ps) #1809

sort(sample_sums(blank_ps)) # ranges from 16 - 63,500 counts

#The extraction reagent blanks from the first 4 plates are relatively low (16 - 629 counts)
#The extraction reagent blanks from the last plate is very high (44,645 and 57,629 counts)
#Buffer tube blanks were run on the last (contaminated) plate and have 56,500 and 63,500 counts


#Percent of each phyla to the total bacteria
sort((table(tax_table(blank_ps)[,2]))/ntaxa(blank_ps)*100)
#Firmicutes (42.3 %), Proteobacteria (19.6 %), Bacteroidota (13.4 %)

#Percent of each class to the total bacteria
sort((table(tax_table(blank_ps)[,3]))/ntaxa(blank_ps)*100)
#Clostridia (35.2 %), Bacteroidia (13.3 %), Alphaproteobacteria (10.1)

#Percent of each order to the total bacteria
sort((table(tax_table(blank_ps)[,4]))/ntaxa(blank_ps)*100)
#Lachnospirales (15.0 %), Oscillospirales (14.2 %), Burkholderiales (6.5 %)

#Percent of each family to the total bacteria
sort((table(tax_table(blank_ps)[,5]))/ntaxa(blank_ps)*100)
#Lachnospiraceae (14.9 %), Ruminococcaceae (6.9 %), Oscillospiraceae (5.5 %)

#Percent of each genus to the total bacteria
sort((table(tax_table(blank_ps)[,6]))/ntaxa(blank_ps)*100)
#Faecalibacterium (1.99%), Ruminococcus torques group (1.8 %), Blautia (1.6 %)

#Percent of each species to the total bacteria
sort((table(tax_table(blank_ps)[,7]))/ntaxa(blank_ps)*100)
#prausnitzii (1.0 %), bacterium (0.44%), hadrus (0.39%)

##Get total numbers of ASVs at each level
get_taxa_unique(blank_ps, "Phylum") #26 phyla total
get_taxa_unique(blank_ps, "Class") #57 Classes total
get_taxa_unique(blank_ps, "Order") #136 orders total
get_taxa_unique(blank_ps, "Family") #199 families total
get_taxa_unique(blank_ps, "Genus") #400 genera total

blank_phy <- tax_glom(blank_ps, taxrank="Phylum")
blank_rel_phy <- microbiome::transform(blank_phy, "compositional")

blank_rel_phy_df <- psmelt(blank_rel_phy)

blank_phy_agg <- aggregate(Abundance ~ Phylum + Blank_Type, data = blank_rel_phy_df, FUN = mean) 

blank_phy_plot <- ggplot(blank_phy_agg, aes(x=Blank_Type, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  rremove("grid")
blank_phy_plot

#Family
blank_fam <- tax_glom(blank_ps, taxrank="Family")
blank_rel_fam <- microbiome::transform(blank_fam, "compositional")
blank_rel_fam_prune = prune_taxa(taxa_sums(blank_rel_fam) > 0.05, blank_rel_fam)

blank_rel_fam_df <- psmelt(blank_rel_fam_prune)

blank_fam_agg <- aggregate(Abundance ~ Family + Blank_Type, data = blank_rel_fam_df, FUN = mean) 

blank_fam_plot <- ggplot(blank_fam_agg, aes(x=Blank_Type, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  rremove("grid")
blank_fam_plot

#Genus
blank_genus <- tax_glom(blank_ps, taxrank="Genus")
blank_rel_genus <- microbiome::transform(blank_genus, "compositional")
blank_rel_genus_prune = prune_taxa(taxa_sums(blank_rel_genus) > 0.1, blank_rel_genus)

blank_rel_genus_df <- psmelt(blank_rel_genus_prune)

blank_genus_agg <- aggregate(Abundance ~ Genus + Blank_Type, data = blank_rel_genus_df, FUN = mean) 

blank_genus_plot <- ggplot(blank_genus_agg, aes(x=Blank_Type, y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)+
  rremove("grid")
blank_genus_plot
```

## Experimental Samples (before decontamination or filtering)
```{r}
sample_ps_contam<-subset_samples(ps, Sample_Type=="Sample")
sample_ps_contam<-prune_taxa(taxa_sums(sample_ps_contam) > 0, sample_ps_contam)
ntaxa(sample_ps_contam) #30,911

sort(sample_sums(sample_ps_contam)) # ranges from 77 - 222,956 counts

#Percent of each phyla to the total bacteria
sort((table(tax_table(sample_ps_contam)[,2]))/ntaxa(sample_ps_contam)*100)
#Proteobacteria (33.5 %), Bacteroidota (25.5 %), Firmicutes (10.1%)

#Percent of each class to the total bacteria
sort((table(tax_table(sample_ps_contam)[,3]))/ntaxa(sample_ps_contam)*100)
#Bacteroidia (24.9 %), Gammaproteobacteria (18.0%), Alphaproteobacteria (10.1)

#Percent of each order to the total bacteria
sort((table(tax_table(sample_ps_contam)[,4]))/ntaxa(sample_ps_contam)*100)
#Burkholderiales (12.6 %), Chitinophagales (8.5 %), Cytophagales (5.5 %)

#Percent of each family to the total bacteria
sort((table(tax_table(sample_ps_contam)[,5]))/ntaxa(sample_ps_contam)*100)
#Comamonadaceae (5.6%), Chitinophagaceae (3.8%), Saprospiraceae (3.8%)

#Percent of each genus to the total bacteria

sort((table(tax_table(sample_ps_contam)[,6]))/ntaxa(sample_ps_contam)*100)
#Doesn't show all the rows

#Percent of each species to the total bacteria
sort((table(tax_table(blank_ps)[,7]))/ntaxa(blank_ps)*100)
#Doesn't show all the rows

##Get total numbers of ASVs at each level
get_taxa_unique(sample_ps_contam, "Phylum") #54 phyla total
get_taxa_unique(sample_ps_contam, "Class") #133 Classes total
get_taxa_unique(sample_ps_contam, "Order") #300 orders total
get_taxa_unique(sample_ps_contam, "Family") #462 families total
get_taxa_unique(sample_ps_contam, "Genus") #1178 genera total

sample_contam_phy <- tax_glom(sample_ps_contam, taxrank="Phylum")
sample_contam_rel_phy <- microbiome::transform(sample_contam_phy, "compositional")
sample_contam_rel_phy_prune = prune_taxa(taxa_sums(sample_contam_rel_phy) > 0.05, sample_contam_rel_phy)

sample_contam_rel_phy_df <- psmelt(sample_contam_rel_phy_prune)

sample_contam_phy_agg <- aggregate(Abundance ~ Phylum + Site + Taxa, data = sample_contam_rel_phy_df, FUN = mean) 

sample_contam_phy_plot <- ggplot(sample_contam_phy_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity")+
  facet_grid(.~Taxa, scales = "free")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
sample_contam_phy_plot

#Family
sample_contam_fam <- tax_glom(sample_ps_contam, taxrank="Family")
sample_contam_rel_fam <- microbiome::transform(sample_contam_fam, "compositional")
sample_contam_rel_fam_prune = prune_taxa(taxa_sums(sample_contam_rel_fam) > 0.2, sample_contam_rel_fam)

sample_contam_rel_fam_df <- psmelt(sample_contam_rel_fam_prune)

sample_contam_fam_agg <- aggregate(Abundance ~ Family + Site +Taxa, data = sample_contam_rel_fam_df, FUN = mean) 

sample_contam_fam_plot <- ggplot(sample_contam_fam_agg, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity")+
  facet_grid(.~Taxa, scales = "free")+
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=6), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
sample_contam_fam_plot
```

## Experimental samples (after decontamination and filtering) {.tabset}

### Relative abundance at different taxonomic levels (all samples combined)
```{r}
#Experimental Samples
ntaxa(sample_ps) #4434
sort((table(tax_table(sample_ps)[,2]))/ntaxa(sample_ps)*100)
#Top 5 abundant phyla are Proteobacteria (38.0 %), Bacteroidota (31.7 %), Firmicutes (8.9 %), Cyanobacteria (6.2 %), Actinobacteria (5.4 %)

sort((table(tax_table(sample_ps)[,5])/ntaxa(sample_ps)*100))
#Top 5 families are Comamonadaceae (8.1 %), Saprospiraceae (6.8 %), Flavobacteriaceae (4.9 %), Rhodobacteraceae (4.4 %), Sphingomonadaceae (3.6 %)

sort(table(tax_table(sample_ps)[,6]))
#top 5 genera: Flavobacterium (4.6 %), Rhodoferax (2.3 %), Tyzzerella (1.5 %), Pseudorhodobacter (1.4), Limnohabitans (1.2 %)

#Sort by proteobacteria
proteo <- subset_taxa(sample_ps, Phylum=="Proteobacteria")
ntaxa(proteo) #1684
sort(table(tax_table(proteo)[,3])) #class. Only 2 classes. Alphaproteobacteria (48.7%) and Gammaproteobacteria (51.1%)
sort(table(tax_table(proteo)[,5])) #family Top 3: Comamonadaceae (21.2 %), Rhodobacteraceae (11.5 %), Sphingomonadaceae (9.5 %) 
sort(table(tax_table(proteo)[,6])) #genera Top 3: Rhodoferax (5.9 %), Pseudorhodobacter (3.6 %), Limnohabitans (3.3 %)

#Sort by Bacteroidota
bac <- subset_taxa(sample_ps, Phylum=="Bacteroidota")
ntaxa(bac) #1404
sort(table(tax_table(bac)[,3])) #class. Top class: Bacteroidia (97.5 %)
sort(table(tax_table(bac)[,5])) #family Top 3: Saprospiraceae (21.4 %), Flavobacteriaceae (15.5 %), Chitinophagaceae (10.3 %)  
sort(table(tax_table(bac)[,6])) #genera Top 3: Flavobacterium (14.7 %), Hymenobacter (3.0 %), Ferruginibacter (2.8 %)

#Sort by Firmicutes
firm <- subset_taxa(sample_ps, Phylum=="Firmicutes")
ntaxa(firm) #393
sort(table(tax_table(firm)[,3])) #class. Top 2: Clostridia (64.4 %), Bacilli (32.8 %) 
sort(table(tax_table(firm)[,5])) #family Top 2: Lachnospiraceae (25.2 %), Ruminococcaceae (15. 5%) 
sort(table(tax_table(firm)[,6])) #genera Top 3: Tyzzerella (17.0 %), Candidatus Soleaferrea (4.8 %), Staphylococcus (4.3 %)


##Get total numbers of ASVs at each level
get_taxa_unique(sample_ps, "Phylum") #32 phyla total
get_taxa_unique(sample_ps, "Class") #70 Classes total
get_taxa_unique(sample_ps, "Order") #170 orders total
get_taxa_unique(sample_ps, "Family") #241 families total
get_taxa_unique(sample_ps, "Genus") #467 genera total
```

### Number of ASVs per invertebrate taxa

```{r}

#Hydropsychidae
hydro<-subset_samples(sample_ps, Family=="Hydropsychidae")
hydro<-prune_taxa(taxa_sums(hydro) > 0, hydro)
ntaxa(hydro) #2541

#Heptageniidae
hep<-subset_samples(sample_ps, Family=="Heptageniidae ")
hep<-prune_taxa(taxa_sums(hep) > 0, hep)
ntaxa(hep) #2604

#Baetidae
baetid<-subset_samples(sample_ps, Family=="Baetidae")
baetid<-prune_taxa(taxa_sums(baetid) > 0, baetid)
ntaxa(baetid) #1449

#Perlidae
perlid<-subset_samples(sample_ps, Family=="Perlidae")
perlid<-prune_taxa(taxa_sums(perlid) > 0, perlid)
ntaxa(perlid) #1693

#Chironomidae
chiro<-subset_samples(sample_ps, Family=="Chironomidae")
chiro<-prune_taxa(taxa_sums(chiro) > 0, chiro)
ntaxa(chiro) #1349

#Araneidae
aran<-subset_samples(sample_ps, Family=="Araneidae")
aran<-prune_taxa(taxa_sums(aran) > 0, aran)
ntaxa(aran) #548

#Tetragnathidae
tetra<-subset_samples(sample_ps, Family=="Tetragnathidae")
tetra<-prune_taxa(taxa_sums(tetra) > 0, tetra)
ntaxa(tetra) #437

#Ephemeroptera
ephem<-subset_samples(sample_ps, Order=="Ephemeroptera")
ephem<-prune_taxa(taxa_sums(ephem) > 0, ephem)
ntaxa(ephem) #3370

#Tricoptera
tricop<-subset_samples(sample_ps, Order=="Tricoptera")
tricop<-prune_taxa(taxa_sums(tricop) > 0, tricop)
ntaxa(tricop) #3361

#Plecoptera
plec<-subset_samples(sample_ps, Order=="Plecoptera")
plec<-prune_taxa(taxa_sums(plec) > 0, plec)
ntaxa(plec) #1693

#Araneae
Spiderss<-subset_samples(sample_ps, Order=="Araneae")
Spiderss<-prune_taxa(taxa_sums(Spiderss) > 0, Spiderss)
ntaxa(Spiderss) #675
nsamples(Spiderss) #69
```

### Total ASVs per site and invertebrate taxa 

```{r}
#Cochrane, n = 60
coch<-subset_samples(sample_ps, Site=="Cochrane")
coch<-prune_taxa(taxa_sums(coch) > 0, coch)
nsamples(coch) #60
ntaxa(coch) #2323

get_taxa_unique(coch, "Phylum") #25 phyla total
get_taxa_unique(coch, "Class") #54 Classes total
get_taxa_unique(coch, "Order") #125 orders total
get_taxa_unique(coch, "Family") #182 families total
get_taxa_unique(coch, "Genus") #333 genera total

sort((table(tax_table(coch)[,2]))/ntaxa(coch)*100)

#Tricoptera at Cochrane
tricoch<-subset_samples(coch, Order=="Tricoptera")
tricoch<-prune_taxa(taxa_sums(tricoch)>0, tricoch)
ntaxa(tricoch) #1234
nsamples(tricoch) #16

get_taxa_unique(tricoch, "Phylum") #19 phyla total
get_taxa_unique(tricoch, "Class") #38 Classes total
get_taxa_unique(tricoch, "Order") #90 orders total
get_taxa_unique(tricoch, "Family") #128 families total
get_taxa_unique(tricoch, "Genus") #202 genera total

#ephemeroptera at Cochrane
ephcoch<-subset_samples(coch, Order=="Ephemeroptera")
ephcoch<-prune_taxa(taxa_sums(ephcoch)>0, ephcoch)
ntaxa(ephcoch) #1246
nsamples(ephcoch) #16

get_taxa_unique(ephcoch, "Phylum") #19 phyla total
get_taxa_unique(ephcoch, "Class") #41 Classes total
get_taxa_unique(ephcoch, "Order") #91 orders total
get_taxa_unique(ephcoch, "Family") #146 families total
get_taxa_unique(ephcoch, "Genus") #233 genera total

#Diptera at Cochrane
dipcoch<-subset_samples(coch, Order=="Diptera")
dipcoch<-prune_taxa(taxa_sums(dipcoch)>0, dipcoch)
ntaxa(dipcoch) #396
nsamples(dipcoch) #9

get_taxa_unique(dipcoch, "Phylum") #10 phyla total
get_taxa_unique(dipcoch, "Class") #25 Classes total
get_taxa_unique(dipcoch, "Order") #71 orders total
get_taxa_unique(dipcoch, "Family") #97 families total
get_taxa_unique(dipcoch, "Genus") #141 genera total

#Plecoptera at Cochrane
pleccoch<-subset_samples(coch, Order=="Plecoptera")
pleccoch<-prune_taxa(taxa_sums(pleccoch)>0, pleccoch)
ntaxa(pleccoch) #566
nsamples(pleccoch) #8

get_taxa_unique(pleccoch, "Phylum") #12 phyla total
get_taxa_unique(pleccoch, "Class") #21 Classes total
get_taxa_unique(pleccoch, "Order") #50 orders total
get_taxa_unique(pleccoch, "Family") #78 families total
get_taxa_unique(pleccoch, "Genus") #104 genera total

#Spiderss at Cochrane
arancoch<-subset_samples(coch, Order=="Araneae")
arancoch<-prune_taxa(taxa_sums(arancoch)>0, arancoch)
ntaxa(arancoch) #189
nsamples(arancoch) #11

get_taxa_unique(arancoch, "Phylum") #7 phyla total
get_taxa_unique(arancoch, "Class") #10 Classes total
get_taxa_unique(arancoch, "Order") #36 orders total
get_taxa_unique(arancoch, "Family") #58 families total
get_taxa_unique(arancoch, "Genus") #88 genera total

#Sunalta, n = 47
sun<-subset_samples(sample_ps, Site=="Sunalta")
sun<-prune_taxa(taxa_sums(sun) > 0, sun)
ntaxa(sun) #2076
nsamples(sun) #47

get_taxa_unique(sun, "Phylum") #25 phyla total
get_taxa_unique(sun, "Class") #55 Classes total
get_taxa_unique(sun, "Order") #119 orders total
get_taxa_unique(sun, "Family") #175 families total
get_taxa_unique(sun, "Genus") #306 genera total

sort((table(tax_table(sun)[,2]))/ntaxa(sun)*100)

#Tricoptera at sunalta
trisun<-subset_samples(sun, Order=="Tricoptera")
trisun<-prune_taxa(taxa_sums(trisun) > 0, trisun)
ntaxa(trisun) #974
nsamples(trisun) #15

get_taxa_unique(trisun, "Phylum") #20 phyla total
get_taxa_unique(trisun, "Class") #33 Classes total
get_taxa_unique(trisun, "Order") #76 orders total
get_taxa_unique(trisun, "Family") #117 families total
get_taxa_unique(trisun, "Genus") #182 genera total

#Ephem in sunalta
ephsun<-subset_samples(sun, Order=="Ephemeroptera")
ephsun<-prune_taxa(taxa_sums(ephsun) > 0, ephsun)
ntaxa(ephsun) #1429
nsamples(ephsun) #16

get_taxa_unique(ephsun, "Phylum") #19 phyla total
get_taxa_unique(ephsun, "Class") #44 Classes total
get_taxa_unique(ephsun, "Order") #99 orders total
get_taxa_unique(ephsun, "Family") #143 families total
get_taxa_unique(ephsun, "Genus") #241 genera total

#Araneae in sunalta
aransun<-subset_samples(sun, Order=="Araneae")
aransun<-prune_taxa(taxa_sums(aransun) > 0, aransun)
ntaxa(aransun) #284
nsamples(aransun) #16

get_taxa_unique(aransun, "Phylum") #8 phyla total
get_taxa_unique(aransun, "Class") #13 Classes total
get_taxa_unique(aransun, "Order") #42 orders total
get_taxa_unique(aransun, "Family") #60 families total
get_taxa_unique(aransun, "Genus") #100 genera total

#Cushing Bridge, n = 61
cushbr<-subset_samples(sample_ps, Site=="Cushing Bridge")
cushbr<-prune_taxa(taxa_sums(cushbr) > 0, cushbr)
ntaxa(cushbr) #2628
nsamples(cushbr)

get_taxa_unique(cushbr, "Phylum") #29 phyla total
get_taxa_unique(cushbr, "Class") #58 Classes total
get_taxa_unique(cushbr, "Order") #139 orders total
get_taxa_unique(cushbr, "Family") #198 families total
get_taxa_unique(cushbr, "Genus") #371 genera total

sort((table(tax_table(cushbr)[,2]))/ntaxa(cushbr)*100)

#Tricoptera in cushing bridge
tricush<-subset_samples(cushbr, Order=="Tricoptera")
tricush<-prune_taxa(taxa_sums(tricush) > 0, tricush)
ntaxa(tricush) #1079
nsamples(tricush) #13

get_taxa_unique(tricush, "Phylum") #20 phyla total
get_taxa_unique(tricush, "Class") #34 Classes total
get_taxa_unique(tricush, "Order") #79 orders total
get_taxa_unique(tricush, "Family") #128 families total
get_taxa_unique(tricush, "Genus") #206 genera total

#Ephemeroptera in cushing bridge
ephcush<-subset_samples(cushbr, Order=="Ephemeroptera")
ephcush<-prune_taxa(taxa_sums(ephcush) > 0, ephcush)
ntaxa(ephcush) #1258
nsamples(ephcush) #16

get_taxa_unique(ephcush, "Phylum") #20 phyla total
get_taxa_unique(ephcush, "Class") #42 Classes total
get_taxa_unique(ephcush, "Order") #102 orders total
get_taxa_unique(ephcush, "Family") #146 families total
get_taxa_unique(ephcush, "Genus") #247 genera total

#Diptera in cushbr
dipcush<-subset_samples(cushbr, Order=="Diptera")
dipcush<-prune_taxa(taxa_sums(dipcush) > 0, dipcush)
ntaxa(dipcush) #645
nsamples(dipcush) #14

get_taxa_unique(dipcush, "Phylum") #16 phyla total
get_taxa_unique(dipcush, "Class") #32 Classes total
get_taxa_unique(dipcush, "Order") #80 orders total
get_taxa_unique(dipcush, "Family") #123 families total
get_taxa_unique(dipcush, "Genus") #163 genera total

#plecoptera in cush br
pleccush<-subset_samples(cushbr, Order=="Plecoptera")
pleccush<-prune_taxa(taxa_sums(pleccush) > 0, pleccush)
ntaxa(pleccush) #1173
nsamples(pleccush) #5

get_taxa_unique(pleccush, "Phylum") #20 phyla total
get_taxa_unique(pleccush, "Class") #39 Classes total
get_taxa_unique(pleccush, "Order") #88 orders total
get_taxa_unique(pleccush, "Family") #123 families total
get_taxa_unique(pleccush, "Genus") #191 genera total

#araneae in cush br
arancush<-subset_samples(cushbr, Order=="Araneae")
arancush<-prune_taxa(taxa_sums(arancush) > 0, arancush)
ntaxa(arancush) #339
nsamples(arancush) #13

get_taxa_unique(arancush, "Phylum") #11 phyla total
get_taxa_unique(arancush, "Class") #20 Classes total
get_taxa_unique(arancush, "Order") #60 orders total
get_taxa_unique(arancush, "Family") #81 families total
get_taxa_unique(arancush, "Genus") #133 genera total

#Graves Bridge, n = 46
grvbr<-subset_samples(sample_ps, Site=="Graves Bridge")
grvbr<-prune_taxa(taxa_sums(grvbr) > 0, grvbr)
ntaxa(grvbr) #2539
nsamples(grvbr) #46

get_taxa_unique(grvbr, "Phylum") #25 phyla total
get_taxa_unique(grvbr, "Class") #54 Classes total
get_taxa_unique(grvbr, "Order") #135 orders total
get_taxa_unique(grvbr, "Family") #191 families total
get_taxa_unique(grvbr, "Genus") #344 genera total

sort((table(tax_table(grvbr)[,2]))/ntaxa(grvbr)*100)

#Tricoptera at graves bridge
trigrvbr<-subset_samples(grvbr, Order=="Tricoptera")
trigrvbr<-prune_taxa(taxa_sums(trigrvbr) > 0, trigrvbr)
ntaxa(trigrvbr) #1557
nsamples(trigrvbr) #15

get_taxa_unique(trigrvbr, "Phylum") #22 phyla total
get_taxa_unique(trigrvbr, "Class") #42 Classes total
get_taxa_unique(trigrvbr, "Order") #101 orders total
get_taxa_unique(trigrvbr, "Family") #143 families total
get_taxa_unique(trigrvbr, "Genus") #226 genera total

ephgrvbr<-subset_samples(grvbr, Order=="Ephemeroptera")
ephgrvbr<-prune_taxa(taxa_sums(ephgrvbr) > 0, ephgrvbr)
ntaxa(ephgrvbr) #1445
nsamples(ephgrvbr) #16

get_taxa_unique(ephgrvbr, "Phylum") #20 phyla total
get_taxa_unique(ephgrvbr, "Class") #44 Classes total
get_taxa_unique(ephgrvbr, "Order") #97 orders total
get_taxa_unique(ephgrvbr, "Family") #151 families total
get_taxa_unique(ephgrvbr, "Genus") #257 genera total

arangrvbr<-subset_samples(grvbr, Order=="Araneae")
arangrvbr<-prune_taxa(taxa_sums(arangrvbr) > 0, arangrvbr)
ntaxa(arangrvbr) #258
nsamples(arangrvbr) #15

get_taxa_unique(arangrvbr, "Phylum") #9 phyla total
get_taxa_unique(arangrvbr, "Class") #15 Classes total
get_taxa_unique(arangrvbr, "Order") #44 orders total
get_taxa_unique(arangrvbr, "Family") #67 families total
get_taxa_unique(arangrvbr, "Genus") #104 genera total


#Policeman Flats, n = 62
pmf<-subset_samples(sample_ps, Site=="Policeman Flats")
pmf<-prune_taxa(taxa_sums(pmf) > 0, pmf)
ntaxa(pmf) #3054
nsamples(pmf) #62

get_taxa_unique(pmf, "Phylum") #27 phyla total
get_taxa_unique(pmf, "Class") #61 Classes total
get_taxa_unique(pmf, "Order") #152 orders total
get_taxa_unique(pmf, "Family") #215 families total
get_taxa_unique(pmf, "Genus") #391 genera total

sort((table(tax_table(pmf)[,2]))/ntaxa(pmf)*100)

tripmf<-subset_samples(pmf, Order=="Tricoptera")
tripmf<-prune_taxa(taxa_sums(tripmf) > 0, tripmf)
ntaxa(tripmf) #1264
nsamples(tripmf) #13

get_taxa_unique(tripmf, "Phylum") #21 phyla total
get_taxa_unique(tripmf, "Class") #41 Classes total
get_taxa_unique(tripmf, "Order") #90 orders total
get_taxa_unique(tripmf, "Family") #123 families total
get_taxa_unique(tripmf, "Genus") #167 genera total

ephpmf<-subset_samples(pmf, Order=="Ephemeroptera")
ephpmf<-prune_taxa(taxa_sums(ephpmf) > 0, ephpmf)
ntaxa(ephpmf) #1770
nsamples(ephpmf) #16

get_taxa_unique(ephpmf, "Phylum") #22 phyla total
get_taxa_unique(ephpmf, "Class") #49 Classes total
get_taxa_unique(ephpmf, "Order") #112 orders total
get_taxa_unique(ephpmf, "Family") #166 families total
get_taxa_unique(ephpmf, "Genus") #276 genera total

dippmf<-subset_samples(pmf, Order=="Diptera")
dippmf<-prune_taxa(taxa_sums(dippmf) > 0, dippmf)
ntaxa(dippmf) #828
nsamples(dippmf) #11

get_taxa_unique(dippmf, "Phylum") #16 phyla total
get_taxa_unique(dippmf, "Class") #35 Classes total
get_taxa_unique(dippmf, "Order") #87 orders total
get_taxa_unique(dippmf, "Family") #129 families total
get_taxa_unique(dippmf, "Genus") #194 genera total

plecpmf<-subset_samples(pmf, Order=="Plecoptera")
plecpmf<-prune_taxa(taxa_sums(plecpmf) > 0, plecpmf)
ntaxa(plecpmf) #750
nsamples(plecpmf) #8

get_taxa_unique(plecpmf, "Phylum") #17 phyla total
get_taxa_unique(plecpmf, "Class") #30 Classes total
get_taxa_unique(plecpmf, "Order") #69 orders total
get_taxa_unique(plecpmf, "Family") #93 families total
get_taxa_unique(plecpmf, "Genus") #129 genera total

aranpmf<-subset_samples(pmf, Order=="Araneae")
aranpmf<-prune_taxa(taxa_sums(aranpmf) > 0, aranpmf)
ntaxa(aranpmf) #401
nsamples(aranpmf) #14

get_taxa_unique(aranpmf, "Phylum") #11 phyla total
get_taxa_unique(aranpmf, "Class") #17 Classes total
get_taxa_unique(aranpmf, "Order") #59 orders total
get_taxa_unique(aranpmf, "Family") #88 families total
get_taxa_unique(aranpmf, "Genus") #148 genera total

#BRR2 (ACWA control), n = 8
BRR2<-subset_samples(sample_ps, Site=="BRR2")
BRR2<-prune_taxa(taxa_sums(BRR2) > 0, BRR2)
ntaxa(BRR2) #1321
nsamples(BRR2) #8

get_taxa_unique(BRR2, "Phylum") #22 phyla total
get_taxa_unique(BRR2, "Class") #48 Classes total
get_taxa_unique(BRR2, "Order") #113 orders total
get_taxa_unique(BRR2, "Family") #131 families total
get_taxa_unique(BRR2, "Genus") #171 genera total

sort((table(tax_table(BRR2)[,2]))/ntaxa(BRR2)*100)

#PCR1 (ACWA 5% effluent stream), n = 20
PCR1<-subset_samples(sample_ps, Site=="PCR1")
PCR1<-prune_taxa(taxa_sums(PCR1) > 0, PCR1)
ntaxa(PCR1) #2133
nsamples(PCR1) #20

get_taxa_unique(PCR1, "Phylum") #26 phyla total
get_taxa_unique(PCR1, "Class") #59 Classes total
get_taxa_unique(PCR1, "Order") #137 orders total
get_taxa_unique(PCR1, "Family") #175 families total
get_taxa_unique(PCR1, "Genus") #269 genera total

sort((table(tax_table(PCR1)[,2]))/ntaxa(PCR1)*100)

triPCR1<-subset_samples(PCR1, Order=="Tricoptera")
triPCR1<-prune_taxa(taxa_sums(triPCR1) > 0, triPCR1)
ntaxa(triPCR1) #1580
nsamples(triPCR1) #8

get_taxa_unique(triPCR1, "Phylum") #25 phyla total
get_taxa_unique(triPCR1, "Class") #54 Classes total
get_taxa_unique(triPCR1, "Order") #122 orders total
get_taxa_unique(triPCR1, "Family") #141 families total
get_taxa_unique(triPCR1, "Genus") #202 genera total

ephPCR1<-subset_samples(PCR1, Order=="Ephemeroptera")
ephPCR1<-prune_taxa(taxa_sums(ephPCR1) > 0, ephPCR1)
ntaxa(ephPCR1) #1224
nsamples(ephPCR1) #12

get_taxa_unique(ephPCR1, "Phylum") #21 phyla total
get_taxa_unique(ephPCR1, "Class") #48 Classes total
get_taxa_unique(ephPCR1, "Order") #107 orders total
get_taxa_unique(ephPCR1, "Family") #148 families total
get_taxa_unique(ephPCR1, "Genus") #223 genera total


#PCR3 (ACWA 15% effluent stream), n = 15
PCR3<-subset_samples(sample_ps, Site=="PCR3")
PCR3<-prune_taxa(taxa_sums(PCR3) > 0, PCR3)
ntaxa(PCR3) #1887
nsamples(PCR3) #15

get_taxa_unique(PCR3, "Phylum") #27 phyla total
get_taxa_unique(PCR3, "Class") #55 Classes total
get_taxa_unique(PCR3, "Order") #128 orders total
get_taxa_unique(PCR3, "Family") #163 families total
get_taxa_unique(PCR3, "Genus") #237 genera total

sort((table(tax_table(PCR3)[,2]))/ntaxa(PCR3)*100)

triPCR3<-subset_samples(PCR3, Order=="Tricoptera")
triPCR3<-prune_taxa(taxa_sums(triPCR3) > 0, triPCR3)
ntaxa(triPCR3) #1339
nsamples(triPCR3) #8

get_taxa_unique(triPCR3, "Phylum") #24 phyla total
get_taxa_unique(triPCR3, "Class") #48 Classes total
get_taxa_unique(triPCR3, "Order") #110 orders total
get_taxa_unique(triPCR3, "Family") #133 families total
get_taxa_unique(triPCR3, "Genus") #179 genera total

ephPCR3<-subset_samples(PCR3, Order=="Ephemeroptera")
ephPCR3<-prune_taxa(taxa_sums(ephPCR3) > 0, ephPCR3)
ntaxa(ephPCR3) #982
nsamples(ephPCR3) #7

get_taxa_unique(ephPCR3, "Phylum") #23 phyla total
get_taxa_unique(ephPCR3, "Class") #44 Classes total
get_taxa_unique(ephPCR3, "Order") #96 orders total
get_taxa_unique(ephPCR3, "Family") #120 families total
get_taxa_unique(ephPCR3, "Genus") #175 genera total

#Larvae
larvae<-subset_samples(sample_ps, Stage=="Larvae")
larvae<-prune_taxa(taxa_sums(larvae) > 0, larvae)
nsamples(larvae) #150

#Adults
Adults<-subset_samples(sample_ps, Stage=="Adults")
Adults<-prune_taxa(taxa_sums(Adults) > 0, Adults)
nsamples(Adults) #100

#69 Spiders samples
```

#### Relative abundance across sites 
```{r}
#Cochrane
#Larvae
cochlar<-subset_samples(coch, Stage=="Larvae")
cochlar<-prune_taxa(taxa_sums(cochlar) > 0, cochlar)
sort((table(tax_table(cochlar)[,2]))/ntaxa(cochlar)*100)
#Adults
cochad<-subset_samples(coch, Stage=="Adults")
cochad<-prune_taxa(taxa_sums(cochad) > 0, cochad)
sort((table(tax_table(cochad)[,2]))/ntaxa(cochad)*100)
#Spiders
cochSpiders<-subset_samples(coch, Stage=="Spiderss")
cochSpiders<-prune_taxa(taxa_sums(cochSpiders) > 0, cochSpiders)
sort((table(tax_table(cochSpiders)[,2]))/ntaxa(cochSpiders)*100)

#Sunalta
#Larvae
sunlar<-subset_samples(sun, Stage=="Larvae")
sunlar<-prune_taxa(taxa_sums(sunlar) > 0, sunlar)
sort((table(tax_table(sunlar)[,2]))/ntaxa(sunlar)*100)
#Adultss
sunad<-subset_samples(sun, Stage=="Adultss")
sunad<-prune_taxa(taxa_sums(sunad) > 0, sunad)
sort((table(tax_table(sunad)[,2]))/ntaxa(sunad)*100)
#Spiders
sunSpiders<-subset_samples(sun, Stage=="Spiders")
sunSpiders<-prune_taxa(taxa_sums(sunSpiders) > 0, sunSpiders)
sort((table(tax_table(sunSpiders)[,2]))/ntaxa(sunSpiders)*100)

#Cushing Bridge
#Larvae
cushbrlar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Cushing Bridge")
cushbrlar<-prune_taxa(taxa_sums(cushbrlar) > 0, cushbrlar)
sort((table(tax_table(cushbrlar)[,2]))/ntaxa(cushbrlar)*100)
#Adults
cushbrad<-subset_samples(cushbr, Stage=="Adults")
cushbrad<-prune_taxa(taxa_sums(cushbrad) > 0, cushbrad)
sort((table(tax_table(cushbrad)[,2]))/ntaxa(cushbrad)*100)
#Spiders
cushbrSpiders<-subset_samples(cushbr, Stage=="Spiders")
cushbrSpiders<-prune_taxa(taxa_sums(cushbrSpiders) > 0, cushbrSpiders)
sort((table(tax_table(cushbrSpiders)[,2]))/ntaxa(cushbrSpiders)*100)

#Graves Bridge
#Larvae
grvbrlar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Graves Bridge")
grvbrlar<-prune_taxa(taxa_sums(grvbrlar) > 0, grvbrlar)
sort((table(tax_table(grvbrlar)[,2]))/ntaxa(grvbrlar)*100)
#Adults
grvbrad<-subset_samples(grvbr, Stage=="Adults")
grvbrad<-prune_taxa(taxa_sums(grvbrad) > 0, grvbrad)
sort((table(tax_table(grvbrad)[,2]))/ntaxa(grvbrad)*100)
#Spiders
grvbrSpiders<-subset_samples(grvbr, Stage=="Spiders")
grvbrSpiders<-prune_taxa(taxa_sums(grvbrSpiders) > 0, grvbrSpiders)
sort((table(tax_table(grvbrSpiders)[,2]))/ntaxa(grvbrSpiders)*100)

#Policeman Flats
#Larvae
pmflar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="Policeman Flats")
pmflar<-prune_taxa(taxa_sums(pmflar) > 0, pmflar)
sort((table(tax_table(pmflar)[,2]))/ntaxa(pmflar)*100)
#Adults
pmfad<-subset_samples(pmf, Stage=="Adults")
pmfad<-prune_taxa(taxa_sums(pmfad) > 0, pmfad)
sort((table(tax_table(pmfad)[,2]))/ntaxa(pmfad)*100)
#Spiders
pmfSpiders<-subset_samples(pmf, Stage=="Spiders")
pmfSpiders<-prune_taxa(taxa_sums(pmfSpiders) > 0, pmfSpiders)
sort((table(tax_table(pmfSpiders)[,2]))/ntaxa(pmfSpiders)*100)

#BRR2
#Larvae
sort((table(tax_table(BRR2)[,2]))/ntaxa(BRR2)*100)

#PCR1
#Larvae
PCR1lar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="PCR1")
PCR1lar<-prune_taxa(taxa_sums(PCR1lar) > 0, PCR1lar)
sort((table(tax_table(PCR1lar)[,2]))/ntaxa(PCR1lar)*100)
#Adults
PCR1ad<-subset_samples(PCR1, Stage=="Adults")
PCR1ad<-prune_taxa(taxa_sums(PCR1ad) > 0, PCR1ad)
sort((table(tax_table(PCR1ad)[,2]))/ntaxa(PCR1ad)*100)

#PCR3
#Larvae
PCR3lar<-subset_samples(sample_ps, Stage=="Larvae" & Site=="PCR3")
PCR3lar<-prune_taxa(taxa_sums(PCR3lar) > 0, PCR3lar)
sort((table(tax_table(PCR3lar)[,2]))/ntaxa(PCR3lar)*100)
```



# Rarefying the data

Rarefying is done to allow comparisons between samples with high versus low sequencing depth. It involves subsampling the samples to an even number of reads for diversity comparisons. The sequencing depth has to be large enough that most samples retain their diversity, otherwise results can become biased. 

Seems like the samples retain most of their diversity around 2000 reads so I will rarefy to that depth.

``` {r Rarefying the data}
#Rarefying to 2000 reads
#How to rarefy to an even sequencing depth (usually the minimum read so that no sequences are removed)
(ps_rare <- phyloseq::rarefy_even_depth(sample_ps, rngseed = 123, replace = FALSE)) 
#Rarefying to the minimum sample depth removes a lot of the ASVs because there was such high variation in number of reads. Some ASVs were removed because they are no longer present in any sample after random subsampling

head(phyloseq::sample_sums(ps_rare)) #2048 bacterial taxa subsampled per sample. Cuts a lot of sequencing depth but some people argue it is better to retain more samples rather than a larger sequencing depth. 

#Rarefying to 5000 reads
ps4 <- subset_samples(ps2, sample_sums(ps2) > 5000) 
ps4
sample_ps1 <- prune_taxa(taxa_sums(ps4) > 5, ps4)
sample_ps1<- prune_taxa(taxa_sums(sample_ps1) > 0, sample_ps1)
sample_ps1 #335 samples, 20934 taxa
ps_rare2 <- phyloseq::rarefy_even_depth(sample_ps1, rngseed = 123, replace = FALSE)

head(phyloseq::sample_sums(ps_rare2)) #5057 bacterial taxa per sample

```

# Relative Abundance 

Mean relative abundance is beneficial for displaying the composition of taxonomic levels at different sites/across taxa in an easy to understand way (unlike absolute abundance) but has some drawbacks including:

-Difficulty in comparing the absolute abundance across studies because most literature doesn't put the sample size of bacteria or samples. The percent abundance isn't very useful to compare if you don't know the total sample size. 
-Difficulty in interpreting the percent abundance across groups because of a lack of a baseline. 
-No error bars, can't get an idea of the variability in the data
-Hard to decipher between small abundance groups because of the colour limitation

The following graphs display mean relative abundance post filtering of very low counts (all pre-processing steps are done, but rarefying was not done). Graphs show all phyla/families above 2-5% of the total ASVs as well as the top 5 most abundant groups. 


#### Blank samples 
```{r}
#Relative Abundance - phylum level
bl_rel_abun_phylum <- tax_glom(blank_ps, taxrank="Phylum")
bl_comp_rel_abun_phylum <- microbiome::transform(bl_rel_abun_phylum, "compositional")
bl_rel_abun_all_prune_phylum = prune_taxa(taxa_sums(bl_comp_rel_abun_phylum) > 0.02, bl_comp_rel_abun_phylum) 

bl_rel_abun_prune_sample_phylum <- psmelt(bl_rel_abun_all_prune_phylum)

bl.sample.df.agg.phylum <- aggregate(Abundance ~ Phylum + Site  + Taxa, data = bl_rel_abun_prune_sample_phylum, FUN = mean) #aggregate to Phylum level, take the mean
bl.sample.df.agg.phylum$Site <- factor(bl.sample.df.agg.phylum$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))
#bl.sample.df.agg.phylum$Taxa <- factor(bl.sample.df.agg.phylum$Taxa, levels=c("Water Blank", "Mesh Blank", "Sheet Blank", "Negative Blank"), labels=c("RNase Free Water Blank", "Mesh Blank" "Sheet Blank", "Negative Blank"))


bl_rel_abun_plot_phylum <- ggplot(bl.sample.df.agg.phylum, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance \n >2% ASVs") +
  facet_grid(.~Taxa, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label=scales::percent)
bl_rel_abun_plot_phylum

top_phyla_bl <- bl.sample.df.agg.phylum %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phyla_bl
top5phybl <- top_phyla_bl$Phylum[1:10]
top5phybl <- bl.sample.df.agg.phylum %>%
    mutate(Phylum = fct_other(Phylum, top5phybl))


top5phyblplot<- ggplot(top5phybl, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Taxa, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(4, "Paired"), "gray35"))+
  scale_y_continuous(label = scales::percent)

top5phyblplot


#Family Level
bl_rel_abun_fam <- aggregate_taxa(blank_ps, "Family")
bl_comp_rel_abun_fam <- microbiome::transform(bl_rel_abun_fam, "compositional")
bl_rel_abun_all_prune_fam = prune_taxa(taxa_sums(bl_comp_rel_abun_fam) > 0.05, bl_comp_rel_abun_fam) 

bl_rel_abun_prune_sample_fam <- psmelt(bl_rel_abun_all_prune_fam)

bl.sample.df.agg.fam <- aggregate(Abundance ~ Family + Site  + Taxa, data = bl_rel_abun_prune_sample_fam, FUN = mean) #aggregate to Phylum level, take the mean
bl.sample.df.agg.fam$Site <- factor(bl.sample.df.agg.fam$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))
bl.sample.df.agg.fam$Taxa <- factor(bl.sample.df.agg.fam$Taxa, levels=c("Water Blank", "Mesh Blank", "Sheet Blank", "Negative Blank"))

top_fam_bl <- bl.sample.df.agg.fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_bl
top5fambl <- top_fam_bl$Family[1:10]
top5fambl <- bl.sample.df.agg.fam %>%
    mutate(Family = fct_other(Family, top5fambl))


top5famplotbl<-ggplot(top5fambl, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Taxa, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(5, "Paired"), "gray35"))+
  scale_y_continuous(label = scales::percent)

top5famplotbl

#Genus

bl_rel_abun_genus <- aggregate_taxa(blank_ps, "Genus")
bl_comp_rel_abun_genus <- microbiome::transform(bl_rel_abun_genus, "compositional")

bl_rel_abun_genus_df <- psmelt(bl_comp_rel_abun_genus)

bl.sample.df.agg.genus <- aggregate(Abundance ~ Genus + Site +Blank_Type, data = bl_rel_abun_genus_df, FUN = mean) #aggregate to Phylum level, take the mean
bl.sample.df.agg.genus$Site <- factor(bl.sample.df.agg.genus$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))

top_genus_bl <- bl.sample.df.agg.genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_genus_bl
top5genusbl <- top_genus_bl$Genus[1:20]
top5genusbl <- bl.sample.df.agg.genus %>%
    mutate(Genus = fct_other(Genus, top5genusbl))


topgenusplotbl<-ggplot(top5genusbl, aes(x=Blank_Type, y=Abundance, fill=Genus)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Site, scales = "free")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(11, "Paired"), "gray35"))+
  scale_y_continuous(label = scales::percent)

topgenusplotbl
```

#### All inverts and Spiders combined 
```{r}
#Phylum
#Non-rarefied all sites (Bow River and ACWA combined)
rel_abun_phylum <- speedyseq::tax_glom(sample_ps, taxrank = "Phylum")
comp_rel_abun_phylum <- transform_sample_counts(rel_abun_phylum, function(x) x/sum(x))
#comp_rel_abun_phylum <- microbiome::transform(rel_abun_phylum, "compositional") #This is the exact same as the code above. "compositional" means transform to relative abundance.
rel_abun_all_prune = prune_taxa(taxa_sums(comp_rel_abun_phylum) > 0.05, comp_rel_abun_phylum) #Includes phyla that are present in more than 5% of samples

rel_abun_prune_sample <- psmelt(rel_abun_all_prune)

sample.df.agg <- aggregate(Abundance ~ Phylum + Site + sample_Family , data = rel_abun_prune_sample, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg$Site <- factor(sample.df.agg$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))


#Relative Abundance Plot - Phylum Level
rel_abun_plot_phylum <- ggplot(sample.df.agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_wrap(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x", nrow=1) + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
 scale_fill_manual(values=c("purple", "#CC9966", "#A6CEE3", "coral", "orange", "#1F78B4", "red", "brown", "#B2DF8A", brewer.pal(3, "Set3"), "darkolivegreen4", brewer.pal(3, "Set2")))+ rremove("grid")
rel_abun_plot_phylum

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Mean relative abundance of phyla greater than 5 percent 3 panel plot.png", height=7, width=8)


#Top 5 phyla relative abundance, all inverts included (Bow River and ACWA combined)
top_phyla <- sample.df.agg %>%
    group_by(Site, Phylum) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phyla
top5 <- top_phyla$Phylum[1:33]
top5
df0 <- sample.df.agg %>%
    mutate(Phylum = fct_other(Phylum, top5))

df0$Phylum <- factor(df0$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"), labels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"))

#df0$Stage <- factor(df0$Stage, levels=c("Larvae", "Adults", "Spiders"), labels=c("Larvae (n = 150)", "Adults (n = 115)", "Spiders (n = 77)"))


top5phylumplot <- ggplot(df0, aes(x=Site, y=Abundance, fill=Phylum)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=6)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")
top5phylumplot

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Mean relative abundance of top 5 phyla across invertebrate taxa ACWA and Bow.png", height=7, width=22)

#Bow River only
bow_ps<-subset_samples(sample_ps, Stream_Type=="Bow River")
rel_abun_phylum_bow <- speedyseq::tax_glom(bow_ps, taxrank = "Phylum")
comp_rel_abun_phylum_bow <- transform_sample_counts(rel_abun_phylum_bow, function(x) x/sum(x))

rel_abun_prune_sample_bow <- psmelt(comp_rel_abun_phylum_bow)

sample.df.agg_bow <- aggregate(Abundance ~ Phylum + Site + sample_Family , data = rel_abun_prune_sample_bow, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_bow$Site <- factor(sample.df.agg_bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))


#Top 5 phyla relative abundance, all inverts included (Bow River and ACWA combined)
top_phyla_bow <- sample.df.agg_bow %>%
    group_by(Site, Phylum) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phyla_bow
top5_bow <- top_phyla_bow$Phylum[1:23]
df0_bow <- sample.df.agg_bow %>%
    mutate(Phylum = fct_other(Phylum, top5_bow))

df0_bow$Phylum <- factor(df0_bow$Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"), labels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Actinobacteriota", "Other", "Proteobacteria"))

top5phylumplot_bow <- ggplot(df0_bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Phylum)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=(aes(angle=45, hjust=1, size=16)),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")+
  xlab("Site")
top5phylumplot_bow

#ggsave(filename="Microbiome_ED/Non decontaminated analysis/Relative Abundance Graphs/All samples/Mean relative abundance of top 5 phyla across invertebrate taxa Bow only.png", height=8, width=22)


#Relative abundance - Class Level
rel_abun_class <- speedyseq::tax_glom(sample_ps, taxrank = "Class")
comp_rel_abun_class <- microbiome::transform(rel_abun_class, "compositional")
rel_abun_all_prune_class = prune_taxa(taxa_sums(comp_rel_abun_class) > 0.05, comp_rel_abun_class) 

rel_abun_prune_sample_class <- psmelt(rel_abun_all_prune_class)

sample.df.agg.class <- aggregate(Abundance ~ Class + Site  + sample_Family, data = rel_abun_prune_sample_class, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg.class$Site <- factor(sample.df.agg.class$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))
#sample.df.agg.class$Stage <- factor(sample.df.agg.class$Stage, levels=c("Larvae", "Adults", "Spiders"), labels=c("Larvae (n = 150)", "Adults (n = 115)", "Spiders (n = 77)"))


rel_abun_plot_class <- ggplot(sample.df.agg.class, aes(x=Site, y=Abundance, fill=Class)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_wrap(.~sample_Family, scale="free_x", nrow=2) + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_discrete(name="Bacterial Class")

rel_abun_plot_class #Too many classes

top_class <- sample.df.agg.class %>%
    group_by(Site, Class) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_class
top5class <- top_class$Class[1:30]
top5classdf <- sample.df.agg.class %>%
    mutate(Class = fct_other(Class, top5class))
top5classdf

top5classdf$Class <- factor(top5classdf$Class, levels=c("Alphaproteobacteria", "Bacteroidia", "Gammaproteobacteria", "Bacilli", "Other", "Clostridia", "Cyanobacteriia"))

top5classplot <- ggplot(top5classdf, aes(x=Site, y=Abundance, fill=Class)) + 
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=6)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c("#A6CEE3", "#1F78B4", "#B2DF8A", "#CC9966", "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+
  rremove("grid")
top5classplot

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Mean relative abundance of top 5 classes across invertebrates.png", height=8, width=22)


#Relative abundance - Family Level


#All samples (Bow and ACWA)

rel_abun_fam <- tax_glom(sample_ps, taxrank = "Family")
comp_rel_abun_fam <- microbiome::transform(rel_abun_fam, "compositional")
rel_abun_all_prune_fam = prune_taxa(taxa_sums(comp_rel_abun_fam) > 0.1, comp_rel_abun_fam) 

rel_abun_prune_sample_fam <- psmelt(rel_abun_all_prune_fam)

sample.df.agg_fam <- aggregate(Abundance ~ Family + Site + sample_Family, data = rel_abun_prune_sample_fam, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_fam$Site <- factor(sample.df.agg_fam$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))

top_fam <- sample.df.agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam <- top_fam$Family[1:85]
top5famdf <- sample.df.agg_fam %>%
    mutate(Family = fct_other(Family, top5fam))

top5famplot <- ggplot(top5famdf, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5famplot

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 families separated by invertebrate taxa type Bow and ACWA.png", height=8, width=24)



#Bow River only 

sample_all_Bow<-subset_samples(sample_ps, Stream_Type=="Bow River")

rel_abun_fam_Bow <- tax_glom(sample_all_Bow, taxrank = "Family")
comp_rel_abun_fam_Bow <- microbiome::transform(rel_abun_fam_Bow, "compositional")
rel_abun_all_prune_fam_Bow = prune_taxa(taxa_sums(comp_rel_abun_fam_Bow) > 0.1, comp_rel_abun_fam_Bow) 

rel_abun_prune_sample_fam_Bow <- psmelt(rel_abun_all_prune_fam_Bow)

sample.df.agg_fam_Bow <- aggregate(Abundance ~ Family + Site + sample_Family, data = rel_abun_prune_sample_fam_Bow, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_fam_Bow$Site <- factor(sample.df.agg_fam_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))


rel_abun_plot_fam_Bow <- ggplot(sample.df.agg_fam_Bow, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >10% ASVs") +
  facet_grid(~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.2, "cm"),
        legend.text=element_text(size=8)) +
  guides(fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)
rel_abun_plot_fam_Bow 

#Top ~22 families, all inverts included
top_fam_Bow <- sample.df.agg_fam_Bow %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_Bow <- top_fam_Bow$Family[1:60]
top5famdf_Bow <- sample.df.agg_fam_Bow %>%
    mutate(Family = fct_other(Family, top5fam_Bow))

top5famplotBow <- ggplot(top5famdf_Bow, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5famplotBow 

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 families larvae, adults, spiders separated by invertebrate taxa type.png", height=8, width=22)


#ACWA only 

sample_all_ACWA<-subset_samples(sample_ps, Stream_Type!="Bow River")

rel_abun_fam_ACWA <- tax_glom(sample_all_ACWA, taxrank = "Family")
comp_rel_abun_fam_ACWA <- microbiome::transform(rel_abun_fam_ACWA, "compositional")

rel_abun_prune_sample_fam_ACWA <- psmelt(comp_rel_abun_fam_ACWA)

sample.df.agg_fam_ACWA <- aggregate(Abundance ~ Family + Site + sample_Family, data = rel_abun_prune_sample_fam_ACWA, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_fam_ACWA$Site <- factor(sample.df.agg_fam_ACWA$Site, levels=c("BRR2", "PCR1", "PCR3"))


rel_abun_plot_fam_ACWA <- ggplot(sample.df.agg_fam_ACWA, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance \n >10% ASVs") +
  facet_grid(~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.2, "cm"),
        legend.text=element_text(size=8)) +
  guides(fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)
rel_abun_plot_fam_ACWA 

#Top ~22 families, all inverts included
top_fam_ACWA <- sample.df.agg_fam_ACWA %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_ACWA
top5fam_ACWA <- top_fam_ACWA$Family[1:50]
top5famdf_ACWA <- sample.df.agg_fam_ACWA %>%
    mutate(Family = fct_other(Family, top5fam_ACWA))

top5famplotACWA <- ggplot(top5famdf_ACWA, aes(x=Site, y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Baetidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "blue", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5famplotACWA

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 families Hydropsychidae, Baetidae ACWA.png", height=8, width=16)



#Genus level

#All samples (Bow and ACWA)

rel_abun_genus <- tax_glom(sample_ps, taxrank = "Genus")
comp_rel_abun_genus <- microbiome::transform(rel_abun_genus, "compositional")
#rel_abun_all_prune_genus = prune_taxa(taxa_sums(comp_rel_abun_genus) > 0.1, comp_rel_abun_genus) 

rel_abun_prune_sample_genus <- psmelt(comp_rel_abun_genus)

sample.df.agg_genus <- aggregate(Abundance ~ Genus + Site + sample_Family, data = rel_abun_prune_sample_genus, FUN = mean) #aggregate to Phylum level, take the mean

#Top ~22 genera, all inverts included
top_genus <- sample.df.agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_genus
top5genus<- top_genus$Genus[1:64]
top5genusdf <- sample.df.agg_genus %>%
    mutate(Genus = fct_other(Genus, top5genus))

top5genusplot <- ggplot(top5genusdf, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae",  "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "blue", "pink", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5genusplot

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 genera separated by invertebrate taxa type Bow and ACWA.png", height=8, width=24)


#Bow River only
rel_abun_genus_Bow <- tax_glom(sample_all_Bow, taxrank = "Genus")
comp_rel_abun_genus_Bow <- microbiome::transform(rel_abun_genus_Bow, "compositional")
rel_abun_all_prune_genus_Bow = prune_taxa(taxa_sums(comp_rel_abun_genus_Bow) > 0.1, comp_rel_abun_genus_Bow) 

rel_abun_prune_sample_genus_Bow <- psmelt(rel_abun_all_prune_genus_Bow)

sample.df.agg_genus_Bow <- aggregate(Abundance ~ Genus + Site + sample_Family, data = rel_abun_prune_sample_genus_Bow, FUN = mean) #aggregate to Phylum level, take the mean
sample.df.agg_genus_Bow$Site <- factor(sample.df.agg_genus_Bow$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

#Top ~22 families, all inverts included
top_genus_Bow <- sample.df.agg_genus_Bow %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_Bow <- top_genus_Bow$Genus[1:63]
top5genusdf_Bow <- sample.df.agg_genus_Bow %>%
    mutate(Genus = fct_other(Genus, top5genus_Bow))

top5genusplotBow <- ggplot(top5genusdf_Bow, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=26)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "blue", "pink", "coral", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5genusplotBow

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 genera separated by invertebrate taxa type Bow only.png", height=8, width=22)

#Alistipes : protective effects against colitis (intestinal inflammation), autism, and cirrhosis (liver fibrosis). However, this genus can also cause dysbiosis by contributing to anxiety, chronic fatigue syndrome, depression, and hypertension. isolated primarily from medical clinical samples.

#Candidatus Bacilloplasma: Candidatus Bacilloplasma are the dominant bacteria in the intestines of healthy shrimp, and a change in its abundance can contribute to shifting the intestinal microbial communities in shrimp suffering pathogens 

#Carnobacterium: lactic acid-producing, Gram-positive rod-shaped bacteria that are rarely isolated in humans and often regarded as non-pathogenic [1]. Instead, they are frequently isolated from the environment and are currently approved for use as a bio-preservative in the food industry

#CM1G08: Not much info on this one

#Emticicia: Joung Y, Seo MA, Kang H, Kim H, Ahn TS, Cho JC, Joh K. Emticicia aquatica sp. nov., a species of the family Cytophagaceae isolated from fresh water. Int J Syst Evol Microbiol 2015; 65:4358-4362.

#Flavobacterium: Flavobacteria are found in soil and fresh water in a variety of environments. Several species are known to cause disease in freshwater fish.

#Hymenobacter: Microorganisms belonging to the genus Hymenobacter (phylum Bacteroidetes, family Cytophagaceae) are aerobic, nonmotile, Gram-negative and rod-shaped bacteria. It has been reported that Hymenobacter cells may hydrolyze gelatin, starch, xylan, Tween 80 and Tween 60, suggesting the production of extracellular proteases, amylases, xylanases and lipases

#Ideonella: Probably dechloratans which is chlorate respiring and commonly found in effluent. Couuld also be sakaiensis which breaks dwon plastic

#Massilia: The Massilia genus is a diverse group that resides in many different environments, has many heterotrophic means of gathering energy, and is commonly found in association with plants.

#Mucinivorans: Mucinivorans is an anaerobic bacterial genus from the family of Rikenellaceae with one known species (Mucinivorans hirudinis). Mucinivorans hirudinis has been isolated from the digestrive tract of the leech Hirudo verbana

#Pseudorhodobacter: Associated with aquatic environments

#Rhodococcus: Rhodococcus spp. strains have peculiar biosynthetic activities that contribute to their strong persistence in harsh and contaminated environments and provide them a competitive advantage over other microorganisms.capacity to biodegrade a wide range of organic compounds, including toxic and recalcitrant molecules such as chlorinated aliphatic and aromatic hydrocarbons, N- and S-heterocyclic compounds, and synthetic polymers (e.g., polyethylene)

#Rhodoferax: commonly found in lakes and rivers. Possibly iron reducing.

#Rikettsia: Rickettsiae and related (rickettsia-like) bacteria (such as Ehrlichia, Anaplasma, and Coxiella burnetii bacteria) are an unusual type of bacteria that cause several similar diseases

#Serratia: Serratia species are opportunistic gram-negative bacteria in the large family, Enterobacteriaceae. Serratia are widespread in the environment, but are not a common component of the human fecal flora. Occurs naturally in soil and water.

#Sphingorhabdus: isolated from freshwater lake

#Spiroplasma: many species of Spiroplasma have been isolated from a wide variety of hosts, mainly arthropods and plants (Clark et al., 1985; Saillard et al., 1987; Regassa and Gasparich, 2006), and have also been shown to be able to infect humans


#ACWA streams only
rel_abun_genus_ACWA <- tax_glom(sample_all_ACWA, taxrank = "Genus")
comp_rel_abun_genus_ACWA <- microbiome::transform(rel_abun_genus_ACWA, "compositional")

rel_abun_prune_sample_genus_ACWA <- psmelt(comp_rel_abun_genus_ACWA)

sample.df.agg_genus_ACWA <- aggregate(Abundance ~ Genus + Site + sample_Family, data = rel_abun_prune_sample_genus_ACWA, FUN = mean) #aggregate to Phylum level, take the mean

#Top ~22 families, all inverts included
top_genus_ACWA <- sample.df.agg_genus_ACWA %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_genus_ACWA
top5genus_ACWA <- top_genus_ACWA$Genus[1:48]
top5genusdf_ACWA <- sample.df.agg_genus_ACWA %>%
    mutate(Genus = fct_other(Genus, top5genus_ACWA))

top5genusplotACWA <- ggplot(top5genusdf_ACWA, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~factor(sample_Family, levels=c("Hydropsychidae", "Baetidae")), scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=25)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "blue", "pink", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
top5genusplotACWA

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Top 22 genera Hydropsychidae, Baetidae ACWA.png", height=8, width=16)

```

#### Across metamorphosis {.tabset}

##### Family level
```{r}
#Hydropsychidae larvae to Trichoptera adults
#Only from the Bow River since there was no adult caddis in the ACWA streams
caddis_ps<-subset_samples(sample_ps, Order=="Trichoptera" & Stream_Type=="Bow River")

caddis_rel_abun_fam <- tax_glom(caddis_ps, taxrank = "Family")
caddis_rel_abun_fam <- microbiome::transform(caddis_rel_abun_fam, "compositional")
#caddis_rel_abun_all_prune_fam = prune_taxa(taxa_sums(comp_rel_abun_fam) > 0.1, comp_rel_abun_fam) 

caddis_rel_abun_fam_df <- psmelt(caddis_rel_abun_fam)

caddis_agg_fam <- aggregate(Abundance ~ Family + Site + Stage, data = caddis_rel_abun_fam_df, FUN = mean) 
caddis_agg_fam$Site <- factor(caddis_agg_fam$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

top_fam_caddis <- caddis_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_caddis <- top_fam_caddis$Family[1:63]
top5famdf_caddis <- caddis_agg_fam %>%
    mutate(Family = fct_other(Family, top5fam_caddis))

caddistopfamplot <- ggplot(top5famdf_caddis, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
caddistopfamplot

#Mayflies (larval heptageniidae and adult ephemeroptera which probably includes both heps and ephemerellids but we can't know for sure because they weren't identified to family level)
mayfly_ps<-subset_samples(sample_ps, Order=="Ephemeroptera" & Stream_Type=="Bow River")

mayfly_rel_abun_fam <- tax_glom(mayfly_ps, taxrank = "Family")
mayfly_rel_abun_fam <- microbiome::transform(mayfly_rel_abun_fam, "compositional")

mayfly_rel_abun_fam_df <- psmelt(mayfly_rel_abun_fam)

mayfly_agg_fam <- aggregate(Abundance ~ Family + Site + Stage, data = mayfly_rel_abun_fam_df, FUN = mean) 
mayfly_agg_fam$Site <- factor(mayfly_agg_fam$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

top_fam_mayfly <- mayfly_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_mayfly <- top_fam_mayfly$Family[1:57]
top5famdf_mayfly <- mayfly_agg_fam %>%
    mutate(Family = fct_other(Family, top5fam_mayfly))

mayflytopfamplot <- ggplot(top5famdf_mayfly, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
mayflytopfamplot


#Baetidae
baetidae_ps<-subset_samples(sample_ps, Family=="Baetidae")

baetidae_rel_abun_fam <- tax_glom(baetidae_ps, taxrank = "Family")
baetidae_rel_abun_fam <- microbiome::transform(baetidae_rel_abun_fam, "compositional")

baetidae_rel_abun_fam_df <- psmelt(baetidae_rel_abun_fam)

baetidae_agg_fam <- aggregate(Abundance ~ Family + Site + Stage, data = baetidae_rel_abun_fam_df, FUN = mean) 
baetidae_agg_fam$Site <- factor(baetidae_agg_fam$Site, levels=c("BRR2", "PCR1", "PCR3"))

top_fam_baetidae <- baetidae_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_baetidae <- top_fam_baetidae$Family[1:32]
top5famdf_baetidae <- baetidae_agg_fam %>%
    mutate(Family = fct_other(Family, top5fam_baetidae))

baetidaetopfamplot <- ggplot(top5famdf_baetidae, aes(x=Site, y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
 theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
baetidaetopfamplot

#Chironomidae
chiro_ps<-subset_samples(sample_ps, Order=="Diptera")

chiro_rel_abun_fam <- tax_glom(chiro_ps, taxrank = "Family")
chiro_rel_abun_fam <- microbiome::transform(chiro_rel_abun_fam, "compositional")

chiro_rel_abun_fam_df <- psmelt(chiro_rel_abun_fam)

chiro_agg_fam <- aggregate(Abundance ~ Family + Site + Stage, data = chiro_rel_abun_fam_df, FUN = mean) 

top_fam_chiro <- chiro_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5fam_chiro <- top_fam_chiro$Family[1:34]
top5famdf_chiro <- chiro_agg_fam %>%
    mutate(Family = fct_other(Family, top5fam_chiro))

chirotopfamplot <- ggplot(top5famdf_chiro, aes(x=Site, y=Abundance, fill=Family)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
chirotopfamplot

#All orders combined family level

caddistopfamplot2<-caddistopfamplot + rremove("xlab") +rremove("ylab")
mayflytopfamplot2<-mayflytopfamplot + rremove("xlab") +rremove("ylab")
baetidaetopfamplot2<-baetidaetopfamplot+ rremove("xlab") +rremove("ylab")
chirotopfamplot2<-chirotopfamplot + rremove("xlab") +rremove("ylab")

ggarrange(caddistopfamplot2, mayflytopfamplot2, baetidaetopfamplot2, chirotopfamplot2, nrow=2, ncol=2, align="v", common.legend = F, labels = c("Trichoptera", "Ephemeroptera", "Baetidae", "Chironomidae"), vjust=1)

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Relative abundance across metamorphosis of invertebrate taxa family level.png", height=10, width=20)
```

##### Genus level
```{r}
#Hydropsychidae larvae to Trichoptera adults
#Only from the Bow River since there was no adult caddis in the ACWA streams
caddis_rel_abun_genus <- tax_glom(caddis_ps, taxrank = "Genus")
caddis_rel_abun_genus <- microbiome::transform(caddis_rel_abun_genus, "compositional")

caddis_rel_abun_genus_df <- psmelt(caddis_rel_abun_genus)

caddis_agg_genus <- aggregate(Abundance ~ Genus + Site + Stage, data = caddis_rel_abun_genus_df, FUN = mean) 
caddis_agg_genus$Site <- factor(caddis_agg_genus$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

top_genus_caddis <- caddis_agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_caddis <- top_genus_caddis$Genus[1:56]
top5genusdf_caddis <- caddis_agg_genus %>%
    mutate(Genus = fct_other(Genus, top5genus_caddis))

caddistopgenusplot <- ggplot(top5genusdf_caddis, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
caddistopgenusplot

#Mayflies (larval heptageniidae and adult ephemeroptera which probably includes both heps and ephemerellids but we can't know for sure because they weren't identified to family level)

mayfly_rel_abun_genus <- tax_glom(mayfly_ps, taxrank = "Genus")
mayfly_rel_abun_genus <- microbiome::transform(mayfly_rel_abun_genus, "compositional")

mayfly_rel_abun_genus_df <- psmelt(mayfly_rel_abun_genus)

mayfly_agg_genus <- aggregate(Abundance ~ Genus + Site + Stage, data = mayfly_rel_abun_genus_df, FUN = mean) 
mayfly_agg_genus$Site <- factor(mayfly_agg_genus$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats"))

top_genus_mayfly <- mayfly_agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_mayfly <- top_genus_mayfly$Genus[1:47]
top5genusdf_mayfly <- mayfly_agg_genus %>%
    mutate(Genus = fct_other(Genus, top5genus_mayfly))

mayflytopgenusplot <- ggplot(top5genusdf_mayfly, aes(x=factor(Site, labels = c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
mayflytopgenusplot

#Baetidae
baetidae_rel_abun_genus <- tax_glom(baetidae_ps, taxrank = "Genus")
baetidae_rel_abun_genus <- microbiome::transform(baetidae_rel_abun_genus, "compositional")

baetidae_rel_abun_genus_df <- psmelt(baetidae_rel_abun_genus)

baetidae_agg_genus <- aggregate(Abundance ~ Genus + Site + Stage, data = baetidae_rel_abun_genus_df, FUN = mean) 
baetidae_agg_genus$Site <- factor(baetidae_agg_genus$Site, levels=c("BRR2", "PCR1", "PCR3"))

top_genus_baetidae <- baetidae_agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_baetidae <- top_genus_baetidae$Genus[1:32]
top5genusdf_baetidae <- baetidae_agg_genus %>%
    mutate(Genus = fct_other(Genus, top5genus_baetidae))

baetidaetopgenusplot <- ggplot(top5genusdf_baetidae, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
baetidaetopgenusplot


#Chironomidae
chiro_rel_abun_genus <- tax_glom(chiro_ps, taxrank = "Genus")
chiro_rel_abun_genus <- microbiome::transform(chiro_rel_abun_genus, "compositional")

chiro_rel_abun_genus_df <- psmelt(chiro_rel_abun_genus)

chiro_agg_genus <- aggregate(Abundance ~ Genus + Site + Stage, data = chiro_rel_abun_genus_df, FUN = mean)

top_genus_chiro <- chiro_agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5genus_chiro <- top_genus_chiro$Genus[1:32]
top5genusdf_chiro <- chiro_agg_genus %>%
    mutate(Genus = fct_other(Genus, top5genus_chiro))

chirotopgenusplot <- ggplot(top5genusdf_chiro, aes(x=Site, y=Abundance, fill=Genus)) +
  geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  facet_grid(.~Stage, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.3, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=12)) +
  guides (fill=guide_legend(nrow=23)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "azure3", "gray35"))+
  scale_y_continuous(label = scales::percent)+
  xlab("Site")+
  rremove("grid")
chirotopgenusplot

#All orders combined genus level
caddistopgenusplot2<-caddistopgenusplot + rremove("xlab") +rremove("ylab")
mayflytopgenusplot2<-mayflytopgenusplot + rremove("xlab") +rremove("ylab")
baetidaetopgenusplot2<-baetidaetopgenusplot+ rremove("xlab") +rremove("ylab")
chirotopgenusplot2<-chirotopgenusplot + rremove("xlab") +rremove("ylab")

ggarrange(caddistopgenusplot2, mayflytopgenusplot2, baetidaetopgenusplot2, chirotopgenusplot2, nrow=2, ncol=2, align="v", common.legend = F, labels = c("Trichoptera", "Ephemeroptera", "Baetidae", "Chironomidae"), vjust=1)

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/All samples/Relative abundance across metamorphosis of invertebrate taxa genus level.png", height=10, width=20)

```

#### All macroinvertebrate Larvae {.tabset} 
```{r}
#Bow River and ACWA

larvae<-subset_samples(sample_ps, Stage == "Larvae")
larvae<-prune_taxa(taxa_sums(larvae) > 0, larvae)
larvae #150 samples, 4062 taxa

larvae_rel_abund_phylum <- tax_glom(larvae, taxrank="Phylum")
comp_rel_abund_phylum_larvae<- microbiome::transform(larvae_rel_abund_phylum, "compositional")
#comp_rel_abund_phylum_larvae_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_larvae) > 0.02, comp_rel_abund_phylum_larvae) 

rel_abund_larvae_df <- psmelt(comp_rel_abund_phylum_larvae)

rel_abund_larvae_agg <- aggregate(Abundance ~ Phylum + Site , data = rel_abund_larvae_df, FUN = mean)

rel_abund_larvae_phylum_plot <- ggplot(rel_abund_larvae_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))+
  scale_fill_discrete(name=NULL)
rel_abund_larvae_phylum_plot


top_phy_larvae<- rel_abund_larvae_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_larvae
top5phylarvae <- top_phy_larvae$Phylum[1:20]
top5phylarvae
top5phylarvae <- rel_abund_larvae_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phylarvae))

#Phyla
top5phylarvaeplot<-ggplot(top5phylarvae, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")

top5phylarvaeplot

#ggsave("Microbiome_ED/Decontaminated analysis/Relative Abundance Graphs/Larvae/Top 4 phyla all larvae pooled.png", width=10, height=8)


#Only Bow River

larvae_bow<-subset_samples(sample_ps, Stage == "Larvae" & Stream_Type=="Bow River")
larvae_bow<-prune_taxa(taxa_sums(larvae_bow) > 0, larvae_bow)
larvae_bow #111 samples, 3654 taxa

larvae_bow_rel_abund_phylum <- tax_glom(larvae_bow, taxrank="Phylum")
comp_rel_abund_phylum_larvae_bow <- microbiome::transform(larvae_bow_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_larvae_bow_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_larvae_bow) > 0.02, comp_rel_abund_phylum_larvae_bow) 

rel_abund_larvae_bow_df <- psmelt(comp_rel_abund_phylum_larvae_bow_prune)

rel_abund_larvae_bow_agg <- aggregate(Abundance ~ Phylum + Site , data = rel_abund_larvae_bow_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_larvae_bow_phylum_plot <- ggplot(rel_abund_larvae_bow_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))+
  scale_fill_discrete(name=NULL)
rel_abund_larvae_bow_phylum_plot
#Some change at Cushing Bridge

top_phy_larvae_bow <- rel_abund_larvae_bow_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_larvae_bow
top5phylarvaebow <- top_phy_larvae_bow$Phylum[1:20]
top5phylarvaebow
top5phylarvaebow <- rel_abund_larvae_bow_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phylarvaebow))

#Phyla
top5phylarvaeplotbow<-ggplot(top5phylarvaebow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phylarvaeplotbow
#Most differences at Cushing Bridge and BRR2. Large increase in Cyanobacteria at Cushing Bridge


#Family
larvae_rel_abund_fam_bow <- tax_glom(larvae_bow, taxrank="Family")
comp_rel_abund_fam_larvae_bow <- microbiome::transform(larvae_rel_abund_fam_bow, "compositional")
comp_rel_abund_fam_larvae_prune_bow = prune_taxa(taxa_sums(comp_rel_abund_fam_larvae_bow) > 0.02, comp_rel_abund_fam_larvae_bow) 

rel_abund_larvae_df_fam_bow <- psmelt(comp_rel_abund_fam_larvae_prune_bow)

rel_abund_larvae_agg_fam_bow <- aggregate(Abundance ~ Family + Site , data = rel_abund_larvae_df_fam_bow, FUN = mean) #aggregate to Phylum level, take the mean

top_fam_larvae_bow <- rel_abund_larvae_agg_fam_bow %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_larvae_bow
top5famlarvae_bow <- top_fam_larvae_bow$Family[1:40]
top5famlarvae_bow
top5famlarvae_bow <- rel_abund_larvae_agg_fam_bow %>%
    mutate(Family = fct_other(Family, top5famlarvae_bow))

#Family
top5famlarvaeplotbow<-ggplot(top5famlarvae_bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family))+ 
                                             #levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=16)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=8, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "gray35"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5famlarvaeplotbow


```

##### 5 panel faceted plot with all larval aquatic invert families
```{r}
#Bow River and ACWA
rel_abund_larvae_agg_fam <- aggregate(Abundance ~ Phylum + Site + sample_Family, data = rel_abund_larvae_df, FUN = mean)

rel_abund_larvae_phylum_plot_lar <- ggplot(rel_abund_larvae_agg_fam, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  facet_wrap(~ sample_Family, scales ="free")+
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))
rel_abund_larvae_phylum_plot_lar

top_phy_larvae_facet <- rel_abund_larvae_agg_fam %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_larvae_facet
top5phylarvae_facet <- top_phy_larvae_facet$Phylum[1:20]
top5phylarvae_facet
top5phylarvae_facet <- rel_abund_larvae_agg_fam %>%
    mutate(Phylum = fct_other(Phylum, top5phylarvae_facet))


top5phylarvaeplot_facet<-ggplot(top5phylarvae_facet, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
  facet_wrap(~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae")), scales="free")+
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  xlab("Site")+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phylarvaeplot_facet 


#Only Bow River

rel_abund_larvae_bow_agg_fam <- aggregate(Abundance ~ Phylum + Site +sample_Family, data = rel_abund_larvae_bow_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_larvae_bow_phylum_plot_lar <- ggplot(rel_abund_larvae_bow_agg_fam, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  facet_wrap(~ sample_Family, scales ="free")+
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))
rel_abund_larvae_bow_phylum_plot_lar

top_phy_larvae_bow_facet <- rel_abund_larvae_bow_agg_fam %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_larvae_bow_facet
top5phylarvae_bowfacet <- top_phy_larvae_bow_facet$Phylum[1:20]
top5phylarvae_bowfacet
top5phylarvae_bowfacet <- rel_abund_larvae_bow_agg_fam %>%
    mutate(Phylum = fct_other(Phylum, top5phylarvae_bowfacet))


top5phylarvaebowplot_facet<-ggplot(top5phylarvae_bowfacet, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
  facet_wrap(~factor(sample_Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae")), scales="free")+
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  xlab("Site")+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phylarvaebowplot_facet 
```

#### All macroinvertebrate Adults {.tabset} 
```{r}

#All samples (Bow and ACWA)

adults<-subset_samples(sample_ps, Stage == "Adults")
adults<-prune_taxa(taxa_sums(adults) > 0, adults)
adults #116 samples, 1989 taxa

adults_rel_abund_phylum <- tax_glom(adults, taxrank="Phylum")
comp_rel_abund_phylum_adults <- microbiome::transform(adults_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_adults_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_adults) > 0.02, comp_rel_abund_phylum_adults) 

rel_abund_adults_df <- psmelt(comp_rel_abund_phylum_adults_prune)

rel_abund_adults_agg <- aggregate(Abundance ~ Phylum + Site, data = rel_abund_adults_df, FUN = mean) 

rel_abund_adults_phylum_plot <- ggplot(rel_abund_adults_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))+
  scale_fill_discrete(name=NULL)
rel_abund_adults_phylum_plot


top_phy_adults <- rel_abund_adults_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_adults
top5phyadults <- top_phy_adults$Phylum[1:15]
top5phyadults <- rel_abund_adults_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phyadults))


top5phyadultsplot<-ggplot(top5phyadults, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "PCR1", "PCR3")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phyadultsplot


#Bow River sites only to start

adults_bow<-subset_samples(sample_ps, Stage == "Adults" & Stream_Type =="Bow River")
adults_bow<-prune_taxa(taxa_sums(adults_bow) > 0, adults_bow)
adults_bow #100 samples, 1847 taxa

adults_rel_abund_phylum_bow <- tax_glom(adults_bow, taxrank="Phylum")
comp_rel_abund_phylum_adults_bow <- microbiome::transform(adults_rel_abund_phylum_bow, "compositional")
comp_rel_abund_phylum_adults_prune_bow = prune_taxa(taxa_sums(comp_rel_abund_phylum_adults_bow) > 0.02, comp_rel_abund_phylum_adults_bow) 

rel_abund_adults_df_bow <- psmelt(comp_rel_abund_phylum_adults_prune_bow)

rel_abund_adults_agg_bow <- aggregate(Abundance ~ Phylum + Site, data = rel_abund_adults_df_bow, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_adults_phylum_plot_bow <- ggplot(rel_abund_adults_agg_bow, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))+
  scale_fill_discrete(name=NULL)
rel_abund_adults_phylum_plot_bow
#Some change at Graves Bridge

top_phy_adults_bow <- rel_abund_adults_agg_bow %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_adults_bow
top5phyadults_bow <- top_phy_adults_bow$Phylum[1:15]
top5phyadults_bow <- rel_abund_adults_agg_bow %>%
    mutate(Phylum = fct_other(Phylum, top5phyadults_bow))


top5phyadultsplot_bow<-ggplot(top5phyadults_bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phyadultsplot_bow
#Higher Cyanobacteria and decrease in Proteobacteria at Graves bridge

#Family
adults_rel_abund_fam_bow <- tax_glom(adults_bow, taxrank="Family")
comp_rel_abund_fam_adults_bow <- microbiome::transform(adults_rel_abund_fam_bow, "compositional")
comp_rel_abund_fam_adults_prune_bow = prune_taxa(taxa_sums(comp_rel_abund_fam_adults_bow) > 0.02, comp_rel_abund_fam_adults_bow) 

rel_abund_adults_df_fam_bow <- psmelt(comp_rel_abund_fam_adults_prune_bow)

rel_abund_adults_agg_fam_bow <- aggregate(Abundance ~ Family + Site, data = rel_abund_adults_df_fam_bow, FUN = mean) 

top_fam_adults_bow <- rel_abund_adults_agg_fam_bow %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_adults_bow
top5famadults_bow <- top_fam_adults_bow$Family[1:42]
top5famadults_bow <- rel_abund_adults_agg_fam_bow %>%
    mutate(Family = fct_other(Family, top5famadults_bow))

top5famadultsplot_bow<-ggplot(top5famadults_bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "right") +
  guides (fill=guide_legend(nrow=16)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=8, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "pink", "darkcyan", "deeppink4", "burlywood3", "dark blue", "cornsilk", "darkgreen", "aliceblue", "darkgoldenrod2", "gray35"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5famadultsplot_bow

```

##### 3 panel facet with Adults orders 
```{r}
#Bow River and ACWA
rel_abund_adults_agg_order <- aggregate(Abundance ~ Phylum + Site + sample_Order, data = rel_abund_adults_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_adults_phylum_plot_order <- ggplot(rel_abund_adults_agg_order, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity", color="black") + 
  facet_wrap(~ sample_Order, scales ="free")+
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))
rel_abund_adults_phylum_plot_order

top_phy_adults_facet <- rel_abund_adults_agg_order %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_adults_facet
top5phyadults_facet <- top_phy_adults_facet$Phylum[1:18]
top5phyadults_facet
top5phyadults_facet <- rel_abund_adults_agg_order %>%
    mutate(Phylum = fct_other(Phylum, top5phyadults_facet))


top5phyadultsplot_facet<-ggplot(top5phyadults_facet, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "PCR1", "PCR3")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
  facet_wrap(~factor(sample_Order, levels=c("Trichoptera", "Ephemeroptera", "Diptera")), scales="free")+
geom_bar(stat="identity", color="black") + ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  xlab("Site")+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phyadultsplot_facet 


#Bow River only
rel_abund_adults_agg_order_bow <- aggregate(Abundance ~ Phylum + Site + sample_Order, data = rel_abund_adults_df_bow, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_adults_phylum_plot_order_bow <- ggplot(rel_abund_adults_agg_order_bow, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  facet_wrap(~ sample_Order, scales ="free")+
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="right", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(ncol=1)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))
rel_abund_adults_phylum_plot_order_bow

top_phy_adults_facet_bow <- rel_abund_adults_agg_order_bow %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_adults_facet_bow
top5phyadults_facet_bow <- top_phy_adults_facet_bow$Phylum[1:18]
top5phyadults_facet_bow
top5phyadults_facet_bow <- rel_abund_adults_agg_order_bow %>%
    mutate(Phylum = fct_other(Phylum, top5phyadults_facet_bow))


top5phyadultsplot_facet_bow<-ggplot(top5phyadults_facet_bow, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
  facet_wrap(~factor(sample_Order, levels=c("Trichoptera", "Ephemeroptera", "Diptera")), scales="free")+
geom_bar(stat="identity", color="black") + ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=8), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  xlab("Site")+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phyadultsplot_facet_bow 
```

#### Spiders {.tabset}
##### Spiders families separated 
```{r}
Spiders_ps<-subset_samples(sample_ps, Stage == "Spiders")
Spiders_ps<-prune_taxa(taxa_sums(Spiders_ps) > 0, Spiders_ps)
Spiders_ps #77 samples, 836 taxa

Spiders_rel_abund_phylum <- tax_glom(Spiders_ps, taxrank="Phylum")
comp_rel_abund_phylum_Spiders <- microbiome::transform(Spiders_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_Spiders_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_Spiders) > 0.02, comp_rel_abund_phylum_Spiders) 

rel_abund_Spiders_df <- psmelt(comp_rel_abund_phylum_Spiders_prune)

rel_abund_Spiders_agg <- aggregate(Abundance ~ Phylum + Site + sample_Family, data = rel_abund_Spiders_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_Spiders_phylum_plot <- ggplot(rel_abund_Spiders_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  facet_grid(~sample_Family, scales="free")+
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=14), axis.text.y=element_text(size=14),
        axis.title=element_text(size=14),
  axis.text.x=element_text(size=8))+
  scale_fill_discrete(name=NULL)
rel_abund_Spiders_phylum_plot
#Slight change at Sunalta or Policeman Flats (increase in cyanobacteria and firmicutes but not much different)

top_phy_Spiders <- rel_abund_Spiders_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_Spiders
top5phySpiders <- top_phy_Spiders$Phylum[1:15]
top5phySpiders
top5phySpiders <- rel_abund_Spiders_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phySpiders))
top5phySpiders


top5physpiderfamplot<- ggplot(top5phySpiders, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  facet_grid(~sample_Family, scales="free")+
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")

top5physpiderfamplot

#ggsave(filename="Microbiome_ED/Non decontaminated analysis/Relative Abundance Graphs/Spiders/Top 4 phyla tetragnaths and araneids.png", height=7, width=10)

#Family separated by taxa


Spiders_rel_abund_fam <- tax_glom(Spiders_ps, taxrank="Family")
comp_rel_abund_fam_Spiders <- microbiome::transform(Spiders_rel_abund_fam, "compositional")
comp_rel_abund_fam_Spiders_prune = prune_taxa(taxa_sums(comp_rel_abund_fam_Spiders) > 0.02, comp_rel_abund_fam_Spiders) 

rel_abund_Spiders_df_fam <- psmelt(comp_rel_abund_fam_Spiders_prune)

rel_abund_Spiders_agg_fam_sep <- aggregate(Abundance ~ Family + Site + sample_Family, data = rel_abund_Spiders_df_fam, FUN = mean) #aggregate to Phylum level, take the mean


top_fam_Spiders_sep <- rel_abund_Spiders_agg_fam_sep %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_Spiders_sep
top5famSpiders_sep <- top_fam_Spiders_sep$Family[1:37]
top5famSpiders_sep
top5famSpiders_sep<- rel_abund_Spiders_agg_fam_sep%>%
    mutate(Family = fct_other(Family, top5famSpiders_sep))
top5famSpiders_sep

top5famspidersplot_sep<-ggplot(top5famSpiders_sep, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family))+
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "right") +
  guides (fill=guide_legend(nrow=14)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "gray35"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")+
  facet_grid(~sample_Family, scales="free")

top5famspidersplot_sep
#Tetragnathidae is more infected by endosymbiont bacteria than araneidae

#ggsave(filename="Relative Abundance Graphs/Spiders/top 13 bacteria families in Araneidae and Tetragnathidae.png", height=7, width=10)


#Genera

Spiders_rel_abund_genus <- tax_glom(Spiders_ps, taxrank="Genus")
comp_rel_abund_genus_Spiders <- microbiome::transform(Spiders_rel_abund_genus, "compositional")
comp_rel_abund_genus_Spiders_prune = prune_taxa(taxa_sums(comp_rel_abund_genus_Spiders) > 0.02, comp_rel_abund_genus_Spiders) 

rel_abund_Spiders_df_genus <- psmelt(comp_rel_abund_genus_Spiders_prune)

rel_abund_Spiders_agg_genus <- aggregate(Abundance ~ Genus + Site + sample_Family, data = rel_abund_Spiders_df_genus, FUN = mean)


top_genus_Spiders_sep <- rel_abund_Spiders_agg_genus %>%
    group_by(Site, Genus) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_genus_Spiders_sep
top5genusSpiders_sep <- top_genus_Spiders_sep$Genus[1:35]
top5genusSpiders_sep
top5genusSpiders_sep<- rel_abund_Spiders_agg_genus%>%
    mutate(Genus = fct_other(Genus, top5genusSpiders_sep))
top5genusSpiders_sep

top5genusspidersplot_sep<-ggplot(top5genusSpiders_sep, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Genus))+
  #), levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "right") +
  guides (fill=guide_legend(nrow=14)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Genera", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "gray35"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")+
  facet_grid(~sample_Family, scales="free")

top5genusspidersplot_sep

#ggsave(filename="Relative Abundance Graphs/Spiders/top 13 genera Araneidae vs Tetragnathidae.png", height=7, width=10)

#Araneidae Phylum
aran_phy<-subset_samples(Spiders_ps, Family=="Araneidae")
rel_abun_aran_phy <- tax_glom(aran_phy, taxrank="Phylum")
comp_rel_abund_phy_aran <- microbiome::transform(rel_abun_aran_phy, "compositional")
comp_rel_abund_phy_aran_prune = prune_taxa(taxa_sums(comp_rel_abund_phy_aran) > 0.02, comp_rel_abund_phy_aran) 

rel_abund_aran_df_phy <- psmelt(comp_rel_abund_phy_aran_prune)

rel_abund_aran_agg_phy <- aggregate(Abundance ~ Phylum + Site, data = rel_abund_aran_df_phy, FUN = mean)


top_phy_aran <- rel_abund_aran_agg_phy %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5phyaran <- top_phy_aran$Phylum[1:12]
top5phyaran <- rel_abund_aran_agg_phy %>%
    mutate(Phylum = fct_other(Phylum, top5phyaran))

top5phyaranplot<-ggplot(top5phyaran, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")
top5phyaranplot

```

##### Spider families combined 

```{r}
rel_abund_Spiders_agg_combined <- aggregate(Abundance ~ Phylum + Site, data = rel_abund_Spiders_df, FUN = mean) #aggregate to Phylum level, take the mean

top_phy_Spiders_combined <- rel_abund_Spiders_agg_combined %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_Spiders_combined
top5phySpiders_combined <- top_phy_Spiders_combined$Phylum[1:15]
top5phySpiders_combined
top5phySpiders_combined <- rel_abund_Spiders_agg_combined %>%
    mutate(Phylum = fct_other(Phylum, top5phySpiders_combined))
top5phySpiders_combined

top5physpidersplot<-ggplot(top5phySpiders_combined, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "right") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")

top5physpidersplot

#ggsave(filename="Relative Abundance Graphs/Spiders/Top 4 phyla spiders combined.png", height=8, width=10)


#Family

Spiders_rel_abund_fam <- tax_glom(Spiders_ps, taxrank="Family")
comp_rel_abund_fam_Spiders <- microbiome::transform(Spiders_rel_abund_fam, "compositional")
comp_rel_abund_fam_Spiders_prune = prune_taxa(taxa_sums(comp_rel_abund_fam_Spiders) > 0.02, comp_rel_abund_fam_Spiders) 

rel_abund_Spiders_df_fam <- psmelt(comp_rel_abund_fam_Spiders_prune)

rel_abund_Spiders_agg_fam <- aggregate(Abundance ~ Family + Site, data = rel_abund_Spiders_df_fam, FUN = mean) #aggregate to Phylum level, take the mean


top_fam_Spiders_combined <- rel_abund_Spiders_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_Spiders_combined
top5famSpiders_combined <- top_fam_Spiders_combined$Family[1:36]
top5famSpiders_combined
top5famSpiders_combined <- rel_abund_Spiders_agg_fam%>%
    mutate(Family = fct_other(Family, top5famSpiders_combined))
top5famSpiders_combined

top5famspidersplot<-ggplot(top5famSpiders_combined, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF")), y=Abundance, fill=Family))+
  #), levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity", color="black") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10), legend.position = "right") +
  guides (fill=guide_legend(nrow=14)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=10),
        axis.text.x=element_text(size=10, color="black", angle=45, vjust=0.5),
                        axis.title=element_text(size=16))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "darkolivegreen4", "gray35"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")+
  xlab("Site")

top5famspidersplot
```

#### 3 panel plot larvae, Adults, Spiders 
```{r}
#Phyla
#Putting all 3 graphs together (Bow River and ACWA), separated by invertebrate taxa
a2<-top5phylarvaeplot_facet + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")

b2<-top5phyadultsplot_facet+ rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")

c2<-top5physpiderfamplot+ rremove("ylab")+rremove("xlab")

ad_lar_spiders<-ggarrange(a2, b2, c2, ncol=3, common.legend = TRUE, legend = "bottom", align="v")

annotate_figure(ad_lar_spiders, bottom = text_grob("Site", size = 16), left = text_grob("Mean Relative Abundance", rot = 90, size=16))

#Saving the file
ggsave(filename="Microbiome_ED/Non decontaminated analysis/Relative Abundance Graphs/All samples/Relative Abundance of Phyla Across Taxa 3 panel Plot separated by invert taxa type.png", height=7.5, width=22)


#Phyla separated by invertebrate host taxa - Bow River only
a1<-top5phylarvaebowplot_facet + rremove("ylab") + rremove("xlab") + rremove("legend.title") + rremove("legend")

b1<-top5phyadultsplot_facet_bow + rremove("ylab") +rremove("xlab") +rremove("legend.title") + rremove("legend")

c1<-top5physpiderfamplot + rremove("ylab")+rremove("xlab")

ad_lar_spiders_facet<-ggarrange(a1, b1, c1, ncol=3, common.legend = TRUE, legend = "bottom", align="v")

annotate_figure(ad_lar_spiders_facet, bottom = text_grob("Site", size = 16), left = text_grob("Mean Relative Abundance", rot = 90, size=16))

#ggsave(filename="Microbiome_ED/Non decontaminated analysis/Relative Abundance Graphs/All samples/Top 4 phyla larvae, adults, spiders, separated by invertebrate taxa type.png", height=8, width=20, bg="white")

#Family level - Bow River only
d2<-top5famlarvaeplot + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")
e2<-top5famadultsplot+ rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")
f2<-top5famspidersplot+ rremove("ylab")+rremove("xlab")

ad_lar_spiders_fam<-ggarrange(d2, e2, f2, labels = c("Larval Insects (n = 150)", "Adult Insects (n = 116)", "Spiders (n = 80)"), font.label=list(size=12), vjust=1.9, hjust=-0.3, ncol=3, nrow=1, common.legend = TRUE, legend = "right", align="v")

annotate_figure(ad_lar_spiders_fam, bottom = text_grob("Site", size = 16), left = text_grob("Mean Relative Abundance", rot = 90, size=16))

```

#### Caddisfly {.tabset}
##### Larval Hydropsychids {.tabset}
```{r}
hydro_lar<-subset_samples(sample_ps, Family == "Hydropsychidae")
hydro_lar<-prune_taxa(taxa_sums(hydro_lar) > 0, hydro_lar)
hydro_lar #60 samples, 2490 taxa

hydro_rel_abund_phylum <- aggregate_taxa(hydro_lar, "Phylum")
comp_rel_abund_phylum_hydro <- microbiome::transform(hydro_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_hydro_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_hydro) > 0.02, comp_rel_abund_phylum_hydro) 

rel_abund_hydro_df <- psmelt(comp_rel_abund_phylum_hydro_prune)

rel_abund_hydro_agg <- aggregate(Abundance ~ Phylum + Site  + Taxa, data = rel_abund_hydro_df, FUN = mean) #aggregate to Phylum level, take the mean
rel_abund_hydro_agg$Site <- factor(rel_abund_hydro_agg$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))
rel_abund_hydro_agg$Taxa<-factor(rel_abund_hydro_agg$Taxa, levels=c("Insect"), labels=c("Hydropsychidae"))


ggplot(rel_abund_hydro_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
#Not much change across any sites
#ggsave(filename= "Relative Abundance Graphs/Larval Hydropsychidae/Phylum greater than 2 percent of samples.png", height=7.5, width=7.0)

#Top 5 Phyla
top_phy_hydro <- rel_abund_hydro_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_hydro
top5phyhydro <- top_phy_hydro$Phylum[1:25]
top5phyhydro <- rel_abund_hydro_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phyhydro))


top5phyhydroplot<-ggplot(top5phyhydro, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.35, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=6)) +
  guides (fill=guide_legend(nrow=8)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "lightcoral"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")

#ggsave(filename= "Relative Abundance Graphs/Larval Hydropsychidae/Phylum top 5.png", height=7.5, width=7.0)

#Family level
hydro_rel_abund_fam <- tax_glom(hydro_lar, taxrank = "Family")
comp_rel_abund_fam_hydro <- microbiome::transform(hydro_rel_abund_fam, "compositional")
comp_rel_abund_fam_hydro_prune = prune_taxa(taxa_sums(comp_rel_abund_fam_hydro) > 0.2, comp_rel_abund_fam_hydro) 

rel_abund_hydro_df_fam <- psmelt(comp_rel_abund_fam_hydro_prune)

rel_abund_hydro_agg_fam <- aggregate(Abundance ~ Family + Site, data = rel_abund_hydro_df_fam, FUN = mean) 

rel_abund_hydro_agg_fam$Site <- factor(rel_abund_hydro_agg_fam$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))

ggplot(rel_abund_hydro_agg_fam, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  rremove("grid")

#ggsave(filename= "Relative Abundance Graphs/Larval Hydropsychidae/Family 20 percent of samples.png", height=7.5, width=7.0)

#Top 5 fam
top_fam_hydro <- rel_abund_hydro_agg_fam %>%
    group_by(Site, Family) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_fam_hydro
top5famhydro <- top_fam_hydro$Family[1:35]
top5famhydro <- rel_abund_hydro_agg_fam %>%
    mutate(Family = fct_other(Family, top5famhydro))


ggplot(top5famhydro, aes(x=Site, y=Abundance, fill=Family))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=9, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=10)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(9, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")+
  ggtitle("Larval Hydropsychidae top 8 most abundant bacterial families")

#ggsave(filename= "Relative Abundance Graphs/Larval Hydropsychidae/Phylum top 5.png", height=7.5, width=7.0)
```

##### Adults Trichoptera
```{r}
tricop_ad<-subset_samples(sample_ps, Order=="Trichoptera" & Stage=="Adults")
tricop_ad<-prune_taxa(taxa_sums(tricop_ad) > 0, tricop_ad)
tricop_ad #36 samples, 1110 taxa

tricop_rel_abund_phylum <- aggregate_taxa(tricop_ad, "Phylum")
comp_rel_abund_phylum_tricop <- microbiome::transform(tricop_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_tricop_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_tricop) > 0.02, comp_rel_abund_phylum_tricop) 

rel_abund_tricop_df <- psmelt(comp_rel_abund_phylum_tricop_prune)

rel_abund_tricop_agg <- aggregate(Abundance ~ Phylum + Site, data = rel_abund_tricop_df, FUN = mean) #aggregate to Phylum level, take the mean
rel_abund_tricop_agg$Site <- factor(rel_abund_tricop_agg$Site, levels=c("Cochrane", "Sunalta", "Cushing Bridge", "Graves Bridge", "Policeman Flats", "BRR2", "PCR1", "PCR3"))


ggplot(rel_abund_tricop_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
#Some change at Policeman Flats

#ggsave(filename= "Relative Abundance Graphs/Adults Trichoptera/Phylum greater than 2 percent of samples.png", height=7.5, width=7.0)


top_phy_tricop <- rel_abund_tricop_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_tricop
top5phytricop <- top_phy_tricop$Phylum[1:15]
top5phytricop <- rel_abund_tricop_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phytricop))


ggplot(top5phytricop, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.35, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=6)) +
  guides (fill=guide_legend(nrow=8)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")

#ggsave(filename= "Relative Abundance Graphs/Adults Trichoptera/Phylum top 5.png", height=7.5, width=7.0)
```

#### Mayfly {.tabset}
##### Larval Heptageniidae 
```{r}
hep_ps<-subset_samples(sample_ps, Family=="Heptageniidae")
hep_lar<-subset_samples(hep_ps, Stage=="Larvae")
hep_lar<-prune_taxa(taxa_sums(hep_lar)>0, hep_lar)
hep_lar #40 samples, 6960 taxa

hep_agg_taxa<-aggregate_taxa(hep_lar, "Phylum")
comp_hep_lar<-microbiome::transform(hep_agg_taxa, "compositional")
hep_prune<-prune_taxa(taxa_sums(comp_hep_lar)>0.02, comp_hep_lar)
hep_df<-psmelt(hep_prune)

hep_agg<-aggregate(Abundance~Phylum+Site+Taxa, data=hep_df, FUN=mean)

rel_abund_hep_lar_plot <- ggplot(hep_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
rel_abund_hep_lar_plot
#A bit of change at Cushing Bridge

top_phy_hep <- hep_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_hep
top5phyhep <- top_phy_hep$Phylum[1:18]
top5phyhep <- hep_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phyhep))


top5phyhepplot<-ggplot(top5phyhep, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Deferribacterota", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.35, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=6)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")

top5phyhepplot
```
##### Adult Ephemeroptera
```{r}
ephem_ps<-subset_samples(sample_ps, Order=="Ephemeroptera" & Stage=="Adults")
ephem_ps<-prune_taxa(taxa_sums(ephem_ps)>0, ephem_ps)
ephem_ps #56 samples, 1088 taxa

ephem_agg_taxa<-tax_glom(ephem_ps, taxrank="Phylum")
comp_ephem_ad<-microbiome::transform(ephem_agg_taxa, "compositional")
ephem_ad_prune<-prune_taxa(taxa_sums(comp_ephem_ad)>0.02, comp_ephem_ad)
ephem_ad_df<-psmelt(ephem_ad_prune)

ephem_agg<-aggregate(Abundance~Phylum+Site, data=ephem_ad_df, FUN=mean)

rel_abund_ephem_ad_plot <- ggplot(ephem_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
rel_abund_ephem_ad_plot

top_phy_ephem <- ephem_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top5phyephem <- top_phy_ephem$Phylum[1:18]
top5phyephem <- ephem_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phyephem))


top5phyephemplot<-ggplot(top5phyephem, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Deferribacterota", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.35, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=6)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")

top5phyephemplot

```

#### Adults and larval Chironomidae 
```{r}
chiro_ps<-subset_samples(sample_ps, Family == "Chironomidae")
chiro_ps<-subset_samples(chiro_ps, Site %in% c("Cochrane", "Cushing Bridge", "Policeman Flats")) 
chiro_ps<-prune_taxa(taxa_sums(chiro_ps) > 0, chiro_ps)
chiro_ps #34 samples, 1349 taxa

chiro_rel_abund_phylum <- aggregate_taxa(chiro_ps, "Phylum")
comp_rel_abund_phylum_chiro <- microbiome::transform(chiro_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_chiro_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_chiro) > 0.02, comp_rel_abund_phylum_chiro) 

rel_abund_chiro_df <- psmelt(comp_rel_abund_phylum_chiro_prune)

rel_abund_chiro_agg <- aggregate(Abundance ~ Phylum + Site  + Taxa, data = rel_abund_chiro_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_chiro_phylum_plot <- ggplot(rel_abund_chiro_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
rel_abund_chiro_phylum_plot
#Not much change across sites here


top_phy_chiro <- rel_abund_chiro_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_chiro
top5phychiro <- top_phy_chiro$Phylum[1:10]
top5phychiro <- rel_abund_chiro_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phychiro))


top5phychiroplot<-ggplot(top5phychiro, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Deferribacterota", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 8, color="black")) + 
  theme(legend.key.size = unit(0.35, "cm"),
        legend.text=element_text(size=6, face="italic"), legend.title=element_text(size=6)) +
  guides (fill=guide_legend(nrow=12)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=10))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent)+ rremove("grid")

top5phychiroplot
```

#### Perlidae 
```{r perlidae}

perl_ps<-subset_samples(sample_ps, Family == "Perlidae")
perl_ps<-subset_samples(perl_ps, Site %in% c("Cochrane", "Cushing Bridge", "Policeman Flats")) 
perl_ps<-prune_taxa(taxa_sums(perl_ps) > 0, perl_ps)
perl_ps #21 samples, 3114 taxa

perl_rel_abund_phylum <- aggregate_taxa(perl_ps, "Phylum")
comp_rel_abund_phylum_perl <- microbiome::transform(perl_rel_abund_phylum, "compositional")
comp_rel_abund_phylum_perl_prune = prune_taxa(taxa_sums(comp_rel_abund_phylum_perl) > 0.02, comp_rel_abund_phylum_perl) 

rel_abund_perl_df <- psmelt(comp_rel_abund_phylum_perl_prune)

rel_abund_perl_agg <- aggregate(Abundance ~ Phylum + Site  + Taxa, data = rel_abund_perl_df, FUN = mean) #aggregate to Phylum level, take the mean

rel_abund_perl_phylum_plot <- ggplot(rel_abund_perl_agg, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >2%") +
  theme(strip.text = element_text(size = 14, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=14),axis.text.y=element_text(size=14), axis.title=element_text(size=14))+
  scale_fill_discrete(name=NULL)
rel_abund_perl_phylum_plot
#Some change at Cushing Bridge

top_phy_perl <- rel_abund_perl_agg %>%
    group_by(Site, Phylum) %>%
    summarize(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_phy_perl
top5phyperl <- top_phy_perl$Phylum[1:11]

top5phyperl <- rel_abund_perl_agg %>%
    mutate(Phylum = fct_other(Phylum, top5phyperl))


top5phyperlplot<-ggplot(top5phyperl, aes(x=Site, y=Abundance, fill=factor(Phylum, levels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria")), order=-as.numeric(Phylum)))+ 
geom_bar(stat="identity") +   ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=8), legend.position = "bottom") +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=8),
        axis.text.x=element_text(size=6, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(3, "Paired"), "gray35", "darkolivegreen4"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5phyperlplot

#Putting all taxa phyla plots together
a<-top5phyhydroplot + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")+coord_cartesian(clip = "off")
b<-top5phyhepplot + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")+coord_cartesian(clip = "off")
c<-top5phychiroplot + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")+coord_cartesian(clip = "off")
d<-top5phyperlplot + rremove("ylab")+rremove("xlab")+rremove("legend.title")+rremove("legend")+coord_cartesian(clip = "off")

fourtaxa<- ggarrange(a, b, c, d, common.legend=T, legend="right", labels = c("Hydropsychidae", "Heptageniidae", "Chironomidae", "Perlidae"), font.label=list(size=9), vjust=1.2, hjust=-0.6, ncol=2, nrow=2)
annotate_figure(fourtaxa, bottom = text_grob("Site", size = 16), left = text_grob("Mean Relative Abundance", rot = 90, size=16)) 

#Saving the file
#ggsave(filename="Relative Abundance of Bacterial Phyla Across Taxa Plot.pdf", plot=last_plot(), dpi=300, height=8, width=7)

```


# Alpha Diversity 

Observed richness is the number of unique ASVs per sample (species richness). Count data ranging from 0 - infinity.
Shannon diversity is a measure of the richness and relative abundance (evenness) of bacteria in a sample (H=[(pi)log(pi)]). Usually ranges between 1.5 - 3.5, rarely above 4.5.
Simpson diversity is a measure of the richness and relative abundance (evenness) of bacteria in a sample (D=n i(ni1)/N(N1)). Ranges between 0-1 (1 being the most diverse, 0 being not diverse). 

When all samples are combined there are only significant differences in alpha diversity between BRR2 and Cochrane, Sunalta, and Cushing Bridge. There are no differences between ACWA stream treatments nor between Bow River sites. 

Invertebrates separated had 

Spiders samples showed no significant difference in richness or alpha diversity between sites when separated from insects. 

Overall, seems as though wastewater effluent does not affect invertebrate or Spiders microbial richness or alpha diversity. 


``` {r Alpha Diversity}
#### All samples combined ####
#Alpha Diversity 
#Check to see if total reads is correlated with observed richness. Should be since you likely have more taxa of bacteria with a higher total abundance. 
ggplot(data = data.frame("total_reads" =  phyloseq::sample_sums(sample_ps),
                         "observed" = phyloseq::estimate_richness(sample_ps, measures = "Observed")[, 1]),
       aes(x = total_reads, y = observed)) +
  geom_point() +
  geom_smooth(method="lm", se = FALSE) +
  labs(x = "\nTotal Reads", y = "Observed Richness\n")
#Looks like there is not a positive correlation btw increasing reads/total abundance and observed richness. This might be because it only takes a certain amount of reads to capture all the unique bacteria so past this point there is no increase. Could also be that either  sequencing errors are causing there to be really low amount of richness even though there are a large amount of reads?

#Alpha diversity with rarefied data (even sequencing depth). 

#Rarefying
(ps_rare <- phyloseq::rarefy_even_depth(sample_ps, rngseed = 123, replace = FALSE))  
head(phyloseq::sample_sums(ps_rare)) #minimum sequencing depth, 2048. #4,379 taxa 318 samples. Eliminated about 10,000 taxa. Quite a bit of the total ASVs. 

#Generate a data frame with diversity measures
adiv_rare <- data.frame(
  "Observed" = phyloseq::estimate_richness(ps_rare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ps_rare, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(ps_rare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(ps_rare)$Site, 
  "Family" = phyloseq::sample_data(ps_rare)$Family, 
  "Taxa"=phyloseq::sample_data(ps_rare)$Taxa,
  "Order" = phyloseq::sample_data(ps_rare)$Order)
head(adiv_rare)

#Plot adiv measures
adiv_rare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metric") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))

#Summarize
adiv_rare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_Simpson = mean(Simpson),
            sd_Simpson = sd(Simpson))


shapiro.test(adiv_rare$Observed) #Not normal
shapiro.test(adiv_rare$Shannon) #Not normal
shapiro.test(adiv_rare$Simpson) #Not normal

#Statistical test. Use Kruskal test bc the data is not normal and we have more than 2 sites (non-parametric ANOVA)
#Observed richness across sites
kruskal.test(Observed ~ Site, data = adiv_rare) #p<0.001. Significant differences across sites
dunnTest(Observed~Site, data=adiv_rare)
#Significant differences in bacterial richness between:
#BRR2 - Cochrane p=0.002
#BRR2 - Cushing Bridge p=0.0004
#BRR2 - Cushing Bridge p=0.01
#BRR2 - Sunalta p=0.0007
#BRR2 - Policeman Flats p=0.03
#Cochrane - PCR1 p=0.024
#Cushing Bridge - PCR1 p=0.0042
#Cochrane - PCR3 p<0.001
#PCR1 - Sunalta  p= 0.009
#PCR3 - Sunalta  p=0.0003
#PCR3 - Policeman Flats p=0.034
#Cushing Bridge - PCR3 p<0.001
#Graves Bridge - PCR3 p=0.01


kruskal.test(Shannon ~ Site, data = adiv_rare) #p=0.003. Significant differences across sites
dunnTest(Shannon~Site, data=adiv_rare)
#Significant differences in shannon diversity between:
#BRR2 - Sunalta p=0.019
#BBR2 - Cushing Bridge p=0.012
#BRR2 - Cochrane p=0.03
#PCR3 - Sunalta  p= 0.012
#Cushing Bridge - PCR3 p= 0.01269122
#Cochrane - PCR3 p= 0.04696715

kruskal.test(Simpson ~ Site, data = adiv_rare) #p=0.014
dunnTest(Simpson~Site, data=adiv_rare) 
#Significant differences in bacterial alpha diversity between:
#BRR2 - Sunalta p=0.028
#BRR2 - Cushing Bridge p=0.025
#BRR2 - Cochrane p=0.047

ps4 <- subset_samples(ps2, sample_sums(ps2) > 5000) 
ps4
sample_ps1 <- prune_taxa(taxa_sums(ps4) > 5, ps4)
sample_ps1<- prune_taxa(taxa_sums(sample_ps1) > 0, sample_ps1)
sample_ps1 #4428 taxa, 294 samples
ps_rare2 <- phyloseq::rarefy_even_depth(sample_ps1, rngseed = 123, replace = FALSE)

head(phyloseq::sample_sums(ps_rare2)) #5180 bacterial taxa per sample

#Rarefied to 5000 reads
adiv_rare2 <- data.frame(
  "Observed" = phyloseq::estimate_richness(ps_rare2, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ps_rare2, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(ps_rare2, measures = "Simpson"),
  "Site" = phyloseq::sample_data(ps_rare2)$Site, 
  "Family" = phyloseq::sample_data(ps_rare2)$Family, 
  "Taxa"=phyloseq::sample_data(ps_rare2)$Taxa)
head(adiv_rare2)

#Plot adiv measures
adiv_rare2 %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metric") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))

#Observed richness across sites
kruskal.test(Observed ~ Site, data = adiv_rare2) #p<0.001. Significant differences across sites
dunnTest(Observed~Site, data=adiv_rare2)
#Same reult as ps_rare (2000 reads)

kruskal.test(Shannon ~ Site, data = adiv_rare2) #p=0.003. Significant differences across sites
dunnTest(Shannon~Site, data=adiv_rare2)
#Same result as ps_rare

kruskal.test(Simpson ~ Site, data = adiv_rare2) #p=0.014
dunnTest(Simpson~Site, data=adiv_rare2)
#Same result as ps_rare.

#No difference in rarefied data to 2000 reads or 5000 reads.

#### Larval and Adult macroinvertebrates and Spiders ####
#### All samples separated by invertebrate family/order
er <- estimate_richness(ps_rare, measures=c("Observed", "Shannon", "Simpson"))
er <- data.frame(er)
er<- rownames_to_column(er, var="sample_id")
metadata_alpha_div<-rownames_to_column(metadata, var="sample_id")

div.df<-left_join(er, metadata_alpha_div, by = "sample_id")

#Observed richness across sites
rich.plot<-ggplot(div.df, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Observed))+
  facet_grid(~factor(Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scales = "free")+
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Observed richness") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()
rich.plot
#Lower observed richness in Spiders

#Shannon diversity across sites
shan.plot<-ggplot(div.df, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Shannon))+
  facet_grid(~factor(Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scales = "free")+
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Shannon Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))
shan.plot

#Spiders have a lower Shannon diversity than insects

#Simpson diversity across sites
simp.plot<-ggplot(div.df, aes(x=factor(Site, labels=c("COCH", "SUN", "CUSHBR", "GRVBR", "PMF", "BRR2", "PCR1", "PCR3")), y=Simpson))+
  facet_grid(~factor(Family, levels=c("Hydropsychidae", "Heptageniidae", "Baetidae", "Chironomidae", "Perlidae", "Trichoptera", "Ephemeroptera", "Diptera", "Araneidae", "Tetragnathidae")), scales = "free")+
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(y = "Simpson Diversity") +
  theme_bw()+
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black")) +
   theme(axis.title.x = element_blank())+
  rremove("legend")+
  rremove("xlab")
simp.plot

#3 panel plot

rich.plot2<-rich.plot+ rremove("xlab") + rremove("legend")
shan.plot2<-shan.plot + rremove("xlab") + rremove("legend")
simp.plot2<-simp.plot + rremove("xlab") + rremove("legend") 

div.plots<-ggarrange(rich.plot2 + theme(axis.text.x = element_blank(),
                     axis.ticks.x = element_blank(),
                     axis.title.x = element_blank()), shan.plot2 + theme(axis.text.x =
                     element_blank(), axis.ticks.x = element_blank(),
                     axis.title.x = element_blank(), strip.background = element_blank(),
  strip.text.x = element_blank()), simp.plot + theme(strip.background = element_blank(),
  strip.text.x = element_blank()), nrow=3, align="v")

annotate_figure(div.plots, bottom = text_grob("Site", size = 12, vjust=0.5))

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Alpha Diversity Graphs/Richness, Shannon, Simpson diversity across invertebrate taxa.png", height=20, width=16, bg="white")

#### Spiders ####
Spiders_psrare<-subset_samples(ps_rare, Taxa == "Spider")

adivSpidersrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(Spiders_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(Spiders_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(Spiders_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(Spiders_psrare)$Site,
  "Taxa" = phyloseq::sample_data(Spiders_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(Spiders_psrare)$Family,
  "Order"= phyloseq::sample_data(Spiders_psrare)$Order)
head(adivSpidersrare)

Spiders.alpha.div.plot<-adivSpidersrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()+rremove("grid")+
  theme_bw()

adivSpidersrare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_simpson = mean(Simpson), 
            sd_simpson = sd(Simpson))

kruskal.test(Observed ~ Site, data = adivSpidersrare) #p = 0.15. No differences across sites.
dunnTest(Observed~Site, data=adivSpidersrare)


kruskal.test(Shannon ~ Site, data = adivSpidersrare) #p=0.06 No sig differences across sites.
dunnTest(Shannon~Site, data=adivSpidersrare) 

kruskal.test(Simpson ~ Site, data = adivSpidersrare) #p=0.06 No sig differences across sites.
dunnTest(Shannon~Site, data=adivSpidersrare) 


#### Larval macroinvertebrates ####
lar_psrare<-subset_samples(ps_rare, Stage=="Larvae")

adivlarrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(lar_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(lar_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(lar_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(lar_psrare)$Site,
  "Taxa" = phyloseq::sample_data(lar_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(lar_psrare)$Family, 
  "Order"= phyloseq::sample_data(lar_psrare)$Order)
head(adivlarrare)

lar.alpha.div.plot<-adivlarrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()+
  rremove("grid")+
  theme_bw()

adivlarrare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_simpson = mean(Simpson),
            sd_simpson = sd(Simpson))

kruskal.test(Observed ~ Site, data = adivlarrare) #p <0.001. sig differences across sites.
dunnTest(Observed~Site, data=adivlarrare)


kruskal.test(Shannon ~ Site, data = adivlarrare) #p<0.001 sig differences across sites.
dunnTest(Shannon~Site, data=adivlarrare) 


kruskal.test(Simpson ~ Site, data = adivlarrare) #p=0.002 sig differences across sites.
dunnTest(Shannon~Site, data=adivlarrare) 

#### Adult macroinvertebrates ####

ad_psrare<-subset_samples(ps_rare, Stage=="Adults")

adivadrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(ad_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ad_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(ad_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(ad_psrare)$Site,
  "Taxa" = phyloseq::sample_data(ad_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(ad_psrare)$Family, 
  "Order"= phyloseq::sample_data(ad_psrare)$Order)
head(adivadrare)

adults.alpha.div.plot<-adivadrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()+
  rremove("grid")+
  theme_bw()
adults.alpha.div.plot

adivadrare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_simpson = mean(Simpson),
            sd_simpson = sd(Simpson))

kruskal.test(Observed ~ Site, data = adivadrare) #p = 0.21. No differences across sites.
dunnTest(Observed~Site, data=adivadrare)


kruskal.test(Shannon ~ Site, data = adivadrare) #p=0.42 No differences across sites.
dunnTest(Shannon~Site, data=adivadrare) 


kruskal.test(Simpson ~ Site, data = adivadrare) #p=0.61 No differences across sites.
dunnTest(Shannon~Site, data=adivadrare) 


#### Larval and Adults Caddisfly ####
hydro_psrare<-subset_samples(ps_rare, Family == "Hydropsychidae")
hydro_psrare<-subset_samples(hydro_psrare, Stage=="Larvae")

#Larvae
adivhydrorare <- data.frame(
  "Observed" = phyloseq::estimate_richness(hydro_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(hydro_psrare, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(hydro_psrare, measures = "InvSimpson"),
  "Simpson" = phyloseq::estimate_richness(hydro_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(hydro_psrare)$Site,
  "Taxa" = phyloseq::sample_data(hydro_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(hydro_psrare)$Family)
head(adivhydrorare)

adivhydrorare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free", nrow=1) +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()


kruskal.test(Observed ~ Site, data = adivhydrorare) #p < 0.001. sig differences across sites.
dunnTest(Observed~Site, data=adivhydrorare)
#Policeman Flats - Sunalta p=0.0004
#PCR1 - Sunalta p=0.0006
#BRR2 - Sunalta p=0.045
#Cushing Bridge - Policeman Flats p=0.008
#Cochrane - Policeman Flats p=0.0026
#Cushing Bridge - PCR1 p=0.012
#Cochrane - PCR1 p=0.004


kruskal.test(Shannon ~ Site, data = adivhydrorare) #p<0.001 sig differences across sites.
dunnTest(Shannon~Site, data=adivhydrorare) 
#Cochrane - Policeman Flats p=0.00065
#Cushing Bridge - Policeman Flats p=0.0023
#PCR1 - Sunalta p=0.026
#Policeman Flats - Sunalta p=0.00023

kruskal.test(Simpson ~ Site, data = adivhydrorare) #p<0.001 sig differences across sites.
dunnTest(Simpson~Site, data=adivhydrorare) 
#Cochrane - Policeman Flats p=0.006
#Cushing Bridge - Policeman Flats p=0.006
#Policeman Flats - Sunalta p=0.00076

adivhydrorare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_simpson = mean(Simpson),
            sd_simpson = sd(Simpson))

#Adults
tricop_rare<-subset_samples(ps_rare, Order=="Tricoptera")
tricop_rare<-subset_samples(ps_rare, Stage=="Adults")

adivtricoprare <- data.frame(
  "Observed" = phyloseq::estimate_richness(tricop_rare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(tricop_rare, measures = "Shannon"),
  "InvSimpson" = phyloseq::estimate_richness(tricop_rare, measures = "InvSimpson"),
  "Simpson" = phyloseq::estimate_richness(tricop_rare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(tricop_rare)$Site,
  "Taxa" = phyloseq::sample_data(tricop_rare)$Taxa, 
  "Family"= phyloseq::sample_data(tricop_rare)$Family)
head(adivhydrorare)

adivtricoprare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free", nrow=1) +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivtricoprare %>%
  group_by(Site) %>%
  dplyr::summarise(mean_observed = mean(Observed),
                   sd_observed = sd(Observed),
            mean_shannon = mean(Shannon),
            sd_shannon = sd(Shannon),
            mean_simpson = mean(Simpson),
            sd_simpson = sd(Simpson))

kruskal.test(Observed ~ Site, data = adivtricoprare) #p = 0.21. 
dunnTest(Observed~Site, data=adivtricoprare)
#No differences

kruskal.test(Shannon ~ Site, data = adivtricoprare) #p = 0.42 
dunnTest(Shannon~Site, data=adivtricoprare) 
#No differences

kruskal.test(Simpson ~ Site, data = adivtricoprare) #p = 0.61 
dunnTest(Simpson~Site, data=adivtricoprare) 
#No differences


#### Larval and Adults mayflies ####
#Larval Heptageniidae 
hep_psrare<-subset_samples(ps_rare, Family == "Heptageniidae")
hep_psrare<-subset_samples(hep_psrare, Stage=="Larvae")

adivheprare <- data.frame(
  "Observed" = phyloseq::estimate_richness(hep_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(hep_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(hep_psrare, measures = "Simpson"),
  "InvSimpson" = phyloseq::estimate_richness(hep_psrare, measures = "InvSimpson"),
  "Site" = phyloseq::sample_data(hep_psrare)$Site,
  "Taxa" = phyloseq::sample_data(hep_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(hep_psrare)$Family)
head(adivheprare)

adivheprare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free", nrow=1) +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivheprare %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))


kruskal.test(Observed ~ Site, data = adivheprare) #p = 0.042.
dunnTest(Observed~Site, data=adivheprare)
#Difference only between Cushing Bridge - Policeman Flats p=0.031

kruskal.test(Shannon ~ Site, data = adivheprare) #p=0.048 
dunnTest(Shannon~Site, data=adivheprare) 
#No differences

kruskal.test(InvSimpson ~ Site, data = adivheprare) #p=0.031
dunnTest(InvSimpson~Site, data=adivheprare)
#Cochrane - Policeman Flats p=0.024

kruskal.test(Simpson ~ Site, data = adivheprare) #p=0.031 sig differences across sites.
dunnTest(Simpson~Site, data=adivheprare) 
#Cochrane - Policeman Flats p=0.024
#Simpson and Inverse Simpson gives the same results just on a different axis


#Adults Ephemeroptera
ephem_psrare<-subset_samples(ps_rare, Order == "Ephemeroptera")
ephem_psrare<-subset_samples(ephem_psrare, Stage == "Adults")

adivephemrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(ephem_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(ephem_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(ephem_psrare, measures = "Simpson"),
  "InvSimpson" = phyloseq::estimate_richness(ephem_psrare, measures = "InvSimpson"),
  "Site" = phyloseq::sample_data(ephem_psrare)$Site,
  "Taxa" = phyloseq::sample_data(ephem_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(ephem_psrare)$Family)
head(adivephemrare)

adivephemrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free", nrow=1) +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()
#Organisms at Cushing Bridge looks much lower in diversity than the rest of the sites
#It's likely that the Adults at Cushing Bridge aren't from the same family since there are two points that have really high richness/diversity but the rest are very low. 

kruskal.test(Observed ~ Site, data = adivephemrare) #p = 0.058.
dunnTest(Observed ~ Site, data = adivephemrare)
#Cushing Bridge - Policeman Flats p=0.047

kruskal.test(Shannon ~ Site, data = adivephemrare) #p=0.21
dunnTest(Shannon ~ Site, data = adivephemrare)
#No differences

kruskal.test(InvSimpson ~ Site, data = adivephemrare) #p=0.19
dunnTest(InvSimpson~ Site, data = adivephemrare)
#No differences

#Baetidae
baetid_psrare<-subset_samples(ps_rare, Family == "Baetidae")
baetid_lar<-subset_samples(baetid_psrare, Stage=="Larvae")

#Larvae
adivbaetidrarelar <- data.frame(
  "Observed" = phyloseq::estimate_richness(baetid_lar, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(baetid_lar, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(baetid_lar, measures = "Simpson"),
  "InvSimpson" = phyloseq::estimate_richness(baetid_lar, measures = "InvSimpson"),
  "Site" = phyloseq::sample_data(baetid_lar)$Site,
  "Taxa" = phyloseq::sample_data(baetid_lar)$Taxa, 
  "Family"= phyloseq::sample_data(baetid_lar)$Family)
head(adivbaetidrarelar)

adivbaetidrarelar %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "InvSimpson", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "InvSimpson", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free", nrow=1) +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

kruskal.test(Observed ~ Site, data = adivbaetidrarelar) #p = 0.73. No diff.


kruskal.test(Shannon ~ Site, data = adivbaetidrarelar) #p=0.91. No diff.


kruskal.test(InvSimpson ~ Site, data = adivbaetidrarelar) #p=0.73. No diff.

#Can't compare adults because the filtering steps removed all PCR3 baetid adults. Only have 4 PCR1 ones left. 

#### Chironomidae ####
chiro_psrare<-subset_samples(ps_rare, Family == "Chironomidae")

#Larvae
chiro_psrare_lar<-subset_samples(chiro_psrare, Stage == "Larvae")

adivchirorarelar <- data.frame(
  "Observed" = phyloseq::estimate_richness(chiro_psrare_lar, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(chiro_psrare_lar, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(chiro_psrare_lar, measures = "Simpson"),
  "Site" = phyloseq::sample_data(chiro_psrare_lar)$Site,
  "Taxa" = phyloseq::sample_data(chiro_psrare_lar)$Taxa, 
  "Family"= phyloseq::sample_data(chiro_psrare_lar)$Family)
head(adivchirorarelar)

adivchirorarelar %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivchirorarelar %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))


shapiro.test(adivchirorarelar$Observed) #normal dist.
summary(aov1<-aov(Observed~Site, data = adivchirorarelar)) #p=0.074
TukeyHSD(aov1) #no differences. 

kruskal.test(Observed ~ Site, data = adivchirorarelar) #p = 0.08
dunnTest(Observed~Site, data=adivchirorarelar)
#No differences. 

#Gives the same result with parametric or non-parametric test


shapiro.test(adivchirorarelar$Shannon) #Normal
summary(aov2<-aov(Shannon~Site, data = adivchirorarelar)) #p=0.124
TukeyHSD(aov2) #no differences

kruskal.test(Shannon ~ Site, data = adivchirorarelar) #p=0.15
dunnTest(Shannon~Site, data=adivchirorarelar) 
#No differences

kruskal.test(Simpson ~ Site, data = adivchirorarelar) #p=0.41
dunnTest(Simpson~Site, data=adivchirorarelar) 
#No differences.

#Adults
chiro_psrare_ad<-subset_samples(chiro_psrare, Stage == "Adults")

adivchiroraread <- data.frame(
  "Observed" = phyloseq::estimate_richness(chiro_psrare_ad, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(chiro_psrare_ad, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(chiro_psrare_ad, measures = "Simpson"),
  "Site" = phyloseq::sample_data(chiro_psrare_ad)$Site,
  "Taxa" = phyloseq::sample_data(chiro_psrare_ad)$Taxa, 
  "Family"= phyloseq::sample_data(chiro_psrare_ad)$Family)
head(adivchiroraread)

adivchiroraread %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivchiroraread %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))


kruskal.test(Observed ~ Site, data = adivchiroraread) #p = 0.60
dunnTest(Observed~Site, data=adivchiroraread)
#No differences. 

kruskal.test(Shannon ~ Site, data = adivchiroraread) #p=0.68
dunnTest(Shannon~Site, data=adivchiroraread) 
#No differences

kruskal.test(Simpson ~ Site, data = adivchiroraread) #p=0.59
dunnTest(Simpson~Site, data=adivchiroraread) 
#No differences.

#### Perlidae ####
perl_psrare<-subset_samples(ps_rare, Family == "Perlidae")

adivperlrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(perl_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(perl_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(perl_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(perl_psrare)$Site,
  "Taxa" = phyloseq::sample_data(perl_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(perl_psrare)$Family)
head(adivperlrare)

adivperlrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivperlrare %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))


kruskal.test(Observed ~ Site, data = adivperlrare) #p = 0.092
dunnTest(Observed~Site, data=adivperlrare)
#No differences. 

kruskal.test(Shannon ~ Site, data = adivperlrare) #p=0.46
dunnTest(Shannon~Site, data=adivperlrare) 
#No differences

kruskal.test(Simpson ~ Site, data = adivperlrare) #p=0.85
dunnTest(Simpson~Site, data=adivperlrare) 
#No differences


#### Araneidae ####
aran_psrare<-subset_samples(ps_rare, Family == "Araneidae")

adivaranrare <- data.frame(
  "Observed" = phyloseq::estimate_richness(aran_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(aran_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(aran_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(aran_psrare)$Site,
  "Taxa" = phyloseq::sample_data(aran_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(aran_psrare)$Family)
head(adivaranrare)

adivaranrare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

kruskal.test(Observed ~ Site, data = adivaranrare) #p = 0.16
dunnTest(Observed~Site, data=adivaranrare)
#No differences

kruskal.test(Shannon ~ Site, data = adivaranrare) #p=0.13
dunnTest(Shannon~Site, data=adivaranrare) 
#No differences

kruskal.test(Simpson ~ Site, data = adivaranrare) #p=0.09
dunnTest(Simpson~Site, data=adivaranrare) 
#No differences


#### Tetragnathidae ####

tetra_psrare<-subset_samples(ps_rare, Family == "Tetragnathidae")

adivtetrarare <- data.frame(
  "Observed" = phyloseq::estimate_richness(tetra_psrare, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(tetra_psrare, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(tetra_psrare, measures = "Simpson"),
  "Site" = phyloseq::sample_data(tetra_psrare)$Site,
  "Taxa" = phyloseq::sample_data(tetra_psrare)$Taxa, 
  "Family"= phyloseq::sample_data(tetra_psrare)$Family)
head(adivtetrarare)

adivtetrarare %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

kruskal.test(Observed ~ Site, data = adivtetrarare) #p = 0.49
dunnTest(Observed~Site, data=adivtetrarare)
#no diff

kruskal.test(Shannon ~ Site, data = adivtetrarare) #p=0.31
dunnTest(Shannon~Site, data=adivtetrarare) 
#No differences

kruskal.test(Simpson ~ Site, data = adivtetrarare) #p=0.27
dunnTest(Simpson~Site, data=adivtetrarare) 
#No differences

#### Non-rarefied alpha diversity ####
#From Joey711 GitHub: You do not want to transform or filter your data before estimating richness, other than quality assurance filtering that would remove non-target sequences. Usually that sort of filtering would be done on the sequence data before it is in the form of a contingency table (table of counts). Basically, the very first table of counts that you have in your workflow is probably the one that you want to use for estimate_richness and plot_richness. There are several different alpha diversity estimators supported in the estimate_richness function, and they all incorporate library size into their estimates in different ways... Except for the "Observed" option, which is simply showing you graphically the OTUs that were observed at least once in each sample. I think this is the one you were actually asking about.To be clear, the other methods require that you use raw counts. Do not use rarefied counts for the Chao-I estimate, or the Shannon index, for example. The "Observed" option is not a method at all, just showing you what is in your data as you provided. If you transform your data, especially if you rarefy, and then you want to estimate richness, the "Observed" result is now the only one still available to you at all. Rarefying reduces the precision with which you would estimate the diversity in the first place, and so this generally shouldn't be done. I suspect, however, that is the workflow you had in mind. All alpha-diversity indices/estimates are aware of differences in sample size (library size, number of reads in this case), because this has always been a problem when attempting to count things, even trees in a forest (see Sanders original paper describing rarefaction, a different technique than rarefying, if rarefying can be called a technique).

#Now trying without rarefaction. 
##alpha diversity metrics

adiv <- data.frame(
  "Observed" = phyloseq::estimate_richness(sample_ps, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(sample_ps, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(sample_ps, measures = "Simpson"),
  "Site" = phyloseq::sample_data(sample_ps)$Site,
  "Taxa" = phyloseq::sample_data(sample_ps)$Taxa, 
  "Family"= phyloseq::sample_data(sample_ps)$Family)
head(adiv)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adiv %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))

#Subsetting insects
insect_ps<-subset_samples(sample_ps, Taxa == "Insect")

adivinsect <- data.frame(
  "Observed" = phyloseq::estimate_richness(insect_ps, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(insect_ps, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(insect_ps, measures = "Simpson"),
  "Site" = phyloseq::sample_data(insect_ps)$Site,
  "Taxa" = phyloseq::sample_data(insect_ps)$Taxa, 
  "Family"= phyloseq::sample_data(insect_ps)$Family)
head(adivinsect)

adivinsect %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivinsect %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))



kruskal.test(Observed ~ Site, data = adivinsect) #p<0.003. Significant differences across sites
dunnTest(Observed~Site, data=adivinsect)
#Diff btw BRR2 and Cochrane, Sunalta, Cushing Bridge. Same as rarefied analysis
kruskal.test(Shannon ~ Site, data = adivinsect) #p=0.008
dunnTest(Shannon~Site, data=adivinsect) 
#Diff btw BRR2 Cushing Bridge. Same as rarefied analysis
kruskal.test(Simpson ~ Site, data = adivinsect) #p=0.035
dunnTest(Simpson~Site, data=adivinsect) 
#No sig difference

#Spiders samples
Spiders_ps<-subset_samples(sample_ps, Taxa == "Spider")

adivSpiders <- data.frame(
  "Observed" = phyloseq::estimate_richness(Spiders_ps, measures = "Observed"),
  "Shannon" = phyloseq::estimate_richness(Spiders_ps, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(Spiders_ps, measures = "Simpson"),
  "Site" = phyloseq::sample_data(Spiders_ps)$Site,
  "Taxa" = phyloseq::sample_data(Spiders_ps)$Taxa, 
  "Family"= phyloseq::sample_data(Spiders_ps)$Family)
head(adivSpiders)

adivSpiders %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Site, y = value)) +
  geom_boxplot(outlier.color = NA) +
  geom_jitter(aes(color = Site), height = 0, width = .2) +
  labs(x = "Site", y = "Diversity metrics") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45, hjust=1, size=8, color="black"))+
  scale_y_log10()

adivSpiders %>%
  group_by(Site) %>%
  dplyr::summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_pd = median(Simpson))

adivSpiders %>%
  group_by(Site) %>%
  dplyr::summarise(meann_observed = mean(Observed),
            mean_shannon = mean(Shannon),
            mean_pd = mean(Simpson))


kruskal.test(Observed ~ Site, data = adivSpiders) #p=0.15.
dunnTest(Observed~Site, data=adivinsect)
#no sig diff
kruskal.test(Shannon ~ Site, data = adivSpiders) #p=0.36
dunnTest(Shannon~Site, data=adivSpiders) 
#no sig diff
kruskal.test(Simpson ~ Site, data = adivSpiders) #p=0.38
dunnTest(Simpson~Site, data=adivSpiders) 
#No sig difference

```

# Beta Diversity
Beta diversity is a measure of dissimilarity between communities. It is useful to see how the bacterial community changes across sites/invertebrate families/exposures/etc. This analysis was done across sites on rarefied data and was split by invertebrate taxa to tease out some variability in the data. A PERMANOVA (function 'adonis') was used to calculate significant differences across sites and beta dispersion (measure of variability) was also tested to give the PERMANOVA results more confidence. The results were either ordinated with a PCoA or NMDS using Bray-Curtis dissimilarity matrices. 


Adults Tricoptera show a potential effect wastewater on community composition (sig. different at Policeman flats and Graves Bridge but not at any other Bow River site). No sig beta dispersion.

Larval Hydropsychidae have different communities among all Bow River sites except Cushing Bridge - Sunalta so this could indicate that differences are due to something else at the site rather than wastewater exposure. At the ACWA streams the control stream was significantly different from both treatment streams indicating an effect of wastewater but the two treatment streams were not significantly different from one another. No sig. beta dispersion when ACWA and Bow River streams are separated.

Baetids show no difference in their composition between PCR1 and PCR3. No significant beta dispersion.

Larval Heptageniidae graph looks like Graves Bridge and Policeman Flats are the most distinct but the PERMANOVA syas all sites are different from eachother. This is likely due to significant beta dispersion. 

Adults Ephemeroptera PERMANOVA results indicate that all sites except Sunalta - Policeman Flats are sig different from one another but the graph only looks like Cushing Bridge is unique. This could also be attributed to the significant beta dispersion. 

Larval Perlidae had significant differences across all sites; Cochrane - Cushing Bridge, Cushing Bridge - Policeman Flats, Cochrane - Policeman Flats. Beta dispersion was not significant. 

Larval and Adults chironomidae were different among Cochrane - Cushing Bridge and Cochrane - Policeman Flats but Cushing Bridge - Policeman Flats was not significant. No sig. beta dispersion. 

Tetragnathidae have no significant differences across sites. No sig beta dispersion.

Araneidae are different at Policeman Flats compared to all other sites but no other differences. No sig beta dispersion.


#### All samples (larvae, Adults, and Spiders combined) 
```{r}

#Rarefying the data
ps_rare <- phyloseq::rarefy_even_depth(sample_ps, rngseed = 123, replace = FALSE)

#PCoA with Bray Curtis distance matrix
ps_rare_log  = transform_sample_counts(ps_rare, function(x) log(1 + x)) #log transforming the data 

dist_all = phyloseq::distance(ps_rare_log, method="bray") #Bray Curtis distance matrix
ordination_all = ordinate(ps_rare_log, method="PCoA", distance=dist_all) #Ordinate using PCoA

#Scree plot of PCoA
scree <- plot_scree(ordination_all)
scree + theme(axis.text.x=element_text(size=8, color="black"), axis.text.y=element_text(size=18, color="black"), 
        axis.text=element_text(size=18, color="black"), axis.title=element_text(size=18)) 
#axis 1, 2, and 3 account for most variation explained (mostly axis 1)

ps_rare_metadata <- data.frame(sample_data(ps_rare_log))

#All together
plot_ordination(ps_rare_log, ordination_all, color="Site", shape="Order") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [11.2%]")+
  ylab("PC2 [6.8%]")+
  facet_grid(~Family)

#Faceted by stage (larvae, adults, spiders)
plot_ordination(ps_rare_log, ordination_all, color="Site", shape="Order") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  xlab("PC1 [11.2%]")+
  ylab("PC2 [6.8%]")+
  facet_wrap(~Stage)
#Points are very clustered together and the axes don't explaina  ton of variation so I will try ordinating with an NMDS as well. 

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/All samples combined/PCoA by site and invertebrate family faceted by life stage.png", height = 7, width=8)


#PERMANOVA by Site
test.adonis <- adonis(dist_all ~ Site*Family, data = ps_rare_metadata)
test.adonis <- as.data.frame(test.adonis$aov.tab)
test.adonis #p<0.001. Significant dissimilarity between sites, families and significant interaction between the two.

#Pairwise comparisons
pairwise.adonis2(dist_all ~ Site+Family, data = ps_rare_metadata) #Gives the same result as above. 


#Testing variance (beta dispersion) differences between sites, sampling months, and sample families:
bd_all_site<-betadisper(dist_all, ps_rare_metadata$Site)
anova(bd_all_site) #Significant differences in beta dispersion among groups
permutest(bd_all_site) #Gives the same result but based on a permutation test. 
#Because the beta dispersion is significant, this gives us less confidence in the PERMANOVA results

#bd_all_fam<-betadisper(dist_all, ps_rare_metadata$Family) #This isn't working for some reason. Error message says" Error in `[<-`(`*tmp*`, i, , value = optim(apply(X, 2, median, na.rm = TRUE),  : subscript out of bounds"
#anova(bd_all_fam)
#permutest(bd_all_fam)

# NMDS 
set.seed(1)
ord_NMDS = ordinate(ps_rare_log, method="NMDS", distance=dist_all)

plot_ordination(ps_rare_log, ord_NMDS, color="Site", label="Sample_ID") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  theme(legend.key.size = unit(0.5, 'cm'), 
        legend.title = element_text(size=10), 
        legend.text = element_text(size=8), 
        legend.position = "bottom")
#Data is overlapping because of extreme outliers
#Remove these for now to see the graph better

#Removing outliers and re-running ordination
ps_nmds_subset_log = subset_samples(ps_rare_log, Sample_ID != "COCH_MB_Araneid_8" & Sample_ID != "CUSHBR_1_Ad_Caddis_MB_6" & Sample_ID !="CUSHBR_1_Ad_Caddis_MB_8" & Sample_ID !="ACWA_5%_MB_Ad_Mayfly_2" )

dist_all_subset = phyloseq::distance(ps_nmds_subset_log, method="bray") 

ord_NMDS_subset = ordinate(physeq=ps_nmds_subset_log, method="NMDS", distance=dist_all_subset)
ord_NMDS_subset 

plot_ordination(ps_nmds_subset_log, ord_NMDS_subset, color="Site", shape="Order", label="Sample_ID") + 
  theme_classic() +
  theme(strip.background = element_blank())


#Even after subsetting there are a few influential points that makes the ordination hard to visualize. A PCoA may be better suited for this data or trying to tease out some variability by separating out different taxa might help. 
```

#### Larvae, Adults, Spiders separated (3 panel plot) 
```{r}
#Larvae
larvae_log<-subset_samples(ps_rare_log, Stage == "Larvae")
dist_larvae_log = phyloseq::distance(larvae_log, method="bray")

#Bow and ACWA sites
#PCoA
pcoa_larvae_log= ordinate(larvae_log, method="PCoA", distance=dist_larvae_log) #Ordinate using PCoA
plot_ordination(larvae_log, pcoa_larvae_log, color="Site", shape="Family") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all larval invertebrates bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [17.0%]")+
  ylab("PC2 [10.0%]")

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/All Larvae/PCoA across sites (Bow and ACWA).png", height=6.5, width=7)

#NMDS
nmds_larvae_log= ordinate(larvae_log, method="NMDS", distance=dist_larvae_log) 
larvae_nmds_plot<-plot_ordination(larvae_log, nmds_larvae_log, color="Site", shape="Family") +
  theme_classic() +
  theme(strip.background = element_blank())
  
larvae_nmds_plot #stress is 0.17

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/All Larvae/NDMS across sites (Bow and ACWA).png", height=6.5, width=7)

#Bow River sites only

larvae_bow<-subset_samples(larvae_log, Stream_Type == "Bow River")
dist_larvae_bow<-phyloseq::distance(larvae_bow, method="bray")

#PCoA
pcoa_larvae_Bow = ordinate(larvae_bow, method="PCoA", distance=dist_larvae_bow)
pcoa_larvae_Bow_plot<-plot_ordination(larvae_bow, pcoa_larvae_Bow, color="Site", shape="Family") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all larval invertebrates bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [18.9 %]")+
  ylab("PC2 [8.9 %]")
  
pcoa_larvae_Bow_plot

#NMDS
nmds_larvae_Bow = ordinate(larvae_bow, method="NMDS", distance=dist_larvae_bow)
nmds_larvae_Bow #stress = 0.16
nmds_larvae_Bow_plot<-plot_ordination(larvae_bow, nmds_larvae_Bow, color="Site", shape="Family") + 
  scale_colour_discrete(guide = "none")+
  theme_classic() +
  theme(strip.background = element_blank(), legend.position = c(0.2, 0.8))
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))
nmds_larvae_Bow_plot

#ACWA sites only

larvae_ACWA<-subset_samples(larvae_log, Stream_Type == "ACWA Streams")
dist_larvae_ACWA<-phyloseq::distance(larvae_ACWA, method="bray")

#PCoA
pcoa_larvae_ACWA = ordinate(larvae_ACWA, method="PCoA", distance=dist_larvae_ACWA)
plot_ordination(larvae_ACWA, pcoa_larvae_ACWA, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all larval invertebrates bacteria based on log transformed Bray-Curtis dissimilarity")


#Adults
Adults_log<-subset_samples(ps_rare_log, Stage == "Adults")
dist_Adults = phyloseq::distance(Adults_log, method="bray")

#PCoA
pcoa_Adults= ordinate(Adults_log, method="PCoA", distance=dist_Adults) #Ordinate using PCoA
pcoa_Adults_plot<-plot_ordination(Adults_log, pcoa_Adults, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  xlab("PC1 [9.8%]")+
  ylab("PC2 [7.7%]")

pcoa_Adults_plot

#Bow River sites only
Adults_bow<-subset_samples(Adults_log, Stream_Type == "Bow River")
dist_Adults_bow = phyloseq::distance(Adults_bow, method="bray")

pcoa_Adults_bow= ordinate(Adults_bow, method="PCoA", distance=dist_Adults_bow) #Ordinate using PCoA
pcoa_Adults_bow_plot<-plot_ordination(Adults_bow, pcoa_Adults_bow, color="Site", shape="Order") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  xlab("PC1 [10.3%]")+
  ylab("PC2 [8.0%]")

pcoa_Adults_bow_plot
  
#NMDS
#All sites (Bow and ACWA)
nmds_Adults= ordinate(Adults_log, method="NMDS", distance=dist_Adults) #Ordinate using PCoA
plot_ordination(Adults_log, nmds_Adults, color="Site", label="Sample_ID") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all Adults invertebrates bacteria based on log transformed Bray-Curtis dissimilarity")

#ggsave(filename = "Beta Diversity Graphs/All Adultss/NMDS across Bow River sites outliers included.png", height=6.5, width=7)

#Removing outliers
Adults_subset<- subset_samples(Adults_log, Sample_ID !="POLFLT_1_Ad_Caddis_MB_4" & Sample_ID !="CUSHBR_1_Ad_Caddis_MB_1" & Sample_ID !="CUSHBR_1_Ad_Caddis_MB_2" & Sample_ID != "CUSHBR_1_Ad_Caddis_MB_8" & Sample_ID !="CUSHBR_1_Ad_Caddis_MB_6" & Sample_ID != "ACWA_5%_MB_Ad_Mayfly_2")

dist_Adults_subset = phyloseq::distance(Adults_subset, method="bray")

nmds_Adults_subset = ordinate(Adults_subset, method="NMDS", distance=dist_Adults_subset) 

nmds_Adults_subset_plot<- plot_ordination(Adults_subset, nmds_Adults_subset, color="Site", shape="Order")+
  theme_classic() +
  theme(strip.background = element_blank())

nmds_Adults_subset_plot


#PCoA with subset
pcoa_adults_subset = ordinate(Adults_subset, method="PCoA", distance = dist_Adults_subset)
plot_ordination(Adults_subset, pcoa_adults_subset, color="Site") + 
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all Adults invertebrates bacteria based on log transformed Bray-Curtis dissimilarity")

#Spiders
#Spiders_log<-subset_samples(ps_rare_log, Stage == "Spiders")
Spiders_log<-subset_samples(ps_rare_log, Order == "Araneae")
dist_Spiders = phyloseq::distance(Spiders_log, method="bray")

#PCoA
pcoa_Spiders = ordinate(Spiders_log, method="PCoA", distance=dist_Spiders) #Ordinate using PCoA
plot_ordination(Spiders_log, pcoa_Spiders, color="Site", shape="Family") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of all Spiders bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [24.3%]")+
  ylab("PC2 [10.2%]")

#ggsave(filename = "Beta Diversity Graphs/All Spiderss/PCoA across Bow River sites.png", height=6.5, width=7)

#NMDS
nmds_Spiders = ordinate(Spiders_log, method="NMDS", distance=dist_Spiders)
nmds_Spiders #stress = 0.17
nmds_Spiders_plot <- plot_ordination(Spiders_log, nmds_Spiders, color="Site", shape="Family") +
  theme_classic() +
  theme(strip.background = element_blank())+
  scale_colour_discrete(guide = "none")
  #stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))

nmds_Spiders_plot

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/All Spiderss/NMDS across Bow River sites.png", height=6.5, width=7)

#Removing one araenid outlier
Spiders_subset<- subset_samples(Spiders_log, Sample_ID != "COCH_MB_Araneid_8")
dist_Spiders_subset<-phyloseq::distance(Spiders_subset, method="bray")

nmds_Spiders_subset = ordinate(Spiders_subset, method="NMDS", distance=dist_Spiders_subset) 
nmds_Spiders_subset #stress = 0.17
nmds_Spiders_plot_subset <- plot_ordination(Spiders_subset, nmds_Spiders_subset, color="Site", shape="Family") + 
  theme_classic() +
  theme(strip.background = element_blank())
nmds_Spiders_plot_subset

#3 panel plot

nsamples(Spiders_subset) #68
nsamples(Adults_subset) #93
nsamples(larvae_log) #150

ggarrange(larvae_nmds_plot, nmds_Adults_subset_plot, nmds_Spiders_plot_subset, nrow=1, ncol=3, common.legend = F, labels=c("Larvae (n = 150)", "Adults (n = 93)", "Spiders (n = 68)"), hjust=-1, align = "v")

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/All samples combined/NMDS and PCoA faceted by sample type across Bow River sites.png", height=7, width=18, bg="white")
```

#### Larval Hydropsychidae
```{r}
#Bow River and ACWA streams combined
hydro_psrare<-subset_samples(ps_rare, Family == "Hydropsychidae")

#Log transform sample counts
hydro_ps_rare_log  = transform_sample_counts(hydro_psrare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_hydro_lar = phyloseq::distance(hydro_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_hydro_lar= ordinate(hydro_ps_rare_log, method="PCoA", distance=dist_hydro_lar) #Ordinate using PCoA
pcoa_hydro_lar

plot_ordination(hydro_ps_rare_log, pcoa_hydro_lar, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of larval Hydropsychidae bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [25.2%]")+
  ylab("PC2 [11.3%]")

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/Larval Hydropsychidae/PCoA across sites (Bow and ACWA).png", height=6.5, width=7)

#Adonis/PERMANOVA test

hydro_psrare_metadata <- data.frame(sample_data(hydro_ps_rare_log))
hydro_adonis <- adonis(dist_hydro_lar ~ Site, data = hydro_psrare_metadata) #This line is causing issues. Something to due with the adonis function. Adonis2 will run but it gives a dataframe with no data.
hydro_adonis_df <- as.data.frame(hydro_adonis$aov.tab)
hydro_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(as.dist(dist_hydro_lar) ~ Site, data = hydro_psrare_metadata)
#All are significantly different from one another except Cushing Bridge and Sunalta.

#Beta dispersion
hydro_lar_bd<-betadisper(dist_hydro_lar, hydro_psrare_metadata$Site)
anova(hydro_lar_bd) #p=0.0012. Significant beta dispersion.

#Separating by ACWA streams only
hydro_psrare_ACWA<-subset_samples(hydro_psrare, Stream_Type=="ACWA Streams")
hydro_ps_rare_log_ACWA  = transform_sample_counts(hydro_psrare_ACWA, function(x) log(1 + x))
dist_hydro_lar_ACWA = phyloseq::distance(hydro_ps_rare_log_ACWA, method="bray")

pcoa_hydro_lar_ACWA= ordinate(hydro_ps_rare_log_ACWA, method="PCoA", distance=dist_hydro_lar_ACWA)
plot_ordination(hydro_ps_rare_log_ACWA, pcoa_hydro_lar_ACWA, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of larval Hydropsychidae bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [16.5%]")+
  ylab("PC2 [10.4%]")

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/Larval Hydropsychidae/PCoA across ACWA streams.png", height=6.5, width=7)

hydro_psrare_metadata_ACWA <- data.frame(sample_data(hydro_ps_rare_log_ACWA))
hydro_adonis_ACWA <- adonis(dist_hydro_lar_ACWA ~ Site, data = hydro_psrare_metadata_ACWA)
hydro_adonis_df_ACWA <- as.data.frame(hydro_adonis_ACWA$aov.tab)
hydro_adonis_df_ACWA #p=0.002

pairwise.adonis2(as.dist(dist_hydro_lar_ACWA) ~ Site, data = hydro_psrare_metadata_ACWA)
#Sig. differences between all sites

hydro_lar_bd_ACWA<-betadisper(dist_hydro_lar_ACWA, hydro_psrare_metadata_ACWA$Site)
anova(hydro_lar_bd_ACWA) #p=0.07. Not significant. Means the PERMANOVA results can be used with more confidence

#Separating by Bow River sites only 
hydro_psrare_Bow<-subset_samples(hydro_psrare, Stream_Type!="ACWA Streams")

hydro_ps_rare_log_Bow  = transform_sample_counts(hydro_psrare_Bow, function(x) log(1 + x))
dist_hydro_lar_Bow = phyloseq::distance(hydro_ps_rare_log_Bow, method="bray")

pcoa_hydro_lar_Bow= ordinate(hydro_ps_rare_log_Bow, method="PCoA", distance=dist_hydro_lar_Bow)
plot_ordination(hydro_ps_rare_log_Bow, pcoa_hydro_lar_Bow, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of larval Hydropsychidae bacteria based on log transformed Bray-Curtis dissimilarity")+
  xlab("PC1 [28.3%]")+
  ylab("PC2 [6.9%]")

#(filename = "Microbiome_ED/Decontamianted analysis/Beta Diversity Graphs/Larval Hydropsychidae/PCoA across Bow River sites.png", height=6.5, width=7)

hydro_psrare_metadata_Bow <- data.frame(sample_data(hydro_ps_rare_log_Bow))
hydro_adonis_Bow <- adonis(dist_hydro_lar_Bow ~ Site, data = hydro_psrare_metadata_Bow)
hydro_adonis_df_Bow <- as.data.frame(hydro_adonis_Bow$aov.tab)
hydro_adonis_df_Bow #p=0.001

pairwise.adonis2(as.dist(dist_hydro_lar_Bow) ~ Site, data = hydro_psrare_metadata_Bow)
#The only non-significant comparison is between cushing bridge and Sunalta p=0.193. 

hydro_lar_bd_Bow<-betadisper(dist_hydro_lar_Bow, hydro_psrare_metadata_Bow$Site)
anova(hydro_lar_bd_Bow) #p=0.03. PERMANOVA results can be used with more confidence.
```

#### Adults Tricoptera 
```{r}
tricop_rare<-subset_samples(ps_rare, Order=="Trichoptera")
tricop_rare<-subset_samples(tricop_rare, Stage=="Adults")

#Log transform sample counts
tricop_ps_rare_log  = transform_sample_counts(tricop_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_tricop_ad = phyloseq::distance(tricop_ps_rare_log, method="bray") #Bray Curtis distance matrix

#NMDS
nmds_tricop_ad= ordinate(tricop_ps_rare_log, method="NMDS", distance=dist_tricop_ad) #Ordinate using PCoA
#stress=0.11
plot_ordination(tricop_ps_rare_log, nmds_tricop_ad, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Adults Tricoptera bacteria based on log transformed Bray-Curtis dissimilarity")

#ggsave(filename = "Beta Diversity Graphs/Adults Tricoptera/NMDS across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test

tricop_psrare_metadata <- data.frame(sample_data(tricop_ps_rare_log))
tricop_adonis <- adonis(dist_tricop_ad ~ Site, data = tricop_psrare_metadata)
tricop_adonis_df <- as.data.frame(tricop_adonis$aov.tab)
tricop_adonis_df #P<0.001

#Pairwise comparison
pairwise.adonis2(as.dist(dist_tricop_ad) ~ Site, data = tricop_psrare_metadata)
#Sunalta - Cushing Bridge, Coch - CushBr, and Coch-Sunalta are not significant, rest are.
#Might show diff composition with wastewater exposure 

#Beta dispersion
tricop_ad_bd<-betadisper(dist_tricop_ad, tricop_psrare_metadata$Site)
anova(tricop_ad_bd) #p=0.43. 
```

#### Baetidae (Larvae and Adults) 
```{r}
baetid_rare<-subset_samples(ps_rare, Family=="Baetidae")

#Log transform sample counts
baetid_ps_rare_log  = transform_sample_counts(baetid_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_baetid = phyloseq::distance(baetid_ps_rare_log, method="bray") #Bray Curtis distance matrix

#NMDS
nmds_baetid= ordinate(baetid_ps_rare_log, method="NMDS", distance=dist_baetid) #Ordinate using PCoA
nmds_baetid #stress = 0.07
plot_ordination(baetid_ps_rare_log, nmds_baetid, color="Site", shape="Stage") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Baetidae bacteria based on log transformed Bray-Curtis dissimilarity")

#ggsave(filename = "Beta Diversity Graphs/Baetidae/NMDS across ACWA sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test

baetid_psrare_metadata <- data.frame(sample_data(baetid_ps_rare_log))
baetid_adonis <- adonis(dist_baetid ~ Site, data = baetid_psrare_metadata)
baetid_adonis_df <- as.data.frame(baetid_adonis$aov.tab)
baetid_adonis_df #p=0.096. Not difference between PCR1 and PCR3 composition

#Beta dispersion
baetid_bd<-betadisper(dist_baetid, baetid_psrare_metadata$Site)
anova(baetid_bd) #p=0.21. 

pairwise.adonis2(as.dist(dist_baetid) ~ Site, data = baetid_psrare_metadata)


```

#### Larval Heptageniidae 
```{r}
hep_rare<-subset_samples(ps_rare, Family=="Heptageniidae")

#Log transform sample counts
hep_ps_rare_log  = transform_sample_counts(hep_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_hep = phyloseq::distance(hep_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_hep= ordinate(hep_ps_rare_log, method="PCoA", distance=dist_hep) #Ordinate using PCoA

plot_ordination(hep_ps_rare_log, pcoa_hep, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Heptageniidae bacteria based on log transformed \n Bray-Curtis dissimilarity")+
  xlab("PC1 [19.2%]")+
  ylab("PC2 [10.4%]")

#ggsave(filename = "Microbiome_ED/Decontaminated analysis/Beta Diversity Graphs/Heptageniidae/PCoA across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test
hep_psrare_metadata <- data.frame(sample_data(hep_ps_rare_log))
hep_adonis <- adonis(dist_hep ~ Site, data = hep_psrare_metadata)
hep_adonis_df <- as.data.frame(hep_adonis$aov.tab)
hep_adonis_df #p=0.001. 

pairwise.adonis2(as.dist(dist_hep) ~ Site, data = hep_psrare_metadata)
#Says all sites are different from one another but based on the graph it looks like Graves Bridge and Policeman Flats are the most distinct.

#Beta dispersion
hep_bd<-betadisper(dist_hep, hep_psrare_metadata$Site)
anova(hep_bd) #p=0.013. Significant beta dispersion. Might be the reason for why sunalta is still sig. different from cushing bridge and cochrane
```

#### Adults Ephemeroptera 
```{r}
ephem_ad_rare<-subset_samples(ps_rare, Order=="Ephemeroptera" & Stage=="Adults")

#Log transform sample counts
ephem_ad_ps_rare_log  = transform_sample_counts(ephem_ad_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_ephem_ad = phyloseq::distance(ephem_ad__ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_ephem_ad= ordinate(ephem_ad_ps_rare_log, method="PCoA", distance=dist_ephem_ad) #Ordinate using PCoA

plot_ordination(ephem_ad_ps_rare_log, pcoa_ephem_ad, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Ephemeroptera bacteria based on log transformed \n Bray-Curtis dissimilarity")

#ggsave(filename = "Beta Diversity Graphs/Heptageniidae/PCoA across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test
ephem_ad_psrare_metadata <- data.frame(sample_data(ephem_ad_ps_rare_log))
ephem_ad_psrare_metadata$Sample_ID #Filtering steps removed all ACWA 15% Adults mayfly and most of 5% stream
ephem_ad_adonis <- adonis(dist_ephem_ad ~ Site, data = ephem_ad_psrare_metadata)
ephem_ad_adonis_df <- as.data.frame(ephem_ad_adonis$aov.tab)
ephem_ad_adonis_df #p=0.001. 

pairwise.adonis2(as.dist(dist_ephem_ad) ~ Site, data = ephem_ad_psrare_metadata)
#Says all sites except Sunalta - Policeman Flats are different from one another but the graph looks like only Cushing Bridge should be unique to the other sites.

#Beta dispersion
ephem_ad_bd<-betadisper(dist_ephem_ad, ephem_ad_psrare_metadata$Site)
anova(ephem_ad_bd) #p=0.034. Significant beta dispersion.
```

#### Perlidae
```{r}
perl_rare<-subset_samples(ps_rare, Order=="Plecoptera")

#Log transform sample counts
perl_ps_rare_log  = transform_sample_counts(perl_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_perl = phyloseq::distance(perl_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_perl= ordinate(perl_ps_rare_log, method="PCoA", distance=dist_perl) #Ordinate using PCoA

plot_ordination(perl_ps_rare_log, pcoa_perl, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Perlidae bacteria based on log transformed \n Bray-Curtis dissimilarity")+
  xlab("PC1 [21.7%]")+
  ylab("PC2 [17.5%]")

#ggsave(filename = "Beta Diversity Graphs/Perlidae/PCoA across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test
perl_psrare_metadata <- data.frame(sample_data(perl_ps_rare_log))
perl_adonis <- adonis(dist_perl ~ Site, data = perl_psrare_metadata)
perl_adonis_df <- as.data.frame(perl_adonis$aov.tab)
perl_adonis_df #p=0.001. 

pairwise.adonis2(as.dist(dist_perl) ~ Site, data = perl_psrare_metadata)
#All sites are different from one another

#Beta dispersion
perl_bd<-betadisper(dist_perl, perl_psrare_metadata$Site)
anova(perl_bd) #p=0.077. Not significant.  
```

#### Chironomidae 
```{r}
chiro_rare<-subset_samples(ps_rare, Family=="Chironomidae")

#Log transform sample counts
chiro_ps_rare_log  = transform_sample_counts(chiro_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_chiro = phyloseq::distance(chiro_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_chiro= ordinate(chiro_ps_rare_log, method="PCoA", distance=dist_chiro) #Ordinate using PCoA

plot_ordination(chiro_ps_rare_log, pcoa_chiro, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Chironomidae bacteria based on log transformed \n Bray-Curtis dissimilarity")
  #Points look U-shaped so I will try NMDS instead

#NMDS
nmds_chiro= ordinate(chiro_ps_rare_log, method="NMDS", distance=dist_chiro) #Ordinate using PCoA

plot_ordination(chiro_ps_rare_log, nmds_chiro, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Chironomidae bacteria based on log transformed \n Bray-Curtis dissimilarity")
  

#ggsave(filename = "Beta Diversity Graphs/Chironomidae/NMDS across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test
chiro_psrare_metadata <- data.frame(sample_data(chiro_ps_rare_log))
chiro_adonis <- adonis(dist_chiro ~ Site, data = chiro_psrare_metadata)
chiro_adonis_df <- as.data.frame(chiro_adonis$aov.tab)
chiro_adonis_df #p=0.001. 

pairwise.adonis2(as.dist(dist_chiro) ~ Site, data = chiro_psrare_metadata)
#Cushing Bridge - Policeman Flats are not different from one another but the others are

#Beta dispersion
chiro_bd<-betadisper(dist_chiro, chiro_psrare_metadata$Site)
anova(chiro_bd) #p=0.53. Not significant.
```

#### Tetragnathidae
```{r}
tetra_rare<-subset_samples(ps_rare, Family=="Tetragnathidae")

#Log transform sample counts
tetra_ps_rare_log  = transform_sample_counts(tetra_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_tetra = phyloseq::distance(tetra_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_tetra= ordinate(tetra_ps_rare_log, method="PCoA", distance=dist_tetra) #Ordinate using PCoA

plot_ordination(tetra_ps_rare_log, pcoa_tetra, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Tetragnathidae bacteria based on log transformed \n Bray-Curtis dissimilarity")
  #Points look U-shaped so I will try NMDS instead

#NMDS
nmds_tetra= ordinate(tetra_ps_rare_log, method="NMDS", distance=dist_tetra) #Ordinate using PCoA

plot_ordination(tetra_ps_rare_log, nmds_tetra, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Tetragnathidae bacteria based on log transformed \n Bray-Curtis dissimilarity")
  

#ggsave(filename = "Beta Diversity Graphs/Tetragnathidae/NMDS across Bow River sites.png", height=6.5, width=7)

#Adonis/PERMANOVA test
tetra_psrare_metadata <- data.frame(sample_data(tetra_ps_rare_log))
tetra_adonis <- adonis(dist_tetra ~ Site, data = tetra_psrare_metadata)
tetra_adonis_df <- as.data.frame(tetra_adonis$aov.tab)
tetra_adonis_df #p=0.17. 

pairwise.adonis2(as.dist(dist_tetra) ~ Site, data = tetra_psrare_metadata)
#No significant differences.

#Beta dispersion
tetra_bd<-betadisper(dist_tetra, tetra_psrare_metadata$Site)
anova(tetra_bd) #p=0.24.
```

#### Araneidae 
```{r}
aran_rare<-subset_samples(ps_rare, Family=="Araneidae")

#Log transform sample counts
aran_ps_rare_log  = transform_sample_counts(aran_rare, function(x) log(1 + x)) #log transforming the data 

#Make distance matrix
dist_aran = phyloseq::distance(aran_ps_rare_log, method="bray") #Bray Curtis distance matrix

#PCoA
pcoa_aran= ordinate(aran_ps_rare_log, method="PCoA", distance=dist_aran) #Ordinate using PCoA

plot_ordination(aran_ps_rare_log, pcoa_aran, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))+
  ggtitle("Beta diversity of Araneidae bacteria based on log transformed \n Bray-Curtis dissimilarity")+
  xlab("PC1 [14.8%]")+
  ylab("PC2 [11.6%]")


#ggsave(filename = "Beta Diversity Graphs/Araneidae/PCoA across Bow River sites.png", height=6.5, width=7)

nmds_aran= ordinate(aran_ps_rare_log, method="NMDS", distance=dist_aran) #Ordinate using PCoA

plot_ordination(aran_ps_rare_log, nmds_aran, color="Site") + #Plots the ordination
  theme_classic() +
  theme(strip.background = element_blank())+
  stat_ellipse(geom="polygon", alpha=0.1, aes(fill=Site))

#Adonis/PERMANOVA test
aran_psrare_metadata <- data.frame(sample_data(aran_ps_rare_log))
aran_adonis <- adonis(dist_aran ~ Site, data = aran_psrare_metadata)
aran_adonis_df <- as.data.frame(aran_adonis$aov.tab)
aran_adonis_df #p=0.019. 

pairwise.adonis2(as.dist(dist_aran) ~ Site, data = aran_psrare_metadata)
#Difference between Cochrane and Policeman Flats p=0.002
#Graves Bridge - Policeman Flats p=0.003
#Cushing Bridge - Policeman Flats p=0.027
#Sunalta - Policeman Flats p=0.007

#Beta dispersion
aran_bd<-betadisper(dist_aran, aran_psrare_metadata$Site)
anova(aran_bd) #p=0.81.

```

## Env fit
```{r}
ord <- metaMDS(varespec)
(fit <- envfit(ord, varechem, perm = 999))
scores(fit, "vectors")
plot(ord)
plot(fit)
plot(fit, p.max = 0.05, col = "red")

```

## Clustering
```{r Clustering}
#### Clustering methods ####

#Hierarchical clustering using Factoextra

#First we need to standardize the data and make a similarity matrix

ps_cluster_subset = subset_samples(ps_rare2, Sample_ID != "POLFLT_1_Ad_Caddis_MB_1" & Sample_ID != "CUSHBR_1_Ad_Caddis_MB_2")
ps_rare_scaled<-tax_scale(ps_cluster_subset) #Scale/standardize the data
dist_bray = phyloseq::distance(ps_rare_scaled, method="bray") #Use Bray Curtis similarity matrix
dist_bray_mat<-as.matrix(dist_bray)


#Algorithm
hclust.out<-hclust(dist_bray, method="ward.D2")

#Dendrogram
dendro <- as.dendrogram(hclust(dist_bray, method = "ward.D2"))
#Provide color codes
fviz_dend(hclust, rect = TRUE, cex = 0.5, show_labels = TRUE , main="Ward Bray-Curtis")
meta <- data.frame(phyloseq::sample_data(ps_rare_scaled))
colorCode <- c("Cochrane" = "red", "Sunalta" = "orange", "Cushing Bridge" = "yellow", "Graves Bridge" = "green", "Policeman Flats" = "blue", "BRR2"="purple", "PCR1"="pink", "PCR3"="brown")

labels_colors(dendro) <- colorCode[meta$Site][order.dendrogram(dendro)]
#Plot
plot(dendro)
#Can't really read the sample names but the colours represent which site they came from. Seems like the colours are all mixing together for the most part, indicating that the sites don't cluster together in terms of their microbiome

colorCode1<-c("Hydropsychidae" = "red", "Heptageniidae"="orange", "Perlidae"="yellow", "Araneidae"="green", "Tetragnathidae"="blue", "Chironomidae"="purple", "Baetidae"="pink")
labels_colors(dendro) <- colorCode1[meta$Family][order.dendrogram(dendro)]
plot(dendro)
#When colours by the family of insect sample it is more obvious that there are clusters. This makes sense because taxa that are more similar will have a more similar microbiome. Variation in clusters from the same family might be because of different genera within the family or individual variation.


#Find the optimal clusters
clusters<-cutree(hclust, k=7)
print_clusters(clusters, 7)

#Visualize the data
fviz_cluster(list(data=dist_bray, cluster=clusters))

#Bootstrapping
kbest.8<-8
cboot.hclust <- clusterboot(dist_bray, clustermethod=hclustCBI,
                           method="ward.D2", k=kbest.8)
table(cboot.hclust$result$partition)
groups <- cboot.hclust$result$partition
table(groups, meta$Site) 
table(groups, meta$Family)


kbest.7<-7
cboot.hclust.7 <- clusterboot(dist_bray, clustermethod=hclustCBI,
                           method="ward.D2", k=kbest.7)
table(cboot.hclust.7$result$partition)
groups.7 <- cboot.hclust.7$result$partition
table(groups.7, meta$Family)

#using pvclust
asvdf<-as.matrix(otu_table(ps_rare_scaled))
asvdf<-as.data.frame(otu_table(asvdf))

pv.out<-parPvclust(cl=NULL, asvdf, method.hclust = "ward.D2",
           method.dist = "correlation", nboot = 10,
           iseed = NULL)
plot(pv.out, hang = -1, cex = 0.5)


#K-means clustering
#K-means clustering requires a pre-defined number of clusters. In the case of my data, I want to know whether samples are clustering based on the collection site or the type of insect collected. In both cases, the number of clusters would be 8. 
meta_df<-as.data.frame(meta)

k <- kmeans(dist_bray, centers = 7, nstart = 100, iter.max=100)
k


dist_bray_mat<-as.matrix(dist_bray)
dist_bray_df<-as.data.frame(dist_bray_mat)

#Elbow method
fviz_nbclust(dist_bray_mat, kmeans, method="wss")+
    geom_vline(xintercept=6,linetype=2)+
  geom_vline(xintercept=4,linetype=2)
#Looks like there is an elbow at 4 or 6 clusters but difficult to tell 

# Perform silhouette analysis and plot the result
fviz_nbclust(dist_bray_mat, kmeans, method = "silhouette")+
    geom_vline(xintercept=7,linetype=2)
#Highest value is at 7 clusters. This is similar to the elbow method. 
#These clusters could be appearing due to site differences (there are 8 sites total) or sample Taxa differences (7 total families of insect samples)

#Gap statistic
fviz_nbclust(dist_bray_mat, kmeans, method="gap_stat")
#Optimal number of clusters is 8. This could either correspond to collection site or taxa. 

#Visulaizing the data (biplot)
fviz_cluster(kmeans(dist_bray_mat, centers = 8, nstart = 100, iter.max=100), data=dist_bray_mat)

wineBest <- FitKMeans(dist_bray_mat, max.clusters=20, nstart=25,  seed=278613)
wineBest

PlotHartigan(wineBest)

table(dist_bray_mat$Site, wineK3N25$cluster)

#Visualizing data with original variables
clusters<-kmeans(dist_bray_mat, centers = 7, nstart = 100, iter.max=100)
clusters<-as.matrix(clusters)
clusters<-as.data.frame(clusters)

diss_cluster<-cbind(clusters, meta_df)
diss_cluster<- dist_bray_mat %>% mutate(cluster=clusters$cluster)

plot.kmeans(k, data=dist_bray_mat)

ASVseqmat<-as.matrix(ASVseq)
rel_abund_mat <- transformCounts(ps, method = "relabundance")

#Cluster dendrogram
d <- phyloseq::distance(ps_rare_log, method="bray", type="samples")
clust <- hclust(d, method="average")
plot(clust)#Hard to read label names

```

# Differential Abundance

Done on non-rarefied data to compare the mean absolute abundance between two ecosystems/samples/sites. The DESeq2 model internally corrects for library size, so transformed or normalized values such as counts scaled by library size should not be used as input. It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order.

Ignoring excess zeros in zero inflated data can lead to overestimation of dispersion and subsequent reduction in statistical power to detect differentially abundant features. To correct for this I used 'poscounts' which calculates a modified geometric mean by taking the nth root of the product of the non-zero counts. The prevalence threshold was increased to 3% to reduce the chance of false positives. Adjusted p-values using Benjamini-Hochberg correction were used to determine significance with an alpha of 0.001.


```{r Differential Abundance with DESeq2}

#Increase prevalence threshold to 3% to reduce the chance of false positives.
sample_ps2<-phyloseq_filter_prevalence(sample_ps, prev.trh = 0.03, abund.trh = 6, threshold_condition = "AND")
sample_ps2<-prune_taxa(taxa_sums(sample_ps2) > 0, sample_ps2)
sample_ps2 #1911 taxa, 318 samples. 

#### Single site comparisons - Bow River ####
#Sunalta - Cochrane
diff_abun_Coch_Sunal = subset_samples(sample_ps2, Site %in% c("Cochrane", "Sunalta"))

deseq_coch_sun <- phyloseq_to_deseq2(diff_abun_Coch_Sunal, ~ Site)
deseq_coch_sun <- DESeq(deseq_coch_sun, test = "Wald", fitType = "parametric", sfType = "poscounts")  #The "poscounts" estimator deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. 'Parametric' is used by default since it works with most data.
plotMA(deseq_coch_sun)
plotDispEsts(deseq_coch_sun)

#Getting results of diff abund. estimates
res_deseq <- results(deseq_coch_sun, cooksCutoff = FALSE)
res_deseq
alpha = 0.001
sigtab = res_deseq[which(res_deseq$padj < alpha), ]
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab), ], "matrix"))

write.csv(as.data.frame(sigtab), file="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_Coch_Sunalta.csv")


#Shrinking effect size
#res_LFC <- lfcShrink(deseq_coch_sun, coef=2, type="apeglm") 
#df_res_lfc <- cbind(as(res_LFC, "data.frame"), as(tax_table(sample_ps)[rownames(res_LFC), ], "matrix"))

#Plotting the log2fold change in bacteria: Sunalta vs Cochrane
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}
x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))
# Genus order
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))
ggplot(sigtab, aes(x=Genus, y=log2FoldChange, color=Family)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_Sunalta_Coch.png", height=6.5, width=7.5)

#Mucinivorans is an anaerobic, mucin-degrading bacterium isolated from the digestive tract of the medicinal leech Hirudo verbana

#Candidatus bacilloplasma is associated with the hindgut wall of the terrestrial isopod Porcellio scaber


#Cushing Bridge - Cochrane
diff_abun_Coch_cushbr <- subset_samples(sample_ps2, Site %in% c("Cochrane", "Cushing Bridge"))

deseq_coch_cushbr <- phyloseq_to_deseq2(diff_abun_Coch_cushbr, ~Site)
estimateSizeFactors(deseq_coch_cushbr, type="poscount")
traceback()
table(rowSums(counts(deseq_coch_cushbr) == 0 ))
table(colSums(counts(deseq_coch_cushbr) == 0))
deseq_coch_cushbr <- DESeq(deseq_coch_cushbr, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_coch_cushbr <- results(deseq_coch_cushbr, cooksCutoff = FALSE)
res_deseq_coch_cushbr
alpha = 0.001
sigtab_coch_cushbr = res_deseq_coch_cushbr[which(res_deseq_coch_cushbr$padj < alpha), ]
sigtab_coch_cushbr = cbind(as(sigtab_coch_cushbr, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_coch_cushbr), ], "matrix"))

write.csv(as.data.frame(sigtab_coch_cushbr), file="deseq_Coch_CushBr.csv")

#Plotting the log2fold change in bacteria: Cushing Bridge vs Cochrane

x_coch_cushbr = tapply(sigtab_coch_cushbr$log2FoldChange, sigtab_coch_cushbr$Phylum, function(x) max(x))
x_coch_cushbr = sort(x_coch_cushbr, TRUE)
sigtab_coch_cushbr$Phylum = factor(as.character(sigtab_coch_cushbr$Phylum), levels=names(x_coch_cushbr))
# Genus order
x_coch_cushbr = tapply(sigtab_coch_cushbr$log2FoldChange, sigtab_coch_cushbr$Genus, function(x) max(x))
x_coch_cushbr = sort(x_coch_cushbr, TRUE)
sigtab_coch_cushbr$Genus = factor(as.character(sigtab_coch_cushbr$Genus), levels=names(x_coch_cushbr))
ggplot(sigtab_coch_cushbr, aes(x=Genus, y=log2FoldChange, color=Class)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))

#ggsave(filename="deseq_Cushbr_Coch.png", height=6.5, width=7.5)


#Graves Bridge - Cochrane
diff_abun_Coch_gravesbr = subset_samples(sample_ps2, Site %in% c("Cochrane", "Graves Bridge"))

deseq_coch_gravesbr <- phyloseq_to_deseq2(diff_abun_Coch_gravesbr, ~ Site)
deseq_coch_gravesbr <- DESeq(deseq_coch_gravesbr, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_coch_gravesbr <- results(deseq_coch_gravesbr, cooksCutoff = FALSE)
alpha = 0.001
sigtab_coch_gravesbr = res_deseq_coch_gravesbr[which(res_deseq_coch_gravesbr$padj < alpha), ]
sigtab_coch_gravesbr = cbind(as(sigtab_coch_gravesbr, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_coch_gravesbr), ], "matrix"))

write.csv(as.data.frame(sigtab_coch_gravesbr), file="Differential Abundance Results/deseq_Coch_GrvBr.csv")

#Plotting the log2fold change in bacteria: Graves Bridge vs Cochrane

x_coch_gravesbr = tapply(sigtab_coch_gravesbr$log2FoldChange, sigtab_coch_gravesbr$Phylum, function(x) max(x))
x_coch_gravesbr = sort(x_coch_gravesbr, TRUE)
sigtab_coch_gravesbr$Phylum = factor(as.character(sigtab_coch_gravesbr$Phylum), levels=names(x_coch_gravesbr))
# Genus order
x_coch_gravesbr = tapply(sigtab_coch_gravesbr$log2FoldChange, sigtab_coch_gravesbr$Genus, function(x) max(x))
x_coch_gravesbr = sort(x_coch_gravesbr, TRUE)
sigtab_coch_gravesbr$Genus = factor(as.character(sigtab_coch_gravesbr$Genus), levels=names(x_coch_gravesbr))
ggplot(sigtab_coch_gravesbr, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))

#ggsave(filename="Differential Abundance Results/deseq_GrvBr_Coch.png", height=6.5, width=7.5)

#Policeman Flats - Cochrane
diff_abun_Coch_pmf = subset_samples(sample_ps2, Site %in% c("Cochrane", "Policeman Flats"))

deseq_coch_pmf <- phyloseq_to_deseq2(diff_abun_Coch_pmf ~ Site)
deseq_coch_pmf <- DESeq(deseq_coch_pmf, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_coch_pmf <- results(deseq_coch_pmf, cooksCutoff = FALSE)
alpha = 0.001
sigtab_coch_pmf = res_deseq_coch_pmf[which(res_deseq_coch_pmf$padj < alpha), ]
sigtab_coch_pmf = cbind(as(sigtab_coch_pmf, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_coch_pmf), ], "matrix"))

write.csv(as.data.frame(sigtab_coch_pmf), file="Differential Abundance Results/deseq_Coch_PMF.csv")

#Plotting the log2fold change in bacteria: Policeman Flats vs Cochrane

x_coch_pmf = tapply(sigtab_coch_pmf$log2FoldChange, sigtab_coch_pmf$Phylum, function(x) max(x))
x_coch_pmf = sort(x_coch_pmf, TRUE)
sigtab_coch_pmf$Phylum = factor(as.character(sigtab_coch_pmf$Phylum), levels=names(x_coch_pmf))
# Genus order
x_coch_pmf = tapply(sigtab_coch_pmf$log2FoldChange, sigtab_coch_pmf$Genus, function(x) max(x))
x_coch_pmf = sort(x_coch_pmf, TRUE)
sigtab_coch_pmf$Genus = factor(as.character(sigtab_coch_pmf$Genus), levels=names(x_coch_pmf))
ggplot(sigtab_coch_pmf, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))

#ggsave(filename="deseq_PMF_Coch.png", height=6.5, width=7.5)


#### Comparing un-exposed sites to exposed sites ####
#Wastewater exposed sites include Policeman Flats and Graves Bridge whereas unexposed sites are Cochrane, Sunalta, and Cushing Bridge.

deseq_exposed <- phyloseq_to_deseq2(sample_ps2 ~ Exposure)
deseq_exposed <- DESeq(deseq_exposed, test = "Wald", fitType = "parametric", sfType = "poscounts")

res_deseq_exposed <- results(deseq_exposed, cooksCutoff = FALSE)
alpha = 0.01
sigtab_exposed = res_deseq_exposed[which(res_deseq_exposed$padj < alpha), ]
sigtab_exposed = cbind(as(sigtab_exposed, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_exposed), ], "matrix"))

write.csv(as.data.frame(sigtab_exposed), file="deseq_Wastewater_Exposed_vs_Unexposed_Sites.csv")


x_exposed = tapply(sigtab_exposed$log2FoldChange, sigtab_exposed$Phylum, function(x) max(x))
x_exposed = sort(x_exposed, TRUE)
sigtab_exposed$Phylum = factor(as.character(sigtab_exposed$Phylum), levels=names(x_exposed))
# Family order
x_exposed = tapply(sigtab_exposed$log2FoldChange, sigtab_exposed$Family, function(x) max(x))
x_exposed = sort(x_exposed, TRUE)
sigtab_exposed$Family = factor(as.character(sigtab_exposed$Family), levels=names(x_exposed))
ggplot(sigtab_exposed, aes(x=Family, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5, size=8)) + theme(axis.text=element_text(size=12), axis.title=element_text(size=12)) +
          theme(legend.text=element_text(size=10, color="black")) + 
          theme(legend.title = element_text(size=12, color="black"))

#### Differential abundance in ACWA streams ####

#BRR2 vs PCR1 
diff_abun_BRR2_PCR1 = subset_samples(sample_ps2, Stream_Type=="ACWA Streams")
diff_abun_BRR2_PCR1 = subset_samples(diff_abun_BRR2_PCR1, Site != "PCR3")

deseq_BRR2_PCR1 <- phyloseq_to_deseq2(diff_abun_BRR2_PCR1, ~ Site)
deseq_BRR2_PCR1 <- DESeq(deseq_BRR2_PCR1, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_BRR2_PCR1 <- results(deseq_BRR2_PCR1, cooksCutoff = FALSE)
alpha = 0.001
sigtab_BRR2_PCR1 = res_deseq_BRR2_PCR1[which(res_deseq_BRR2_PCR1$padj < alpha), ]
sigtab_BRR2_PCR1 = cbind(as(sigtab_BRR2_PCR1, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_BRR2_PCR1), ], "matrix"))

write.csv(as.data.frame(sigtab_BRR2_PCR1), file="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_BRR2_PCR1.csv")

#Plotting the log2fold change in bacteria: PCR1 vs BRR2

x_BRR2_PCR1 = tapply(sigtab_BRR2_PCR1$log2FoldChange, sigtab_BRR2_PCR1$Phylum, function(x) max(x))
x_BRR2_PCR1 = sort(x_BRR2_PCR1, TRUE)
sigtab_BRR2_PCR1$Phylum = factor(as.character(sigtab_BRR2_PCR1$Phylum), levels=names(x_BRR2_PCR1))
# Genus order
x_BRR2_PCR1 = tapply(sigtab_BRR2_PCR1$log2FoldChange, sigtab_BRR2_PCR1$Genus, function(x) max(x))
x_BRR2_PCR1 = sort(x_BRR2_PCR1, TRUE)
sigtab_BRR2_PCR1$Genus = factor(as.character(sigtab_BRR2_PCR1$Genus), levels=names(x_BRR2_PCR1))
ggplot(sigtab_BRR2_PCR1, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))+
  ggtitle("Differential Abundance of PCR1/BRR2")

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_PCR1_BRR2.png", height=6.5, width=7.5)

#verifying that these aren't false positives
ACWA<-subset_samples(sample_ps2, Stream_Type=="ACWA Streams")

#Flavobacterium
ACWA_Flav<-subset_taxa(ACWA, Genus=="Flavobacterium")

#Need to merge the DESeq2 results with the taxonomy table of the sample_ps2 ACWA phyloseq object so that I can plot the taxa to see if they are a real result and not a false positive. Use left_join/merge_taxa function?
merge_taxa(taxa_table(ACWA_Flav))
left_join(sigtab_BRR2_PCR1, )


ACWA_Flav_df<-psmelt(ACWA_Flav)
ACWA_Flav_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Flav_df, FUN=mean)

ggplot(ACWA_Flav_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at PCR1 then PCR3. Mean total abundance/sample is not very high though. 

#Checking for Falvobacterium in Bow River samples
Bow_Flav<-subset_taxa(Bow, Genus=="Flavobacterium")
Bow_Flav<-prune_taxa(taxa_sums(Bow_Flav) > 0, Bow_Flav)
Bow_Flav_df<-psmelt(Bow_Flav)

Bow_Flav_agg<-aggregate(Abundance~Genus+Site, data=Bow_Flav_df, FUN=mean)

ggplot(Bow_Flav_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Sunalta but the mean total abundance is super low in Bow River sites. 

#Flavobacterium can sometimes be found in wastewater 
#Flavobacterium have a prominent role in freshwater bacterioplankton communities due to their high rates of substrate uptake and growth, growth on algal-derived substrates and high mortality rates from bacterivory.


#Limnohabitans
ACWA_Lim<-subset_taxa(ACWA, Genus=="Limnohabitans")
ACWA_Lim_df<-psmelt(ACWA_Lim)
ACWA_Lim_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Lim_df, FUN=mean)

ggplot(ACWA_Lim_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at PCR1 then PCR3. Low in BRR2. Overall mean total abundance is low for all 3 streams. 

#Checking for Limnohabitans in Bow River samples
Bow_Lim<-subset_taxa(Bow, Genus=="Limnohabitans")
Bow_Lim<-prune_taxa(taxa_sums(Bow_Lim) > 0, Bow_Lim)
Bow_Lim_df<-psmelt(Bow_Lim)

Bow_Lim_agg<-aggregate(Abundance~Genus+Site, data=Bow_Lim_df, FUN=mean)

ggplot(Bow_Lim_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge then Policeman Flats. 


#Ideonella
#Not sure of the species but probably is Ideonella dechloratans because it is perchlorate reducing and perchlorate is found in wastewater since it's commonly used to treat effluent. 
#Could also be Ideonella sakaiensis which able to break down plastic. Potentially has more plastic in the wastewater streams? 
#Or Ideonella azotifigens? It is nitrogen fixing

#PCR1 and PCR3 had a higher abundance of this bacteria compared to the control streams.
ACWA_Ideon<-subset_taxa(ACWA, Genus=="Ideonella")
table(tax_table(ACWA_Ideon)[,7])
ACWA_Ideon_df<-psmelt(ACWA_Ideon)
ACWA_Ideon_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Ideon_df, FUN=mean)
ggplot(ACWA_Ideon_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")+
  ylab("Mean Total Abundance")
  #Much higher in PCR1 and PCR3 streams

#Checking for Ideonella in Bow River samples
Bow<-subset_samples(sample_ps2, Stream_Type=="Bow River")
Bow_Ideon<-subset_taxa(Bow, Genus=="Ideonella")
Bow_Ideon<-prune_taxa(taxa_sums(Bow_Ideon) > 0, Bow_Ideon)
Bow_Ideon_df<-psmelt(Bow_Ideon)

Bow_Ideon_agg<-aggregate(Abundance~Genus+Site, data=Bow_Ideon_df, FUN=mean)

ggplot(Bow_Ideon_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Cushing Bridge had the highest mean total abundance of Ideonella. Lowest at Cochrane and Sunalta.

#Rhodoferax
#Oxidizes acetate with with the reduction of Fe(3)
ACWA_Rhodo<-subset_taxa(ACWA, Genus=="Rhodoferax")
ACWA_Rhodo_df<-psmelt(ACWA_Rhodo)
ACWA_Rhodo_agg<-aggregate(Abundance~Genus+Site, data=ACWA_Rhodo_df, FUN=mean)
ggplot(ACWA_Rhodo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest mean total abundance at PCR3. Could potentially indicate an anoxic environment.

#Checking for Rhodoferax in Bow River samples
Bow_Rhodo<-subset_taxa(Bow, Genus=="Rhodoferax")
Bow_Rhodo<-prune_taxa(taxa_sums(Bow_Rhodo) > 0, Bow_Rhodo)
Bow_Rhodo_df<-psmelt(Bow_Rhodo)

Bow_Rhodo_agg<-aggregate(Abundance~Genus+Site, data=Bow_Rhodo_df, FUN=mean)

ggplot(Bow_Rhodo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge. Similar across all other sites. 


#Pajaroellobacter (abortibovis?) only one known species
#Is the agent for Epizootic bovine abortion (EBA), an arthropod-borne bacterial disease that causes significant economic loss for cattle producers in the western United States

ACWA_paja<-subset_taxa(ACWA, Genus=="Pajaroellobacter")
ACWA_paja_df<-psmelt(ACWA_paja)
ACWA_paja_agg<-aggregate(Abundance~Genus+Site, data=ACWA_paja_df, FUN=mean)
ggplot(ACWA_paja_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Mean total abundance was highest at PCR1 but the same between PCR3 and BRR2

#Checking for Paja in Bow River samples
Bow_paja<-subset_taxa(Bow, Genus=="Pajaroellobacter")
Bow_paja<-prune_taxa(taxa_sums(Bow_paja) > 0, Bow_paja)
Bow_paja_df<-psmelt(Bow_paja)

Bow_paja_agg<-aggregate(Abundance~Genus+Site, data=Bow_paja_df, FUN=mean)

ggplot(Bow_paja_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Mean abundance is much higher at Policeman Flats compared to other sites. Cochrane is virtually 0. 


#Novosphingobium
ACWA_novo<-subset_taxa(ACWA, Genus=="Novosphingobium")
ACWA_novo_df<-psmelt(ACWA_novo)
ACWA_novo_agg<-aggregate(Abundance~Genus+Site, data=ACWA_novo_df, FUN=mean)
ggplot(ACWA_novo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at BRR2... but the differential abundance test said the opposite.

#Checking for Novosphingobium in Bow River samples
Bow_novo<-subset_taxa(Bow, Genus=="Novosphingobium")
Bow_novo<-prune_taxa(taxa_sums(Bow_novo) > 0, Bow_novo)
Bow_novo_df<-psmelt(Bow_novo)

Bow_novo_agg<-aggregate(Abundance~Genus+Site, data=Bow_novo_df, FUN=mean)

ggplot(Bow_novo_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Policeman Flats


#Alistipes
ACWA_Ali<-subset_taxa(ACWA, Genus=="Alistipes")
ACWA_Ali_df<-psmelt(ACWA_Ali)

ACWA_Ali_agg<-aggregate(Abundance ~ Genus + Site, data = ACWA_Ali_df, FUN = mean)

ggplot(ACWA_Ali_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Lowest at PCR1. Same at BRR2 and PCR3. 

#Checking for Alistipes in Bow River samples
Bow_ali<-subset_taxa(Bow, Genus=="Alistipes")
Bow_ali<-prune_taxa(taxa_sums(Bow_ali) > 0, Bow_ali)
Bow_ali_df<-psmelt(Bow_ali)

Bow_ali_agg<-aggregate(Abundance~Genus+Site, data=Bow_ali_df, FUN=mean)

ggplot(Bow_ali_agg, aes(x=Site, y=Abundance))+
  geom_col()+
  rremove("grid")
#Highest at Graves Bridge and second highest at Policeman Flats. 

#Alistipes is known to be pathogenic in colorectal cancer and is associated with mentalsigns of depression (Parker et al., 2020)


#BRR2 vs PCR3 
diff_abun_BRR2_PCR3 = subset_samples(sample_ps2, Stream_Type=="ACWA Streams")
diff_abun_BRR2_PCR3 = subset_samples(diff_abun_BRR2_PCR3, Site != "PCR1")

deseq_BRR2_PCR3 <- phyloseq_to_deseq2(diff_abun_BRR2_PCR3, ~ Site)
deseq_BRR2_PCR3 <- DESeq(deseq_BRR2_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_BRR2_PCR3 <- results(deseq_BRR2_PCR3, cooksCutoff = FALSE)
alpha = 0.001
sigtab_BRR2_PCR3 = res_deseq_BRR2_PCR3[which(res_deseq_BRR2_PCR3$padj < alpha), ]
sigtab_BRR2_PCR3 = cbind(as(sigtab_BRR2_PCR3, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_BRR2_PCR3), ], "matrix"))

write.csv(as.data.frame(sigtab_BRR2_PCR3), file="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_BRR2_PCR3.csv")

#Plotting the log2fold change in bacteria: PCR3 vs BRR2

x_BRR2_PCR3 = tapply(sigtab_BRR2_PCR3$log2FoldChange, sigtab_BRR2_PCR3$Phylum, function(x) max(x))
x_BRR2_PCR3 = sort(x_BRR2_PCR3, TRUE)
sigtab_BRR2_PCR3$Phylum = factor(as.character(sigtab_BRR2_PCR3$Phylum), levels=names(x_BRR2_PCR3))
# Genus order
x_BRR2_PCR3 = tapply(sigtab_BRR2_PCR3$log2FoldChange, sigtab_BRR2_PCR3$Genus, function(x) max(x))
x_BRR2_PCR3 = sort(x_BRR2_PCR3, TRUE)
sigtab_BRR2_PCR3$Genus = factor(as.character(sigtab_BRR2_PCR3$Genus), levels=names(x_BRR2_PCR3))
ggplot(sigtab_BRR2_PCR3, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))+
  ggtitle("Differential Abundance of PCR3/BRR2")

#ggsave(filename="Microbiome_ED/Decontaminated analysis/deseq_PCR3_BRR2.png", height=6.5, width=7.5)


#Clostridium sensu stricto 1


#Gemmobacter


#Rhodobacter




#PCR1 vs PCR3
diff_abun_PCR1_PCR3 = subset_samples(sample_ps2, Stream_Type=="ACWA Streams")
diff_abun_PCR1_PCR3 = subset_samples(diff_abun_PCR1_PCR3, Site != "BRR2")

deseq_PCR1_PCR3 <- phyloseq_to_deseq2(diff_abun_PCR1_PCR3, ~ Site)
deseq_PCR1_PCR3 <- DESeq(deseq_PCR1_PCR3, test = "Wald", fitType = "parametric", sfType = "poscounts")  

#Getting results of diff abund. estimates
res_deseq_PCR1_PCR3 <- results(deseq_PCR1_PCR3, cooksCutoff = FALSE)
res_deseq_PCR1_PCR3 #PCR3/PCR1
alpha = 0.001
sigtab_PCR1_PCR3 = res_deseq_PCR1_PCR3[which(res_deseq_PCR1_PCR3$padj < alpha), ]
sigtab_PCR1_PCR3 = cbind(as(sigtab_PCR1_PCR3, "data.frame"), as(tax_table(sample_ps)[rownames(sigtab_PCR1_PCR3), ], "matrix"))

write.csv(as.data.frame(sigtab_PCR1_PCR3), file="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_PCR1_PCR3.csv")

#Plotting the log2fold change in bacteria: PCR3 vs PCR1

x_PCR1_PCR3 = tapply(sigtab_PCR1_PCR3$log2FoldChange, sigtab_PCR1_PCR3$Phylum, function(x) max(x))
x_PCR1_PCR3 = sort(x_PCR1_PCR3, TRUE)
sigtab_PCR1_PCR3$Phylum = factor(as.character(sigtab_PCR1_PCR3$Phylum), levels=names(x_PCR1_PCR3))
# Family order
x_PCR1_PCR3 = tapply(sigtab_PCR1_PCR3$log2FoldChange, sigtab_PCR1_PCR3$Family, function(x) max(x))
x_PCR1_PCR3 = sort(x_PCR1_PCR3, TRUE)
sigtab_PCR1_PCR3$Family = factor(as.character(sigtab_PCR1_PCR3$Family), levels=names(x_PCR1_PCR3))
ggplot(sigtab_PCR1_PCR3, aes(x=Family, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) + theme(axis.text=element_text(size=15), axis.title=element_text(size=15)) +
          theme(legend.text=element_text(size=15, color="black")) + 
          theme(legend.title = element_text(size=15, color="black"))+
  ggtitle("Differential Abundance of PCR3/PCR1")

#ggsave(filename="Microbiome_ED/Decontaminated analysis/Differential Abundance Results/deseq_PCR3_PCR1.png", height=6.5, width=7.5)
```

# Endosymbionts

Endosymbiont genera were filtered out from the phyloseq object 

```{r Endosymbiont Analysis}
#### Filtering for endosymbiont bacteria ####
tax_select(sample_ps, tax_list="Buchnera") #5 taxa.
tax_select(sample_ps, tax_list="Candidatus_Cardinium") #1 taxa
tax_select(sample_ps, tax_list="Candidatus") #315 taxa.
tax_select(sample_ps, tax_list="Hamiltonella") #1 taxa. 
tax_select(sample_ps, tax_list="Rickettsia") #731 taxa. 
tax_select(sample_ps, tax_list="Rickettsiella") #5 taxa. 
tax_select(sample_ps, tax_list="Wolbachia") #33 taxa. 
tax_select(sample_ps, tax_list="Spiroplasma") #30 taxa 
tax_select(sample_ps, tax_list="Arsenophonus") #1 taxa

endo<-subset_taxa(sample_ps, Genus=="Buchnera" | Genus=="Candidatus" | Genus=="Hamiltonella" | Genus=="Rickettsia" | Genus=="Rickettsiella" | Genus =="Wolbachia" | Genus=="Spiroplasma" | Genus=="Arsenophonus")

endo<-prune_taxa(taxa_sums(endo) > 0, endo) #143 taxa, 340 samples
endo

#### All samples together (insects and Spiderss) ####
endo_agglom <- speedyseq::tax_glom(endo, taxrank = "Genus")
endo_rel_abun<- transform_sample_counts(endo_agglom, function(x) x/sum(x))
endo_df <- psmelt(endo_rel_abun)
sample.df.agg.endo <- aggregate(Abundance ~ Phylum + Family + Genus + Site  + sample_Order, data = endo_df, FUN = mean)

#Phylum
ggplot(sample.df.agg.endo, aes(x=Site, y=Abundance, fill=Phylum)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Family
ggplot(sample.df.agg.endo, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  facet_grid(~sample_Order, scale="free_x")+
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=1)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Family", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Genus
ggplot(sample.df.agg.endo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=2)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Endosymbionts increased at Cushing Bridge, especially Buchnera, and was the only site with Rickettsiella present. The ACWA streams had a relatively low amount of endosymbionts but increased Wolbachia in PCR3 compared to PCR1 or BRR2. Spiroplasma was highest at Policeman Flats, Rickettsia did not change much but was a bit lower at BRR2.

#Separated by sample order and Site
 ggplot(sample.df.agg.endo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=2)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")+ facet_grid(~sample_Order, scale="free_x")
 #Big change in diptera at cushing bridge
 #Changes across most sites in ephemeroptera
 #Changes at cushing bridge and PCR1 in tricoptera

 #Separated by order but not site
 ggplot(sample.df.agg.endo, aes(x=sample_Order, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
   xlab("Sample Taxonomic Order")+
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=2)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")
 
 #Ephemeroptera and tricoptera had the highest overall mean relative abundance of endosymbionts. Tricoptera did not have Buchnera and had increased Rickettsia compared to other orders. Spiderss were the only organisms that had Arsenophonus but it comprised a very small portion of the total abundance. 
 
#### Spiderss only ####
Spidersendo<-subset_samples(endo, Taxa=="Spiders")
Spidersendo<-prune_taxa(taxa_sums(Spidersendo) > 0, Spidersendo)
Spidersendo #90 taxa, 78 samples

Spidersagglom<-speedyseq::tax_glom(Spidersendo, taxrank = "Genus")
Spidersendo_rel_abun<- transform_sample_counts(Spidersagglom, function(x) x/sum(x))
Spidersendo_df <- psmelt(Spidersendo_rel_abun)
sample.df.agg.Spidersendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = Spidersendo_df, FUN = mean)

ggplot(sample.df.agg.Spidersendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")
#Spiderss only had Buchnera at Sunalta and Graves Bridge as well as Arsenophonus at Graves Bridge. Other endosymbionts look relatively unchanged across sites. Slightly less Spiroplasma in the middle sites (Sunalta, Cush br, graves br)

#Tetragnathidae
tetraendo<-subset_samples(endo, Family=="Tetragnathidae")
tetraendo<-prune_taxa(taxa_sums(tetraendo) > 0, tetraendo)
tetraendo #79 taxa, 40 samples

tetraagglom<-speedyseq::tax_glom(tetraendo, taxrank = "Genus")
tetraendo_rel_abun<- transform_sample_counts(tetraagglom, function(x) x/sum(x))
tetraendo_df <- psmelt(tetraendo_rel_abun)
sample.df.agg.tetraendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = tetraendo_df, FUN = mean)

ggplot(sample.df.agg.tetraendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")
#Wolbachia is low at Cochrane and increases at Sunalta. Policeman Flats has increase in spiroplasma relative to other sites

#Araneidae
aranendo<-subset_samples(endo, Family=="Araneidae")
aranendo<-prune_taxa(taxa_sums(aranendo) > 0, aranendo)
aranendo #32 taxa, 38 samples

aranagglom<-speedyseq::tax_glom(aranendo, taxrank = "Genus")
aranendo_rel_abun<- transform_sample_counts(aranagglom, function(x) x/sum(x))
aranendo_df <- psmelt(aranendo_rel_abun)
sample.df.agg.aranendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = aranendo_df, FUN = mean)

ggplot(sample.df.agg.aranendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Sunalta has much lower Wolbachia and increase in Rickettsia. Graves Bridge has presence of Buchnera. Cochrane and Policeman Flats look very similar in endosymbiont composition. 

#### Macroinvertebrates ####
#Tricoptera
tricop<-subset_samples(endo, Order=="Tricoptera")
tricopendo<-prune_taxa(taxa_sums(tricop) > 0, tricop)
tricopendo #40 taxa, 92 samples

tricopagglom<-speedyseq::tax_glom(tricopendo, taxrank = "Genus")
tricopendo_rel_abun<- transform_sample_counts(tricopagglom, function(x) x/sum(x))
tricopendo_df <- psmelt(tricopendo_rel_abun)
sample.df.agg.tricopendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = tricopendo_df, FUN = mean)

ggplot(sample.df.agg.tricopendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")
#Rickettsia is consistent across all sites. Rickettsiella and spiroplasma only occurred at Cushing Bridge and Wolbachia only occurred at PCR1. 

#Diptera
dipt<-subset_samples(endo, Order=="Diptera")
diptendo<-prune_taxa(taxa_sums(dipt) > 0, dipt)
diptendo #27 taxa, 38 samples

diptagglom<-speedyseq::tax_glom(diptendo, taxrank = "Genus")
diptendo_rel_abun<- transform_sample_counts(diptagglom, function(x) x/sum(x))
diptendo_df <- psmelt(diptendo_rel_abun)
sample.df.agg.diptendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = diptendo_df, FUN = mean)

ggplot(sample.df.agg.diptendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Increase/presence of Rickettsiella and Buchnera at Cushing Bridge but no Wolbachia

#Ephemeroptera
ephem<-subset_samples(endo, Order=="Ephemeroptera")
ephemendo<-prune_taxa(taxa_sums(ephem) > 0, ephem)
ephemendo #21 taxa, 111 samples

ephemagglom<-speedyseq::tax_glom(ephemendo, taxrank = "Genus")
ephemendo_rel_abun<- transform_sample_counts(ephemagglom, function(x) x/sum(x))
ephemendo_df <- psmelt(ephemendo_rel_abun)
sample.df.agg.ephemendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = ephemendo_df, FUN = mean)

ggplot(sample.df.agg.ephemendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")

#Bucnera is only at Cushing Bridge. No Rickettsiella among any sites.

#Plecoptera
plecendo<-subset_samples(endo, Order=="Plecoptera")
plecendo<-prune_taxa(taxa_sums(plecendo) > 0, plecendo)
plecendo #143 taxa, 21 samples

plecagglom<-speedyseq::tax_glom(plecendo, taxrank = "Genus")
plecendo_rel_abun<- transform_sample_counts(plecagglom, function(x) x/sum(x))
plecendo_df <- psmelt(plecendo_rel_abun)
sample.df.agg.plecendo <- aggregate(Abundance ~ Phylum + Family+ Genus + Site  + Order, data = plecendo_df, FUN = mean)

ggplot(sample.df.agg.plecendo, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean relative abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_y_continuous(label = scales::percent)+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))+
  rremove("grid")
#No endosymbionts in plecoptera

```

# Wastewater associated bacteria

```{r Filtering for wastewater derived bacteria/pathogens}

#### Cyanotoxin producing Cyanobacteria ####
#Type of cyanobacteria that produce cyanotoxins 

cyano<-subset_taxa(sample_ps, Phylum=="Cyanobacteria")
cyano<-prune_taxa(taxa_sums(cyano) > 0, cyano)
sort(table(tax_table(cyano)[,5])) #Family
sort(table(tax_table(cyano)[,6])) #Genus

cyanofam<-speedyseq::tax_glom(cyano, taxrank = "Family")
cyano_rel_abun <- transform_sample_counts(cyanofam, function(x) x/sum(x))
#cyano_rel_abun<-microbiome::transform(cyanophylum, "compositional")
cyano_df <- psmelt(cyano_rel_abun)
sample.df.agg.cyano <- aggregate(Abundance ~ Site + Family, data = cyano_df, FUN = mean)

ggplot(sample.df.agg.cyano, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity", color="black") + 
  ylab("Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))



#Filtering for the cyanotoxin producing genera
tax_select(cyano, tax_list="Dolichospermum") #No taxa matched (previously Anabaena)
tax_select(cyano, tax_list="Anabaena") #6 taxa
tax_select(cyano, tax_list="Planktothrix NIVA-CYA 15") #3 taxa
tax_select(cyano, tax_list="Synechocytis") #3 taxa
tax_select(cyano, tax_list="Synechococcus") #No taxa matched

#Anabaena
ana<-tax_select(sample_ps, tax_list="Anabaena")
ana_rel_abun<-microbiome::transform(ana, "compositional")
ana_df <- psmelt(ana_rel_abun)
sample.df.agg.ana <- aggregate(Abundance ~ Site + Taxa, data = ana_df, FUN = mean)
ggplot(sample.df.agg.ana, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance") +
  facet_wrap(.~sample_Family, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))
 

#Leptolyngbya
lepto<-tax_select(sample_ps, tax_list="Leptolyngbya")
lepto_rel_abun<-microbiome::transform(lepto, "compositional")
lepto_df <- psmelt(lepto_rel_abun)
sample.df.agg.lepto <- aggregate(Abundance ~ Site  + Taxa, data = lepto_df, FUN = mean) 

ggplot(sample.df.agg.lepto, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >5%") +
  facet_wrap(.~Taxa, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))

#Increases downstream in the bow river (highest at Policeman Flats, lowest at Cochrane) and highest at PCR1 but not PCR3

#Planktothrix NIVA-CYA 15
plank<-subset_taxa(sample_ps, Genus=="Planktothrix NIVA-CYA 15")
plank_rel_abun<-microbiome::transform(plank, "compositional")
plank_df <- psmelt(plank_rel_abun)

sample.df.agg.plank <- aggregate(Abundance ~ Site  + Taxa, data = plank_df, FUN = mean) 

##Relative Abundance Plot - Phylum Level
rel_abun_plot_phylum_plank <- ggplot(sample.df.agg.plank, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance >5%") +
  facet_wrap(.~Taxa, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Phylum", values = c(brewer.pal(12, "Paired"), "grey35", "orange", "purple", "violetred2", "blue", "brown", "red"))
rel_abun_plot_phylum_plank

#Not found in Spiders or at Sunalta. Increased at Graves Bridge but low/not present at other sites. Much higher in the ACWA streams. Could be due to high effluent exposure?


#Microcystis 
microcy<-subset_taxa(sample_ps, Genus=="Microcystis PCC-7914")
microcy_df<-psmelt(microcy)
microcyagg<-aggregate(Abundance ~ Site  + Taxa, data = microcy_df, FUN = mean) 
ggplot(microcyagg, aes(x=Site, y=Abundance))+
  geom_point()
#Only found in PCR3 (maybe from high effluent exposure)

#Aphanizomenon
tax_select(sample_ps, tax_list="Aphanizomenon MDT14a") #No taxa matched

#Schizo
tax_select(sample_ps, tax_list="Schizothrix LEGE 07164") #No taxa matched

tax_select(sample_ps, tax_list="Dolichospermum")

#### Filtering for wastewater effluent-associated bacteria and potential pathogens ####
#Wastewater effluent bacteria/pathogens based on several papers:
#(Cyprowski et al., 2018; Kristensen et al., 2020; Leight et al., 2018; Millar et al., 2022)

effluent_bac<-subset_taxa(sample_ps, Genus=="Bifidobacterium" | Genus=="Clostridium" | Genus=="Propionibacterium" | Genus=="Peptostreptococcus" | Genus=="Actinomyces" | Genus=="Arcobacter" | Genus=="Ruminococcus" | Genus=="Escherichia coli" | Genus=="Campylobacter" | Genus=="Salmonella" | Genus=="Shigella" | Genus=="Trichococcus" | Genus=="Streptococcus" | Genus=="Blautia" | Genus=="Enterococcus" | Genus=="Vibrio vulnificus" | Genus=="Aeromonas" | Genus=="Legionella" | Genus=="Fusibacter" | Genus=="Bacillus" | Genus=="Romboutsia" | Genus=="Cloacibacterium" | Genus=="Pseudomonas" | Genus=="Acetoanaerobium" | Genus=="Acidaminococcus" | Genus=="Aquimonas" | Genus=="AUTHM297" | Genus=="Bact-08" | Genus == "BD1-7 clade" | Genus=="C1-B045" | Genus=="Candidatus Cloacimonas" | Genus=="Candidatus Paenicardinium" | Genus=="Candidatus Protochlamydia" | Genus=="Chiayiivirga" | Genus=="Desulfobacter" | Genus=="Flavitalea" | Genus=="Lelliottia" | Genus=="Leptotrichia" | Genus=="Mesotoga" | Genus=="Neochlamydia" | Genus=="Ottowia" | Genus=="Planctopirus" | Genus=="Prevotellaceae UCG-004" | Genus=="Proteiniclasticum" | Genus=="SC103" | Genus=="Steroidobacter" | Genus=="Succinivibrio" | Genus=="SWB02" | Genus=="Thermovirga" | Genus=="Turneriella" | Genus=="U29-B03" | Genus=="XBB1006" | Genus=="Shewanella" | Genus=="Sphingobacterium" | Genus=="Vibrio")
effluent_bac<-prune_samples(sample_sums(effluent_bac) > 0, effluent_bac) 
effluent_bac #54 taxa 196 samples

sort(table(tax_table(effluent_bac)[,6])) #Genera
sort(table(tax_table(effluent_bac)[,7])) #Species

#### Relative abundance plots with all effluent bacterial taxa combined ####

#Family
#All taxonomic orders combined
effluent_bac_fam<-tax_glom(effluent_bac, taxrank="Family")
effluent_bac_rel_abun<-transform_sample_counts(effluent_bac_fam, function(x) x/sum(x)) #Have to use relative abundance for this so that the number of samples taken doesn't influence the interpretation
#effluent_bac_rel_abun<-microbiome::transform(effluent_bac, "compositional")
effluentbac_df <- psmelt(effluent_bac_rel_abun)
sample.df.agg.effluentbac.fam <- aggregate(Abundance ~ Site  + Family, data = effluentbac_df, FUN = mean)

ggplot(sample.df.agg.effluentbac.fam, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity") +
  ylab("Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=8, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=2)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(size=8, color="black"),
                        axis.title=element_text(size=10))+ rremove("grid")


#Family level, host orders separated
sample.df.agg.effluentbac.order <- aggregate(Abundance ~ Site  + Family + sample_Order, data = effluentbac_df, FUN = mean)

ggplot(sample.df.agg.effluentbac.order, aes(x=Site, y=Abundance, fill=Family)) + geom_bar(stat="identity") + 
  ylab("Relative Abundance") +
  facet_wrap(.~sample_Order, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=2)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+ rremove("grid")


#Genus
#All orders combined
effluent_bac_genus<-tax_glom(effluent_bac, taxrank="Genus")
effluent_bac_rel_abun_genus<-transform_sample_counts(effluent_bac_genus, function(x) x/sum(x))
effluentbac_df_genus <- psmelt(effluent_bac_rel_abun_genus)
sample.df.agg.effluentbac.genus <- aggregate(Abundance ~ Site  + Genus, data = effluentbac_df_genus, FUN = mean)

ggplot(sample.df.agg.effluentbac.genus, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+ rremove("grid")

#High amount of Romboutsia in the ACWA streams

#Orders separated
sample.df.agg.effluentbac.genus.order <- aggregate(Abundance ~ Site  + Genus + sample_Order, data = effluentbac_df_genus, FUN = mean)
ggplot(sample.df.agg.effluentbac.genus.order, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  facet_wrap(.~sample_Order, scale="free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+ rremove("grid")

#### Individual effluent bacteria ####

#The following graphs will be the mean total abundance (instead of sum) of each individual effluent-bacteria/pathogen to control for the sample size.

#Trichococcus
#Total abundance
tricho<-subset_taxa(sample_ps, Genus=="Trichococcus")
tricho_df <- psmelt(tricho)
tricho_agg <- aggregate(Abundance ~ Site, data = tricho_df, FUN = mean)

ggplot(tricho_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ggtitle("Trichococcus")

#Highest mean total abundance at PCR3 then Cushing Bridge. None at BRR2 or PCR1


#Aeromonas
aero<-subset_taxa(sample_ps, Genus=="Aeromonas")
sort(table(tax_table(aero)[,7])) #Sobria and veronii are potential human pathogens causing nausea, vomiting, diahrea 
aero_df <- psmelt(aero)
aero_agg <- aggregate(Abundance ~ Site, data = aero_df, FUN = mean)

ggplot(aero_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ggtitle("Aeromonas")
#High in Policeman Flats and Cushing Bridge. Low everywhere else. Not found in PCR3 effluent stream... possibly not due to wastewater effluent exposure.

#Ruminococcus
rumino<-subset_taxa(sample_ps, Genus=="Ruminococcus")
rumino_df <- psmelt(rumino)
rumino_agg <- aggregate(Abundance ~ Site, data = rumino_df, FUN = mean)

ggplot(rumino_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Ruminococcus")

#Ruminococcus decreased slightly with wastewater exposure (why would it be highest at Cochrane? Maybe agricultural runoff/ cabin septic tanks?)


#Enterococcus
entero<-subset_taxa(sample_ps, Genus=="Enterococcus")
entero_df <- psmelt(entero)
entero_agg <- aggregate(Abundance ~ Site, data = entero_df, FUN = mean)

ggplot(entero_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Enterococcus")

#Highest mean total abundance at Cochrane

#Blautia
blaut<-subset_taxa(sample_ps, Genus=="Blautia")
blaut_df <- psmelt(blaut)
blaut_agg <- aggregate(Abundance ~ Site, data = blaut_df, FUN = mean)

ggplot(blaut_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Blautia")

#Highest mean total abundance at Cochrane 

#Streptococcus
strep<-subset_taxa(sample_ps, Genus=="Streptococcus")
strep_df <- psmelt(strep)
strep_agg <- aggregate(Abundance ~ Site , data = strep_df, FUN = mean)

ggplot(strep_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid") + ggtitle("Streptococcus")
#Much lower at Graves Bridge and ACWA streams


#Streptococcus gallolyticus
strep_gallo<-subset_taxa(sample_ps, Species=="gallolyticus")
sgallo <- psmelt(strep_gallo)
agg_gallo <- aggregate(Abundance ~ Site, data = sgallo, FUN = mean)

ggplot(agg_gallo, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid") + ggtitle("Streptococcus gallolyticus")

#Highest at PCR1 in ACWA streams and Cochrane at Bow River sites

#Cloacibacterium
cloac<-subset_taxa(sample_ps, Genus=="Cloacibacterium")
cloac_df <- psmelt(cloac)
cloac_agg <- aggregate(Abundance ~ Site, data = cloac_df, FUN = mean)

ggplot(cloac_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Cloacibacterium")

#Highest at Policeman Flats

#Romboutsia
romb<-subset_taxa(sample_ps, Genus=="Romboutsia")
romb_df <- psmelt(romb)
romb_agg <- aggregate(Abundance ~ Site, data = romb_df, FUN = mean)

ggplot(romb_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Romboutsia")
#Higher at PCR1 and PCR3 in ACWA. A bit higher at Sunalta and Graves Bridge (doesn't seem to be wastewater exposure driving that in the Bow)

#Bacillus
bacillus<-subset_taxa(sample_ps, Genus=="Bacillus")
bacillus_df <- psmelt(bacillus)
bacillus_agg <- aggregate(Abundance ~ Site, data = bacillus_df, FUN = mean)

ggplot(bacillus_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Total Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Bacillus")

#Highest at Graves Bridge

#Pseudomonas
pseudo<-subset_taxa(sample_ps, Genus=="Pseudomonas")
pseudo_df <- psmelt(pseudo)
pseudo_agg <- aggregate(Abundance ~ Site, data = pseudo_df, FUN = mean)

ggplot(pseudo_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Pseudomonas")

#Highest at Cushing Bridge then Policeman Flats

#Yersinia pestis (plague causing bacteria in mammals)
pestis<-subset_taxa(sample_ps, Species=="pestis")
pestis<-prune_taxa(taxa_sums(pestis) > 0, pestis)
sample_sums(pestis)

pestis_df <- psmelt(pestis)
pestis_agg <- aggregate(Abundance ~ Site, data = pestis_df, FUN = mean)

ggplot(pestis_agg, aes(x=Site, y=Abundance)) + geom_bar(stat="identity") + 
  ylab("Mean Relative Abundance") +
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.position="bottom", legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=4)) +
  theme(axis.title.y=element_text(size=10),
        axis.text.y=element_text(size=6),
        axis.text.x=element_text(angle=45, hjust=1, size=6, color="black"),
                        axis.title=element_text(size=10))+
  rremove("grid")+ ggtitle("Yersinia pestis")
#Found at Cochrane mostly and a bit at Policeman Flats. Only found in stonefly. None found at Cushing Bridge.


#Top 5 effluent derived bacteria
top_eff_bac <- sample.df.agg.effluentbac.genus %>%
    group_by(Site, Genus) %>%
    summarise(Mean = mean(Abundance)) %>%
    arrange(-Mean)
top_eff_bac
top5effbac <- top_eff_bac$Genus[1:10]
top5effbac<- sample.df.agg.effluentbac.genus %>%
    mutate(Genus = fct_other(Genus, top5effbac))


top5effbac$Genus <- factor(top5effbac$Genus, levels=c("Bacillus", "Enterococcus", "Aeromonas", "Streptococcus", "Other", "Romboutsia")) #labels=c("Bacteroidota", "Cyanobacteria", "Firmicutes", "Other", "Proteobacteria"))

top5effbacplot <- ggplot(top5effbac, aes(x=Site, y=Abundance, fill=Genus)) + geom_bar(stat="identity") +
  ylab("Mean Relative Abundance") +
  #facet_grid(.~Taxa, scale="free_x", space = "free_x") + 
  theme(strip.text = element_text(size = 16, color="black")) + 
  theme(legend.key.size = unit(0.5, "cm"),
        legend.text=element_text(size=10, face="italic"), legend.title=element_text(size=10)) +
  guides (fill=guide_legend(nrow=6)) +
  theme(axis.title.y=element_text(size=16),
        axis.text.y=element_text(size=16),
        axis.text.x=element_text(angle=45, hjust=1, size=16, color="black"),
                        axis.title=element_text(size=18))+
  scale_fill_manual(name="Bacterial Genus", values = c(brewer.pal(4, "Paired"), "gray35", "lightcoral"))+
  scale_y_continuous(label = scales::percent) + rremove("grid")

top5effbacplot

#Graves Bridge looks like it's the Bow River site with the most amount of change in the common effluent-associated bacteria (increase in Bacillus). However, the Bow River sites do not have much of the same bacteria as the ACWA streams (less Romboutsia, Aeromonas, and Enterococcus but more Bacillus). 
```



